--- output/outieee.c.orig	2009-02-27 00:04:02.000000000 +0100
+++ output/outieee.c	2009-03-01 01:04:30.000000000 +0100
@@ -1127,7 +1127,7 @@
         if ((buffer[i] & 0xff) > 31)
             checksum += buffer[i];
     va_end(ap);
-    fprintf(ofp, buffer);
+    fprintf(ofp, "%s", buffer);
 }
 
 /*
--- ndisasm.c.orig	2009-02-27 00:04:02.000000000 +0100
+++ ndisasm.c	2009-03-01 01:12:08.000000000 +0100
@@ -85,7 +85,7 @@
                     p++;
                     break;
                 case 'h':
-                    fprintf(stderr, help);
+                    fprintf(stderr, "%s", help);
                     return 0;
                 case 'r':
                 case 'v':
--- rdoff/rdflib.c.orig	2009-02-27 00:04:02.000000000 +0100
+++ rdoff/rdflib.c	2009-03-01 01:16:19.000000000 +0100
@@ -133,7 +133,7 @@
 
     if (argc < 3 || !strncmp(argv[1], "-h", 2)
         || !strncmp(argv[1], "--h", 3)) {
-        printf(usage);
+        printf("%s", usage);
         exit(1);
     }
 
