diff -Nur glob2-0.9.2-old/SConstruct glob2-0.9.2/SConstruct
--- glob2-0.9.2-old/SConstruct	2008-02-24 16:24:55.699765720 +0200
+++ glob2-0.9.2/SConstruct	2008-02-24 16:25:19.246186119 +0200
@@ -7,7 +7,7 @@
 isLinuxPlatform = sys.platform=='linux2'
 isDarwinPlatform = sys.platform=='darwin'
 
-
+SConsignFile()
 def establish_options(env):
     opts = Options('options_cache.py')
     opts.Add("CXXFLAGS", "Manually add to the CXXFLAGS", "-g")
