--- setup.cfg	2006-11-15 19:26:26.000000000 +0200
+++ setup-new.cfg	2007-04-12 14:39:45.000000000 +0300
@@ -7,8 +7,8 @@
 # for wrapping OpenLDAP 2 libs
 [_ldap]
 
-library_dirs = /usr/local/openldap-2.3/lib
-include_dirs = /usr/local/openldap-2.3/include /usr/include/sasl
+library_dirs = /usr/lib
+include_dirs = /usr/include /usr/include/sasl
 
 extra_compile_args = 
 extra_objects = 
@@ -27,8 +27,8 @@
 
 # Installation options
 [install]
-compile = 1
-optimize = 1
+compile = 0
+optimize = 0
 
 # Linux distributors/packagers should adjust these settings
 [bdist_rpm]
