diff -Naur src.orig/plasma-applet-kprayertime4.desktop src/plasma-applet-kprayertime4.desktop
--- src.orig/plasma-applet-kprayertime4.desktop	2009-12-26 18:06:16.184824591 +0200
+++ src/plasma-applet-kprayertime4.desktop	2009-12-26 18:06:47.493947450 +0200
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Name=KPrayertime Widget
 Comment=An Islamic prayer times widget
+Comment[tr]=Namaz vakitleri programcığı
 Type=Service
 X-KDE-ServiceTypes=Plasma/Applet
 Icon=kprayertime-icon
