Index: kdepimlibs-4.4.5/akonadi/agentmanager_p.h
===================================================================
--- kdepimlibs-4.4.5.orig/akonadi/agentmanager_p.h
+++ kdepimlibs-4.4.5/akonadi/agentmanager_p.h
@@ -76,7 +76,7 @@ class AgentManagerPrivate
     void synchronize( const AgentInstance& );
     void synchronizeCollectionTree( const AgentInstance& );
 
-    void serviceOwnderChanged( const QString &name, const QString &oldOwner, const QString &newOwner );
+    void serviceOwnerChanged( const QString &name, const QString &oldOwner, const QString &newOwner );
 
     AgentType fillAgentType( const QString &identifier ) const;
     AgentInstance fillAgentInstance( const QString &identifier ) const;
Index: kdepimlibs-4.4.5/akonadi/agentmanager.cpp
===================================================================
--- kdepimlibs-4.4.5.orig/akonadi/agentmanager.cpp
+++ kdepimlibs-4.4.5/akonadi/agentmanager.cpp
@@ -326,7 +326,7 @@ void AgentManagerPrivate::createDBusInte
   }
 }
 
-void AgentManagerPrivate::serviceOwnderChanged( const QString &service, const QString &oldOwner, const QString &newOwner )
+void AgentManagerPrivate::serviceOwnerChanged( const QString &service, const QString &oldOwner, const QString &newOwner )
 {
   Q_UNUSED( newOwner );
   if ( service == QLatin1String( AKONADI_DBUS_CONTROL_SERVICE ) && oldOwner.isEmpty() )
