--- config/commence.in	2008-02-07 03:31:06.000000000 +0200
+++ config/commence.in	2008-02-07 03:31:51.000000000 +0200
@@ -79,7 +79,7 @@
 ## defined. The main makefile is free to redefine these to something else.
 DOCDIR=$(docdir)
 SUBDOCDIR=
-EXAMPLEDIR=$(docdir)/hdf5/examples/c
+EXAMPLEDIR=$(docdir)/examples/c
 LIB=
 LIB_SRC=
 LIB_OBJ=
--- c++/config/commence.in	2008-02-07 03:31:06.000000000 +0200
+++ c++/config/commence.in	2008-02-07 03:32:28.000000000 +0200
@@ -80,7 +80,7 @@
 ## complains if it sees a reference to a variable which has never been
 ## defined. The main makefile is free to redefine these to something else.
 DOCDIR=$(docdir)
-EXAMPLEDIR=$(docdir)/hdf5/examples/c++
+EXAMPLEDIR=$(docdir)/examples/c++
 LIB=
 LIB_SRC=
 LIB_OBJ=
--- fortran/config/commence.in	2008-02-07 03:31:06.000000000 +0200
+++ fortran/config/commence.in	2008-02-07 03:32:50.000000000 +0200
@@ -88,7 +88,7 @@
 ## complains if it sees a reference to a variable which has never been
 ## defined. The main makefile is free to redefine these to something else.
 DOCDIR=$(docdir)
-EXAMPLEDIR=$(docdir)/hdf5/examples/fortran
+EXAMPLEDIR=$(docdir)/examples/fortran
 LIB=
 LIB_SRC=
 LIB_OBJ=
