--- Fig.ad.orig	2006-04-05 15:50:51.000000000 +0300
+++ Fig.ad	2006-04-05 15:51:49.000000000 +0300
@@ -18,13 +18,11 @@
 
 ! Browser - put your favorite browser here.  
 ! 		This is for viewing the xfig html reference.
-! For netscape, this command will open the help pages in a running netscape,
-!     or start a new netscape if one isn't already running
-Fig.browser:	netscape -remote 'openFile(%f)' || netscape %f
+Fig.browser:	firefox %f
 
 ! pdfviewer - put your favorite pdf viewer here.  
 !		This is for viewing the xfig how-to guide and man pages
-Fig.pdfviewer:			acroread %f
+Fig.pdfviewer:			kpdf %f
 
 ! Spell check program - put your favorite spelling check program here.
 !		It must write the misspelled words to standard output.
