Index: kpdf/shell/shell.cpp
===================================================================
--- kpdf/shell/shell.cpp	(revision 582561)
+++ kpdf/shell/shell.cpp	(working copy)
@@ -116,8 +116,13 @@
     if ( m_part )
     {
         bool openOk = m_part->openURL( url );
-        if ( openOk ) m_recent->addURL( url );
-        else m_recent->removeURL( url );
+        if ( openOk )
+        {
+          m_recent->addURL( url );
+          setCaption(url.fileName());
+        }
+        else
+          m_recent->removeURL( url );
     }
 }
 
