diff -Naur truecrypt-6.1a-source.orig/Common/Crypto.h truecrypt-6.1a-source/Common/Crypto.h
--- truecrypt-6.1a-source.orig/Common/Crypto.h	2009-05-04 08:27:06.000000000 +0300
+++ truecrypt-6.1a-source/Common/Crypto.h	2009-05-04 08:28:30.000000000 +0300
@@ -29,6 +29,10 @@
 
 #include "Tcdefs.h"
 
+#ifndef BOOL
+#define BOOL BOOL1
+#endif
+
 #ifdef __cplusplus
 extern "C" {
 #endif
diff -Naur truecrypt-6.1a-source.orig/Makefile truecrypt-6.1a-source/Makefile
--- truecrypt-6.1a-source.orig/Makefile	2009-05-04 08:27:06.000000000 +0300
+++ truecrypt-6.1a-source/Makefile	2009-05-04 08:30:14.000000000 +0300
@@ -50,6 +50,7 @@
 endif
 
 C_CXX_FLAGS += -DBOOL=int -DFALSE=0 -DTRUE=1
+C_CXX_FLAGS += -DBOOL1=int
 C_CXX_FLAGS += -D__int8=char -D__int16=short -D__int32=int '-D__int64=long long'  # Tested in PlatformTest
 
 export LFLAGS :=
