diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kdegraphics-4.1.80-orig/libs/libkdcraw/CMakeLists.txt kdegraphics-4.1.80/libs/libkdcraw/CMakeLists.txt
--- kdegraphics-4.1.80-orig/libs/libkdcraw/CMakeLists.txt	2008-11-28 20:56:13.000000000 +0200
+++ kdegraphics-4.1.80/libs/libkdcraw/CMakeLists.txt	2008-11-28 20:56:31.000000000 +0200
@@ -33,7 +33,7 @@
 
 # Library ABI version used by linker.
 # For details : http://www.gnu.org/software/libtool/manual.html#Updating-version-info
-SET(DCRAW_LIB_SO_CUR_VERSION "7")
+SET(DCRAW_LIB_SO_CUR_VERSION "6")
 SET(DCRAW_LIB_SO_REV_VERSION "0")
 SET(DCRAW_LIB_SO_AGE_VERSION "0")
 
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kdegraphics-4.1.80-orig/libs/libkipi/CMakeLists.txt kdegraphics-4.1.80/libs/libkipi/CMakeLists.txt
--- kdegraphics-4.1.80-orig/libs/libkipi/CMakeLists.txt	2008-11-28 21:07:32.000000000 +0200
+++ kdegraphics-4.1.80/libs/libkipi/CMakeLists.txt	2008-11-28 21:07:49.000000000 +0200
@@ -32,7 +32,7 @@
 
 # Library ABI version used by linker.
 # For details : http://www.gnu.org/software/libtool/manual.html#Updating-version-info
-SET(KIPI_LIB_SO_CUR_VERSION "6")
+SET(KIPI_LIB_SO_CUR_VERSION "5")
 SET(KIPI_LIB_SO_REV_VERSION "0")
 SET(KIPI_LIB_SO_AGE_VERSION "0")
 
