Index: linux-2.6.18/include/linux/config.h
===================================================================
--- linux-2.6.18.orig/include/linux/config.h
+++ linux-2.6.18/include/linux/config.h
@@ -3,6 +3,7 @@
 /* This file is no longer in use and kept only for backward compatibility.
  * autoconf.h is now included via -imacros on the commandline
  */
+#warning Including config.h is deprecated.
 #include <linux/autoconf.h>
 
 #endif
Index: linux-2.6.18/include/asm-i386/tsc.h
===================================================================
--- linux-2.6.18.orig/include/asm-i386/tsc.h
+++ linux-2.6.18/include/asm-i386/tsc.h
@@ -6,7 +6,6 @@
 #ifndef _ASM_i386_TSC_H
 #define _ASM_i386_TSC_H
 
-#include <linux/config.h>
 #include <asm/processor.h>
 
 /*
Index: linux-2.6.18/include/asm-i386/dwarf2.h
===================================================================
--- linux-2.6.18.orig/include/asm-i386/dwarf2.h
+++ linux-2.6.18/include/asm-i386/dwarf2.h
@@ -1,8 +1,6 @@
 #ifndef _DWARF2_H
 #define _DWARF2_H
 
-#include <linux/config.h>
-
 #ifndef __ASSEMBLY__
 #warning "asm/dwarf2.h should be only included in pure assembly files"
 #endif
Index: linux-2.6.18/net/ipv4/netfilter/ip_conntrack_sip.c
===================================================================
--- linux-2.6.18.orig/net/ipv4/netfilter/ip_conntrack_sip.c
+++ linux-2.6.18/net/ipv4/netfilter/ip_conntrack_sip.c
@@ -8,7 +8,6 @@
  * published by the Free Software Foundation.
  */
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/ctype.h>
 #include <linux/skbuff.h>
Index: linux-2.6.18/net/ipv4/af_inet.c
===================================================================
--- linux-2.6.18.orig/net/ipv4/af_inet.c
+++ linux-2.6.18/net/ipv4/af_inet.c
@@ -67,7 +67,6 @@
  *		2 of the License, or (at your option) any later version.
  */
 
-#include <linux/config.h>
 #include <linux/err.h>
 #include <linux/errno.h>
 #include <linux/types.h>
Index: linux-2.6.18/net/ipv4/ipconfig.c
===================================================================
--- linux-2.6.18.orig/net/ipv4/ipconfig.c
+++ linux-2.6.18/net/ipv4/ipconfig.c
@@ -31,7 +31,6 @@
  *              --  Josef Siemes <jsiemes@web.de>, Aug 2002
  */
 
-#include <linux/config.h>
 #include <linux/types.h>
 #include <linux/string.h>
 #include <linux/kernel.h>
Index: linux-2.6.18/net/ipv4/raw.c
===================================================================
--- linux-2.6.18.orig/net/ipv4/raw.c
+++ linux-2.6.18/net/ipv4/raw.c
@@ -38,8 +38,7 @@
  *		as published by the Free Software Foundation; either version
  *		2 of the License, or (at your option) any later version.
  */
- 
-#include <linux/config.h> 
+
 #include <linux/types.h>
 #include <asm/atomic.h>
 #include <asm/byteorder.h>
Index: linux-2.6.18/net/ipv4/tcp_veno.c
===================================================================
--- linux-2.6.18.orig/net/ipv4/tcp_veno.c
+++ linux-2.6.18/net/ipv4/tcp_veno.c
@@ -9,7 +9,6 @@
  * 	See http://www.ntu.edu.sg/home5/ZHOU0022/papers/CPFu03a.pdf
  */
 
-#include <linux/config.h>
 #include <linux/mm.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
Index: linux-2.6.18/net/ipv4/tcp_lp.c
===================================================================
--- linux-2.6.18.orig/net/ipv4/tcp_lp.c
+++ linux-2.6.18/net/ipv4/tcp_lp.c
@@ -35,7 +35,6 @@
  * Version: $Id: tcp_lp.c,v 1.24 2006/09/05 20:22:53 hswong3i Exp $
  */
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <net/tcp.h>
 
Index: linux-2.6.18/net/core/wireless.c
===================================================================
--- linux-2.6.18.orig/net/core/wireless.c
+++ linux-2.6.18/net/core/wireless.c
@@ -72,7 +72,6 @@
 
 /***************************** INCLUDES *****************************/
 
-#include <linux/config.h>		/* Not needed ??? */
 #include <linux/module.h>
 #include <linux/types.h>		/* off_t */
 #include <linux/netdevice.h>		/* struct ifreq, dev_get_by_name() */
Index: linux-2.6.18/include/linux/vmstat.h
===================================================================
--- linux-2.6.18.orig/include/linux/vmstat.h
+++ linux-2.6.18/include/linux/vmstat.h
@@ -3,7 +3,6 @@
 
 #include <linux/types.h>
 #include <linux/percpu.h>
-#include <linux/config.h>
 #include <linux/mmzone.h>
 #include <asm/atomic.h>
 
Index: linux-2.6.18/include/linux/unwind.h
===================================================================
--- linux-2.6.18.orig/include/linux/unwind.h
+++ linux-2.6.18/include/linux/unwind.h
@@ -12,8 +12,6 @@
  * is not much point in implementing the full Dwarf2 unwind API.
  */
 
-#include <linux/config.h>
-
 struct module;
 
 #ifdef CONFIG_STACK_UNWIND
Index: linux-2.6.18/include/net/netdma.h
===================================================================
--- linux-2.6.18.orig/include/net/netdma.h
+++ linux-2.6.18/include/net/netdma.h
@@ -20,7 +20,6 @@
  */
 #ifndef NETDMA_H
 #define NETDMA_H
-#include <linux/config.h>
 #ifdef CONFIG_NET_DMA
 #include <linux/dmaengine.h>
 #include <linux/skbuff.h>
Index: linux-2.6.18/drivers/media/video/pvrusb2/pvrusb2-hdw-internal.h
===================================================================
--- linux-2.6.18.orig/drivers/media/video/pvrusb2/pvrusb2-hdw-internal.h
+++ linux-2.6.18/drivers/media/video/pvrusb2/pvrusb2-hdw-internal.h
@@ -33,7 +33,6 @@
 
 */
 
-#include <linux/config.h>
 #include <linux/videodev2.h>
 #include <linux/i2c.h>
 #include <linux/mutex.h>
Index: linux-2.6.18/drivers/scsi/scsi.h
===================================================================
--- linux-2.6.18.orig/drivers/scsi/scsi.h
+++ linux-2.6.18/drivers/scsi/scsi.h
@@ -20,8 +20,6 @@
 #ifndef _SCSI_H
 #define _SCSI_H
 
-#include <linux/config.h>	    /* for CONFIG_SCSI_LOGGING */
-
 #include <scsi/scsi_cmnd.h>
 #include <scsi/scsi_device.h>
 #include <scsi/scsi_eh.h>
Index: linux-2.6.18/drivers/net/irda/mcs7780.c
===================================================================
--- linux-2.6.18.orig/drivers/net/irda/mcs7780.c
+++ linux-2.6.18/drivers/net/irda/mcs7780.c
@@ -45,7 +45,6 @@
 
 #include <linux/module.h>
 #include <linux/moduleparam.h>
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/types.h>
 #include <linux/errno.h>
Index: linux-2.6.18/drivers/net/irda/w83977af_ir.c
===================================================================
--- linux-2.6.18.orig/drivers/net/irda/w83977af_ir.c
+++ linux-2.6.18/drivers/net/irda/w83977af_ir.c
@@ -40,7 +40,6 @@
  ********************************************************************/
 
 #include <linux/module.h>
-#include <linux/config.h> 
 #include <linux/kernel.h>
 #include <linux/types.h>
 #include <linux/skbuff.h>
Index: linux-2.6.18/drivers/net/smc911x.c
===================================================================
--- linux-2.6.18.orig/drivers/net/smc911x.c
+++ linux-2.6.18/drivers/net/smc911x.c
@@ -55,8 +55,6 @@ static const char version[] =
 			 )
 #endif
 
-
-#include <linux/config.h>
 #include <linux/init.h>
 #include <linux/module.h>
 #include <linux/kernel.h>
Index: linux-2.6.18/drivers/net/netx-eth.c
===================================================================
--- linux-2.6.18.orig/drivers/net/netx-eth.c
+++ linux-2.6.18/drivers/net/netx-eth.c
@@ -17,7 +17,6 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-#include <linux/config.h>
 #include <linux/init.h>
 #include <linux/module.h>
 #include <linux/kernel.h>
Index: linux-2.6.18/drivers/net/wan/cycx_main.c
===================================================================
--- linux-2.6.18.orig/drivers/net/wan/cycx_main.c
+++ linux-2.6.18/drivers/net/wan/cycx_main.c
@@ -40,7 +40,6 @@
 * 1998/08/08	acme		Initial version.
 */
 
-#include <linux/config.h>	/* OS configuration options */
 #include <linux/stddef.h>	/* offsetof(), etc. */
 #include <linux/errno.h>	/* return codes */
 #include <linux/string.h>	/* inline memset(), etc. */
Index: linux-2.6.18/drivers/net/wan/sdla.c
===================================================================
--- linux-2.6.18.orig/drivers/net/wan/sdla.c
+++ linux-2.6.18/drivers/net/wan/sdla.c
@@ -32,7 +32,6 @@
  *		2 of the License, or (at your option) any later version.
  */
 
-#include <linux/config.h> /* for CONFIG_DLCI_MAX */
 #include <linux/module.h>
 #include <linux/kernel.h>
 #include <linux/types.h>
Index: linux-2.6.18/drivers/net/wan/dlci.c
===================================================================
--- linux-2.6.18.orig/drivers/net/wan/dlci.c
+++ linux-2.6.18/drivers/net/wan/dlci.c
@@ -28,7 +28,6 @@
  *		2 of the License, or (at your option) any later version.
  */
 
-#include <linux/config.h> /* for CONFIG_DLCI_COUNT */
 #include <linux/module.h>
 #include <linux/kernel.h>
 #include <linux/types.h>
Index: linux-2.6.18/drivers/net/phy/vitesse.c
===================================================================
--- linux-2.6.18.orig/drivers/net/phy/vitesse.c
+++ linux-2.6.18/drivers/net/phy/vitesse.c
@@ -12,7 +12,6 @@
  *
  */
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
 #include <linux/mii.h>
Index: linux-2.6.18/drivers/net/phy/smsc.c
===================================================================
--- linux-2.6.18.orig/drivers/net/phy/smsc.c
+++ linux-2.6.18/drivers/net/phy/smsc.c
@@ -14,7 +14,6 @@
  *
  */
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
 #include <linux/mii.h>
Index: linux-2.6.18/drivers/net/hp100.c
===================================================================
--- linux-2.6.18.orig/drivers/net/hp100.c
+++ linux-2.6.18/drivers/net/hp100.c
@@ -111,7 +111,6 @@
 #include <linux/etherdevice.h>
 #include <linux/skbuff.h>
 #include <linux/types.h>
-#include <linux/config.h>	/* for CONFIG_PCI */
 #include <linux/delay.h>
 #include <linux/init.h>
 #include <linux/bitops.h>
Index: linux-2.6.18/drivers/net/3c501.c
===================================================================
--- linux-2.6.18.orig/drivers/net/3c501.c
+++ linux-2.6.18/drivers/net/3c501.c
@@ -120,7 +120,6 @@ static const char version[] =
 #include <linux/slab.h>
 #include <linux/string.h>
 #include <linux/errno.h>
-#include <linux/config.h>	/* for CONFIG_IP_MULTICAST */
 #include <linux/spinlock.h>
 #include <linux/ethtool.h>
 #include <linux/delay.h>
Index: linux-2.6.18/drivers/scsi/aic7xxx_old.c
===================================================================
--- linux-2.6.18.orig/drivers/scsi/aic7xxx_old.c
+++ linux-2.6.18/drivers/scsi/aic7xxx_old.c
@@ -249,8 +249,6 @@
 #include <linux/stat.h>
 #include <linux/slab.h>        /* for kmalloc() */
 
-#include <linux/config.h>        /* for CONFIG_PCI */
-
 #define AIC7XXX_C_VERSION  "5.2.6"
 
 #define ALL_TARGETS -1
Index: linux-2.6.18/drivers/scsi/hptiop.c
===================================================================
--- linux-2.6.18.orig/drivers/scsi/hptiop.c
+++ linux-2.6.18/drivers/scsi/hptiop.c
@@ -15,7 +15,6 @@
  *
  * For more information, visit http://www.highpoint-tech.com
  */
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/types.h>
 #include <linux/string.h>
Index: linux-2.6.18/drivers/scsi/dpt_i2o.c
===================================================================
--- linux-2.6.18.orig/drivers/scsi/dpt_i2o.c
+++ linux-2.6.18/drivers/scsi/dpt_i2o.c
@@ -46,7 +46,6 @@ MODULE_DESCRIPTION("Adaptec I2O RAID Dri
 
 #include <linux/stat.h>
 #include <linux/slab.h>		/* for kmalloc() */
-#include <linux/config.h>	/* for CONFIG_PCI */
 #include <linux/pci.h>		/* for PCI support */
 #include <linux/proc_fs.h>
 #include <linux/blkdev.h>
Index: linux-2.6.18/drivers/scsi/libata-eh.c
===================================================================
--- linux-2.6.18.orig/drivers/scsi/libata-eh.c
+++ linux-2.6.18/drivers/scsi/libata-eh.c
@@ -32,7 +32,6 @@
  *
  */
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <scsi/scsi.h>
 #include <scsi/scsi_host.h>
Index: linux-2.6.18/arch/alpha/kernel/setup.c
===================================================================
--- linux-2.6.18.orig/arch/alpha/kernel/setup.c
+++ linux-2.6.18/arch/alpha/kernel/setup.c
@@ -21,7 +21,6 @@
 #include <linux/a.out.h>
 #include <linux/screen_info.h>
 #include <linux/delay.h>
-#include <linux/config.h>	/* CONFIG_ALPHA_LCA etc */
 #include <linux/mc146818rtc.h>
 #include <linux/console.h>
 #include <linux/cpu.h>
Index: linux-2.6.18/arch/alpha/kernel/systbls.S
===================================================================
--- linux-2.6.18.orig/arch/alpha/kernel/systbls.S
+++ linux-2.6.18/arch/alpha/kernel/systbls.S
@@ -4,7 +4,6 @@
  * The system call table. 
  */
 
-#include <linux/config.h>			/* CONFIG_OSF4_COMPAT */
 #include <asm/unistd.h>
 
 	.data
Index: linux-2.6.18/arch/arm/kernel/crunch.c
===================================================================
--- linux-2.6.18.orig/arch/arm/kernel/crunch.c
+++ linux-2.6.18/arch/arm/kernel/crunch.c
@@ -10,7 +10,6 @@
  */
 
 #include <linux/module.h>
-#include <linux/config.h>
 #include <linux/types.h>
 #include <linux/kernel.h>
 #include <linux/signal.h>
Index: linux-2.6.18/arch/arm/kernel/iwmmxt-notifier.c
===================================================================
--- linux-2.6.18.orig/arch/arm/kernel/iwmmxt-notifier.c
+++ linux-2.6.18/arch/arm/kernel/iwmmxt-notifier.c
@@ -15,7 +15,6 @@
  */
 
 #include <linux/module.h>
-#include <linux/config.h>
 #include <linux/types.h>
 #include <linux/kernel.h>
 #include <linux/signal.h>
Index: linux-2.6.18/arch/arm/mach-at91rm9200/board-1arm.c
===================================================================
--- linux-2.6.18.orig/arch/arm/mach-at91rm9200/board-1arm.c
+++ linux-2.6.18/arch/arm/mach-at91rm9200/board-1arm.c
@@ -18,7 +18,6 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-#include <linux/config.h>
 #include <linux/types.h>
 #include <linux/init.h>
 #include <linux/mm.h>
Index: linux-2.6.18/arch/arm/mach-at91rm9200/board-carmeva.c
===================================================================
--- linux-2.6.18.orig/arch/arm/mach-at91rm9200/board-carmeva.c
+++ linux-2.6.18/arch/arm/mach-at91rm9200/board-carmeva.c
@@ -19,7 +19,6 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-#include <linux/config.h>
 #include <linux/types.h>
 #include <linux/init.h>
 #include <linux/mm.h>
Index: linux-2.6.18/arch/arm/mach-at91rm9200/board-eb9200.c
===================================================================
--- linux-2.6.18.orig/arch/arm/mach-at91rm9200/board-eb9200.c
+++ linux-2.6.18/arch/arm/mach-at91rm9200/board-eb9200.c
@@ -19,7 +19,6 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-#include <linux/config.h>
 #include <linux/types.h>
 #include <linux/init.h>
 #include <linux/mm.h>
Index: linux-2.6.18/arch/arm/mach-at91rm9200/board-kafa.c
===================================================================
--- linux-2.6.18.orig/arch/arm/mach-at91rm9200/board-kafa.c
+++ linux-2.6.18/arch/arm/mach-at91rm9200/board-kafa.c
@@ -18,7 +18,6 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-#include <linux/config.h>
 #include <linux/types.h>
 #include <linux/init.h>
 #include <linux/mm.h>
Index: linux-2.6.18/arch/arm/mach-at91rm9200/board-kb9202.c
===================================================================
--- linux-2.6.18.orig/arch/arm/mach-at91rm9200/board-kb9202.c
+++ linux-2.6.18/arch/arm/mach-at91rm9200/board-kb9202.c
@@ -19,7 +19,6 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-#include <linux/config.h>
 #include <linux/types.h>
 #include <linux/init.h>
 #include <linux/mm.h>
Index: linux-2.6.18/arch/arm/mach-ep93xx/edb9302.c
===================================================================
--- linux-2.6.18.orig/arch/arm/mach-ep93xx/edb9302.c
+++ linux-2.6.18/arch/arm/mach-ep93xx/edb9302.c
@@ -10,7 +10,6 @@
  * your option) any later version.
  */
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/init.h>
 #include <linux/mm.h>
Index: linux-2.6.18/arch/arm/mach-ep93xx/edb9315.c
===================================================================
--- linux-2.6.18.orig/arch/arm/mach-ep93xx/edb9315.c
+++ linux-2.6.18/arch/arm/mach-ep93xx/edb9315.c
@@ -10,7 +10,6 @@
  * your option) any later version.
  */
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/init.h>
 #include <linux/mm.h>
Index: linux-2.6.18/arch/arm/mach-ep93xx/edb9315a.c
===================================================================
--- linux-2.6.18.orig/arch/arm/mach-ep93xx/edb9315a.c
+++ linux-2.6.18/arch/arm/mach-ep93xx/edb9315a.c
@@ -10,7 +10,6 @@
  * your option) any later version.
  */
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/init.h>
 #include <linux/mm.h>
Index: linux-2.6.18/arch/arm/mach-lh7a40x/clcd.c
===================================================================
--- linux-2.6.18.orig/arch/arm/mach-lh7a40x/clcd.c
+++ linux-2.6.18/arch/arm/mach-lh7a40x/clcd.c
@@ -8,7 +8,7 @@
  *  version 2 as published by the Free Software Foundation.
  *
  */
-#include <linux/config.h>
+
 #include <linux/init.h>
 #include <linux/device.h>
 #include <linux/dma-mapping.h>
Index: linux-2.6.18/arch/arm/mach-lh7a40x/clocks.c
===================================================================
--- linux-2.6.18.orig/arch/arm/mach-lh7a40x/clocks.c
+++ linux-2.6.18/arch/arm/mach-lh7a40x/clocks.c
@@ -8,7 +8,6 @@
  *
  */
 
-#include <linux/config.h>
 #include <linux/cpufreq.h>
 #include <asm/hardware.h>
 #include <asm/arch/clocks.h>
Index: linux-2.6.18/arch/arm/mach-omap2/pm-domain.c
===================================================================
--- linux-2.6.18.orig/arch/arm/mach-omap2/pm-domain.c
+++ linux-2.6.18/arch/arm/mach-omap2/pm-domain.c
@@ -15,7 +15,6 @@
  * published by the Free Software Foundation.
  */
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/init.h>
 #include <linux/clk.h>
Index: linux-2.6.18/arch/arm/mach-pnx4008/gpio.c
===================================================================
--- linux-2.6.18.orig/arch/arm/mach-pnx4008/gpio.c
+++ linux-2.6.18/arch/arm/mach-pnx4008/gpio.c
@@ -14,7 +14,6 @@
  * or implied.
  */
 
-#include <linux/config.h>
 #include <linux/types.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
Index: linux-2.6.18/arch/arm/mach-pnx4008/sleep.S
===================================================================
--- linux-2.6.18.orig/arch/arm/mach-pnx4008/sleep.S
+++ linux-2.6.18/arch/arm/mach-pnx4008/sleep.S
@@ -11,7 +11,6 @@
  * or implied.
  */
 
-#include <linux/config.h>
 #include <linux/linkage.h>
 #include <asm/assembler.h>
 #include <asm/hardware.h>
Index: linux-2.6.18/arch/arm/mach-pnx4008/time.c
===================================================================
--- linux-2.6.18.orig/arch/arm/mach-pnx4008/time.c
+++ linux-2.6.18/arch/arm/mach-pnx4008/time.c
@@ -11,7 +11,6 @@
  * or implied.
  */
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/init.h>
 #include <linux/delay.h>
Index: linux-2.6.18/arch/arm/mach-pxa/leds-trizeps4.c
===================================================================
--- linux-2.6.18.orig/arch/arm/mach-pxa/leds-trizeps4.c
+++ linux-2.6.18/arch/arm/mach-pxa/leds-trizeps4.c
@@ -10,7 +10,6 @@
  * published by the Free Software Foundation.
  */
 
-#include <linux/config.h>
 #include <linux/init.h>
 
 #include <asm/hardware.h>
Index: linux-2.6.18/arch/arm26/lib/ecard.S
===================================================================
--- linux-2.6.18.orig/arch/arm26/lib/ecard.S
+++ linux-2.6.18/arch/arm26/lib/ecard.S
@@ -7,7 +7,6 @@
  * it under the terms of the GNU General Public License version 2 as
  * published by the Free Software Foundation.
  */
-#include <linux/config.h> /* for CONFIG_CPU_nn */
 #include <linux/linkage.h>
 #include <asm/assembler.h>
 #include <asm/hardware.h>
Index: linux-2.6.18/arch/arm26/lib/io-acorn.S
===================================================================
--- linux-2.6.18.orig/arch/arm26/lib/io-acorn.S
+++ linux-2.6.18/arch/arm26/lib/io-acorn.S
@@ -7,7 +7,6 @@
  * it under the terms of the GNU General Public License version 2 as
  * published by the Free Software Foundation.
  */
-#include <linux/config.h> /* for CONFIG_CPU_nn */
 #include <linux/linkage.h>
 #include <asm/assembler.h>
 #include <asm/hardware.h>
Index: linux-2.6.18/arch/frv/kernel/time.c
===================================================================
--- linux-2.6.18.orig/arch/frv/kernel/time.c
+++ linux-2.6.18/arch/frv/kernel/time.c
@@ -10,7 +10,6 @@
  * 2 of the License, or (at your option) any later version.
  */
 
-#include <linux/config.h> /* CONFIG_HEARTBEAT */
 #include <linux/module.h>
 #include <linux/errno.h>
 #include <linux/sched.h>
Index: linux-2.6.18/arch/h8300/kernel/time.c
===================================================================
--- linux-2.6.18.orig/arch/h8300/kernel/time.c
+++ linux-2.6.18/arch/h8300/kernel/time.c
@@ -16,7 +16,6 @@
  *		"A Kernel Model for Precision Timekeeping" by Dave Mills
  */
 
-#include <linux/config.h> /* CONFIG_HEARTBEAT */
 #include <linux/errno.h>
 #include <linux/module.h>
 #include <linux/sched.h>
Index: linux-2.6.18/arch/m32r/mm/mmu.S
===================================================================
--- linux-2.6.18.orig/arch/m32r/mm/mmu.S
+++ linux-2.6.18/arch/m32r/mm/mmu.S
@@ -6,7 +6,6 @@
 
 /* $Id: mmu.S,v 1.15 2004/03/16 02:56:27 takata Exp $ */
 
-#include <linux/config.h>	/* CONFIG_MMU */
 #include <linux/linkage.h>
 #include <asm/assembler.h>
 #include <asm/smp.h>
Index: linux-2.6.18/arch/m68k/kernel/time.c
===================================================================
--- linux-2.6.18.orig/arch/m68k/kernel/time.c
+++ linux-2.6.18/arch/m68k/kernel/time.c
@@ -10,7 +10,6 @@
  *		"A Kernel Model for Precision Timekeeping" by Dave Mills
  */
 
-#include <linux/config.h> /* CONFIG_HEARTBEAT */
 #include <linux/errno.h>
 #include <linux/module.h>
 #include <linux/sched.h>
Index: linux-2.6.18/arch/m68knommu/platform/532x/config.c
===================================================================
--- linux-2.6.18.orig/arch/m68knommu/platform/532x/config.c
+++ linux-2.6.18/arch/m68knommu/platform/532x/config.c
@@ -17,7 +17,6 @@
 
 /***************************************************************************/
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/sched.h>
 #include <linux/param.h>
Index: linux-2.6.18/arch/m68knommu/platform/68328/romvec.S
===================================================================
--- linux-2.6.18.orig/arch/m68knommu/platform/68328/romvec.S
+++ linux-2.6.18/arch/m68knommu/platform/68328/romvec.S
@@ -10,8 +10,6 @@
  * Copyright 2006 Greg Ungerer <gerg@snapgear.com>
  */
 
-#include <linux/config.h>
-
 .global _start
 .global _buserr
 .global trap
Index: linux-2.6.18/arch/parisc/kernel/head.S
===================================================================
--- linux-2.6.18.orig/arch/parisc/kernel/head.S
+++ linux-2.6.18/arch/parisc/kernel/head.S
@@ -12,8 +12,6 @@
  * Initial Version 04-23-1999 by Helge Deller <deller@gmx.de>
  */
 
-#include <linux/config.h>	/* for CONFIG_SMP */
-
 #include <asm/asm-offsets.h>
 #include <asm/psw.h>
 #include <asm/pdc.h>
Index: linux-2.6.18/arch/parisc/kernel/syscall.S
===================================================================
--- linux-2.6.18.orig/arch/parisc/kernel/syscall.S
+++ linux-2.6.18/arch/parisc/kernel/syscall.S
@@ -6,7 +6,6 @@
  * thanks to Philipp Rumpf, Mike Shaver and various others
  * sorry about the wall, puffin..
  */
-#include <linux/config.h> /* for CONFIG_SMP */
 
 #include <asm/asm-offsets.h>
 #include <asm/unistd.h>
Index: linux-2.6.18/arch/powerpc/platforms/83xx/mpc834x_itx.c
===================================================================
--- linux-2.6.18.orig/arch/powerpc/platforms/83xx/mpc834x_itx.c
+++ linux-2.6.18/arch/powerpc/platforms/83xx/mpc834x_itx.c
@@ -11,7 +11,6 @@
  * option) any later version.
  */
 
-#include <linux/config.h>
 #include <linux/stddef.h>
 #include <linux/kernel.h>
 #include <linux/init.h>
Index: linux-2.6.18/arch/powerpc/platforms/85xx/mpc85xx_cds.c
===================================================================
--- linux-2.6.18.orig/arch/powerpc/platforms/85xx/mpc85xx_cds.c
+++ linux-2.6.18/arch/powerpc/platforms/85xx/mpc85xx_cds.c
@@ -11,7 +11,6 @@
  * option) any later version.
  */
 
-#include <linux/config.h>
 #include <linux/stddef.h>
 #include <linux/kernel.h>
 #include <linux/init.h>
Index: linux-2.6.18/arch/powerpc/platforms/cell/cbe_regs.c
===================================================================
--- linux-2.6.18.orig/arch/powerpc/platforms/cell/cbe_regs.c
+++ linux-2.6.18/arch/powerpc/platforms/cell/cbe_regs.c
@@ -6,8 +6,6 @@
  * (c) 2006 Benjamin Herrenschmidt <benh@kernel.crashing.org>, IBM Corp.
  */
 
-
-#include <linux/config.h>
 #include <linux/percpu.h>
 #include <linux/types.h>
 
Index: linux-2.6.18/arch/powerpc/platforms/cell/ras.c
===================================================================
--- linux-2.6.18.orig/arch/powerpc/platforms/cell/ras.c
+++ linux-2.6.18/arch/powerpc/platforms/cell/ras.c
@@ -1,6 +1,5 @@
 #define DEBUG
 
-#include <linux/config.h>
 #include <linux/types.h>
 #include <linux/kernel.h>
 #include <linux/smp.h>
Index: linux-2.6.18/arch/powerpc/platforms/embedded6xx/mpc7448_hpc2.c
===================================================================
--- linux-2.6.18.orig/arch/powerpc/platforms/embedded6xx/mpc7448_hpc2.c
+++ linux-2.6.18/arch/powerpc/platforms/embedded6xx/mpc7448_hpc2.c
@@ -18,7 +18,6 @@
  * 2 of the License, or (at your option) any later version.
  */
 
-#include <linux/config.h>
 #include <linux/stddef.h>
 #include <linux/kernel.h>
 #include <linux/pci.h>
Index: linux-2.6.18/arch/powerpc/sysdev/tsi108_dev.c
===================================================================
--- linux-2.6.18.orig/arch/powerpc/sysdev/tsi108_dev.c
+++ linux-2.6.18/arch/powerpc/sysdev/tsi108_dev.c
@@ -9,7 +9,6 @@
  * option) any later version.
  */
 
-#include <linux/config.h>
 #include <linux/stddef.h>
 #include <linux/kernel.h>
 #include <linux/init.h>
Index: linux-2.6.18/arch/ppc/amiga/time.c
===================================================================
--- linux-2.6.18.orig/arch/ppc/amiga/time.c
+++ linux-2.6.18/arch/ppc/amiga/time.c
@@ -1,4 +1,3 @@
-#include <linux/config.h> /* CONFIG_HEARTBEAT */
 #include <linux/errno.h>
 #include <linux/sched.h>
 #include <linux/kernel.h>
Index: linux-2.6.18/arch/ppc/platforms/4xx/cpci405.h
===================================================================
--- linux-2.6.18.orig/arch/ppc/platforms/4xx/cpci405.h
+++ linux-2.6.18/arch/ppc/platforms/4xx/cpci405.h
@@ -13,7 +13,6 @@
 #ifndef __CPCI405_H__
 #define __CPCI405_H__
 
-#include <linux/config.h>
 #include <platforms/4xx/ibm405gp.h>
 #include <asm/ppcboot.h>
 
Index: linux-2.6.18/arch/sparc/kernel/ioport.c
===================================================================
--- linux-2.6.18.orig/arch/sparc/kernel/ioport.c
+++ linux-2.6.18/arch/sparc/kernel/ioport.c
@@ -25,7 +25,6 @@
  * <zaitcev> Sounds reasonable
  */
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/sched.h>
 #include <linux/kernel.h>
Index: linux-2.6.18/arch/sparc/kernel/of_device.c
===================================================================
--- linux-2.6.18.orig/arch/sparc/kernel/of_device.c
+++ linux-2.6.18/arch/sparc/kernel/of_device.c
@@ -1,4 +1,3 @@
-#include <linux/config.h>
 #include <linux/string.h>
 #include <linux/kernel.h>
 #include <linux/init.h>
Index: linux-2.6.18/arch/sparc64/kernel/auxio.c
===================================================================
--- linux-2.6.18.orig/arch/sparc64/kernel/auxio.c
+++ linux-2.6.18/arch/sparc64/kernel/auxio.c
@@ -5,7 +5,6 @@
  * Refactoring for unified NCR/PCIO support 2002 Eric Brower (ebrower@usa.net)
  */
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/kernel.h>
 #include <linux/init.h>
Index: linux-2.6.18/arch/sparc64/kernel/of_device.c
===================================================================
--- linux-2.6.18.orig/arch/sparc64/kernel/of_device.c
+++ linux-2.6.18/arch/sparc64/kernel/of_device.c
@@ -1,4 +1,3 @@
-#include <linux/config.h>
 #include <linux/string.h>
 #include <linux/kernel.h>
 #include <linux/init.h>
Index: linux-2.6.18/arch/sparc64/kernel/prom.c
===================================================================
--- linux-2.6.18.orig/arch/sparc64/kernel/prom.c
+++ linux-2.6.18/arch/sparc64/kernel/prom.c
@@ -15,7 +15,6 @@
  *      2 of the License, or (at your option) any later version.
  */
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/types.h>
 #include <linux/string.h>
Index: linux-2.6.18/arch/um/drivers/hostaudio_kern.c
===================================================================
--- linux-2.6.18.orig/arch/um/drivers/hostaudio_kern.c
+++ linux-2.6.18/arch/um/drivers/hostaudio_kern.c
@@ -3,7 +3,6 @@
  * Licensed under the GPL
  */
 
-#include "linux/config.h"
 #include "linux/module.h"
 #include "linux/init.h"
 #include "linux/slab.h"
Index: linux-2.6.18/arch/um/drivers/net_kern.c
===================================================================
--- linux-2.6.18.orig/arch/um/drivers/net_kern.c
+++ linux-2.6.18/arch/um/drivers/net_kern.c
@@ -5,7 +5,6 @@
  * Licensed under the GPL.
  */
 
-#include "linux/config.h"
 #include "linux/kernel.h"
 #include "linux/netdevice.h"
 #include "linux/rtnetlink.h"
Index: linux-2.6.18/arch/um/drivers/slip_kern.c
===================================================================
--- linux-2.6.18.orig/arch/um/drivers/slip_kern.c
+++ linux-2.6.18/arch/um/drivers/slip_kern.c
@@ -1,4 +1,3 @@
-#include "linux/config.h"
 #include "linux/kernel.h"
 #include "linux/stddef.h"
 #include "linux/init.h"
Index: linux-2.6.18/arch/um/drivers/ssl.c
===================================================================
--- linux-2.6.18.orig/arch/um/drivers/ssl.c
+++ linux-2.6.18/arch/um/drivers/ssl.c
@@ -3,7 +3,6 @@
  * Licensed under the GPL
  */
 
-#include "linux/config.h"
 #include "linux/fs.h"
 #include "linux/tty.h"
 #include "linux/tty_driver.h"
Index: linux-2.6.18/arch/um/drivers/stdio_console.c
===================================================================
--- linux-2.6.18.orig/arch/um/drivers/stdio_console.c
+++ linux-2.6.18/arch/um/drivers/stdio_console.c
@@ -3,7 +3,6 @@
  * Licensed under the GPL
  */
 
-#include "linux/config.h"
 #include "linux/posix_types.h"
 #include "linux/tty.h"
 #include "linux/tty_flip.h"
Index: linux-2.6.18/arch/um/drivers/ubd_kern.c
===================================================================
--- linux-2.6.18.orig/arch/um/drivers/ubd_kern.c
+++ linux-2.6.18/arch/um/drivers/ubd_kern.c
@@ -20,7 +20,6 @@
 #define MAJOR_NR UBD_MAJOR
 #define UBD_SHIFT 4
 
-#include "linux/config.h"
 #include "linux/module.h"
 #include "linux/blkdev.h"
 #include "linux/hdreg.h"
Index: linux-2.6.18/arch/um/include/mconsole_kern.h
===================================================================
--- linux-2.6.18.orig/arch/um/include/mconsole_kern.h
+++ linux-2.6.18/arch/um/include/mconsole_kern.h
@@ -6,7 +6,6 @@
 #ifndef __MCONSOLE_KERN_H__
 #define __MCONSOLE_KERN_H__
 
-#include "linux/config.h"
 #include "linux/list.h"
 #include "mconsole.h"
 
Index: linux-2.6.18/arch/um/include/mode_kern.h
===================================================================
--- linux-2.6.18.orig/arch/um/include/mode_kern.h
+++ linux-2.6.18/arch/um/include/mode_kern.h
@@ -6,8 +6,6 @@
 #ifndef __MODE_KERN_H__
 #define __MODE_KERN_H__
 
-#include "linux/config.h"
-
 #ifdef CONFIG_MODE_TT
 #include "mode_kern_tt.h"
 #endif
Index: linux-2.6.18/arch/um/include/skas/mmu-skas.h
===================================================================
--- linux-2.6.18.orig/arch/um/include/skas/mmu-skas.h
+++ linux-2.6.18/arch/um/include/skas/mmu-skas.h
@@ -6,7 +6,6 @@
 #ifndef __SKAS_MMU_H
 #define __SKAS_MMU_H
 
-#include "linux/config.h"
 #include "mm_id.h"
 #include "asm/ldt.h"
 
Index: linux-2.6.18/arch/um/include/sysdep-ppc/ptrace.h
===================================================================
--- linux-2.6.18.orig/arch/um/include/sysdep-ppc/ptrace.h
+++ linux-2.6.18/arch/um/include/sysdep-ppc/ptrace.h
@@ -5,7 +5,6 @@
 #ifndef __SYS_PTRACE_PPC_H
 #define __SYS_PTRACE_PPC_H
 
-#include "linux/config.h"
 #include "linux/types.h"
 
 /* the following taken from <asm-ppc/ptrace.h> */
Index: linux-2.6.18/arch/um/include/um_uaccess.h
===================================================================
--- linux-2.6.18.orig/arch/um/include/um_uaccess.h
+++ linux-2.6.18/arch/um/include/um_uaccess.h
@@ -6,7 +6,6 @@
 #ifndef __ARCH_UM_UACCESS_H
 #define __ARCH_UM_UACCESS_H
 
-#include "linux/config.h"
 #include "choose-mode.h"
 
 #ifdef CONFIG_MODE_TT
Index: linux-2.6.18/arch/um/kernel/init_task.c
===================================================================
--- linux-2.6.18.orig/arch/um/kernel/init_task.c
+++ linux-2.6.18/arch/um/kernel/init_task.c
@@ -3,7 +3,6 @@
  * Licensed under the GPL
  */
 
-#include "linux/config.h"
 #include "linux/mm.h"
 #include "linux/module.h"
 #include "linux/sched.h"
Index: linux-2.6.18/arch/um/kernel/irq.c
===================================================================
--- linux-2.6.18.orig/arch/um/kernel/irq.c
+++ linux-2.6.18/arch/um/kernel/irq.c
@@ -5,7 +5,6 @@
  *	Copyright (C) 1992, 1998 Linus Torvalds, Ingo Molnar
  */
 
-#include "linux/config.h"
 #include "linux/kernel.h"
 #include "linux/module.h"
 #include "linux/smp.h"
Index: linux-2.6.18/arch/um/kernel/ksyms.c
===================================================================
--- linux-2.6.18.orig/arch/um/kernel/ksyms.c
+++ linux-2.6.18/arch/um/kernel/ksyms.c
@@ -3,7 +3,6 @@
  * Licensed under the GPL
  */
 
-#include "linux/config.h"
 #include "linux/module.h"
 #include "linux/string.h"
 #include "linux/smp_lock.h"
Index: linux-2.6.18/arch/um/kernel/process_kern.c
===================================================================
--- linux-2.6.18.orig/arch/um/kernel/process_kern.c
+++ linux-2.6.18/arch/um/kernel/process_kern.c
@@ -4,7 +4,6 @@
  * Licensed under the GPL
  */
 
-#include "linux/config.h"
 #include "linux/kernel.h"
 #include "linux/sched.h"
 #include "linux/interrupt.h"
Index: linux-2.6.18/arch/um/kernel/signal.c
===================================================================
--- linux-2.6.18.orig/arch/um/kernel/signal.c
+++ linux-2.6.18/arch/um/kernel/signal.c
@@ -3,7 +3,6 @@
  * Licensed under the GPL
  */
 
-#include "linux/config.h"
 #include "linux/stddef.h"
 #include "linux/sys.h"
 #include "linux/sched.h"
Index: linux-2.6.18/arch/um/kernel/skas/mem.c
===================================================================
--- linux-2.6.18.orig/arch/um/kernel/skas/mem.c
+++ linux-2.6.18/arch/um/kernel/skas/mem.c
@@ -3,7 +3,6 @@
  * Licensed under the GPL
  */
 
-#include "linux/config.h"
 #include "linux/mm.h"
 #include "asm/pgtable.h"
 #include "mem_user.h"
Index: linux-2.6.18/arch/um/kernel/skas/mmu.c
===================================================================
--- linux-2.6.18.orig/arch/um/kernel/skas/mmu.c
+++ linux-2.6.18/arch/um/kernel/skas/mmu.c
@@ -3,7 +3,6 @@
  * Licensed under the GPL
  */
 
-#include "linux/config.h"
 #include "linux/sched.h"
 #include "linux/list.h"
 #include "linux/spinlock.h"
Index: linux-2.6.18/arch/um/kernel/skas/tlb.c
===================================================================
--- linux-2.6.18.orig/arch/um/kernel/skas/tlb.c
+++ linux-2.6.18/arch/um/kernel/skas/tlb.c
@@ -6,7 +6,6 @@
 
 #include "linux/stddef.h"
 #include "linux/sched.h"
-#include "linux/config.h"
 #include "linux/mm.h"
 #include "asm/page.h"
 #include "asm/pgtable.h"
Index: linux-2.6.18/arch/um/kernel/smp.c
===================================================================
--- linux-2.6.18.orig/arch/um/kernel/smp.c
+++ linux-2.6.18/arch/um/kernel/smp.c
@@ -3,7 +3,6 @@
  * Licensed under the GPL
  */
 
-#include "linux/config.h"
 #include "linux/percpu.h"
 #include "asm/pgalloc.h"
 #include "asm/tlb.h"
Index: linux-2.6.18/arch/um/kernel/sysrq.c
===================================================================
--- linux-2.6.18.orig/arch/um/kernel/sysrq.c
+++ linux-2.6.18/arch/um/kernel/sysrq.c
@@ -3,7 +3,6 @@
  * Licensed under the GPL
  */
 
-#include "linux/config.h"
 #include "linux/sched.h"
 #include "linux/kernel.h"
 #include "linux/module.h"
Index: linux-2.6.18/arch/um/kernel/trap.c
===================================================================
--- linux-2.6.18.orig/arch/um/kernel/trap.c
+++ linux-2.6.18/arch/um/kernel/trap.c
@@ -8,7 +8,6 @@
 #include "linux/sched.h"
 #include "linux/mm.h"
 #include "linux/spinlock.h"
-#include "linux/config.h"
 #include "linux/init.h"
 #include "linux/ptrace.h"
 #include "asm/semaphore.h"
Index: linux-2.6.18/arch/um/kernel/tt/gdb_kern.c
===================================================================
--- linux-2.6.18.orig/arch/um/kernel/tt/gdb_kern.c
+++ linux-2.6.18/arch/um/kernel/tt/gdb_kern.c
@@ -4,7 +4,6 @@
  */
 
 #include "linux/init.h"
-#include "linux/config.h"
 #include "mconsole_kern.h"
 
 #ifdef CONFIG_MCONSOLE
Index: linux-2.6.18/arch/um/kernel/tt/mem.c
===================================================================
--- linux-2.6.18.orig/arch/um/kernel/tt/mem.c
+++ linux-2.6.18/arch/um/kernel/tt/mem.c
@@ -4,7 +4,6 @@
  */
 
 #include "linux/stddef.h"
-#include "linux/config.h"
 #include "linux/mm.h"
 #include "asm/uaccess.h"
 #include "mem_user.h"
Index: linux-2.6.18/arch/um/kernel/um_arch.c
===================================================================
--- linux-2.6.18.orig/arch/um/kernel/um_arch.c
+++ linux-2.6.18/arch/um/kernel/um_arch.c
@@ -3,7 +3,6 @@
  * Licensed under the GPL
  */
 
-#include "linux/config.h"
 #include "linux/kernel.h"
 #include "linux/sched.h"
 #include "linux/notifier.h"
Index: linux-2.6.18/arch/um/sys-i386/ldt.c
===================================================================
--- linux-2.6.18.orig/arch/um/sys-i386/ldt.c
+++ linux-2.6.18/arch/um/sys-i386/ldt.c
@@ -4,7 +4,6 @@
  */
 
 #include "linux/stddef.h"
-#include "linux/config.h"
 #include "linux/sched.h"
 #include "linux/slab.h"
 #include "linux/types.h"
Index: linux-2.6.18/arch/um/sys-i386/sysrq.c
===================================================================
--- linux-2.6.18.orig/arch/um/sys-i386/sysrq.c
+++ linux-2.6.18/arch/um/sys-i386/sysrq.c
@@ -3,7 +3,6 @@
  * Licensed under the GPL
  */
 
-#include "linux/config.h"
 #include "linux/kernel.h"
 #include "linux/smp.h"
 #include "linux/sched.h"
Index: linux-2.6.18/arch/um/sys-i386/tls.c
===================================================================
--- linux-2.6.18.orig/arch/um/sys-i386/tls.c
+++ linux-2.6.18/arch/um/sys-i386/tls.c
@@ -3,7 +3,6 @@
  * Licensed under the GPL
  */
 
-#include "linux/config.h"
 #include "linux/kernel.h"
 #include "linux/sched.h"
 #include "linux/slab.h"
Index: linux-2.6.18/arch/v850/kernel/time.c
===================================================================
--- linux-2.6.18.orig/arch/v850/kernel/time.c
+++ linux-2.6.18/arch/v850/kernel/time.c
@@ -10,7 +10,6 @@
  *		"A Kernel Model for Precision Timekeeping" by Dave Mills
  */
 
-#include <linux/config.h> /* CONFIG_HEARTBEAT */
 #include <linux/errno.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
Index: linux-2.6.18/arch/x86_64/boot/video.S
===================================================================
--- linux-2.6.18.orig/arch/x86_64/boot/video.S
+++ linux-2.6.18/arch/x86_64/boot/video.S
@@ -11,8 +11,6 @@
  *
  */
 
-#include <linux/config.h> /* for CONFIG_VIDEO_* */
-
 /* Enable autodetection of SVGA adapters and modes. */
 #undef CONFIG_VIDEO_SVGA
 
Index: linux-2.6.18/arch/x86_64/ia32/ia32_binfmt.c
===================================================================
--- linux-2.6.18.orig/arch/x86_64/ia32/ia32_binfmt.c
+++ linux-2.6.18/arch/x86_64/ia32/ia32_binfmt.c
@@ -6,7 +6,6 @@
  * of ugly preprocessor tricks. Talk about very very poor man's inheritance.
  */ 
 #include <linux/types.h>
-#include <linux/config.h> 
 #include <linux/stddef.h>
 #include <linux/rwsem.h>
 #include <linux/sched.h>
Index: linux-2.6.18/arch/x86_64/kernel/pci-calgary.c
===================================================================
--- linux-2.6.18.orig/arch/x86_64/kernel/pci-calgary.c
+++ linux-2.6.18/arch/x86_64/kernel/pci-calgary.c
@@ -21,7 +21,6 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA
  */
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/init.h>
 #include <linux/types.h>
Index: linux-2.6.18/arch/x86_64/kernel/tce.c
===================================================================
--- linux-2.6.18.orig/arch/x86_64/kernel/tce.c
+++ linux-2.6.18/arch/x86_64/kernel/tce.c
@@ -21,7 +21,6 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA
  */
 
-#include <linux/config.h>
 #include <linux/types.h>
 #include <linux/slab.h>
 #include <linux/mm.h>
Index: linux-2.6.18/arch/x86_64/kernel/x8664_ksyms.c
===================================================================
--- linux-2.6.18.orig/arch/x86_64/kernel/x8664_ksyms.c
+++ linux-2.6.18/arch/x86_64/kernel/x8664_ksyms.c
@@ -1,7 +1,6 @@
 /* Exports for assembly files.
    All C exports should go in the respective C files. */
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/smp.h>
 
Index: linux-2.6.18/arch/x86_64/lib/delay.c
===================================================================
--- linux-2.6.18.orig/arch/x86_64/lib/delay.c
+++ linux-2.6.18/arch/x86_64/lib/delay.c
@@ -8,7 +8,6 @@
  *	depends wildly on alignment on many x86 processors. 
  */
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/sched.h>
 #include <linux/delay.h>
Index: linux-2.6.18/arch/x86_64/lib/thunk.S
===================================================================
--- linux-2.6.18.orig/arch/x86_64/lib/thunk.S
+++ linux-2.6.18/arch/x86_64/lib/thunk.S
@@ -6,7 +6,6 @@
 	 * $Id: thunk.S,v 1.2 2002/03/13 20:06:58 ak Exp $
 	 */
 
-	#include <linux/config.h>
 	#include <linux/linkage.h>
 	#include <asm/dwarf2.h>
 	#include <asm/calling.h>			
Index: linux-2.6.18/drivers/block/cciss.c
===================================================================
--- linux-2.6.18.orig/drivers/block/cciss.c
+++ linux-2.6.18/drivers/block/cciss.c
@@ -20,7 +20,6 @@
  *
  */
 
-#include <linux/config.h>	/* CONFIG_PROC_FS */
 #include <linux/module.h>
 #include <linux/interrupt.h>
 #include <linux/types.h>
Index: linux-2.6.18/drivers/block/cpqarray.c
===================================================================
--- linux-2.6.18.orig/drivers/block/cpqarray.c
+++ linux-2.6.18/drivers/block/cpqarray.c
@@ -19,7 +19,6 @@
  *    Questions/Comments/Bugfixes to iss_storagedev@hp.com
  *
  */
-#include <linux/config.h>	/* CONFIG_PROC_FS */
 #include <linux/module.h>
 #include <linux/types.h>
 #include <linux/pci.h>
Index: linux-2.6.18/drivers/char/ftape/lowlevel/fdc-io.c
===================================================================
--- linux-2.6.18.orig/drivers/char/ftape/lowlevel/fdc-io.c
+++ linux-2.6.18/drivers/char/ftape/lowlevel/fdc-io.c
@@ -26,7 +26,6 @@
  *      Linux.
  */
 
-#include <linux/config.h> /* for CONFIG_FT_* */
 #include <linux/errno.h>
 #include <linux/sched.h>
 #include <linux/ioport.h>
Index: linux-2.6.18/drivers/char/ftape/zftape/zftape-rw.c
===================================================================
--- linux-2.6.18.orig/drivers/char/ftape/zftape/zftape-rw.c
+++ linux-2.6.18/drivers/char/ftape/zftape/zftape-rw.c
@@ -24,7 +24,6 @@
  *      zftape.
  */
 
-#include <linux/config.h> /* for CONFIG_ZFT_DFLT_BLK_SZ */
 #include <linux/errno.h>
 #include <linux/mm.h>
 
Index: linux-2.6.18/drivers/char/ftape/zftape/zftape-rw.h
===================================================================
--- linux-2.6.18.orig/drivers/char/ftape/zftape/zftape-rw.h
+++ linux-2.6.18/drivers/char/ftape/zftape/zftape-rw.h
@@ -28,7 +28,6 @@
  *
  */
 
-#include <linux/config.h> /* for CONFIG_ZFT_DFLT_BLK_SZ */
 #include "../zftape/zftape-buffers.h"
 
 #define SEGMENTS_PER_TAPE  (ft_segments_per_track * ft_tracks_per_tape)
Index: linux-2.6.18/drivers/char/hw_random/ixp4xx-rng.c
===================================================================
--- linux-2.6.18.orig/drivers/char/hw_random/ixp4xx-rng.c
+++ linux-2.6.18/drivers/char/hw_random/ixp4xx-rng.c
@@ -15,7 +15,6 @@
  */
 
 #include <linux/kernel.h>
-#include <linux/config.h>
 #include <linux/types.h>
 #include <linux/module.h>
 #include <linux/moduleparam.h>
Index: linux-2.6.18/drivers/char/nsc_gpio.c
===================================================================
--- linux-2.6.18.orig/drivers/char/nsc_gpio.c
+++ linux-2.6.18/drivers/char/nsc_gpio.c
@@ -7,7 +7,6 @@
    Copyright (c) 2005      Jim Cromie <jim.cromie@gmail.com>
 */
 
-#include <linux/config.h>
 #include <linux/fs.h>
 #include <linux/module.h>
 #include <linux/errno.h>
Index: linux-2.6.18/drivers/char/pcmcia/synclink_cs.c
===================================================================
--- linux-2.6.18.orig/drivers/char/pcmcia/synclink_cs.c
+++ linux-2.6.18/drivers/char/pcmcia/synclink_cs.c
@@ -35,7 +35,6 @@
 
 #define MAX_DEVICE_COUNT 4
 
-#include <linux/config.h>	
 #include <linux/module.h>
 #include <linux/errno.h>
 #include <linux/signal.h>
Index: linux-2.6.18/drivers/char/sx.c
===================================================================
--- linux-2.6.18.orig/drivers/char/sx.c
+++ linux-2.6.18/drivers/char/sx.c
@@ -203,9 +203,7 @@
 #define RCS_ID "$Id: sx.c,v 1.33 2000/03/08 10:01:02 wolff, pvdl Exp $"
 #define RCS_REV "$Revision: 1.33 $"
 
-
 #include <linux/module.h>
-#include <linux/config.h> 
 #include <linux/kdev_t.h>
 #include <linux/kernel.h>
 #include <linux/sched.h>
Index: linux-2.6.18/drivers/char/synclink.c
===================================================================
--- linux-2.6.18.orig/drivers/char/synclink.c
+++ linux-2.6.18/drivers/char/synclink.c
@@ -63,7 +63,6 @@
 #define MAX_PCI_DEVICES 10
 #define MAX_TOTAL_DEVICES 20
 
-#include <linux/config.h>	
 #include <linux/module.h>
 #include <linux/errno.h>
 #include <linux/signal.h>
Index: linux-2.6.18/drivers/char/watchdog/pcwd.c
===================================================================
--- linux-2.6.18.orig/drivers/char/watchdog/pcwd.c
+++ linux-2.6.18/drivers/char/watchdog/pcwd.c
@@ -49,7 +49,6 @@
  *	More info available at http://www.berkprod.com/ or http://www.pcwatchdog.com/
  */
 
-#include <linux/config.h>	/* For CONFIG_WATCHDOG_NOWAYOUT/... */
 #include <linux/module.h>	/* For module specific items */
 #include <linux/moduleparam.h>	/* For new moduleparam's */
 #include <linux/types.h>	/* For standard types (like size_t) */
Index: linux-2.6.18/drivers/char/watchdog/pcwd_pci.c
===================================================================
--- linux-2.6.18.orig/drivers/char/watchdog/pcwd_pci.c
+++ linux-2.6.18/drivers/char/watchdog/pcwd_pci.c
@@ -31,7 +31,6 @@
  *	Includes, defines, variables, module parameters, ...
  */
 
-#include <linux/config.h>	/* For CONFIG_WATCHDOG_NOWAYOUT/... */
 #include <linux/module.h>	/* For module specific items */
 #include <linux/moduleparam.h>	/* For new moduleparam's */
 #include <linux/types.h>	/* For standard types (like size_t) */
Index: linux-2.6.18/drivers/hwmon/w83791d.c
===================================================================
--- linux-2.6.18.orig/drivers/hwmon/w83791d.c
+++ linux-2.6.18/drivers/hwmon/w83791d.c
@@ -32,7 +32,6 @@
     program which treats the 83791d as an 83781d).
 */
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/init.h>
 #include <linux/slab.h>
Index: linux-2.6.18/drivers/i2c/busses/i2c-ocores.c
===================================================================
--- linux-2.6.18.orig/drivers/i2c/busses/i2c-ocores.c
+++ linux-2.6.18/drivers/i2c/busses/i2c-ocores.c
@@ -9,7 +9,6 @@
  * kind, whether express or implied.
  */
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
 #include <linux/sched.h>
Index: linux-2.6.18/drivers/ide/pci/generic.c
===================================================================
--- linux-2.6.18.orig/drivers/ide/pci/generic.c
+++ linux-2.6.18/drivers/ide/pci/generic.c
@@ -23,7 +23,6 @@
 
 #undef REALLY_SLOW_IO		/* most systems can safely undef this */
 
-#include <linux/config.h> /* for CONFIG_BLK_DEV_IDEPCI */
 #include <linux/types.h>
 #include <linux/module.h>
 #include <linux/kernel.h>
Index: linux-2.6.18/drivers/ide/pci/rz1000.c
===================================================================
--- linux-2.6.18.orig/drivers/ide/pci/rz1000.c
+++ linux-2.6.18/drivers/ide/pci/rz1000.c
@@ -17,7 +17,6 @@
 
 #undef REALLY_SLOW_IO		/* most systems can safely undef this */
 
-#include <linux/config.h> /* for CONFIG_BLK_DEV_IDEPCI */
 #include <linux/types.h>
 #include <linux/module.h>
 #include <linux/kernel.h>
Index: linux-2.6.18/drivers/leds/leds-ams-delta.c
===================================================================
--- linux-2.6.18.orig/drivers/leds/leds-ams-delta.c
+++ linux-2.6.18/drivers/leds/leds-ams-delta.c
@@ -8,7 +8,6 @@
  * published by the Free Software Foundation.
  */
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/init.h>
 #include <linux/platform_device.h>
Index: linux-2.6.18/drivers/rtc/rtc-max6902.c
===================================================================
--- linux-2.6.18.orig/drivers/rtc/rtc-max6902.c
+++ linux-2.6.18/drivers/rtc/rtc-max6902.c
@@ -19,7 +19,6 @@
  *                - Initial driver creation.
  */
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/version.h>
 
Index: linux-2.6.18/drivers/scsi/imm.h
===================================================================
--- linux-2.6.18.orig/drivers/scsi/imm.h
+++ linux-2.6.18/drivers/scsi/imm.h
@@ -66,7 +66,6 @@
  */
 /* ------ END OF USER CONFIGURABLE PARAMETERS ----- */
 
-#include  <linux/config.h>
 #include  <linux/stddef.h>
 #include  <linux/module.h>
 #include  <linux/kernel.h>
Index: linux-2.6.18/drivers/scsi/ppa.h
===================================================================
--- linux-2.6.18.orig/drivers/scsi/ppa.h
+++ linux-2.6.18/drivers/scsi/ppa.h
@@ -73,7 +73,6 @@
  */
 /* ------ END OF USER CONFIGURABLE PARAMETERS ----- */
 
-#include  <linux/config.h>
 #include  <linux/stddef.h>
 #include  <linux/module.h>
 #include  <linux/kernel.h>
Index: linux-2.6.18/drivers/serial/cpm_uart/cpm_uart.h
===================================================================
--- linux-2.6.18.orig/drivers/serial/cpm_uart/cpm_uart.h
+++ linux-2.6.18/drivers/serial/cpm_uart/cpm_uart.h
@@ -16,7 +16,6 @@
 #ifndef CPM_UART_H
 #define CPM_UART_H
 
-#include <linux/config.h>
 #include <linux/platform_device.h>
 #include <linux/fs_uart_pd.h>
 
Index: linux-2.6.18/drivers/serial/netx-serial.c
===================================================================
--- linux-2.6.18.orig/drivers/serial/netx-serial.c
+++ linux-2.6.18/drivers/serial/netx-serial.c
@@ -17,8 +17,6 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-#include <linux/config.h>
-
 #if defined(CONFIG_SERIAL_NETX_CONSOLE) && defined(CONFIG_MAGIC_SYSRQ)
 #define SUPPORT_SYSRQ
 #endif
Index: linux-2.6.18/drivers/spi/spi_s3c24xx.c
===================================================================
--- linux-2.6.18.orig/drivers/spi/spi_s3c24xx.c
+++ linux-2.6.18/drivers/spi/spi_s3c24xx.c
@@ -13,7 +13,6 @@
 
 //#define DEBUG
 
-#include <linux/config.h>
 #include <linux/init.h>
 #include <linux/spinlock.h>
 #include <linux/workqueue.h>
Index: linux-2.6.18/drivers/spi/spi_s3c24xx_gpio.c
===================================================================
--- linux-2.6.18.orig/drivers/spi/spi_s3c24xx_gpio.c
+++ linux-2.6.18/drivers/spi/spi_s3c24xx_gpio.c
@@ -11,7 +11,6 @@
  *
 */
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/init.h>
 #include <linux/delay.h>
Index: linux-2.6.18/drivers/usb/input/usbtouchscreen.c
===================================================================
--- linux-2.6.18.orig/drivers/usb/input/usbtouchscreen.c
+++ linux-2.6.18/drivers/usb/input/usbtouchscreen.c
@@ -32,7 +32,6 @@
 
 //#define DEBUG
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/slab.h>
 #include <linux/input.h>
Index: linux-2.6.18/drivers/usb/misc/appledisplay.c
===================================================================
--- linux-2.6.18.orig/drivers/usb/misc/appledisplay.c
+++ linux-2.6.18/drivers/usb/misc/appledisplay.c
@@ -20,7 +20,6 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
  */
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/errno.h>
 #include <linux/init.h>
Index: linux-2.6.18/drivers/usb/misc/sisusbvga/sisusb.c
===================================================================
--- linux-2.6.18.orig/drivers/usb/misc/sisusbvga/sisusb.c
+++ linux-2.6.18/drivers/usb/misc/sisusbvga/sisusb.c
@@ -36,7 +36,6 @@
  *
  */
 
-#include <linux/config.h>
 #include <linux/mutex.h>
 #include <linux/module.h>
 #include <linux/kernel.h>
Index: linux-2.6.18/drivers/usb/misc/sisusbvga/sisusb_con.c
===================================================================
--- linux-2.6.18.orig/drivers/usb/misc/sisusbvga/sisusb_con.c
+++ linux-2.6.18/drivers/usb/misc/sisusbvga/sisusb_con.c
@@ -47,7 +47,6 @@
  *
  */
 
-#include <linux/config.h>
 #include <linux/mutex.h>
 #include <linux/module.h>
 #include <linux/kernel.h>
Index: linux-2.6.18/fs/isofs/namei.c
===================================================================
--- linux-2.6.18.orig/fs/isofs/namei.c
+++ linux-2.6.18/fs/isofs/namei.c
@@ -6,7 +6,6 @@
  *  (C) 1991  Linus Torvalds - minix filesystem
  */
 
-#include <linux/config.h>	/* Joliet? */
 #include <linux/smp_lock.h>
 #include "isofs.h"
 
Index: linux-2.6.18/fs/nfs/namespace.c
===================================================================
--- linux-2.6.18.orig/fs/nfs/namespace.c
+++ linux-2.6.18/fs/nfs/namespace.c
@@ -6,8 +6,6 @@
  * NFS namespace
  */
 
-#include <linux/config.h>
-
 #include <linux/dcache.h>
 #include <linux/mount.h>
 #include <linux/namei.h>
Index: linux-2.6.18/fs/nfs/nfs4namespace.c
===================================================================
--- linux-2.6.18.orig/fs/nfs/nfs4namespace.c
+++ linux-2.6.18/fs/nfs/nfs4namespace.c
@@ -6,8 +6,6 @@
  * NFSv4 namespace
  */
 
-#include <linux/config.h>
-
 #include <linux/dcache.h>
 #include <linux/mount.h>
 #include <linux/namei.h>
Index: linux-2.6.18/fs/nfs/nfsroot.c
===================================================================
--- linux-2.6.18.orig/fs/nfs/nfsroot.c
+++ linux-2.6.18/fs/nfs/nfsroot.c
@@ -69,7 +69,6 @@
  *	Fabian Frederick:	Option parser rebuilt (using parser lib)
 */
 
-#include <linux/config.h>
 #include <linux/types.h>
 #include <linux/string.h>
 #include <linux/kernel.h>
Index: linux-2.6.18/fs/nfs/super.c
===================================================================
--- linux-2.6.18.orig/fs/nfs/super.c
+++ linux-2.6.18/fs/nfs/super.c
@@ -15,7 +15,6 @@
  *
  */
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/init.h>
 
Index: linux-2.6.18/include/asm-arm/arch-lh7a40x/clocks.h
===================================================================
--- linux-2.6.18.orig/include/asm-arm/arch-lh7a40x/clocks.h
+++ linux-2.6.18/include/asm-arm/arch-lh7a40x/clocks.h
@@ -8,8 +8,6 @@
  *
  */
 
-#include <linux/config.h>
-
 #ifndef __ASM_ARCH_CLOCKS_H
 #define __ASM_ARCH_CLOCKS_H
 
Index: linux-2.6.18/include/asm-arm/pgtable-nommu.h
===================================================================
--- linux-2.6.18.orig/include/asm-arm/pgtable-nommu.h
+++ linux-2.6.18/include/asm-arm/pgtable-nommu.h
@@ -13,7 +13,6 @@
 
 #ifndef __ASSEMBLY__
 
-#include <linux/config.h>
 #include <linux/slab.h>
 #include <asm/processor.h>
 #include <asm/page.h>
Index: linux-2.6.18/include/asm-powerpc/irq.h
===================================================================
--- linux-2.6.18.orig/include/asm-powerpc/irq.h
+++ linux-2.6.18/include/asm-powerpc/irq.h
@@ -9,7 +9,6 @@
  * 2 of the License, or (at your option) any later version.
  */
 
-#include <linux/config.h>
 #include <linux/threads.h>
 #include <linux/list.h>
 #include <linux/radix-tree.h>
Index: linux-2.6.18/include/asm-powerpc/pci-bridge.h
===================================================================
--- linux-2.6.18.orig/include/asm-powerpc/pci-bridge.h
+++ linux-2.6.18/include/asm-powerpc/pci-bridge.h
@@ -6,7 +6,6 @@
 #include <asm-ppc/pci-bridge.h>
 #else
 
-#include <linux/config.h>
 #include <linux/pci.h>
 #include <linux/list.h>
 
Index: linux-2.6.18/include/asm-x86_64/calgary.h
===================================================================
--- linux-2.6.18.orig/include/asm-x86_64/calgary.h
+++ linux-2.6.18/include/asm-x86_64/calgary.h
@@ -24,7 +24,6 @@
 #ifndef _ASM_X86_64_CALGARY_H
 #define _ASM_X86_64_CALGARY_H
 
-#include <linux/config.h>
 #include <linux/spinlock.h>
 #include <linux/device.h>
 #include <linux/dma-mapping.h>
Index: linux-2.6.18/init/do_mounts.h
===================================================================
--- linux-2.6.18.orig/init/do_mounts.h
+++ linux-2.6.18/init/do_mounts.h
@@ -1,4 +1,3 @@
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/init.h>
 #include <linux/syscalls.h>
Index: linux-2.6.18/kernel/rtmutex-debug.c
===================================================================
--- linux-2.6.18.orig/kernel/rtmutex-debug.c
+++ linux-2.6.18/kernel/rtmutex-debug.c
@@ -16,7 +16,6 @@
  *
  * See rt.c in preempt-rt for proper credits and further information
  */
-#include <linux/config.h>
 #include <linux/sched.h>
 #include <linux/delay.h>
 #include <linux/module.h>
Index: linux-2.6.18/kernel/rtmutex-tester.c
===================================================================
--- linux-2.6.18.orig/kernel/rtmutex-tester.c
+++ linux-2.6.18/kernel/rtmutex-tester.c
@@ -6,7 +6,6 @@
  *  Copyright (C) 2006, Timesys Corp., Thomas Gleixner <tglx@timesys.com>
  *
  */
-#include <linux/config.h>
 #include <linux/kthread.h>
 #include <linux/module.h>
 #include <linux/sched.h>
Index: linux-2.6.18/mm/filemap.h
===================================================================
--- linux-2.6.18.orig/mm/filemap.h
+++ linux-2.6.18/mm/filemap.h
@@ -12,7 +12,6 @@
 #include <linux/mm.h>
 #include <linux/highmem.h>
 #include <linux/uio.h>
-#include <linux/config.h>
 #include <linux/uaccess.h>
 
 size_t
Index: linux-2.6.18/mm/slab.c
===================================================================
--- linux-2.6.18.orig/mm/slab.c
+++ linux-2.6.18/mm/slab.c
@@ -86,7 +86,6 @@
  *	All object allocations for a node occur from node specific slab lists.
  */
 
-#include	<linux/config.h>
 #include	<linux/slab.h>
 #include	<linux/mm.h>
 #include	<linux/poison.h>
Index: linux-2.6.18/mm/vmstat.c
===================================================================
--- linux-2.6.18.orig/mm/vmstat.c
+++ linux-2.6.18/mm/vmstat.c
@@ -9,7 +9,6 @@
  *		Christoph Lameter <christoph@lameter.com>
  */
 
-#include <linux/config.h>
 #include <linux/mm.h>
 #include <linux/module.h>
 #include <linux/cpu.h>
Index: linux-2.6.18/net/atm/atm_sysfs.c
===================================================================
--- linux-2.6.18.orig/net/atm/atm_sysfs.c
+++ linux-2.6.18/net/atm/atm_sysfs.c
@@ -1,6 +1,5 @@
 /* ATM driver model support. */
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/init.h>
 #include <linux/kobject.h>
Index: linux-2.6.18/net/core/dev_mcast.c
===================================================================
--- linux-2.6.18.orig/net/core/dev_mcast.c
+++ linux-2.6.18/net/core/dev_mcast.c
@@ -21,7 +21,6 @@
  *	2 of the License, or (at your option) any later version.
  */
 
-#include <linux/config.h> 
 #include <linux/module.h> 
 #include <asm/uaccess.h>
 #include <asm/system.h>
Index: linux-2.6.18/sound/aoa/soundbus/sysfs.c
===================================================================
--- linux-2.6.18.orig/sound/aoa/soundbus/sysfs.c
+++ linux-2.6.18/sound/aoa/soundbus/sysfs.c
@@ -1,4 +1,3 @@
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/stat.h>
 /* FIX UP */
Index: linux-2.6.18/sound/core/memory.c
===================================================================
--- linux-2.6.18.orig/sound/core/memory.c
+++ linux-2.6.18/sound/core/memory.c
@@ -20,7 +20,6 @@
  *
  */
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <asm/io.h>
 #include <asm/uaccess.h>
Index: linux-2.6.18/arch/i386/kernel/nmi.c
===================================================================
--- linux-2.6.18.orig/arch/i386/kernel/nmi.c
+++ linux-2.6.18/arch/i386/kernel/nmi.c
@@ -13,7 +13,6 @@
  *  Mikael Pettersson	: PM converted to driver model. Disable/enable API.
  */
 
-#include <linux/config.h>
 #include <linux/delay.h>
 #include <linux/interrupt.h>
 #include <linux/module.h>
Index: linux-2.6.18/arch/i386/lib/delay.c
===================================================================
--- linux-2.6.18.orig/arch/i386/lib/delay.c
+++ linux-2.6.18/arch/i386/lib/delay.c
@@ -11,7 +11,6 @@
  */
 
 #include <linux/module.h>
-#include <linux/config.h>
 #include <linux/sched.h>
 #include <linux/delay.h>
 
Index: linux-2.6.18/drivers/usb/input/pid.c
===================================================================
--- linux-2.6.18.orig/drivers/usb/input/pid.c
+++ linux-2.6.18/drivers/usb/input/pid.c
@@ -24,7 +24,6 @@
  * e-mail - mail your message to <rdamazio@lsi.usp.br>
  */
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/slab.h>
 #include <linux/kernel.h>
Index: linux-2.6.18/arch/i386/boot/video.S
===================================================================
--- linux-2.6.18.orig/arch/i386/boot/video.S
+++ linux-2.6.18/arch/i386/boot/video.S
@@ -11,8 +11,6 @@
  *
  */
 
-#include <linux/config.h> /* for CONFIG_VIDEO_* */
-
 /* Enable autodetection of SVGA adapters and modes. */
 #undef CONFIG_VIDEO_SVGA
 
Index: linux-2.6.18/drivers/media/video/cx88/cx88-tvaudio.c
===================================================================
--- linux-2.6.18.orig/drivers/media/video/cx88/cx88-tvaudio.c
+++ linux-2.6.18/drivers/media/video/cx88/cx88-tvaudio.c
@@ -52,7 +52,6 @@
 #include <linux/init.h>
 #include <linux/smp_lock.h>
 #include <linux/delay.h>
-#include <linux/config.h>
 #include <linux/kthread.h>
 
 #include "cx88.h"
Index: linux-2.6.18/drivers/media/video/pvrusb2/pvrusb2-main.c
===================================================================
--- linux-2.6.18.orig/drivers/media/video/pvrusb2/pvrusb2-main.c
+++ linux-2.6.18/drivers/media/video/pvrusb2/pvrusb2-main.c
@@ -20,7 +20,6 @@
  *
  */
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/errno.h>
 #include <linux/slab.h>
Index: linux-2.6.18/drivers/media/video/pvrusb2/pvrusb2-sysfs.c
===================================================================
--- linux-2.6.18.orig/drivers/media/video/pvrusb2/pvrusb2-sysfs.c
+++ linux-2.6.18/drivers/media/video/pvrusb2/pvrusb2-sysfs.c
@@ -19,7 +19,6 @@
  *
  */
 
-#include <linux/config.h>
 #include <linux/string.h>
 #include <linux/slab.h>
 #include <asm/semaphore.h>
Index: linux-2.6.18/drivers/net/phy/fixed.c
===================================================================
--- linux-2.6.18.orig/drivers/net/phy/fixed.c
+++ linux-2.6.18/drivers/net/phy/fixed.c
@@ -13,7 +13,6 @@
  * option) any later version.
  *
  */
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/sched.h>
 #include <linux/string.h>
