--- modules/pam_console/50-default.perms	2006-11-10 19:28:21.102872456 +0200
+++ modules/pam_console/50-default.perms	2006-11-10 19:28:42.514605962 +0200
@@ -33,10 +33,10 @@
 <dvb>=/dev/dvb/adapter*/*
 
 # permission definitions
-<console>  0660 <floppy>     0660 root.floppy
+<console>  0660 <floppy>     0660 root.removable
 #<console>  0600 <sound>      0600 root
 <console>  0600 <cdrom>      0660 root.disk
-<console>  0600 <pilot>      0660 root.uucp
+<console>  0600 <pilot>      0660 root.pnp
 <console>  0600 <jaz>        0660 root.disk
 <console>  0600 <zip>        0660 root.disk
 <console>  0600 <ls120>      0660 root.disk
