Index: stream/tvi_v4l2.c
===================================================================
--- stream/tvi_v4l2.c	(revision 26920)
+++ stream/tvi_v4l2.c	(working copy)
@@ -774,6 +774,20 @@
             return TVI_CONTROL_FALSE;
         }
         return TVI_CONTROL_TRUE;
+    case TVI_CONTROL_VID_CHK_WIDTHHEIGHT:
+        return TVI_CONTROL_TRUE;
+    case TVI_CONTROL_VID_SET_WIDTHHEIGHT:
+        if (getfmt(priv) < 0) return TVI_CONTROL_FALSE;
+        priv->format.fmt.pix.width = ((int *)arg)[0];
+        priv->format.fmt.pix.height = ((int *)arg)[1];
+        mp_msg(MSGT_TV, MSGL_V, "%s: set width x height: %d x %d\n", info.short_name,
+               priv->format.fmt.pix.width, priv->format.fmt.pix.height);
+        if (ioctl(priv->video_fd, VIDIOC_S_FMT, &priv->format) < 0) {
+            mp_msg(MSGT_TV, MSGL_ERR, "%s: ioctl set width&height failed: %s\n",
+                   info.short_name, strerror(errno));
+            return TVI_CONTROL_FALSE;
+        }
+        return TVI_CONTROL_TRUE;
         case TVI_CONTROL_VID_GET_BRIGHTNESS:
             control.id = V4L2_CID_BRIGHTNESS;
             if (get_control(priv, &control, 1) == TVI_CONTROL_TRUE) {
Index: stream/tv.c
===================================================================
--- stream/tv.c	(revision 26920)
+++ stream/tv.c	(working copy)
@@ -318,6 +318,7 @@
 static int open_tv(tvi_handle_t *tvh)
 {
     int i;
+    int wh[2];
     const tvi_functions_t *funcs = tvh->functions;
     int tv_fmt_list[] = {
       IMGFMT_YV12,
@@ -406,6 +407,13 @@
 #endif
 
     /* limits on w&h are norm-dependent -- JM */
+    if (tvh->tv_param->width != -1 && tvh->tv_param->height != -1 &&
+        funcs->control(tvh->priv, TVI_CONTROL_VID_CHK_WIDTHHEIGHT, &tvh->tv_param->width) == TVI_CONTROL_TRUE)
+    {
+        wh[0]=tvh->tv_param->width;
+        wh[1]=tvh->tv_param->height;
+        funcs->control(tvh->priv, TVI_CONTROL_VID_SET_WIDTHHEIGHT, &wh);
+    } else {
     /* set width */
     if (tvh->tv_param->width != -1)
     {
@@ -429,6 +437,7 @@
 	    funcs->control(tvh->priv, TVI_CONTROL_VID_GET_HEIGHT, &tvh->tv_param->height);
 	}
     }
+    }
 
     if (funcs->control(tvh->priv, TVI_CONTROL_IS_TUNER, 0) != TVI_CONTROL_TRUE)
     {
Index: stream/tv.h
===================================================================
--- stream/tv.h	(revision 26920)
+++ stream/tv.h	(working copy)
@@ -174,6 +174,8 @@
 #define TVI_CONTROL_VID_SET_PICTURE	0x11e
 #define TVI_CONTROL_VID_SET_GAIN	0x11f
 #define TVI_CONTROL_VID_GET_GAIN	0x120
+#define TVI_CONTROL_VID_CHK_WIDTHHEIGHT	0x121
+#define TVI_CONTROL_VID_SET_WIDTHHEIGHT	0x122
 
 /* TUNER controls */
 #define TVI_CONTROL_TUN_GET_FREQ	0x201

