diff -Nur mplayer-20080625-old/libavcodec/h263.c mplayer-20080625/libavcodec/h263.c
--- mplayer-20080625-old/libavcodec/h263.c	2008-06-25 23:40:55.293406493 +0300
+++ mplayer-20080625/libavcodec/h263.c	2008-06-25 23:42:07.208406654 +0300
@@ -5752,7 +5752,7 @@
         s->divx_build= build;
         s->divx_packed= e==3 && last=='p';
         if(s->divx_packed)
-            av_log(s->avctx, AV_LOG_WARNING, "Invalid and inefficient vfw-avi packed B frames detected\n");
+            av_log(s->avctx, AV_LOG_DEBUG, "Invalid and inefficient vfw-avi packed B frames detected\n");
     }
 
     /* ffmpeg detection */
