diff -Nur mplayer-20080820-old/libmpdemux/demux_nut.c mplayer-20080820/libmpdemux/demux_nut.c
--- mplayer-20080820-old/libmpdemux/demux_nut.c	2008-08-20 14:51:34.000000000 +0300
+++ mplayer-20080820/libmpdemux/demux_nut.c	2008-08-20 14:51:43.000000000 +0300
@@ -86,8 +86,6 @@
 	while ((ret = nut_read_headers(nut, &s, NULL)) == NUT_ERR_EAGAIN);
 	if (ret) {
 		mp_msg(MSGT_HEADER, MSGL_ERR, "NUT error: %s\n", nut_error(ret));
-		nut_demuxer_uninit(nut);
-		free(priv);
 		return NULL;
 	}
 
