diff -Naur -x '*~' -x '*.orig' heimdal-1.4/lib/kadm5/check-cracklib.pl heimdal-1.4-passwd-check//lib/kadm5/check-cracklib.pl
--- heimdal-1.4/lib/kadm5/check-cracklib.pl	2010-09-13 09:23:34.000000000 +0200
+++ heimdal-1.4-passwd-check//lib/kadm5/check-cracklib.pl	2010-09-15 19:53:28.000000000 +0200
@@ -36,8 +36,8 @@
 use Digest::MD5;
 
 # NEED TO CHANGE THESE TO MATCH YOUR SYSTEM
-my $database = '/usr/lib/cracklib_dict';
-my $historydb = '/var/heimdal/historydb';
+my $database = '/usr/share/cracklib';
+my $historydb = '/var/lib/heimdal/historydb';
 # NEED TO CHANGE THESE TO MATCH YOUR SYSTEM
 
 # seconds password reuse allowed (to catch retries from clients)
diff -Naur -x '*~' -x '*.orig' heimdal-1.4/lib/kadm5/Makefile.am heimdal-1.4-passwd-check//lib/kadm5/Makefile.am
--- heimdal-1.4/lib/kadm5/Makefile.am	2010-09-13 09:23:34.000000000 +0200
+++ heimdal-1.4-passwd-check//lib/kadm5/Makefile.am	2010-09-15 19:53:28.000000000 +0200
@@ -14,10 +14,10 @@
 check_PROGRAMS = default_keys
 noinst_PROGRAMS = test_pw_quality
 
-noinst_LTLIBRARIES = sample_passwd_check.la
+pkglib_LTLIBRARIES = sample_passwd_check.la
 
 sample_passwd_check_la_SOURCES = sample_passwd_check.c
-sample_passwd_check_la_LDFLAGS = -module
+sample_passwd_check_la_LDFLAGS = -module -avoid-version
 
 libkadm5srv_la_LIBADD = \
 	$(LIB_com_err) ../krb5/libkrb5.la \
diff -Naur -x '*~' -x '*.orig' heimdal-1.4/lib/kadm5/sample_passwd_check.c heimdal-1.4-passwd-check//lib/kadm5/sample_passwd_check.c
--- heimdal-1.4/lib/kadm5/sample_passwd_check.c	2010-09-13 09:23:34.000000000 +0200
+++ heimdal-1.4-passwd-check//lib/kadm5/sample_passwd_check.c	2010-09-15 19:53:28.000000000 +0200
@@ -79,7 +79,7 @@
     strings[1] = NULL;
     memcpy(s, password->data, password->length);
     s[password->length] = '\0';
-    msg = FascistCheck(s, DICTPATH, strings);
+    msg = FascistCheck(s, DICTPATH);
     memset(s, 0, password->length);
     free(s);
     return msg;
