--- vobcopy.c	2008-01-05 01:37:21.000000000 +0200
+++ vobcopy.c	2008-02-07 14:04:59.000000000 +0200
@@ -519,7 +519,7 @@
       strcpy( tmp_path, pwd );
       strcat( tmp_path, "vobcopy.bla" );
       fprintf( stderr, "[Hint] Quiet mode - All messages will now end up in %s\n", tmp_path );
-      if ( ( temp = open( tmp_path , O_CREAT | O_EXCL ) ) == -1 )
+      if ( ( temp = open( tmp_path , O_CREAT | O_EXCL, S_IRWXU ) ) == -1 )
 	{
 	  printf( "[Error] Error: %s\n", strerror( errno ) );
 	  if ( errno == EEXIST )
@@ -575,7 +575,7 @@
       strcat( logfile_name, VERSION );
       strcat( logfile_name, ".log" );
       strcat( logfile_path, logfile_name );
-      if ( ( temp = open ( logfile_path , O_CREAT | O_EXCL ) ) == -1 )
+      if ( ( temp = open ( logfile_path , O_CREAT | O_EXCL, S_IRWXU ) ) == -1 )
 	{
 	  printf( "[Error] Error: %s\n", strerror( errno ) );
 	  if ( errno == EEXIST )
