diff -ur nfs-utils-1.0.10.orig/tools/getkversion/getkversion.c nfs-utils-1.0.10/tools/getkversion/getkversion.c
--- nfs-utils-1.0.10.orig/tools/getkversion/getkversion.c	2006-08-07 09:40:50.000000000 +0300
+++ nfs-utils-1.0.10/tools/getkversion/getkversion.c	2007-03-10 23:11:04.726721249 +0200
@@ -14,6 +14,8 @@
 int
 main(void)	/* This is for Dan Popp ;) */
 {
+#ifdef UTS_RELEASE
 	printf("%s\n", UTS_RELEASE);
+#endif
 	return 0;
 }
