Index: colibri-0.1.1_20100518/kcm/alignmentselector.cpp
===================================================================
--- colibri-0.1.1_20100518.orig/kcm/alignmentselector.cpp
+++ colibri-0.1.1_20100518/kcm/alignmentselector.cpp
@@ -79,7 +79,7 @@ AlignmentSelector::AlignmentSelector(QWi
     QDesktopWidget* desktop = QApplication::desktop();
     QRect rect = desktop->availableGeometry(desktop->screenNumber(this));
     qreal ratio = qreal(rect.width()) / rect.height();
-    setFixedSize(300, 300 / ratio + baseHeight);
+    setFixedSize(220, 220 / ratio + baseHeight);
 
     QGridLayout* layout = new QGridLayout(this);
 

