diff -Nur gecko-mediaplayer-0.6.0-old/configure gecko-mediaplayer-0.6.0/configure
--- gecko-mediaplayer-0.6.0-old/configure	2008-03-30 13:50:31.284562871 +0300
+++ gecko-mediaplayer-0.6.0/configure	2008-03-30 13:51:18.151438023 +0300
@@ -6506,7 +6506,7 @@
 _ACEOF
 
 
-ALL_LINGUAS="es fr pl pt_BR"
+ALL_LINGUAS="es fr pl pt_BR tr"
 # On IRIX 5.3, sys/types and inttypes.h are conflicting.
 
 
diff -Nur gecko-mediaplayer-0.6.0-old/configure.in gecko-mediaplayer-0.6.0/configure.in
--- gecko-mediaplayer-0.6.0-old/configure.in	2008-03-30 13:50:31.284562871 +0300
+++ gecko-mediaplayer-0.6.0/configure.in	2008-03-30 13:51:09.279786719 +0300
@@ -23,7 +23,7 @@
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE", [GETTEXT package name])
 
 dnl Add the languages which your application supports here.
-ALL_LINGUAS="es fr pl pt_BR"
+ALL_LINGUAS="es fr pl pt_BR tr"
 AM_GLIB_GNU_GETTEXT
 
 
