diff -Nuar luckybackup-0.4.1.orig/menu/luckybackup-gnome-su.desktop luckybackup-0.4.1/menu/luckybackup-gnome-su.desktop
--- luckybackup-0.4.1.orig/menu/luckybackup-gnome-su.desktop	2010-06-15 14:41:48.000000000 +0000
+++ luckybackup-0.4.1/menu/luckybackup-gnome-su.desktop	2010-07-04 11:11:41.646373039 +0000
@@ -53,7 +53,7 @@
 Comment[sl]=Ustvarjanje varnostnih kopij in usklajevanje z uporabo rsync
 Comment[sv]=Backup & sync your data with the power of rsync
 Comment[tr]=rsync gücüyle verinizi yedekleyin & şipşaklayın
-Exec=su-to-root -X -c /usr/bin/luckybackup
+Exec=xdg-su -c /usr/bin/luckybackup
 Terminal=false
 Type=Application
 Icon=/usr/share/pixmaps/luckybackup.png
