diff -Nur et131x-1.2.3-old/et131x_isr.c et131x-1.2.3/et131x_isr.c
--- et131x-1.2.3-old/et131x_isr.c	2007-07-06 18:10:27.275575075 +0300
+++ et131x-1.2.3/et131x_isr.c	2007-07-06 18:14:36.824637850 +0300
@@ -335,9 +335,8 @@
    REUSE INFORMATION :
         
  *****************************************************************************/
-void et131x_isr_handler( struct work_struct *work )
+void et131x_isr_handler( ET131X_ADAPTER *pAdapter )
 {
-    ET131X_ADAPTER *pAdapter = container_of( work, ET131X_ADAPTER, task );
     
     INT_STATUS_t GlobStatus = pAdapter->Stats.InterruptStatus;
     /*-----------------------------------------------------------------------*/
