Index: comar/src/iksemel.c
===================================================================
--- comar/src/iksemel.c	(revision 27400)
+++ comar/src/iksemel.c	(revision 27401)
@@ -1822,8 +1822,9 @@
 				if (len < FILE_IO_BUF_SIZE) {
 					if (0 == feof (f)) {
 						ret = IKS_FILE_RWERR;
-						len = 0;
+						break;
 					}
+					if (len == 0) ret = IKS_OK;
 					done = 1;
 				}
 				if (len > 0) {

Property changes on: comar/src
___________________________________________________________________
Added: svn:mergeinfo
   Merged /trunk/piksemel/src:r7562

