diff -ur libkdcraw-0.1.1.orig/libkdcraw/dcraw/dcraw.c libkdcraw-0.1.1/libkdcraw/dcraw/dcraw.c
--- libkdcraw-0.1.1.orig/libkdcraw/dcraw/dcraw.c	2007-06-13 03:01:45.000000000 +0300
+++ libkdcraw-0.1.1/libkdcraw/dcraw/dcraw.c	2007-06-25 20:21:50.000000000 +0300
@@ -46,7 +46,7 @@
 #include <jpeglib.h>
 #endif
 #ifndef NO_LCMS
-#include <lcms.h>
+#include <lcms/lcms.h>
 #endif
 #ifdef LOCALEDIR
 #include <libintl.h>
