diff -ur ktorrent-2.1beta1.orig/apps/ktorrent/ktorrent.cpp ktorrent-2.1beta1/apps/ktorrent/ktorrent.cpp
--- ktorrent-2.1beta1.orig/apps/ktorrent/ktorrent.cpp	2006-11-21 21:08:57.000000000 +0200
+++ ktorrent-2.1beta1/apps/ktorrent/ktorrent.cpp	2006-11-23 09:56:12.000000000 +0200
@@ -444,23 +444,23 @@
 	KAction* pref = KStdAction::preferences(this, SLOT(optionsPreferences()), actionCollection());
 
 	m_start = new KAction(
-			i18n("to start", "Start"), "ktstart",0,this, SLOT(startDownload()),
+			i18n("to start", "Start"), "1rightarrow",0,this, SLOT(startDownload()),
 			actionCollection(), "Start");
 
 	m_stop = new KAction(
-			i18n("to stop", "Stop"), "ktstop",0,this, SLOT(stopDownload()),
+			i18n("to stop", "Stop"), "stop",0,this, SLOT(stopDownload()),
 			actionCollection(), "Stop");
 
 	m_remove = new KAction(
-			i18n("Remove"), "ktremove",0,this, SLOT(removeDownload()),
+			i18n("Remove"), "edittrash",0,this, SLOT(removeDownload()),
 			actionCollection(), "Remove");
 	
 	m_startall = new KAction(
-			i18n("to start all", "Start All"), "ktstart_all",0,this, SLOT(startAllDownloads()),
+			i18n("to start all", "Start All"), "2rightarrow",0,this, SLOT(startAllDownloads()),
 			actionCollection(), "Start all");
 	
 	m_stopall = new KAction(
-			i18n("to stop all", "Stop All"), "ktstop_all",0,this, SLOT(stopAllDownloads()),
+			i18n("to stop all", "Stop All"), "player_stop",0,this, SLOT(stopAllDownloads()),
 			actionCollection(), "Stop all");
 	
 	m_pasteurl = new KAction(
diff -ur ktorrent-2.1beta1.orig/apps/ktorrent/ktorrentview.cpp ktorrent-2.1beta1/apps/ktorrent/ktorrentview.cpp
--- ktorrent-2.1beta1.orig/apps/ktorrent/ktorrentview.cpp	2006-11-21 21:08:57.000000000 +0200
+++ ktorrent-2.1beta1/apps/ktorrent/ktorrentview.cpp	2006-11-23 09:55:18.000000000 +0200
@@ -142,19 +142,19 @@
 	menu = new KPopupMenu(this);
 	
 	stop_id = menu->insertItem(
-			iload->loadIconSet("ktstop",KIcon::Small),i18n("to stop", "Stop"),
+			iload->loadIconSet("stop",KIcon::Small),i18n("to stop", "Stop"),
 			this,SLOT(stopDownloads()));
 
 	start_id = menu->insertItem(
-			iload->loadIconSet("ktstart",KIcon::Small),i18n("to start", "Start"),
+			iload->loadIconSet("1rightarrow",KIcon::Small),i18n("to start", "Start"),
 			this,SLOT(startDownloads()));
 
 	remove_id = menu->insertItem(
-			iload->loadIconSet("ktremove",KIcon::Small),i18n("Remove Torrent"),
+			iload->loadIconSet("edittrash",KIcon::Small),i18n("Remove Torrent"),
 			this,SLOT(removeDownloads()));
 	
 	remove_all_id = menu->insertItem(
-			iload->loadIconSet("ktremove",KIcon::Small),i18n("Remove Torrent and Data"),
+			iload->loadIconSet("remove",KIcon::Small),i18n("Remove Torrent and Data"),
 			this,SLOT(removeDownloadsAndData()));
 	
 	queue_id = menu->insertItem(
