Index: pisi-1.1/pisi/delta.py
===================================================================
--- pisi-1.1/pisi/delta.py	(revision 14765)
+++ pisi-1.1/pisi/delta.py	(revision 14767)
@@ -110,6 +110,10 @@
     for h in files_delta:
         deltas.extend(hashto_files[h])
 
+    # Directory hashes are None. There was a bug with PolicyKit that should have an empty directory.
+    if hashto_files.has_key(None):
+        deltas.extend(hashto_files[None])
+
     return deltas
 
 def find_relocations(oldfiles, newfiles):
