diff -Naur metaxy-openBibleViewer-e3176fd.orig/openBibleViewer.desktop metaxy-openBibleViewer-e3176fd/openBibleViewer.desktop
--- metaxy-openBibleViewer-e3176fd.orig/openBibleViewer.desktop	2010-04-16 22:49:25.102310280 +0300
+++ metaxy-openBibleViewer-e3176fd/openBibleViewer.desktop	2010-04-17 00:09:13.033309344 +0300
@@ -2,6 +2,7 @@
 Name=openBibleViewer
 Exec=openBibleViewer %u
 GenericName=Bible Study Tool
+GenericName[tr]=İncil Uygulaması
 GenericName[en]=Bible Study Tool
 GenericName[de]=Bibelstudium
 Terminal=false
