diff -Naur pkgconfig.orig/glib-1.2.10/glibconfig-sysdefs.h pkgconfig/glib-1.2.10/glibconfig-sysdefs.h
--- pkgconfig.orig/glib-1.2.10/glibconfig-sysdefs.h	1970-01-01 02:00:00.000000000 +0200
+++ pkgconfig/glib-1.2.10/glibconfig-sysdefs.h	2010-02-11 13:42:40.000000000 +0200
@@ -0,0 +1,6 @@
+#define GLIB_SYSDEF_POLLIN =1
+#define GLIB_SYSDEF_POLLOUT =4
+#define GLIB_SYSDEF_POLLPRI =2
+#define GLIB_SYSDEF_POLLERR =8
+#define GLIB_SYSDEF_POLLHUP =16
+#define GLIB_SYSDEF_POLLNVAL =32
