--- ktranslator/ktranslator.kcfg.orig	2005-10-03 02:45:45.000000000 +0300
+++ ktranslator/ktranslator.kcfg	2006-02-18 21:54:51.000000000 +0200
@@ -95,7 +95,7 @@
     <entry name="SitesToSearch" type="StringList">
       <label>Sites to look for the word in Internet.</label>
       <whatsthis>Sites to look for the word in Internet.</whatsthis>
-      <default>http://en.wikipedia.org/wiki/\\{@},http://en.wiktionary.org/wiki/\\{@}</default>
+      <default>http://www.seslisozluk.com/?word=\\{@}&amp;go_search=Search,http://en.wikipedia.org/wiki/\\{@},http://en.wiktionary.org/wiki/\\{@}</default>
     </entry>
   </group>
 </kcfg>
