--- build.xml	2007-07-05 05:48:19.000000000 +0300
+++ build.xml2	2008-01-23 06:42:52.000000000 +0200
@@ -6,7 +6,7 @@
 <project default="jar" name="commons-cli" basedir=".">
   <property name="defaulttargetdir" value="target">
   </property>
-  <property name="libdir" value="target/lib">
+  <property name="libdir" value=".">
   </property>
   <property name="classesdir" value="target/classes">
   </property>
@@ -40,7 +40,7 @@
     <available property="Junit.present" classname="junit.framework.Test">
     </available>
   </target>
-  <target name="compile" description="o Compile the code" depends="get-deps">
+  <target name="compile" description="o Compile the code">
     <mkdir dir="${classesdir}">
     </mkdir>
     <javac destdir="${classesdir}" deprecation="true" debug="true" optimize="false" excludes="**/package.html">
