Index: tools/Linux/xbmc.desktop
===================================================================
--- tools/Linux/xbmc.desktop.orig
+++ tools/Linux/xbmc.desktop
@@ -1,10 +1,15 @@
 [Desktop Entry]
 Version=1.0
+Encoding=UTF-8
 Name=XBMC Media Center
+Name[tr]=XBMC Medya Merkezi
 GenericName=Media Center
-Comment=Manage and view your media 
+GenericName[tr]=Medya Merkezi
+Comment=Manage and view your media
+Comment[tr]=Medya dosyalarınızı görüntüleyin ve yönetin
 Exec=xbmc
 Icon=xbmc
 Terminal=false
 Type=Application
+StartupNotify=false
 Categories=AudioVideo;Video;Player;TV;
