--- gs-gpl-8.54.dfsg.1.orig/base/gdevxini.c	2006-10-18 03:39:14.000000000 +0900
+++ gs-gpl-8.54.dfsg.1/base/gdevxini.c	2006-10-18 03:40:40.000000000 +0900
@@ -902,7 +902,10 @@
 	dev->is_open = false;
     xdev->IsPageDevice = values.IsPageDevice;
     code = gx_default_put_params(dev, plist);
-    dev->is_open = values.is_open; /* saved value */
+    /* Prevent us from preventing the device closure if the size changed;
+     * that may require us to rebuild the buffer (see x_set_buffer). */
+    if (xdev->width == values.width && xdev->height == values.height)
+      dev->is_open = values.is_open; /* saved value */
     if (code < 0) {		/* Undo setting of .IsPageDevice */
 	xdev->IsPageDevice = save_is_page;
 	return code;
