--- gs-gpl-8.01.orig/base/gdevbbox.c	2004-08-15 13:36:14.000000000 +0900
+++ gs-gpl-8.01/base/gdevbbox.c	2004-08-15 13:36:51.000000000 +0900
@@ -222,7 +222,7 @@
 
     if (tdev != 0)
 	gx_device_copy_params((gx_device *)bdev, tdev);
-    if (remap_colors) {
+    if (remap_colors && bdev->is_open) {
 	bdev->black = gx_device_black((gx_device *)bdev);
 	bdev->white = gx_device_white((gx_device *)bdev);
 	bdev->transparent =
