--- oscripts/menus/ptions.lua.orig	2007-03-01 00:12:04.000000000 +0200
+++ scripts/menus/options.lua	2007-03-06 23:20:55.000000000 +0200
@@ -247,15 +247,16 @@
   end
      
   print(StratagusTranslation)
-  b = AddLanguage("English", "en", 1.5)
+  b = AddLanguage("English", "en", 1.7)
   if StratagusTranslation == "" then
      b:setMarked(true)
   end
-  AddLanguage("Français", "fr", 2.5)
-  AddLanguage("Suomi", "fi", 3.5)
-  AddLanguage("Deutsch", "de", 4.5)
-  AddLanguage("Polski", "pl", 5.5)
-  AddLanguage("Dansk", "da", 6.5)
+  AddLanguage("Türkçe", "tr", 2.7)
+  AddLanguage("Français", "fr", 3.7)
+  AddLanguage("Suomi", "fi", 4.7)
+  AddLanguage("Deutsch", "de", 5.7)
+  AddLanguage("Polski", "pl", 6.7)
+  AddLanguage("Dansk", "da", 7.7)
 
   menu:addButton(_("~!OK"), "o", Video.Width / 2 - 100, Video.Height - 100,
     function() menu:stop() end)
