--- advance/menu/emulator.cc.orig	2009-03-27 15:48:35.000000000 +0200
+++ advance/menu/emulator.cc	2009-03-27 15:49:46.000000000 +0200
@@ -1335,15 +1335,9 @@
 
 	string config_file = slash_add(ref_dir) + "mame.ini";
 #else
-	char* home = getenv("HOME");
-	if (!home || !*home) {
-		target_err("Environment variable HOME not set.\n");
-		return false;
-	}
-
-	string ref_dir = slash_add(home) + ".mame";
+	string ref_dir = "";
 
-	string config_file = path_import(slash_add(ref_dir) + "mame.ini");
+	string config_file = path_import("/etc/sdlmame/mame.ini");
 #endif
 
 	if (!validate_config_file(config_file)) {
