--- kdelibs/khtml/ecma/kjs_css.cpp	2009/02/07 18:21:18	922888
+++ 4.2/kdelibs/khtml/ecma/kjs_css.cpp	2009/03/09 14:38:22	937328
@@ -222,12 +222,12 @@
   CSSStyleDeclarationImpl &styleDecl = *m_impl;
 
   if (propertyName == "cssText") {
-    styleDecl.setCssText(value->toString(exec).domString());
+    styleDecl.setCssText(valueToStringWithNullCheck(exec, value));
   }
   else {
     bool pxSuffix;
     QString prop = cssPropertyName(propertyName, &pxSuffix);
-    QString propvalue = value->toString(exec).qstring();
+    QString propvalue = valueToStringWithNullCheck(exec, value).string();
 
     if (pxSuffix)
       propvalue += QLatin1String("px");
