--- setup/Makefile.in~	2008-07-23 13:16:34.000000000 +0200
+++ setup/Makefile.in	2008-08-06 16:06:09.000000000 +0200
@@ -261,7 +261,7 @@
 CLEANFILES = $(setup_MOC) $(setup_UIC) ./setup_qrc.cpp
 @HAVE_LIBEAY32_TRUE@LIBSSL_LDADD = -leay32
 @HAVE_LIBSSL_TRUE@LIBSSL_LDADD = -lssl -lcrypto
-@BUILD_WIN32_FALSE@PLATFORM_LDADD = ../lib/libitalc_core-static.a
+@BUILD_WIN32_FALSE@PLATFORM_LDADD = -L../lib/ -litalc_core
 @BUILD_WIN32_TRUE@PLATFORM_LDADD = setup_win_resources.o -L../lib/ -litalc_core
 setup_LDADD = $(QT_LDADD) $(QT_LIB_GUI) $(LIBSSL_LDADD) -ljpeg $(PLATFORM_LDADD)
 setup_LDFLAGS = $(LDFLAGS) -mwindows
