diff -Naur qalculate_applet-0.5.1-orig/plasma-applet_qalculate.desktop qalculate_applet-0.5.1/plasma-applet_qalculate.desktop
--- qalculate_applet-0.5.1-orig/plasma-applet_qalculate.desktop	2009-07-23 16:18:25.213596757 +0300
+++ qalculate_applet-0.5.1/plasma-applet_qalculate.desktop	2009-07-23 16:19:15.596571776 +0300
@@ -3,6 +3,7 @@
 Name[x-test]=xxQalculate_appletxx
 Comment=A calculator that uses the powerful Qalculate! library
 Comment[it]=Una calcolatrice che utilizza la libreria Qalculate!
+Comment[tr]=Güçlü Qalculate! kitaplığını kullanan bir hesap makinesi
 Comment[x-test]=xxPlasma Qalculate_appletxx
 Icon=qalculate-applet
 Type=Service
