diff -Naur unixODBC-2.2.11.orig/odbcinstQ/Makefile.in unixODBC-2.2.11/odbcinstQ/Makefile.in
--- unixODBC-2.2.11.orig/odbcinstQ/Makefile.in	2006-01-18 11:23:47.000000000 +0200
+++ unixODBC-2.2.11/odbcinstQ/Makefile.in	2006-01-18 14:39:13.000000000 +0200
@@ -210,19 +210,19 @@
 @QT_TRUE@    -DSYSTEM_FILE_PATH=\"@sysconfdir@\" -DDEFLIB_PATH=\"@libdir@\" $(INCLTDL)
 
 
-@QT_TRUE@libodbcinstQ_la_LIBADD = @X_LDFLAGS@ \
-@QT_TRUE@	@QT_LDFLAGS@ \
-@QT_TRUE@	@USER_LDFLAGS@ \
-@QT_TRUE@	@LIB_QT@ \
-@QT_TRUE@	../odbcinst/libodbcinst.la \
-@QT_TRUE@	../extras/libodbcextraslc.la \
-@QT_TRUE@	../DriverManager/libodbc.la 
+@QT_TRUE at libodbcinstQ_la_LIBADD = @X_LDFLAGS@ \
+@QT_TRUE@   @QT_LDFLAGS@ \
+@QT_TRUE@   @USER_LDFLAGS@ \
+@QT_TRUE@   @LIB_QT@ \
+@QT_TRUE@    ../ini/libinilc.la \
+@QT_TRUE@   ../odbcinst/libodbcinst.la \
+@QT_TRUE@   ../DriverManager/libodbc.la
 
 
-@QT_TRUE@libodbcinstQ_la_DEPENDENCIES = \
-@QT_TRUE@	../odbcinst/libodbcinst.la \
-@QT_TRUE@	../extras/libodbcextraslc.la \
-@QT_TRUE@	../DriverManager/libodbc.la
+@QT_TRUE at libodbcinstQ_la_DEPENDENCIES = \
+@QT_TRUE@    ../ini/libinilc.la \
+@QT_TRUE@   ../odbcinst/libodbcinst.la \
+@QT_TRUE@   ../DriverManager/libodbc.la
 
 
 @QT_TRUE@CLEANFILES = \
