Index: armagetronad-0.2.8.2.1/src/tools/tError.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/tools/tError.cpp	2008-01-27 23:46:55.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/tools/tError.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -26,6 +26,7 @@
 */
 
 #include "config.h"
+#include <cstdlib>
 #include <iostream>
 #if HAVE_UNISTD_H
 #include <unistd.h>
Index: armagetronad-0.2.8.2.1/src/tools/tRandom.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/tools/tRandom.cpp	2008-01-27 23:46:55.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/tools/tRandom.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -34,6 +34,8 @@
 #include    <stdlib.h>
 #endif
 
+#include <cstdlib>
+
 #include    "tError.h"
 #include    "tRecorder.h"
 
Index: armagetronad-0.2.8.2.1/src/tron/gAIBase.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/tron/gAIBase.cpp	2008-01-27 23:46:56.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/tron/gAIBase.cpp	2008-01-28 12:52:44.000000000 -0500
@@ -43,6 +43,7 @@
 #include "tRandom.h"
 #include "tRecorder.h"
 #include <stdlib.h>
+#include <memory>
 
 #define AI_REACTION          0 
 #define AI_EMERGENCY         1 
Index: armagetronad-0.2.8.2.1/src/tron/gCycle.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/tron/gCycle.cpp	2008-01-27 23:46:56.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/tron/gCycle.cpp	2008-01-28 12:53:17.000000000 -0500
@@ -52,6 +52,7 @@
 #include "tMath.h"
 #include <stdlib.h>
 #include <fstream>
+#include <memory>
 
 #ifndef DEDICATED
 #define DONTDOIT
Index: armagetronad-0.2.8.2.1/src/network/md5.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/network/md5.cpp	2008-01-27 23:46:56.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/network/md5.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -78,6 +78,7 @@
 
 #include <string>
 #include <memory>
+#include <cstring>
 
 #ifdef TEST
 /*
Index: armagetronad-0.2.8.2.1/src/network/nKrawall.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/network/nKrawall.cpp	2008-01-27 23:46:56.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/network/nKrawall.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -44,6 +44,7 @@
 
 #include <stdlib.h>
 #include <string>
+#include <cstring>
 
 bool nKrawall::MayRequirePassword(tString& adress, unsigned int port)
 {
Index: armagetronad-0.2.8.2.1/src/network/nNetObject.h
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/network/nNetObject.h	2008-01-27 23:46:56.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/network/nNetObject.h	2008-01-28 12:52:34.000000000 -0500
@@ -31,6 +31,7 @@
 #include "nNetwork.h"
 #include "tArray.h"
 #include "tConsole.h"
+#include <cstring>
 
 class nObserver;
 
Index: armagetronad-0.2.8.2.1/src/network/nNetwork.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/network/nNetwork.cpp	2008-01-27 23:46:56.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/network/nNetwork.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -48,6 +48,7 @@
 #endif
 
 #include <deque>
+#include <cstring>
 
 // debug watchs
 #ifdef DEBUG
Index: armagetronad-0.2.8.2.1/src/network/nSocket.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/network/nSocket.cpp	2008-01-27 23:46:56.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/network/nSocket.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -44,6 +44,7 @@
 #include <errno.h>
 #include <stdlib.h>
 #include <vector>
+#include <cstring>
 
 #ifndef WIN32
 #include <arpa/inet.h> 
Index: armagetronad-0.2.8.2.1/src/render/rFont.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/render/rFont.cpp	2008-01-27 23:46:55.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/render/rFont.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -29,6 +29,7 @@
 #include "rScreen.h"
 #include "tConfiguration.h"
 #include <ctype.h>
+#include <cstring>
 
 #ifndef DEDICATED
 #include "rRender.h"
Index: armagetronad-0.2.8.2.1/src/render/rModel.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/render/rModel.cpp	2008-01-27 23:46:55.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/render/rModel.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -28,6 +28,7 @@
 #include "rModel.h"
 #include <string>
 #include <fstream>
+#include <cstring>
 #include <stdlib.h>
 #include "rScreen.h"
 #include "tString.h"
Index: armagetronad-0.2.8.2.1/src/render/rScreen.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/render/rScreen.cpp	2008-01-27 23:46:55.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/render/rScreen.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -30,6 +30,7 @@
 #include "defs.h"
 
 #include <string>
+#include <cstring>
 #include "rTexture.h"
 #include "rScreen.h"
 #include "rSysdep.h"
Index: armagetronad-0.2.8.2.1/src/tools/tArray.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/tools/tArray.cpp	2008-01-27 23:46:55.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/tools/tArray.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -32,6 +32,7 @@
 #include <iostream>
 #include <stdlib.h>
 #include <string>
+#include <cstring>
 #include "tArray.h"
 
 void GrowingArrayBase::ComplainIfFull(){
Index: armagetronad-0.2.8.2.1/src/tools/tCommandLine.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/tools/tCommandLine.cpp	2008-01-27 23:46:55.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/tools/tCommandLine.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -34,6 +34,7 @@
 #include    "tLocale.h"
 #include    "tConfiguration.h"
 #include    "tException.h"
+#include    <cstring>
 
 #ifdef WIN32
 #include    <windows.h>
Index: armagetronad-0.2.8.2.1/src/tools/tConfiguration.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/tools/tConfiguration.cpp	2008-01-27 23:46:55.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/tools/tConfiguration.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -33,6 +33,7 @@
 #include <stdlib.h>
 #include <ctype.h>
 #include <string>
+#include <cstring>
 #include <sstream>
 #include "tString.h"
 #include "tToDo.h"
Index: armagetronad-0.2.8.2.1/src/tools/tCrypt.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/tools/tCrypt.cpp	2008-01-27 23:46:55.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/tools/tCrypt.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -29,6 +29,7 @@
 #include "tMemManager.h"
 
 #include <string>
+#include <cstring>
 
 class tCryptDummy: public tCrypt
 {
Index: armagetronad-0.2.8.2.1/src/tools/tDirectories.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/tools/tDirectories.cpp	2008-01-27 23:46:55.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/tools/tDirectories.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -32,6 +32,7 @@
 #include <sys/types.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <cstring>
 
 #include "tLocale.h"
 #include "tDirectories.h"
Index: armagetronad-0.2.8.2.1/src/tools/tLocale.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/tools/tLocale.cpp	2008-01-27 23:46:55.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/tools/tLocale.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -33,6 +33,7 @@
 
 #include <fstream>
 #include <string>
+#include <cstring>
 #include <map>
 
 class tLocaleSubItem; // identifies a single string in a single language
Index: armagetronad-0.2.8.2.1/src/tools/tMemManager.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/tools/tMemManager.cpp	2008-01-27 23:46:55.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/tools/tMemManager.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -33,6 +33,7 @@
 #include <iostream>
 #include <sstream>
 #include <stdio.h>  // need basic C IO since STL IO does memory management
+#include <cstring>
 #include "tMemManager.h"
 #include "tError.h"
 
Index: armagetronad-0.2.8.2.1/src/tools/tString.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/tools/tString.cpp	2008-01-27 23:46:55.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/tools/tString.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -31,6 +31,7 @@
 #include "tConfiguration.h"
 #include <ctype.h>
 #include <string>
+#include <cstring>
 #include <iostream>
 
 tString::tString(){
Index: armagetronad-0.2.8.2.1/src/ui/uInput.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/ui/uInput.cpp	2008-01-27 23:46:55.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/ui/uInput.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -33,6 +33,7 @@
 #include "rConsole.h"
 #include "uMenu.h"
 #include "tSysTime.h"
+#include <cstring>
 
 bool su_mouseGrab = false;
 
Index: armagetronad-0.2.8.2.1/src/ui/uInputQueue.cpp
===================================================================
--- armagetronad-0.2.8.2.1.orig/src/ui/uInputQueue.cpp	2008-01-27 23:46:55.000000000 -0500
+++ armagetronad-0.2.8.2.1/src/ui/uInputQueue.cpp	2008-01-28 12:52:34.000000000 -0500
@@ -29,6 +29,7 @@
 #include "rScreen.h"
 #include "tConfiguration.h"
 #include <iostream>
+#include <cstring>
 
 #ifndef DEDICATED
 #include "rSDL.h"
