--- configure.in.orig	2010-07-15 15:11:15.726629843 +0300
+++ configure.in	2010-07-15 15:11:35.249917506 +0300
@@ -17,7 +17,7 @@
 
 # Check for wx
 AM_OPTIONS_WXCONFIG
-AM_PATH_WXCONFIG(2.6.0, [wxWin=1], [wxWin=0], [], [--unicode=no])
+AM_PATH_WXCONFIG(2.6.0, [wxWin=1], [wxWin=0], [], [--unicode=yes])
      if test "$wxWin" != 1; then
         AC_MSG_ERROR([
                 wxWidgets must be installed on your system
@@ -29,7 +29,7 @@
                 equivalent variable and wxWidgets version is 2.6.0 or above.
         ])
      fi
-WX_INCLUDES="`wx-config --unicode=no --cxxflags | awk '{print $1 \"  \" $2}'`"
+WX_INCLUDES="`wx-config --unicode=yes --cxxflags | awk '{print $1 \"  \" $2}'`"
 
 # Checks for header files.
 AC_HEADER_STDC
--- configure.orig	2010-07-15 15:12:29.403885261 +0300
+++ configure	2010-07-15 15:12:49.025881003 +0300
@@ -4942,15 +4942,15 @@
     WX_VERSION=""
 
     min_wx_version=2.6.0
-    if test -z "--unicode=no" ; then
+    if test -z "--unicode=yes" ; then
       echo "$as_me:$LINENO: checking for wxWidgets version >= $min_wx_version" >&5
 echo $ECHO_N "checking for wxWidgets version >= $min_wx_version... $ECHO_C" >&6
     else
-      echo "$as_me:$LINENO: checking for wxWidgets version >= $min_wx_version (--unicode=no)" >&5
-echo $ECHO_N "checking for wxWidgets version >= $min_wx_version (--unicode=no)... $ECHO_C" >&6
+      echo "$as_me:$LINENO: checking for wxWidgets version >= $min_wx_version (--unicode=yes)" >&5
+echo $ECHO_N "checking for wxWidgets version >= $min_wx_version (--unicode=yes)... $ECHO_C" >&6
     fi
 
-    WX_CONFIG_WITH_ARGS="$WX_CONFIG_PATH $wx_config_args --unicode=no "
+    WX_CONFIG_WITH_ARGS="$WX_CONFIG_PATH $wx_config_args --unicode=yes "
 
     WX_VERSION=`$WX_CONFIG_WITH_ARGS --version 2>/dev/null`
     wx_config_major_version=`echo $WX_VERSION | \
@@ -5099,7 +5099,7 @@
         " >&2;}
    { (exit 1); exit 1; }; }
      fi
-WX_INCLUDES="`wx-config --unicode=no --cxxflags | awk '{print $1 \"  \" $2}'`"
+WX_INCLUDES="`wx-config --unicode=yes --cxxflags | awk '{print $1 \"  \" $2}'`"
 
 # Checks for header files.
 
