--- src/gtk/button.cpp	2006-05-01 06:38:40.000000000 +0200
+++ src/gtk/button.cpp	2007-09-22 21:41:32.000000000 +0200
@@ -89,7 +89,7 @@
             right_border += default_border->right;
             top_border += default_border->top;
             bottom_border += default_border->bottom;
-            g_free( default_border );
+            gtk_border_free( default_border);
         }
 #else
         left_border = 6;
--- src/gtk/window.cpp	2006-05-01 06:38:40.000000000 +0200
+++ src/gtk/window.cpp	2007-09-22 21:41:32.000000000 +0200
@@ -3276,7 +3276,7 @@
                 right_border += default_border->right;
                 top_border += default_border->top;
                 bottom_border += default_border->bottom;
-                g_free( default_border );
+                gtk_border_free(default_border);
             }
 #else
             left_border = 6;
