diff -Naur firefoxnotify-1.5.4-orig/src/chrome/content/download_complete_notify.py firefoxnotify-1.5.4/src/chrome/content/download_complete_notify.py
--- firefoxnotify-1.5.4-orig/src/chrome/content/download_complete_notify.py	2009-06-08 10:23:25.536105859 +0300
+++ firefoxnotify-1.5.4/src/chrome/content/download_complete_notify.py	2009-06-08 10:36:44.935105404 +0300
@@ -38,7 +38,7 @@
 
 OPEN_COMMAND = "xdg-open"
 SUMMARY = _("Download Complete")
-BODY = _('%(title)s')
+BODY = _('"%(title)s" file is successfully downloaded to "%(location)s".')
 
 
 logging.basicConfig(level=logging.DEBUG)
@@ -101,6 +101,7 @@
                                       body,
                                       get_icon(),
                                       )
+        self.notif.set_timeout(10000)
         self.notif.connect('closed', self._cleanup)
         self.notif.set_hint_string("category", "transfer.complete")
         # Note: This won't work until we get the pynotify instance to be
