--- python/Makefile.in.orig	2009-01-07 06:24:26.000000000 +0200
+++ python/Makefile.in	2009-01-07 06:26:15.000000000 +0200
@@ -441,13 +441,13 @@
 	    $(pygsfPYTHON_INSTALL) "$$b$$p" "$(DESTDIR)$(pygsfdir)/$$f"; \
 	  else :; fi; \
 	done; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pygsfdir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pygsfdir)" $$dlist; \
-	  fi; \
-	else :; fi
+#	if test -n "$$dlist"; then \
+#	  if test -z "$(DESTDIR)"; then \
+#	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pygsfdir)" $$dlist; \
+#	  else \
+#	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pygsfdir)" $$dlist; \
+#	  fi; \
+#	else :; fi
 
 uninstall-pygsfPYTHON:
 	@$(NORMAL_UNINSTALL)
