--- nepomuk/common/strigiconfigfile.cpp.orig	2008-07-06 14:22:28.000000000 +0300
+++ nepomuk/common/strigiconfigfile.cpp	2008-07-06 14:22:44.000000000 +0300
@@ -287,7 +287,7 @@
         repo.setIndexDir( QDir::homePath() + "/.strigi/clucene" );
 #endif
         repo.addIndexedDirectory( QDir::homePath() );
-        repo.addIndexedDirectory( QDir::homePath() + "/.kde" );
+        repo.addIndexedDirectory( QDir::homePath() + "/.kde4" );
         addRepository( repo );
 
         // in case there are no repositories and no filters we also create
