--- yakuake/src/main_window.h~	2006-07-21 15:55:52.000000000 +0300
+++ yakuake/src/main_window.h	2006-07-21 15:57:39.000000000 +0300
@@ -254,7 +254,7 @@
 
     /*
     ** Shows a passive popup with the given text */
-    void    showPopup(const QString & text, int time = 5000);
+    void    showPopup(const QString & text, int time = 3*1000);
 
 
     /*
--- yakuake/src/main_window.cpp~	2006-07-21 15:55:52.000000000 +0300
+++ yakuake/src/main_window.cpp	2006-07-21 15:59:15.000000000 +0300
@@ -131,6 +131,7 @@
 
     // Displays a popup window ----------------------------
 
+    connect(&popup,SIGNAL(clicked()),&popup,SLOT(hide()));
     showPopup(i18n("Application successfully started!\nPress %1 to use it...").arg(global_key->shortcut("AccessKey").toString()));
 }
 
