diff -up bzflag-2.0.10/configure.ac.plugins bzflag-2.0.10/configure.ac
--- bzflag-2.0.10/configure.ac.plugins	2007-11-15 23:11:36.000000000 +0100
+++ bzflag-2.0.10/configure.ac	2007-12-20 11:10:41.000000000 +0100
@@ -119,6 +119,10 @@ AC_DISABLE_SHARED
 
 if test x$enable_shared != xno ; then
    AC_DEFINE(_USE_BZ_API, 1, [Enable plugins])
+   bzplugindir="$libdir/bzflag/plugins"
+   AC_SUBST(bzplugindir)
+   CFLAGS="$CFLAGS -DBZ_PLUGIN_DIR=\\\"$bzplugindir/\\\""
+   CXXFLAGS="$CXXFLAGS -DBZ_PLUGIN_DIR=\\\"$bzplugindir/\\\""
 fi
 AM_CONDITIONAL(BUILD_PLUGINS, test "x$enable_shared" != "xno")
 
diff -up bzflag-2.0.10/src/bzfs/bzfsPlugins.cxx.plugins bzflag-2.0.10/src/bzfs/bzfsPlugins.cxx
--- bzflag-2.0.10/src/bzfs/bzfsPlugins.cxx.plugins	2007-11-15 23:10:56.000000000 +0100
+++ bzflag-2.0.10/src/bzfs/bzfsPlugins.cxx	2007-12-20 11:10:57.000000000 +0100
@@ -31,7 +31,7 @@ std::string extension = ".dll";
 std::string globalPluginDir = ".\\plugins\\";
 #else
 std::string extension = ".so";
-std::string globalPluginDir = "$(prefix)/lib/bzflag/";
+std::string globalPluginDir = BZ_PLUGIN_DIR;
 #endif
 
 typedef std::map<std::string, bz_APIPluginHandler*> tmCustomPluginMap;
diff -up bzflag-2.0.10/plugins/chathistory/Makefile.am.plugins bzflag-2.0.10/plugins/chathistory/Makefile.am
--- bzflag-2.0.10/plugins/chathistory/Makefile.am.plugins	2007-11-15 23:11:30.000000000 +0100
+++ bzflag-2.0.10/plugins/chathistory/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = chathistory.la
+bzplugin_LTLIBRARIES = chathistory.la
 
 chathistory_la_SOURCES = chathistory.cpp
 chathistory_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/regFlag/Makefile.am.plugins bzflag-2.0.10/plugins/regFlag/Makefile.am
--- bzflag-2.0.10/plugins/regFlag/Makefile.am.plugins	2007-11-15 23:11:31.000000000 +0100
+++ bzflag-2.0.10/plugins/regFlag/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = regFlag.la
+bzplugin_LTLIBRARIES = regFlag.la
 
 regFlag_la_SOURCES = regFlag.cpp
 regFlag_la_CXXFLAGS= -I$(top_srcdir)/include -I$(top_srcdir)/plugins/plugin_utils
diff -up bzflag-2.0.10/plugins/wwzones/Makefile.am.plugins bzflag-2.0.10/plugins/wwzones/Makefile.am
--- bzflag-2.0.10/plugins/wwzones/Makefile.am.plugins	2007-11-15 23:11:31.000000000 +0100
+++ bzflag-2.0.10/plugins/wwzones/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = wwzones.la
+bzplugin_LTLIBRARIES = wwzones.la
 
 wwzones_la_SOURCES = wwzones.cpp
 wwzones_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/rabbitTimer/Makefile.am.plugins bzflag-2.0.10/plugins/rabbitTimer/Makefile.am
--- bzflag-2.0.10/plugins/rabbitTimer/Makefile.am.plugins	2007-11-15 23:11:30.000000000 +0100
+++ bzflag-2.0.10/plugins/rabbitTimer/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = rabbitTimer.la
+bzplugin_LTLIBRARIES = rabbitTimer.la
 
 rabbitTimer_la_SOURCES = rabbitTimer.cpp
 rabbitTimer_la_CXXFLAGS= -I$(top_srcdir)/include -I$(top_srcdir)/plugins/plugin_utils
diff -up bzflag-2.0.10/plugins/RogueGenocide/Makefile.am.plugins bzflag-2.0.10/plugins/RogueGenocide/Makefile.am
--- bzflag-2.0.10/plugins/RogueGenocide/Makefile.am.plugins	2007-11-15 23:11:31.000000000 +0100
+++ bzflag-2.0.10/plugins/RogueGenocide/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = RogueGenocide.la
+bzplugin_LTLIBRARIES = RogueGenocide.la
 
 RogueGenocide_la_SOURCES = RogueGenocide.cpp
 RogueGenocide_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/logDetail/Makefile.am.plugins bzflag-2.0.10/plugins/logDetail/Makefile.am
--- bzflag-2.0.10/plugins/logDetail/Makefile.am.plugins	2007-11-15 23:11:30.000000000 +0100
+++ bzflag-2.0.10/plugins/logDetail/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = logDetail.la
+bzplugin_LTLIBRARIES = logDetail.la
 
 logDetail_la_SOURCES = logDetail.cpp
 logDetail_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/killall/Makefile.am.plugins bzflag-2.0.10/plugins/killall/Makefile.am
--- bzflag-2.0.10/plugins/killall/Makefile.am.plugins	2007-11-15 23:11:30.000000000 +0100
+++ bzflag-2.0.10/plugins/killall/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = killall.la
+bzplugin_LTLIBRARIES = killall.la
 
 killall_la_SOURCES = killall.cpp
 killall_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/rabidRabbit/Makefile.am.plugins bzflag-2.0.10/plugins/rabidRabbit/Makefile.am
--- bzflag-2.0.10/plugins/rabidRabbit/Makefile.am.plugins	2007-11-15 23:11:31.000000000 +0100
+++ bzflag-2.0.10/plugins/rabidRabbit/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = rabidRabbit.la
+bzplugin_LTLIBRARIES = rabidRabbit.la
 
 rabidRabbit_la_SOURCES = rabidRabbit.cpp
 rabidRabbit_la_CXXFLAGS= -I$(top_srcdir)/include -I$(top_srcdir)/plugins/plugin_utils
diff -up bzflag-2.0.10/plugins/airspawn/Makefile.am.plugins bzflag-2.0.10/plugins/airspawn/Makefile.am
--- bzflag-2.0.10/plugins/airspawn/Makefile.am.plugins	2007-11-15 23:11:31.000000000 +0100
+++ bzflag-2.0.10/plugins/airspawn/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = airspawn.la
+bzplugin_LTLIBRARIES = airspawn.la
 
 airspawn_la_SOURCES = airspawn.cpp
 airspawn_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/vocaliser/Makefile.am.plugins bzflag-2.0.10/plugins/vocaliser/Makefile.am
--- bzflag-2.0.10/plugins/vocaliser/Makefile.am.plugins	2007-11-15 23:11:30.000000000 +0100
+++ bzflag-2.0.10/plugins/vocaliser/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = vocaliser.la
+bzplugin_LTLIBRARIES = vocaliser.la
 
 vocaliser_la_SOURCES = vocaliser.cpp
 vocaliser_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/hiddenAdmin/Makefile.am.plugins bzflag-2.0.10/plugins/hiddenAdmin/Makefile.am
--- bzflag-2.0.10/plugins/hiddenAdmin/Makefile.am.plugins	2007-11-15 23:11:30.000000000 +0100
+++ bzflag-2.0.10/plugins/hiddenAdmin/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = hiddenAdmin.la
+bzplugin_LTLIBRARIES = hiddenAdmin.la
 
 hiddenAdmin_la_SOURCES = hiddenAdmin.cpp
 hiddenAdmin_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/SAMPLE_PLUGIN/Makefile.am.plugins bzflag-2.0.10/plugins/SAMPLE_PLUGIN/Makefile.am
--- bzflag-2.0.10/plugins/SAMPLE_PLUGIN/Makefile.am.plugins	2007-11-15 23:11:31.000000000 +0100
+++ bzflag-2.0.10/plugins/SAMPLE_PLUGIN/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = SAMPLE_PLUGIN.la
+bzplugin_LTLIBRARIES = SAMPLE_PLUGIN.la
 
 SAMPLE_PLUGIN_la_SOURCES = SAMPLE_PLUGIN.cpp
 SAMPLE_PLUGIN_la_CXXFLAGS= -I$(top_srcdir)/include -I$(top_srcdir)/plugins/plugin_utils
diff -up bzflag-2.0.10/plugins/keepaway/Makefile.am.plugins bzflag-2.0.10/plugins/keepaway/Makefile.am
--- bzflag-2.0.10/plugins/keepaway/Makefile.am.plugins	2007-11-15 23:11:30.000000000 +0100
+++ bzflag-2.0.10/plugins/keepaway/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = keepaway.la
+bzplugin_LTLIBRARIES = keepaway.la
 
 keepaway_la_SOURCES = keepaway.cpp
 keepaway_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/koth/Makefile.am.plugins bzflag-2.0.10/plugins/koth/Makefile.am
--- bzflag-2.0.10/plugins/koth/Makefile.am.plugins	2007-11-15 23:11:30.000000000 +0100
+++ bzflag-2.0.10/plugins/koth/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = koth.la
+bzplugin_LTLIBRARIES = koth.la
 
 koth_la_SOURCES = koth.cpp
 koth_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/HoldTheFlag/Makefile.am.plugins bzflag-2.0.10/plugins/HoldTheFlag/Makefile.am
--- bzflag-2.0.10/plugins/HoldTheFlag/Makefile.am.plugins	2007-11-15 23:11:31.000000000 +0100
+++ bzflag-2.0.10/plugins/HoldTheFlag/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = HoldTheFlag.la
+bzplugin_LTLIBRARIES = HoldTheFlag.la
 
 HoldTheFlag_la_SOURCES = HoldTheFlag.cpp
 HoldTheFlag_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/nagware/Makefile.am.plugins bzflag-2.0.10/plugins/nagware/Makefile.am
--- bzflag-2.0.10/plugins/nagware/Makefile.am.plugins	2007-11-15 23:11:31.000000000 +0100
+++ bzflag-2.0.10/plugins/nagware/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = nagware.la
+bzplugin_LTLIBRARIES = nagware.la
 
 nagware_la_SOURCES = nagware.cpp
 nagware_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/timedctf/Makefile.am.plugins bzflag-2.0.10/plugins/timedctf/Makefile.am
--- bzflag-2.0.10/plugins/timedctf/Makefile.am.plugins	2007-11-15 23:11:31.000000000 +0100
+++ bzflag-2.0.10/plugins/timedctf/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = timedctf.la
+bzplugin_LTLIBRARIES = timedctf.la
 
 timedctf_la_SOURCES = timedctf.cpp
 timedctf_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/Phoenix/Makefile.am.plugins bzflag-2.0.10/plugins/Phoenix/Makefile.am
--- bzflag-2.0.10/plugins/Phoenix/Makefile.am.plugins	2007-11-15 23:11:30.000000000 +0100
+++ bzflag-2.0.10/plugins/Phoenix/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = Phoenix.la
+bzplugin_LTLIBRARIES = Phoenix.la
 
 Phoenix_la_SOURCES = Phoenix.cpp
 Phoenix_la_CXXFLAGS= -I$(top_srcdir)/include -I$(top_srcdir)/plugins/plugin_utils
diff -up bzflag-2.0.10/plugins/serverControl/Makefile.am.plugins bzflag-2.0.10/plugins/serverControl/Makefile.am
--- bzflag-2.0.10/plugins/serverControl/Makefile.am.plugins	2007-11-15 23:11:30.000000000 +0100
+++ bzflag-2.0.10/plugins/serverControl/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = serverControl.la
+bzplugin_LTLIBRARIES = serverControl.la
 
 serverControl_la_SOURCES = serverControl.cpp
 serverControl_la_CXXFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/plugins/plugin_utils
diff -up bzflag-2.0.10/plugins/flagStay/Makefile.am.plugins bzflag-2.0.10/plugins/flagStay/Makefile.am
--- bzflag-2.0.10/plugins/flagStay/Makefile.am.plugins	2007-11-15 23:11:31.000000000 +0100
+++ bzflag-2.0.10/plugins/flagStay/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = flagStay.la
+bzplugin_LTLIBRARIES = flagStay.la
 
 flagStay_la_SOURCES = flagStay.cpp
 flagStay_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/shockwaveDeath/Makefile.am.plugins bzflag-2.0.10/plugins/shockwaveDeath/Makefile.am
--- bzflag-2.0.10/plugins/shockwaveDeath/Makefile.am.plugins	2007-11-15 23:11:31.000000000 +0100
+++ bzflag-2.0.10/plugins/shockwaveDeath/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = shockwaveDeath.la
+bzplugin_LTLIBRARIES = shockwaveDeath.la
 
 shockwaveDeath_la_SOURCES = shockwaveDeath.cpp
 shockwaveDeath_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/playHistoryTracker/Makefile.am.plugins bzflag-2.0.10/plugins/playHistoryTracker/Makefile.am
--- bzflag-2.0.10/plugins/playHistoryTracker/Makefile.am.plugins	2007-11-15 23:11:31.000000000 +0100
+++ bzflag-2.0.10/plugins/playHistoryTracker/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = playHistoryTracker.la
+bzplugin_LTLIBRARIES = playHistoryTracker.la
 
 playHistoryTracker_la_SOURCES = playHistoryTracker.cpp
 playHistoryTracker_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/torBlock/Makefile.am.plugins bzflag-2.0.10/plugins/torBlock/Makefile.am
--- bzflag-2.0.10/plugins/torBlock/Makefile.am.plugins	2007-11-15 23:11:31.000000000 +0100
+++ bzflag-2.0.10/plugins/torBlock/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = torBlock.la
+bzplugin_LTLIBRARIES = torBlock.la
 
 torBlock_la_SOURCES = torBlock.cpp
 torBlock_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/recordmatch/Makefile.am.plugins bzflag-2.0.10/plugins/recordmatch/Makefile.am
--- bzflag-2.0.10/plugins/recordmatch/Makefile.am.plugins	2007-11-15 23:11:30.000000000 +0100
+++ bzflag-2.0.10/plugins/recordmatch/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = recordmatch.la
+bzplugin_LTLIBRARIES = recordmatch.la
 
 recordmatch_la_SOURCES = recordmatch.cpp
 recordmatch_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/teamflagreset/Makefile.am.plugins bzflag-2.0.10/plugins/teamflagreset/Makefile.am
--- bzflag-2.0.10/plugins/teamflagreset/Makefile.am.plugins	2007-11-15 23:11:30.000000000 +0100
+++ bzflag-2.0.10/plugins/teamflagreset/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = teamflagreset.la
+bzplugin_LTLIBRARIES = teamflagreset.la
 
 teamflagreset_la_SOURCES = teamflagreset.cpp
 teamflagreset_la_CPPFLAGS = -I$(top_srcdir)/include
diff -up bzflag-2.0.10/plugins/python/Makefile.am.plugins bzflag-2.0.10/plugins/python/Makefile.am
--- bzflag-2.0.10/plugins/python/Makefile.am.plugins	2007-11-15 23:11:30.000000000 +0100
+++ bzflag-2.0.10/plugins/python/Makefile.am	2007-12-20 11:09:04.000000000 +0100
@@ -1,5 +1,5 @@
 # change EXTRA to lib to enable this plugin
-# lib_LTLIBRARIES = PyBZFlag.la
+# bzplugin_LTLIBRARIES = PyBZFlag.la
 EXTRA_LTLIBRARIES = PyBZFlag.la
 
 PyBZFlag_la_LDFLAGS = -module -avoid-version
