--- ext/dba/config.m4	2006-08-29 19:20:55.000000000 +0200
+++ ext/dba/config.m4	2006-08-29 19:21:22.000000000 +0200
@@ -463,7 +463,7 @@
 
 AC_ARG_WITH(cdb,
 [  --with-cdb[=DIR]          DBA: Include CDB support],[
-  if test "$withval" = "yes" || test "$HAVE_DBA" = "1"; then
+  if test "$withval" = "yes" && test "$HAVE_DBA" = "1"; then
     PHP_DBA_BUILTIN_CDB
   elif test "$withval" != "no"; then
     PHP_DBA_STD_BEGIN
