diff -ur SuiteSparse-old/AMD/Makefile SuiteSparse-2.1.1/AMD/Makefile
--- SuiteSparse-old/AMD/Makefile	2006-09-18 12:31:10.000000000 +0300
+++ SuiteSparse-2.1.1/AMD/Makefile	2006-09-18 12:31:42.000000000 +0300
@@ -10,7 +10,6 @@
 # Do not compile the FORTRAN versions, or MATLAB interface.
 demo:
 	( cd Source ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 
 # Compile all C code, including the C-callable routine and the mexFunctions.
 # Do not compile the FORTRAN versions.
diff -ur SuiteSparse-old/CHOLMOD/Makefile SuiteSparse-2.1.1/CHOLMOD/Makefile
--- SuiteSparse-old/CHOLMOD/Makefile	2006-09-18 12:31:10.000000000 +0300
+++ SuiteSparse-2.1.1/CHOLMOD/Makefile	2006-09-18 12:32:35.000000000 +0300
@@ -12,7 +12,6 @@
 # Compile the C-callable libraries and the Demo programs.
 all:
 	( cd Lib ; $(MAKE) )
-	( cd Demo ; $(MAKE) )
 
 # Compile the C-callable libraries only.
 library:
diff -ur SuiteSparse-old/CSparse/Makefile SuiteSparse-2.1.1/CSparse/Makefile
--- SuiteSparse-old/CSparse/Makefile	2006-09-18 12:31:10.000000000 +0300
+++ SuiteSparse-2.1.1/CSparse/Makefile	2006-09-18 12:33:12.000000000 +0300
@@ -2,7 +2,6 @@
 
 C:
 	( cd Source ; $(MAKE) )
-	( cd Demo ; $(MAKE) )
 
 all: C tcov
 
diff -ur SuiteSparse-old/CXSparse/Makefile SuiteSparse-2.1.1/CXSparse/Makefile
--- SuiteSparse-old/CXSparse/Makefile	2006-09-18 12:31:10.000000000 +0300
+++ SuiteSparse-2.1.1/CXSparse/Makefile	2006-09-18 12:33:06.000000000 +0300
@@ -2,7 +2,6 @@
 
 C:
 	( cd Source ; $(MAKE) )
-	( cd Demo ; $(MAKE) )
 
 all: C tcov
 
diff -ur SuiteSparse-old/KLU/Makefile SuiteSparse-2.1.1/KLU/Makefile
--- SuiteSparse-old/KLU/Makefile	2006-09-18 12:31:10.000000000 +0300
+++ SuiteSparse-2.1.1/KLU/Makefile	2006-09-18 12:32:23.000000000 +0300
@@ -4,7 +4,6 @@
 
 library:
 	( cd Lib ; $(MAKE) )
-	( cd Demo ; $(MAKE) )
 
 clean:
 	( cd Demo ; $(MAKE) clean )
diff -ur SuiteSparse-old/UMFPACK/Makefile SuiteSparse-2.1.1/UMFPACK/Makefile
--- SuiteSparse-old/UMFPACK/Makefile	2006-09-18 12:31:10.000000000 +0300
+++ SuiteSparse-2.1.1/UMFPACK/Makefile	2006-09-18 12:32:09.000000000 +0300
@@ -14,7 +14,7 @@
 	( cd ../AMD ; $(MAKE) )
 	( cd ../AMD/MATLAB ; $(MAKE) )
 	( cd Source ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
+
 	( cd MATLAB ; $(MAKE) )
 	- cat Doc/License
 
@@ -22,7 +22,6 @@
 library:
 	( cd ../AMD ; $(MAKE) )
 	( cd Source ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 	- cat Doc/License
 
 # compile the FORTRAN interface and demo program
--- SuiteSparse-old/CAMD/Makefile	2006-09-18 12:31:10.000000000 +0300
+++ SuiteSparse-2.1.1/CAMD/Makefile	2006-09-18 14:18:24.000000000 +0300
@@ -10,7 +10,6 @@
 # Do not compile the FORTRAN versions, or MATLAB interface.
 demo:
 	( cd Source ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 
 # Compile all C code, including the C-callable routine and the mexFunctions.
 # Do not compile the FORTRAN versions.
