--- SuiteSparse-old/CHOLMOD/Lib/Makefile	2006-09-18 12:31:10.000000000 +0300
+++ SuiteSparse-2.1.1/CHOLMOD/Lib/Makefile	2006-09-18 12:35:35.000000000 +0300
@@ -116,7 +116,7 @@
 $(OBJ): $(INC)
 
 I = -I../../AMD/Include -I../../AMD/Source -I../../COLAMD \
-	-I$(METIS_PATH)/Lib -I../../CCOLAMD -I../../CAMD/Include -I../Include \
+	-I/usr/include/metis -I../../CCOLAMD -I../../CAMD/Include -I../Include \
 	-I../../UFconfig
 
 
--- SuiteSparse-old/Makefile	2006-09-18 12:31:10.000000000 +0300
+++ SuiteSparse-2.1.1/Makefile	2006-09-18 13:31:55.000000000 +0300
@@ -19,7 +19,7 @@
 	( cd CSparse ; $(MAKE) )
 	( cd CXSparse ; $(MAKE) )
 #	( cd LPDASA ; $(MAKE) )
-	( cd metis-4.0 ; $(MAKE) )
+#	( cd metis-4.0 ; $(MAKE) )
 #	( cd PARAKLETE ; $(MAKE) )
 
 library: default
