Index: kdebase/admin/acinclude.m4.in
===================================================================
--- kdebase.orig/admin/acinclude.m4.in
+++ kdebase/admin/acinclude.m4.in
@@ -1635,7 +1635,7 @@ if AC_TRY_EVAL(ac_link) && test -s conft
 else
   echo "configure: failed program was:" >&AC_FD_CC
   cat conftest.$ac_ext >&AC_FD_CC
-  ac_qt_libraries="NO"
+  #ac_qt_libraries="NO"
 fi
 rm -f conftest*
 CXXFLAGS="$ac_cxxflags_safe"
