Index: kdebase/kdm/configure.in.in
===================================================================
--- kdebase.orig/kdm/configure.in.in
+++ kdebase/kdm/configure.in.in
@@ -14,7 +14,7 @@ acimake:
 	@echo "XBINDIR=\"/usr/bin\" XLIBDIR=\"$(LIBDIR)\""
 
 ]EOF
-    if imake -I/usr/lib/X11/config -DTOPDIR=/etc/X11 -DCURDIR=. /etc/X11 >&5 2>&1 && test -f Makefile; then
+    if imake -I/usr/share/X11/config -DTOPDIR=/etc/X11 -DCURDIR=. /etc/X11 >&5 2>&1 && test -f Makefile; then
 	kde_cv_defines_imake=`${MAKE-make} acimake 2> /dev/null | grep -v "^make"`
 	kde_cv_defines_imake_version=$imkv
     else
