Index: kdebase/kdesktop/Makefile.am
===================================================================
--- kdebase.orig/kdesktop/Makefile.am
+++ kdebase/kdesktop/Makefile.am
@@ -2,7 +2,7 @@
 
 INCLUDES = -I$(top_srcdir)/kcontrol/background -I$(top_srcdir)/libkonq -I$(top_srcdir)/kdmlib $(all_includes)
 kdesktop_la_LDFLAGS  = $(all_libraries) -module $(KDE_PLUGIN)
-kdesktop_la_LIBADD   = libkdesktopsettings.la $(top_builddir)/libkonq/libkonq.la $(top_builddir)/kcontrol/background/libbgnd.la $(top_builddir)/kdmlib/libdmctl.la $(LIB_XSS) $(LIB_KUTILS) $(LIB_XCURSOR)
+kdesktop_la_LIBADD   = /usr/kde/3.5/lib/libkutils.so libkdesktopsettings.la $(top_builddir)/libkonq/libkonq.la $(top_builddir)/kcontrol/background/libbgnd.la $(top_builddir)/kdmlib/libdmctl.la $(LIB_XSS) /usr/kde/3.5/lib/libkutils.so $(LIB_KUTILS) $(LIB_XCURSOR)
 
 SUBDIRS = . lock pics patterns programs init kwebdesktop
 
@@ -38,8 +38,8 @@ kcheckrunning_LDFLAGS = $(all_libraries)
 kcheckrunning_LDADD = $(LIB_X11)
 
 kxdglauncher_SOURCES = kxdglauncher.cpp
-kxdglauncher_LDFLAGS = $(all_libraries)
-kxdglauncher_LDADD = $(LIB_KUTILS)
+kxdglauncher_LDFLAGS = /usr/kde/3.5/lib/libkutils.so $(all_libraries)
+kxdglauncher_LDADD = /usr/kde/3.5/lib/libkutils.so $(LIB_KUTILS)
 
 METASOURCES = AUTO
 
