Index: kdebase/kappfinder/Makefile.am
===================================================================
--- kdebase.orig/kappfinder/Makefile.am
+++ kdebase/kappfinder/Makefile.am
@@ -8,7 +8,7 @@ kappfinder_LDFLAGS = $(LIB_KDEUI) $(all_
 noinst_PROGRAMS = kappfinder_install
 kappfinder_install_SOURCES = main_install.cpp common.cpp
 kappfinder_install_LDADD = $(LIB_KDEUI)
-kappfinder_install_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+kappfinder_install_LDFLAGS = $(LIB_KDEUI) /usr/kde/3.5/lib/libkdeui.so $(all_libraries) $(KDE_RPATH)
 
 INCLUDES = $(all_includes)
 METASOURCES = AUTO
