Index: kdebase/kwin/workspace.cpp
===================================================================
--- kdebase.orig/kwin/workspace.cpp
+++ kdebase/kwin/workspace.cpp
@@ -1036,7 +1036,7 @@ void Workspace::loadDesktopSettings()
         groupname.sprintf("Desktops-screen-%d", screen_number);
     KConfigGroupSaver saver(c,groupname);
 
-    int n = c->readNumEntry("Number", 4);
+    int n = c->readNumEntry("Number", 2);
     number_of_desktops = n;
     delete workarea;
     workarea = new QRect[ n + 1 ];
