Index: kdebase/ksplashml/wndmain.cpp
===================================================================
--- kdebase.orig/ksplashml/wndmain.cpp
+++ kdebase/ksplashml/wndmain.cpp
@@ -95,7 +95,7 @@ void KSplash::slotReadProperties( KConfi
   if (mTheme.isEmpty())
   {
     config->setGroup( "KSplash" );
-    mTheme = config->readEntry( "Theme", "Default" );
+    mTheme = config->readEntry( "Theme", "Pardus" );
   }
   loadTheme( mTheme ); // Guaranteed to return a valid theme.
 }
