Index: kdebase/kcontrol/ebrowsing/plugins/ikws/searchproviders/uluzilla.desktop
===================================================================
--- /dev/null
+++ kdebase/kcontrol/ebrowsing/plugins/ikws/searchproviders/uluzilla.desktop
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Encoding=UTF-8
+Keys=ulubug,uluzilla
+Name=Uludağ Bug Database Bug Number Search
+Name[tr]=Uludağ Hata Veritabanı Hata Numarasıyla Arama
+Query=http://bugs.uludag.org.tr/show_bug.cgi?id=\\{@}
+ServiceTypes=SearchProvider
+Type=Service
