Index: kdebase/kdesktop/krootwm.cc
===================================================================
--- kdebase.orig/kdesktop/krootwm.cc
+++ kdebase/kdesktop/krootwm.cc
@@ -92,6 +92,7 @@ KRootWm::KRootWm(KDesktop* _desktop) : T
               m_pDesktop->iconView(), TQT_SLOT( slotNewMenuActivated() ) );
   }
 
+  /* Disabled by Pardus Linux
   if (kapp->authorizeKAction("bookmarks"))
   {
      bookmarks = new KActionMenu( i18n("Bookmarks"), "bookmark", m_actionCollection, "bookmarks" );
@@ -106,7 +107,11 @@ KRootWm::KRootWm(KDesktop* _desktop) : T
      bookmarks = 0;
      bookmarkMenu = 0;
   }
-  
+  */
+
+  bookmarks = 0;
+  bookmarkMenu = 0;
+
   // The windowList and desktop menus can be part of a menubar (Mac style)
   // so we create them here
   desktopMenu = new TQPopupMenu;
@@ -128,13 +133,14 @@ KRootWm::KRootWm(KDesktop* _desktop) : T
         m_actionCollection->insert( action );
   }
 #endif
-
+  /* Disabled by Pardus Linux
   if (kapp->authorize("run_command"))
   {
      new KAction(i18n("Run Command..."), "run", 0, m_pDesktop, TQT_SLOT( slotExecuteCommand() ), m_actionCollection, "exec" );
      new KAction(i18n("Open Terminal Here..." ), "terminal", CTRL+Key_T, this, TQT_SLOT( slotOpenTerminal() ),
 	m_actionCollection, "open_terminal" );
   }
+  */
 
   if (!KGlobal::config()->isImmutable())
   {
@@ -185,6 +191,7 @@ KRootWm::KRootWm(KDesktop* _desktop) : T
      new KAction(i18n("Refresh Desktop"), "desktop", 0, this, TQT_SLOT( slotRefreshDesktop() ),
                  m_actionCollection, "refresh" );
   }
+  /* Disabled by Pardus Linux
   // Icons in sync with kicker
   if (kapp->authorize("lock_screen"))
   {
@@ -207,6 +214,7 @@ KRootWm::KRootWm(KDesktop* _desktop) : T
                       TQT_SLOT( slotLockNNewSession() ), m_actionCollection, "lockNnewsession" );
       }
   }
+  */
 
   initConfig();
 }
