Index: kdebase/kcontrol/kdm/kdm-font.cpp
===================================================================
--- kdebase.orig/kcontrol/kdm/kdm-font.cpp
+++ kdebase/kcontrol/kdm/kdm-font.cpp
@@ -80,7 +80,7 @@ void KDMFontWidget::makeReadOnly()
   stdFontChooser->button()->setEnabled(false);
   failFontChooser->button()->setEnabled(false);
   greetingFontChooser->button()->setEnabled(false);
-  aacb->setEnabled(false);
+  aacb->setEnabled(true);
 }
 
 void KDMFontWidget::configChanged()
@@ -90,9 +90,9 @@ void KDMFontWidget::configChanged()
 
 void KDMFontWidget::set_def()
 {
-  stdFontChooser->setFont(TQFont("Sans Serif", 10));
-  failFontChooser->setFont(TQFont("Sans Serif", 10, TQFont::Bold));
-  greetingFontChooser->setFont(TQFont("Sans Serif", 22));
+  stdFontChooser->setFont(TQFont("DejaVu Sans", 10));
+  failFontChooser->setFont(TQFont("DejaVu Sans", 10, TQFont::Bold));
+  greetingFontChooser->setFont(TQFont("DejaVu Sans", 20));
 }
 
 void KDMFontWidget::save()
@@ -121,7 +121,7 @@ void KDMFontWidget::load()
   font = greetingFontChooser->font();
   greetingFontChooser->setFont(config->readFontEntry("GreetFont",  &font));
 
-  aacb->setChecked(config->readBoolEntry("AntiAliasing"));
+  aacb->setChecked(config->readBoolEntry("AntiAliasing", true));
 }
 
 
