Index: kdebase/kfind/kquery.cpp
===================================================================
--- kdebase.orig/kfind/kquery.cpp
+++ kdebase/kfind/kquery.cpp
@@ -252,7 +252,7 @@ void KQuery::processQuery( KFileItem* fi
        bool foundmeta=false;
        TQString filename = file->url().path();
 
-       if(filename.startsWith("/dev/"))
+       if(filename.startsWith("/dev/") || filename.startsWith("/proc/") || filename.startsWith("/sys/"))
           return;
 
        KFileMetaInfo metadatas(filename);
