Index: kdebase/applnk/Makefile.am
===================================================================
--- kdebase.orig/applnk/Makefile.am
+++ kdebase/applnk/Makefile.am
@@ -1,6 +1,6 @@
 SUBDIRS = compat
 
-xdg_menu_DATA = kde-settings.menu kde-information.menu kde-screensavers.menu
+xdg_menu_DATA = kde-information.menu kde-screensavers.menu
 
 xdg_mergedmenu_DATA = kde-essential.menu
 xdg_mergedmenudir = $(xdg_menudir)/applications-merged
