Index: kdebase/kdm/backend/Makefile.am
===================================================================
--- kdebase.orig/kdm/backend/Makefile.am
+++ kdebase/kdm/backend/Makefile.am
@@ -4,7 +4,7 @@ CPPFLAGS = $(USER_INCLUDES) $(X_INCLUDES
 LDFLAGS = $(USER_LDFLAGS) $(X_LDFLAGS) $(X_RPATH) $(KRB4_RPATH) $(KRB5_RPATH)
 LDADD = $(LIB_X11) -lXau $(LIBXDMCP) $(PASSWDLIBS) $(LIBSHADOW) $(LIBGEN) \
     $(LIB_LIBS) $(KRB4_LIBS) $(KRB5_LIBS) $(DBUS_LIBS) $(LIBSOCKET) $(LIBRESOLV) \
-    $(LIBUCB) $(LIBUTIL) $(LIBPOSIX4)
+    $(LIBUCB) $(LIBUTIL) $(LIBPOSIX4) -laudit
 
 bin_PROGRAMS = kdm
 kdm_SOURCES = \
