Index: kdebase/kdm/kfrontend/genkdmconf.c
===================================================================
--- kdebase.orig/kdm/kfrontend/genkdmconf.c
+++ kdebase/kdm/kfrontend/genkdmconf.c
@@ -675,6 +675,9 @@ static const char def_session2[] =
 "    [ -f $HOME/.profile ] && . $HOME/.profile\n"
 "    ;;\n"
 "esac\n"
+"[ -f /etc/X11/Xmodmap ] && xmodmap /etc/X11/Xmodmap\n"
+"[ -f $HOME/.Xmodmap ] && xmodmap $HOME/.Xmodmap\n"
+"\n"
 "# invoke global X session script\n"
 ". /etc/X11/Xsession\n";
 
