Index: kdebase/kcontrol/kio/kcookiespolicies.cpp
===================================================================
--- kdebase.orig/kcontrol/kio/kcookiespolicies.cpp
+++ kdebase/kcontrol/kio/kcookiespolicies.cpp
@@ -291,7 +291,7 @@ void KCookiesPolicies::load()
   cookiesEnabled( enableCookies );
 
   KCookieAdvice::Value advice = KCookieAdvice::strToAdvice (cfg.readEntry(
-                                               "CookieGlobalAdvice", "Ask"));
+                                               "CookieGlobalAdvice", "Accept"));
   switch (advice)
   {
     case KCookieAdvice::Accept:
