Index: kdebase/kcontrol/fonts/fonts.cpp
===================================================================
--- kdebase.orig/kcontrol/fonts/fonts.cpp
+++ kdebase/kcontrol/fonts/fonts.cpp
@@ -524,11 +524,11 @@ KFonts::KFonts(TQWidget *parent, const c
 
   // Keep in sync with kdelibs/kdecore/kglobalsettings.cpp
 
-  TQFont f0("Sans Serif", 10);
-  TQFont f1("Monospace", 10);
-  TQFont f2("Sans Serif", 10);
-  TQFont f3("Sans Serif", 10, TQFont::Bold);
-  TQFont f4("Sans Serif", 10);
+  TQFont f0("DejaVu Sans", 10);
+  TQFont f1("DejaVu Sans Mono", 10);
+  TQFont f2("DejaVu Sans", 10);
+  TQFont f3("DejaVu Sans", 9, TQFont::Bold);
+  TQFont f4("DejaVu Sans", 10);
 
   f0.setPointSize(10);
   f1.setPointSize(10);
