Index: kdebase/ksplashml/kcmksplash/installer.cpp
===================================================================
--- kdebase.orig/ksplashml/kcmksplash/installer.cpp
+++ kdebase/ksplashml/kcmksplash/installer.cpp
@@ -261,7 +261,7 @@ void SplashInstaller::load( bool useDefa
   KConfig cnf("ksplashrc");
   cnf.setReadDefaults( useDefaults );
   cnf.setGroup("KSplash");
-  TQString curTheme = cnf.readEntry("Theme","Default");
+  TQString curTheme = cnf.readEntry("Theme","Pardus");
   mThemesList->setCurrentItem(findTheme(curTheme));
   emit changed( useDefaults );
 }
