Index: kdebase/kcontrol/konq/desktopbehavior_impl.cpp
===================================================================
--- kdebase.orig/kcontrol/konq/desktopbehavior_impl.cpp
+++ kdebase/kcontrol/konq/desktopbehavior_impl.cpp
@@ -328,7 +328,7 @@ void DesktopBehavior::load( bool useDefa
     g_pConfig->setGroup( "General" );
     vrootBox->setChecked( g_pConfig->readBoolEntry( "SetVRoot", false ) );
     iconsEnabledBox->setChecked( g_pConfig->readBoolEntry( "Enabled", true ) );
-    autoLineupIconsBox->setChecked( g_pConfig->readBoolEntry( "AutoLineUpIcons", false ) );
+    autoLineupIconsBox->setChecked( g_pConfig->readBoolEntry( "AutoLineUpIcons", true ) );
 
     //
     g_pConfig->setGroup( "Mouse Buttons" );
