Index: kdebase/kcontrol/arts/arts.cpp
===================================================================
--- kdebase.orig/kcontrol/arts/arts.cpp
+++ kdebase/kcontrol/arts/arts.cpp
@@ -69,7 +69,7 @@ static bool startArts()
 	config->setGroup("Arts");
 	bool startServer = config->readBoolEntry("StartServer",true);
 	bool startRealtime = config->readBoolEntry("StartRealtime",true);
-	TQString args = config->readEntry("Arguments","-F 10 -S 4096 -s 60 -m artsmessage -c drkonqi -l 3 -f");
+	TQString args = config->readEntry("Arguments","-F 10 -S 4096 -s 3 -m artsmessage -c drkonqi -l 3 -f");
 
 	delete config;
 
@@ -260,7 +260,7 @@ void KArtsModule::load( bool useDefaults
 	networkTransparent->setChecked(config->readBoolEntry("NetworkTransparent",false));
 	fullDuplex->setChecked(config->readBoolEntry("FullDuplex",false));
 	autoSuspend->setChecked(config->readBoolEntry("AutoSuspend",true));
-	suspendTime->setValue(config->readNumEntry("SuspendTime",60));
+	suspendTime->setValue(config->readNumEntry("SuspendTime",3));
 	deviceName->setText(config->readEntry("DeviceName",TQString::null));
 	customDevice->setChecked(!deviceName->text().isEmpty());
 	hardware->addOptions->setText(config->readEntry("AddOptions",TQString::null));
