Index: kdebase/libkonq/konq_settings.cc
===================================================================
--- kdebase.orig/libkonq/konq_settings.cc
+++ kdebase/libkonq/konq_settings.cc
@@ -90,7 +90,7 @@ void KonqFMSettings::init( KConfig * con
 
   d->m_iconTextWidth = config->readNumEntry( "TextWidth", DEFAULT_TEXTWIDTH );
   if ( d->m_iconTextWidth == DEFAULT_TEXTWIDTH )
-    d->m_iconTextWidth = TQFontMetrics(m_standardFont).width("0000000000");
+    d->m_iconTextWidth = TQFontMetrics(m_standardFont).width("0000000000000");
 
   m_iconTextHeight = config->readNumEntry( "TextHeight", 0 );
   if ( m_iconTextHeight == 0 ) {
