Index: kdebase/konqueror/konq_frame.cc
===================================================================
--- kdebase.orig/konqueror/konq_frame.cc
+++ kdebase/konqueror/konq_frame.cc
@@ -99,9 +99,9 @@ KonqFrameStatusBar::KonqFrameStatusBar(
     addWidget( m_progressBar, 0, true /*permanent->right align*/ );
 
 //     // FIXME: This was added by OpenSUSE; someone needs to figure out what it does and how to fix it!
-//     StatusBarNetworkStatusIndicator * indicator = new StatusBarNetworkStatusIndicator( this, "networkstatusindicator" );
-//     addWidget( indicator, 0, false );
-//     indicator->init();
+     StatusBarNetworkStatusIndicator * indicator = new StatusBarNetworkStatusIndicator( this, "networkstatusindicator" );
+     addWidget( indicator, 0, false );
+     indicator->init();
 
     fontChange(TQFont());
     installEventFilter( this );
