Index: kdebase/kwin/kcmkwin/kwinoptions/mouse.cpp
===================================================================
--- kdebase.orig/kwin/kcmkwin/kwinoptions/mouse.cpp
+++ kdebase/kwin/kcmkwin/kwinoptions/mouse.cpp
@@ -530,7 +530,7 @@ const char* KTitleBarActionsConfig::func
 void KTitleBarActionsConfig::load()
 {
   config->setGroup("Windows");
-  setComboText(coTiDbl, config->readEntry("TitlebarDoubleClickCommand","Shade").ascii());
+  setComboText(coTiDbl,config->readEntry("TitlebarDoubleClickCommand","Maximize").ascii());
   for (int t = 0; t < 3; ++t)
     setComboText(coMax[t],config->readEntry(cnf_Max[t], tbl_Max[t]).ascii());
 
Index: kdebase/kwin/options.cpp
===================================================================
--- kdebase.orig/kwin/options.cpp
+++ kdebase/kwin/options.cpp
@@ -128,7 +128,7 @@ unsigned long Options::updateSettings()
     electric_borders = config->readNumEntry("ElectricBorders", 0);
     electric_border_delay = config->readNumEntry("ElectricBorderDelay", 150);
 
-    OpTitlebarDblClick = windowOperation( config->readEntry("TitlebarDoubleClickCommand", "Shade"), true );
+    OpTitlebarDblClick = windowOperation(config->readEntry("TitlebarDoubleClickCommand", "Maximize"), true );
     d->OpMaxButtonLeftClick = windowOperation( config->readEntry("MaximizeButtonLeftClickCommand", "Maximize"), true );
     d->OpMaxButtonMiddleClick = windowOperation( config->readEntry("MaximizeButtonMiddleClickCommand", "Maximize (vertical only)"), true );
     d->OpMaxButtonRightClick = windowOperation( config->readEntry("MaximizeButtonRightClickCommand", "Maximize (horizontal only)"), true );
