Index: kdebase/kdmlib/kgreet_classic.cpp
===================================================================
--- kdebase.orig/kdmlib/kgreet_classic.cpp
+++ kdebase/kdmlib/kgreet_classic.cpp
@@ -26,21 +26,30 @@ Foundation, Inc., 51 Franklin Street, Fi
 #include "themer/kdmthemer.h"
 #include "themer/kdmitem.h"
 
+#include <kglobal.h>
 #include <klocale.h>
 #include <klineedit.h>
-#include <kpassdlg.h>
+//#include <kpassdlg.h>
 #include <kuser.h>
 
 #include <tqregexp.h>
 #include <tqlayout.h>
 #include <tqlabel.h>
 
-class KDMPasswordEdit : public KPasswordEdit {
+class KDMPasswordEdit : public KLineEdit {
 public:
-	KDMPasswordEdit( TQWidget *parent ) : KPasswordEdit( parent, 0 ) {}
-	KDMPasswordEdit( KPasswordEdit::EchoModes echoMode, TQWidget *parent ) : KPasswordEdit( echoMode, parent, 0 ) {}
+    KDMPasswordEdit( TQWidget *parent ) : KLineEdit( parent )
+    {
+        setEchoMode(TQLineEdit::Password);
+    }
+
+    KDMPasswordEdit( TQLineEdit::EchoMode echoMode, TQWidget *parent ) : KLineEdit( parent )
+    {
+        setEchoMode(echoMode);
+    }
+
 protected:
-	virtual void contextMenuEvent( TQContextMenuEvent * ) {}
+    virtual void contextMenuEvent( TQContextMenuEvent * ) {}
 };
 
 static int echoMode;
@@ -106,8 +115,7 @@ KClassicGreeter::KClassicGreeter( KGreet
 		if (echoMode == -1)
 			passwdEdit = new KDMPasswordEdit( parent );
 		else
-			passwdEdit = new KDMPasswordEdit( (KPasswordEdit::EchoModes)echoMode,
-			                                  parent );
+            passwdEdit = new KDMPasswordEdit( (TQLineEdit::EchoMode)echoMode, parent );
 		connect( passwdEdit, TQT_SIGNAL(textChanged( const TQString & )),
 		         TQT_SLOT(slotActivity()) );
 		connect( passwdEdit, TQT_SIGNAL(lostFocus()), TQT_SLOT(slotActivity()) );
@@ -134,8 +142,8 @@ KClassicGreeter::KClassicGreeter( KGreet
 	}
 	if (func != Authenticate) {
 		if (echoMode == -1) {
-			passwd1Edit = new KDMPasswordEdit( (KPasswordEdit::EchoModes)echoMode, parent );
-			passwd2Edit = new KDMPasswordEdit( (KPasswordEdit::EchoModes)echoMode, parent );
+            passwd1Edit = new KDMPasswordEdit( (TQLineEdit::EchoMode)echoMode, parent );
+            passwd2Edit = new KDMPasswordEdit( (TQLineEdit::EchoMode)echoMode, parent );
 		} else {
 			passwd1Edit = new KDMPasswordEdit( parent );
 			passwd2Edit = new KDMPasswordEdit( parent );
@@ -239,16 +247,16 @@ KClassicGreeter::returnData()
 		                          KGreeterPluginHandler::IsUser );
 		break;
 	case 1:
-		handler->gplugReturnText( passwdEdit->password(),
+		handler->gplugReturnText( passwdEdit->text().utf8(),
 		                          KGreeterPluginHandler::IsPassword |
 		                          KGreeterPluginHandler::IsSecret );
 		break;
 	case 2:
-		handler->gplugReturnText( passwd1Edit->password(),
+		handler->gplugReturnText( passwd2Edit->text().utf8(),
 		                          KGreeterPluginHandler::IsSecret );
 		break;
 	default: // case 3:
-		handler->gplugReturnText( passwd2Edit->password(),
+		handler->gplugReturnText( passwd2Edit->text().utf8(),
 		                          KGreeterPluginHandler::IsNewPassword |
 		                          KGreeterPluginHandler::IsSecret );
 		break;
@@ -397,11 +405,11 @@ KClassicGreeter::revive()
 	// assert( !running );
 	setActive2( true );
 	if (authTok) {
-		passwd1Edit->erase();
-		passwd2Edit->erase();
+		passwd1Edit->clear();
+		passwd2Edit->clear();
 		passwd1Edit->setFocus();
 	} else {
-		passwdEdit->erase();
+		passwdEdit->clear();
 		if (loginEdit && loginEdit->isEnabled())
 			passwdEdit->setEnabled( true );
 		else {
@@ -418,7 +426,7 @@ void // virtual
 KClassicGreeter::clear()
 {
 	// assert( !running && !passwd1Edit );
-	passwdEdit->erase();
+	passwdEdit->clear();
 	if (loginEdit) {
 		loginEdit->clear();
 		loginEdit->setFocus();
@@ -477,6 +485,20 @@ static bool init( const TQString &,
                   void *ctx )
 {
 	echoMode = getConf( ctx, "EchoMode", TQVariant( -1 ) ).toInt();
+    switch (echoMode)
+    {
+        case (0):
+        case (1):
+            echoMode = TQLineEdit::Password;
+            break;
+
+        case (2):
+            echoMode = TQLineEdit::NoEcho;
+            break;
+
+        default:
+            echoMode = TQLineEdit::Password;
+    }
 	KGlobal::locale()->insertCatalogue( "kgreet_classic" );
 	return true;
 }
Index: kdebase/kdmlib/kgreet_classic.h
===================================================================
--- kdebase.orig/kdmlib/kgreet_classic.h
+++ kdebase/kdmlib/kgreet_classic.h
@@ -31,7 +31,7 @@ Foundation, Inc., 51 Franklin Street, Fi
 #include <tqobject.h>
 
 class KLineEdit;
-class KPasswordEdit;
+//class KPasswordEdit;
 class KSimpleConfig;
 class TQGridLayout;
 class TQLabel;
@@ -75,7 +75,7 @@ class KClassicGreeter : public TQObject,
 
 	TQLabel *loginLabel, *passwdLabel, *passwd1Label, *passwd2Label;
 	KLineEdit *loginEdit;
-	KPasswordEdit *passwdEdit, *passwd1Edit, *passwd2Edit;
+	KLineEdit *passwdEdit, *passwd1Edit, *passwd2Edit;
 	KSimpleConfig *stsFile;
 	TQString fixedUser, curUser;
 	Function func;
Index: kdebase/kdmlib/kgreet_winbind.cpp
===================================================================
--- kdebase.orig/kdmlib/kgreet_winbind.cpp
+++ kdebase/kdmlib/kgreet_winbind.cpp
@@ -26,11 +26,11 @@ Foundation, Inc., 51 Franklin Street, Fi
 #include "themer/kdmthemer.h"
 #include "themer/kdmitem.h"
 
+#include <kglobal.h>
 #include <klocale.h>
 #include <kdebug.h>
 #include <kcombobox.h>
 #include <klineedit.h>
-#include <kpassdlg.h>
 #include <kuser.h>
 #include <kprocio.h>
 
@@ -40,10 +40,16 @@ Foundation, Inc., 51 Franklin Street, Fi
 
 #include <stdlib.h>
 
-class KDMPasswordEdit : public KPasswordEdit {
+class KDMPasswordEdit : public KLineEdit {
 public:
-	KDMPasswordEdit( TQWidget *parent ) : KPasswordEdit( parent, 0 ) {}
-	KDMPasswordEdit( KPasswordEdit::EchoModes echoMode, TQWidget *parent ) : KPasswordEdit( echoMode, parent, 0 ) {}
+    KDMPasswordEdit( TQWidget *parent ) : KLineEdit( parent )
+    {
+        setEchoMode(TQLineEdit::Password);
+    }
+    KDMPasswordEdit( TQLineEdit::EchoMode echoMode, TQWidget *parent ) : KLineEdit( parent )
+    {
+        setEchoMode(echoMode);
+    }
 protected:
 	virtual void contextMenuEvent( TQContextMenuEvent * ) {}
 };
@@ -149,7 +155,7 @@ KWinbindGreeter::KWinbindGreeter( KGreet
 		if (echoMode == -1)
 			passwdEdit = new KDMPasswordEdit( parent );
 		else
-			passwdEdit = new KDMPasswordEdit( (KPasswordEdit::EchoModes)echoMode,
+			passwdEdit = new KDMPasswordEdit( (QLineEdit::EchoMode)echoMode,
 			                                  parent );
 		connect( passwdEdit, TQT_SIGNAL(textChanged( const TQString & )),
 		         TQT_SLOT(slotActivity()) );
@@ -179,8 +185,8 @@ KWinbindGreeter::KWinbindGreeter( KGreet
 	}
 	if (func != Authenticate) {
 		if (echoMode == -1) {
-			passwd1Edit = new KDMPasswordEdit( (KPasswordEdit::EchoModes)echoMode, parent );
-			passwd2Edit = new KDMPasswordEdit( (KPasswordEdit::EchoModes)echoMode, parent );
+            passwd1Edit = new KDMPasswordEdit( (TQLineEdit::EchoMode)echoMode, parent );
+            passwd2Edit = new KDMPasswordEdit( (TQLineEdit::EchoMode)echoMode, parent );
 		} else {
 			passwd1Edit = new KDMPasswordEdit( parent );
 			passwd2Edit = new KDMPasswordEdit( parent );
@@ -320,16 +326,16 @@ KWinbindGreeter::returnData()
 		                          KGreeterPluginHandler::IsUser );
 		break;
 	case 1:
-		handler->gplugReturnText( passwdEdit->password(),
+		handler->gplugReturnText( passwdEdit->text().utf8(),
 		                          KGreeterPluginHandler::IsPassword |
 		                          KGreeterPluginHandler::IsSecret );
 		break;
 	case 2:
-		handler->gplugReturnText( passwd1Edit->password(),
+		handler->gplugReturnText( passwd1Edit->text().utf8(),
 		                          KGreeterPluginHandler::IsSecret );
 		break;
 	default: // case 3:
-		handler->gplugReturnText( passwd2Edit->password(),
+		handler->gplugReturnText( passwd2Edit->text().utf8(),
 		                          KGreeterPluginHandler::IsNewPassword |
 		                          KGreeterPluginHandler::IsSecret );
 		break;
@@ -478,11 +484,11 @@ KWinbindGreeter::revive()
 	// assert( !running );
 	setActive2( true );
 	if (authTok) {
-		passwd1Edit->erase();
-		passwd2Edit->erase();
+		passwd1Edit->clear();
+		passwd2Edit->clear();
 		passwd1Edit->setFocus();
 	} else {
-		passwdEdit->erase();
+		passwdEdit->clear();
 		if (loginEdit && loginEdit->isEnabled())
 			passwdEdit->setEnabled( true );
 		else {
@@ -499,7 +505,7 @@ void // virtual
 KWinbindGreeter::clear()
 {
 	// assert( !running && !passwd1Edit );
-	passwdEdit->erase();
+	passwdEdit->clear();
 	if (loginEdit) {
 		domainCombo->setCurrentItem( defaultDomain );
 		slotChangedDomain( defaultDomain );
@@ -624,6 +630,20 @@ static bool init( const TQString &,
                   void *ctx )
 {
 	echoMode = getConf( ctx, "EchoMode", TQVariant( -1 ) ).toInt();
+    switch (echoMode)
+    {
+        case (0):
+        case (1):
+            echoMode = QLineEdit::Password;
+            break;
+
+        case (2):
+            echoMode = QLineEdit::NoEcho;
+            break;
+
+        default:
+            echoMode = QLineEdit::Password;
+    }
 	staticDomains = TQStringList::split( ':', getConf( ctx, "winbind.Domains", TQVariant( "" ) ).toString() );
 	if (!staticDomains.contains("<local>"))
 		staticDomains << "<local>";
Index: kdebase/kdmlib/kgreet_winbind.h
===================================================================
--- kdebase.orig/kdmlib/kgreet_winbind.h
+++ kdebase/kdmlib/kgreet_winbind.h
@@ -33,7 +33,6 @@ Foundation, Inc., 51 Franklin Street, Fi
 
 class KComboBox;
 class KLineEdit;
-class KPasswordEdit;
 class KSimpleConfig;
 class TQGridLayout;
 class TQLabel;
@@ -84,7 +83,7 @@ class KWinbindGreeter : public TQObject,
 	TQLabel *domainLabel, *loginLabel, *passwdLabel, *passwd1Label, *passwd2Label;
 	KComboBox *domainCombo;
 	KLineEdit *loginEdit;
-	KPasswordEdit *passwdEdit, *passwd1Edit, *passwd2Edit;
+	KLineEdit *passwdEdit, *passwd1Edit, *passwd2Edit;
 	KSimpleConfig *stsFile;
 	TQString fixedDomain, fixedUser, curUser;
 	TQStringList allUsers, mDomainListing;
