Index: kdebase/kdm/kfrontend/kgreeter.cpp
===================================================================
--- kdebase.orig/kdm/kfrontend/kgreeter.cpp
+++ kdebase/kdm/kfrontend/kgreeter.cpp
@@ -170,8 +170,8 @@ KGreeter::KGreeter( bool framed )
 		userView = new UserListView( framed, this );
 		connect( userView, TQT_SIGNAL(clicked( TQListViewItem * )),
 		         TQT_SLOT(slotUserClicked( TQListViewItem * )) );
-		connect( userView, TQT_SIGNAL(doubleClicked( TQListViewItem * )),
-		         TQT_SLOT(accept()) );
+		// connect( userView, TQT_SIGNAL(doubleClicked( TQListViewItem * )),
+		         // TQT_SLOT(accept()) );
 	}
 	if (_userCompletion)
 		userList = new TQStringList;
