Index: kdebase/kioslave/media/mediamanager/halbackend.cpp
===================================================================
--- kdebase.orig/kioslave/media/mediamanager/halbackend.cpp
+++ kdebase/kioslave/media/mediamanager/halbackend.cpp
@@ -546,7 +546,18 @@ void HALBackend::setVolumeProperties(Med
     char* name = libhal_volume_policy_compute_display_name(halDrive, halVolume, m_halStoragePolicy);
     TQString volume_name = TQString::fromUtf8(name);
     TQString media_name = volume_name;
-    medium->setLabel(media_name);
+    /* media_name contains something like "501M Removable Media" or "Blank CD-R"
+       The former needs special handling for correct translation
+    */
+    if (media_name.find(TQRegExp("^[0-9]+\\.?[0-9]*[KMGT] (Removable )?Media$")) > -1)
+    {
+        TQString pattern = media_name.section(" ", 1);
+        media_name.replace(pattern, i18n(pattern.ascii()));
+        medium->setLabel(media_name);
+    }
+    else
+        medium->setLabel(i18n(media_name.ascii()));
+ 
     free(name);
 
     TQString mimeType;
Index: kdebase/kioslave/media/mediamanager/halbackend.h
===================================================================
--- kdebase.orig/kioslave/media/mediamanager/halbackend.h
+++ kdebase/kioslave/media/mediamanager/halbackend.h
@@ -34,6 +34,7 @@
 #include <tqobject.h>
 #include <tqstringlist.h>
 #include <tqstring.h>
+#include <tqregexp.h>
 
 #include <config.h>
 
