Index: kdebase/ksmserver/shutdowndlg.cpp
===================================================================
--- kdebase.orig/ksmserver/shutdowndlg.cpp
+++ kdebase/ksmserver/shutdowndlg.cpp
@@ -554,6 +554,9 @@ KSMShutdownDlg::KSMShutdownDlg( TQWidget
 			btnReboot->setPopup(targets);
 			connect( targets, TQT_SIGNAL(activated(int)), TQT_SLOT(slotReboot(int)) );
 			}
+            else {
+                TQToolTip::add( btnReboot, i18n( "<qt><h3>Restart Computer</h3><p>Log out of the current session and restart the computer</p></qt>" ) );
+            }
 	
 	
 			if (canSuspend && !disableSuspend)
