Index: kdebase/kioslave/media/mediamanager/halbackend.cpp
===================================================================
--- kdebase.orig/kioslave/media/mediamanager/halbackend.cpp
+++ kdebase/kioslave/media/mediamanager/halbackend.cpp
@@ -520,7 +520,7 @@ void HALBackend::setVolumeProperties(Med
             medium->mountableState(
                 libhal_volume_get_device_file(halVolume),		/* Device node */
                 clearUdiString,
-                libhal_volume_get_mount_point(halClearVolume),		/* Mount point */
+                TQString::fromUtf8(libhal_volume_get_mount_point(halClearVolume)),   /* Mount point */
                 libhal_volume_get_fstype(halClearVolume),		/* Filesystem type */
                 libhal_volume_is_mounted(halClearVolume) );		/* Mounted ? */
         else
@@ -536,7 +536,7 @@ void HALBackend::setVolumeProperties(Med
         kdDebug(1219) << "HALBackend::setVolumeProperties : normal volume" << endl;
         medium->mountableState(
             libhal_volume_get_device_file(halVolume),		/* Device node */
-            libhal_volume_get_mount_point(halVolume),		/* Mount point */
+            TQString::fromUtf8(libhal_volume_get_mount_point(halVolume)),   /* Mount point */
             libhal_volume_get_fstype(halVolume),		/* Filesystem type */
             libhal_volume_is_mounted(halVolume) );		/* Mounted ? */
     }
