Index: kdebase/kioslave/media/mediamanager/halbackend.cpp
===================================================================
--- kdebase.orig/kioslave/media/mediamanager/halbackend.cpp
+++ kdebase/kioslave/media/mediamanager/halbackend.cpp
@@ -981,7 +981,7 @@ TQStringList HALBackend::mountoptions(co
 
     if (valids.contains("shortname"))
     {
-        TQString svalue = config.readEntry("shortname", "lower").lower();
+        TQString svalue = config.readEntry("shortname", "mixed").lower();
         if (svalue == "winnt")
             result << "shortname=winnt";
         else if (svalue == "win95")
