Index: kdebase/kdm/config.def
===================================================================
--- kdebase.orig/kdm/config.def
+++ kdebase/kdm/config.def
@@ -75,7 +75,7 @@
 # define DEF_SERVER_CMD XBINDIR "/X -T -force"
 #elif defined(__linux__) || defined(__GNU__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__)
 /* we just assume that any free *nix installation has a recent xfree86/xorg */
-# define DEF_SERVER_CMD XBINDIR "/X -br"
+# define DEF_SERVER_CMD XBINDIR "/X -nr"
 #else
 # define DEF_SERVER_CMD XBINDIR "/X"
 #endif
