Index: kdebase/kcontrol/kfontinst/viewpart/FontViewPart.cpp
===================================================================
--- kdebase.orig/kcontrol/kfontinst/viewpart/FontViewPart.cpp
+++ kdebase/kcontrol/kfontinst/viewpart/FontViewPart.cpp
@@ -198,7 +198,7 @@ void CFontViewPart::previewStatus(bool s
     itsInstallButton->setShown(st && itsShowInstallButton);
     itsToolsFrame->setShown(itsInstallButton->isShown()||itsFaceSelector->isShown());
     itsChangeTextAction->setEnabled(st);
-    itsPrintAction->setEnabled(st && KFI_KIO_FONTS_PROTOCOL==m_url.protocol());
+    itsPrintAction->setEnabled(st || KFI_KIO_FONTS_PROTOCOL==m_url.protocol());
 }
 
 void CFontViewPart::install()
