Index: kdebase/admin/acinclude.m4.in
===================================================================
--- kdebase.orig/admin/acinclude.m4.in
+++ kdebase/admin/acinclude.m4.in
@@ -3753,7 +3753,7 @@ AC_LIBTOOL_CXX
 # LIBTOOL_SHELL="/bin/sh ./libtool"
 LIBTOOL_SHELL="libtool"
 #  LIBTOOL="$LIBTOOL --silent"
-KDE_PLUGIN="-avoid-version -module -no-undefined \$(KDE_NO_UNDEFINED) \$(KDE_RPATH) \$(KDE_MT_LDFLAGS)"
+KDE_PLUGIN="-avoid-version -module -no-undefined \$(KDE_RPATH) \$(KDE_MT_LDFLAGS)"
 AC_SUBST(KDE_PLUGIN)
 
 # This hack ensures that libtool creates shared libs for kunittest plugins. By default check_LTLIBRARIES makes static libs.
