Index: kdebase/kcontrol/screensaver/scrnsave.cpp
===================================================================
--- kdebase.orig/kcontrol/screensaver/scrnsave.cpp
+++ kdebase/kcontrol/screensaver/scrnsave.cpp
@@ -101,11 +101,13 @@ KScreenSaver::KScreenSaver(TQWidget *par
     // Add KDE specific screensaver path
     TQString relPath="System/ScreenSavers/";
     KServiceGroup::Ptr servGroup = KServiceGroup::baseGroup( "screensavers" );
+#if 0
     if (servGroup)
     {
       relPath=servGroup->relPath();
       kdDebug() << "relPath=" << relPath << endl;
     }
+#endif
 
     KGlobal::dirs()->addResourceType("scrsav",
                                      KGlobal::dirs()->kde_default("apps") +
