Index: kdebase/kcontrol/konq/desktopbehavior_impl.cpp
===================================================================
--- kdebase.orig/kcontrol/konq/desktopbehavior_impl.cpp
+++ kdebase/kcontrol/konq/desktopbehavior_impl.cpp
@@ -254,7 +254,7 @@ void DesktopBehavior::fillMediaListView(
     KMimeType::List mimetypes = KMimeType::allMimeTypes();
     TQValueListIterator<KMimeType::Ptr> it2(mimetypes.begin());
     g_pConfig->setGroup( "Media" );
-    enableMediaBox->setChecked(g_pConfig->readBoolEntry("enabled",false));
+    enableMediaBox->setChecked(g_pConfig->readBoolEntry("enabled",true));
     TQString excludedMedia=g_pConfig->readEntry("exclude","media/hdd_mounted,media/hdd_unmounted,media/floppy_unmounted,media/cdrom_unmounted,media/floppy5_unmounted");
     for (; it2 != mimetypes.end(); ++it2) {
         if ( ((*it2)->name().startsWith("media/")) )
