--- sudo/auth/pam.c	2005/06/25 19:29:17	1.51
+++ sudo/auth/pam.c	2006/09/29 15:53:42	1.52
@@ -249,7 +249,12 @@ sudo_conv(num_msg, msg, response, appdat
 		    p = pm->msg;
 		/* Read the password. */
 		pass = tgetpass(p, def_passwd_timeout * 60, flags);
-		pr->resp = estrdup(pass ? pass : "");
+		if (pass == NULL) {
+		    /* We got ^C instead of a password; abort quickly. */
+		    nil_pw = 1;
+		    return(PAM_CONV_ERR);
+		}
+		pr->resp = estrdup(pass);
 		if (*pr->resp == '\0')
 		    nil_pw = 1;		/* empty password */
 		else
