diff -Nur ode-0.9-old/Makefile.am ode-0.9/Makefile.am
--- ode-0.9-old/Makefile.am	2008-04-21 00:03:20.000000000 +0300
+++ ode-0.9/Makefile.am	2008-04-21 00:04:43.000000000 +0300
@@ -26,12 +26,11 @@
 if USE_SONAME
 install-exec-hook:
 	ln -s $(libdir)/@ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
-	$(libdir)/libode.so
+	$(DESTDIR)$(libdir)/libode.so
 	ln -s $(libdir)/@ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
-	$(libdir)/@ODE_SONAME@
+	$(DESTDIR)$(libdir)/@ODE_SONAME@
 	ln -s $(libdir)/@ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
-	$(libdir)/@ODE_SONAME@.@ODE_REVISION@
-	/sbin/ldconfig
+	$(DESTDIR)$(libdir)/@ODE_SONAME@.@ODE_REVISION@
 else
 install-exec-hook:
 endif
