diff -Naur zekr.orig/zekr.sh zekr/zekr.sh
--- zekr.orig/zekr.sh	2011-02-01 21:52:35.427000057 +0200
+++ zekr/zekr.sh	2011-02-01 21:59:50.952000058 +0200
@@ -6,7 +6,7 @@
 # See http://zekr.org/faq.html#linux for more details.
 
 # change the following line to a proper Mozilla-GTK2 installation directory.
-export MOZILLA_FIVE_HOME=/usr/lib/firefox
+export MOZILLA_FIVE_HOME=/usr/lib/xulrunner-2.0
 
 if [ -z "$MOZILLA_FIVE_HOME" ]; then
 	echo "MOZILLA_FIVE_HOME environment variable is not set. Please set it to a Mozilla GTK2 1.4+ installation directory. See http://zekr.org/faq.html#linux for more info."
