--- gs-gpl-8.01.orig/src/gdevepsn.c	2004-08-15 15:26:08.000000000 +0900
+++ gs-gpl-8.01/src/gdevepsn.c	2004-08-15 15:26:54.000000000 +0900
@@ -105,7 +105,7 @@
   prn_device(prn_std_procs, "epson",
 	DEFAULT_WIDTH_10THS, DEFAULT_HEIGHT_10THS,
 	X_DPI, Y_DPI,
-	0.2, 0.0, 0.0, 0.0,			/* margins */
+	0.25, 0.02, 0.25, 0.4,			/* margins */
 	1, epson_print_page);
 
 /* Mid-res (interleaved, 1 pass per line) 9-pin device */
