Index: xkeyboard-config-1.2/symbols/pc
===================================================================
--- xkeyboard-config-1.2.orig/symbols/pc
+++ xkeyboard-config-1.2/symbols/pc
@@ -66,10 +66,37 @@ xkb_symbols "pc105" {
     key <HYPR> {	[ NoSymbol, Hyper_L	]	};
     modifier_map Mod4   { <HYPR> };
 
+    key  <I04> { [   XF86LightBulb ] };
+    key  <I10> { [   XF86AudioPrev ] };
+    key  <I19> { [   XF86AudioNext ] };
+    key  <I1F> { [     XF86LaunchA ] };
+    key  <I20> { [   XF86AudioMute ] };
+    key  <I21> { [  XF86Calculator ] };
+    key  <I22> { [  XF86AudioPause ] };
+    key  <I24> { [   XF86AudioStop ] };
+    key  <I25> { [       XF86Sleep ] };
+    key  <I2E> { [ XF86AudioLowerVolume ] };
+    key  <I30> { [ XF86AudioRaiseVolume ] };
+    key  <I32> { [         XF86WWW ] };
+    key  <K6C> { [       XF86Eject ] };
+    key  <K6D> { [     XF86LaunchC ] };
+    key  <I54> { [     XF86LaunchE ] };
     key <OUTP> { [ XF86Display ] };
     key <KITG> { [ XF86KbdLightOnOff ] };
     key <KIDN> { [ XF86KbdBrightnessDown ] };
     key <KIUP> { [ XF86KbdBrightnessUp ] };
+    key  <I5F> { [     XF86Standby ] };
+    key  <I65> { [      XF86Search ] };
+    key  <I66> { [   XF86Favorites ] };
+    key  <I67> { [     XF86Refresh ] };
+    key  <I68> { [        XF86Stop ] };
+    key  <I69> { [     XF86Forward ] };
+    key  <I6A> { [        XF86Back ] };
+    key  <I6B> { [  XF86MyComputer ] };
+    key  <I6C> { [        XF86Mail ] };
+    key  <I6D> { [  XF86AudioMedia ] };
+    key  <I71> { [     XF86LaunchB ] };
+    key  <I75> { [     XF86Launch0 ] };
 };
 
 partial hidden alphanumeric_keys
