diff -ur SuiteSparse-old/CSparse/Source/Makefile SuiteSparse-2.1.1/CSparse/Source/Makefile
--- SuiteSparse-old/CSparse/Source/Makefile	2006-09-18 12:31:10.000000000 +0300
+++ SuiteSparse-2.1.1/CSparse/Source/Makefile	2006-09-18 13:10:41.000000000 +0300
@@ -1,6 +1,6 @@
 # Modify the "-O" optimization option for best performance (-O3 on Linux):
 CC = cc
-CFLAGS = -O
+CFLAGS = $(SYSTEM_CFLAGS)
 
 AR = ar cr
 RANLIB = ranlib
diff -ur SuiteSparse-old/CXSparse/Source/Makefile SuiteSparse-2.1.1/CXSparse/Source/Makefile
--- SuiteSparse-old/CXSparse/Source/Makefile	2006-09-18 12:31:10.000000000 +0300
+++ SuiteSparse-2.1.1/CXSparse/Source/Makefile	2006-09-18 13:20:27.000000000 +0300
@@ -1,6 +1,6 @@
 # Modify the "-O" optimization option for best performance (-O3 on Linux):
 CC = cc
-CFLAGS = -O
+CFLAGS = $(SYSTEM_CFLAGS)
 I = -I../../UFconfig
 
 AR = ar cr
diff -ur SuiteSparse-old/UFconfig/UFconfig.mk SuiteSparse-2.1.1/UFconfig/UFconfig.mk
--- SuiteSparse-old/UFconfig/UFconfig.mk	2006-09-18 12:31:10.000000000 +0300
+++ SuiteSparse-2.1.1/UFconfig/UFconfig.mk	2006-09-18 13:10:05.000000000 +0300
@@ -32,7 +32,7 @@
 # performance.  You should select the optimization parameters that are best
 # for your system.  On Linux, use "CFLAGS = -O3 -fexceptions" for example.
 CC = cc
-CFLAGS = -O
+CFLAGS = $(SYSTEM_CFLAGS)
 
 # ranlib, and ar, for generating libraries
 RANLIB = ranlib
