--- qgit/src/src.pro.qmake	2008-06-07 16:37:41.000000000 +0200
+++ qgit/src/src.pro	2008-07-17 14:53:54.000000000 +0200
@@ -38,13 +38,13 @@
 
 unix {
     TARGET = qgit
-    target.path = ~/bin
+    target.path = /usr/bin
     CONFIG += x11
 }
 
 HAVE_GCC {
-	QMAKE_CXXFLAGS_RELEASE += -s -O2 -Wno-non-virtual-dtor -Wno-long-long -pedantic -Wconversion
-	QMAKE_CXXFLAGS_DEBUG += -g3 -ggdb -O0 -Wno-non-virtual-dtor -Wno-long-long -pedantic -Wconversion
+	QMAKE_CXXFLAGS_RELEASE +=
+	QMAKE_CXXFLAGS_DEBUG +=
 }
 
 ENABLE_CONSOLE_MSG {
