diff -Nur hplip-1.7.1-old/prnt/hpijs/ljcolor.cpp hplip-1.7.1/prnt/hpijs/ljcolor.cpp
--- hplip-1.7.1-old/prnt/hpijs/ljcolor.cpp	2007-02-05 21:26:37.374883422 +0200
+++ hplip-1.7.1/prnt/hpijs/ljcolor.cpp	2007-02-05 21:28:19.790313910 +0200
@@ -272,7 +272,7 @@
 DRIVER_ERROR HeaderLJColor::StartSend ()
 {
     DRIVER_ERROR err;
-    char    res[64];
+    char    res[67];
     int     iRes;
 
     iRes = thePrintContext->EffectiveResolutionY ();
