Index: xorg-video-7.2/xf86-video-sis-0.10.1/src/sis_vb.c
===================================================================
--- xorg-video-7.2.orig/xf86-video-sis-0.10.1/src/sis_vb.c
+++ xorg-video-7.2/xf86-video-sis-0.10.1/src/sis_vb.c
@@ -405,7 +405,7 @@ void SISLCDPreInit(ScrnInfoPtr pScrn, Bo
 	  pSiS->LCDheight = 1024;
 	  if(CR37 & 0x10) pSiS->VBLCDFlags |= VB_LCD_EXPANDING;
 	  xf86DrvMsg(pScrn->scrnIndex, X_PROBED,
-		"Detected LCD panel (%dx%d, type %d, %sexpanding, RGB%d)\n",
+		"Detected LCD/plasma panel (%dx%d, type %d, %sexpanding, RGB%d)\n",
 		pSiS->LCDwidth, pSiS->LCDheight,
 		((CR36 & 0xf0) >> 4),
 		(CR37 & 0x10) ? "" : "non-",
@@ -464,7 +464,7 @@ void SISLCDPreInit(ScrnInfoPtr pScrn, Bo
 	     pSiS->LCDwidth = pSiS->SiS_Pr->CP_MaxX;
 	     if(CR37 & 0x10) pSiS->VBLCDFlags |= VB_LCD_EXPANDING;
 	     xf86DrvMsg(pScrn->scrnIndex, X_PROBED,
-		"Detected LCD/Plasma panel (max. X %d Y %d, pref. %dx%d, RGB%d)\n",
+		"Detected LCD/plasma panel (%dx%d, pref. %dx%d, RGB%d)\n",
 		pSiS->SiS_Pr->CP_MaxX, pSiS->SiS_Pr->CP_MaxY,
 		pSiS->SiS_Pr->CP_PreferredX, pSiS->SiS_Pr->CP_PreferredY,
 		(CR37 & 0x01) ? 18 : 24);
