--- webalizer-2.01-10/dns_resolv.c.overflow	2004-01-08 14:51:04.000000000 +0000
+++ webalizer-2.01-10/dns_resolv.c	2004-01-08 14:51:11.000000000 +0000
@@ -156,7 +156,7 @@
          strncpy (log_rec->hostname,
                   ((struct dnsRecord *)response.data)->hostName,
                   MAXHOST);
-         log_rec->hostname[MAXHOST]=0;
+         log_rec->hostname[MAXHOST-1]=0;
          if (debug_mode)
             fprintf(stderr," found: %s (%ld)\n",
              log_rec->hostname, alignedRecord.timeStamp);
