--- directory/c-sdk/ldap/libraries/libldap/os-ip.c	2002-04-02 09:44:45.000000000 +0200
+++ directory/c-sdk/ldap/libraries/libldap/os-ip.c.new	2007-01-30 18:10:39.270576000 +0100
@@ -554,6 +554,7 @@
 #endif /* GETHOSTBYNAME_BUF_T */
 
 	connected = 0;
+	ldhpbuf_allocd = NULL;
 	parse_err = ldap_x_hostlist_first( hostlist, defport, &host, &port,
             &status );
 	while ( !connected && LDAP_SUCCESS == parse_err && host != NULL ) {
