diff -Nur ode-0.6-old/drawstuff/dstest/dstest.cpp ode-0.6/drawstuff/dstest/dstest.cpp
--- ode-0.6-old/drawstuff/dstest/dstest.cpp	2006-07-17 17:57:35.023078280 +0300
+++ ode-0.6/drawstuff/dstest/dstest.cpp	2006-07-17 17:57:43.621771080 +0300
@@ -116,7 +116,7 @@
   fn.step = &simLoop;
   fn.command = command;
   fn.stop = 0;
-  fn.path_to_textures = 0;	// uses default
+  fn.path_to_textures = "/usr/share//examples";	// uses default
 
   // run simulation
   dsSimulationLoop (argc,argv,400,400,&fn);
diff -Nur ode-0.6-old/ode/test/test_basket.cpp ode-0.6/ode/test/test_basket.cpp
--- ode-0.6-old/ode/test/test_basket.cpp	2006-07-17 17:57:35.039075848 +0300
+++ ode-0.6/ode/test/test_basket.cpp	2006-07-17 17:57:43.475793272 +0300
@@ -210,7 +210,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     fn.path_to_textures = argv[1];
 
diff -Nur ode-0.6-old/ode/test/test_boxstack.cpp ode-0.6/ode/test/test_boxstack.cpp
--- ode-0.6-old/ode/test/test_boxstack.cpp	2006-07-17 17:57:35.039075848 +0300
+++ ode-0.6/ode/test/test_boxstack.cpp	2006-07-17 17:57:43.479792664 +0300
@@ -468,7 +468,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.6-old/ode/test/test_buggy.cpp ode-0.6/ode/test/test_buggy.cpp
--- ode-0.6-old/ode/test/test_buggy.cpp	2006-07-17 17:57:35.039075848 +0300
+++ ode-0.6/ode/test/test_buggy.cpp	2006-07-17 17:57:43.482792208 +0300
@@ -219,7 +219,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.6-old/ode/test/test_chain1.c ode-0.6/ode/test/test_chain1.c
--- ode-0.6-old/ode/test/test_chain1.c	2006-07-17 17:57:35.039075848 +0300
+++ ode-0.6/ode/test/test_chain1.c	2006-07-17 17:57:43.483792056 +0300
@@ -129,7 +129,7 @@
   fn.step = &simLoop;
   fn.command = 0;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.6-old/ode/test/test_chain2.cpp ode-0.6/ode/test/test_chain2.cpp
--- ode-0.6-old/ode/test/test_chain2.cpp	2006-07-17 17:57:35.039075848 +0300
+++ ode-0.6/ode/test/test_chain2.cpp	2006-07-17 17:57:43.485791752 +0300
@@ -122,7 +122,7 @@
   fn.step = &simLoop;
   fn.command = 0;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.6-old/ode/test/test_collision.cpp ode-0.6/ode/test/test_collision.cpp
--- ode-0.6-old/ode/test/test_collision.cpp	2006-07-17 17:57:35.039075848 +0300
+++ ode-0.6/ode/test/test_collision.cpp	2006-07-17 17:57:43.496790080 +0300
@@ -1284,7 +1284,7 @@
     fn.step = &simLoop;
     fn.command = &command;
     fn.stop = 0;
-    fn.path_to_textures = "../../drawstuff/textures";
+    fn.path_to_textures = "/usr/share//examples";
 
     dsSetSphereQuality (3);
     dsSetCapsuleQuality (8);
diff -Nur ode-0.6-old/ode/test/test_crash.cpp ode-0.6/ode/test/test_crash.cpp
--- ode-0.6-old/ode/test/test_crash.cpp	2006-07-17 17:57:35.040075696 +0300
+++ ode-0.6/ode/test/test_crash.cpp	2006-07-17 17:57:43.502789168 +0300
@@ -609,7 +609,7 @@
 	fn.step = &simLoop;
 	fn.command = &command;
 	fn.stop = 0;
-	fn.path_to_textures = "../../drawstuff/textures";
+	fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.6-old/ode/test/test_cyl.cpp ode-0.6/ode/test/test_cyl.cpp
--- ode-0.6-old/ode/test/test_cyl.cpp	2006-07-17 17:57:35.040075696 +0300
+++ ode-0.6/ode/test/test_cyl.cpp	2006-07-17 17:57:43.504788864 +0300
@@ -230,7 +230,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.6-old/ode/test/test_cylvssphere.cpp ode-0.6/ode/test/test_cylvssphere.cpp
--- ode-0.6-old/ode/test/test_cylvssphere.cpp	2006-07-17 17:57:35.040075696 +0300
+++ ode-0.6/ode/test/test_cylvssphere.cpp	2006-07-17 17:57:43.506788560 +0300
@@ -172,7 +172,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     fn.path_to_textures = argv[1];
 
diff -Nur ode-0.6-old/ode/test/test_friction.cpp ode-0.6/ode/test/test_friction.cpp
--- ode-0.6-old/ode/test/test_friction.cpp	2006-07-17 17:57:35.040075696 +0300
+++ ode-0.6/ode/test/test_friction.cpp	2006-07-17 17:57:43.508788256 +0300
@@ -167,7 +167,7 @@
   fn.step = &simLoop;
   fn.command = 0;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.6-old/ode/test/test_hinge.cpp ode-0.6/ode/test/test_hinge.cpp
--- ode-0.6-old/ode/test/test_hinge.cpp	2006-07-17 17:57:35.040075696 +0300
+++ ode-0.6/ode/test/test_hinge.cpp	2006-07-17 17:57:43.509788104 +0300
@@ -125,7 +125,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.6-old/ode/test/test_I.cpp ode-0.6/ode/test/test_I.cpp
--- ode-0.6-old/ode/test/test_I.cpp	2006-07-17 17:57:35.040075696 +0300
+++ ode-0.6/ode/test/test_I.cpp	2006-07-17 17:57:43.512787648 +0300
@@ -235,7 +235,7 @@
   fn.step = &simLoop;
   fn.command = 0;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.6-old/ode/test/test_joints.cpp ode-0.6/ode/test/test_joints.cpp
--- ode-0.6-old/ode/test/test_joints.cpp	2006-07-17 17:57:35.040075696 +0300
+++ ode-0.6/ode/test/test_joints.cpp	2006-07-17 17:57:43.521786280 +0300
@@ -960,7 +960,7 @@
   fn.step = &simLoop;
   fn.command = 0;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if (argc>=2)
     fn.path_to_textures = argv[1];
 
diff -Nur ode-0.6-old/ode/test/test_motor.cpp ode-0.6/ode/test/test_motor.cpp
--- ode-0.6-old/ode/test/test_motor.cpp	2006-07-17 17:57:35.040075696 +0300
+++ ode-0.6/ode/test/test_motor.cpp	2006-07-17 17:57:43.523785976 +0300
@@ -140,7 +140,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc>=2)
   {
     fn.path_to_textures = argv[1];
diff -Nur ode-0.6-old/ode/test/test_moving_trimesh.cpp ode-0.6/ode/test/test_moving_trimesh.cpp
--- ode-0.6-old/ode/test/test_moving_trimesh.cpp	2006-07-17 17:57:35.040075696 +0300
+++ ode-0.6/ode/test/test_moving_trimesh.cpp	2006-07-17 17:57:43.538783696 +0300
@@ -1886,7 +1886,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.6-old/ode/test/test_slider.cpp ode-0.6/ode/test/test_slider.cpp
--- ode-0.6-old/ode/test/test_slider.cpp	2006-07-17 17:57:35.041075544 +0300
+++ ode-0.6/ode/test/test_slider.cpp	2006-07-17 17:57:43.575778072 +0300
@@ -136,7 +136,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.6-old/ode/test/test_space.cpp ode-0.6/ode/test/test_space.cpp
--- ode-0.6-old/ode/test/test_space.cpp	2006-07-17 17:57:35.041075544 +0300
+++ ode-0.6/ode/test/test_space.cpp	2006-07-17 17:57:43.577777768 +0300
@@ -201,7 +201,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.6-old/ode/test/test_space_stress.cpp ode-0.6/ode/test/test_space_stress.cpp
--- ode-0.6-old/ode/test/test_space_stress.cpp	2006-07-17 17:57:35.041075544 +0300
+++ ode-0.6/ode/test/test_space_stress.cpp	2006-07-17 17:57:43.580777312 +0300
@@ -396,7 +396,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.6-old/ode/test/test_step.cpp ode-0.6/ode/test/test_step.cpp
--- ode-0.6-old/ode/test/test_step.cpp	2006-07-17 17:57:35.041075544 +0300
+++ ode-0.6/ode/test/test_step.cpp	2006-07-17 17:57:43.582777008 +0300
@@ -174,7 +174,7 @@
   fn.step = &simLoop;
   fn.command = 0;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.6-old/ode/test/test_trimesh.cpp ode-0.6/ode/test/test_trimesh.cpp
--- ode-0.6-old/ode/test/test_trimesh.cpp	2006-07-17 17:57:35.041075544 +0300
+++ ode-0.6/ode/test/test_trimesh.cpp	2006-07-17 17:57:43.586776400 +0300
@@ -446,7 +446,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share//examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
