--- bin/fingerprint-helper/fingerprint-helper.pro.orig	2011-01-27 14:10:
+++ bin/fingerprint-helper/fingerprint-helper.pro	    2011-01-27 14:11:36.356000455 +0200
@@ -6,7 +6,7 @@
 INCLUDEPATH += . ../../include ../../upek/include
 
 isEmpty(PREFIX) {
-	PREFIX = /usr/local
+	PREFIX = /usr
 }
 isEmpty(LIB) {
 	LIB = lib
--- bin/fingerprint-identifier/fingerprint-identifier.pro.orig	2011-01-27 14:13:44.782000536 +0200
+++ bin/fingerprint-identifier/fingerprint-identifier.pro	2011-01-27 14:13:59.951000538 +0200
@@ -5,7 +5,7 @@
 INCLUDEPATH += . ../../include ../../upek/include
 
 isEmpty(PREFIX) {
-	PREFIX = /usr/local
+	PREFIX = /usr
 }
 isEmpty(LIB) {
 	LIB = lib
--- bin/fingerprint-plugin/fingerprint-plugin.pro.orig	2011-01-27 14:15:09.347000536 +0200
+++ bin/fingerprint-plugin/fingerprint-plugin.pro	2011-01-27 14:15:33.329000535 +0200
@@ -5,7 +5,7 @@
 INCLUDEPATH += . ../../include ../../upek/include
 
 isEmpty(PREFIX) {
-	PREFIX = /usr/local
+	PREFIX = /usr
 }
 isEmpty(LIB) {
 	LIB = lib
--- bin/fingerprint-polkit-agent/fingerprint-polkit-agent.pro.orig	2011-01-27 14:16:29.720000538 +0200
+++ bin/fingerprint-polkit-agent/fingerprint-polkit-agent.pro	2011-01-27 14:16:47.310000528 +0200
@@ -5,7 +5,7 @@
 INCLUDEPATH += . ../../include ../../upek/include
 
 isEmpty(PREFIX) {
-    PREFIX = /usr/local
+    PREFIX = /usr
 }
 isEmpty(LIB) {
     LIB = lib
--- bin/fingerprint-gui/fingerprint-gui.pro.orig	2011-01-27 14:18:14.066000538 +0200
+++ bin/fingerprint-gui/fingerprint-gui.pro	2011-01-27 14:18:33.602000557 +0200
@@ -6,7 +6,7 @@
 INCLUDEPATH += . ../../include ../../upek/include
 
 isEmpty(PREFIX) {
-	PREFIX = /usr/local
+	PREFIX = /usr
 }
 isEmpty(LIB) {
 	LIB = lib
--- bin/fingerprint-pam/fingerprint-pam.pro.orig	2011-01-27 14:19:31.786000534 +0200
+++ bin/fingerprint-pam/fingerprint-pam.pro	2011-01-27 14:19:45.808000483 +0200
@@ -5,7 +5,7 @@
 INCLUDEPATH += . ../../include ../../upek/include
 
 isEmpty(PREFIX) {
-	PREFIX = /usr/local
+	PREFIX = /usr
 }
 isEmpty(LIB) {
 	LIB = lib
