diff -Nur gkrellm-2.3.2-old/server/Makefile gkrellm-2.3.2/server/Makefile
--- gkrellm-2.3.2-old/server/Makefile	2009-05-06 22:38:11.000000000 +0300
+++ gkrellm-2.3.2/server/Makefile	2009-05-06 22:41:36.000000000 +0300
@@ -117,11 +117,11 @@
 all:	gkrellmd
 
 gkrellmd: $(OBJS) $(EXTRAOBJS)
-	$(CC) $(OBJS) $(EXTRAOBJS) -o gkrellmd $(LIBS) $(LINK_FLAGS)
+	$(CC) $(LINK_FLAGS) $(OBJS) $(EXTRAOBJS) -o gkrellmd $(LIBS)
 
 static: $(OBJS) $(EXTRAOBJS)
-	$(CC) $(OBJS) $(EXTRAOBJS) -o gkrellmd.static -static \
-		$(LIBS) $(LINK_FLAGS)
+	$(CC) $(LINK_FLAGS) $(OBJS) $(EXTRAOBJS) -o gkrellmd.static -static \
+		$(LIBS)
 
 freebsd2:
 	$(MAKE) GTK_CONFIG=gtk12-config \
diff -Nur gkrellm-2.3.2-old/src/Makefile gkrellm-2.3.2/src/Makefile
--- gkrellm-2.3.2-old/src/Makefile	2009-05-06 22:38:11.000000000 +0300
+++ gkrellm-2.3.2/src/Makefile	2009-05-06 22:38:17.000000000 +0300
@@ -125,11 +125,11 @@
 all:	gkrellm
 
 gkrellm: check_env $(OBJS) $(UNIXOBJS) $(EXTRAOBJS)
-	$(CC) $(OBJS) $(UNIXOBJS) $(EXTRAOBJS) -o gkrellm $(LIBS) $(LINK_FLAGS)
+	$(CC) $(LINK_FLAGS) $(OBJS) $(UNIXOBJS) $(EXTRAOBJS) -o gkrellm $(LIBS)
 
 static: check_env $(OBJS) $(UNIXOBJS) $(EXTRAOBJS)
-	$(CC) $(OBJS) $(UNIXOBJS) $(EXTRAOBJS) -o gkrellm.static -static \
-		$(LIBS) $(LINK_FLAGS)
+	$(CC) $(LINK_FLAGS) $(OBJS) $(UNIXOBJS) $(EXTRAOBJS) -o gkrellm.static -static \
+		$(LIBS)
 
 freebsd2:
 ifeq ($(HAVE_SSL),1)
