diff -Naur luminance-2.0-SVN691.orig/luminance.desktop luminance-2.0-SVN691/luminance.desktop
--- luminance-2.0-SVN691.orig/luminance.desktop	2009-12-26 15:41:05.821947178 +0200
+++ luminance-2.0-SVN691/luminance.desktop	2009-12-26 15:48:56.239823141 +0200
@@ -3,12 +3,14 @@
 Name=LuminanceHDR
 Name[en_US]=LuminanceHDR
 GenericName=HDR imaging
+GenericName[tr]=HDR Resim Düzenleyici
 GenericName[de]=Workflow für HDR Bilder
 GenericName[en_US]=HDR imaging
 GenericName[it]=Workflow per immagini HDR
 GenericName[ru]=Работа с HDR-фотографией
 Categories=Graphics;RasterGraphics;Photography;Viewer;Qt;
 Comment=Create and tonemap HDR images
+Comment[tr]=HDR resim dosyaları oluşturun ve ton haritalamalarını düzenleyin
 Comment[de]=Luminance ist eine grafische Benutzeroberfläche um HDR Bilder zu erzeugen und um die Farbabbildung durchzuführen
 Comment[en_US]=Luminance is a graphical user interface used to create and tonemap HDR images
 Comment[hu_HU]=Ez a program nagy megvilágítástartományú (HDR) képek előállítására és megtekintésére szolgál.
