diff -ur xorg-video-7.2_rc3.orig/xf86-video-unichrome-0.2.6/src/via_drmclient.h xorg-video-7.2_rc3/xf86-video-unichrome-0.2.6/src/via_drmclient.h
--- xorg-video-7.2_rc3.orig/xf86-video-unichrome-0.2.6/src/via_drmclient.h	2006-08-24 17:18:28.000000000 +0300
+++ xorg-video-7.2_rc3/xf86-video-unichrome-0.2.6/src/via_drmclient.h	2007-01-24 23:49:13.230470892 +0200
@@ -26,8 +26,7 @@
 
 #include "drm.h"
 #include "xf86drm.h"
-
-typedef CARD32 uint32_t;
+#include "stdint.h"
 
 #ifdef X_NEED_DRMLOCK
 #define drm_hw_lock_t drmLock
