diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kdebase-4.1.80-orig/apps/konsole/data/profiles/Shell.profile kdebase-4.1.80/apps/konsole/data/profiles/Shell.profile
--- kdebase-4.1.80-orig/apps/konsole/data/profiles/Shell.profile	2008-11-29 22:52:49.000000000 +0200
+++ kdebase-4.1.80/apps/konsole/data/profiles/Shell.profile	2008-11-29 22:54:17.000000000 +0200
@@ -122,7 +122,7 @@
 RemoteTabTitleFormat=%h : %u
 
 [Appearance]
-ColorScheme=DarkPastels
+ColorScheme=Linux
 
 [Scrolling]
 ScrollMode=FixedLineCount
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kdebase-4.1.80-orig/apps/konsole/src/Profile.cpp kdebase-4.1.80/apps/konsole/src/Profile.cpp
--- kdebase-4.1.80-orig/apps/konsole/src/Profile.cpp	2008-11-29 22:52:49.000000000 +0200
+++ kdebase-4.1.80/apps/konsole/src/Profile.cpp	2008-11-29 22:53:49.000000000 +0200
@@ -157,7 +157,7 @@
     setProperty(ShowNewAndCloseTabButtons,false);
 
     setProperty(KeyBindings,"default");
-    setProperty(ColorScheme,"DarkPastels");
+    setProperty(ColorScheme,"Linux");
     setProperty(Font,KGlobalSettings::fixedFont());
 
     setProperty(HistoryMode,FixedSizeHistory);
