--- pwc-v4l.c.orig	2006-05-24 19:16:35.000000000 +0300
+++ pwc-v4l.c	2006-05-24 19:17:01.000000000 +0300
@@ -128,7 +128,6 @@
 	    .step    = 1,
 	    .default_value = 0,
 	},
-#if XAWTV_HAS_BEEN_FIXED
         {
 	    .id      = V4L2_CID_PRIVATE_SAVE_USER,
 	    .type    = V4L2_CTRL_TYPE_BUTTON,
@@ -210,7 +209,6 @@
 	    .step    = 1,
 	    .default_value = 0,
 	},
-#endif
 };
 
 #if CONFIG_PWC_DEBUG 
