--- etc/mailcap	2007-02-25 17:18:37.000000000 +0200
+++ etc/mailcap	2007-03-08 19:04:31.000000000 +0200
@@ -1,8 +1,12 @@
 # Compressed Archives
 application/x-pisi; pm-install '%s'; -n "$DISPLAY" ; nametemplate=%s.pisi
+application/gzip; ark '%s'; -n "$DISPLAY" ; nametemplate=%s.gz
 application/x-gzip; ark '%s'; -n "$DISPLAY" ; nametemplate=%s.gz
+application/zip; ark '%s'; -n "$DISPLAY" ; nametemplate=%s.zip
 application/x-zip; ark '%s'; -n "$DISPLAY" ; nametemplate=%s.zip
+application/tar; ark '%s'; -n "$DISPLAY" ; nametemplate=%s.tar
 application/x-tar; ark '%s'; -n "$DISPLAY" ; nametemplate=%s.tar
+application/bzip2; ark '%s'; -n "$DISPLAY" ; nametemplate=%s.bz2
 application/x-bzip2; ark '%s'; -n "$DISPLAY" ; nametemplate=%s.bz2
 
 # Powerpoint
@@ -166,6 +170,12 @@
 text/plain; kwrite '%s'; -n "$DISPLAY" ; nametemplate=%s.log
 application/txt; kwrite '%s'; -n "$DISPLAY" ; nametemplate=%s.log
 
-# Rest
+text/x-patch; kwrite '%s'; -n "$DISPLAY" ; nametemplate=%s.patch
+text/x-patch; kwrite '%s'; -n "$DISPLAY" ; nametemplate=%s.diff
+
+# Po files
 text/x-pofile; kbabel '%s'; -n "$DISPLAY" ; nametemplate=%s.po
 text/x-pofile; kbabel '%s'; -n "$DISPLAY" ; nametemplate=%s.pot
+
+# Bittorrent
+application/x-bittorrent; ktorrent '%s'; -n "$DISPLAY" ; nametemplate=%s.torrent
