--- kdelibs-4.2.85/khtml/rendering/render_table.h~	2009-05-06 12:14:46.000000000 +0200
+++ kdelibs-4.2.85/khtml/rendering/render_table.h	2009-05-07 18:51:03.000000000 +0200
@@ -281,8 +281,8 @@
     //True if any of the cells has a span of 0
     bool containsSpansZero;
 
-    signed short cRow;
-    ushort cCol;
+    unsigned int cRow;
+    unsigned int cCol;
     bool needCellRecalc;
 
     void recalcCells();
