Index: trunk/KDE/kdebindings/python/pykde4/sip/kparts/factory.sip
===================================================================
--- python/pykde4/sip/kparts/factory.sip	(revision 976473)
+++ python/pykde4/sip/kparts/factory.sip	(revision 976474)
@@ -37,7 +37,7 @@
 
 protected:
     virtual KParts::Part*   createPartObject (QWidget* parentWidget = 0, QObject* parent /Transfer/ = 0, const char* classname = "KParts::Part", const QStringList& args = QStringList()) = 0;
-    virtual QObject*        createObject (QObject* parent /Transfer/ = 0, const char* classname = "QObject", const QStringList& args = QStringList());
+//ig    virtual QObject*        createObject (QObject* parent /Transfer/ = 0, const char* classname = "QObject", const QStringList& args = QStringList());
 };   // Factory
 
 };   // KParts
Index: trunk/KDE/kdebindings/python/pykde4/sip/kdecore/kptydevice.sip
===================================================================
--- python/pykde4/sip/kdecore/kptydevice.sip	(revision 976473)
+++ python/pykde4/sip/kdecore/kptydevice.sip	(revision 976474)
@@ -48,7 +48,7 @@
 protected:
 //ig    virtual qint64       readData (char*, qint64);
 //ig    virtual qint64       readLineData (char*, qint64);
-    virtual qint64       writeData (const char*, qint64);
+//ig    virtual qint64       writeData (const char*, qint64);
 
 };  // class KPtyDevice
 
