--- Mercator/RandCache.h~       2008-01-21 17:19:05.000000000 +0000
+++ Mercator/RandCache.h        2008-01-21 17:20:04.000000000 +0000
@@ -6,6 +6,7 @@
 #define MERCATOR_RANDCACHE_H

 #include <vector>
+#include <cstdlib>
 #include <wfmath/MersenneTwister.h>

 // construct with something like:
--- Mercator/AreaShader.cpp~    2008-01-21 17:20:24.000000000 +0000
+++ Mercator/AreaShader.cpp     2008-01-21 17:22:42.000000000 +0000
@@ -12,6 +12,7 @@
 #include "Mercator/Segment.h"
 #include "Mercator/Surface.h"

+#include <algorithm>
 #include <set>
 #include <iostream>

--- Mercator/AreaShader.h~      2008-01-21 17:21:02.000000000 +0000
+++ Mercator/AreaShader.h       2008-01-21 17:22:52.000000000 +0000
@@ -22,7 +22,7 @@
     virtual bool checkIntersect(const Segment &) const;
 private:
     /// helper to shader a single area into the surface
-    void shadeArea(Surface& s, const Area* const s) const;
+    void shadeArea(Surface& s, const Area* const ar) const;
     
     int m_layer;
 };
