--- bugzilla-4.0/Bugzilla/Constants.pm        2009-02-03 10:02:53.000000000 +0000
+++ bugzilla-4.0-rw/Bugzilla/Constants.pm     2009-02-18 17:59:52.000000000 +0000
@@ -539,9 +539,9 @@
         'cgi_path'    => $libpath,
         'templatedir' => "$libpath/template",
         'project'     => $project,
-        'localconfig' => "$libpath/$localconfig",
-        'datadir'     => "$libpath/$datadir",
-        'attachdir'   => "$libpath/$datadir/attachments",
+        'localconfig' => "/etc/bugzilla/$localconfig",
+        'datadir'     => "/var/lib/bugzilla/$datadir",
+        'attachdir'   => "/var/lib/bugzilla/$datadir/attachments",
         'skinsdir'    => "$libpath/skins",
         'graphsdir'   => "$libpath/graphs",
         # $webdotdir must be in the web server's tree somewhere. Even if you use a 
@@ -550,7 +550,7 @@
         # change showdependencygraph.cgi to set image_url to the correct 
         # location.
         # The script should really generate these graphs directly...
-        'webdotdir'   => "$libpath/$datadir/webdot",
+        'webdotdir'   => "/var/lib/bugzilla/$datadir/webdot",
         'extensionsdir' => "$libpath/extensions",
     };
 }
