--- rts/build/scons/config.py.orig	2007-12-11 20:23:11.000000000 +0200
+++ rts/build/scons/config.py	2007-12-18 22:32:45.000000000 +0200
@@ -212,7 +212,7 @@
 	if env['platform'] == 'windows':
 		guess_include_path(env, conf, 'Java', 'java')
 		return
-	possible_dirs = []
+	possible_dirs = ["/opt/sun-jdk/include"]
 	for root in ["/usr/local/lib/jvm", "/usr/local/lib64/jvm", "/usr/lib/jvm", "/usr/lib64/jvm", "/usr/java"]:
 		if os.path.exists(root) and os.path.isdir(root):
 			dirs = os.listdir(root)
