--- rts/spring.desktop.orig	2007-08-11 00:58:47.000000000 +0300
+++ rts/spring.desktop	2007-12-18 22:27:49.000000000 +0200
@@ -2,8 +2,11 @@
 Name=Spring
 Comment=An open source RTS with similar gameplay to TA
 Comment[de]=Multiplayerstrategie der nächsten Generation
+Comment[tr]=TA benzeri oyun tarzı ile açık kaynak bir strateji oyunu
+GenericName=Realtime Strategy Game
+GenericName[tr]=Gerçek Zamanlı Strateji Oyunu
 Exec=spring
-Icon=spring.png
+Icon=spring
 Terminal=false
 Type=Application
 Categories=Application;Game;StrategyGame;
