From 195eb8fc1423486847cfc892dfb01beb735d18b7 Mon Sep 17 00:00:00 2001
From: Brian Paul <brian.paul@tungstengraphics.com>
Date: Thu, 28 Aug 2008 15:15:41 -0600
Subject: [PATCH] mesa: bump MAX_INSN to 350

---
 src/mesa/tnl/t_vp_build.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git Mesa-7.0.4/src/mesa/tnl/t_vp_build.c Mesa-7.0.4/src/mesa/tnl/t_vp_build.c
index b7bc197..ddb0d51 100644
--- Mesa-7.0.4/src/mesa/tnl/t_vp_build.c
+++ Mesa-7.0.4/src/mesa/tnl/t_vp_build.c
@@ -235,7 +235,7 @@ static struct state_key *make_state_key( GLcontext *ctx )
  */
 #define PREFER_DP4 0
 
-#define MAX_INSN 256
+#define MAX_INSN 350
 
 /* Use uregs to represent registers internally, translate to Mesa's
  * expected formats on emit.  
-- 
1.6.0.2

