Index: udev-126/rules/rules.d/50-udev-default.rules
===================================================================
--- udev-126.orig/rules/rules.d/50-udev-default.rules
+++ udev-126/rules/rules.d/50-udev-default.rules
@@ -64,7 +64,7 @@ SUBSYSTEM=="scsi", KERNEL=="[0-9]*:[0-9]
 SUBSYSTEM=="scsi", KERNEL=="[0-9]*:[0-9]*", ACTION=="add", ATTR{type}=="1", ATTR{timeout}="900"
 KERNEL=="hd*", SUBSYSTEMS=="ide", ATTRS{media}=="floppy", OPTIONS+="all_partitions"
 KERNEL=="fd[0-9]",		GROUP="floppy"
-KERNEL=="fd[0-9]", ACTION=="add", ATTRS{cmos}=="?*", RUN+="create_floppy_devices -c -t $attr{cmos} -m %M -M 0640 -G floppy $root/%k"
+KERNEL=="fd[0-9]", ACTION=="add", ATTRS{cmos}=="?*", RUN+="create_floppy_devices -c -t $attr{cmos} -m %M -M 0640 -G pnp $root/%k"
 KERNEL=="sch[0-9]*",		GROUP="disk"
 KERNEL=="sg[0-9]*",		ATTRS{type}!="3|6", GROUP="disk", MODE="0640"
 KERNEL=="ht[0-9]*|nht[0-9]*",	GROUP="disk"
