--- faad2/configure.in.mpeg4ipplugin	2006-10-22 13:17:01.000000000 +0200
+++ faad2/configure.in	2006-10-22 13:23:07.000000000 +0200
@@ -85,6 +85,8 @@
 AM_CONDITIONAL(HAVE_MPEG4IP_PLUG, false)
 if test x$WITHMPEG4IP = xyes; then
    if test x$external_mp4v2 = xyes; then
+      MPEG4IP_PLAYER_PLUGIN_DIR=`mpeg4ip-config --player-plugin-dir`
+      AC_SUBST(MPEG4IP_PLAYER_PLUGIN_DIR)
       AM_CONDITIONAL(HAVE_MPEG4IP_PLUG, true)
       AC_MSG_NOTICE("Building MPEG4IP plugin")
    else
