diff -up OCSNG_UNIX_SERVER-1.3.2/ocsreports/blacklist.php.orig OCSNG_UNIX_SERVER-1.3.2/ocsreports/blacklist.php
--- OCSNG_UNIX_SERVER-1.3.2/ocsreports/blacklist.php.orig	2010-05-13 08:00:03.000000000 +0200
+++ OCSNG_UNIX_SERVER-1.3.2/ocsreports/blacklist.php	2010-05-13 08:00:17.000000000 +0200
@@ -13,7 +13,7 @@ if (isset($_GET['choise'])) $_POST['choo
 ?>
 	<script language=javascript>
 		function confirme(did){
-			if(confirm("<?echo $l->g(640)?> id "+did+" ?"))
+			if(confirm("<?php echo $l->g(640)?> id "+did+" ?"))
 				window.location="index.php?multi=<?php echo $_GET["multi"]?>&choise=<?php echo $_POST['choose_blacklist']?>&suppAcc="+did;
 		}
 		function convertToUpper(v_string){
