Index: src/gui/ScrSmolt.py
===================================================================
--- src/gui/ScrSmolt.py	(revision 29642)
+++ src/gui/ScrSmolt.py	(revision 29675)
@@ -56,8 +56,7 @@
             self.setRowColor(self.ui.tableWidget, labelItem)
             self.ui.tableWidget.setItem(row, 0, labelItem)
 
-            _value = QtCore.QVariant(value).toString()
-            dataItem = QtGui.QTableWidgetItem(_value)
+            dataItem = QtGui.QTableWidgetItem(str(value))
             self.setRowColor(self.ui.tableWidget, dataItem)
             self.ui.tableWidget.setItem(row, 1, dataItem)
 
