diff -Nuar R-2.8.1.orig/m4/R.m4 R-2.8.1/m4/R.m4
--- R-2.8.1.orig/m4/R.m4	2008-11-12 11:23:33.000000000 +0200
+++ R-2.8.1/m4/R.m4	2009-02-13 10:37:13.000000000 +0200
@@ -258,7 +258,7 @@
 ## the FreeBSD acroread port.
 AC_DEFUN([R_PROG_PDFVIEWER],
 [AC_PATH_PROGS(R_PDFVIEWER,
-               [${R_PDFVIEWER} acroread acroread4 evince xpdf gv gnome-gv ggv kghostview open gpdf])
+               [${R_PDFVIEWER} xdg-open acroread acroread4 evince xpdf gv gnome-gv ggv kghostview open gpdf])
 if test -z "${R_PDFVIEWER}"; then
   warn_pdfviewer="I could not determine a PDF viewer"
   AC_MSG_WARN([${warn_pdfviewer}])
