diff -Nur boson-all-0.13-old/code/boson/data/boson-boinfo.desktop boson-all-0.13/code/boson/data/boson-boinfo.desktop
--- boson-all-0.13-old/code/boson/data/boson-boinfo.desktop	2006-10-04 22:45:41.001021685 +0300
+++ boson-all-0.13/code/boson/data/boson-boinfo.desktop	2006-10-04 22:47:07.410885389 +0300
@@ -2,6 +2,8 @@
 Encoding=UTF-8
 Name=Boson Info
 GenericName=Boson System Information
+Name[tr]=Boson Bilgi
+GenericName[tr]=Boson Sistem Bilgisi
 Exec=boinfo
 Icon=boson
 Path=
diff -Nur boson-all-0.13-old/code/boson/data/boson-boson.desktop boson-all-0.13/code/boson/data/boson-boson.desktop
--- boson-all-0.13-old/code/boson/data/boson-boson.desktop	2006-10-04 22:45:41.001021685 +0300
+++ boson-all-0.13/code/boson/data/boson-boson.desktop	2006-10-04 22:47:32.759031885 +0300
@@ -4,9 +4,11 @@
 Name[de]=Boson
 Name[fr]=Boson
 Name[it]=Boson
+Name[tr]=Boson
 GenericName=Real-time strategy game
 GenericName[de]=Echtzeit Strategiespiel
 GenericName[fr]=Jeu de strategie en temps reel
+GenericName[tr]=Gerçekzamanlı Strateji Oyunu
 Exec=boson
 Icon=boson
 X-DocPath=boson/index.html
diff -Nur boson-all-0.13-old/code/boson/data/boson-bounit.desktop boson-all-0.13/code/boson/data/boson-bounit.desktop
--- boson-all-0.13-old/code/boson/data/boson-bounit.desktop	2006-10-04 22:45:41.000021837 +0300
+++ boson-all-0.13/code/boson/data/boson-bounit.desktop	2006-10-04 22:48:40.141788149 +0300
@@ -2,6 +2,8 @@
 Encoding=UTF-8
 GenericName=Unit Editor for Boson
 Name=BoUnit
+GenericName[tr]=Boson Araç/Görevli Düzenleyici
+Name[tr]=BoUnit
 Exec=bounit
 Icon=boson
 Path=
