--- kioslave/trash/trashimpl.cpp	2009/01/26 16:31:17	916986
+++ kioslave/trash/trashimpl.cpp	2009/01/26 16:41:07	916987
@@ -766,8 +766,10 @@
         return 0;
 
     KMountPoint::Ptr mp = KMountPoint::currentMountPoints().findByPath( origPath );
-    if (!mp)
+    if (!mp) {
+        kDebug() << "KMountPoint found no mount point for" << origPath;
         return 0;
+    }
     QString mountPoint = mp->mountPoint();
     const QString trashDir = trashForMountPoint( mountPoint, true );
     kDebug() << "mountPoint=" << mountPoint << " trashDir=" << trashDir;
@@ -861,7 +863,7 @@
         if ( (S_ISDIR(buff.st_mode)) // must be a dir
              && (!S_ISLNK(buff.st_mode)) // not a symlink
              && ((buff.st_mode & requiredBits) == requiredBits)
-             && (::access(rootTrashDir_c, W_OK))
+             && (::access(rootTrashDir_c, W_OK) == 0) // must be user-writable
             ) {
             const QString trashDir = rootTrashDir + '/' + QString::number( uid );
             const QByteArray trashDir_c = QFile::encodeName( trashDir );
