--- configure	2007-10-17 01:55:36.000000000 +0300
+++ configure	2007-12-04 18:26:17.000000000 +0200
@@ -24632,7 +24632,7 @@
 #CXXFLAGS="${GMP_CFLAGS} ${NTL_CFLAGS} ${GIVARO_CFLAGS} ${SACLIB_CFLAGS} ${CXXFLAGS}"
 #AC_SUBST(CXXFLAGS)
 
-ac_config_files="$ac_config_files Makefile linbox-config examples/Makefile examples/fields/Makefile examples/Readme-make gmp++/Makefile doc/Makefile linbox/Makefile linbox/algorithms/Makefile linbox/fflas/Makefile linbox/ffpack/Makefile linbox/blackbox/Makefile linbox/element/Makefile linbox/field/Makefile linbox/matrix/Makefile linbox/randiter/Makefile linbox/ring/Makefile linbox/solutions/Makefile linbox/switch/Makefile linbox/util/Makefile linbox/util/gmp++/Makefile linbox/util/formats/Makefile linbox/vector/Makefile tests/Makefile tests/data/Makefile interfaces/Makefile interfaces/driver/Makefile interfaces/maple/Makefile interfaces/kaapi/Makefile macros/Makefile"
+ac_config_files="$ac_config_files Makefile linbox-config examples/Makefile examples/fields/Makefile examples/Readme-make gmp++/Makefile doc/Makefile linbox/Makefile linbox/algorithms/Makefile linbox/fflas/Makefile linbox/ffpack/Makefile linbox/blackbox/Makefile linbox/element/Makefile linbox/field/Makefile linbox/matrix/Makefile linbox/randiter/Makefile linbox/ring/Makefile linbox/solutions/Makefile linbox/switch/Makefile linbox/util/Makefile linbox/util/gmp++/Makefile linbox/util/formats/Makefile linbox/vector/Makefile tests/Makefile tests/data/Makefile interfaces/Makefile interfaces/driver/Makefile interfaces/kaapi/Makefile macros/Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
