diff -ur unionfs-1.4.orig/stale_inode.c unionfs-1.4/stale_inode.c
--- unionfs-1.4.orig/stale_inode.c	2006-11-05 01:56:59.000000000 +0200
+++ unionfs-1.4/stale_inode.c	2007-06-04 13:31:43.539575673 +0300
@@ -10,7 +10,6 @@
  *  $Id: stale_inode.c,v 1.13 2006/03/21 09:22:11 jsipek Exp $
  */
 
-#include <linux/config.h>
 #include <linux/version.h>
 
 #include <linux/fs.h>
diff -ur unionfs-1.4.orig/unionfs.h unionfs-1.4/unionfs.h
--- unionfs-1.4.orig/unionfs.h	2006-11-05 01:56:59.000000000 +0200
+++ unionfs-1.4/unionfs.h	2007-06-04 13:31:36.088708377 +0300
@@ -3,7 +3,6 @@
 
 #ifdef __KERNEL__
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/sched.h>
 #include <linux/kernel.h>
