--- wlan.c.org  2008-01-15 13:12:54.000000000 +0200
+++ wlan.c      2008-01-15 13:13:20.000000000 +0200
@@ -39,7 +39,7 @@

 #include <linux/version.h>
 #if LINUX_VERSION_CODE <= KERNEL_VERSION(2, 6, 18)
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #endif
 #include <linux/types.h>
 #include <linux/if_arp.h>
--- conv.c.org  2008-01-15 13:13:35.000000000 +0200
+++ conv.c      2008-01-15 13:13:45.000000000 +0200
@@ -32,7 +32,7 @@

 #include <linux/version.h>
 #if LINUX_VERSION_CODE <= KERNEL_VERSION(2, 6, 18)
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #endif
 #include <linux/skbuff.h>
 #include <linux/if_arp.h>
--- ioctl.c.org 2008-01-15 13:13:53.000000000 +0200
+++ ioctl.c     2008-01-15 13:14:02.000000000 +0200
@@ -32,7 +32,7 @@

 #include <linux/version.h>
 #if LINUX_VERSION_CODE <= KERNEL_VERSION(2, 6, 18)
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #endif
 #include <linux/kernel.h>
 #include <linux/types.h>
--- common.c.org        2008-01-15 13:14:12.000000000 +0200
+++ common.c    2008-01-15 13:14:21.000000000 +0200
@@ -32,7 +32,7 @@

 #include <linux/version.h>
 #if LINUX_VERSION_CODE <= KERNEL_VERSION(2, 6, 18)
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #endif
 #include <linux/module.h>
 #include <linux/kernel.h>
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #endif

 /* Linux 2.6.18+ uses <linux/utsrelease.h> */
--- usb.c.org   2008-01-15 13:14:51.000000000 +0200
+++ usb.c       2008-01-15 13:15:03.000000000 +0200
@@ -46,7 +46,7 @@

 #include <linux/version.h>
 #if LINUX_VERSION_CODE <= KERNEL_VERSION(2, 6, 18)
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #endif
 #include <linux/types.h>
 #include <linux/module.h>
