Index: kdeplasma-addons-4.4.4/applets/lancelot/app/src/models/FolderModel.cpp
===================================================================
--- kdeplasma-addons-4.4.4.orig/applets/lancelot/app/src/models/FolderModel.cpp
+++ kdeplasma-addons-4.4.4/applets/lancelot/app/src/models/FolderModel.cpp
@@ -30,7 +30,7 @@ FolderModel::FolderModel(QString dirPath
     if (!m_dirPath.endsWith(QDir::separator())) {
         m_dirPath += QDir::separator();
     }
-    m_dirPath = m_dirPath.replace("//", "/");
+    //m_dirPath = m_dirPath.replace("//", "/");
 
     load();
 
