Index: kdeplasma-addons-4.4.4/applets/lancelot/app/src/parts/LancelotPart.cpp
===================================================================
--- kdeplasma-addons-4.4.4.orig/applets/lancelot/app/src/parts/LancelotPart.cpp
+++ kdeplasma-addons-4.4.4/applets/lancelot/app/src/parts/LancelotPart.cpp
@@ -242,7 +242,7 @@ void LancelotPart::createConfigurationIn
 {
     QWidget * widget = new QWidget();
     m_config.setupUi(widget);
-    m_config.panelIcon->setVisible(m_list->parentItem() == NULL);
+    m_config.panelIcon->setVisible(m_root->parentItem() == NULL);
 
     KConfigGroup kcg = config();
 
@@ -255,7 +255,7 @@ void LancelotPart::createConfigurationIn
     m_config.setIconClickActivation(
             kcg.readEntry("iconClickActivation", true));
     m_config.setContentsClickActivation(
-            kcg.readEntry("contentsClickActivation", m_list->parentItem() == NULL));
+            kcg.readEntry("contentsClickActivation", m_root->parentItem() == NULL));
     m_config.setContentsExtenderPosition(
             (Lancelot::ExtenderPosition)
             kcg.readEntry("contentsExtenderPosition",
@@ -286,7 +286,7 @@ void LancelotPart::applyConfig()
     }
     m_iconClickActivation = kcg.readEntry("iconClickActivation", true);
 
-    if (!kcg.readEntry("contentsClickActivation", m_list->parentItem() == NULL)) {
+    if (!kcg.readEntry("contentsClickActivation", m_root->parentItem() == NULL)) {
        m_list->setExtenderPosition(
                (Lancelot::ExtenderPosition)
                kcg.readEntry("contentsExtenderPosition",
