diff -Naur zenity-2.24.1.orig/help/Makefile.am zenity-2.24.1/help/Makefile.am
--- zenity-2.24.1.orig/help/Makefile.am	2008-12-13 02:18:01.000000000 +0200
+++ zenity-2.24.1/help/Makefile.am	2009-03-06 21:09:58.000000000 +0200
@@ -6,4 +6,4 @@
 DOC_INCLUDES = 
 DOC_FIGURES = figures/zenity-calendar-screenshot.png figures/zenity-entry-screenshot.png figures/zenity-error-screenshot.png figures/zenity-fileselection-screenshot.png figures/zenity-information-screenshot.png figures/zenity-list-screenshot.png figures/zenity-notification-screenshot.png figures/zenity-progress-screenshot.png figures/zenity-question-screenshot.png figures/zenity-text-screenshot.png figures/zenity-warning-screenshot.png
 
-DOC_LINGUAS = bg de en_GB es fi fr oc ru sv uk
+DOC_LINGUAS = de en_GB es fr
diff -Naur zenity-2.24.1.orig/po/LINGUAS zenity-2.24.1/po/LINGUAS
--- zenity-2.24.1.orig/po/LINGUAS	2008-12-13 02:18:01.000000000 +0200
+++ zenity-2.24.1/po/LINGUAS	2009-03-06 21:12:30.000000000 +0200
@@ -2,83 +2,16 @@
 #
 # please keep this list sorted alphabetically
 #
-am
-ar
-as
-az
-be
-be@latin
-bg
-bn
-bn_IN
-bs
 ca
-cs
-cy
-da
 de
-dz
-el
 en_CA
 en_GB
 es
-et
-eu
-fa
-fi
 fr
-ga
-gl
-gu
-he
-hi
-hr
-hu
-id
-is
 it
-ja
-ka
-kn
-ko
-ku
-lt
-lv
-mg
-mi
-mk
-ml
-mn
-mr
-ms
-nb
-ne
 nl
-nn
-oc
-or
-pa
 pl
 pt
 pt_BR
-ro
-ru
-rw
-si
-sk
-sl
-sq
-sr
-sr@latin
 sv
-ta
-te
-th
 tr
-ug
-uk
-vi
-xh
-zh_CN
-zh_HK
-zh_TW
