diff -up kdenlive-0.5/kdenlive/ktrackview.cpp.gcc43 kdenlive-0.5/kdenlive/ktrackview.cpp
--- kdenlive-0.5/kdenlive/ktrackview.cpp.gcc43	2008-02-23 21:05:37.000000000 +0900
+++ kdenlive-0.5/kdenlive/ktrackview.cpp	2008-02-23 21:06:50.000000000 +0900
@@ -17,6 +17,7 @@
 
 #include <iostream>
 #include <math.h>
+#include <cstdlib>
 
 #include <qpainter.h>
 #include <qcursor.h>
diff -up kdenlive-0.5/kdenlive/kruler.cpp.gcc43 kdenlive-0.5/kdenlive/kruler.cpp
--- kdenlive-0.5/kdenlive/kruler.cpp.gcc43	2008-02-23 21:02:35.000000000 +0900
+++ kdenlive-0.5/kdenlive/kruler.cpp	2008-02-23 21:02:47.000000000 +0900
@@ -23,6 +23,7 @@
 #include <klocale.h>
 
 #include <iostream>
+#include <cstdlib>
 #include <cmath>
 
 #include <kdebug.h>
diff -up kdenlive-0.5/kdenlive/krender.cpp.gcc43 kdenlive-0.5/kdenlive/krender.cpp
--- kdenlive-0.5/kdenlive/krender.cpp.gcc43	2008-02-23 21:00:45.000000000 +0900
+++ kdenlive-0.5/kdenlive/krender.cpp	2008-02-23 21:01:00.000000000 +0900
@@ -27,6 +27,7 @@ extern "C" {
 }
 
 #include <iostream>
+#include <cstdlib>
 
 // ffmpeg Header files
 
diff -up kdenlive-0.5/kdenlive/kdenlive.cpp.gcc43 kdenlive-0.5/kdenlive/kdenlive.cpp
--- kdenlive-0.5/kdenlive/kdenlive.cpp.gcc43	2008-02-23 20:58:23.000000000 +0900
+++ kdenlive-0.5/kdenlive/kdenlive.cpp	2008-02-23 20:58:42.000000000 +0900
@@ -121,7 +121,7 @@
 #include "trackpaneltransitionresizefunction.h"
 #include "trackpaneltransitionmovefunction.h"
 
-
+#include <cstdlib>
 
 #define ID_STATUS_MSG 1
 #define ID_EDITMODE_MSG 2
