diff -Nur widelands-old/src/wui/interactive_base.cc widelands/src/wui/interactive_base.cc
--- widelands-old/src/wui/interactive_base.cc	2009-10-22 10:12:39.197571016 +0300
+++ widelands/src/wui/interactive_base.cc	2009-10-22 10:14:13.126655302 +0300
@@ -90,8 +90,8 @@
 	//  Switch to the new graphics system now, if necessary.
 	WLApplication::get()->init_graphics
 		(get_xres(), get_yres(),
-		 global_s.get_int("depth", 16),
-		 global_s.get_bool("fullscreen", false),
+		 global_s.get_int("depth", 32),
+		 global_s.get_bool("fullscreen", true),
 		 global_s.get_bool("hw_improvements", false),
 		 global_s.get_bool("double_buffer", false)
 #if HAS_OPENGL
@@ -177,7 +177,7 @@
  */
 int32_t Interactive_Base::get_xres()
 {
-	return g_options.pull_section("global").get_int("xres", 640);
+	return g_options.pull_section("global").get_int("xres", 1024);
 }
 
 
@@ -188,7 +188,7 @@
  */
 int32_t Interactive_Base::get_yres()
 {
-	return g_options.pull_section("global").get_int("yres", 480);
+	return g_options.pull_section("global").get_int("yres", 768);
 }
 
 
