Index: kvm-84/qemu/qemu-kvm-x86.c
===================================================================
--- kvm-84.orig/qemu/qemu-kvm-x86.c
+++ kvm-84/qemu/qemu-kvm-x86.c
@@ -111,9 +111,9 @@ static int get_msr_entry(struct kvm_msr_
 }
 
 #ifdef TARGET_X86_64
-#define MSR_COUNT 9
+#define MSR_COUNT 10
 #else
-#define MSR_COUNT 5
+#define MSR_COUNT 6
 #endif
 
 static void set_v8086_seg(struct kvm_segment *lhs, const SegmentCache *rhs)
