Index: xorg-video-7.2/xf86-video-savage-2.2.0/src/savage_driver.c
===================================================================
--- xorg-video-7.2.orig/xf86-video-savage-2.2.1/src/savage_driver.c
+++ xorg-video-7.2/xf86-video-savage-2.2.1/src/savage_driver.c
@@ -1043,6 +1043,11 @@ static void SavageGetPanelInfo(ScrnInfoP
 	sTechnology = "STN";
     }
 
+    // one stable message for panel info
+    xf86DrvMsg(pScrn->scrnIndex, X_PROBED,
+                "Panel size is %d x %d\n",
+                panelX, panelY);
+
     xf86DrvMsg(pScrn->scrnIndex, X_PROBED,
 		   "%dx%d %s LCD panel detected %s\n", 
 		   panelX, panelY, sTechnology,
