diff -urNp module-init-tools-3.4_orig/depmod.c module-init-tools-3.4/depmod.c
--- module-init-tools-3.4_orig/depmod.c	2007-10-07 17:51:46.000000000 -0400
+++ module-init-tools-3.4/depmod.c	2007-10-07 18:18:14.000000000 -0400
@@ -1053,8 +1053,7 @@ static void read_toplevel_config(const c
 
 	/* Try defaults. */
 	for (i = 0; i < ARRAY_SIZE(default_configs); i++) {
-		if (read_config(default_configs[i], basedir, search, overrides))
-			return;
+		read_config(default_configs[i], basedir, search, overrides);
 	}
 }
 
diff -urNp module-init-tools-3.4_orig/modprobe.c module-init-tools-3.4/modprobe.c
--- module-init-tools-3.4_orig/modprobe.c	2007-10-07 17:57:23.000000000 -0400
+++ module-init-tools-3.4/modprobe.c	2007-10-07 18:18:14.000000000 -0400
@@ -1251,9 +1251,8 @@ static void read_toplevel_config(const c
 
 	/* Try defaults. */
 	for (i = 0; i < ARRAY_SIZE(default_configs); i++) {
-		if (read_config(default_configs[i], name, dump_only, removing,
-				options, commands, aliases, blacklist))
-			return;
+		read_config(default_configs[i], name, dump_only, removing,
+				options, commands, aliases, blacklist);
 	}
 }
 
