--- mfile.orig	2007-12-22 01:47:44.000000000 +0200
+++ mfile	2007-12-22 01:48:41.000000000 +0200
@@ -380,21 +380,25 @@
 
 
 install:
-	mkdir -p -m 755 $(LIBDIR)
-	cp -p ntl.a $(LIBDIR)/libntl.a
-	- chmod a+r $(LIBDIR)/libntl.a
-	mkdir -p -m 755 $(INCLUDEDIR)
-	- rm -r $(INCLUDEDIR)/NTL
-	mkdir -m 755 $(INCLUDEDIR)/NTL
-	cp -p ../include/NTL/*.h $(INCLUDEDIR)/NTL
-	- chmod -R a+r $(INCLUDEDIR)/NTL
-	mkdir -p -m 755 $(DOCDIR)
-	- rm -r $(DOCDIR)/NTL
-	mkdir -m 755 $(DOCDIR)/NTL
-	cp -p ../doc/*.txt $(DOCDIR)/NTL
-	cp -p ../doc/*.html $(DOCDIR)/NTL
-	cp -p ../doc/*.gif $(DOCDIR)/NTL
-	- chmod -R a+r $(DOCDIR)/NTL
+	mkdir -p -m 755 $(DESTDIR)$(LIBDIR)
+	cp -p libntl.so $(DESTDIR)$(LIBDIR)/libntl.so
+	- chmod 755 $(DESTDIR)$(LIBDIR)/libntl.so
+	mkdir -p -m 755 $(DESTDIR)$(INCLUDEDIR)
+	- rm -r $(DESTDIR)$(INCLUDEDIR)/NTL
+	mkdir -m 755 $(DESTDIR)$(INCLUDEDIR)/NTL
+	cp -p ../include/NTL/*.h $(DESTDIR)$(INCLUDEDIR)/NTL
+	- chmod -R a+r $(DESTDIR)$(INCLUDEDIR)/NTL
+	mkdir -p -m 755 $(DESTDIR)$(DOCDIR)
+	- rm -r $(DESTDIR)$(DOCDIR)
+	mkdir -m 755 $(DESTDIR)$(DOCDIR)
+	mkdir -p -m 755 $(DESTDIR)$(DOCDIR)/html
+	- rm -r $(DESTDIR)$(DOCDIR)/html
+	mkdir -m 755 $(DESTDIR)$(DOCDIR)/html
+	cp -p ../doc/*.txt $(DESTDIR)$(DOCDIR)
+	cp -p ../doc/*.html $(DESTDIR)$(DOCDIR)/html
+	cp -p ../doc/*.gif $(DESTDIR)$(DOCDIR)/html
+	- chmod -R a+r $(DESTDIR)$(DOCDIR)
+	- chmod -R a+r $(DESTDIR)$(DOCDIR)/html
 
 
 uninstall:
