--- configure.ac~	2006-09-14 10:53:41.000000000 +0300
+++ configure.ac	2006-09-14 10:55:08.000000000 +0300
@@ -39,21 +39,8 @@

 export PKG_CONFIG_PATH

-dnl
-dnl check for FLTK
-dnl
-AM_PATH_FLTK(have_fltk=yes, have_fltk=no)
-
-dnl
-dnl check for Nvidia SDK
-dnl
-if test $have_fltk = "yes"; then
-  AM_PATH_NVSDK(have_nvsdk=yes, have_nvsdk=no)
-  if test $have_nvsdk = "yes"; then
-    AC_DEFINE(HAVE_FRAGMENT_SHADERS)
-  fi
-else have_nvsdk = "no"
-fi
+have_fltk=no
+have_nvsdk=no

 dnl Checks for libraries.

