Index: kdesdk/kmtrace/Makefile.am
===================================================================
--- kdesdk.orig/kmtrace/Makefile.am
+++ kdesdk/kmtrace/Makefile.am
@@ -23,13 +23,13 @@ INCLUDES = $(all_includes)
 
 bin_PROGRAMS = kmtrace demangle kmmatch
 kmtrace_SOURCES = kmtrace.cpp
-kmtrace_LDFLAGS = $(all_libraries)
+kmtrace_LDFLAGS = /usr/kde/3.5/lib/libkdecore.so $(all_libraries)
 
 demangle_SOURCES = demangle.cpp
-demangle_LDFLAGS = $(all_libraries)
+demangle_LDFLAGS = /usr/kde/3.5/lib/libkdecore.so $(all_libraries)
 
 kmmatch_SOURCES = match.cpp
-kmmatch_LDFLAGS = $(all_libraries)
+kmmatch_LDFLAGS = /usr/kde/3.5/lib/libkdecore.so $(all_libraries)
 
 bin_SCRIPTS = kminspector
 
