Subject: stop Akonadi dropping connections to the mysql server after 8 hours
From: wstephenson@suse.de
Patch-upstream: r1094139
Index: server/src/storage/mysql-global.conf
===================================================================
--- server/src/storage/mysql-global.conf	(revision 1094138)
+++ server/src/storage/mysql-global.conf	(revision 1094139)
@@ -52,3 +52,6 @@ innodb_buffer_pool_size=80M
 innodb_log_file_size=64M
 innodb_flush_log_at_trx_commit=2
 
+# Do not drop the connection to the DB after 8 hours of inactivity
+wait_timeout=1296000
+
