diff -up openbabel-2.2.1b3-20090215-r2890/test/wrapper.sh.chk openbabel-2.2.1b3-20090215-r2890/test/wrapper.sh
--- openbabel-2.2.1b3-20090215-r2890/test/wrapper.sh.chk	2009-01-12 20:10:33.000000000 +0100
+++ openbabel-2.2.1b3-20090215-r2890/test/wrapper.sh	2009-03-01 16:19:28.000000000 +0100
@@ -16,11 +16,11 @@ TESTS="${TESTS} logp_psa math"
 TESTS="${TESTS} mol phmodel residue rings"
 TESTS="${TESTS} smarts smilesmatch unitcell"
 TESTS="${TESTS} cml.sh test-set.sh"
-if [ "x${srcdir}" != "x" ]; then
-  TESTS="${TESTS} ${srcdir}/inchi.pl ${srcdir}/inchi2.pl"
-else
-  TESTS="${TESTS} inchi.pl inchi2.pl"
-fi
+#if [ "x${srcdir}" != "x" ]; then
+#  TESTS="${TESTS} ${srcdir}/inchi.pl ${srcdir}/inchi2.pl"
+#else
+#  TESTS="${TESTS} inchi.pl inchi2.pl"
+#fi
 PROVE=prove
 
 echo "top srcdir: .${topsrcdir}."
@@ -45,4 +45,4 @@ if [ -d ../src/formats/.libs ]; then
     fi
 fi
 
-${PROVE} "$@" ${TESTS}
+${PROVE} --verbose "$@" ${TESTS}
diff -up openbabel-2.2.0/test/bond.cpp.chk openbabel-2.2.0/test/bond.cpp
--- openbabel-2.2.0/test/bond.cpp.chk	2008-06-13 22:48:05.000000000 +0200
+++ openbabel-2.2.0/test/bond.cpp	2008-07-06 21:45:44.000000000 +0200
@@ -58,7 +58,7 @@ int main(int argc,char *argv[])
 
   bond1.SetBegin(&begin1);
   bond1.SetEnd(&end1);
-  cout << "ok 4\n";
+  cout << "ok 3\n";
 
   return(0);
 }
 
