Index: pisi/api.py
===================================================================
--- pisi/api.py (revision 19948)
+++ pisi/api.py (working copy)
@@ -581,7 +581,7 @@
     ctx.ui.info(_('* Index file written'))

 def add_repo(name, indexuri, at = None):
-    if not re.match("^[a-zA-Z0-9\\-\\_]*$", name):
+    if not re.match("^[a-zA-Z0-9\\-\\_\\.]*$", name):
         raise pisi.Error(_('Not a valid repo name.'))
     repodb = pisi.db.repodb.RepoDB()
     if repodb.has_repo(name):
