diff -pruN kdelibs-3.5.9-orig/dcop/dcopidl/Makefile.am kdelibs-3.5.9/dcop/dcopidl/Makefile.am
--- kdelibs-3.5.9-orig/dcop/dcopidl/Makefile.am	2008-03-23 19:32:14.000000000 +0200
+++ kdelibs-3.5.9/dcop/dcopidl/Makefile.am	2008-03-23 19:33:45.000000000 +0200
@@ -11,7 +11,7 @@ dcopidl_SOURCES = main.cpp yacc.cc scann
 noinst_HEADERS = yacc.cc.h
 
 dcopidl_LDADD = $(LIB_QT)
-dcopidl_LDFLAGS = $(KDE_RPATH)
+dcopidl_LDFLAGS = $(AM_LDFLAGS) $(KDE_RPATH)
 
 ####### Build rules
 
diff -pruN kdelibs-3.5.9-orig/dcop/dcopidl2cpp/Makefile.am kdelibs-3.5.9/dcop/dcopidl2cpp/Makefile.am
--- kdelibs-3.5.9-orig/dcop/dcopidl2cpp/Makefile.am	2008-03-23 19:32:14.000000000 +0200
+++ kdelibs-3.5.9/dcop/dcopidl2cpp/Makefile.am	2008-03-23 19:37:45.000000000 +0200
@@ -10,7 +10,7 @@ dcopidl2cpp_SOURCES = main.cpp skel.cpp 
 noinst_HEADERS = main.h
 
 dcopidl2cpp_LDADD = $(LIB_QT)
-dcopidl2cpp_LDFLAGS = $(KDE_RPATH)
+dcopidl2cpp_LDFLAGS = $(AM_LDFLAGS) $(KDE_RPATH)
 
 dcopidl_output.kidl: ../dcopidl/dcopidl $(srcdir)/dcopidl_test.h
 	../dcopidl/dcopidl $(srcdir)/dcopidl_test.h > dcopidl_output.kidl
