diff -pruN libofa-0.9.3-orig/lib/signal_op.cpp libofa-0.9.3/lib/signal_op.cpp
--- libofa-0.9.3-orig/lib/signal_op.cpp	2008-04-04 06:03:31.000000000 +0300
+++ libofa-0.9.3/lib/signal_op.cpp	2008-04-04 06:04:33.000000000 +0300
@@ -13,6 +13,7 @@
 
 
 #include <math.h>
+#include <cstdlib>
 #include "signal_op.h"
 #include "AFLIB/aflibConverter.h"
 #include "error_op.h"
diff -pruN libofa-0.9.3-orig/examples/example.cpp libofa-0.9.3/examples/example.cpp
--- libofa-0.9.3-orig/examples/example.cpp	2008-04-04 06:09:14.000000000 +0300
+++ libofa-0.9.3/examples/example.cpp	2008-04-04 06:10:56.000000000 +0300
@@ -6,6 +6,7 @@
    No rights reserved.
 
 -------------------------------------------------------------------*/
+#include <cstring>
 
 #include "protocol.h"
 
diff -pruN libofa-0.9.3-orig/examples/protocol.cpp libofa-0.9.3/examples/protocol.cpp
--- libofa-0.9.3-orig/examples/protocol.cpp	2008-04-04 06:09:14.000000000 +0300
+++ libofa-0.9.3/examples/protocol.cpp	2008-04-04 06:11:44.000000000 +0300
@@ -9,6 +9,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string>
+#include <cstring>
 #include <map>
 #include <expat.h>
 #include <curl/curl.h>
