diff -Naur busybox-1.01/networking/udhcp/options.c busybox-1.01-new/networking/udhcp/options.c
--- busybox-1.01/networking/udhcp/options.c	2005-10-15 18:27:52.836298040 -0500
+++ busybox-1.01-new/networking/udhcp/options.c	2005-10-15 18:27:00.624235480 -0500
@@ -28,7 +28,7 @@
 	{"bootsize",	OPTION_U16,				0x0d},
 	{"domain",	OPTION_STRING | OPTION_REQ,		0x0f},
 	{"swapsvr",	OPTION_IP,				0x10},
-	{"rootpath",	OPTION_STRING,				0x11},
+	{"rootpath",	OPTION_STRING | OPTION_REQ,		0x11},
 	{"ipttl",	OPTION_U8,				0x17},
 	{"mtu",		OPTION_U16,				0x1a},
 	{"broadcast",	OPTION_IP | OPTION_REQ,			0x1c},
