--- autoopts/Makefile.in	2007-10-07 19:58:11.000000000 +0300
+++ autoopts/Makefile.in	2007-12-02 12:17:50.000000000 +0200
@@ -322,7 +322,6 @@
 
 INST_MANS = autoopts-config.1 $(GENMAN)
 INST_M4DATA = autoopts.m4
-INST_ND_PKGDATA = $(libsrc)
 INST_PKGCFG = autoopts.pc
 INST_PKGDATA = autoopts.m4 $(CLIENT_TPLS)
 INST_LIBS = libopts.la libguileopts.la
--- autoopts/autoopts-config.in	2007-10-07 19:58:00.000000000 +0300
+++ autoopts/autoopts-config.in	2007-12-02 12:20:35.000000000 +0200
@@ -26,16 +26,13 @@
      autogen="${bindir}/autogen${exeext}"
      ldflags="-L${libdir} -lopts"
         libs="${ldflags}"
-      libsrc="${pkgdatadir}/libopts-${dotver}.tar.gz"
- static_libs="${libdir}/libopts.a"
       cflags="-I${includedir}"
 test -n "${ldopts}" && ldflags="${ldopts}${libdir} ${ldflags}"
 test "${includedir}" = "/usr/include" && cflags=""
 optlist="\
     autogen     bindir      cflags      datadir     datarootdir dotver
     everything  exec_prefix exeext      includedir  ldflags     ldopts
-    libdir      libs        libsrc      package     pkgdatadir  prefix
-    static_libs version"
+    libdir      libs        package     pkgdatadir  prefix      version"
 
 usage()
 {
@@ -81,8 +78,6 @@
   autogen     ) val="${val} ${autogen}"      ;;
   ldflags     ) val="${val} ${ldflags}"      ;;
   libs        ) val="${val} ${libs}"         ;;
-  libsrc      ) val="${val} ${libsrc}"       ;;
-  static_libs ) val="${val} ${static_libs}"  ;;
   cflags      ) val="${val} ${cflags}"       ;;
   everything  )
      for o in ${optlist}
