--- configure.in	2007-01-19 22:17:39.000000000 +0200
+++ configure.in	2007-03-11 18:05:47.000000000 +0200
@@ -4986,13 +4986,7 @@
 dnl Checks specific to contrib libs (MMedia)
 dnl ------------------------------------------
 
-dnl Check for ESD:
 EXTRALIBS_ESD=
-AC_CHECK_LIB(esd, esd_close, [
-    AC_CHECK_HEADERS([esd.h], [
-    EXTRALIBS_ESD="-lesd"
-  ])
-])
 AC_SUBST(EXTRALIBS_ESD)
 
 dnl check for known CD-ROM interface
