diff -ur ClanLib-0.6.5.orig/Sources/Display/Input/DirectFB/keyboard_directfb.cpp ClanLib-0.6.5/Sources/Display/Input/DirectFB/keyboard_directfb.cpp
--- ClanLib-0.6.5.orig/Sources/Display/Input/DirectFB/keyboard_directfb.cpp	2002-06-07 22:36:33.000000000 +0300
+++ ClanLib-0.6.5/Sources/Display/Input/DirectFB/keyboard_directfb.cpp	2006-09-05 19:07:18.000000000 +0300
@@ -242,8 +242,7 @@
     case DIKI_SHIFT_L:     return CL_KEY_LSHIFT;
     case DIKI_SHIFT_R:     return CL_KEY_RSHIFT;
     case DIKI_ALT_L:       return CL_KEY_ALT;
-    case DIKI_ALT_R:       return CL_KEY_ALT;
-    case DIKI_ALTGR:       return CL_KEY_ALTGR;
+    case DIKI_ALT_R:       return CL_KEY_ALTGR;
     case DIKI_TAB:         return CL_KEY_TAB;
     case DIKI_ENTER:       return CL_KEY_ENTER;
     case DIKI_SPACE:       return CL_KEY_SPACE;
@@ -337,7 +336,7 @@
     case CL_KEY_LSHIFT:    return DIKI_SHIFT_L;
     case CL_KEY_RSHIFT:    return DIKI_SHIFT_R;
     case CL_KEY_ALT:       return DIKI_ALT_L;
-    case CL_KEY_ALTGR:     return DIKI_ALTGR;
+    case CL_KEY_ALTGR:     return DIKI_ALT_R;
     case CL_KEY_TAB:       return DIKI_TAB;
     case CL_KEY_ENTER:     return DIKI_ENTER;
     case CL_KEY_SPACE:     return DIKI_SPACE;
