--- smoke/qt/CMakeLists.txt~	2008-05-15 20:37:08.000000000 +0200
+++ smoke/qt/CMakeLists.txt	2008-05-26 13:42:22.000000000 +0200
@@ -144,7 +144,7 @@
 
 # Don't use the KDE ${LIB_INSTALL_DIR} macro here as this file is
 # for Qt only installs
-install(TARGETS smokeqt DESTINATION ${CMAKE_INSTALL_PREFIX}/lib )
+install(TARGETS smokeqt DESTINATION ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX} )
 install(FILES qt_smoke.h DESTINATION include/smoke/qt)
 
 configure_file(qtguess.pl.cmake ${CMAKE_CURRENT_BINARY_DIR}/qtguess.pl @ONLY)
--- smoke/soprano/CMakeLists.txt~	2008-05-21 10:32:37.000000000 +0200
+++ smoke/soprano/CMakeLists.txt	2008-05-26 13:44:05.000000000 +0200
@@ -72,7 +72,7 @@
 
 set_target_properties(smokesoprano PROPERTIES VERSION 2.0.0 SOVERSION 2 )
 
-install(TARGETS smokesoprano DESTINATION ${CMAKE_INSTALL_PREFIX}/lib )
+install(TARGETS smokesoprano DESTINATION ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX} )
 install(FILES soprano_smoke.h DESTINATION include/smoke/soprano)
 
 FOREACH(incname ${QT_INCLUDES} ${SOPRANO_INCLUDE_DIR})
--- smoke/qsci/CMakeLists.txt~	2008-05-15 20:37:08.000000000 +0200
+++ smoke/qsci/CMakeLists.txt	2008-05-26 13:45:25.000000000 +0200
@@ -70,7 +70,7 @@
 
 set_target_properties(smokeqsci PROPERTIES VERSION 2.0.0 SOVERSION 2 )
 
-install(TARGETS smokeqsci DESTINATION ${CMAKE_INSTALL_PREFIX}/lib )
+install(TARGETS smokeqsci DESTINATION ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX} )
 install(FILES qsci_smoke.h DESTINATION include/smoke/qsci)
 
 FOREACH(incname ${QT_INCLUDES} ${QSCINTILLA_INCLUDE_DIR})
--- ruby/qtruby/src/CMakeLists.txt~	2008-05-15 20:37:07.000000000 +0200
+++ ruby/qtruby/src/CMakeLists.txt	2008-05-26 13:47:07.000000000 +0200
@@ -33,7 +33,7 @@
 
 set_target_properties(qtruby4 PROPERTIES PREFIX "")
 
-install(TARGETS qtruby4shared LIBRARY DESTINATION lib )
+install(TARGETS qtruby4shared LIBRARY DESTINATION lib${LIB_SUFFIX} )
 install(TARGETS qtruby4 DESTINATION ${CUSTOM_RUBY_SITE_ARCH_DIR})
 
 install(FILES qtruby.h marshall.h marshall_basetypes.h marshall_complex.h marshall_primitives.h marshall_types.h smokeruby.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include/qtruby)
