--- setup.cfg	2007-06-08 11:34:18.000000000 +0300
+++ setup2.cfg	2007-10-21 20:54:12.000000000 +0300
@@ -19,12 +19,12 @@
 
 # If the build system does not find the mx.DateTime headers, try 
 # uncommenting the following line and setting its value to the right path.
-#mx_include_dir=
+mx_include_dir=/usr/lib/python2.4/site-packages/mx/DateTime/mxDateTime
 
 # For Windows only
 # Set to 1 if the PostgreSQL library was built with OpenSSL
 # Required to link in OpenSSL libraries and dependencies
-have_ssl=0
+have_ssl=1
 
 # "pg_config" is the preferred method to locate PostgreSQL headers and
 # libraries needed to build psycopg2. If pg_config is not in the path or
@@ -36,7 +36,7 @@
 # postgresql headers and libraries. Some extra checks on sys.platform will
 # still be done in setup.py.
 # The next line is the default as used on psycopg author Debian laptop:
-#include_dirs=/usr/include/postgresql:/usr/include/postgresql/server
+include_dirs=/usr/include/postgresql:/usr/include/postgresql/server
 
 # Uncomment next line on Mandrake 10.x (and comment previous ones):
 #include_dirs=/usr/include/pgsql/8.0:/usr/include/pgsql/8.0/server 
