diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur hplip-2.8.6b-orig/Makefile.in hplip-2.8.6b/Makefile.in
--- hplip-2.8.6b-orig/Makefile.in	2008-07-26 09:52:38.000000000 +0300
+++ hplip-2.8.6b/Makefile.in	2008-07-26 09:57:13.000000000 +0300
@@ -6091,7 +6091,7 @@
 install-docDATA: $(doc_DATA)
 	@$(NORMAL_INSTALL)
 	test -z "$(docdir)" || $(MKDIR_P) "$(DESTDIR)$(docdir)"
-	@list='$(doc_DATA)'; for p in $$list; do \
+	@list=''; for p in $$list; do \
 	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
 	  f=$(am__strip_dir) \
 	  echo " $(docDATA_INSTALL) '$$d$$p' '$(DESTDIR)$(docdir)/$$f'"; \
@@ -6100,7 +6100,7 @@
 
 uninstall-docDATA:
 	@$(NORMAL_UNINSTALL)
-	@list='$(doc_DATA)'; for p in $$list; do \
+	@list=''; for p in $$list; do \
 	  f=$(am__strip_dir) \
 	  echo " rm -f '$(DESTDIR)$(docdir)/$$f'"; \
 	  rm -f "$(DESTDIR)$(docdir)/$$f"; \
@@ -6404,7 +6404,7 @@
 	install-dist_ripSCRIPTS install-dist_rulesDATA \
 	install-dist_scanSCRIPTS install-dist_uiSCRIPTS \
 	install-dist_unrelDATA install-dist_www0DATA \
-	install-dist_www3DATA install-dist_www4DATA install-docDATA \
+	install-dist_www3DATA install-dist_www4DATA \
 	install-hpPROGRAMS install-hp_mkuriPROGRAMS \
 	install-hplip_confDATA install-hplip_desktopDATA \
 	install-hplip_systrayDATA install-hplipjsPROGRAMS \
@@ -6506,7 +6506,7 @@
 	install-dist_ripSCRIPTS install-dist_rulesDATA \
 	install-dist_scanSCRIPTS install-dist_uiSCRIPTS \
 	install-dist_unrelDATA install-dist_www0DATA \
-	install-dist_www3DATA install-dist_www4DATA install-docDATA \
+	install-dist_www3DATA install-dist_www4DATA \
 	install-dvi install-dvi-am install-exec install-exec-am \
 	install-hpPROGRAMS install-hp_mkuriPROGRAMS \
 	install-hplip_confDATA install-hplip_desktopDATA \
