--- pythoncad.desktop.orig	2009-12-09 11:28:58.000000000 +0200
+++ pythoncad.desktop	2010-03-18 13:16:10.297351296 +0200
@@ -1,9 +1,11 @@
 [Desktop Entry]
-Name=PyCAD
-Comment=PythonCAD
-Icon=gtkpycad
+Name=PythonCAD
+GenericName=CAD software
+GenericName[tr]=CAD uygulaması
+Comment=Draw CAD files using python
+Comment[tr]=Python temelli cad uygulaması
+Icon=pythoncad
 Exec=pythoncad
 Terminal=false
 Type=Application
-X-Desktop-File-Install-Version=0.3
-Categories=Office;X-Red-Hat-Base;
+Categories=Application;Graphics;
