diff --git a/tools/wineprefixcreate.in b/tools/wineprefixcreate.in
index 7ac5c8c..867f3d7 100644
--- a/tools/wineprefixcreate.in
+++ b/tools/wineprefixcreate.in
@@ -156,6 +156,7 @@ for i in \
     "$CROOT/windows/command" \
     "$CROOT/windows/fonts" \
     "$CROOT/windows/inf" \
+    "$CROOT/windows/resources" \
     "$CROOT/windows/system" \
     "$CROOT/windows/system32" \
     "$CROOT/windows/system32/drivers" \
