Index: dns.c
===================================================================
--- dns.c.orig
+++ dns.c
@@ -900,7 +900,7 @@ void resendrequest(struct resolve *rp,in
   } else if (type == T_PTR) {
     switch ( af ) {
     case AF_INET:
-    sprintf(tempstring,"%u.%u.%u.%u.in-addr.arpa",
+    snprintf(tempstring, sizeof(tempstring),"%u.%u.%u.%u.in-addr.arpa",
 	    ((byte *)&rp->ip)[3],
 	    ((byte *)&rp->ip)[2],
 	    ((byte *)&rp->ip)[1],
