--- sample-config/nagios.cfg.in	2007-12-14 19:10:09.000000000 +0100
+++ sample-config/nagios.cfg.in.oden	2008-01-14 16:18:53.000000000 +0100
@@ -18,7 +18,8 @@
 
 log_file=@localstatedir@/nagios.log
 
-
+# load command definitions for plugins
+cfg_dir=@sysconfdir@/plugins.d
 
 # OBJECT CONFIGURATION FILE(S)
 # These are the object configuration files in which you define hosts,
@@ -27,13 +28,13 @@
 # if you wish (as shown below), or keep them all in a single config file.
 
 # You can specify individual object config files as shown below:
-cfg_file=@sysconfdir@/objects/commands.cfg
-cfg_file=@sysconfdir@/objects/contacts.cfg
-cfg_file=@sysconfdir@/objects/timeperiods.cfg
-cfg_file=@sysconfdir@/objects/templates.cfg
+#cfg_file=@sysconfdir@/objects/commands.cfg
+#cfg_file=@sysconfdir@/objects/contacts.cfg
+#cfg_file=@sysconfdir@/objects/timeperiods.cfg
+#cfg_file=@sysconfdir@/objects/templates.cfg
 
 # Definitions for monitoring the local (Linux) host
-cfg_file=@sysconfdir@/objects/localhost.cfg
+#cfg_file=@sysconfdir@/objects/localhost.cfg
 
 # Definitions for monitoring a Windows machine
 #cfg_file=@sysconfdir@/objects/windows.cfg
@@ -44,17 +45,17 @@
 # Definitions for monitoring a network printer
 #cfg_file=@sysconfdir@/objects/printer.cfg
 
+# load configuration files on the fly
+cfg_dir=@sysconfdir@/conf.d
 
 # You can also tell Nagios to process all config files (with a .cfg
 # extension) in a particular directory by using the cfg_dir
 # directive as shown below:
 
-#cfg_dir=@sysconfdir@/servers
-#cfg_dir=@sysconfdir@/printers
-#cfg_dir=@sysconfdir@/switches
-#cfg_dir=@sysconfdir@/routers
-
-
+cfg_dir=@sysconfdir@/servers
+cfg_dir=@sysconfdir@/printers
+cfg_dir=@sysconfdir@/switches
+cfg_dir=@sysconfdir@/routers
 
 
 # OBJECT CACHE FILE
