--- Makefile.in	2007-11-28 09:38:48.000000000 +0200
+++ Makefile.in	2007-12-27 20:45:52.000000000 +0200
@@ -123,13 +123,11 @@
 install-all: install-asy install-doc
 
 install-asy: asy asy-keywords.el
-	${INSTALL} -d $(bindir) $(asydir) $(GUIdir) $(exampledir) \
+	${INSTALL} -d $(bindir) $(asydir) $(exampledir) \
           $(animationsdir)
 	-${INSTALL} -d $(latexdir)
 	${INSTALL} -p -m 755 $(NAME) $(bindir)
 	${INSTALL} -p -m 644 base/*.asy $(addprefix base/,$(EXTRA)) $(asydir)
-	${INSTALL} -p -m 755 GUI/*.py $(GUIdir)
-	ln -sf @datadir@/asymptote/GUI/xasy.py $(bindir)/$(XNAME)
 	${INSTALL} -p -m 644 examples/*.asy examples/piicon.eps doc/*.asy \
 	  doc/extra/*.asy doc/*.csv doc/*.dat doc/latexusage.tex $(exampledir)
 	${INSTALL} -p -m 644 examples/animations/*.asy $(animationsdir)
