Index: code/networkmanager/base.py
===================================================================
--- code/networkmanager/base.py	(revision 27519)
+++ code/networkmanager/base.py	(revision 27520)
@@ -35,6 +35,11 @@
 ANIMATION_TIME = 200
 DEFAULT_HEIGHT = 16777215
 
+def cropText(text):
+    if len(text) <= 32:
+        return text
+    return "%s...%s" % (text[:16], text[-16:])
+
 class MainManager(QtGui.QWidget):
     def __init__(self, parent, standAlone=True, app=None):
         QtGui.QWidget.__init__(self, parent)
@@ -432,7 +437,7 @@
         if len(devices) == 1:
             ui.deviceList.hide()
             ui.labelDeviceDescription.show()
-            ui.labelDeviceDescription.setText(devices[device])
+            ui.labelDeviceDescription.setText(cropText(devices[device]))
         else:
             ui.deviceList.show()
             ui.labelDeviceDescription.hide()
@@ -445,7 +450,7 @@
         ui.lineConnectionName.setText(data["name"])
 
         if "device_name" in data:
-            ui.labelDeviceDescription.setText(data["device_name"])
+            ui.labelDeviceDescription.setText(cropText(data["device_name"]))
         if "device_id" in data:
             index = ui.deviceList.findText(data["device_id"])
             if index != -1:
Index: ui/main.ui
===================================================================
--- ui/main.ui	(revision 27519)
+++ ui/main.ui	(revision 27520)
@@ -123,9 +123,9 @@
       <property name="geometry">
        <rect>
         <x>0</x>
-        <y>-123</y>
-        <width>861</width>
-        <height>483</height>
+        <y>0</y>
+        <width>863</width>
+        <height>508</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_6">
@@ -195,6 +195,10 @@
                   </property>
                  </widget>
                 </item>
+               </layout>
+              </item>
+              <item>
+               <layout class="QHBoxLayout" name="horizontalLayout_9">
                 <item>
                  <widget class="KSqueezedTextLabel" name="labelDeviceDescription">
                   <property name="sizePolicy">
@@ -208,10 +212,6 @@
                   </property>
                  </widget>
                 </item>
-               </layout>
-              </item>
-              <item>
-               <layout class="QHBoxLayout" name="horizontalLayout_9">
                 <item>
                  <widget class="QComboBox" name="deviceList">
                   <property name="minimumSize">
