Index: data/kcm_networkmanager.desktop
===================================================================
--- data/kcm_networkmanager.desktop	(revision 27544)
+++ data/kcm_networkmanager.desktop	(revision 27545)
@@ -17,12 +17,14 @@
 Name[es]=Conexiones de Red
 Name[de]=Netzwerkverbindungen
 Name[nl]=Netwerkbeheer
+Name[fr]=Gestionnaire de Réseau
 Comment=Network Manager
 Comment[sv]=Nätverkshanteraren
 Comment[es]=Administrador de Red
 Comment[de]=Netzwerkverbindung konfigurieren
 Comment[tr]=Ağ Yöneticisi
 Comment[nl]=Beheerder voor netwerk
+Comment[fr]=Gestionnaire de Réseau
 X-KDE-Keywords=python network manager dns ethernet wireless ip dhcp internet
 X-KDE-Keywords[tr]=python ağ yönetici dns ethernet kablosuz ip dhcp internet
 X-KDE-Keywords[nl]=python netwerkbeheer dns ethernet draadloos ip dhcp internet
Index: data/networkmanager.desktop
===================================================================
--- data/networkmanager.desktop	(revision 27544)
+++ data/networkmanager.desktop	(revision 27545)
@@ -4,14 +4,17 @@
 Name[nl]=Netwerkbeheer
 Name[de]=Netzwerkverbindungen
 Name[es]=Conexiones de Red
+Name[fr]=Gestionnaire de Réseau
 GenericName=Network Manager
 GenericName[tr]=Ağ Yöneticisi
 GenericName[nl]=Netwerkbeheer
 GenericName[de]=Netzwerk
+GenericName[fr]=Gestionnaire de Réseau
 Comment=An application for managing network connections.
 Comment[tr]=Ağ bağlantılarını yönetmenizi sağlayan bir uygulama.
 Comment[nl]=Beheerder voor netwerkverbindingen.
 Comment[de]=Netzwerkverbindung konfigurieren.
+Comment[fr]=Configuration du réseau
 Exec=network-manager
 Icon=applications-internet
 Type=Application
