diff -Nur ffmpeg-old/libavcodec/h264.c ffmpeg/libavcodec/h264.c
--- ffmpeg-old/libavcodec/h264.c	2008-08-15 11:52:39.000000000 +0300
+++ ffmpeg/libavcodec/h264.c	2008-08-15 11:52:59.000000000 +0300
@@ -7534,7 +7534,7 @@
         cnt = *(p++); // Number of pps
         for (i = 0; i < cnt; i++) {
             nalsize = AV_RB16(p) + 2;
-            if(decode_nal_units(h, p, nalsize)  != nalsize) {
+            if(decode_nal_units(h, p, nalsize) < 0) {
                 av_log(avctx, AV_LOG_ERROR, "Decoding pps %d from avcC failed\n", i);
                 return -1;
             }
