diff -Naur salatinoid.orig/applet/data/plasma-applet-salatinoid.desktop salatinoid/applet/data/plasma-applet-salatinoid.desktop
--- salatinoid.orig/applet/data/plasma-applet-salatinoid.desktop	2009-02-09 14:45:03.188060000 +0200
+++ salatinoid/applet/data/plasma-applet-salatinoid.desktop	2009-10-01 12:59:01.673432627 +0300
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Name=Salatinoid
 Comment=Islamic Prayer Times Plasmoid
+Comment[tr]=Namaz Vakitleri Programcığı
 Type=Service
 ServiceTypes=Plasma/Applet
 Icon=salatinoid
