diff -Naur salatinoid.orig/applet/plasma-applet-salatinoid.cpp salatinoid/applet/plasma-applet-salatinoid.cpp
--- salatinoid.orig/applet/plasma-applet-salatinoid.cpp	2009-02-11 13:37:10.013057000 +0200
+++ salatinoid/applet/plasma-applet-salatinoid.cpp	2009-10-01 12:51:02.456307914 +0300
@@ -32,10 +32,10 @@
 
 Salatinoid::Salatinoid(QObject *parent, const QVariantList &args) : Plasma::Applet(parent, args) {
     setHasConfigurationInterface(true);
-    prayerLabelFont = QFont("Utopia",14,true);
-    limitLabelFont  = QFont("Utopia",10,false);
-    prayerTimeFont  = QFont("Arial",8,false);
-    timeLeftFont    = QFont("Arial",9,true);
+    prayerLabelFont = QFont("DejavuSans",14,true);
+    limitLabelFont  = QFont("DejavuSans",12,false);
+    prayerTimeFont  = QFont("DejavuSans-Bold",9,false);
+    timeLeftFont    = QFont("DejavuSans-Bold",9,false);
     prayerNames[0] = i18n("Fajr");
     prayerNames[1] = i18n("Sunrise");
     prayerNames[2] = i18n("Zuhr");
