Description: Remove unnecessary dependencies from embryo and dbus-1.
Author: Davide Cavalca <davide@geexbox.org>
Forwarded: yes
Origin: upstream, http://hg.geexbox.org/enna/diff/4726da40700f/configure.ac

Index: enna-0.4.0/configure.ac
===================================================================
--- enna-0.4.0.orig/configure.ac	2010-01-04 15:33:53.000000000 +0100
+++ enna-0.4.0/configure.ac	2010-01-05 10:25:54.000000000 +0100
@@ -43,15 +43,12 @@
    evas >= 0.9.9.063
    ecore >= 0.9.9.063
    ecore-evas >= 0.9.9.063
-   ecore-txt >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   embryo >= 0.9.9.063
    edje >= 0.9.93.063
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
   ]
 )
 
Index: enna-0.4.0/configure
===================================================================
--- enna-0.4.0.orig/configure	2010-01-04 15:57:45.000000000 +0100
+++ enna-0.4.0/configure	2010-01-05 10:26:09.000000000 +0100
@@ -14051,15 +14051,12 @@
    evas >= 0.9.9.063
    ecore >= 0.9.9.063
    ecore-evas >= 0.9.9.063
-   ecore-txt >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   embryo >= 0.9.9.063
    edje >= 0.9.93.063
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 \""; } >&5
   ($PKG_CONFIG --exists --print-errors "
@@ -14068,15 +14065,12 @@
    evas >= 0.9.9.063
    ecore >= 0.9.9.063
    ecore-evas >= 0.9.9.063
-   ecore-txt >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   embryo >= 0.9.9.063
    edje >= 0.9.93.063
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 ") 2>&5
   ac_status=$?
@@ -14088,15 +14082,12 @@
    evas >= 0.9.9.063
    ecore >= 0.9.9.063
    ecore-evas >= 0.9.9.063
-   ecore-txt >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   embryo >= 0.9.9.063
    edje >= 0.9.93.063
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 " 2>/dev/null`
 else
@@ -14117,15 +14108,12 @@
    evas >= 0.9.9.063
    ecore >= 0.9.9.063
    ecore-evas >= 0.9.9.063
-   ecore-txt >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   embryo >= 0.9.9.063
    edje >= 0.9.93.063
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 \""; } >&5
   ($PKG_CONFIG --exists --print-errors "
@@ -14134,15 +14122,12 @@
    evas >= 0.9.9.063
    ecore >= 0.9.9.063
    ecore-evas >= 0.9.9.063
-   ecore-txt >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   embryo >= 0.9.9.063
    edje >= 0.9.93.063
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 ") 2>&5
   ac_status=$?
@@ -14154,15 +14139,12 @@
    evas >= 0.9.9.063
    ecore >= 0.9.9.063
    ecore-evas >= 0.9.9.063
-   ecore-txt >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   embryo >= 0.9.9.063
    edje >= 0.9.93.063
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 " 2>/dev/null`
 else
@@ -14189,15 +14171,12 @@
    evas >= 0.9.9.063
    ecore >= 0.9.9.063
    ecore-evas >= 0.9.9.063
-   ecore-txt >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   embryo >= 0.9.9.063
    edje >= 0.9.93.063
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 "`
         else
@@ -14207,15 +14186,12 @@
    evas >= 0.9.9.063
    ecore >= 0.9.9.063
    ecore-evas >= 0.9.9.063
-   ecore-txt >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   embryo >= 0.9.9.063
    edje >= 0.9.93.063
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 "`
         fi
@@ -14228,15 +14204,12 @@
    evas >= 0.9.9.063
    ecore >= 0.9.9.063
    ecore-evas >= 0.9.9.063
-   ecore-txt >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   embryo >= 0.9.9.063
    edje >= 0.9.93.063
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 ) were not met:
 
