Description: Fix the enna.desktop file
Author: Alexis Saettler <alexis@saettler.org>
Forwarded: yes
Origin: upstream, http://hg.geexbox.org/enna/diff/03efa4a8aea0/data/other/enna.desktop

Index: enna-0.4.0/data/other/enna.desktop
===================================================================
--- enna-0.4.0.orig/data/other/enna.desktop	2010-02-14 16:32:31.000000000 +0100
+++ enna-0.4.0/data/other/enna.desktop	2010-02-14 16:34:09.000000000 +0100
@@ -7,5 +7,5 @@
 TryExec=enna
 Exec=enna
 Terminal=false
-Categories=ENLIGHTENMENT;Application;AudioVideo;Player;
-StartupWmClass=enna
+Categories=Application;AudioVideo;Player;
+StartupWMClass=enna
