diff -Naur dosbox-0.72/src/dos/cdrom_image.cpp dosbox-0.72_patched/src/dos/cdrom_image.cpp
--- dosbox-0.72/src/dos/cdrom_image.cpp	2007-08-22 14:54:35.000000000 +0300
+++ dosbox-0.72_patched/src/dos/cdrom_image.cpp	2008-05-23 18:18:04.000000000 +0300
@@ -31,6 +31,8 @@
 #include "drives.h"
 #include "support.h"
 
+#include <limits>
+
 #if !defined(WIN32)
 #include <libgen.h>
 #else
diff -Naur dosbox-0.72/src/hardware/gameblaster.cpp dosbox-0.72_patched/src/hardware/gameblaster.cpp
--- dosbox-0.72/src/hardware/gameblaster.cpp	2007-01-10 12:17:37.000000000 +0200
+++ dosbox-0.72_patched/src/hardware/gameblaster.cpp	2008-05-23 18:22:31.000000000 +0300
@@ -25,6 +25,8 @@
 #include "setup.h"
 #include "pic.h"
 
+#include <cstring>
+
 #define LEFT	0x00
 #define RIGHT	0x01
 #define CMS_BUFFER_SIZE 128
diff -Naur dosbox-0.72/src/hardware/tandy_sound.cpp dosbox-0.72_patched/src/hardware/tandy_sound.cpp
--- dosbox-0.72/src/hardware/tandy_sound.cpp	2007-06-29 21:21:12.000000000 +0300
+++ dosbox-0.72_patched/src/hardware/tandy_sound.cpp	2008-05-23 18:28:48.000000000 +0300
@@ -29,6 +29,8 @@
 #include "pic.h"
 #include "dma.h"
 
+#include <cstring>
+
 #define DAC_CLOCK 3570000
 #define MAX_OUTPUT 0x7fff
 #define STEP 0x10000
diff -Naur dosbox-0.72/src/shell/shell_cmds.cpp dosbox-0.72_patched/src/shell/shell_cmds.cpp
--- dosbox-0.72/src/shell/shell_cmds.cpp	2007-08-17 20:58:46.000000000 +0300
+++ dosbox-0.72_patched/src/shell/shell_cmds.cpp	2008-05-23 18:31:00.000000000 +0300
@@ -28,6 +28,8 @@
 #include "../dos/drives.h"
 #include "support.h"
 
+#include <cstdlib>
+
 static SHELL_Cmd cmd_list[]={
 {	"CHDIR",	1,			&DOS_Shell::CMD_CHDIR,		"SHELL_CMD_CHDIR_HELP"},
 {	"CD",		0,			&DOS_Shell::CMD_CHDIR,		"SHELL_CMD_CHDIR_HELP"},
