diff -ur xorg-server-7.2.orig/Mesa-7.0/src/glx/x11/glxext.c xorg-server-7.2/Mesa-7.0/src/glx/x11/glxext.c
--- xorg-server-7.2.orig/Mesa-7.0/src/glx/x11/glxext.c	2006-11-15 16:55:47.000000000 +0200
+++ xorg-server-7.2/Mesa-7.0/src/glx/x11/glxext.c	2007-03-27 10:14:18.768661025 +0300
@@ -666,8 +666,9 @@
 	if ( do_delete && (m->visualID != 0) ) {
 	    do_delete = GL_FALSE;
 
-	    fprintf(stderr, "libGL warning: 3D driver claims to not support "
-		    "visual 0x%02x\n", m->visualID);
+		if (m->visualRating != GLX_NON_CONFORMANT_VISUAL_EXT || getenv("LIBGL_DEBUG"))
+			fprintf(stderr, "libGL warning: 3D driver claims to not support "
+				"visual 0x%02x\n", m->visualID);
 	}
 
 	if ( do_delete ) {
