--- configure	2007-02-10 00:09:36.000000000 +0200
+++ configure	2007-03-05 16:05:29.000000000 +0200
@@ -23509,35 +23509,12 @@
 { echo "$as_me:$LINENO: checking for BlueZ bluetooth support" >&5
 echo $ECHO_N "checking for BlueZ bluetooth support... $ECHO_C" >&6; }
 
-# Check whether --with-bluez was given.
-if test "${with_bluez+set}" = set; then
-  withval=$with_bluez;
-else
-  use_bluez=auto
-fi
-
-if test "x$use_bluez" != "xno"; then
-        if test "x$have_bluez" = xno && test "x$use_bluez" = xyes; then
-                { { echo "$as_me:$LINENO: error: BlueZ explicitly requested but no support found" >&5
-echo "$as_me: error: BlueZ explicitly requested but no support found" >&2;}
-   { (exit 1); exit 1; }; }
-        fi
-        if test "x$have_bluez" = xyes; then
-
 cat >>confdefs.h <<\_ACEOF
 #define HAVE_BLUEZ 1
 _ACEOF
 
-                { echo "$as_me:$LINENO: result: enabled" >&5
-echo "${ECHO_T}enabled" >&6; }
-                msg_bluez="yes"
-        else
-                { echo "$as_me:$LINENO: result: not found" >&5
-echo "${ECHO_T}not found" >&6; }
-        fi
-fi
-
-
+echo "$as_me:$LINENO: result: enabled" >&5
+msg_bluez="yes"
 
 if test x$have_bluez = "xyes"; then
   WITH_BLUEZ_TRUE=
