From ef1ca4800ee75044243d0eb2346240c50c28f0aa Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Fri, 15 Aug 2008 14:07:28 -0400
Subject: [PATCH] Uninclude xf86Version.h

---
 src/s3_driver.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/src/s3_driver.c b/src/s3_driver.c
index 86b336c..696e8d4 100644
--- xf86-video-s3-0.6.0/src/s3_driver.c
+++ xf86-video-s3-0.6.0/src/s3_driver.c
@@ -43,7 +43,6 @@
 #include "xf86_OSproc.h"
 #include "xf86Pci.h"
 #include "xf86PciInfo.h"
-#include "xf86Version.h"
 #include "xf86Resources.h"
 #include "xf86fbman.h"
 #include "xf86cmap.h"
-- 
1.5.6.5

