diff -ur Linux-PAM-0.99.6.3.orig/modules/pam_limits/pam_limits.c Linux-PAM-0.99.6.3/modules/pam_limits/pam_limits.c
--- Linux-PAM-0.99.6.3.orig/modules/pam_limits/pam_limits.c	2006-08-05 21:02:44.000000000 +0300
+++ Linux-PAM-0.99.6.3/modules/pam_limits/pam_limits.c	2008-01-10 03:29:56.000000000 +0200
@@ -50,6 +50,11 @@
 #define LIMITS_DEF_DEFAULT  4 /* limit was set by an default entry */
 #define LIMITS_DEF_NONE     5 /* this limit was not set yet */
 
+/* Hack to test new rlimit values */
+#define RLIMIT_NICE 13
+#define RLIMIT_RTPRIO 14
+#define RLIM_NLIMITS 15
+
 static const char *limits_def_names[] = {
        "USER",
        "GROUP",
