diff -Nur ltmodem-8.31_alpha10-alk8-old/modules/docs/ltmodem.rules ltmodem-8.31_alpha10-alk8/modules/docs/ltmodem.rules
--- ltmodem-8.31_alpha10-alk8-old/modules/docs/ltmodem.rules	2006-09-15 02:43:20.399740896 +0300
+++ ltmodem-8.31_alpha10-alk8/modules/docs/ltmodem.rules	2006-09-15 02:43:36.701262688 +0300
@@ -2,5 +2,5 @@
 # UDEV rule for ltmodem
 #  creates symlink /dev/modem to /dev/ttyLT?, and takes care of permissions
 
-KERNEL="ttyLTM[0-9]", NAME="%k", MODE="0660", GROUP="dialout", SYMLINK="modem"
+KERNEL=="ttyLTM[0-9]", NAME="%k", MODE="0660", GROUP="dialout", SYMLINK="modem"
 
