diff -Nur nut-2.0.5-old/configure.in nut-2.0.5/configure.in
--- nut-2.0.5-old/configure.in	2007-06-03 23:21:02.556225932 +0300
+++ nut-2.0.5/configure.in	2007-06-03 23:22:16.299015324 +0300
@@ -17,7 +17,7 @@
 dnl Pull this in for things that need to replace it later
 if (test -f include/version)
 then
-	NUT_VERSION=`head -1 include/version`
+	NUT_VERSION=`head -n 1 include/version`
 	AC_MSG_RESULT($NUT_VERSION)
 else
 	AC_MSG_ERROR([Can't read include/version - broken source distribution?])
