diff -Nur nut-2.0.5-old/drivers/snmp-ups.h nut-2.0.5/drivers/snmp-ups.h
--- nut-2.0.5-old/drivers/snmp-ups.h	2007-06-15 16:21:13.242492230 +0300
+++ nut-2.0.5/drivers/snmp-ups.h	2007-06-15 16:22:31.502594885 +0300
@@ -194,3 +194,8 @@
 const char *OID_pwr_status;
 int g_pwr_battery;
 int pollfreq; /* polling frequency */
+
+/* these symbols are referenced but not defined, blah */
+int allow_severity = LOG_DEBUG;
+int deny_severity = LOG_ERR;
+
