diff -Nur nut-2.0.5-old/drivers/snmp-ups.c nut-2.0.5/drivers/snmp-ups.c
--- nut-2.0.5-old/drivers/snmp-ups.c	2007-06-20 19:48:50.543550465 +0300
+++ nut-2.0.5/drivers/snmp-ups.c	2007-06-20 19:51:39.113923873 +0300
@@ -37,6 +37,7 @@
 #include "ietfmib.h"
 #include "mgemib.h"
 #include "netvisionmib.h"
+#include "nibumib.h"
 #include "pwmib.h"
 
 /* pointer to the Snmp2Nut lookup table */
@@ -113,7 +114,7 @@
 {
   upsdebugx(1, "entering upsdrv_makevartable()");
 
-  addvar(VAR_VALUE, SU_VAR_MIBS, "Set MIB compliance (default=ietf, allowed mge,apcc,netvision,pw)");
+  addvar(VAR_VALUE, SU_VAR_MIBS, "Set MIB compliance (default=ietf, allowed mge,apcc,netvision,pw,nibu)");
   addvar(VAR_VALUE | VAR_SENSITIVE, SU_VAR_COMMUNITY, "Set community name (default=public)");
   addvar(VAR_VALUE, SU_VAR_VERSION, "Set SNMP version (default=v1, allowed v2c)");
   addvar(VAR_VALUE, SU_VAR_POLLFREQ, "Set polling frequency in seconds, to reduce network flow (default=30)");
@@ -539,6 +540,11 @@
 		snmp_info = &pw_mib[0];
 		OID_pwr_status = "";
 /*             read_mibconf("pw"); */
+	} else if (!strcmp(mib, "nibu")) {
+		upsdebugx(1, "load_mib2nut: using nibu mib");
+		snmp_info = &nibu_mib[0];
+		OID_pwr_status = "";
+/*             read_mibconf("nibu"); */
 	} else {
 		upsdebugx(1, "load_mib2nut: using ietf (RFC 1628) mib");
 		snmp_info = &ietf_mib[0];
