diff -Nur nut-2.0.5-old/configure.in nut-2.0.5/configure.in
--- nut-2.0.5-old/configure.in	2007-06-03 23:24:00.711142268 +0300
+++ nut-2.0.5/configure.in	2007-06-03 23:24:25.320401092 +0300
@@ -45,7 +45,7 @@
 dnl fails to pass better values in --with arguments
 
 GD_INCLUDES=""
-GD_LIBS="-L/usr/X11R6/lib -lgd -lpng -lz -ljpeg -lfreetype -lm -lXpm -lX11"
+GD_LIBS="-lgd -lpng -lz -ljpeg -lfreetype -lm"
 
 dnl +-------------------------------------------------------------------
 
