diff -Nuar a2ps-4.14.orig/configure a2ps-4.14/configure
--- a2ps-4.14.orig/configure	2009-03-26 16:06:35.000000000 +0200
+++ a2ps-4.14/configure	2009-03-26 16:06:56.000000000 +0200
@@ -32064,7 +32064,7 @@
 if test "${a2_cv_com_acroread4+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  $acroread -helpall >conftest.help
+  acroread4 -helpall >conftest.help
   if fgrep -e -size   conftest.help >/dev/null &&
      fgrep -e -shrink conftest.help >/dev/null; then
      a2_cv_com_acroread4=yes
diff -Nuar a2ps-4.14.orig/configure.in a2ps-4.14/configure.in
--- a2ps-4.14.orig/configure.in	2009-03-26 16:06:35.000000000 +0200
+++ a2ps-4.14/configure.in	2009-03-26 16:06:56.000000000 +0200
@@ -327,7 +327,7 @@
 # If we have acroread, is it a version which supports -shrink etc.?
 if test "x$COM_acroread" = x; then
   AC_CACHE_CHECK(whether acroread is acroread 4, a2_cv_com_acroread4,
-  [$acroread -helpall >conftest.help
+  [acroread4 -helpall >conftest.help
   if fgrep -e -size   conftest.help >/dev/null &&
      fgrep -e -shrink conftest.help >/dev/null; then
      a2_cv_com_acroread4=yes
