--- runtime/runtime.h-dist	2006-08-24 18:12:29.000000000 +0200
+++ runtime/runtime.h	2006-08-24 18:13:12.000000000 +0200
@@ -46,6 +46,11 @@ static void _stp_dbug (char *func, int l
 #define kbug(args...) ;
 #endif /* DEBUG */
 
+/* 2.6.18 kernel has no longer for_each_cpu */
+#ifndef for_each_cpu
+#define for_each_cpu for_each_possible_cpu
+#endif
+
 /* atomic globals */
 static atomic_t _stp_transport_failures = ATOMIC_INIT (0);
 
