Index: actions.py
===================================================================
--- actions.py	(revision 85706)
+++ actions.py	(working copy)
@@ -1,79 +1,42 @@
 #!/usr/bin/python
 # -*- coding: utf-8 -*-
 #
-# Copyright 2005-2009 TUBITAK/UEKAE
+# Copyright 2005-2010 TUBITAK/UEKAE
 # Licensed under the GNU General Public License, version 2.
 # See the file http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt
 
-from pisi.actionsapi import autotools
+from pisi.actionsapi import crosstools
 from pisi.actionsapi import pisitools
 from pisi.actionsapi import shelltools
 from pisi.actionsapi import get
 
-_build="i686-pc-linux-gnu"
-_host="arm-cortex_a8-linux-gnueabi"
-_target=_host
-
-# ugly hard-coded stuff, unfortunately..
-_RootDir="/pardus-arm"
-_ToolchainDir = "/home/memre/x-tools"
-
-# Pardus-ARM preparation
-def prepare():
-    shelltools.export("LC_ALL", "C")
-    shelltools.export("CXXFLAGS", "")
-    shelltools.export("CFLAGS", "%s -I%s/usr/include \
-                       -L%s/usr/lib -L%s/lib \
-                       -I../include/linux/ " % \
-                       (get.CFLAGS(), _RootDir, _RootDir, _RootDir))
-    shelltools.export("LDFLAGS", "%s -L%s/usr/lib -L%s/lib" % (get.LDFLAGS(), _RootDir, _RootDir))
-
-    shelltools.export("CC",     "%s-gcc" % _target)
-    shelltools.export("CXX",    "%s-g++" % _target)
-    shelltools.export("AR",     "%s-ar"  % _target)
-    shelltools.export("AS",     "%s-as"  % _target)
-    shelltools.export("LD",     "%s-ld"  % _target)
-    shelltools.export("RANLIB", "%s-ranlib" % _target)
-    shelltools.export("STRIP",  "%s-strip"  % _target)
-    shelltools.export("LIBTOOL","%s-libtool" % _target)
-
 def setup():
-    # Pardus-ARM preparation
-    prepare()
+    crosstools.configure()
 
-    for dir in ("ip", "misc", "tc"):
-        #pisitools.dosed("%s/Makefile" % dir, "0755 -s", "0755")
-        shelltools.system("cp %s/Makefile{,.orig}; \
-                           sed 's/0755 -s/0755/' %s/Makefile.orig > %s/Makefile" % \
-                           (dir, dir, dir))
-
-    shelltools.system("cp misc/Makefile{,.orig}; \
-                       sed '/^TARGETS/s@arpd@@g' misc/Makefile.orig > misc/Makefile")
-
-    #pisitools.dosed("misc/Makefile", \
-                    #"^TARGETS=ss nstat ifstat rtacct arpd lnstat", \
-                    #"TARGETS=ss nstat ifstat rtacct lnstat")
-
-    pisitools.dosed("Makefile", "-O2", get.CFLAGS())
-    pisitools.dosed("tc/m_ipt.c", "/usr/local", "/usr")
-
 def build():
-    # Pardus-ARM preparation
-    prepare()
+    shelltools.export("LC_ALL", "C")
 
-    autotools.make('CC="%s" \
-                    RPM_OPT_FLAGS="%s" \
-                    DESTDIR="%s" \
-                    DOCDIR="%s" \
-                    ROOTDIR="%s"' \
-                    % (get.CC(), \
-                       get.CFLAGS(), \
-                       get.installDIR(),
-                       get.docDIR(), \
-                       _RootDir))
+    crosstools.environment["CXXFLAGS"] = ""
+    crosstools.environment["instDIR"] = get.installDIR()
+    crosstools.environment["workDIR"] = get.workDIR()
+    crosstools.environment["srcVER"] = get.srcVERSION()
+    crosstools.environment["CFLAGS"] = "-Du32=__u32 -O2 \
+                                        -I%(workDIR)s/iproute2-%(srcVER)s/include/linux \
+                                        -I%(workDIR)s/iproute2-%(srcVER)s/include/ \
+                                        -I%(RootDir)s/usr/include \
+                                        -L%(RootDir)s/usr/lib -L%(RootDir)s/lib" % crosstools.environment
+    crosstools.environment["LDFLAGS"] = get.LDFLAGS()
+    crosstools.environment["LDFLAGS"] = "%(LDFLAGS)s -L%(RootDir)s/usr/lib -L%(RootDir)s/lib"  % crosstools.environment
+    crosstools.prepare()
 
+    crosstools.make('CC="%(CC)s" \
+                     KERNEL_INCLUDE="%(RootDir)s/usr/include" \
+                     RPM_OPT_FLAGS="%(CFLAGS)s" \
+                     CFLAGS="%(CFLAGS)s" \
+                     LDFLAGS="%(LDFLAGS)s"' % crosstools.environment)
+
 def install():
-    autotools.rawInstall('DESTDIR="%s" \
+    crosstools.rawInstall('DESTDIR="%s" \
                           SBINDIR="/sbin" \
                           DOCDIR="/%s/%s" \
                           MANDIR="/usr/share/man" \
Index: pspec.xml
===================================================================
--- pspec.xml	(revision 85706)
+++ pspec.xml	(working copy)
@@ -15,7 +15,7 @@
         <Archive sha1sum="e7dec62a4d6646f3fded4f5d34718e9fd98615e4" type="tarbz2">http://devresources.linux-foundation.org/dev/iproute2/download/iproute2-2.6.31.tar.bz2</Archive>
         <BuildDependencies>
             <Dependency>iptables</Dependency>
-            <Dependency>linux-atm-lib</Dependency>
+            <Dependency>linux-atm</Dependency>
         </BuildDependencies>
         <Patches>
             <Patch level="1">man-pages.patch</Patch>
@@ -27,9 +27,8 @@
             <Patch level="1">iproute2-2.6.29-IPPROTO_IP_for_SA.patch</Patch>
             <Patch level="1">iproute2-example-cbq-service.patch</Patch>
 
-<!--             <Patch level="1">ldflags.patch</Patch> -->
             <Patch level="1">new-flex-fix.patch</Patch>
-            <Patch>ipaddrfix.patch</Patch>
+            <!--<Patch>ipaddrfix.patch</Patch>-->
         </Patches>
     </Source>
 
