diff -Nur goonies_r1-0-1-old/src/TheGoonies.cpp goonies_r1-0-1/src/TheGoonies.cpp
--- goonies_r1-0-1-old/src/TheGoonies.cpp	2007-08-25 18:06:33.473636945 +0300
+++ goonies_r1-0-1/src/TheGoonies.cpp	2007-08-25 18:10:05.120461738 +0300
@@ -100,7 +100,7 @@
 	// Load the room/music table:
 	{
 		FILE *fp;
-		fp=fopen("maps/musics.txt","r");
+		fp=fopen("/usr/share/goonies/maps/musics.txt","r");
 		if (fp!=0) {
 			int n;
 			char tmp[80],tmp2[80],tmp3[80];
@@ -127,7 +127,7 @@
 	// Load the room/fratelli table:
 	{
 		FILE *fp;
-		fp=fopen("maps/fratellies.txt","r");
+		fp=fopen("/usr/share/goonies/maps/fratellies.txt","r");
 		if (fp!=0) {
 			int n;
 			char tmp[80],tmp2[80];
@@ -235,7 +235,7 @@
 	// Load the room/fratelli table:
 	{
 		FILE *fp;
-		fp=fopen("maps/fratellies.txt","r");
+		fp=fopen("/usr/share/goonies/maps/fratellies.txt","r");
 		if (fp!=0) {
 			int n;
 			char tmp[80],tmp2[80];
@@ -284,7 +284,7 @@
 
 void TheGoonies::map_filename(char *fn,int level,int roomx,int roomy)
 {
-	sprintf(fn,"maps/level%i-%i-%i.gmp",level,roomx,roomy);
+	sprintf(fn,"/usr/share/goonies/maps/level%i-%i-%i.gmp",level,roomx,roomy);
 } /* TheGoonies::map_filename */ 
 
 
