diff -Nur goonies_r1-0-1-old/src/GLTManager.cpp goonies_r1-0-1/src/GLTManager.cpp
--- goonies_r1-0-1-old/src/GLTManager.cpp	2007-08-25 18:06:33.421644849 +0300
+++ goonies_r1-0-1/src/GLTManager.cpp	2007-08-25 18:12:39.566982298 +0300
@@ -64,7 +64,7 @@
 		char filename[256];
 		GLTile *t;
 
-		sprintf(filename,"graphics/%s.png",name->get());
+		sprintf(filename,"/usr/share/goonies/graphics/%s.png",name->get());
 		t=new GLTile(filename);	
 		t->set_clamp();
 		n=new GLTManagerNode();
@@ -153,7 +153,7 @@
 			n->m_tile=t=new GLTile(s);
 		} // if
 
-		sprintf(filename,"graphics/%s.nfo",name->get());
+		sprintf(filename,"/usr/share/goonies/graphics/%s.nfo",name->get());
 		{
 			FILE *fp;
 			fp=fopen(filename,"r");
@@ -210,7 +210,7 @@
 		char filename[256];
 		GLTile *t;
 
-		sprintf(filename,"graphics/%s.png",name->get());
+		sprintf(filename,"/usr/share/goonies/graphics/%s.png",name->get());
 		t=new GLTile(filename);
 		t->set_smooth();
 //		t->set_clamp();
diff -Nur goonies_r1-0-1-old/src/TheGooniesApp.cpp goonies_r1-0-1/src/TheGooniesApp.cpp
--- goonies_r1-0-1-old/src/TheGooniesApp.cpp	2007-08-25 18:06:33.473636945 +0300
+++ goonies_r1-0-1/src/TheGooniesApp.cpp	2007-08-25 18:12:54.003787570 +0300
@@ -109,9 +109,9 @@
 //	m_sfx_menu_select=Sound_create_sound("sound/sfx/cursor_select");
 //	m_sfx_menu_move=Sound_create_sound("sound/sfx/cursor_move");
 
-	font_extract("font","graphics/font.png",10+13+13+10+14+8,"1234567890abcdefghijklmnopqrstuvwxyzXXXXXXXXXX-./:\"#$%!?:;.,'`[]{|}ñ");
-	font_extract("font_hl","graphics/font_highlighted.png",10+13+13+10+14+8,"1234567890abcdefghijklmnopqrstuvwxyzXXXXXXXXXX-./:\"#$%!?:;.,'`[]{|}ñ");
-	font_extract("font_hud","graphics/font_hud.png",11,"0123456789-");
+	font_extract("font","/usr/share/goonies/graphics/font.png",10+13+13+10+14+8,"1234567890abcdefghijklmnopqrstuvwxyzXXXXXXXXXX-./:\"#$%!?:;.,'`[]{|}ñ");
+	font_extract("font_hl","/usr/share/goonies/graphics/font_highlighted.png",10+13+13+10+14+8,"1234567890abcdefghijklmnopqrstuvwxyzXXXXXXXXXX-./:\"#$%!?:;.,'`[]{|}ñ");
+	font_extract("font_hud","/usr/share/goonies/graphics/font_hud.png",11,"0123456789-");
 
 	m_test_game=0;
 	load_configuration();
