diff -Nur ufoai-2.0_rc5-old/source/ports/linux/snd_arts.c ufoai-2.0_rc5/source/ports/linux/snd_arts.c
--- ufoai-2.0_rc5-old/source/ports/linux/snd_arts.c	2006-09-29 04:51:48.268189598 +0300
+++ ufoai-2.0_rc5/source/ports/linux/snd_arts.c	2006-09-29 04:53:13.308261543 +0300
@@ -23,7 +23,7 @@
 
 */
 
-#include <artsc.h>
+#include <artsc/artsc.h>
 
 #include "../../client/client.h"
 #include "../../client/snd_loc.h"
