diff -Nur ufoai-2.0-RC6-source-old/client/snd_ref.c ufoai-2.0-RC6-source/client/snd_ref.c
--- ufoai-2.0-RC6-source-old/client/snd_ref.c	2006-12-04 17:58:17.444935784 +0200
+++ ufoai-2.0-RC6-source/client/snd_ref.c	2006-12-04 18:29:38.156024238 +0200
@@ -345,7 +345,7 @@
 		Com_Printf("not initializing.\n");
 	} else {
 		snd_volume = Cvar_Get("snd_volume", "0.7", CVAR_ARCHIVE, "Sound volume - default is 0.7");
-		snd_khz = Cvar_Get("snd_khz", "48", CVAR_ARCHIVE, "Khz value for sound renderer - default is 48");
+		snd_khz = Cvar_Get("snd_khz", "44", CVAR_ARCHIVE, "Khz value for sound renderer - default is 44");
 		snd_loadas8bit = Cvar_Get("snd_loadas8bit", "0", CVAR_ARCHIVE, NULL);
 		snd_mixahead = Cvar_Get("snd_mixahead", "0.2", CVAR_ARCHIVE, NULL);
 		snd_show = Cvar_Get("snd_show", "0", 0, NULL);
@@ -357,7 +357,7 @@
 
 		{
 			char fn[MAX_QPATH];
-			snd_ref = Cvar_Get("snd_ref", "sdl", CVAR_ARCHIVE, "Sound renderer libary name - default is sdl");
+			snd_ref = Cvar_Get("snd_ref", "arts", CVAR_ARCHIVE, "Sound renderer libary name - default is arts");
 			/* don't restart right again */
 			snd_ref->modified = qfalse;
 
