--- screenlets-manager.desktop.orig	2008-03-17 12:16:47.000000000 +0200
+++ desktop-menu/screenlets-manager.desktop	2008-03-17 12:19:26.000000000 +0200
@@ -1,14 +1,16 @@
 [Desktop Entry]
 Type=Application
-Categories=Settings;
+Categories=Application;Utility;
 Name=Screenlets
 Name[de]=Screenlets
 Name[et]=Screenlets
+Name[tr]=Screenlets
 Comment=A graphical tool to manage your Screenlets.
 Comment[de]=Ein grafisches Werkzeug, um Ihre Screenlets zu verwalten.
 Comment[et]=Graafiline tööriist Screenlets vidinate haldamiseks
+GenericName[tr]=Masaüstü Araçları
 Encoding=UTF-8
 Icon=/usr/share/icons/screenlets.svg
 Exec=/usr/share/screenlets-manager/screenlets-manager.py > /dev/null
-StartupNotify=true
+StartupNotify=false
 
