Index: folks-0.2.1/tests/telepathy/Makefile.am
===================================================================
--- folks-0.2.1.orig/tests/telepathy/Makefile.am
+++ folks-0.2.1/tests/telepathy/Makefile.am
@@ -46,6 +46,7 @@ contact_retrieval_SOURCES = \
 	$(NULL)
 
 contact_retrieval_LDADD = \
+	$(GEE_LIBS) \
 	$(top_builddir)/tests/lib/telepathy/contactlist/libtp-test-contactlist.la \
 	$(top_builddir)/folks/libfolks.la
 
Index: folks-0.2.1/tests/telepathy/Makefile.in
===================================================================
--- folks-0.2.1.orig/tests/telepathy/Makefile.in
+++ folks-0.2.1/tests/telepathy/Makefile.in
@@ -282,6 +282,7 @@ contact_retrieval_SOURCES = \
 	$(NULL)
 
 contact_retrieval_LDADD = \
+	$(GEE_LIBS) \
 	$(top_builddir)/tests/lib/telepathy/contactlist/libtp-test-contactlist.la \
 	$(top_builddir)/folks/libfolks.la
 
