--- src/qalculate_kde.desktop.orig	2008-09-04 15:17:14.000000000 +0300
+++ src/qalculate_kde.desktop	2008-09-04 15:19:31.000000000 +0300
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Name=Qalculate!
 GenericName=Scientific Calculator
+GenericName[tr]=Bilimsel Hesap makinası
 Exec=qalculate-kde
 Icon=qalculate_kde
 Type=Application
@@ -12,3 +13,4 @@
 X-DCOP-ServiceName=qalculate_kde
 X-KDE-StartupNotify=true
 Comment=Powerful and easy to use desktop calculator
+Comment[tr]=Güçlü ve kolay masaüstü hesap makinası
