diff -ur qalculate-kde-0.9.4.orig/src/qalculate_kde.desktop qalculate-kde-0.9.4/src/qalculate_kde.desktop
--- qalculate-kde-0.9.4.orig/src/qalculate_kde.desktop	2006-03-17 09:33:21.000000000 +0200
+++ qalculate-kde-0.9.4/src/qalculate_kde.desktop	2006-11-03 21:13:56.000000000 +0200
@@ -6,9 +6,9 @@
 Icon=qalculate_kde
 Type=Application
 DocPath=qalculate_kde/index.html
-Categories=Qt;KDE;Utility;Science;
+Categories=Qt;KDE;Utility;
 Terminal=false
 X-DCOP-ServiceType=Unique
 X-DCOP-ServiceName=qalculate_kde
 X-KDE-StartupNotify=true
-Comment=Powerful and easy to use desktop calculator
\ Dosya sonunda yenisatır yok.
+Comment=Powerful and easy to use desktop calculator
