diff -Nur knutclient-0.9.3-old/src/knutdock.h knutclient-0.9.3/src/knutdock.h
--- knutclient-0.9.3-old/src/knutdock.h	2008-04-08 15:09:54.000000000 +0300
+++ knutclient-0.9.3/src/knutdock.h	2008-04-08 15:10:57.000000000 +0300
@@ -322,7 +322,7 @@
  *
  * @since  0.1
  **/
-    void paintNumbers (int numbers, int x, int x, QPainter* p ); //x,y levy horni roh cisla 1
+    void paintNumbers (int numbers, int x1, int x2, QPainter* p ); //x,y levy horni roh cisla 1
 
 /**
  * @internal
