Index: src/via_driver.c
===================================================================
--- xf86-video-openchrome-0.2.902/src/via_driver.c (revision 557)
+++ xf86-video-openchrome-0.2.902/src/via_driver.c (revision 565)
@@ -601,4 +601,7 @@
         xf86Msg(X_NOTICE,
                 "For support, please refer to http://www.openchrome.org/.\n");
+#ifdef BUILDCOMMENT
+        xf86Msg(X_NOTICE, BUILDCOMMENT"\n");
+#endif
     }
     return scrn != NULL;
