Index: src/via_driver.c
===================================================================
--- xf86-video-openchrome-0.2.902/src/via_driver.c	(revision 562)
+++ xf86-video-openchrome-0.2.902/src/via_driver.c	(working copy)
@@ -817,12 +817,8 @@
             pVia->DRIIrqEnable = FALSE;
             break;
         case VIA_K8M800:
-            pVia->agpEnable = FALSE;
             pVia->DRIIrqEnable = FALSE;
             break;
-        case VIA_VM800:
-            pVia->agpEnable = FALSE;
-            break;
         case VIA_K8M890:
             pVia->VideoEngine = VIDEO_ENGINE_CME;
             pVia->agpEnable = FALSE;
