--- src/raster/raster.c	2004-12-05 23:12:15.000000000 +0200
+++ src/raster/raster.c	2007-05-25 22:20:04.000000000 +0300
@@ -379,6 +379,13 @@
 {
     raster_list_t *rlist;
 
+    // Lame but works for now, see http://landover.no-ip.com/128/viewtopic.php?pid=2280 -- cartman
+    static int magicfix=1;
+    if (magicfix) {
+        ActiveRasters = NULL;
+        magicfix=0;
+    }
+
     if (realize_canvas(raster) < 0)
         return -1;
 
