diff -urNad ghostscript-8.61.dfsg.1~/src/seexec.c ghostscript-8.61.dfsg.1/src/seexec.c
--- ghostscript-8.61.dfsg.1~/src/seexec.c	2007-09-25 14:31:24.000000000 +0100
+++ ghostscript-8.61.dfsg.1/src/seexec.c	2007-12-10 12:23:31.000000000 +0000
@@ -215,8 +215,10 @@
  * so that it will stay under the limit even after adding min_in_size
  * for a subsequent filter in a pipeline.  Note that we have to specify
  * a size of at least 128 so that filter_read won't round it up.
+ * The value of 132 is small enough for the sample file of the bug 689577 but
+ * still sufficient for comparefiles/fonttest.pdf .
  */
 const stream_template s_exD_template = {
-    &st_exD_state, s_exD_init, s_exD_process, 8, 200,
+    &st_exD_state, s_exD_init, s_exD_process, 8, 132,
     NULL, s_exD_set_defaults
 };
