--- src/main.cc	2007-03-29 14:29:45.000000000 +0300
+++ src/main.cc	2007-05-19 18:19:16.000000000 +0300
@@ -120,7 +120,7 @@
 	    "                              log is written to stdout.") << endl;
   cout << _(" -D, --daemon                 Run as daemon.") << endl;
   cout << _(" -s, --split=N                Download a file using N connections. N must be\n"
-	    "                              between 1 and 5. This option affects all URLs.\n"
+	    "                              between 1 and 500. This option affects all URLs.\n"
 	    "                              Thus, aria2 connects to each URL with\n"
 	    "                              N connections.\n"
 	    "                              Default: 1") << endl;
--- src/OptionHandlerFactory.cc	2007-03-27 19:16:44.000000000 +0300
+++ src/OptionHandlerFactory.cc	2007-05-19 18:20:37.000000000 +0300
@@ -77,7 +77,7 @@
   handlers.push_back(new DefaultOptionHandler(PREF_DIR));
   handlers.push_back(new DefaultOptionHandler(PREF_OUT));
   handlers.push_back(new LogOptionHandler(PREF_LOG));
-  handlers.push_back(new NumberOptionHandler(PREF_SPLIT, 1, 5));
+  handlers.push_back(new NumberOptionHandler(PREF_SPLIT, 1, 500));
   handlers.push_back(new NumberOptionHandler(PREF_TIMEOUT, 1, 600));
   handlers.push_back(new NumberOptionHandler(PREF_MAX_TRIES, 0));
   handlers.push_back(new BooleanOptionHandler(PREF_FTP_PASV));
