diff -up ./src/globals/globals.cxx.old ./src/globals/globals.cxx
--- ./src/globals/globals.cxx.old	2011-06-09 17:12:59.000000000 -0400
+++ ./src/globals/globals.cxx	2011-06-12 00:20:44.546248229 -0400
@@ -112,7 +112,10 @@ const struct mode_info_t mode_info[NUM_M
 
 	{ MODE_SSB, &ssb_modem, "SSB", "SSB", "", "SSB", "" },
 	{ MODE_WWV, &wwv_modem, "WWV", "WWV", "", "", "" },
-        { MODE_ANALYSIS, &anal_modem, "ANALYSIS", _("Freq Analysis"), "", "", "" }
+	// N.B. it is not valid to use an _("NLS") string in this table!!
+	// ... doing so will break the Fl_menu_item table 'menu_'.  -Kamal
+	// { MODE_ANALYSIS, &anal_modem, "ANALYSIS", _("Freq Analysis"), "", "", "" }
+        { MODE_ANALYSIS, &anal_modem, "ANALYSIS", "Freq Analysis", "", "", "" }
 };
 
 std::ostream& operator<<(std::ostream& s, const qrg_mode_t& m)
