Index: Makefile.in
===================================================================
--- Makefile.in.orig
+++ Makefile.in
@@ -47,7 +47,7 @@ distclean: clean
 
 DATE=$(shell date "+%B %Y")
 
-MAN1DIR = man/man1
+MAN1DIR = share/man/man1
 
 install: doxywizard_install 
 	$(INSTTOOL) -d $(INSTALL)/bin
Index: addon/doxywizard/Makefile.in
===================================================================
--- addon/doxywizard/Makefile.in.orig
+++ addon/doxywizard/Makefile.in
@@ -41,7 +41,7 @@ clean: Makefile.doxywizard 
 distclean: clean
 	$(RM) Makefile.doxywizard version.cpp config.l config.h portable.h portable.cpp
 
-MAN1DIR = man/man1
+MAN1DIR = share/man/man1
 
 install:
 	$(INSTTOOL) -d $(INSTALL)/bin	
