Index: desktop/recoll-searchgui.desktop
===================================================================
--- desktop/recoll-searchgui.desktop.orig
+++ desktop/recoll-searchgui.desktop
@@ -1,8 +1,10 @@
 [Desktop Entry]
-Categories=Utility;Filesystem;Database;
+Categories=Utility;Filesystem;Database;System;AudioVideo;
 Comment=Find documents by specifying search terms
+Comment[tr]=Belgeler içerisinde belirtilen ifadeleri arar
 Exec=recoll
 GenericName=Local Text Search
+GenericName[tr]=Yerel Metin Arayıcı
 Icon=recoll
 Name=Recoll
 Terminal=false
