--- index/Makefile.orig	2010-09-26 17:41:09.000000000 +0000
+++ index/Makefile	2010-11-25 01:04:30.000000000 +0000
@@ -12,7 +12,7 @@
 	       $(BSTATIC) $(LIBXAPIAN) $(LIBXAPIANSTATICEXTRA) \
 	       $(LIBICONV) $(BDYNAMIC) \
 	       $(LIBFAM) \
-	       $(X_LIBS) $(X_PRE_LIBS) $(X_LIBX11) $(X_EXTRA_LIBS) \
+	       $(X_LIBS) $(X_LIBX11) $(X_EXTRA_LIBS) \
 	       $(LIBSYS)
 recollindex.o : recollindex.cpp
 	$(CXX) $(ALL_CXXFLAGS) -c -o recollindex.o $<
