Index: telepathy-glib-0.10.6/telepathy-glib/telepathy-glib.pc.in
===================================================================
--- telepathy-glib-0.10.6.orig/telepathy-glib/telepathy-glib.pc.in
+++ telepathy-glib-0.10.6/telepathy-glib/telepathy-glib.pc.in
@@ -6,7 +6,7 @@ includedir=@includedir@
 Name: Telepathy-GLib
 Description: GLib utility library for the Telepathy framework
 Version: @VERSION@
-Requires: pkg-config >= 0.21
+Requires: 
 Requires.private: dbus-glib-1 >= 0.73, glib-2.0 >= 2.20, gobject-2.0 >= 2.20, gio-2.0 >= 2.20
 Libs: -L${libdir} -ltelepathy-glib
 Cflags: -I${includedir}/telepathy-1.0
