Index: kdelibs/kdeui/kbugreport.cpp
===================================================================
--- kdelibs.orig/kdeui/kbugreport.cpp
+++ kdelibs/kdeui/kbugreport.cpp
@@ -275,14 +275,7 @@ KBugReport::~KBugReport()
 
 void KBugReport::updateURL()
 {
-    KURL url ( "http://bugs.pearsoncomputing.net/enter_bug.cgi" );
-    url.addQueryItem( "product", "KDE 3.5" );
-    url.addQueryItem( "op_sys", d->os );
-    url.addQueryItem( "cf_kde_compiler", KDE_COMPILER_VERSION );
-    url.addQueryItem( "cf_kde_version", d->kde_version );
-    url.addQueryItem( "cf_kde_appversion", m_strVersion );
-    url.addQueryItem( "cf_kde_package", d->appcombo->currentText() );
-    url.addQueryItem( "cf_kde_kbugreport", "1" );
+    KURL url ( "http://bugs.pardus.org.tr/enter_bug.cgi" );
     d->url = url;
 }
 
