Index: xorg-video-7.2/xf86-video-intel-2.1.1/src/i830_accel.c
===================================================================
--- xorg-video-7.2.orig/xf86-video-intel-2.1.1/src/i830_accel.c
+++ xorg-video-7.2/xf86-video-intel-2.1.1/src/i830_accel.c
@@ -173,6 +173,8 @@ I830Sync(ScrnInfoPtr pScrn)
    if (I810_DEBUG & (DEBUG_VERBOSE_ACCEL | DEBUG_VERBOSE_SYNC))
       ErrorF("I830Sync\n");
 
+   if (!pScrn->vtSema) 
+     return; 
 #ifdef XF86DRI
    /* VT switching tries to do this.
     */
Index: xorg-video-7.2/xf86-video-intel-2.1.1/src/i810_accel.c
===================================================================
--- xorg-video-7.2.orig/xf86-video-intel-2.1.1/src/i810_accel.c
+++ xorg-video-7.2/xf86-video-intel-2.1.1/src/i810_accel.c
@@ -259,6 +259,8 @@ I810Sync(ScrnInfoPtr pScrn)
    if (I810_DEBUG & (DEBUG_VERBOSE_ACCEL | DEBUG_VERBOSE_SYNC))
       ErrorF("I810Sync\n");
 
+   if (!pScrn->vtSema) 
+     return; 
 #ifdef XF86DRI
    /* VT switching tries to do this.  
     */
