From b3ba121ed546e45d4840d57b4293f1d98f02509d Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Wed, 15 Aug 2007 18:28:50 +1000
Subject: [PATCH] intel: don't setup texOffsetStart unless using EXA

---
 src/i830_dri.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/i830_dri.c b/src/i830_dri.c
index ca1190c..0febb6c 100644
--- xf86-video-intel-2.1.1/src/i830_dri.c
+++ xf86-video-intel-2.1.1/src/i830_dri.c
@@ -602,6 +602,7 @@ I830DRIScreenInit(ScreenPtr pScreen)
 #endif
 #if DRIINFO_MAJOR_VERSION > 5 || \
     (DRIINFO_MAJOR_VERSION == 5 && DRIINFO_MINOR_VERSION >= 3)
+      if (pI830->useEXA)
 	 pDRIInfo->texOffsetStart = I830TexOffsetStart;
 #endif
 
-- 
1.5.3.5

