--- xf86-video-openchrome-0.2.900/src/via_driver.c.orig	2007-11-23 09:56:39.000000000 +0200
+++ xf86-video-openchrome-0.2.900/src/via_driver.c	2007-11-23 09:56:29.000000000 +0200
@@ -419,6 +419,7 @@
 };
 
 XF86ModuleData openchromeModuleData = {&VIAVersRec, VIASetup, NULL};
+XF86ModuleData viaModuleData = {&VIAVersRec, VIASetup, NULL};
 
 static pointer VIASetup(pointer module, pointer opts, int *errmaj, int *errmin)
 {
