diff -Naur imagination-2.1.orig/imagination.desktop.in imagination-2.1/imagination.desktop.in
--- imagination-2.1.orig/imagination.desktop.in	2010-04-17 00:53:35.753186626 +0300
+++ imagination-2.1/imagination.desktop.in	2010-04-17 01:06:34.012184829 +0300
@@ -2,7 +2,8 @@
 Type=Application
 Version=1.0
 _Name=Imagination
-_GenericName=A DVD slideshow maker
+_GenericName=DVD Slideshow Maker
+_GenericName[tr]=Slayt Şov Uygulaması
 _Comment=A lightweight DVD slideshow maker using GTK+2
 Exec=imagination %F
 Icon=imagination
diff -Naur imagination-2.1.orig/imagination.desktop.in.in imagination-2.1/imagination.desktop.in.in
--- imagination-2.1.orig/imagination.desktop.in.in	2010-04-17 00:53:35.753186626 +0300
+++ imagination-2.1/imagination.desktop.in.in	2010-04-17 01:06:52.651311652 +0300
@@ -2,7 +2,8 @@
 Type=Application
 Version=1.0
 _Name=Imagination
-_GenericName=A DVD slideshow maker
+_GenericName=DVD Slideshow Maker
+_GenericName[tr]=Slayt Şov Uygulaması
 _Comment=A lightweight DVD slideshow maker using GTK+2
 Exec=imagination %F
 Icon=imagination
