--- src/kxmame.desktop.orig	2008-10-08 16:02:48.000000000 +0300
+++ src/kxmame.desktop	2008-10-08 16:06:25.000000000 +0300
@@ -6,6 +6,9 @@
 Type=Application
 DocPath=kxmame/kxmame.html
 Categories=Qt;KDE;Game;ArcadeGame
-Comment=Arcade Machine Emulator Frontend
-Comment[it]=Frontend Emulatore Macchine Arcade
+Comment=Arcade Machine Emulator 
+Comment[tr]=Arcade Frontend Emülatör
+Comment[it]=Frontend Emulatore Machine Arcade
+GenericName= Arcade Frontend Machine Emulator
+GenericName[tr]= Arcade Emülatör
 Terminal=false
