Index: kdebluetooth4-0.3/src/trayicon.cpp
===================================================================
--- kdebluetooth4-0.3.orig/src/trayicon.cpp
+++ kdebluetooth4-0.3/src/trayicon.cpp
@@ -52,7 +52,7 @@ KBlueTray::KBlueTray(const QString& path
 	confGroup = new KConfigGroup(config, "General");
 	KConfigGroup obexServerConfig(config, "ObexServer");
 	if(!obexServerConfig.hasKey("savePath")) {
-		obexServerConfig.writeEntry("savePath", QDir::homePath() + "/.kbluetooth4");
+		obexServerConfig.writeEntry("savePath", QDir::homePath());
 		obexServerConfig.sync();
 	}
 	if(!obexServerConfig.hasKey("Autostart")) {
