--- src/cups/genppd.c	2007-03-02 14:01:15.000000000 +0200
+++ src/cups/genppd.c	2007-03-07 17:57:35.000000000 +0200
@@ -276,7 +276,9 @@
   const stp_printer_t	*p;		/* Printer driver */
   const char		*lang = "C";
   char			*s;
+#ifdef ENABLE_NLS
   char			**all_langs = getlangs();
+#endif
   char			filename[1024],		/* Filename */
 			ppd_location[1024];	/* Installed location */
 
--- src/cups/genppd.c	2007-03-07 18:00:21.000000000 +0200
+++ src/cups/genppd.c	2007-03-07 18:01:52.000000000 +0200
@@ -476,7 +476,9 @@
       verbose = 0;
       break;
     case 'c':
+#ifdef ENABLE_NLS
       baselocaledir = optarg;
+#endif
 #  ifdef DEBUG
       fprintf(stderr, "DEBUG: baselocaledir: %s\n", baselocaledir);
 #  endif
