--- yakuake/src/yakuake.desktop	2006-03-21 20:14:26.000000000 +0200
+++ yakuake/src/yakuake.desktop	2006-10-26 00:52:28.000000000 +0300
@@ -14,3 +14,4 @@
 Comment[pt_BR]=Um emulador de terminal no estilo do Quake, baseado na tecnologia do Konsole do KDE. 
 Comment[sv]=En Quake-liknande terminalemulator baserad på teknologi från KDE:s Terminal.
 Comment[tr]=KDE Konsole teknolojisi üzerine kurulu Quake stili bir terminal emülatörü
+X-KDE-StartupNotify=false
