--- cdrtools-2.01.01a33.orig/cdrecord/cdrecord.dfl
+++ cdrtools-2.01.01a33/cdrecord/cdrecord.dfl
@@ -6,8 +6,11 @@
 # 
 # The default device, if not specified elswhere
 #
-#CDR_DEVICE=yamaha
-CDR_DEVICE=default
+# Do not set CDR_DEVICE by default. If unset, cdrecord will
+# choose the correct drive, if only one is present!
+#
+# CDR_DEVICE=yamaha
+# CDR_DEVICE=default
 
 # 
 # The default speed, if not specified elswhere
