diff -Naur Fotowall-0.9.orig/3rdparty/posterazor/posterazor.pri Fotowall-0.9/3rdparty/posterazor/posterazor.pri
--- Fotowall-0.9.orig/3rdparty/posterazor/posterazor.pri	2009-12-19 15:01:39.429406552 +0200
+++ Fotowall-0.9/3rdparty/posterazor/posterazor.pri	2009-12-19 15:06:24.194284200 +0200
@@ -39,14 +39,14 @@
     posterazor.qrc
 
 TRANSLATIONS += \
-    de.ts \
-    en.ts \
-    es.ts \
-    fi.ts \
-    fr.ts \
-    it.ts \
-    nl.ts \
-    pl.ts \
-    pt_BR.ts \
-    pt_PT.ts \
-    zh_CN.ts
+    3rdparty/posterazor/de.ts \
+    3rdparty/posterazor/en.ts \
+    3rdparty/posterazor/es.ts \
+    3rdparty/posterazor/fi.ts \
+    3rdparty/posterazor/fr.ts \
+    3rdparty/posterazor/it.ts \
+    3rdparty/posterazor/nl.ts \
+    3rdparty/posterazor/pl.ts \
+    3rdparty/posterazor/pt_BR.ts \
+    3rdparty/posterazor/pt_PT.ts \
+    3rdparty/posterazor/zh_CN.ts
