--- xmlsec1-1.2.9/configure.in	2005-07-13 05:18:00.000000000 +0300
+++ xmlsec1-fixed/configure.in	2006-01-01 00:22:27.000000000 +0200
@@ -540,7 +540,7 @@
     AC_MSG_RESULT(no)
     NSS_FOUND="without"
 elif test "z$with_nss" = "z" -a "z$with_nspr" = "z" -a "z$with_mozilla_ver" = "z" -a "z$PKG_CONFIG_ENABLED" = "zyes" ; then
-    PKG_CHECK_MODULES(NSS, mozilla-nspr >= $MOZILLA_MIN_VERSION mozilla-nss >= $MOZILLA_MIN_VERSION,
+    PKG_CHECK_MODULES(NSS, nspr >= $MOZILLA_MIN_VERSION nss >= $MOZILLA_MIN_VERSION,
 	[NSS_FOUND=yes],
 	[NSS_FOUND=no])
 fi
