--- festival-1.4.3.orig/src/arch/festival/festival.cc
+++ festival-1.4.3/src/arch/festival/festival.cc
@@ -346,6 +351,7 @@
     if (mplayer_supported)
 	proclaim_module("mplayeraudio");
     
+#if 0 /* /usr/lib/festival/etc/machine -- ??? */
     // Add etc-dir path and machine specific directory etc/$OSTYPE
     char *etcdir = walloc(char,strlen(festival_libdir)+strlen("etc/")+
 			  strlen(FTOSTYPE)+3);
@@ -364,6 +370,7 @@
     
     wfree(etcdir);
     wfree(etcdircommon);
+#endif
     return;
 }
 
