--- src/arch/festival/festival.cc
+++ src/arch/festival/festival.cc
@@ -341,8 +341,6 @@
 	proclaim_module("freebsd16audio");
     if (linux16_supported)
 	proclaim_module("linux16audio");
-    if (macosx_supported)
-	proclaim_module("macosxaudio");
     if (win32audio_supported)
 	proclaim_module("win32audio");
     if (mplayer_supported)
--- src/modules/MultiSyn/TargetCostRescoring.cc
+++ src/modules/MultiSyn/TargetCostRescoring.cc
@@ -109,7 +109,7 @@
   for( EST_Litem *li = scores.head(); li != 0; li = next(li) ){
     float cand_dur = scores(li)._dur; 
     //    cerr << scores(li)._cand->score << " ";
-    scores(li)._cand->score += (mult * abs( cand_dur - meandur ) );
+    scores(li)._cand->score += (mult * abs(int( cand_dur - meandur ) ) );
     //    cerr << scores(li)._cand->score << endl;
   }
 }
