diff -Naur amsn-0.98.3.orig/plugins.tcl amsn-0.98.3/plugins.tcl
--- amsn-0.98.3.orig/plugins.tcl	2010-03-16 01:12:09.011146633 +0200
+++ amsn-0.98.3/plugins.tcl	2010-03-16 03:37:20.548902180 +0200
@@ -1313,7 +1313,7 @@
 			#logic: if this is the first time the user logged in, then 
 			#the configuration list is empty which means load_config
 			#makes loadedplugins empty
-			set loadedplugins [list "Nudge" "Winks" "Cam Shooter" "remind" "Notes" "ColoredNicks"]
+			set loadedplugins [list "Nudge" "Winks" "Cam Shooter" "remind" "Notes" "ColoredNicks" "Desktop Integration"]
 
 			status_log "Plugins System: load_config: No plugins.xml]\n" red
 		}
