diff -Naur amsn-0.98.3.orig/config.tcl amsn-0.98.3/config.tcl
--- amsn-0.98.3.orig/config.tcl	2010-03-16 01:12:09.013147097 +0200
+++ amsn-0.98.3/config.tcl	2010-03-16 01:40:05.551148027 +0200
@@ -175,8 +175,8 @@
 
 		::config::setKey encoding auto						;#ANSN encoding
 
-		::config::setKey textsize 2							;#User text size
-		::config::setKey mychatfont "{Helvetica} {} 000000"	;#User chat font
+       ::config::setKey textsize 0                         ;#User text size
+       ::config::setKey mychatfont "{DeJavu\ Sans} {} 000000"  ;#User chat font
 		::config::setKey winchatsize "350x320"		;#Default chat window size
 		::config::setKey winchatoutheight "200"		;#Default chat window output height
 		::config::setKey savechatwinsize 1			;#Save chat window sizes when changed?
@@ -267,7 +267,7 @@
 		::config::setKey enablebanner 1		;#Show or Hide AMSN Banner (By default Show)
 		::config::setKey startontray 0		;#Start amsn on tray icon only (hide contact list)
 		::config::setKey storename 1			;#Store original nick in a variable when go to custom states to revert it when go back
-		::config::setKey strictfonts 0		;#Use strict fonts' size in _ALL_ AMSN's fonts (Disabled by default)
+       ::config::setKey strictfonts 1      ;#Use strict fonts' size in _ALL_ AMSN's fonts (Disabled by default)
 		if {[OnMaemo] } {
 			::config::setKey sngdblclick 1		;#Use single or double click to open a message window (0 double, 1 single)
 		} else {
@@ -599,7 +599,7 @@
 		} elseif { [OnWin] } {
 			setGlobalKey basefont [list Arial 10 normal]
 		} else {
-			setGlobalKey basefont [list Helvetica 11 normal]	;#AMSN base font
+           setGlobalKey basefont [list DeJavu\ Sans 10 normal] ;#AMSN base font
 		}
 		setGlobalKey dpi 0	;# The screen's DPI if we want to force it. 0 to let it use default
 
