--- imagination.desktop.in.orig	2011-04-23 22:43:00.829000006 +0300
+++ imagination.desktop.in	2011-04-23 22:44:58.970000006 +0300
@@ -3,9 +3,11 @@
 Version=1.0
 _Name=Imagination
 _GenericName=A DVD slideshow maker
+GenericName[tr]=Slayt gösteri uygulaması
 _Comment=A lightweight DVD slideshow maker using GTK+2
+Comment[tr]=Hafif, GTK+2 tabanlı DVD slayt gösteri oluşturma uygulaması
 Exec=imagination %F
 Icon=imagination
 Terminal=false
-Categories=AudioVideo;
+Categories=GTK;Application;Graphics;
 StartupNotify=true
