--- CMakeLists.txt.orig	2008-01-31 21:01:24.000000000 +0200
+++ CMakeLists.txt	2008-01-31 21:01:43.000000000 +0200
@@ -130,14 +130,7 @@
 if (MINGW)
       set ( USE_PORTAUDIO 1 )
 else (MINGW)
-   PKGCONFIG (portaudio-2.0 19 PORTAUDIO_INCDIR PORAUDIO_LIBDIR PORTAUDIO_LIB PORTAUDIO_CPP )
-   if (PORTAUDIO_INCDIR)
-	message("portaudio detected")
-      set ( USE_PORTAUDIO 1 )
-   else (PORTAUDIO_INCDIR)
-      message("optional package portaudio-2.0 Version 19 not found (package portaudio19-dev)\n")
       set ( USE_PORTAUDIO 0 )
-   endif (PORTAUDIO_INCDIR)
 endif (MINGW)
 
 # produce config.h file
