diff -Nuar artha-1.0.2.orig//data/artha.desktop artha-1.0.2/data/artha.desktop
--- artha-1.0.2.orig//data/artha.desktop	2011-05-12 12:05:54.840999993 +0300
+++ artha-1.0.2/data/artha.desktop	2011-05-12 12:06:23.348999993 +0300
@@ -2,7 +2,9 @@
 Version=1.0
 Name=Artha
 GenericName=Thesaurus
+GenericName[tr]=Eşanlamlılar Sözlüğü
 Comment=Handy thesaurus based on WordNet
+Comment[tr]=Wordnet tabanlı kullanışlı eşanlamlılar sözlüğü
 Exec=artha
 Icon=artha
 Terminal=false
