Index: newsbeuter-2.3/src/controller.cpp
===================================================================
--- newsbeuter-2.3.orig/src/controller.cpp
+++ newsbeuter-2.3/src/controller.cpp
@@ -38,7 +38,7 @@
 #include <sys/stat.h>
 #include <pwd.h>
 
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 
 #include <libxml/xmlversion.h>
 #include <libxml/parser.h>
Index: newsbeuter-2.3/src/view.cpp
===================================================================
--- newsbeuter-2.3.orig/src/view.cpp
+++ newsbeuter-2.3/src/view.cpp
@@ -40,7 +40,7 @@
 #include <config.h>
 #include <sys/param.h>
 #include <string.h>
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 
 
 extern "C" {
