diff -Naur kdocker-4.0.orig/kdocker.pro kdocker-4.0/kdocker.pro
--- kdocker-4.0.orig/kdocker.pro	2009-09-16 22:05:54.398192945 +0300
+++ kdocker-4.0/kdocker.pro	2009-09-16 22:23:47.634317680 +0300
@@ -7,7 +7,7 @@
 LIBPATH += /usr/X11/lib
 LIBS = -lX11 -lXpm -lXmu
 
-INSTALL_PATH = /usr/local/share/kdocker
+INSTALL_PATH = /usr/share/kdocker
 
 ICONS_PATH = $$INSTALL_PATH/icons
 icons.path = $$ICONS_PATH
@@ -16,7 +16,7 @@
 desktop.path = /usr/share/applications
 desktop.files = kdocker.desktop
 
-target.path = /usr/local/bin
+target.path = /usr/bin
 
 INSTALLS += target icons desktop
 
