--- build/photofilmstrip.desktop.orig	2010-03-14 14:59:50.000000000 +0200
+++ build/photofilmstrip.desktop	2010-04-26 00:32:09.375931169 +0300
@@ -3,11 +3,13 @@
 Encoding=UTF-8
 Name=PhotoFilmStrip
 GenericName=Slideshow Creator
+GenericName[tr]=Slayt oluşturucu
 Comment=Create animated slideshows
+Comment[tr]=Slayt gösterisi oluşturabilirsiniz
 Exec=photofilmstrip
 Icon=/usr/share/pixmaps/photofilmstrip.xpm
 Terminal=false
 Type=Application
 StartupNotify=false
 Categories=AudioVideo;
-OnlyShowIn=GNOME;XFCE;
+OnlyShowIn=GNOME;XFCE;KDE
