Index: k9copy-2.3.7-Source/k9copy_assistant.desktop
===================================================================
--- k9copy-2.3.7-Source.orig/k9copy_assistant.desktop
+++ k9copy-2.3.7-Source/k9copy_assistant.desktop
@@ -1,5 +1,6 @@
 #!/usr/bin/env xdg-open
 [Desktop Entry]
+X-KDE-Solid-Predicate=[[ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Data|VideoDvd' ] OR [ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Data|VideoDvd' ]]
 Encoding=UTF-8
 Type=Application
 Name=k9copy assistant
@@ -13,7 +13,6 @@ GenericName=Video DVD Backup
 GenericName[fr]=Backup de DVD Video
 GenericName[tr]=DVD Vidyo Yedekleme Aracı
 GenericName[ru]=Копирование с видео-DVD
-MimeTypes=
 Terminal=false
 Icon=k9copy
 X-DocPath=k9copy/index.html
@@ -17,4 +18,4 @@ Terminal=false
 Icon=k9copy
 X-DocPath=k9copy/index.html
 Categories=Qt;KDE;AudioVideo;DiscBurning;
-X-KDE-StartupNotify=true
\ Dosya sonunda yenisatır yok.
+X-KDE-StartupNotify=true
Index: k9copy-2.3.7-Source/k9copy_assistant_open.desktop
===================================================================
--- k9copy-2.3.7-Source.orig/k9copy_assistant_open.desktop
+++ k9copy-2.3.7-Source/k9copy_assistant_open.desktop
@@ -1,3 +1,9 @@
+[Desktop Entry]
+X-KDE-Solid-Predicate=[[ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Data|VideoDvd' ] OR [ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Data|VideoDvd' ]]
+Type=Service
+Actions=open;
+Name=Copy with k9copy (wizard)
+
 [Desktop Action open]
 Exec=k9copy --input %f --assistant
 Icon=k9copy
@@ -6,8 +12,12 @@ Name[fr]=Copier avec k9copy (wizard)
 Name[de]=Kopieren mit k9copy (wizard)
 Name[tr]=k9copy ile kopyala (wizard)
 Name[ru]=Копировать при помощи k9copy (Мастер)
-
-[Desktop Entry]
-X-KDE-Solid-Predicate=[[ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Data|VideoDvd' ] OR [ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Data|VideoDvd' ]]
-Type=Service
-Actions=open;
\ Dosya sonunda yenisatır yok.
+Comment=DVD9 to DVD5 backup tool
+Comment[fr]=Backup de DVD9 en DVD5
+Comment[tr]=DVD9 DVD5 yedekleme aracı
+Comment[ru]=Инструмент для копирования с DVD9 на DVD5
+GenericName=Copy with k9copy (wizard)
+GenericName[fr]=Copier avec k9copy (wizard)
+GenericName[de]=Kopieren mit k9copy (wizard)
+GenericName[tr]=k9copy ile kopyala (wizard)
+GenericName[ru]=Копировать при помощи k9copy (Мастер)
Index: k9copy-2.3.7-Source/k9copy.desktop
===================================================================
--- k9copy-2.3.7-Source.orig/k9copy.desktop
+++ k9copy-2.3.7-Source/k9copy.desktop
@@ -1,5 +1,6 @@
 #!/usr/bin/env xdg-open
 [Desktop Entry]
+X-KDE-Solid-Predicate=[[ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Data|VideoDvd' ] OR [ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Data|VideoDvd' ]]
 Encoding=UTF-8
 Type=Application
 Name=k9copy
@@ -13,7 +13,6 @@ GenericName=Video DVD Backup
 GenericName[fr]=Backup de DVD Video
 GenericName[tr]=DVD Vidyo Yedekleme Aracı
 GenericName[ru]=Копирование с видео-DVD
-MimeTypes=
 Terminal=false
 Icon=k9copy
 X-DocPath=k9copy/index.html
@@ -17,4 +18,4 @@ Terminal=false
 Icon=k9copy
 X-DocPath=k9copy/index.html
 Categories=Qt;KDE;AudioVideo;DiscBurning;
-X-KDE-StartupNotify=true
\ Dosya sonunda yenisatır yok.
+X-KDE-StartupNotify=true
Index: k9copy-2.3.7-Source/k9copy_open.desktop
===================================================================
--- k9copy-2.3.7-Source.orig/k9copy_open.desktop
+++ k9copy-2.3.7-Source/k9copy_open.desktop
@@ -1,3 +1,9 @@
+[Desktop Entry]
+X-KDE-Solid-Predicate=[[ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Data|VideoDvd' ] OR [ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Data|VideoDvd' ]]
+Type=Service
+Actions=open;
+Name=Copy with k9copy
+
 [Desktop Action open]
 Exec=k9copy --input %f 
 Icon=k9copy
@@ -7,7 +13,3 @@ Name[de]=Kopieren mit k9copy
 Name[tr]=k9copy ile kopyala
 Name[ru]=Копировать при помощи k9copy
 
-[Desktop Entry]
-X-KDE-Solid-Predicate=[[ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Data|VideoDvd' ] OR [ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Data|VideoDvd' ]]
-Type=Service
-Actions=open;
\ Dosya sonunda yenisatır yok.
