diff -Nur et131x-1.2.3-2-old/et131x_netdev.c et131x-1.2.3-2/et131x_netdev.c
--- et131x-1.2.3-2-old/et131x_netdev.c	2008-03-22 20:48:21.000000000 +0200
+++ et131x-1.2.3-2/et131x_netdev.c	2008-03-22 20:50:33.000000000 +0200
@@ -1605,7 +1605,7 @@
 
     if( pAdapter->vlgrp )
     {
-        pAdapter->vlgrp->vlan_devices_arrays[vid] = NULL;    
+        vlan_group_set_device(pAdapter->vlgrp, vid, NULL);
     }
 
 
