diff -Nur rt2570-cvs-2007040503-old/Module/rt2570sw.h rt2570-cvs-2007040503/Module/rt2570sw.h
--- rt2570-cvs-2007040503-old/Module/rt2570sw.h	2007-04-05 14:22:23.037060347 +0300
+++ rt2570-cvs-2007040503/Module/rt2570sw.h	2007-04-05 14:32:56.442768043 +0300
@@ -120,7 +120,6 @@
 	{ USB_DEVICE(0x0411, 0x0067)},	/* Melco */		\
 	{ USB_DEVICE(0x050d, 0x7050)},	/* Belkin */		\
 	{ USB_DEVICE(0x050d, 0x7051)},	/* Belkin */		\
-	{ USB_DEVICE(0x050d, 0x705a)},  /* Belkin */		\
 	{ USB_DEVICE(0x06f8, 0xe000)}, /* GUILLEMOT */		\
 	{ USB_DEVICE(0x0707, 0xee13)},	/* SMC */		\
 	{ USB_DEVICE(0x0b05, 0x1706)},	/* ASUS */		\
@@ -136,7 +135,6 @@
 	{ USB_DEVICE(0x13b1, 0x001a)},  /* Cisco Systems */	\
 	{ USB_DEVICE(0x148f, 0x1706)},	/* Ralink */		\
 	{ USB_DEVICE(0x148f, 0x2570)},	/* Ralink */		\
-	{ USB_DEVICE(0x148f, 0x2573)},	/* CNET CWD-854 */	\
 	{ USB_DEVICE(0x148f, 0x9020)},	/* Ralink */		\
 	{ USB_DEVICE(0x14b2, 0x3c02)},	/* Conceptronic */	\
 	{ USB_DEVICE(0x14f8, 0x2570)},	/* Eminent */		\
