--- GCC/config_cmake/gcc_platform.sh~	2007-12-19 06:34:51.000000000 -0600
+++ GCC/config_cmake/gcc_platform.sh	2007-12-19 06:35:45.000000000 -0600
@@ -42,7 +42,7 @@
 SET(extra_options ${extra_options})
 SET(c_target_objs ${c_target_objs})
 SET(cxx_target_objs ${cxx_target_objs})
-SET(target_cpu_default ${target_cpu_default})
+SET(target_cpu_default "${target_cpu_default}")
 SET(out_host_hook_obj ${out_host_hook_obj})
 EOF
 