diff -Naur kamoso.orig/src/kamoso.desktop kamoso/src/kamoso.desktop
--- kamoso.orig/src/kamoso.desktop	2009-09-10 02:15:08.532057614 +0300
+++ kamoso/src/kamoso.desktop	2009-09-10 02:28:28.669416310 +0300
@@ -1,7 +1,9 @@
 [Desktop Entry]
 Name=Kamoso
 GenericName=Picture retriever
+GenericName[tr]=Fotoğraf Çekme Uygulaması
 Comment=Take any picture with your web cam
+Comment[tr]=Web kameranızla fotoğraf çekin
 Exec=kamoso
 MimeType=application/x-kamoso;
 Icon=kamoso
