--- bookreader.desktop	2006-05-12 21:44:36.000000000 +0300
+++ src/bookreader.desktop	2007-06-30 00:38:27.000000000 +0300
@@ -4,6 +4,8 @@
 Exec=bookreader %i %m -caption "%c"
 Icon=bookreader
 Type=Application
+GenericName=E-book Reader
+GenericName[tr]=E-kitap Okuyucu
 DocPath=bookreader/bookreader.html
 Comment=A KDE KPart Application
 Comment[br]=Ur meziant Kpart evit KDE
@@ -35,5 +37,6 @@
 Comment[sv]=Ett KDE Kpart-program
 Comment[ta]=கெடி கெபார்ட்  பயன்பாடு 
 Comment[tg]=Ба кор андохтани KPart барои KDE
+Comment[tr]=E-kitapları okumayı kolaylaştıran bir uygulama
 Comment[zh_CN]=一个 KDE KPart 应用程序
 Terminal=false
