diff -ur bookreader-0.2/src/bookmarksdlg.h bookreader-0.2-patched/src/bookmarksdlg.h
--- bookreader-0.2/src/bookmarksdlg.h	2006-09-24 03:58:09.000000000 +0300
+++ bookreader-0.2-patched/src/bookmarksdlg.h	2008-08-26 10:19:44.000000000 +0300
@@ -24,6 +24,8 @@
 #include <qdialog.h>
 #include "bookmark.h"
 
+#include <algorithm>
+
 class QVBoxLayout;
 class QHBoxLayout;
 class QGridLayout;
diff -ur bookreader-0.2/src/bookwidget.h bookreader-0.2-patched/src/bookwidget.h
--- bookreader-0.2/src/bookwidget.h	2007-07-04 01:28:29.000000000 +0300
+++ bookreader-0.2-patched/src/bookwidget.h	2008-08-26 10:20:15.000000000 +0300
@@ -29,6 +29,8 @@
 #include "renderer.h"
 #include "bookmark.h"
 
+#include <algorithm>
+
 class QStringList;
 class QRect;
 class KProgressDialog;
diff -ur bookreader-0.2/src/renderer.h bookreader-0.2-patched/src/renderer.h
--- bookreader-0.2/src/renderer.h	2007-07-04 01:28:29.000000000 +0300
+++ bookreader-0.2-patched/src/renderer.h	2008-08-26 10:19:56.000000000 +0300
@@ -27,6 +27,7 @@
 #include <qsize.h>
 #include <qtimer.h>
 #include <vector>
+#include <memory>
 class QStringList;
 
 class TextLine
