Index: glibc-2.10.1/localedata/Makefile
===================================================================
--- glibc-2.10.1.orig/localedata/Makefile
+++ glibc-2.10.1/localedata/Makefile
@@ -209,7 +209,7 @@ INSTALL-SUPPORTED-LOCALES=$(addprefix in
 
 # Sometimes the whole collection of locale files should be installed.
 LOCALEDEF=I18NPATH=. GCONV_PATH=$(common-objpfx)iconvdata LC_ALL=C \
-$(common-objpfx)elf/ld.so --library-path $(rpath-link) $(common-objpfx)locale/localedef --no-archive
+localedef --no-archive
 install-locales: $(INSTALL-SUPPORTED-LOCALES)
 
 install-locales-dir:
