Index: dependencies/arts/qtmcop/Makefile
===================================================================
--- dependencies.orig/arts/qtmcop/Makefile
+++ dependencies/arts/qtmcop/Makefile
@@ -732,7 +732,7 @@ uninstall-am: uninstall-artsincludeHEADE
 
 #>+ 3
 qiomanager_p.moc: $(srcdir)/qiomanager_p.h
-	$(TQTMOC) $(MOC) $(srcdir)/qiomanager_p.h qiomanager_p.moc
+	$(TQTMOC) $(MOC) -o qiomanager_p.moc $(srcdir)/qiomanager_p.h
 
 #>+ 2
 mocs: qiomanager_p.moc
