--- apps/nsplugins/kcm_nsplugins.cpp.sav	2008-11-20 16:04:52.000000000 +0100
+++ apps/nsplugins/kcm_nsplugins.cpp	2009-10-06 16:02:03.000000000 +0200
@@ -86,7 +86,7 @@ extern "C"
     KDE_EXPORT void kcminit_nsplugin()
     {
         KConfigGroup config(KSharedConfig::openConfig( "kcmnspluginrc", KConfig::NoGlobals ), "Misc");
-        if( true ) // I don't think this needs to be configurable now
+        if( config.readEntry( "StartupCheck", true ))
         {
             bool update = false;
             QStringList searchPaths = getSearchPaths();
