--- fltk-1.1.x-r5750/configure.in.rpath	2007-02-10 15:21:02.000000000 -0600
+++ fltk-1.1.x-r5750/configure.in	2007-04-29 12:47:29.000000000 -0500
@@ -180,9 +180,9 @@
             GLDSONAME="libfltk_gl.so.$FL_API_VERSION"
             IMGDSONAME="libfltk_images.so.$FL_API_VERSION"
             DSOCOMMAND="\$(CXX) \$(DSOFLAGS) -Wl,-soname,\$@ \$(LDLIBS) -shared -fPIC $DEBUGFLAG -o"
-	    if test "x$libdir" != "x/usr/lib"; then
-		    DSOLINK="-Wl,-rpath,$libdir"
-            fi
+#	    if test "x$libdir" != "x/usr/lib"; then
+#		    DSOLINK="-Wl,-rpath,$libdir"
+#	    fi
 	    ;;
 	AIX*)
             DSONAME="libfltk_s.a"
--- fltk-1.1.x-r5750/configure.rpath	2007-03-23 00:00:35.000000000 -0500
+++ fltk-1.1.x-r5750/configure	2007-04-29 12:47:59.000000000 -0500
@@ -1482,9 +1482,9 @@
             GLDSONAME="libfltk_gl.so.$FL_API_VERSION"
             IMGDSONAME="libfltk_images.so.$FL_API_VERSION"
             DSOCOMMAND="\$(CXX) \$(DSOFLAGS) -Wl,-soname,\$@ \$(LDLIBS) -shared -fPIC $DEBUGFLAG -o"
-	    if test "x$libdir" != "x/usr/lib"; then
-		    DSOLINK="-Wl,-rpath,$libdir"
-            fi
+#	    if test "x$libdir" != "x/usr/lib"; then
+#		    DSOLINK="-Wl,-rpath,$libdir"
+#	    fi
 	    ;;
 	AIX*)
             DSONAME="libfltk_s.a"
