--- data/Makefile.am.orig	2007-12-24 01:48:19.000000000 +0200
+++ data/Makefile.am	2009-02-07 23:14:02.000000000 +0200
@@ -13,10 +13,6 @@
 		-e 's,[@]PACKAGE[@],$(PACKAGE),g' \
 	 <$(srcdir)/minbar.schemas.in.in >minbar.schemas.in
 
-install-data-local:
-	GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) gconftool-2 --makefile-install-rule  minbar.schemas 2>&1 > /dev/null
-	gconftool-2 --shutdown
-
 uninstall-local:
 	GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) gconftool-2 --makefile-uninstall-rule minbar.schemas
 
