diff -Naur usbmanager.orig/data/usbmanager.desktop usbmanager/data/usbmanager.desktop
--- usbmanager.orig/data/usbmanager.desktop	2010-03-14 17:31:06.423366792 +0200
+++ usbmanager/data/usbmanager.desktop	2010-03-14 17:32:44.472388526 +0200
@@ -1,16 +1,14 @@
 [Desktop Entry]
 Name=USBManager
+GenericName=Storage Device Manager
+GenericName[tr]=Hafıza Aygıtları Yöneticisi
 Comment=A simple USB storage device manager
+Comment[tr]=USB hafıza aygıtı yöneticisi
 Comment[pt_BR]=Um gerenciador simples de dispositivos de armazenamento USB
 TryExec=usbmanager
 Exec=usbmanager
-Icon=/usr/share/icons/hicolor/64x64/apps/usbmanager.png
+Icon=usbmanager
 Type=Application
-X-GNOME-DocPath=
-X-GNOME-Bugzilla-Bugzilla=
-X-GNOME-Bugzilla-Product=
-X-GNOME-Bugzilla-Component=
-X-GNOME-Bugzilla-Version=
 Categories=GNOME;GTK;Utility;System;
 StartupNotify=true
 X-Ubuntu-Gettext-Domain=usbmanager
