diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kdebase-4.2.2-orig/apps/dolphin/src/dolphin_generalsettings.kcfg kdebase-4.2.2/apps/dolphin/src/dolphin_generalsettings.kcfg
--- kdebase-4.2.2-orig/apps/dolphin/src/dolphin_generalsettings.kcfg	2009-04-12 05:41:03.000000000 +0300
+++ kdebase-4.2.2/apps/dolphin/src/dolphin_generalsettings.kcfg	2009-04-12 05:52:23.000000000 +0300
@@ -38,7 +38,7 @@
         </entry>
         <entry name="RenameInline" type="Bool">
             <label context="@label">Rename inline</label>
-            <default>false</default>
+            <default>true</default>
         </entry>
         <entry name="ShowSelectionToggle" type="Bool">
             <label context="@label">Show selection toggle</label>
@@ -46,7 +46,7 @@
         </entry>
         <entry name="ShowCopyMoveMenu" type="Bool">
             <label context="@label">Show 'Copy To' and 'Move To' commands in context menu</label>
-            <default>false</default>
+            <default>true</default>
         </entry>
         <entry name="ViewPropsTimestamp" type="DateTime" >
             <label context="@label">Timestamp since when the view properties are valid</label>
@@ -65,7 +65,7 @@
         </entry>
         <entry name="ShowSpaceInfo" type="Bool">
             <label context="@label">Show the space information in the statusbar</label>
-            <default>false</default>
+            <default>true</default>
         </entry>
     </group>
 </kcfg>
