--- locale/Makefile.in.orig	2007-12-13 18:14:14.000000000 +0200
+++ locale/Makefile.in	2008-01-01 03:16:10.000000000 +0200
@@ -144,7 +144,7 @@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-locale_DATA = de.dat en.dat fr.dat hu.dat it.dat pl.dat sk.dat sr.dat
+locale_DATA = de.dat en.dat fr.dat hu.dat it.dat pl.dat sk.dat sr.dat tr.dat
 EXTRA_DIST = $(locale_DATA:.dat=.tmpl)
 CLEANFILES = $(locale_DATA)
 SUFFIXES = .dat .tmpl
