--- mplayerplug-in-3.40.orig/Source/plugin-threads.cpp
+++ mplayerplug-in-3.40/Source/plugin-threads.cpp
@@ -1746,7 +1746,7 @@
 		 i);
 
 	// if the play count is exactly 1 then tell mplayer to loop continuously and to use the same frame buffer (should eliminate blinking)
-	if (i == 1 || local_list->loop == 1) {
+	if (i == 1 || local_td->instance->loop == 1) {
 	    loop_single = 1;
 	    local_td->argv[argc++] = strdup("-loop");
 	    snprintf(buffer, 32, "%i", local_td->instance->loop);
