diff -Nur mplayerplug-in-old/po/Makefile.in mplayerplug-in/po/Makefile.in
--- mplayerplug-in-old/po/Makefile.in	2007-05-19 02:35:54.241101424 +0300
+++ mplayerplug-in/po/Makefile.in	2007-05-19 02:36:12.380343839 +0300
@@ -1,5 +1,5 @@
-LANGUAGES=da de en_US es fr hu it ja ko nb pl pt_BR nl ru se wa zh_CN
-MOFILES=da.mo de.mo en_US.mo es.mo fr.mo hu.mo it.mo ja.mo ko.mo nb.po pl.mo pt_BR.mo nl.mo ru.mo se.mo wa.mo zh_CN.mo
+LANGUAGES=da de en_US es fr hu it ja ko nb pl pt_BR nl ru se tr wa zh_CN
+MOFILES=da.mo de.mo en_US.mo es.mo fr.mo hu.mo it.mo ja.mo ko.mo nb.po pl.mo pt_BR.mo nl.mo ru.mo se.mo tr.mo wa.mo zh_CN.mo
 INSTALL = @INSTALL@
 prefix=@prefix@
 LOCALEDIR=@datadir@/locale
