Index: qtgui/i18n/recoll_tr.ts
===================================================================
--- qtgui/i18n/recoll_tr.ts.orig
+++ qtgui/i18n/recoll_tr.ts
@@ -139,7 +139,7 @@
         <source>.
 Click Cancel if you want to edit the configuration file before indexing starts, or Ok to let it proceed.</source>
         <translation>.
-İndekseleme başlamadan yapılandırmayı düzenlemek için İptal düğmesine basın ya da Tamam düğmesine basarak işleme izin verin.</translation>
+İndeksleme başlamadan yapılandırmayı düzenlemek için İptal düğmesine basın ya da Tamam düğmesine basarak işleme izin verin.</translation>
     </message>
 </context>
 <context>
@@ -268,7 +268,7 @@ İndekseleme başlamadan yapılandırmay
     </message>
     <message>
         <source>Beagle web history</source>
-        <translation type="unfinished"></translation>
+        <translation>Beagle ağ geçmişi</translation>
     </message>
 </context>
 <context>
@@ -373,7 +373,7 @@ Lütfen mimeconf dosyasını kontrol edi
     </message>
     <message>
         <source>All</source>
-        <translation type="unfinished"></translation>
+        <translation>Hepsi</translation>
     </message>
     <message>
         <source>media</source>
@@ -389,19 +389,19 @@ Lütfen mimeconf dosyasını kontrol edi
     </message>
     <message>
         <source>presentation</source>
-        <translation type="unfinished"></translation>
+        <translation>sunu</translation>
     </message>
     <message>
         <source>spreadsheet</source>
-        <translation type="unfinished"></translation>
+        <translation>sunu dosyası</translation>
     </message>
     <message>
         <source>text</source>
-        <translation type="unfinished"></translation>
+        <translation>metin</translation>
     </message>
     <message>
         <source>sorted</source>
-        <translation type="unfinished"></translation>
+        <translation>sıralanmış</translation>
     </message>
     <message>
         <source>filtered</source>
@@ -411,7 +411,7 @@ Lütfen mimeconf dosyasını kontrol edi
         <source>External applications/commands needed and not found for indexing your file types:
 
 </source>
-        <translation type="unfinished"></translation>
+    <translation>Dosya tiplerinizi indekslemek için gerekli dış programlar/komutlar bulunamadı</translation>
     </message>
     <message>
         <source>No helpers found missing</source>
