--- build.xml	2008-01-13 14:03:07.000000000 +0200
+++ build.xml2	2008-01-13 14:03:58.000000000 +0200
@@ -6,7 +6,7 @@
 <project default="jar" name="bcel" basedir=".">
   <property name="defaulttargetdir" value="target">
   </property>
-  <property name="libdir" value="target/lib">
+  <property name="libdir" value="/usr/share/java">
   </property>
   <property name="classesdir" value="target/classes">
   </property>
@@ -20,7 +20,7 @@
   </property>
   <path id="build.classpath">
     <fileset dir="${libdir}">
-      <include name="**/*.jar">
+      <include name="regexp.jar">
       </include>
     </fileset>
   </path>
@@ -36,7 +36,7 @@
     <available property="Junit.present" classname="junit.framework.Test">
     </available>
   </target>
-  <target name="compile" description="o Compile the code" depends="get-deps">
+  <target name="compile" description="o Compile the code">
     <mkdir dir="${classesdir}">
     </mkdir>
     <javac destdir="${classesdir}" deprecation="true" debug="true" optimize="false" excludes="**/package.html">
