diff -r c32abbaccdac xen/common/symbols.c
--- a/xen/common/symbols.c	Tue Jan 09 19:24:12 2007 +0000
+++ b/xen/common/symbols.c	Wed Jan 10 13:52:15 2007 +0200
@@ -94,7 +94,7 @@ const char *symbols_lookup(unsigned long
     unsigned long symbol_end = 0;
 
     /* This kernel should never had been booted. */
-    BUG_ON(!symbols_addresses);
+    BUG_ON(symbols_num_syms == 0);
 
     namebuf[KSYM_NAME_LEN] = 0;
     namebuf[0] = 0;
