--- kdegraphics/kpdf/xpdf/xpdf/Lexer.cc	2008/02/09 23:00:50	773029
+++ kdegraphics/kpdf/xpdf/xpdf/Lexer.cc	2008/02/18 19:22:19	776758
@@ -298,7 +298,7 @@
 	  n = 0;
 	  
 	  // we are growing see if the document is not malformed and we are growing too much
-	  if (objNum > 0)
+	  if (objNum > 0 && xref != NULL)
 	  {
 	    int newObjNum = xref->getNumEntry(curStr.streamGetPos());
 	    if (newObjNum != objNum)
