diff -ur kdegraphics-3.5.9.orig/kdvi/fontpool.cpp kdegraphics-3.5.9/kdvi/fontpool.cpp
--- kdegraphics-3.5.9.orig/kdvi/fontpool.cpp	2008-03-02 17:18:37.000000000 +0200
+++ kdegraphics-3.5.9/kdvi/fontpool.cpp	2008-03-02 17:19:15.000000000 +0200
@@ -111,6 +111,9 @@
   kdDebug(4300) << "fontPool::~fontPool() called" << endl;
 #endif
 
+  // need to manually clear the fonts _before_ freetype gets unloaded
+  fontList.clear();
+
 #ifdef HAVE_FREETYPE
   if (FreeType_could_be_loaded == true)
     FT_Done_FreeType( FreeType_library );
