Index: gst/equalizer/gstiirequalizer.c
===================================================================
RCS file: /cvs/gstreamer/gst-plugins-good/gst/equalizer/gstiirequalizer.c,v
retrieving revision 1.28
diff -u -p -a -u -p -r1.28 gstiirequalizer.c
--- gst/equalizer/gstiirequalizer.c	31 May 2008 08:37:00 -0000	1.28
+++ gst/equalizer/gstiirequalizer.c	21 Oct 2008 15:30:20 -0000
@@ -470,6 +470,7 @@ update_coefficients (GstIirEqualizer * e
   for (i = 0; i < equ->freq_band_count; i++) {
     setup_filter (equ, equ->bands[i]);
   }
+  equ->need_new_coefficients = FALSE;
 }
 
 void

