Index: procps-3.2.8/Makefile
===================================================================
--- procps-3.2.8.orig/Makefile
+++ procps-3.2.8/Makefile
@@ -230,6 +230,8 @@ $(MANFILES) : all
 install: $(filter-out $(SKIP) $(addprefix $(DESTDIR),$(SKIP)),$(INSTALL))
 	cd $(usr/bin) && $(ln_f) skill snice
 	cd $(usr/proc/bin) && $(ln_f) pgrep pkill
+	mkdir -p $(usr/include)/proc
+	cp proc/*h $(usr/include)/proc
 
 ############ prog.c --> prog.o
 
