Index: x11vnc-0.9.12/x11vnc/x11vnc.desktop
===================================================================
--- x11vnc-0.9.12.orig/x11vnc/x11vnc.desktop
+++ x11vnc-0.9.12/x11vnc/x11vnc.desktop
@@ -1,6 +1,10 @@
 [Desktop Entry]
 Name=X11VNC Server
+Name[tr]=X11VNC Sunucusu
+GenericName=X11VNC Server
+GenericName[tr]=X11VNC Sunucusu
 Comment=Share this desktop by VNC
+Comment[tr]=Masaüstünü VNC ile paylaş
 Exec=x11vnc -gui tray=setpass -rfbport PROMPT -bg -o %%HOME/.x11vnc.log.%%VNCDISPLAY
 Icon=computer
 Terminal=false
