diff -Nur trackballs-1.1.2-old/po/Makefile.in.in trackballs-1.1.2/po/Makefile.in.in
--- trackballs-1.1.2-old/po/Makefile.in.in	2006-08-20 11:00:47.333181520 +0300
+++ trackballs-1.1.2/po/Makefile.in.in	2006-08-20 11:02:29.311678432 +0300
@@ -22,7 +22,7 @@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
 #localedir = $(datadir)/${PACKAGE}/locale
-localedir = $(datadir)/locale
+localedir =/usr/share/locale
 gettextsrcdir = $(datadir)/gettext/po
 
 INSTALL = @INSTALL@
diff -Nur trackballs-1.1.2-old/share/Makefile.in trackballs-1.1.2/share/Makefile.in
--- trackballs-1.1.2-old/share/Makefile.in	2006-08-20 11:00:47.521152944 +0300
+++ trackballs-1.1.2/share/Makefile.in	2006-08-20 11:02:07.005069552 +0300
@@ -108,7 +108,7 @@
 WINRESOURCES = @WINRESOURCES@
 with_highscores = @with_highscores@
 
-SUBDIRS = fonts levels sfx images icons music
+SUBDIRS = fonts levels sfx images  music
 pkgdata_DATA = highScores
 man_MANS = trackballs.6
 EXTRA_DIST = $(pkgdata_DATA) trackballs.6
diff -Nur trackballs-1.1.2-old/src/Makefile.in trackballs-1.1.2/src/Makefile.in
--- trackballs-1.1.2-old/src/Makefile.in	2006-08-20 11:00:47.458162520 +0300
+++ trackballs-1.1.2/src/Makefile.in	2006-08-20 11:02:29.309678736 +0300
@@ -122,7 +122,7 @@
 
 snarfcppopts = $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(CPPFLAGS) $(CFLAGS)
 
-localedir = $(datadir)/locale
+localedir =/usr/share/locale
 DEFS = -DLOCALEDIR=\"$(localedir)\" @DEFS@
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../config.h
