diff -Nur ov511-2.32-old/saa7111-new.c ov511-2.32/saa7111-new.c
--- ov511-2.32-old/saa7111-new.c	2007-01-19 16:45:51.082633107 +0200
+++ ov511-2.32/saa7111-new.c	2007-01-19 16:45:59.872296875 +0200
@@ -30,6 +30,7 @@
 #include <linux/string.h>
 #include <linux/errno.h>
 #include <linux/slab.h>
+#include <linux/ioctl.h>
 
 #include <linux/i2c.h>
 #include <linux/videodev.h>
diff -Nur ov511-2.32-old/tda7313.c ov511-2.32/tda7313.c
--- ov511-2.32-old/tda7313.c	2007-01-19 16:45:51.083632955 +0200
+++ ov511-2.32/tda7313.c	2007-01-19 16:45:59.872296875 +0200
@@ -19,6 +19,7 @@
 #include <linux/slab.h>
 #include <linux/videodev.h>
 #include <linux/i2c.h>
+#include <linux/ioctl.h>
 
 #include "id.h"
 #include "audiochip.h"
