diff -Nur ov511-2.32-old/ov511_core.c ov511-2.32/ov511_core.c
--- ov511-2.32-old/ov511_core.c	2008-03-22 19:46:52.000000000 +0200
+++ ov511-2.32/ov511_core.c	2008-03-22 19:47:15.000000000 +0200
@@ -4980,7 +4980,6 @@
 	.owner =	THIS_MODULE,
 	.name =		"OV511 USB Camera",
 	.type =		VID_TYPE_CAPTURE,
-	.hardware =	VID_HARDWARE_OV511,
 	.fops =		&ov511_fops,
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 0)
 	.release =	video_device_release,
diff -Nur ov511-2.32-old/ovfx2.c ov511-2.32/ovfx2.c
--- ov511-2.32-old/ovfx2.c	2008-03-22 19:46:52.000000000 +0200
+++ ov511-2.32/ovfx2.c	2008-03-22 19:47:28.000000000 +0200
@@ -2961,7 +2961,6 @@
 	.owner =	THIS_MODULE,
 	.name =		"OVFX2 USB Camera",
 	.type =		VID_TYPE_CAPTURE,
-	.hardware =	VID_HARDWARE_OV511,	/* FIXME */
 	.fops =		&ovfx2_fops,
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 0)
 	.release =	video_device_release,
