--- admin/cvs.sh.orig	2008-09-22 13:29:27.000000000 +0300
+++ admin/cvs.sh	2008-09-22 13:20:33.000000000 +0300
@@ -70,7 +70,7 @@
     echo "*** KDE requires automake $required_automake_version"
     exit 1
     ;;
-  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9*)
+  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.10*)
     echo "*** $AUTOMAKE_STRING found."
     UNSERMAKE=no
     ;;
