Index: akiaLinuxPaketv1.3/akia.desktop
===================================================================
--- akiaLinuxPaketv1.3.orig/akia.desktop
+++ akiaLinuxPaketv1.3/akia.desktop
@@ -6,7 +6,7 @@ Name[tr]=AKiS Kart İzleme Aracı
 GenericName=SmartCard Viewer
 GenericName[tr]=Akıllı Kart İzleme Aracı
 Comment=View and modify files,folders and objects in a AKiS smartcard
-Exec=java -jar /usr/share/akia/akia.jar
+Exec=/usr/bin/akia
 Icon=/usr/share/pixmaps/akia.png
 StartupNotify=true
 Type=Application
