diff --git a/privileges/hal-power-cpufreq.privilege b/privileges/hal-power-cpufreq.privilege
index cee08b4..49df022 100644
--- a/privileges/hal-power-cpufreq.privilege
+++ b/privileges/hal-power-cpufreq.privilege
@@ -3,9 +3,9 @@ # This privilege specifies who is allowe
 # via the org.freedesktop.Hal.Device.CPUFreq interface
 
 [Privilege]
-RequiredPrivileges=desktop-console
-SufficientPrivileges=
-Allow=uid:__all__
+RequiredPrivileges=
+SufficientPrivileges=desktop-console
+Allow=uid:root
 Deny=
 CanObtain=True
 CanGrant=True
diff -ur hal-0.5.8_git20060822.orig/privileges/hal-power-hibernate.privilege hal-0.5.8_git20060822/privileges/hal-power-hibernate.privilege
--- hal-0.5.8_git20060822.orig/privileges/hal-power-hibernate.privilege	2006-08-22 16:49:40.000000000 +0300
+++ hal-0.5.8_git20060822/privileges/hal-power-hibernate.privilege	2006-09-04 11:18:21.000000000 +0300
@@ -4,7 +4,7 @@
 [Privilege]
 RequiredPrivileges=desktop-console
 SufficientPrivileges=
-Allow=uid:__all__
+Allow=gid:power
 Deny=
 CanObtain=True
 CanGrant=True
diff -ur hal-0.5.8_git20060822.orig/privileges/hal-power-poweroff.privilege hal-0.5.8_git20060822/privileges/hal-power-poweroff.privilege
--- hal-0.5.8_git20060822.orig/privileges/hal-power-poweroff.privilege	2006-08-22 16:49:40.000000000 +0300
+++ hal-0.5.8_git20060822/privileges/hal-power-poweroff.privilege	2006-09-04 11:18:22.000000000 +0300
@@ -4,7 +4,7 @@
 [Privilege]
 RequiredPrivileges=desktop-console
 SufficientPrivileges=
-Allow=uid:__all__
+Allow=gid:power
 Deny=
 CanObtain=True
 CanGrant=True
diff -ur hal-0.5.8_git20060822.orig/privileges/hal-power-reboot.privilege hal-0.5.8_git20060822/privileges/hal-power-reboot.privilege
--- hal-0.5.8_git20060822.orig/privileges/hal-power-reboot.privilege	2006-08-22 16:49:40.000000000 +0300
+++ hal-0.5.8_git20060822/privileges/hal-power-reboot.privilege	2006-09-04 11:18:22.000000000 +0300
@@ -4,7 +4,7 @@
 [Privilege]
 RequiredPrivileges=desktop-console
 SufficientPrivileges=
-Allow=uid:__all__
+Allow=gid:power
 Deny=
 CanObtain=True
 CanGrant=True
diff -ur hal-0.5.8_git20060822.orig/privileges/hal-power-standby.privilege hal-0.5.8_git20060822/privileges/hal-power-standby.privilege
--- hal-0.5.8_git20060822.orig/privileges/hal-power-standby.privilege	2006-09-04 11:17:58.000000000 +0300
+++ hal-0.5.8_git20060822/privileges/hal-power-standby.privilege	2006-09-04 11:18:22.000000000 +0300
@@ -4,7 +4,7 @@
 [Privilege]
 RequiredPrivileges=desktop-console
 SufficientPrivileges=
-Allow=uid:__all__
+Allow=gid:power
 Deny=
 CanObtain=True
 CanGrant=True
diff -ur hal-0.5.8_git20060822.orig/privileges/hal-power-suspend.privilege hal-0.5.8_git20060822/privileges/hal-power-suspend.privilege
--- hal-0.5.8_git20060822.orig/privileges/hal-power-suspend.privilege	2006-08-22 16:49:40.000000000 +0300
+++ hal-0.5.8_git20060822/privileges/hal-power-suspend.privilege	2006-09-04 11:18:23.000000000 +0300
@@ -4,7 +4,7 @@
 [Privilege]
 RequiredPrivileges=desktop-console
 SufficientPrivileges=
-Allow=uid:__all__
+Allow=gid:power
 Deny=
 CanObtain=True
 CanGrant=True
diff -ur hal-0.5.8_git20060822.orig/privileges/hal-storage-fixed-mount-all-options.privilege hal-0.5.8_git20060822/privileges/hal-storage-fixed-mount-all-options.privilege
--- hal-0.5.8_git20060822.orig/privileges/hal-storage-fixed-mount-all-options.privilege	2006-08-22 16:49:40.000000000 +0300
+++ hal-0.5.8_git20060822/privileges/hal-storage-fixed-mount-all-options.privilege	2006-09-04 11:18:23.000000000 +0300
@@ -7,7 +7,7 @@
 [Privilege]
 RequiredPrivileges=desktop-console
 SufficientPrivileges=
-Allow=
+Allow=gid:disk
 Deny=
 CanObtain=True
 CanGrant=True
diff -ur hal-0.5.8_git20060822.orig/privileges/hal-storage-fixed-mount.privilege hal-0.5.8_git20060822/privileges/hal-storage-fixed-mount.privilege
--- hal-0.5.8_git20060822.orig/privileges/hal-storage-fixed-mount.privilege	2006-08-22 16:49:40.000000000 +0300
+++ hal-0.5.8_git20060822/privileges/hal-storage-fixed-mount.privilege	2006-09-04 11:18:23.000000000 +0300
@@ -9,7 +9,7 @@
 [Privilege]
 RequiredPrivileges=desktop-console
 SufficientPrivileges=
-Allow=
+Allow=gid:disk
 Deny=
 CanObtain=True
 CanGrant=True
diff -ur hal-0.5.8_git20060822.orig/privileges/hal-storage-removable-mount-all-options.privilege hal-0.5.8_git20060822/privileges/hal-storage-removable-mount-all-options.privilege
--- hal-0.5.8_git20060822.orig/privileges/hal-storage-removable-mount-all-options.privilege	2006-08-22 16:49:40.000000000 +0300
+++ hal-0.5.8_git20060822/privileges/hal-storage-removable-mount-all-options.privilege	2006-09-04 11:18:24.000000000 +0300
@@ -7,7 +7,7 @@
 [Privilege]
 RequiredPrivileges=desktop-console
 SufficientPrivileges=
-Allow=
+Allow=gid:removable
 Deny=
 CanObtain=True
 CanGrant=True
diff -ur hal-0.5.8_git20060822.orig/privileges/hal-storage-removable-mount.privilege hal-0.5.8_git20060822/privileges/hal-storage-removable-mount.privilege
--- hal-0.5.8_git20060822.orig/privileges/hal-storage-removable-mount.privilege	2006-08-22 16:49:40.000000000 +0300
+++ hal-0.5.8_git20060822/privileges/hal-storage-removable-mount.privilege	2006-09-04 11:18:24.000000000 +0300
@@ -8,7 +8,7 @@
 [Privilege]
 RequiredPrivileges=desktop-console
 SufficientPrivileges=
-Allow=uid:__all__
+Allow=gid:removable
 Deny=
 CanObtain=True
 CanGrant=True
