diff -Nur mplayer-20081003-old/libvo/x11_common.c mplayer-20081003/libvo/x11_common.c
--- mplayer-20081003-old/libvo/x11_common.c	2008-10-03 19:20:17.000000000 +0300
+++ mplayer-20081003/libvo/x11_common.c	2008-10-03 19:22:54.000000000 +0300
@@ -1231,7 +1231,7 @@
                                    unsigned int width, unsigned int height,
                                    int depth, Colormap col_map)
 {
-    unsigned long xswamask = CWBackingStore | CWBorderPixel;
+    unsigned long xswamask = CWBorderPixel;
     XSetWindowAttributes xswa;
     Window ret_win;
 
@@ -1242,7 +1242,7 @@
     }
     xswa.background_pixel = 0;
     xswa.border_pixel = 0;
-    xswa.backing_store = Always;
+    xswa.backing_store = NotUseful;
     xswa.bit_gravity = StaticGravity;
 
     ret_win =
