Index: help/help_mp-tr.h
===================================================================
--- help/help_mp-tr.h	(revision 28228)
+++ help/help_mp-tr.h	(working copy)
@@ -1516,7 +1516,7 @@
 // network.c
 
 #define MSGTR_MPDEMUX_NW_UnknownAF "Bilinmeyen adres sınıfı %d\n"
-#define MSGTR_MPDEMUX_NW_ResolvingHostForAF "%s çözülüyor ( %s için )...\n"
+#define MSGTR_MPDEMUX_NW_ResolvingHostForAF "Adres çözülüyor %s , %s için...\n"
 #define MSGTR_MPDEMUX_NW_CantResolv "%s için ad çözülemedi: %s\n"
 #define MSGTR_MPDEMUX_NW_ConnectingToServer "%s[%s] sunucusuna bağlanılıyor: %d...\n"
 #define MSGTR_MPDEMUX_NW_CantConnect2Server "%s ile sunucuya bağlanılamadı\n"
