--- share/LaTeX/Makefile.am.orig	2008-10-13 13:17:38.000000000 +0300
+++ share/LaTeX/Makefile.am	2008-10-13 13:19:31.000000000 +0300
@@ -8,11 +8,11 @@
 
 install-cfg:
 	@if test $(PLAINTEX) != no ; then \
-	  installdir=`kpsexpand '$$TEXMFLOCAL'/tex/latex/gnuplot |sed "s%^.*share/%$(prefix)/share/%"` ; \
+	  installdir=`echo /usr/share/texmf-dist/tex/latex/gnuplot` ; \
 	  $(top_srcdir)/mkinstalldirs $(DESTDIR)$$installdir ; \
 	  if [ ! -f $(DESTDIR)$$installdir/gnuplot.cfg ]; then \
 	    $(INSTALL_DATA) $(srcdir)/gnuplot.cfg $(DESTDIR)$$installdir/gnuplot.cfg && \
-	    texhash; \
+	    /bin/true; \
 	  fi ; \
 	  fi
 
