diff -Nur gdb-6.4.90.dfsg.orig/gdb/maint.c gdb-6.4.90.dfsg/gdb/maint.c
--- gdb-6.4.90.dfsg.orig/gdb/maint.c	2006-02-23 18:33:43.000000000 +0000
+++ gdb-6.4.90.dfsg/gdb/maint.c	2006-09-30 16:42:40.000000000 +0100
@@ -851,14 +851,14 @@
 
   add_cmd ("deprecate", class_maintenance, maintenance_deprecate, _("\
 Deprecate a command.  Note that this is just in here so the \n\
-testsuite can check the comamnd deprecator. You probably shouldn't use this,\n\
+testsuite can check the command deprecator. You probably shouldn't use this,\n\
 rather you should use the C function deprecate_cmd().  If you decide you \n\
 want to use it: maintenance deprecate 'commandname' \"replacement\". The \n\
 replacement is optional."), &maintenancelist);
 
   add_cmd ("undeprecate", class_maintenance, maintenance_undeprecate, _("\
 Undeprecate a command.  Note that this is just in here so the \n\
-testsuite can check the comamnd deprecator. You probably shouldn't use this,\n\
+testsuite can check the command deprecator. You probably shouldn't use this,\n\
 If you decide you want to use it: maintenance undeprecate 'commandname'"),
 	   &maintenancelist);
 
diff -Nur gdb-6.4.90.dfsg.orig/gdb/remote.c gdb-6.4.90.dfsg/gdb/remote.c
--- gdb-6.4.90.dfsg.orig/gdb/remote.c	2006-05-05 21:08:45.000000000 +0100
+++ gdb-6.4.90.dfsg/gdb/remote.c	2006-09-30 16:43:32.000000000 +0100
@@ -4795,7 +4795,7 @@
       do_cleanups (old_chain);
     }
   if (mismatched > 0)
-    warning (_("One or more sections of the remote executable does not match\n\
+    warning (_("One or more sections of the remote executable do not match\n\
 the loaded file\n"));
   if (args && !matched)
     printf_filtered (_("No loaded section named '%s'.\n"), args);
