--- examples/poker/python-poker3d.desktop~	2008-06-29 22:18:13.000000000 +0300
+++ examples/poker/python-poker3d.desktop	2008-06-29 22:16:36.000000000 +0300
@@ -2,12 +2,13 @@
 Encoding=UTF-8
 Name=poker3D
 Name[fr]=poker3D
-Exec=/usr/games/poker2d
+Exec=poker3d
 Icon=poker3D
 Miniicon=poker3D
 DocPath=poker3D/index.html
 GenericName=Multiplayer online poker
 GenericName[fr]=Poker multijoueur en ligne
+GenericName[tr]=Çok oyunculu çevrimiçi poker oyunu
 Terminal=false
 Type=Application
 X-KDE-StartupNotify=false
