--- src/maf/renderbin.cpp~	2008-02-21 20:38:37.000000000 -0800
+++ src/maf/renderbin.cpp	2008-02-21 20:38:53.000000000 -0800
@@ -26,6 +26,7 @@
  *
  */
 
+#include <string.h>
 #include <maf/StdAfx.h>
 
 #ifndef MAF_USE_VS_PCH
--- src/maf/wnc_image.cpp~	2008-02-21 20:49:49.000000000 -0800
+++ src/maf/wnc_image.cpp	2008-02-21 20:50:11.000000000 -0800
@@ -36,6 +36,7 @@
  *
  */
 
+#include <string.h>
 #include <maf/StdAfx.h>
 
 #ifndef MAF_USE_VS_PCH
--- src/vserial/meshlayer.cpp~	2008-02-21 20:59:14.000000000 -0800
+++ src/vserial/meshlayer.cpp	2008-02-21 20:59:42.000000000 -0800
@@ -26,6 +26,7 @@
  *
  */
 
+#include <string.h>
 #include <vserial/stdafx.h>
 
 #ifndef UNDERWARE_VSERIAL_USE_PCH
--- src/vserial/mesh_saver.cpp~	2008-02-21 21:15:58.000000000 -0800
+++ src/vserial/mesh_saver.cpp	2008-02-21 21:16:08.000000000 -0800
@@ -26,6 +26,7 @@
  *
  */
 
+#include <string.h>
 #include <vserial/stdafx.h>
 
 #ifndef UNDERWARE_VSERIAL_USE_PCH
--- src/vserial/rle.cpp~	2008-02-21 21:23:47.000000000 -0800
+++ src/vserial/rle.cpp	2008-02-21 21:23:58.000000000 -0800
@@ -26,6 +26,7 @@
  *
  */
 
+#include <stdlib.h>
 #include <vserial/stdafx.h>
 
 #ifndef UNDERWARE_VSERIAL_USE_PCH
--- src/vserial/scene_saver.cpp~	2008-02-21 21:32:46.000000000 -0800
+++ src/vserial/scene_saver.cpp	2008-02-21 21:32:57.000000000 -0800
@@ -26,6 +26,7 @@
  *
  */
 
+#include <string.h>
 #include <vserial/stdafx.h>
 
 #ifndef UNDERWARE_VSERIAL_USE_PCH
--- src/vserial/string.cpp~	2008-02-21 21:39:35.000000000 -0800
+++ src/vserial/string.cpp	2008-02-21 21:39:43.000000000 -0800
@@ -26,6 +26,7 @@
  *
  */
 
+#include <string.h>
 #include <vserial/stdafx.h>
 
 #ifndef UNDERWARE_VSERIAL_USE_PCH
--- osgplugins/ddsrle/src/rle.cpp~	2008-02-21 21:49:47.000000000 -0800
+++ osgplugins/ddsrle/src/rle.cpp	2008-02-21 21:58:42.000000000 -0800
@@ -26,6 +26,8 @@
  *
  */
 
+#include <stdlib.h>
+#include <limits.h>
 #include <memory.h>
 #include <vector>
 
--- dmctp/dmctp_sound/dmctp_sound.cpp~	2008-02-21 22:09:06.000000000 -0800
+++ dmctp/dmctp_sound/dmctp_sound.cpp	2008-02-21 22:09:18.000000000 -0800
@@ -26,6 +26,7 @@
  *
  */
 
+#include <stdlib.h>
 #include <string>
 #include <iostream>
 #include <vector>
