--- backend/avision.c.orig	2008-02-09 11:42:46.000000000 +0100
+++ backend/avision.c	2008-02-27 13:35:18.000000000 +0100
@@ -4375,8 +4375,9 @@ set_calib_data (Avision_Scanner* s, stru
 static uint8_t*
 sort_and_average (struct calibration_format* format, uint8_t* data)
 {
-  int stride, i, line;
-  int elements_per_line;
+  int stride = 1;
+  int i, line;
+  int elements_per_line = format->pixel_per_line * format->channels;
   
   uint8_t *sort_data, *avg_data;
   
