--- camorama.desktop.in.orig	2007-06-14 17:05:33.000000000 +0300
+++ camorama.desktop.in	2009-09-21 11:56:23.228477940 +0300
@@ -1,11 +1,13 @@
 [Desktop Entry]
 _Name=Camorama Webcam Viewer
+_Name[tr]=Camorama Webcam Görüntüleyicisi
 _Comment=View, alter and save images from a webcam
+_Comment[tr]=webcamden resim kaydedebilir, görüntüleyebilirsiniz
 Exec=camorama
 Terminal=0
 Type=Application
-Icon=camorama.png
-Categories=Application;Graphics
+Icon=camorama
+Categories=Application;GNOME;AudioVideo;
 StartupNotify=true
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=camorama
