diff -Nur ov511-2.32-old/ov511_core.c ov511-2.32/ov511_core.c
--- ov511-2.32-old/ov511_core.c	2008-03-22 19:42:20.000000000 +0200
+++ ov511-2.32/ov511_core.c	2008-03-22 19:44:52.000000000 +0200
@@ -26,7 +26,7 @@
  * option) any later version. NO WARRANTY OF ANY KIND is expressed or implied.
  */
 
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 
 /* 2.6 Doesn't support /proc/video, but this is still defined somewhere */
diff -Nur ov511-2.32-old/ov511_decomp.c ov511-2.32/ov511_decomp.c
--- ov511-2.32-old/ov511_decomp.c	2008-03-22 19:42:20.000000000 +0200
+++ ov511-2.32/ov511_decomp.c	2008-03-22 19:44:52.000000000 +0200
@@ -13,7 +13,7 @@
 
 #define __NO_VERSION__
 
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 
 #include "ov511.h"
diff -Nur ov511-2.32-old/ov511.h ov511-2.32/ov511.h
--- ov511-2.32-old/ov511.h	2008-03-22 19:42:20.000000000 +0200
+++ ov511-2.32/ov511.h	2008-03-22 19:44:52.000000000 +0200
@@ -14,7 +14,7 @@
 #define HAVE_V4L2
 #endif
 
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <asm/uaccess.h>
 #include <linux/videodev.h>
 #include <linux/smp_lock.h>
diff -Nur ov511-2.32-old/ov518_decomp.c ov511-2.32/ov518_decomp.c
--- ov511-2.32-old/ov518_decomp.c	2008-03-22 19:42:20.000000000 +0200
+++ ov511-2.32/ov518_decomp.c	2008-03-22 19:44:52.000000000 +0200
@@ -14,7 +14,7 @@
 
 #define __NO_VERSION__
 
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 
 #include "ov511.h"
diff -Nur ov511-2.32-old/ov6x20.c ov511-2.32/ov6x20.c
--- ov511-2.32-old/ov6x20.c	2008-03-22 19:42:20.000000000 +0200
+++ ov511-2.32/ov6x20.c	2008-03-22 19:44:52.000000000 +0200
@@ -12,7 +12,7 @@
 /* This file is not a module yet */
 #define __NO_VERSION__
 
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 #include <linux/slab.h>
 #include "ovcamchip_priv.h"
diff -Nur ov511-2.32-old/ov6x30.c ov511-2.32/ov6x30.c
--- ov511-2.32-old/ov6x30.c	2008-03-22 19:42:20.000000000 +0200
+++ ov511-2.32/ov6x30.c	2008-03-22 19:44:52.000000000 +0200
@@ -12,7 +12,7 @@
 /* This file is not a module yet */
 #define __NO_VERSION__
 
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 #include <linux/slab.h>
 #include "ovcamchip_priv.h"
diff -Nur ov511-2.32-old/ov76be.c ov511-2.32/ov76be.c
--- ov511-2.32-old/ov76be.c	2008-03-22 19:42:20.000000000 +0200
+++ ov511-2.32/ov76be.c	2008-03-22 19:44:52.000000000 +0200
@@ -12,7 +12,7 @@
 /* This file is not a module yet */
 #define __NO_VERSION__
 
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 #include <linux/slab.h>
 #include "ovcamchip_priv.h"
diff -Nur ov511-2.32-old/ov7x10.c ov511-2.32/ov7x10.c
--- ov511-2.32-old/ov7x10.c	2008-03-22 19:42:20.000000000 +0200
+++ ov511-2.32/ov7x10.c	2008-03-22 19:44:52.000000000 +0200
@@ -14,7 +14,7 @@
 /* This file is not a module yet */
 #define __NO_VERSION__
 
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 #include <linux/slab.h>
 #include "ovcamchip_priv.h"
diff -Nur ov511-2.32-old/ov7x20.c ov511-2.32/ov7x20.c
--- ov511-2.32-old/ov7x20.c	2008-03-22 19:42:20.000000000 +0200
+++ ov511-2.32/ov7x20.c	2008-03-22 19:44:52.000000000 +0200
@@ -14,7 +14,7 @@
 /* This file is not a module yet */
 #define __NO_VERSION__
 
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 #include <linux/slab.h>
 #include "ovcamchip_priv.h"
diff -Nur ov511-2.32-old/ovcamchip_core.c ov511-2.32/ovcamchip_core.c
--- ov511-2.32-old/ovcamchip_core.c	2008-03-22 19:42:20.000000000 +0200
+++ ov511-2.32/ovcamchip_core.c	2008-03-22 19:44:52.000000000 +0200
@@ -9,7 +9,7 @@
  * option) any later version. NO WARRANTY OF ANY KIND is expressed or implied.
  */
 
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 #include <linux/init.h>
 #include <linux/module.h>
diff -Nur ov511-2.32-old/ovfx2.c ov511-2.32/ovfx2.c
--- ov511-2.32-old/ovfx2.c	2008-03-22 19:42:20.000000000 +0200
+++ ov511-2.32/ovfx2.c	2008-03-22 19:44:52.000000000 +0200
@@ -21,7 +21,7 @@
  * option) any later version. NO WARRANTY OF ANY KIND is expressed or implied.
  */
 
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 
 /* 2.6 Doesn't support /proc/video, but this is still defined somewhere */
diff -Nur ov511-2.32-old/ovfx2.h ov511-2.32/ovfx2.h
--- ov511-2.32-old/ovfx2.h	2008-03-22 19:42:20.000000000 +0200
+++ ov511-2.32/ovfx2.h	2008-03-22 19:44:52.000000000 +0200
@@ -14,7 +14,7 @@
 #define HAVE_V4L2
 #endif
 
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <asm/uaccess.h>
 #include <linux/videodev.h>
 #include <linux/smp_lock.h>
diff -Nur ov511-2.32-old/saa7111-new.c ov511-2.32/saa7111-new.c
--- ov511-2.32-old/saa7111-new.c	2008-03-22 19:42:20.000000000 +0200
+++ ov511-2.32/saa7111-new.c	2008-03-22 19:44:52.000000000 +0200
@@ -22,7 +22,7 @@
    Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
