diff -Nur ov511-2.32-old/ov511_core.c ov511-2.32/ov511_core.c
--- ov511-2.32-old/ov511_core.c	2008-03-22 20:29:19.000000000 +0200
+++ ov511-2.32/ov511_core.c	2008-03-22 20:30:19.000000000 +0200
@@ -1698,7 +1698,6 @@
 	.id =			I2C_ALGO_SMBUS,
 #endif
 	.smbus_xfer =		ov518_smbus_xfer,
-	.algo_control =		ov51x_i2c_control,
 	.functionality =	ov518_i2c_func,
 };
 
@@ -1708,7 +1707,6 @@
 	.id =			I2C_ALGO_SMBUS,
 #endif
 	.smbus_xfer =		ov511_smbus_xfer,
-	.algo_control =		ov51x_i2c_control,
 	.functionality =	ov511_i2c_func,
 };
 
diff -Nur ov511-2.32-old/ovfx2.c ov511-2.32/ovfx2.c
--- ov511-2.32-old/ovfx2.c	2008-03-22 20:29:19.000000000 +0200
+++ ov511-2.32/ovfx2.c	2008-03-22 20:30:19.000000000 +0200
@@ -1005,7 +1005,6 @@
 	.id =			I2C_ALGO_SMBUS,
 #endif
 	.smbus_xfer =		ovfx2_smbus_xfer,
-	.algo_control =		ovfx2_i2c_control,
 	.functionality =	ovfx2_i2c_func,
 };
 
