diff -Nur RutilTv0.15-old/configure.sh RutilTv0.15/configure.sh
--- RutilTv0.15-old/configure.sh	2007-11-10 20:27:55.836986471 +0200
+++ RutilTv0.15/configure.sh	2007-11-10 20:29:10.581623559 +0200
@@ -294,12 +294,8 @@
 # within this script too to apply different policy for each user. 
 # This script is not executed with root privileges.
 #
-if which gksu > /dev/null; then
-    exec gksu \"$prefix/$helper_prefix$helper_name \$1\"
-elif which kdesu > /dev/null; then
-    exec kdesu \"$prefix/$helper_prefix$helper_name \$1\"
-else
-    echo "$helper_launcher_name : Cannot find a launcher."
-fi" > $helper_launcher_name
+
+exec kdesu \"$prefix/$helper_prefix$helper_name \$1\"" \
+> $helper_launcher_name
     echo done
 fi
