diff -Nur RutilTv0.16-old/include/CWirelessMsgHandler.h RutilTv0.16/include/CWirelessMsgHandler.h
--- RutilTv0.16-old/include/CWirelessMsgHandler.h	2008-04-02 20:23:17.000000000 +0300
+++ RutilTv0.16/include/CWirelessMsgHandler.h	2008-04-02 20:25:51.000000000 +0300
@@ -23,6 +23,7 @@
 
 #include <string>
 #include <stdexcept> // std::bad_alloc
+#include <memory> // std::auto_ptr
 
 #include "IMsgHandler.h"
 #include "IDriver.h"
diff -Nur RutilTv0.16-old/include/private/RTDrivers.hxx RutilTv0.16/include/private/RTDrivers.hxx
--- RutilTv0.16-old/include/private/RTDrivers.hxx	2008-04-02 20:23:17.000000000 +0300
+++ RutilTv0.16/include/private/RTDrivers.hxx	2008-04-02 20:25:51.000000000 +0300
@@ -19,6 +19,9 @@
     \author Romain BONDUE
     \date 30/08/2005
     \warning This file should only be included from RTDrivers.h */
+
+#include <cstring> // memcpy()
+
 inline nsWireless::CRTDriver::CRTDriver (const std::string& DeviceName)
                                                 throw (nsErrors::CSystemExc)
     : CWE17Driver (DeviceName) {}
diff -Nur RutilTv0.16-old/lib/src/Su.cxx RutilTv0.16/lib/src/Su.cxx
--- RutilTv0.16-old/lib/src/Su.cxx	2008-04-02 20:23:17.000000000 +0300
+++ RutilTv0.16/lib/src/Su.cxx	2008-04-02 20:25:51.000000000 +0300
@@ -23,6 +23,7 @@
 #include <sstream>
 #include <memory>
 #include <csignal>
+#include <cstring> // memset()
 
 extern"C"{
 #include <sys/types.h>
