Index: libgphoto2-2.4.2/camlibs/ptp2/library.c
===================================================================
--- libgphoto2-2.4.2.orig/camlibs/ptp2/library.c
+++ libgphoto2-2.4.2/camlibs/ptp2/library.c
@@ -853,6 +853,7 @@ static struct {
 
 	/* from Mike Meyer <mwm@mired.org> */
 	{"Apple:iPhone (PTP mode)",		0x05ac, 0x1290, PTPBUG_DCIM_WRONG_PARENT},
+    {"Apple:iPhone 3G (PTP mode)",   0x05ac, 0x1292, PTPBUG_DCIM_WRONG_PARENT},
 	/* https://sourceforge.net/tracker/index.php?func=detail&aid=1869653&group_id=158745&atid=809061 */
 	{"Pioneer:DVR-LX60D",			0x08e4, 0x0142, 0},
 };
