diff -ur dbus-python-0.82.2.orig/configure dbus-python-0.82.2/configure
--- dbus-python-0.82.2.orig/configure	2007-08-01 22:05:32.000000000 +0300
+++ dbus-python-0.82.2/configure	2007-08-02 23:57:44.000000000 +0300
@@ -9822,7 +9822,7 @@
 
 { echo "$as_me:$LINENO: checking for Python headers using $PYTHON-config --includes" >&5
 echo $ECHO_N "checking for Python headers using $PYTHON-config --includes... $ECHO_C" >&6; }
-PYTHON_INCLUDES=`$PYTHON-config --includes`
+PYTHON_INCLUDES=`echo -I/usr/include/python2.4`
 if test $? = 0; then
   { echo "$as_me:$LINENO: result: $PYTHON_INCLUDES" >&5
 echo "${ECHO_T}$PYTHON_INCLUDES" >&6; }
