Index: libreport-2.0.6/src/gtk-helpers/Makefile.am
===================================================================
--- libreport-2.0.6.orig/src/gtk-helpers/Makefile.am
+++ libreport-2.0.6/src/gtk-helpers/Makefile.am
@@ -26,6 +26,7 @@ libreport_gtk_la_LDFLAGS = \
 libreport_gtk_la_LIBADD = \
     $(GTK_LIBS) \
     $(GLIB_LIBS) \
+    $(LIBREPORT_LIBS) \
     $(GNOME_KEYRING_LIBS)
 
 EXTRA_DIST = libreport-gtk.pc.in
Index: libreport-2.0.6/src/lib/Makefile.am
===================================================================
--- libreport-2.0.6.orig/src/lib/Makefile.am
+++ libreport-2.0.6/src/lib/Makefile.am
@@ -116,6 +116,7 @@ libabrt_web_la_LIBADD = \
     $(GLIB_LIBS) \
     $(CURL_LIBS) \
     $(LIBXML_LIBS) \
+    $(LIBREPORT_LIBS) \
     $(XMLRPC_LIBS) $(XMLRPC_CLIENT_LIBS)
 
 MAN_TXT = \
