Index: libreport-2.0.6/src/plugins/Makefile.am
===================================================================
--- libreport-2.0.6.orig/src/plugins/Makefile.am
+++ libreport-2.0.6/src/plugins/Makefile.am
@@ -169,7 +169,8 @@ reporter_kerneloops_CPPFLAGS = \
     -Wall -Wwrite-strings -Werror
 reporter_kerneloops_LDADD = \
     ../lib/libabrt_web.la \
-    ../lib/libreport.la
+    ../lib/libreport.la \
+    $(CURL_LIBS)
 
 reporter_mailx_SOURCES = \
     reporter-mailx.c
