Index: libreport-2.0.6/src/lib/dump_dir.c
===================================================================
--- libreport-2.0.6.orig/src/lib/dump_dir.c
+++ libreport-2.0.6/src/lib/dump_dir.c
@@ -510,10 +510,8 @@ void dd_create_basic_files(struct dump_d
     if (release)
         return;
 
-    release = load_text_file("/etc/system-release",
+    release = load_text_file("/etc/pardus-release",
             DD_LOAD_TEXT_RETURN_NULL_ON_FAILURE);
-    if (!release)
-        release = load_text_file("/etc/redhat-release", /*flags:*/ 0);
     dd_save_text(dd, FILENAME_OS_RELEASE, release);
     free(release);
 }
Index: libreport-2.0.6/src/report-python/__init__.py
===================================================================
--- libreport-2.0.6.orig/src/report-python/__init__.py
+++ libreport-2.0.6/src/report-python/__init__.py
@@ -21,8 +21,8 @@ from report.io import TextIO, GTKIO, New
 
 import os
 
-SYSTEM_RELEASE_PATHS = ["/etc/system-release","/etc/redhat-release"]
-SYSTEM_RELEASE_DEPS = ["system-release", "redhat-release"]
+SYSTEM_RELEASE_PATHS = ["/etc/pardus-release"]
+SYSTEM_RELEASE_DEPS = ["pardus-release"]
 
 _hardcoded_default_product = ""
 _hardcoded_default_version = ""
