diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kgraphviewer-2.0.2-kde4.2.0-orig/src/kgrapheditor.desktop kgraphviewer-2.0.2-kde4.2.0/src/kgrapheditor.desktop
--- kgraphviewer-2.0.2-kde4.2.0-orig/src/kgrapheditor.desktop	2009-03-17 12:45:50.000000000 +0200
+++ kgraphviewer-2.0.2-kde4.2.0/src/kgrapheditor.desktop	2009-03-17 12:48:47.000000000 +0200
@@ -12,6 +12,8 @@
 Type=Application
 Categories=Qt;KDE;Application;Graphics;
 DocPath=kgrapheditor/kgrapheditor.html
+GenericName=Graph Editor
+GenericName[tr]=Graf Düzenleyici
 Comment=A Graphviz dot graph editor for KDE
 Comment[cs]=Editor bodových grafů Graphviz pro KDE
 Comment[da]=Et program til redigering af Graphviz dot til KDE
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kgraphviewer-2.0.2-kde4.2.0-orig/src/kgraphviewer.desktop kgraphviewer-2.0.2-kde4.2.0/src/kgraphviewer.desktop
--- kgraphviewer-2.0.2-kde4.2.0-orig/src/kgraphviewer.desktop	2009-03-17 12:45:50.000000000 +0200
+++ kgraphviewer-2.0.2-kde4.2.0/src/kgraphviewer.desktop	2009-03-17 12:48:46.000000000 +0200
@@ -12,6 +12,8 @@
 Type=Application
 Categories=Qt;KDE;Application;Graphics;
 DocPath=kgraphviewer/index.html
+GenericName=Graph Viewer
+GenericName[tr]=Graf Gösterici
 Comment=A Graphviz dot graph viewer for KDE
 Comment[bg]=Програма за преглед на Graphviz-графики в KDE
 Comment[cs]=Prohlížeč bodových grafů Graphviz pro KDE
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kgraphviewer-2.0.2-kde4.2.0-orig/src/kgrapheditor.desktop kgraphviewer-2.0.2-kde4.2.0/src/kgrapheditor.desktop
--- kgraphviewer-2.0.2-kde4.2.0-orig/src/kgrapheditor.desktop	2009-03-17 13:08:43.000000000 +0200
+++ kgraphviewer-2.0.2-kde4.2.0/src/kgrapheditor.desktop	2009-03-17 13:09:10.000000000 +0200
@@ -8,7 +8,7 @@
 Name[tr]=Kgrapheditor
 Name[x-test]=xxkgrapheditorxx
 Exec=kgrapheditor %U
-Icon=kgrapheditor
+Icon=kgraphviewer
 Type=Application
 Categories=Qt;KDE;Application;Graphics;
 DocPath=kgrapheditor/kgrapheditor.html
