Index: release/freedesktop/blender.desktop
===================================================================
--- release/freedesktop/blender.desktop.orig
+++ release/freedesktop/blender.desktop
@@ -5,7 +5,9 @@ GenericName[es]=modelador 3D
 GenericName[de]=3D Modellierer
 GenericName[fr]=modeleur 3D
 GenericName[ru]=Редактор 3D-моделей
+GenericName[tr]=3D modelleme programı
 Comment=3D modeling, animation, rendering and post-production
+Comment[tr]=3B modelleme, animasyon, rendering ve üretim sonrası uygulaması
 Comment[es]=modelado 3D, animación, renderizado y post-producción
 Exec=blender
 Icon=blender
