diff -Nur KoboDeluxe-0.4pre10-old/prefs.cpp KoboDeluxe-0.4pre10/prefs.cpp
--- KoboDeluxe-0.4pre10-old/prefs.cpp	2007-04-04 21:42:51.351080538 +0300
+++ KoboDeluxe-0.4pre10/prefs.cpp	2007-04-04 21:45:25.597631497 +0300
@@ -70,10 +70,10 @@
 	yesno("fullscreen", fullscreen, 0); desc("Fullscreen Display");
 	key("videodriver", videodriver, GFX_DRIVER_SDL2D);
 			desc("Display Driver");
-	key("width", width, 320); desc("Horizontal Resolution");
-	key("height", height, 240); desc("Vertical Resolution");
+	key("width", width, 640); desc("Horizontal Resolution");
+	key("height", height, 480); desc("Vertical Resolution");
 	key("depth", depth, 0); desc("Display Depth");
-	key("maxfps", max_fps, 0); desc("Maximum FPS");
+	key("maxfps", max_fps, 50); desc("Maximum FPS");
 	key("buffer", buffermode, GFX_BUFFER_SEMITRIPLE);
 			desc("Display Buffer Mode");
 	key("internalres", internalres, 1); desc("Texture Resolution");
