diff -ur binutils-2.18.50.0.7.orig/bfd/Makefile.am binutils-2.18.50.0.7/bfd/Makefile.am
--- binutils-2.18.50.0.7.orig/bfd/Makefile.am	2008-03-14 20:23:01.000000000 +0200
+++ binutils-2.18.50.0.7/bfd/Makefile.am	2008-05-08 22:58:20.000000000 +0300
@@ -764,7 +764,7 @@
 libbfd_la_SOURCES = $(BFD32_LIBS_CFILES)
 libbfd_la_DEPENDENCIES = $(OFILES) ofiles
 libbfd_la_LIBADD = `cat ofiles` @WIN32LIBADD@
-libbfd_la_LDFLAGS = -release `cat libtool-soversion` @WIN32LDFLAGS@
+libbfd_la_LDFLAGS = -Wl,-Bsymbolic-functions -release `cat libtool-soversion` @WIN32LDFLAGS@
 
 # libtool will build .libs/libbfd.a.  We create libbfd.a in the build
 # directory so that we don't have to convert all the programs that use
diff -ur binutils-2.18.50.0.7.orig/bfd/Makefile.in binutils-2.18.50.0.7/bfd/Makefile.in
--- binutils-2.18.50.0.7.orig/bfd/Makefile.in	2008-04-03 19:54:04.000000000 +0300
+++ binutils-2.18.50.0.7/bfd/Makefile.in	2008-05-08 22:58:20.000000000 +0300
@@ -957,7 +957,7 @@
 libbfd_la_SOURCES = $(BFD32_LIBS_CFILES)
 libbfd_la_DEPENDENCIES = $(OFILES) ofiles
 libbfd_la_LIBADD = `cat ofiles` @WIN32LIBADD@
-libbfd_la_LDFLAGS = -release `cat libtool-soversion` @WIN32LDFLAGS@
+libbfd_la_LDFLAGS = -Wl,-Bsymbolic-functions -release `cat libtool-soversion` @WIN32LDFLAGS@
 
 # libtool will build .libs/libbfd.a.  We create libbfd.a in the build
 # directory so that we don't have to convert all the programs that use
diff -ur binutils-2.18.50.0.7.orig/opcodes/Makefile.am binutils-2.18.50.0.7/opcodes/Makefile.am
--- binutils-2.18.50.0.7.orig/opcodes/Makefile.am	2008-03-14 20:23:03.000000000 +0200
+++ binutils-2.18.50.0.7/opcodes/Makefile.am	2008-05-08 22:58:25.000000000 +0300
@@ -371,7 +371,7 @@
 # hardcoded path to libbfd.so instead of relying on the entries in libbfd.la.
 libopcodes_la_DEPENDENCIES = $(OFILES) @SHARED_DEPENDENCIES@
 libopcodes_la_LIBADD = $(OFILES) @SHARED_LIBADD@
-libopcodes_la_LDFLAGS = -release `cat ../bfd/libtool-soversion` @SHARED_LDFLAGS@
+libopcodes_la_LDFLAGS = -Wl,-Bsymbolic-functions -release `cat ../bfd/libtool-soversion` @SHARED_LDFLAGS@
 
 # libtool will build .libs/libopcodes.a.  We create libopcodes.a in
 # the build directory so that we don't have to convert all the
diff -ur binutils-2.18.50.0.7.orig/opcodes/Makefile.in binutils-2.18.50.0.7/opcodes/Makefile.in
--- binutils-2.18.50.0.7.orig/opcodes/Makefile.in	2008-04-03 19:54:05.000000000 +0300
+++ binutils-2.18.50.0.7/opcodes/Makefile.in	2008-05-08 22:58:25.000000000 +0300
@@ -592,7 +592,7 @@
 # hardcoded path to libbfd.so instead of relying on the entries in libbfd.la.
 libopcodes_la_DEPENDENCIES = $(OFILES) @SHARED_DEPENDENCIES@
 libopcodes_la_LIBADD = $(OFILES) @SHARED_LIBADD@
-libopcodes_la_LDFLAGS = -release `cat ../bfd/libtool-soversion` @SHARED_LDFLAGS@
+libopcodes_la_LDFLAGS = -Wl,-Bsymbolic-functions -release `cat ../bfd/libtool-soversion` @SHARED_LDFLAGS@
 
 # libtool will build .libs/libopcodes.a.  We create libopcodes.a in
 # the build directory so that we don't have to convert all the
