diff -Naur zekr.orig/zekr.sh zekr/zekr.sh
--- zekr.orig/zekr.sh	2010-04-02 01:08:47.766717009 +0300
+++ zekr/zekr.sh	2010-04-02 01:09:54.213717553 +0300
@@ -6,7 +6,7 @@
 # See http://zekr.org/faq.html#linux for more details.
 
 # uncomment and change the following line to a proper Mozilla-GTK2 installation directory.
-export MOZILLA_FIVE_HOME=/usr/lib/firefox
+export MOZILLA_FIVE_HOME=/usr/lib/xulrunner-1.9.2
 
 if [ -z "$MOZILLA_FIVE_HOME" ]; then
 	echo "MOZILLA_FIVE_HOME environment variable is not set. Please set it to a Mozilla GTK2 1.4+ installation directory. See http://zekr.org/faq.html#linux for more info."
