--- xqf.desktop.in	2004-03-14 17:20:10.000000000 +0200
+++ xqf.desktop-yeni.in	2006-07-08 10:33:50.000000000 +0300
@@ -1,8 +1,11 @@
 [Desktop Entry]
 Encoding=UTF-8
 _Name=XQF
+_Name[tr]=XQF
 _GenericName=XQF Game Server Browser
+_GenerıcName[tr]=XQF oyun sunucu tarayıcısı
 _Comment=Locate and connect to game servers
+_Comment[tr]=Oyun sunucularının yerini öğren ve bağlan
 Exec=xqf
 Icon=xqf_32x32.png
 StartupNotify=true
