Index: linux-2.6.25/drivers/input/serio/i8042-x86ia64io.h
===================================================================
--- linux-2.6.25.orig/drivers/input/serio/i8042-x86ia64io.h
+++ linux-2.6.25/drivers/input/serio/i8042-x86ia64io.h
@@ -118,6 +118,14 @@ static struct dmi_system_id __initdata i
 			DMI_MATCH(DMI_PRODUCT_VERSION, "VS2005R2"),
 		},
 	},
+	{
+		.ident = "Gigabyte M912",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "GIGABYTE"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "M912"),
+			DMI_MATCH(DMI_PRODUCT_VERSION, "01"),
+		},
+	},
 	{ }
 };
 
