diff -ur dvd+rw-tools-6.1.orig/growisofs_mmc.cpp dvd+rw-tools-6.1/growisofs_mmc.cpp
--- dvd+rw-tools-6.1.orig/growisofs_mmc.cpp	2006-01-26 15:04:09.000000000 +0200
+++ dvd+rw-tools-6.1/growisofs_mmc.cpp	2006-04-29 00:18:03.000000000 +0300
@@ -1412,9 +1412,7 @@
     blocks += 15, blocks &= ~15;
 
     if (blocks <= split)
-	fprintf (stderr,":-( more than 50%% of space will be *wasted*!\n"
-			"    use single layer media for this recording\n"),
-	exit (FATAL_START(EMEDIUMTYPE));
+	fprintf (stderr,":-( more than 50%% of space will be *wasted*!\n");
 
     blocks /= 16;
     blocks += 1;
