diff -Nur splashutils-misc-0.1.3-old/fbtruetype/Makefile splashutils-misc-0.1.3/fbtruetype/Makefile
--- splashutils-misc-0.1.3-old/fbtruetype/Makefile	2006-09-18 22:03:07.270771808 +0300
+++ splashutils-misc-0.1.3/fbtruetype/Makefile	2006-09-18 22:05:14.018503224 +0300
@@ -19,7 +19,7 @@
 
 LIBDIR   = $(PREFIX)/$(LIB)
 
-PROGRAMS = fbtruetype fbtruetype.static
+PROGRAMS = fbtruetype
 OBJECTS  = fbtruetype.o messages.o console.o ttf.o luxisri.o
 
 LDFLAGS  =  -L$(LIBDIR)
@@ -47,20 +47,16 @@
 mostlyclean:
 	-rm -f *.o core
 clean:	mostlyclean
-	-rm -f fbtruetype fbtruetype.static
+	-rm -f fbtruetype
 
 fbtruetype: $(OBJECTS) 
 	@rm -f fbtruetype
 	$(LINKD) $(LDFLAGS) $(OBJECTS) $(LIBSD)
-fbtruetype.static: $(OBJECTS)
-	@rm -f fbtruetype.static
-	$(LINKS) $(LDFLAGS) $(OBJECTS) $(LIBSS)
 
 install:
 	$(INSTALL_PROG) -D fbtruetype $(DESTDIR)/sbin/fbtruetype
-	$(INSTALL_PROG) -D fbtruetype.static $(DESTDIR)/sbin/fbtruetype.static
 
 strip: $(PROGRAMS)
 	$(STRIP) -s $(PROGRAMS)
 
-.PHONY: mostlyclean clean fbtruetype fbtruetype.static all strip
+.PHONY: mostlyclean clean fbtruetype all strip
