diff -Nur lua-5.0.3-old/Makefile lua-5.0.3/Makefile
--- lua-5.0.3-old/Makefile	2006-07-17 18:48:40.455061784 +0300
+++ lua-5.0.3/Makefile	2006-07-17 18:50:01.068806632 +0300
@@ -38,8 +38,8 @@
 
 # shared libraries (for Linux)
 so:
-	ld -o lib/liblua.so.$V -shared src/*.o
-	ld -o lib/liblualib.so.$V -shared src/lib/*.o
+	$(CC) -o lib/liblua.so.$V -shared src/*.o
+	$(CC) -o lib/liblualib.so.$V -shared src/lib/*.o $(EXTRA_LIBS)
 	cd lib; ln -fs liblua.so.$V liblua.so; ln -fs liblualib.so.$V liblualib.so
 
 # binaries using shared libraries
