diff -Nur lua-5.0.3-old/etc/lua.pc lua-5.0.3/etc/lua.pc
--- lua-5.0.3-old/etc/lua.pc	1970-01-01 02:00:00.000000000 +0200
+++ lua-5.0.3/etc/lua.pc	2006-07-17 18:40:24.502458040 +0300
@@ -0,0 +1,12 @@
+prefix=/usr
+exec_prefix=\${prefix}
+includedir=\${prefix}/include
+libdir=\${exec_prefix}/$(get_libdir)
+interpreter=\${exec_prefix}/bin/lua
+compiler=\${exec_prefix}/bin/luac
+
+Name: Lua
+Description: An extension programming language
+Version: ${PV}
+Cflags: -I\${includedir}
+Libs: -L\${libdir} -llua -llualib -ldl -lm
