diff -Nur kbib/kbib.desktop kbib/kbib.desktop
--- kbib/kbib.desktop	2007-05-18 07:21:37.000000000 +0300
+++ kbib/kbib.desktop	2007-05-28 01:36:23.000000000 +0300
@@ -1,7 +1,9 @@
 [Desktop Entry]
 Comment= A bibliography reference manager
+Comment[tr]= Bir kaynakça dipnot yöneticisi
 Exec=kbib %i %m -caption "%c"
 GenericName=Bibliography Manager
+GenericName[tr]=Kaynakça Yöneticisi
 Icon=kbib
 MimeType=text/x-bibtex
 Name=KBib
diff -Nur kbib/kbib-end.desktop kbib/kbib-end.desktop
--- kbib/kbib-end.desktop	2006-11-12 05:33:10.000000000 +0200
+++ kbib/kbib-end.desktop	2007-05-28 01:46:40.000000000 +0300
@@ -1,5 +1,6 @@
 [Desktop Entry]
 Comment= BibTex management program
+Comment[tr]= BibTex yönetim programı
 Exec=kbib -i %f -t=end
 GenericName=
 Icon=kbib
diff -Nur kbib/kbib-isi.desktop kbib/kbib-isi.desktop
--- kbib/kbib-isi.desktop	2006-11-12 05:33:10.000000000 +0200
+++ kbib/kbib-isi.desktop	2007-05-28 01:46:30.000000000 +0300
@@ -1,5 +1,6 @@
 [Desktop Entry]
 Comment= BibTex management program
+Comment[tr]= BibTex yönetim programı
 Exec=kbib -i %f -t=isi
 GenericName=
 Icon=kbib
diff -Nur kbib/kbib-ris.desktop kbib/kbib-ris.desktop
--- kbib/kbib-ris.desktop	2006-11-12 05:33:10.000000000 +0200
+++ kbib/kbib-ris.desktop	2007-05-28 01:47:40.000000000 +0300
@@ -1,5 +1,6 @@
 [Desktop Entry]
 Comment= BibTex management program
+Comment[tr]=BibTex yönetim programı
 Exec=kbib -i %f -t=ris
 GenericName=
 Icon=kbib
diff -Nur kbib/x-bibtex.desktop kbib/x-bibtex.desktop
--- kbib/x-bibtex.desktop	2006-11-12 05:33:10.000000000 +0200
+++ kbib/x-bibtex.desktop	2007-05-28 01:50:57.000000000 +0300
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Comment=Bibliographic Data (Bibtex)
+Comment[tr]=Kaynakça Verisi(Bibtex)
 Icon=tex
 MimeType=text/x-bibtex
 Patterns=*.bib
diff -Nur kbib/x-endnote-refer.desktop kbib/x-endnote-refer.desktop
--- kbib/x-endnote-refer.desktop	2006-11-12 05:33:10.000000000 +0200
+++ kbib/x-endnote-refer.desktop	2007-05-28 01:50:40.000000000 +0300
@@ -1,5 +1,6 @@
 [Desktop Entry]
 Comment=Bibliographic Data(Endnote Refer format)
+Comment[tr]=Kaynakça Verisi(Endnote Dipnot biçemi)
 Hidden=false
 Icon=tex
 MimeType=application/x-endnote-refer
diff -Nur kbib/x-inst-for-scientific-info.desktop kbib/x-inst-for-scientific-info.desktop
--- kbib/x-inst-for-scientific-info.desktop	2006-11-12 05:33:10.000000000 +0200
+++ kbib/x-inst-for-scientific-info.desktop	2007-05-28 01:50:09.000000000 +0300
@@ -1,5 +1,6 @@
 [Desktop Entry]
 Comment=Bibliographic Data(ISI format)
+Comment[tr]=Kaynakça Verisi(ISI biçemi)
 Hidden=false
 Icon=tex
 MimeType=application/x-inst-for-scientific-info
diff -Nur kbib/x-isibib.desktop kbib/x-isibib.desktop
--- kbib/x-isibib.desktop	2006-11-12 05:33:10.000000000 +0200
+++ kbib/x-isibib.desktop	2007-05-28 01:49:57.000000000 +0300
@@ -1,5 +1,6 @@
 [Desktop Entry]
 Comment=Bibliographic Data(ISI format)
+Comment[tr]=Kaynakça Verisi(ISI biçemi)
 Hidden=false
 Icon=tex
 MimeType=text/x-isibib
diff -Nur kbib/x-research-info-systems.desktop kbib/x-research-info-systems.desktop
--- kbib/x-research-info-systems.desktop	2006-11-12 05:33:10.000000000 +0200
+++ kbib/x-research-info-systems.desktop	2007-05-28 01:49:42.000000000 +0300
@@ -1,5 +1,6 @@
 [Desktop Entry]
 Comment=Bibliographic Data(RIS format)
+Comment[tr]=Kaynakça Verisi(RIS biçemi)
 Hidden=false
 Icon=tex
 MimeType=application/x-research-info-systems
diff -Nur kbib/x-risbib.desktop kbib/x-risbib.desktop
--- kbib/x-risbib.desktop	2006-11-12 05:33:10.000000000 +0200
+++ kbib/x-risbib.desktop	2007-05-28 01:51:25.000000000 +0300
@@ -1,5 +1,6 @@
 [Desktop Entry]
 Comment=Bibliographic Data(RIS format)
+Comment[tr]=Kaynakça Verisi(RIS biçemi)
 Hidden=false
 Icon=tex
 MimeType=text/x-risbib
