--- kerry/src/kcm/search.cpp	2007-01-15 16:20:25.000000000 +0200
+++ kerry/src/kcm/search.cpp.new	2007-03-04 23:43:26.000000000 +0200
@@ -67,8 +67,7 @@
     globalKeys = new KGlobalAccel(this);
     globalKeys->insert( "Program:kerry", i18n("Kerry Beagle Search") );
 
-    KShortcut showDialogShortcut = KShortcut(ALT+Key_Space);
-    showDialogShortcut.append( KKey( Key_F12 ) );
+    KShortcut showDialogShortcut = KShortcut(Key_F9);
     globalKeys->insert( "Show Kerry Dialog", i18n("Show Search Dialog"), QString::null, showDialogShortcut, showDialogShortcut, 0, 0);
     globalKeys->insert( "Search Primary Selection with Kerry", i18n("Search Primary Selection"), QString::null, CTRL+ALT+Key_Space, CTRL+ALT+Key_Space, 0, 0);
 
