diff -Naur screenie-git.orig/screenie.desktop screenie-git/screenie.desktop
--- screenie-git.orig/screenie.desktop	2009-03-22 21:58:58.000000000 +0200
+++ screenie-git/screenie.desktop	2009-03-22 22:01:55.000000000 +0200
@@ -1,7 +1,11 @@
 [Desktop Entry]
 Name=Screenie
+GenericName=Take Fancy Screenshots
+GenericName[tr]=Ekran Görüntüsü Yakalayıcı
 Comment=A fancy screenshot composer
-Exec=/usr/bin/screenie
+Comment[tr]=Özel ekran görüntüleri oluşturun
+Exec=screenie
 Icon=screenie
+Terminal=false
 Type=Application
-Categories=Graphics;3DGraphics;
+Categories=Application;Graphics;Qt;
