diff -Nuar gdm-2.18.0.orig/config/gdm.conf.in gdm-2.18.0/config/gdm.conf.in
--- gdm-2.18.0.orig/config/gdm.conf.in	2007-03-25 08:14:01.000000000 +0300
+++ gdm-2.18.0/config/gdm.conf.in	2007-03-25 08:29:12.000000000 +0300
@@ -338,11 +338,11 @@
 # greeter.
 Browser=false
 # The default picture in the browser.
-#DefaultFace=@pixmapdir@/nobody.png
+DefaultFace=@pixmapdir@/gdm-default-face.png
 # User ID's less than the MinimalUID value will not be included in the face
 # browser or in the gdmselection list for Automatic/Timed login.  They will not
 # be displayed regardless of the settings for Include and Exclude.
-#MinimalUID=100
+MinimalUID=1000
 # Users listed in Include will be included in the face browser and in the
 # gdmsetup selection list for Automatic/Timed login.  Users should be separated
 # by commas.
@@ -359,9 +359,9 @@
 # large numbers of users and this feature should not be used in such
 # environments.  The setting of IncludeAll does nothing if Include is set to a
 # non-empty value.
-#IncludeAll=false
+IncludeAll=true
 # If user or user.png exists in this dir it will be used as his picture.
-#GlobalFaceDir=@datadir@/pixmaps/faces/
+GlobalFaceDir=@datadir@/pixmaps/faces/
 
 # File which contains the locale we show to the user.  Likely you want to use
 # the one shipped with GDM and edit it.  It is not a standard locale.alias
