diff -Naur kdocker-4.0.orig/kdocker.desktop kdocker-4.0/kdocker.desktop
--- kdocker-4.0.orig/helpers/kdocker.desktop	2009-09-16 22:05:54.398192945 +0300
+++ kdocker-4.0/helpers/kdocker.desktop	2009-09-16 22:26:10.217320143 +0300
@@ -1,7 +1,9 @@
 [Desktop Entry]
 Name=KDocker
 GenericName=Docking utility
+GenericName[tr]=Sistem Tepsisi Uygulaması
 Comment=Docks any application into system tray
+Comment[tr]=İstediğiniz uygulamayı sistem tepsisine küçültün
 Exec=kdocker
 TryExec=kdocker
 Icon=kdocker
