--- dialog-1.0-20060221/makefile.in.org	2006-01-19 00:42:18.000000000 +0200
+++ dialog-1.0-20060221/makefile.in	    2006-06-01 12:02:50.000000000 +0300
@@ -27,7 +27,7 @@
 localedir       = $(prefix)/@DATADIRNAME@/locale
 
 CFLAGS		= @CFLAGS@
-CPPFLAGS	= @CPPFLAGS@ @DEFS@ -I. -I$(srcdir) -DLOCALEDIR=\"$(localedir)\"
+CPPFLAGS	= @CPPFLAGS@ @DEFS@ -I. -I$(srcdir) -DLOCALEDIR=\"$(localedir)\" -DENABLE_NLS
 EXTRA_CFLAGS	= @EXTRA_CFLAGS@
 CC		= @CC@
 AR		= @AR@
