diff -Nur scummvm-1.1.1-old/Makefile.common scummvm-1.1.1/Makefile.common
--- scummvm-1.1.1-old/Makefile.common	2010-05-26 19:31:46.724993458 +0300
+++ scummvm-1.1.1/Makefile.common	2010-05-26 19:31:57.303993350 +0300
@@ -141,7 +141,7 @@
 # Build rule for NASM assembler files
 %.o: %.asm
 	$(QUIET)$(MKDIR) $(*D)
-	$(QUIET_NASM)$(NASM) -O1 $(NASMFLAGS) -g -o $*.o $(<)
+	$(QUIET_NASM)$(NASM) -O1 $(NASMFLAGS)  -o $*.o $(<)
 endif
 
 # Include the dependency tracking files.
