---
 arch/i386/boot/video.S |    4 ++++
 1 file changed, 4 insertions(+)

Index: linux-2.6.18/arch/i386/boot/video.S
===================================================================
--- linux-2.6.18.orig/arch/i386/boot/video.S
+++ linux-2.6.18/arch/i386/boot/video.S
@@ -125,8 +125,12 @@ video:	pushw	%ds		# We use different seg
 	call	mode_set			# Set the mode
 	jc	vid1
 
+#if 0
 	leaw	badmdt, %si			# Invalid mode ID
 	call	prtstr
+#else
+	jmp	vid1
+#endif /* CONFIG_VIDEO_IGNORE_BAD_MODE */
 vid2:	call	mode_menu
 vid1:
 #ifdef CONFIG_VIDEO_RETAIN
