diff -ur mpg123-1.2.0/src/libmpg123/dct36_3dnowext.S mpg123-noexec/src/libmpg123/dct36_3dnowext.S
--- mpg123-1.2.0/src/libmpg123/dct36_3dnowext.S	2008-02-01 01:09:42.000000000 +0200
+++ mpg123-noexec/src/libmpg123/dct36_3dnowext.S	2008-02-03 07:43:38.000000000 +0200
@@ -508,3 +508,5 @@
 	leave
 	ret
 	/* .size	ASM_NAME(dct36_3dnowext), .-ASM_NAME(dct36_3dnowext) */
+
+.section .note.GNU-stack,"",%progbits
diff -ur mpg123-1.2.0/src/libmpg123/dct36_3dnow.S mpg123-noexec/src/libmpg123/dct36_3dnow.S
--- mpg123-1.2.0/src/libmpg123/dct36_3dnow.S	2008-02-01 01:09:42.000000000 +0200
+++ mpg123-noexec/src/libmpg123/dct36_3dnow.S	2008-02-03 07:43:38.000000000 +0200
@@ -501,3 +501,5 @@
 	movl %ebp,%esp
 	popl %ebp
 	ret
+
+.section .note.GNU-stack,"",%progbits
diff -ur mpg123-1.2.0/src/libmpg123/dct64_3dnowext.S mpg123-noexec/src/libmpg123/dct64_3dnowext.S
--- mpg123-1.2.0/src/libmpg123/dct64_3dnowext.S	2008-02-01 01:09:42.000000000 +0200
+++ mpg123-noexec/src/libmpg123/dct64_3dnowext.S	2008-02-03 07:43:38.000000000 +0200
@@ -710,3 +710,5 @@
 	leave
 	ret
 	/* .size	ASM_NAME(dct64_3dnowext), .-ASM_NAME(dct64_3dnowext) */
+
+.section .note.GNU-stack,"",%progbits
diff -ur mpg123-1.2.0/src/libmpg123/dct64_3dnow.S mpg123-noexec/src/libmpg123/dct64_3dnow.S
--- mpg123-1.2.0/src/libmpg123/dct64_3dnow.S	2008-02-01 01:09:42.000000000 +0200
+++ mpg123-noexec/src/libmpg123/dct64_3dnow.S	2008-02-03 07:43:38.000000000 +0200
@@ -709,3 +709,5 @@
 	
         ret
 
+
+.section .note.GNU-stack,"",%progbits
diff -ur mpg123-1.2.0/src/libmpg123/dct64_mmx.S mpg123-noexec/src/libmpg123/dct64_mmx.S
--- mpg123-1.2.0/src/libmpg123/dct64_mmx.S	2008-02-01 01:09:42.000000000 +0200
+++ mpg123-noexec/src/libmpg123/dct64_mmx.S	2008-02-03 07:43:38.000000000 +0200
@@ -809,3 +809,5 @@
 	ret
 	
 
+
+.section .note.GNU-stack,"",%progbits
diff -ur mpg123-1.2.0/src/libmpg123/dct64_sse.S mpg123-noexec/src/libmpg123/dct64_sse.S
--- mpg123-1.2.0/src/libmpg123/dct64_sse.S	2008-02-01 01:09:42.000000000 +0200
+++ mpg123-noexec/src/libmpg123/dct64_sse.S	2008-02-03 07:43:38.000000000 +0200
@@ -555,3 +555,5 @@
 	popl	%ebp
 	ret
 	/* .size	ASM_NAME(dct64_sse), .-ASM_NAME(dct64_sse) */
+
+.section .note.GNU-stack,"",%progbits
diff -ur mpg123-1.2.0/src/libmpg123/decode_3dnowext.S mpg123-noexec/src/libmpg123/decode_3dnowext.S
--- mpg123-1.2.0/src/libmpg123/decode_3dnowext.S	2008-02-01 01:09:42.000000000 +0200
+++ mpg123-noexec/src/libmpg123/decode_3dnowext.S	2008-02-03 07:43:38.000000000 +0200
@@ -2,3 +2,5 @@
 #define MPL_DCT64 ASM_NAME(dct64_3dnowext)
 #define SYNTH_NAME ASM_NAME(synth_1to1_3dnowext_asm)
 #include "decode_sse3d.h"
+
+.section .note.GNU-stack,"",%progbits
diff -ur mpg123-1.2.0/src/libmpg123/decode_3dnow.S mpg123-noexec/src/libmpg123/decode_3dnow.S
--- mpg123-1.2.0/src/libmpg123/decode_3dnow.S	2008-02-01 01:09:42.000000000 +0200
+++ mpg123-noexec/src/libmpg123/decode_3dnow.S	2008-02-03 07:43:38.000000000 +0200
@@ -278,3 +278,5 @@
 	popl %ebp
 	addl $24,%esp
 	ret
+
+.section .note.GNU-stack,"",%progbits
diff -ur mpg123-1.2.0/src/libmpg123/decode_i586_dither.S mpg123-noexec/src/libmpg123/decode_i586_dither.S
--- mpg123-1.2.0/src/libmpg123/decode_i586_dither.S	2008-02-01 01:09:42.000000000 +0200
+++ mpg123-noexec/src/libmpg123/decode_i586_dither.S	2008-02-03 07:43:38.000000000 +0200
@@ -366,3 +366,5 @@
 /* The stack must be now: 0=back 4=bandptr 8=channel 12=out 16=buffs 20=bo */
 	ret
 
+
+.section .note.GNU-stack,"",%progbits
diff -ur mpg123-1.2.0/src/libmpg123/decode_i586.S mpg123-noexec/src/libmpg123/decode_i586.S
--- mpg123-1.2.0/src/libmpg123/decode_i586.S	2008-02-01 01:09:42.000000000 +0200
+++ mpg123-noexec/src/libmpg123/decode_i586.S	2008-02-03 07:43:38.000000000 +0200
@@ -333,3 +333,5 @@
 	addl $12,%esp
 	ret
 
+
+.section .note.GNU-stack,"",%progbits
diff -ur mpg123-1.2.0/src/libmpg123/decode_mmx.S mpg123-noexec/src/libmpg123/decode_mmx.S
--- mpg123-1.2.0/src/libmpg123/decode_mmx.S	2008-02-01 01:09:42.000000000 +0200
+++ mpg123-noexec/src/libmpg123/decode_mmx.S	2008-02-03 07:43:38.000000000 +0200
@@ -123,3 +123,5 @@
         ret
 
 
+
+.section .note.GNU-stack,"",%progbits
diff -ur mpg123-1.2.0/src/libmpg123/decode_sse.S mpg123-noexec/src/libmpg123/decode_sse.S
--- mpg123-1.2.0/src/libmpg123/decode_sse.S	2008-02-01 01:09:42.000000000 +0200
+++ mpg123-noexec/src/libmpg123/decode_sse.S	2008-02-03 07:43:38.000000000 +0200
@@ -2,3 +2,5 @@
 #define MPL_DCT64 ASM_NAME(dct64_sse)
 #define SYNTH_NAME ASM_NAME(synth_1to1_sse_asm)
 #include "decode_sse3d.h"
+
+.section .note.GNU-stack,"",%progbits
diff -ur mpg123-1.2.0/src/libmpg123/equalizer_3dnow.S mpg123-noexec/src/libmpg123/equalizer_3dnow.S
--- mpg123-1.2.0/src/libmpg123/equalizer_3dnow.S	2008-02-01 01:09:42.000000000 +0200
+++ mpg123-noexec/src/libmpg123/equalizer_3dnow.S	2008-02-03 07:43:38.000000000 +0200
@@ -66,3 +66,5 @@
 	popl %ebx
 	popl %esi
 	ret
+
+.section .note.GNU-stack,"",%progbits
diff -ur mpg123-1.2.0/src/libmpg123/getcpuflags.S mpg123-noexec/src/libmpg123/getcpuflags.S
--- mpg123-1.2.0/src/libmpg123/getcpuflags.S	2008-02-01 01:09:42.000000000 +0200
+++ mpg123-noexec/src/libmpg123/getcpuflags.S	2008-02-03 07:43:38.000000000 +0200
@@ -77,3 +77,5 @@
 	movl %ebp,%esp
 	popl %ebp
 	ret
+
+.section .note.GNU-stack,"",%progbits
diff -ur mpg123-1.2.0/src/libmpg123/tabinit_mmx.S mpg123-noexec/src/libmpg123/tabinit_mmx.S
--- mpg123-1.2.0/src/libmpg123/tabinit_mmx.S	2008-02-01 01:09:42.000000000 +0200
+++ mpg123-noexec/src/libmpg123/tabinit_mmx.S	2008-02-03 07:43:38.000000000 +0200
@@ -207,3 +207,5 @@
 	popl %edi
 	ret
 
+
+.section .note.GNU-stack,"",%progbits
