--- configure.orig	2007-11-13 00:46:37.000000000 +0200
+++ configure	2009-10-02 21:58:39.228771828 +0300
@@ -6859,19 +6859,19 @@
 
 
 # --- DOCDIR ---
-DOCDIR='${prefix}/${PACKAGE}/doc'
+DOCDIR='${prefix}/${DATADIRNAME}/doc/${PACKAGE}/html'
 
 
 # --- FONTDIR ---
-FONTDIR='${prefix}/${PACKAGE}/font'
+FONTDIR='${prefix}/${DATADIRNAME}/${PACKAGE}/font'
 
 
 # --- TOOLDIR ---
-TOOLDIR='${prefix}/${PACKAGE}/tool'
+TOOLDIR='${prefix}/${DATADIRNAME}/${PACKAGE}/tool'
 
 
 # --- EXAMPLEDIR ---
-EXAMPLEDIR='${prefix}/${PACKAGE}/example'
+EXAMPLEDIR='${prefix}/${DATADIRNAME}/doc/${PACKAGE}/examples'
 
 
 
