--- src/global.c.orig	2007-11-08 05:47:28.000000000 +0200
+++ src/global.c	2009-10-02 22:07:18.437772999 +0300
@@ -71,14 +71,14 @@
 	/* 設定ファイルのパスの取得 */
 	GetConfigPath(sagcad_user.ConfigPath);
 
-	strcpy(sagcad_user.BrowserPath, "netscape");
+	strcpy(sagcad_user.BrowserPath, "x-www-browser");
 	home_of_user(home);
 	strcpy(sagcad_user.CadPath, home);
 	strcpy(sagcad_user.KeepCadPath, home);
 	strcpy(sagcad_user.NcPath, "CADPATH");
 
-	strcpy(sagcad_user.browser_command, "mozilla %s");
-	strcpy(sagcad_user.editor_command, "gvim %s");
+	strcpy(sagcad_user.browser_command, "x-www-browser %s");
+	strcpy(sagcad_user.editor_command, "kwrite %s");
 	strcpy(sagcad_user.print_command, "lpr %s");
 
 
