diff -Nuar sagcad-0.9.14.orig/tools/Makefile.am sagcad-0.9.14/tools/Makefile.am
--- sagcad-0.9.14.orig/tools/Makefile.am	2007-11-06 16:03:34.000000000 +0200
+++ sagcad-0.9.14/tools/Makefile.am	2009-10-02 23:13:09.653771882 +0300
@@ -9,7 +9,7 @@
 	test.ops
 
 #install-data-hook:
-	chmod 777 $(tooldatadir)
+#	chmod 777 $(tooldatadir)
 #	chmod 755 $(tooldata_DATA)
 
 EXTRA_DIST = $(tooldata_DATA)
diff -Nuar sagcad-0.9.14.orig/tools/Makefile.in sagcad-0.9.14/tools/Makefile.in
--- sagcad-0.9.14.orig/tools/Makefile.in	2007-11-13 00:46:37.000000000 +0200
+++ sagcad-0.9.14/tools/Makefile.in	2009-10-02 23:14:03.062776165 +0300
@@ -536,7 +536,7 @@
 
 
 #install-data-hook:
-	chmod 777 $(tooldatadir)
+#	chmod 777 $(tooldatadir)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -Nuar sagcad-0.9.14.orig/tools/ruby/Makefile.am sagcad-0.9.14/tools/ruby/Makefile.am
--- sagcad-0.9.14.orig/tools/ruby/Makefile.am	2007-11-06 16:03:34.000000000 +0200
+++ sagcad-0.9.14/tools/ruby/Makefile.am	2009-10-02 23:14:44.277772748 +0300
@@ -8,7 +8,7 @@
 	test.rb
 
 #install-data-hook:
-	chmod 777 $(tooldatadir)
+#	chmod 777 $(tooldatadir)
 #	chmod 755 $(tooldata_SCRIPTS)
 
 EXTRA_DIST = $(tooldata_SCRIPTS)
diff -Nuar sagcad-0.9.14.orig/tools/ruby/Makefile.in sagcad-0.9.14/tools/ruby/Makefile.in
--- sagcad-0.9.14.orig/tools/ruby/Makefile.in	2007-11-13 00:46:37.000000000 +0200
+++ sagcad-0.9.14/tools/ruby/Makefile.in	2009-10-02 23:17:48.926774282 +0300
@@ -374,7 +374,7 @@
 
 
 #install-data-hook:
-	chmod 777 $(tooldatadir)
+#	chmod 777 $(tooldatadir)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
