diff -Naur imdboid.orig/imdbqueryengine.cpp imdboid/imdbqueryengine.cpp
--- imdboid.orig/imdbqueryengine.cpp	2010-02-09 21:03:08.489001935 +0200
+++ imdboid/imdbqueryengine.cpp	2010-02-09 21:05:42.061249486 +0200
@@ -166,7 +166,7 @@
   // for more images maybe interesting: 
   //http://www.imdb.com/title/tt0099685/mediaindex 
   QStringList line;
-  reg.setPattern( QString("\"poster\" href=\"/rg/action-box-title/primary-photo/media/rm[0-9]*/tt[0-9]*\" title=\"[a-zA-Z0-9\ ]*\"><img border=\"0\" alt=\"[a-zA-Z0-9\ ]*\" title=\"[a-zA-Z0-9\ ]*\" src=\"http://ia.media-imdb.com/images/[A-Z]/([A-Za-z0-9@._]*|[A-Za-z0-9@.]*|[A-Za-z0-9._]*).jpg\" /></a>") );
+  reg.setPattern( QString("\"poster\" href=\"/rg/action-box-title/primary-photo/media/rm[0-9]*/tt[0-9]*\" title=\"[a-zA-Z0-9\\ ]*\"><img border=\"0\" alt=\"[a-zA-Z0-9\\ ]*\" title=\"[a-zA-Z0-9\\ ]*\" src=\"http://ia.media-imdb.com/images/[A-Z]/([A-Za-z0-9@._]*|[A-Za-z0-9@.]*|[A-Za-z0-9._]*).jpg\" /></a>") );
   while( ( pos = reg.indexIn(data,pos))  != -1 ) {
     kDebug() << reg.capturedTexts();
     pos += reg.matchedLength();
