diff -Naur banshee-1-1.9.0.orig/src/Clients/Booter/banshee-1.linux.in banshee-1-1.9.0/src/Clients/Booter/banshee-1.linux.in
--- banshee-1-1.9.0.orig/src/Clients/Booter/banshee-1.linux.in	2010-11-14 00:27:29.512973288 +0200
+++ banshee-1-1.9.0/src/Clients/Booter/banshee-1.linux.in	2010-11-14 00:28:33.025161548 +0200
@@ -6,7 +6,7 @@
 MONO_EXE="@expanded_libdir@/@PACKAGE@/$exec_asm"
 BANSHEE_EXEC_NAME=$(basename $0)
 BANSHEE_CONFIG_DIR="${XDG_CONFIG_HOME:-$HOME/.config}/banshee-1"
-
+export XLIB_SKIP_ARGB_VISUALS=1
 export LD_LIBRARY_PATH=@expanded_libdir@/@PACKAGE@:@expanded_libdir@/@PACKAGE@/Extensions:@expanded_libdir@/@PACKAGE@/Backends:@expanded_libdir@${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
 export GST_PLUGIN_PATH=@expanded_libdir@/@PACKAGE@/gstreamer-0.10${GST_PLUGIN_PATH:+:$GST_PLUGIN_PATH}
 if [ $BANSHEE_EXEC_NAME = "muinshee" ]; then
