Index: data/diskmanager.desktop
===================================================================
--- data/diskmanager.desktop	(revision 27548)
+++ data/diskmanager.desktop	(revision 27550)
@@ -12,7 +12,7 @@
 GenericName[es]=Administrador de Montaje de particiones
 GenericName[de]=Disk-mount Manager
 GenericName[nl]=Schijfbeheer
-GenericName[fr]=Gestionnaire de disques
+GenericName[fr]=Gestionnaire des Disques
 Comment=An application for managing partitions and their mount paths.
 Comment[tr]=Disk bölümlerini ve bağlandıkları dizinleri yönetmenizi sağlayan bir uygulama.
 Comment[sv]=Ett program för att hantera partitioner och deras monteringspunkter.
