diff --git a/setup.py b/setup.py
index a1713e8..356c7d4 100644
--- a/setup.py
+++ b/setup.py
@@ -884,7 +884,7 @@ class PyBuildExt(build_ext):
                   and find_file("gdbm/ndbm.h", inc_dirs, []) is not None):
                 exts.append( Extension('dbm', ['dbmmodule.c'],
                                        define_macros=[('HAVE_GDBM_NDBM_H',None)],
-                                       libraries = ['gdbm'] ) )
+                                       libraries = ['gdbm','gdbm_compat'] ) )
             elif db_incs is not None:
                 exts.append( Extension('dbm', ['dbmmodule.c'],
                                        library_dirs=dblib_dir,
