diff -ur gparted-0.3.1.orig/gparted.desktop gparted-0.3.1/gparted.desktop
--- gparted-0.3.1.orig/gparted.desktop	2006-09-10 16:35:42.000000000 +0300
+++ gparted-0.3.1/gparted.desktop	2006-12-10 17:42:56.000000000 +0200
@@ -51,8 +51,28 @@
 Comment[pa]=ਗਨੋਮ ਭਾਗ ਸੰਪਾਦਕ
 Comment[pt_BR]=Editor de Partições do GNOME
 Comment[sv]=Partitionsredigerare för GNOME
-Comment[tr]=GNOME Bölümleme Düzenleyicisi
+Comment[tr]=Disk Bölümleme Aracı
 Comment[vi]=Bộ hiệu chỉnh phân vùng Gnome
+GenericName=GNOME Partition Editor
+GenericName[bg]=Редактор на дялове на диск за GNOME
+GenericName[ca]=Editor de particions del GNOME
+GenericName[dz]=ཇི་ནོམ་བར་བཅད་ཞུན་དགཔ།
+GenericName[en_CA]=GNOME Partition Editor
+GenericName[es]=Editor de particiones de GNOME
+GenericName[fi]=Gnomen levyosioiden muokkain
+GenericName[fr]=Éditeur de partition de Gnome
+GenericName[gl]=Editor de particións de Gnome
+GenericName[hu]=GNOME partíciószerkesztő
+GenericName[ja]=Gnome パーティション編集ツール
+GenericName[lt]=GNOME skirsnių redaktorius
+GenericName[lv]=GNOME partīciju redaktors
+GenericName[mk]=Уредувач на партиции за Гном
+GenericName[nl]=Gnome Partitiebewerker
+GenericName[pa]=ਗਨੋਮ ਭਾਗ ਸੰਪਾਦਕ
+GenericName[pt_BR]=Editor de Partições do GNOME
+GenericName[sv]=Partitionsredigerare för GNOME
+GenericName[tr]=Disk Bölümleme Aracı
+GenericName[vi]=Bộ hiệu chỉnh phân vùng Gnome
 Exec=gparted
 Icon=gparted.png
 Terminal=false
