Index: system-config-printer-1.1.8/system-config-printer.py
===================================================================
--- system-config-printer-1.1.8.orig/system-config-printer.py
+++ system-config-printer-1.1.8/system-config-printer.py
@@ -1458,7 +1458,7 @@ class GUI(GtkGUI, monitor.Watcher):
 
                 emblem = None
                 if name == self.default_printer:
-                    emblem = 'emblem-default'
+                    emblem = 'dialog-ok'
                 elif name == userdef:
                     emblem = 'emblem-favorite'
 
