diff -urN gtkdatabox-0.8.2.0.orig/examples/Makefile.am gtkdatabox-0.8.2.0/examples/Makefile.am
--- gtkdatabox-0.8.2.0.orig/examples/Makefile.am	2007-12-18 10:36:44.000000000 -0800
+++ gtkdatabox-0.8.2.0/examples/Makefile.am	2007-12-18 10:38:00.000000000 -0800
@@ -41,7 +41,7 @@
 			-I$(top_srcdir)/gtk
 
 AM_CFLAGS			=\
-			-O2\
+			$(RPM_OPT_FLAGS)\
 			-W -Wall -pedantic\
 			-DG_DISABLE_DEPRECATED\
 			-DGDK_DISABLE_DEPRECATED\
diff -urN gtkdatabox-0.8.2.0.orig/examples/Makefile.in gtkdatabox-0.8.2.0/examples/Makefile.in
--- gtkdatabox-0.8.2.0.orig/examples/Makefile.in	2007-12-18 10:36:44.000000000 -0800
+++ gtkdatabox-0.8.2.0/examples/Makefile.in	2007-12-18 10:38:00.000000000 -0800
@@ -264,7 +264,7 @@
 			-I$(top_srcdir)/gtk
 
 AM_CFLAGS = \
-			-O2\
+			$(RPM_OPT_FLAGS)\
 			-W -Wall -pedantic\
 			-DG_DISABLE_DEPRECATED\
 			-DGDK_DISABLE_DEPRECATED\
diff -urN gtkdatabox-0.8.2.0.orig/gtk/Makefile.am gtkdatabox-0.8.2.0/gtk/Makefile.am
--- gtkdatabox-0.8.2.0.orig/gtk/Makefile.am	2007-12-18 10:36:44.000000000 -0800
+++ gtkdatabox-0.8.2.0/gtk/Makefile.am	2007-12-18 10:38:00.000000000 -0800
@@ -48,7 +48,7 @@
 			-I$(top_srcdir)
 
 AM_CFLAGS			=\
-			-O2\
+			$(RPM_OPT_FLAGS)\
 			-W -Wall -pedantic\
 			-DG_DISABLE_DEPRECATED\
 			-DGDK_DISABLE_DEPRECATED\
diff -urN gtkdatabox-0.8.2.0.orig/gtk/Makefile.in gtkdatabox-0.8.2.0/gtk/Makefile.in
--- gtkdatabox-0.8.2.0.orig/gtk/Makefile.in	2007-12-18 10:36:44.000000000 -0800
+++ gtkdatabox-0.8.2.0/gtk/Makefile.in	2007-12-18 10:38:00.000000000 -0800
@@ -246,7 +246,7 @@
 			-I$(top_srcdir)
 
 AM_CFLAGS = \
-			-O2\
+			$(RPM_OPT_FLAGS)\
 			-W -Wall -pedantic\
 			-DG_DISABLE_DEPRECATED\
 			-DGDK_DISABLE_DEPRECATED\
