--- data/menu/freetuxtv.desktop.in.orig	2010-07-20 17:23:46.000000000 +0000
+++ data/menu/freetuxtv.desktop.in	2010-08-22 15:22:18.541274734 +0000
@@ -1,7 +1,9 @@
 [Desktop Entry]
 _Name=FreetuxTV Television Channels Player
 _GenericName=Television Channels Player
+_GenericName[tr]=İnternet Tv
 _Comment=Watch television channels from internet
+_Comment[tr]=İnternetten Tv izleyabilirsiniz
 Categories=AudioVideo;TV;
 Encoding=UTF-8
 Exec=freetuxtv
