diff -up gtkpod-0.99.13SVN/src/prefs.c.use-xdg-open gtkpod-0.99.13SVN/src/prefs.c
--- gtkpod-0.99.13SVN/src/prefs.c.use-xdg-open	2008-11-18 18:21:27.000000000 -0500
+++ gtkpod-0.99.13SVN/src/prefs.c	2008-11-18 18:22:06.000000000 -0500
@@ -180,8 +180,8 @@ static void set_default_preferences()
     prefs_set_int("delete_local_file", TRUE);
     prefs_set_int("delete_database", TRUE);
     prefs_set_string("initial_mountpoint", "/mnt/ipod");
-    prefs_set_string ("path_play_now", "xmms %s");
-    prefs_set_string ("path_play_enqueue", "xmms -e %s");
+    prefs_set_string ("path_play_now", "xdg-open %s");
+    prefs_set_string ("path_play_enqueue", "");
     prefs_set_string ("path_mserv_trackinfo_root", "/var/lib/mserv/trackinfo/");
     prefs_set_string ("coverart_display_bg_color", "#000000");
     prefs_set_string ("coverart_display_fg_color", "#FFFFFF");
