Index: scripts/sync-tomboy.sh
===================================================================
--- scripts/sync-tomboy.sh	(Revision 2382)
+++ scripts/sync-tomboy.sh	(Arbeitskopie)
@@ -12,7 +12,7 @@
 # with the following defaults: 
 
 IPOD_MOUNT=/media/ipod                    # mountpoint of ipod
-NOTESPATH=~/.tomboy			  # path to folder containing notes
+NOTESPATH=~/.local/share/tomboy		  # path to folder containing notes
 ENCODING=ISO-8859-15                      # encoding used by ipod
 
 # Unless called with "-e=none" this script requires "iconv" available
