diff -r 712276328f75 virtinst/VirtualGraphics.py
--- a/virtinst/VirtualGraphics.py	Wed Apr 08 10:27:20 2009 -0400
+++ b/virtinst/VirtualGraphics.py	Thu Apr 09 09:48:36 2009 -0400
@@ -55,8 +55,9 @@
     def get_keymap(self):
         return self._keymap
     def set_keymap(self, val):
-        if not val:
-            val = _util.default_keymap()
+        if val == None:
+            self._keymap = val
+            return
         if not val or type(val) != type("string"):
             raise ValueError, _("Keymap must be a string")
         if len(val) > 16:
