--- configure.orig	2008-10-16 12:38:48.000000000 +0300
+++ configure	2008-12-14 19:43:55.000000000 +0200
@@ -284,30 +284,30 @@
 
 ID=`id -u`
 if [ $ID -eq 0 ]; then
-	'${prefix}'/bin/mountmanager
+	/usr/bin/mountmanager
 elif [ -n "$KDESU_PROG" ]; then
 	if [[ "$1" = "--debug" ]]; then
-		$KDESU_PROG -d -n '${prefix}'/bin/mountmanager
+		$KDESU_PROG -d -n /usr/bin/mountmanager
 	else
-		$KDESU_PROG -d -n "'${prefix}'/bin/mountmanager 2> /var/log/MountManager.log"
+		$KDESU_PROG -d -n "/usr/bin/mountmanager 2> /var/log/MountManager.log"
 	fi
 elif [ -n "$KDESUDO_PROG" ]; then
 	if [[ "$1" = "--debug" ]]; then
-		$KDESUDO_PROG -d -n '${prefix}'/bin/mountmanager
+		$KDESUDO_PROG -d -n /usr/bin/mountmanager
 	else
-		$KDESUDO_PROG -d -n "'${prefix}'/bin/mountmanager 2> /var/log/MountManager.log"
+		$KDESUDO_PROG -d -n "/usr/bin/mountmanager 2> /var/log/MountManager.log"
 	fi
 elif [ -n "$GTKSU_PROG" ]; then
 	if [[ "$1" = "--debug" ]]; then
-		$GTKSU_PROG '${prefix}'/bin/mountmanager
+		$GTKSU_PROG /usr/bin/mountmanager
 	else
-		$GTKSU_PROG "'${prefix}'/bin/mountmanager 2> /var/log/MountManager.log"
+		$GTKSU_PROG "/usr/bin/mountmanager 2> /var/log/MountManager.log"
 	fi
 elif [ -n "$GKSU_PROG" ]; then
 	if [[ "$1" = "--debug" ]]; then
-		$GKSU_PROG '${prefix}'/bin/mountmanager
+		$GKSU_PROG /usr/bin/mountmanager
 	else
-		$GKSU_PROG "'${prefix}'/bin/mountmanager 2> /var/log/MountManager.log"
+		$GKSU_PROG "/usr/bin/mountmanager 2> /var/log/MountManager.log"
 	fi
 elif [ -n "$KDIALOG_PROG" ]; then
 	$KDIALOG_PROG "Install kdesu, kdesudo,gtksu or gksu program to get root permissions"
