--- konversation/src/osd.cpp
+++ konversation/src/osd.cpp
@@ -241,7 +241,7 @@
 void OSDWidget::show()
 {
     // Don't show the OSD widget when the desktop is locked
-    if ( isKDesktopLockRunning() )
+    if ( isKDesktopLockRunning() == Locked )
     {
         minReached();                             // don't queue the message
         return;
@@ -441,7 +439,7 @@
         kdWarning() << k_funcinfo << ": Strange return value from screensaver. "
             << "Assuming screensaver is active." << endl;
         // Err on the side of safety.
-        return DCOPError;
+        return Locked;
     }
 }
 

