Index: linux-2.6.31/drivers/staging/rt2870/rt2870.h
===================================================================
--- linux-2.6.31.orig/drivers/staging/rt2870/rt2870.h
+++ linux-2.6.31/drivers/staging/rt2870/rt2870.h
@@ -115,6 +115,7 @@
 	{USB_DEVICE(0x083A,0x7522)}, /* Arcadyan */		\
 	{USB_DEVICE(0x0CDE,0x0022)}, /* ZCOM */			\
 	{USB_DEVICE(0x0586,0x3416)}, /* Zyxel */		\
+	{USB_DEVICE(0x0586,0x341a)}, /* Zyxel */		\
 	{USB_DEVICE(0x0CDE,0x0025)}, /* Zyxel */		\
 	{USB_DEVICE(0x1740,0x9701)}, /* EnGenius */		\
 	{USB_DEVICE(0x1740,0x9702)}, /* EnGenius */		\
