diff -Naur sparklemedia-0.2.orig/app/sparklemedia.desktop sparklemedia-0.2/app/sparklemedia.desktop
--- sparklemedia-0.2.orig/app/sparklemedia.desktop	2010-01-16 02:46:24.832272855 +0200
+++ sparklemedia-0.2/app/sparklemedia.desktop	2010-01-16 02:47:10.622273778 +0200
@@ -4,8 +4,10 @@
 Encoding=UTF-8
 Name=Sparkle Media
 GenericName=Media Player
+GenericName[tr]=Medya Oynatıcı
 Exec=sparklemedia
 Comment=Media Manager and Player
+Comment[tr]=Medya Yöneticisi ve Oynatıcısı
 Icon=sparklemedia
 Terminal=false
 Categories=Qt;KDE;AudioVideo;Audio;Video;Player;
