Index: kdelibs/kio/kio/kdirwatch.cpp
===================================================================
--- kdelibs.orig/kio/kio/kdirwatch.cpp
+++ kdelibs/kio/kio/kdirwatch.cpp
@@ -71,7 +71,11 @@
 #include <linux/types.h>
 // Linux kernel headers are documented to not compile
 #define _S390_BITOPS_H
+#define flock linux_flock
+#define flock64 linux_flock64
 #include <sys/inotify.h>
+#undef flock
+#undef flock64
 
 #ifndef __NR_inotify_init
 #if defined(__i386__)
