Index: pisi/atomicoperations.py
===================================================================
--- pisi/atomicoperations.py	(revision 21148)
+++ pisi/atomicoperations.py	(revision 21149)
@@ -337,9 +337,11 @@
                 if not os.path.exists(destdir):
                     os.makedirs(destdir)
 
-                if os.path.islink(old_path): 
-                    if not os.path.lexists(new_path):
-                        os.symlink(os.readlink(old_path), new_path)
+                if os.path.lexists(new_path):
+                    os.unlink(new_path)
+
+                if os.path.islink(old_path):
+                    os.symlink(os.readlink(old_path), new_path)
                 else:
                     shutil.copy(old_path, new_path)
 
