Index: pisi/operations/upgrade.py
===================================================================
--- pisi/operations/upgrade.py	(revision 21190)
+++ pisi/operations/upgrade.py	(revision 21191)
@@ -209,7 +209,7 @@
         Bp = set()
         for x in B:
             pkg = packagedb.get_package(x)
-            rev_deps = installdb.get_rev_deps(x)
+            rev_deps = packagedb.get_rev_deps(x)
             for (rev_dep, depinfo) in rev_deps:
                 # add only installed but unsatisfied reverse dependencies
                 if (installdb.has_package(rev_dep) and 
