diff -Nur gspcav1-20070508-old/Pixart/pac207.h gspcav1-20070508/Pixart/pac207.h
--- gspcav1-20070508-old/Pixart/pac207.h	2007-11-01 10:37:43.727448887 +0200
+++ gspcav1-20070508/Pixart/pac207.h	2007-11-01 10:39:02.544466878 +0200
@@ -134,7 +134,7 @@
     id[1] = id[1] & 0x0f;
     PDEBUG(2, " Pixart Sensor ID 0x%02X Chips ID 0x%02X !!\n", id[0],
 	   id[1]);
-    if (id[0] != 0x27 || id[1] != 0x00)
+    if (id[0] != 0x27 || (id[1] != 0x00 && id[1] != 0x08))
 	return -ENODEV;
 
     return 0;
