--- gnuradio-3.2.2/configure.ac	2010-01-20 09:54:08.000000000 +0000
+++ gnuradio-3.2.2/configure.ac	2010-01-20 09:55:35.000000000 +0000
@@ -121,7 +121,9 @@
 AC_ENABLE_SHARED	dnl do build shared libraries
 AC_DISABLE_STATIC	dnl don't build static libraries
 m4_ifdef([LT_INIT],[LT_INIT],[AC_PROG_LIBTOOL])
+AC_LANG_PUSH([Fortran])
 GR_FORTRAN
+AC_LANG_POP([Fortran])
 
 GR_NO_UNDEFINED		dnl do we need the -no-undefined linker flag
 GR_SCRIPTING
