--- install.py.orig	2009-05-03 10:39:41.000000000 +0200
+++ install.py	2009-05-25 22:32:05.164131243 +0200
@@ -204,7 +204,7 @@ def createGlobalPluginsDir():
     fname = os.path.join(pdir, "__init__.py")
     if not os.path.exists(fname):
         if not os.path.exists(pdir):
-            os.mkdir(pdir,  0755)
+            os.makedirs(pdir,  0755)
         f = open(fname, "wb")
         f.write(\
 '''# -*- coding: utf-8 -*-
