--- amsn-remote~	2007-04-26 09:21:48.000000000 +0300
+++ amsn-remote	2008-07-02 21:45:26.000000000 +0300
@@ -3,7 +3,7 @@
 ### Remote controller program
 ###
 
-source amsncore.tcl
+source /usr/share/amsn/amsncore.tcl
 
 set initialize_amsn 1
 
--- amsn-remote-CLI~	2007-04-26 09:21:48.000000000 +0300
+++ amsn-remote-CLI	2008-07-02 21:45:50.000000000 +0300
@@ -5,7 +5,7 @@
 
 set initialize_amsn 1
 
-source amsncore.tcl
+source /usr/share/amsn/amsncore.tcl
 
 proc readsocket { } {
 	global sock
