--- config.tcl~	2008-05-29 01:28:36.000000000 +0300
+++ config.tcl	2008-07-02 21:47:10.000000000 +0300
@@ -229,7 +229,7 @@
 		::config::setKey rightdelimiter \]		;#Right Timestamps' delimiter ']' by default
 		::config::setKey default_ns_server "messenger.hotmail.com:1863"
 		::config::setKey start_ns_server "messenger.hotmail.com:1863"
-		::config::setKey activeautoupdate 1		;#Active the auto update
+		::config::setKey activeautoupdate 0		;#Active the auto update
 		::config::setKey allowbadwords 1		;#Disable censure on nicks
 		::config::setKey enablebanner 1		;#Show or Hide AMSN Banner (By default Show)
 		::config::setKey startontray 0		;#Start amsn on tray icon only (hide contact list)
