diff -ur poppler-0.8.0.orig/qt4/demos/Makefile.am poppler-0.8.0/qt4/demos/Makefile.am
--- poppler-0.8.0.orig/qt4/demos/Makefile.am	2008-03-22 19:08:37.000000000 +0200
+++ poppler-0.8.0/qt4/demos/Makefile.am	2008-04-22 11:53:32.000000000 +0300
@@ -16,7 +16,7 @@
 
 # This rule lets GNU make create any *.moc from the equivalent *.h
 .h.moc:
-	moc -i $< -o $@
+	moc-qt4 -i $< -o $@
 
 noinst_PROGRAMS = poppler_qt4viewer
 
diff -ur poppler-0.8.0.orig/qt4/src/Makefile.am poppler-0.8.0/qt4/src/Makefile.am
--- poppler-0.8.0.orig/qt4/src/Makefile.am	2008-04-22 11:52:54.000000000 +0300
+++ poppler-0.8.0/qt4/src/Makefile.am	2008-04-22 11:53:18.000000000 +0300
@@ -63,7 +63,7 @@
 
 # This rule lets GNU make create any *.moc from the equivalent *.h
 .h.moc:
-	moc -i $< -o $@
+	moc-qt4 -i $< -o $@
 
 poppler-optcontent.lo: poppler-optcontent.moc
 
