diff -Nur magicor-1.0-rc1-old/magicor/editor/gtkgui.py magicor-1.0-rc1/magicor/editor/gtkgui.py
--- magicor-1.0-rc1-old/magicor/editor/gtkgui.py	2006-11-03 19:19:48.474070835 +0200
+++ magicor-1.0-rc1/magicor/editor/gtkgui.py	2006-11-03 19:20:20.773160627 +0200
@@ -394,7 +394,7 @@
     def __init__(self, config, loadfile = None):
         Editor.__init__(self, config, loadfile)
         self.glade = glade.XML("%s/editor/magicor-editor.glade"
-                               %config.get("path", "data"))
+                               %config.get("data_path", "data"))
         self.glade.signal_autoconnect(self)
         if loadfile:
             self.setTitle(os.path.basename(loadfile))
