--- makefile.unix	2006-05-15 19:53:34.000000000 +0300
+++ makefile.unix	2007-02-05 13:38:16.469365792 +0200
@@ -52,7 +52,7 @@
 
 # Uncomment to use DRC MIPS3 engine (faster emulation, but only works on x86-
 # compatible CPUs).
-# X86_MIPS3_DRC = 1
+X86_MIPS3_DRC = 1
 
 # Uncomment to use DRC PowerPC engine (faster emulation, but only works on x86-
 # compatible CPUs).
@@ -73,7 +73,7 @@
 # On Linux-i386 you can enable this without causing problems on non
 # MMX capable processors: the Linux code will autodetect if an MMX
 # capable processor is present and use the right code.
-# EFFECT_MMX_ASM = 1
+EFFECT_MMX_ASM = 1
 
 
 ###########################################################################
@@ -118,7 +118,7 @@
 
 # If you want to use whatever CFLAGS are currently set in your 
 # environment, then comment this out.
-CFLAGS =
+#CFLAGS =
 
 
 ###########################################################################
@@ -127,7 +127,7 @@
 ###########################################################################
 
 # GCC on x86
-CFLAGS = -O2
+#CFLAGS = -O2
 
 # GCC on x86 with some optimizations
 # CFLAGS = -O2 -mtune=i686 -fomit-frame-pointer -fstrength-reduce -ffast-math
@@ -181,13 +181,13 @@
   #########################################################################
   
   # Check for C89 + GNU extensions compliance.
-  CFLAGS += -std=gnu89
+#  CFLAGS += -std=gnu89
 
   # Show all warnings.
-  CFLAGS += -Wall
+#  CFLAGS += -Wall
 
   # Don't warn about unused variables.
-  CFLAGS += -Wno-unused
+#  CFLAGS += -Wno-unused
 
   # Warn about declarations after statements.
   # CFLAGS += -Wdeclaration-after-statement
@@ -500,7 +500,7 @@
 
 # The user and group to be used for installation.
 INSTALL_USER =         root
-INSTALL_GROUP =        bin
+INSTALL_GROUP =        root
 
 INSTALL_PROGRAM_DIR =	$(INSTALL) -d -o $(INSTALL_USER) -g $(INSTALL_GROUP) -m  755
 INSTALL_MAN_DIR =	$(INSTALL) -d -o $(INSTALL_USER) -g $(INSTALL_GROUP) -m  755
@@ -560,10 +560,10 @@
 # Select destination directory for your compiled program, manual page,
 # and binary distribution.
 ifndef PREFIX
-PREFIX = /usr/local
+PREFIX = /usr
 endif
 BINDIR = $(PREFIX)/bin
-MANDIR = $(PREFIX)/man/man6
+MANDIR = $(PREFIX)/share/man/man6
 
 # This is the default path for ROMs and other data files.
 XMAMEROOT = $(PREFIX)/share/x$(TARGET)
@@ -670,7 +670,7 @@
 
 # Enable the standard joystick driver, which should work in Linux, OpenBSD, 
 # NetBSD, and FreeBSD.
-# JOY_STANDARD = 1
+JOY_STANDARD = 1
 
 # Linux FM-TOWNS game pad joystick emulation support.  Thanks to Osamu 
 # Kurati.
@@ -694,7 +694,7 @@
 # events as lightguns. This can be tablets, touchpads or lightguns. You
 # must have the input layer and support for your devices (e.g. USB) for
 # this to work.
-# LIGHTGUN_ABS_EVENT = 1
+LIGHTGUN_ABS_EVENT = 1
 
 # Uncomment this if you get errors about a input_absinfo being undefined.
 # Some versions of the Linux headers define this, others don't, and it 
@@ -712,9 +712,9 @@
 # Choose any additonal sound drivers you want to include, besides the 
 # native sound driver for your system.
 # SOUND_ESOUND = 1
-# SOUND_ALSA = 1
-# SOUND_ARTS_TEIRA = 1
-# SOUND_ARTS_SMOTEK = 1
+SOUND_ALSA = 1
+SOUND_ARTS_TEIRA = 1
+SOUND_ARTS_SMOTEK = 1
 # SOUND_SDL = 1
 # SOUND_WAVEOUT = 1
 
@@ -765,7 +765,7 @@
 # X11_DGA = 1
 
 # Build support for OpenGL rendering into the X-Window version
-# X11_OPENGL = 1
+X11_OPENGL = 1
 
 # Build support for rendering through Glide into the X-Window version
 # X11_GLIDE = 1
@@ -782,8 +782,8 @@
 # X11LIB		= -L/usr/lib/X11
 
 # Standard location for XFree86
-X11INC		= -I/usr/X11R6/include
-X11LIB		= -L/usr/X11R6/lib
+X11INC		= -I/usr/include/X11
+X11LIB		= -L/usr/lib
 
 # Standard location for Sun systems
 # X11INC		= -I/usr/openwin/include
@@ -799,10 +799,10 @@
 
 # Uncomment this to use XInput devices, e.g. additional mice as 
 # trackballs.
-# XINPUT_DEVICES = 1
+XINPUT_DEVICES = 1
 
 # Enable Xinerama support
-# X11_XINERAMA = 1
+X11_XINERAMA = 1
 
 
 ##############################################################################
