diff -ur kvm-74.orig/kernel/Makefile kvm-74/kernel/Makefile
--- kvm-74.orig/kernel/Makefile	2008-08-27 11:53:37.000000000 +0300
+++ kvm-74/kernel/Makefile	2008-08-28 17:14:47.000000000 +0300
@@ -89,11 +89,6 @@
 install:
 	mkdir -p $(DESTDIR)/$(INSTALLDIR)
 	cp $(ARCH_DIR)/*.ko $(DESTDIR)/$(INSTALLDIR)
-	for i in $(ORIGMODDIR)/drivers/kvm/*.ko \
-		 $(ORIGMODDIR)/arch/$(ARCH_DIR)/kvm/*.ko; do \
-		if [ -f "$$i" ]; then mv "$$i" "$$i.orig"; fi; \
-	done
-	/sbin/depmod -a
 
 tmpspec = .tmp.kvm-kmod.spec
 
