--- scipy/fftpack/setup.py	2007-09-22 10:56:38.000000000 +0300
+++ scipy/fftpack/setup.py	2007-12-19 15:14:57.000000000 +0200
@@ -28,7 +28,7 @@
 
     config.add_extension('_fftpack',
         sources=sources,
-        libraries=['dfftpack'],
+        libraries=['dfftpack','python2.5'],
         extra_info=[fft_opt_info, djbfft_info],
         depends=['src/zfft_djbfft.c', 'src/zfft_fftpack.c', 'src/zfft_fftw.c',
             'src/zfft_fftw3.c', 'src/zfft_mkl.c', 
