--- scipy/linsolve/umfpack/umfpack.i	2007-09-22 10:55:21.000000000 +0300
+++ scipy/linsolve/umfpack/umfpack.i	2007-12-19 15:08:24.000000000 +0200
@@ -10,13 +10,13 @@
 */
 
 %{
-#include <umfpack.h>
+#include <suitesparse/umfpack.h>
 #include "numpy/arrayobject.h"
 %}
 
 %feature("autodoc", "1");
 
-#include <umfpack.h>
+#include <suitesparse/umfpack.h>
 
 %init %{
     import_array();
@@ -189,18 +189,18 @@
     double Info [ANY]
 };
 
-%include <umfpack.h>
-%include <umfpack_solve.h>
-%include <umfpack_defaults.h>
-%include <umfpack_triplet_to_col.h>
-%include <umfpack_col_to_triplet.h>
-%include <umfpack_transpose.h>
-%include <umfpack_scale.h>
-
-%include <umfpack_report_symbolic.h>
-%include <umfpack_report_numeric.h>
-%include <umfpack_report_info.h>
-%include <umfpack_report_control.h>
+%include <suitesparse/umfpack.h>
+%include <suitesparse/umfpack_solve.h>
+%include <suitesparse/umfpack_defaults.h>
+%include <suitesparse/umfpack_triplet_to_col.h>
+%include <suitesparse/umfpack_col_to_triplet.h>
+%include <suitesparse/umfpack_transpose.h>
+%include <suitesparse/umfpack_scale.h>
+
+%include <suitesparse/umfpack_report_symbolic.h>
+%include <suitesparse/umfpack_report_numeric.h>
+%include <suitesparse/umfpack_report_info.h>
+%include <suitesparse/umfpack_report_control.h>
 
 /*
   The order is important below!
@@ -212,8 +212,8 @@
     void **Numeric
 }
 
-%include <umfpack_symbolic.h>
-%include <umfpack_numeric.h>
+%include <suitesparse/umfpack_symbolic.h>
+%include <suitesparse/umfpack_numeric.h>
 
 
 OPAQUE_ARGINOUT( void * )
@@ -222,8 +222,8 @@
     void **Numeric
 }
 
-%include <umfpack_free_symbolic.h>
-%include <umfpack_free_numeric.h>
+%include <suitesparse/umfpack_free_symbolic.h>
+%include <suitesparse/umfpack_free_numeric.h>
 
 
 
@@ -245,7 +245,7 @@
     long *n_col,
     long *nz_udiag
 };
-%include <umfpack_get_lunz.h>
+%include <suitesparse/umfpack_get_lunz.h>
 
 
 ARRAY_IN( double, double, DOUBLE )
@@ -269,6 +269,6 @@
     int Q [ ]
 };
 %apply int  *OUTPUT { int *do_recip};
-%include <umfpack_get_numeric.h>
+%include <suitesparse/umfpack_get_numeric.h>
 
 #endif
