diff -Nur goonies_r1-0-1-old/src/TheGooniesApp.cpp goonies_r1-0-1/src/TheGooniesApp.cpp
--- goonies_r1-0-1-old/src/TheGooniesApp.cpp	2007-09-09 18:38:12.000000000 +0300
+++ goonies_r1-0-1/src/TheGooniesApp.cpp	2007-09-09 18:38:23.000000000 +0300
@@ -80,7 +80,7 @@
 
 	m_screen_dx=640;
 	m_screen_dy=480;
-	fullscreen=true;
+	fullscreen=false;
 	
 	m_GLTM=new GLTManager();
 	m_SFXM=new SFXManager();
