Index: groff-1.20.1/contrib/mom/Makefile.sub
===================================================================
--- groff-1.20.1.orig/contrib/mom/Makefile.sub
+++ groff-1.20.1/contrib/mom/Makefile.sub
@@ -18,7 +18,7 @@
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
 # These may be overridden if cross-compiling.
-GROFFBIN=$(top_builddir)/src/roff/groff/groff
+GROFFBIN=echo
 GROFF_BIN_PATH=`echo $(groff_bin_dirs) | sed -e 's|  *|$(SH_SEP)|g'`
 
 groff_bin_dirs=\
Index: groff-1.20.1/doc/Makefile.in
===================================================================
--- groff-1.20.1.orig/doc/Makefile.in
+++ groff-1.20.1/doc/Makefile.in
@@ -35,8 +35,8 @@ top_builddir=@abs_top_builddir@
 pnmtops=@pnmtops_nosetpage@
 
 # These may be overridden if cross-compiling.
-TROFFBIN=$(top_builddir)/src/roff/troff/troff
-GROFFBIN=$(top_builddir)/src/roff/groff/groff
+TROFFBIN=echo
+GROFFBIN=echo
 GROFF_BIN_PATH=`echo $(groff_bin_dirs) | sed -e 's|  *|:|g'`
 
 # info files are distributed with the groff package, so makeinfo is
Index: groff-1.20.1/doc/Makefile.sub
===================================================================
--- groff-1.20.1.orig/doc/Makefile.sub
+++ groff-1.20.1/doc/Makefile.sub
@@ -18,7 +18,7 @@
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
 # These may be overridden if cross-compiling.
-GROFFBIN=$(top_builddir)/src/roff/groff/groff
+GROFFBIN=echo
 GROFF_BIN_PATH=`echo $(groff_bin_dirs) | sed -e 's|  *|$(SH_SEP)|g'`
 
 groff_bin_dirs=\
Index: groff-1.20.1/contrib/hdtbl/Makefile.sub
===================================================================
--- groff-1.20.1.orig/contrib/hdtbl/Makefile.sub
+++ groff-1.20.1/contrib/hdtbl/Makefile.sub
@@ -17,7 +17,7 @@
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
 # These may be overridden if cross-compiling.
-GROFFBIN=$(top_builddir)/src/roff/groff/groff
+GROFFBIN=echo
 GROFF_BIN_PATH=`echo $(groff_bin_dirs) | sed -e 's|  *|$(SH_SEP)|g'`
 
 groff_bin_dirs=\
