Index: blueman-1.10/blueman/Constants.py.in
===================================================================
--- blueman-1.10.orig/blueman/Constants.py.in
+++ blueman-1.10/blueman/Constants.py.in
@@ -8,7 +8,7 @@ UI_PATH = "@pkgdatadir@/ui"
 OBEX_BROWSE_AVAILABLE = True
 DHCP_CONFIG_FILE = "@dhconfig@"
 
-DEF_BROWSE_COMMAND = "nautilus --browser obex://[%d]"
+DEF_BROWSE_COMMAND = "dolphin obex://[%d]"
 
 import os
 import gettext
