diff -Naur converseen_0.2.orig/res/converseen.desktop converseen_0.2/res/converseen.desktop
--- converseen_0.2.orig/res/converseen.desktop	2009-12-20 01:32:07.869558185 +0200
+++ converseen_0.2/res/converseen.desktop	2009-12-20 01:44:40.909407482 +0200
@@ -1,13 +1,14 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=Converseen
-GenericName=Converseen
+GenericName=Batch Image Converter
+GenericName[tr]=Toplu Grafik Dönüştürücü
 Name[en_GB]=Converseen
 Name[it]=Converseen
 Comment=Batch image converter
 Comment[it]=Convertitore batch di immagini
 Exec=converseen
-Icon=/usr/share/pixmaps/converseen.png
+Icon=converseen
 StartupNotify=true
 X-KDE-StartupNotify=true
 Terminal=false
