Index: procps-3.2.8/proc/module.mk
===================================================================
--- procps-3.2.8.orig/proc/module.mk
+++ procps-3.2.8/proc/module.mk
@@ -96,7 +96,7 @@ endif
 #################### install rules ###########################
 
 $(lib)$(SOFILE) : proc/$(SONAME)
-	$(install) --mode a=rx $< $@
+	$(install) -m 555 $< $@
 
 ifneq ($(SOLINK),$(SOFILE))
 .PHONY: $(lib)$(SOLINK)
@@ -115,7 +115,7 @@ ldconfig : $(lib)$(SONAME) $(lib)$(SOLIN
 	$(ldconfig)
 
 $(usr/lib)$(ANAME) : proc/$(ANAME)
-	$(install) --mode a=r $< $@
+	$(install) -m 444 $< $@
 
 # Junk anyway... supposed to go in /usr/include/$(NAME)
 #INSTALL += $(addprefix $(include),$(HDRFILES))
