--- ext/odbc/config.m4  2006-01-23 23:55:13.000000000 +0200
+++ ext.orig/odbc/config.m4     2006-01-23 23:56:08.000000000 +0200
@@ -440,9 +440,9 @@
   fi
   if test "$withval" != "no"; then
     PHP_ADD_LIBRARY_WITH_PATH(iodbc, $withval/lib)
-    PHP_ADD_INCLUDE($withval/include, 1)
+    PHP_ADD_INCLUDE($withval/include/iodbc, 1)
     ODBC_TYPE=iodbc
-    ODBC_INCLUDE=-I$withval/include
+    ODBC_INCLUDE=-I$withval/include/iodbc
     ODBC_LFLAGS=-L$withval/$PHP_LIBDIR
     ODBC_LIBS=-liodbc
     AC_DEFINE(HAVE_IODBC,1,[ ])
