diff -ur alsa-driver-1.0.13.orig/alsa-kernel/pci/hda/hda_intel.c alsa-driver-1.0.13/alsa-kernel/pci/hda/hda_intel.c
--- alsa-driver-1.0.13.orig/alsa-kernel/pci/hda/hda_intel.c	2006-09-29 14:40:36.000000000 +0300
+++ alsa-driver-1.0.13/alsa-kernel/pci/hda/hda_intel.c	2006-11-18 02:14:32.000000000 +0200
@@ -55,7 +55,7 @@
 static int position_fix;
 static int probe_mask = -1;
 static int single_cmd;
-static int disable_msi;
+static int disable_msi = 1;
 
