--- xen-3.0.0/tools/examples/xend-config.sxp.min	2005-12-06 15:05:29.000000000 +0100
+++ xen-3.0.0/tools/examples/xend-config.sxp	2005-12-06 20:33:21.000000000 +0100
@@ -105,7 +105,7 @@
 # Dom0 will balloon out when needed to free memory for domU.
 # dom0-min-mem is the lowest memory level (in MB) dom0 will get down to.
 # If dom0-min-mem=0, dom0 will never balloon out.
-(dom0-min-mem 196)
+(dom0-min-mem 256)
 
 # In SMP system, dom0 will use dom0-cpus # of CPUS
 # If dom0-cpus = 0, dom0 will take all cpus available
