diff -urNad cdrtools-2.01+01a03~/libscg/scsi-linux-sg.c cdrtools-2.01+01a03/libscg/scsi-linux-sg.c
--- cdrtools-2.01+01a03~/libscg/scsi-linux-sg.c	2005-05-16 23:48:37.000000000 +0200
+++ cdrtools-2.01+01a03/libscg/scsi-linux-sg.c	2006-08-17 12:01:37.000000000 +0200
@@ -1291,7 +1291,7 @@
 			scglocal(scgp)->isold = 1;
 			return (sg_rwsend(scgp));
 		}
-		if (sp->ux_errno == ENXIO || sp->ux_errno == EPERM ||
+		if (sp->ux_errno == ENXIO || /* sp->ux_errno == EPERM || */
 		    sp->ux_errno == EINVAL || sp->ux_errno == EACCES) {
 			return (-1);
 		}
