Index: boson-0.13/boson/info/boinfo_linux.cpp
===================================================================
--- boson-0.13.orig/code/boson/info/boinfo_linux.cpp	2005-07-26 21:53:07.000000000 +0000
+++ boson-0.13/code/boson/info/boinfo_linux.cpp	2007-06-29 12:19:41.000000000 +0000
@@ -72,8 +72,10 @@
 	insert(OSVersionString, kernelVersion);
  }
 
+#ifdef UTS_RELEASE
  kernelVersion = UTS_RELEASE;
  insert(BoInfo::OSVersionString + BoInfo::CompileOffset, kernelVersion);
+#endif
 
  QString kernelModules = readFile("/proc/modules");
  if (!kernelModules.isEmpty()) {
