diff --git a/libs/pixertool/CMakeLists.txt b/libs/pixertool/CMakeLists.txt
--- a/libs/pixertool/CMakeLists.txt
+++ b/libs/pixertool/CMakeLists.txt
@@ -54,6 +54,10 @@
 	endif(APPLE)
 endif (UNIX AND NOT WIN32)
 
+if (_SWSCALE_FOUND)
+	set(HAVE_SWSCALE 1)
+endif(_SWSCALE_FOUND)
+
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config.h)
 
 ow_create_project_binary()

