diff --git a/wifo/phapi/phcodec-avcodec-wrapper.c b/wifo/phapi/phcodec-avcodec-wrapper.c
--- a/wifo/phapi/phcodec-avcodec-wrapper.c
+++ b/wifo/phapi/phcodec-avcodec-wrapper.c
@@ -123,7 +123,9 @@
 	encoder_t->context->dct_algo = FF_DCT_AUTO;
 	encoder_t->context->idct_algo = FF_IDCT_AUTO;
 	encoder_t->context->opaque = opaque;
+#if LIBAVCODEC_VERSION_INT < ((52<<16)+(0<<8)+0)
 	encoder_t->context->rtp_mode = 1;
+#endif
 	encoder_t->context->rtp_payload_size = 1000;
 	encoder_t->context->rtp_callback = phcodec_avcodec_video_rtp_callback;
 
