diff -ur kdemultimedia-3.5.0.orig/kaudiocreator/encoderconfigimp.cpp kdemultimedia-3.5.0/kaudiocreator/encoderconfigimp.cpp
--- kdemultimedia-3.5.0.orig/kaudiocreator/encoderconfigimp.cpp	2005-09-10 11:20:05.000000000 +0300
+++ kdemultimedia-3.5.0/kaudiocreator/encoderconfigimp.cpp	2006-01-25 23:11:20.000000000 +0200
@@ -51,7 +51,7 @@
     encPrefs->writeConfig();
 
     encPrefs = EncoderPrefs::prefs("Encoder_1");
-    encPrefs->setEncoderName(i18n("Lame"));
+    encPrefs->setEncoderName(i18n("MP3 (Lame)"));
     encPrefs->setCommandLine("lame --preset standard --tt %{title} --ta %{artist} --tl %{albumtitle} --ty %{year} --tn %{number} --tg %{genre} %f %o");
     encPrefs->setExtension("mp3");
     encPrefs->setPercentLength(2);
diff -ur kdemultimedia-3.5.0.orig/kaudiocreator/encoder.cpp kdemultimedia-3.5.0/kaudiocreator/encoder.cpp
--- kdemultimedia-3.5.0.orig/kaudiocreator/encoder.cpp	2005-09-10 11:20:05.000000000 +0300
+++ kdemultimedia-3.5.0/kaudiocreator/encoder.cpp	2006-01-25 23:15:05.000000000 +0200
@@ -55,7 +55,9 @@
 	QString currentEncoderGroup = QString("Encoder_%1").arg(encoder);
 	prefs = EncoderPrefs::prefs(currentEncoderGroup);
 	if ( !EncoderPrefs::hasPrefs(currentEncoderGroup) ) {
+    /*
 		KMessageBox::sorry(0, i18n("No encoder has been selected.\nPlease select an encoder in the configuration."), i18n("No Encoder Selected"));
+    */
 		prefs->setCommandLine(QString::null);
 	}				
 }
