--- qcommon/files.c.orig	2007-08-21 20:17:06.000000000 +0300
+++ qcommon/files.c	2007-10-13 15:05:26.000000000 +0300
@@ -981,7 +981,7 @@
 	// basedir <path>
 	// allows the game to run from outside the data tree
 	//
-	fs_basedir = Cvar_Get ("basedir", ".", CVAR_NOSET);
+	fs_basedir = Cvar_Get ("basedir", "/usr/share/alienarena2007", CVAR_NOSET);
 
 	//
 	// start up with baseq2 by default
--- unix/sys_unix.c.orig	2007-10-13 18:51:00.000000000 +0300
+++ unix/sys_unix.c	2007-10-13 18:50:01.000000000 +0300
@@ -308,6 +308,7 @@
 	saved_euid = geteuid();
 	seteuid(getuid());
 
+	chdir("/usr/share/alienarena2007");
 	Qcommon_Init(argc, argv);
 
 	fcntl(0, F_SETFL, fcntl (0, F_GETFL, 0) | FNDELAY);
