--- qcommon/files.c.orig	2007-10-13 12:00:18.000000000 +0300
+++ qcommon/files.c	2007-10-13 12:02:09.000000000 +0300
@@ -591,7 +591,7 @@
 ================
 FS_AddHomeAsGameDirectory
 
-Use ~/.codered/dir as fs_gamedir
+Use ~/.alienarena2007/dir as fs_gamedir
 ================
 */
 void FS_AddHomeAsGameDirectory (char *dir)
@@ -600,7 +600,7 @@
 	char *homedir=getenv("HOME");
 	if(homedir)
 	{
-		int len = snprintf(gdir,sizeof(gdir),"%s/.codered/%s/", homedir, dir);
+		int len = snprintf(gdir,sizeof(gdir),"%s/.alienarena2007/%s/", homedir, dir);
 		Com_Printf("using %s for writing\n",gdir);
 		FS_CreatePath (gdir);
 
