From c171af1abc453776ebb0445d65c346f628716db6 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Fatih=20A=C5=9F=C4=B1c=C4=B1?= <fatih@pardus.org.tr>
Date: Fri, 6 Feb 2009 04:32:22 +0200
Subject: [PATCH 01/20] Disable FBC on 945GM

---
 src/i830_driver.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git xf86-video-intel-2.4.3/src/i830_driver.c xf86-video-intel-2.4.3/src/i830_driver.c
index 0a747c1..a25b532 100644
--- xf86-video-intel-2.4.3/src/i830_driver.c
+++ xf86-video-intel-2.4.3/src/i830_driver.c
@@ -2872,7 +2872,7 @@ I830ScreenInit(int scrnIndex, ScreenPtr pScreen, int argc, char **argv)
    }
 
    /* Enable FB compression if possible */
-   if (i830_fb_compression_supported(pI830))
+   if (i830_fb_compression_supported(pI830) && !IS_I945GM(pI830))
        pI830->fb_compression = TRUE;
    else
        pI830->fb_compression = FALSE;
-- 
1.6.2

