diff -Naur KDVDCreator.0.3.2/src/kdcconfig.kcfg KDVDCreator/src/kdcconfig.kcfg
--- KDVDCreator.0.3.2/src/kdcconfig.kcfg	2008-11-17 18:36:12.000000000 +0200
+++ KDVDCreator/src/kdcconfig.kcfg	2009-02-16 11:33:22.000000000 +0200
@@ -27,7 +27,7 @@
 	</group>
 	<group name="AudioVideo" >
 		<entry name="Driver" type="String">
-			<default>xv</default>
+			<default>gl</default>
 		</entry>
 	</group>
 	<group name="LookAndFeel" >
