diff -Nur hcfpcimodem-1.14full-old/modules/GPL/oscompat.h hcfpcimodem-1.14full/modules/GPL/oscompat.h
--- hcfpcimodem-1.14full-old/modules/GPL/oscompat.h	2007-12-05 18:31:49.585503830 +0200
+++ hcfpcimodem-1.14full/modules/GPL/oscompat.h	2007-12-05 18:32:39.074980286 +0200
@@ -70,7 +70,7 @@
 
 #include <linux/version.h>
 #ifdef FOUND_LINUX_CONFIG
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #endif
 #include <linux/types.h>
 
diff -Nur hcfpcimodem-1.14full-old/modules/GPL/serial_cnxt.c hcfpcimodem-1.14full/modules/GPL/serial_cnxt.c
--- hcfpcimodem-1.14full-old/modules/GPL/serial_cnxt.c	2007-12-05 18:31:49.584503982 +0200
+++ hcfpcimodem-1.14full/modules/GPL/serial_cnxt.c	2007-12-05 18:32:46.456858070 +0200
@@ -63,7 +63,7 @@
  */
 #include <linux/version.h>
 #ifdef FOUND_LINUX_CONFIG
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #endif
 #include <linux/module.h>
 #include <linux/errno.h>
diff -Nur hcfpcimodem-1.14full-old/modules/GPL/serial_core.c hcfpcimodem-1.14full/modules/GPL/serial_core.c
--- hcfpcimodem-1.14full-old/modules/GPL/serial_core.c	2007-12-05 18:31:49.585503830 +0200
+++ hcfpcimodem-1.14full/modules/GPL/serial_core.c	2007-12-05 18:32:54.064701502 +0200
@@ -29,7 +29,7 @@
  *
  */
 #include <linux/version.h>
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/module.h>
 #include <linux/errno.h>
 #include <linux/signal.h>
diff -Nur hcfpcimodem-1.14full-old/modules/GPL/serial_core.h hcfpcimodem-1.14full/modules/GPL/serial_core.h
--- hcfpcimodem-1.14full-old/modules/GPL/serial_core.h	2007-12-05 18:31:49.585503830 +0200
+++ hcfpcimodem-1.14full/modules/GPL/serial_core.h	2007-12-05 18:33:03.645245038 +0200
@@ -53,7 +53,7 @@
 
 #ifdef __KERNEL__
 
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/interrupt.h>
 #include <linux/serial.h>
 
