Index: gmult-5.3/gmult/DigitChooser.cpp
===================================================================
--- gmult-5.3.orig/gmult/DigitChooser.cpp
+++ gmult-5.3/gmult/DigitChooser.cpp
@@ -16,6 +16,8 @@
  *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
+#include <cstring>
+
 #include "DigitChooser.h"
 #include "config.h"
 #include "gettext.h"
Index: gmult-5.3/gmult/GtkMult.cpp
===================================================================
--- gmult-5.3.orig/gmult/GtkMult.cpp
+++ gmult-5.3/gmult/GtkMult.cpp
@@ -1,3 +1,5 @@
+#include <cstring>
+
 #include "GtkMult.h"
 #include "GtkMultTable.h"
 #include "config.h"
Index: gmult-5.3/gmult/GtkMultTable.cpp
===================================================================
--- gmult-5.3.orig/gmult/GtkMultTable.cpp
+++ gmult-5.3/gmult/GtkMultTable.cpp
@@ -16,6 +16,8 @@
  *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
+#include <cstring>
+
 #include "GtkMultTable.h"
 #include "MultPuzzle.h"
 #include "CharBox.h"
