Index: gwenhywfar-3.11.3/configure.ac
===================================================================
--- gwenhywfar-3.11.3.orig/configure.ac
+++ gwenhywfar-3.11.3/configure.ac
@@ -811,7 +811,7 @@ if test "$enable_visibility" = "yes"; th
   AC_MSG_CHECKING(if symbol visibility is supported)
   visibility_cflags=""
   case `$CC --version|head -n1|cut -f1 -d " "` in 
-    gcc*)
+    *gcc*)
        dnl Determine version number (watch out: the outside square
        dnl parenteses are vitally important here!)
        gccversion=["` ${CC} -dumpversion `"]
