Index: vlc-0.9.0-svn/libs/loader/wrapper.h
===================================================================
--- vlc-0.9.0-svn.orig/libs/loader/wrapper.h
+++ vlc-0.9.0-svn/libs/loader/wrapper.h
@@ -7,10 +7,6 @@ typedef struct {
   uint32_t edi, esi, ebp, esp, ebx, edx, ecx, eax;
 } reg386_t;
 
-typedef int (*wrapper_func_t)(void *stack_base, int stack_size, reg386_t *reg,  uint32_t *flags);
-
-extern wrapper_func_t report_entry, report_ret;
-
 extern void (*wrapper_target)(void);
 
 extern int wrapper(void);
