--- src/Makefile.in.orig	2006-08-28 02:04:51.000000000 +0300
+++ src/Makefile.in	2007-02-09 23:46:29.000000000 +0200
@@ -29,7 +29,7 @@
 CFLAGS	= @CFLAGS@ -I$(srcdir) -I$(builddir)
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS	= -L. @LDFLAGS@
-LIBS	= -lst @LIBS@
+LIBS	= -lst -lgsm @LIBS@
 
 # Other macros.
 
@@ -69,7 +69,7 @@
 SUNAUOBJ_1  = sunaudio.o
 ALSAOBJ_0   =
 ALSAOBJ_1   = alsa.o
-EXTRAOBJS   = $(OSSOBJ_$(NEED_OSS)) $(SUNAUOBJ_$(NEED_SUNAU)) $(ALSAOBJ_$(NEED_ALSA)) $(GSMOBJ_$(GSM_SUPPORT))
+EXTRAOBJS   = $(OSSOBJ_$(NEED_OSS)) $(SUNAUOBJ_$(NEED_SUNAU)) $(ALSAOBJ_$(NEED_ALSA))
 
 LIBOBJS = $(FOBJ) $(EOBJ) getopt.o getopt1.o handlers.o misc.o stio.o \
 	  util.o $(EXTRAOBJS)
@@ -80,7 +80,7 @@
 # target is just an easy way to make sure that makefile builds
 # the objects instead of this makefile.
 GSM_0     =
-GSM_1     = libgsm/libgsm.a
+GSM_1     = -lgsm
 GSM_ext   = 
 PLAY_0    =
 PLAY_1    = play
