Index: xf86-video-openchrome-0.2.903/src/via_driver.c
===================================================================
--- xf86-video-openchrome-0.2.903/src/via_driver.c.orig
+++ xf86-video-openchrome-0.2.903/src/via_driver.c
@@ -820,14 +820,12 @@ VIASetupDefaultOptions(ScrnInfoPtr pScrn
             pVia->DRIIrqEnable = FALSE;
             break;
         case VIA_K8M800:
-            pVia->agpEnable = FALSE;
             pVia->DRIIrqEnable = FALSE;
             break;
         case VIA_PM800:
             pVia->VideoEngine = VIDEO_ENGINE_CME;
             break;
         case VIA_VM800:
-            pVia->agpEnable = FALSE;
             break;
         case VIA_K8M890:
             pVia->VideoEngine = VIDEO_ENGINE_CME;
