diff -u basket/src/basket_config_apps.desktop src/basket_config_apps.desktop
--- basket/src/basket_config_apps.desktop	2007-01-29 12:23:53.000000000 +0200
+++ src/basket_config_apps.desktop	2007-02-27 00:38:43.000000000 +0200
@@ -23,6 +23,7 @@
 Name[nn]=Program
 Name[pt]=Aplicações
 Name[ru]=Приложения
+Name[tr]=Uygulamalar
 Comment=Applications
 Comment[de]=Anwendungen
 Comment[es]=Aplicaciones
@@ -33,3 +34,4 @@
 Comment[nn]=Program
 Comment[pt]=Aplicações
 Comment[ru]=Приложения
+Comment[tr]=Uygulamalar
diff -u basket/src/basket_config_baskets.desktop src/basket_config_baskets.desktop
--- basket/src/basket_config_baskets.desktop	2007-01-29 12:24:49.000000000 +0200
+++ src/basket_config_baskets.desktop	2007-02-27 00:39:33.000000000 +0200
@@ -23,6 +23,7 @@
 Name[nn]=Korger
 Name[pt]=Cestos
 Name[ru]=Корзины
+Name[tr]=Sepetler
 Comment=Baskets
 Comment[de]=Körbe
 Comment[es]=Cestas
@@ -33,3 +34,4 @@
 Comment[nn]=Korger
 Comment[pt]=Cestos
 Comment[ru]=Корзины
+Comment[tr]=Sepetler
\ Dosya sonunda yenisatır yok.
diff -u basket/src/basket_config_general.desktop src/basket_config_general.desktop
--- basket/src/basket_config_general.desktop	2007-01-29 12:25:46.000000000 +0200
+++ src/basket_config_general.desktop	2007-02-27 00:41:25.000000000 +0200
@@ -23,6 +23,7 @@
 Name[nn]=Generelt
 Name[pt]=Geral
 Name[ru]=Общие
+Name[tr]=Genel
 Comment=General
 Comment[de]=Allgemein
 Comment[es]=General
@@ -33,3 +34,4 @@
 Comment[nn]=Generelt
 Comment[pt]=Geral
 Comment[ru]=Общие
+Comment[de]=Genel
\ Dosya sonunda yenisatır yok.
diff -u basket/src/basket_config_new_notes.desktop src/basket_config_new_notes.desktop
--- basket/src/basket_config_new_notes.desktop	2007-01-29 12:26:29.000000000 +0200
+++ src/basket_config_new_notes.desktop	2007-02-27 00:40:44.000000000 +0200
@@ -23,6 +23,7 @@
 Name[nn]=Nye notat
 Name[pt]=Anotações novas
 Name[ru]=Новые заметки
+Name[tr]=Yeni Notlar
 Comment=New Notes
 Comment[de]=Neue Notizen
 Comment[es]=Notas nuevas
@@ -33,3 +34,4 @@
 Comment[nn]=Nye notat
 Comment[pt]=Anotações novas
 Comment[ru]=Новые заметки
+Comment[tr]=Yeni Notlar
\ Dosya sonunda yenisatır yok.
diff -u basket/src/basket_config_notes_appearance.desktop src/basket_config_notes_appearance.desktop
--- basket/src/basket_config_notes_appearance.desktop	2007-01-29 12:27:26.000000000 +0200
+++ src/basket_config_notes_appearance.desktop	2007-02-27 00:36:45.000000000 +0200
@@ -22,6 +22,7 @@
 Name[nb]=Notatutseende
 Name[nn]=Notatutsjånad
 Name[ru]=Вид
+Name[tr]=Not Görünümleri
 Comment=Notes Appearance
 Comment[de]=Erscheinungsbild der Notizen
 Comment[es]=Apariencia de las notas
@@ -31,3 +32,4 @@
 Comment[nb]=Notatutseende
 Comment[nn]=Notatutsjånad
 Comment[ru]=Вид
+Comment[tr]=Not Görünümleri
\ Dosya sonunda yenisatır yok.
diff -u basket/src/basket.desktop src/basket.desktop
--- basket/src/basket.desktop	2007-01-29 12:29:05.000000000 +0200
+++ src/basket.desktop	2007-02-27 00:37:43.000000000 +0200
@@ -17,6 +17,7 @@
 Name[pt]=Bloco de Notas BasKet
 Name[ru]=Альбом заметок BasKet
 Name[xx]=xxBasKet Note Padsxx
+Name[tr]=BasKet
 
 Comment=Taking care of your ideas.
 Comment[de]=Bewahren Sie Ihre Ideen.
@@ -29,3 +30,4 @@
 Comment[pt]=Cuidando das suas ideias.
 Comment[ru]=Заботимся о ваших идеях.
 Comment[xx]=xxTaking care of your ideas.xx
+Comment[tr]=Not Tutucu
\ Dosya sonunda yenisatır yok.
