diff -Nur slim-1.3.0.orig/Makefile slim-1.3.0/Makefile
--- slim-1.3.0.orig/Makefile	2008-01-28 08:42:27.000000000 +0200
+++ slim-1.3.0/Makefile	2008-01-28 09:00:28.000000000 +0200
@@ -5,7 +5,7 @@
 #######################################################
 CXX=/usr/bin/g++
 CC=/usr/bin/gcc
-CFLAGS=-Wall -I. -I/usr/X11R6/include -I/usr/include/freetype2 -I/usr/include/freetype2/config -I/usr/include/libpng12 -I/usr/include
+CFLAGS=-Wall -I. -I/usr/X11R6/include -I/usr/include/freetype2 -I/usr/include/freetype2/config -I/usr/include/libpng12 -I/usr/include $(ADDITIONAL_CFLAGS)
 CXXFLAGS=$(CFLAGS)
 LDFLAGS=-L/usr/X11R6/lib -lXft -lX11 -lfreetype -lXrender -lfontconfig -lpng12 -lz -lm -lcrypt -lXmu -lpng -ljpeg
 CUSTOM=-DHAVE_SHADOW
@@ -15,7 +15,7 @@
 endif
 PREFIX=/usr
 CFGDIR=/etc
-MANDIR=/usr/man
+MANDIR=/usr/share/man
 DESTDIR=
 #######################################################
 
