diff -BrauN samba-3.3.2.orig/source/nsswitch/pam_winbind.c samba-3.3.2/source/nsswitch/pam_winbind.c
--- samba-3.3.2.orig/source/nsswitch/pam_winbind.c	2009-03-12 10:47:20.000000000 +0100
+++ samba-3.3.2/source/nsswitch/pam_winbind.c	2009-04-08 15:04:02.000000000 +0200
@@ -914,7 +914,7 @@
 	/* now check for the global password policy */
 	/* good catch from Ralf Haferkamp: an expiry of "never" is translated
 	 * to -1 */
-	if (policy->expire <= 0) {
+	if (policy->expire == -1) {
 		return;
 	}
 
