diff -Naur minitube.orig/minitube.desktop minitube/minitube.desktop
--- minitube.orig/minitube.desktop	2009-11-17 00:35:45.133338469 +0200
+++ minitube/minitube.desktop	2009-11-17 00:37:54.847211774 +0200
@@ -2,7 +2,9 @@
 Name=Minitube
 Name[en_US]=Minitube
 Comment=YouTube client
+Comment[tr]=YouTube İstemcisi
 GenericName=YouTube client
+GenericName[tr]=YouTube İstemcisi
 GenericName[en_US]=YouTube client
 Exec=minitube
 Terminal=false
