diff -Nur x264-snapshot-20080402-2245-old/common/x86/cpu-32.asm x264-snapshot-20080402-2245/common/x86/cpu-32.asm
--- x264-snapshot-20080402-2245-old/common/x86/cpu-32.asm	2008-04-04 10:56:24.000000000 +0300
+++ x264-snapshot-20080402-2245/common/x86/cpu-32.asm	2008-04-04 10:58:13.000000000 +0300
@@ -88,3 +88,7 @@
     leave
     ret
 
+
+%ifidn __OUTPUT_FORMAT__,elf
+        section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nur x264-snapshot-20080402-2245-old/common/x86/dct-32.asm x264-snapshot-20080402-2245/common/x86/dct-32.asm
--- x264-snapshot-20080402-2245-old/common/x86/dct-32.asm	2008-04-04 10:56:24.000000000 +0300
+++ x264-snapshot-20080402-2245/common/x86/dct-32.asm	2008-04-04 11:00:30.000000000 +0300
@@ -526,3 +526,7 @@
 
 ADD_NxN_IDCT x264_add16x16_idct8_sse2, x264_add8x8_idct8_sse2, 128, 8
 
+
+%ifidn __OUTPUT_FORMAT__,elf
+        section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nur x264-snapshot-20080402-2245-old/common/x86/deblock-a.asm x264-snapshot-20080402-2245/common/x86/deblock-a.asm
--- x264-snapshot-20080402-2245-old/common/x86/deblock-a.asm	2008-04-04 10:56:24.000000000 +0300
+++ x264-snapshot-20080402-2245/common/x86/deblock-a.asm	2008-04-04 11:00:16.000000000 +0300
@@ -590,3 +590,7 @@
     TRANSPOSE8x4_STORE PASS8ROWS(t5, r0, r1, t6)
     RET
 
+
+%ifidn __OUTPUT_FORMAT__,elf
+        section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nur x264-snapshot-20080402-2245-old/common/x86/mc-a2.asm x264-snapshot-20080402-2245/common/x86/mc-a2.asm
--- x264-snapshot-20080402-2245-old/common/x86/mc-a2.asm	2008-04-04 10:56:24.000000000 +0300
+++ x264-snapshot-20080402-2245/common/x86/mc-a2.asm	2008-04-04 11:01:26.000000000 +0300
@@ -389,3 +389,7 @@
         movdqa [r0 + r2 + 48], xmm3
     jg .copy64
     REP_RET
+
+%ifidn __OUTPUT_FORMAT__,elf
+        section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nur x264-snapshot-20080402-2245-old/common/x86/mc-a.asm x264-snapshot-20080402-2245/common/x86/mc-a.asm
--- x264-snapshot-20080402-2245-old/common/x86/mc-a.asm	2008-04-04 10:56:24.000000000 +0300
+++ x264-snapshot-20080402-2245/common/x86/mc-a.asm	2008-04-04 10:59:56.000000000 +0300
@@ -701,3 +701,7 @@
     jnz .height_loop1_w8
     REP_RET
 
+
+%ifidn __OUTPUT_FORMAT__,elf
+        section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nur x264-snapshot-20080402-2245-old/common/x86/pixel-32.asm x264-snapshot-20080402-2245/common/x86/pixel-32.asm
--- x264-snapshot-20080402-2245-old/common/x86/pixel-32.asm	2008-04-04 10:56:24.000000000 +0300
+++ x264-snapshot-20080402-2245/common/x86/pixel-32.asm	2008-04-04 11:00:47.000000000 +0300
@@ -458,3 +458,7 @@
     emms
     ret
 
+
+%ifidn __OUTPUT_FORMAT__,elf
+        section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nur x264-snapshot-20080402-2245-old/common/x86/pixel-a.asm x264-snapshot-20080402-2245/common/x86/pixel-a.asm
--- x264-snapshot-20080402-2245-old/common/x86/pixel-a.asm	2008-04-04 10:56:24.000000000 +0300
+++ x264-snapshot-20080402-2245/common/x86/pixel-a.asm	2008-04-04 11:01:06.000000000 +0300
@@ -1756,3 +1756,7 @@
     RET
 %endif ; ARCH
 
+
+%ifidn __OUTPUT_FORMAT__,elf
+        section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nur x264-snapshot-20080402-2245-old/common/x86/predict-a.asm x264-snapshot-20080402-2245/common/x86/predict-a.asm
--- x264-snapshot-20080402-2245-old/common/x86/predict-a.asm	2008-04-04 10:56:24.000000000 +0300
+++ x264-snapshot-20080402-2245/common/x86/predict-a.asm	2008-04-04 10:58:54.000000000 +0300
@@ -597,3 +597,7 @@
     PRED16x16_DC_SSE2 [pw_8 GLOBAL], 4
     REP_RET
 
+
+%ifidn __OUTPUT_FORMAT__,elf
+        section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nur x264-snapshot-20080402-2245-old/common/x86/quant-a.asm x264-snapshot-20080402-2245/common/x86/quant-a.asm
--- x264-snapshot-20080402-2245-old/common/x86/quant-a.asm	2008-04-04 10:56:24.000000000 +0300
+++ x264-snapshot-20080402-2245/common/x86/quant-a.asm	2008-04-04 10:58:39.000000000 +0300
@@ -325,3 +325,7 @@
 DEQUANT sse2, 4, 4, 2
 DEQUANT sse2, 8, 6, 2
 
+
+%ifidn __OUTPUT_FORMAT__,elf
+        section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nur x264-snapshot-20080402-2245-old/common/x86/x86inc-32.asm x264-snapshot-20080402-2245/common/x86/x86inc-32.asm
--- x264-snapshot-20080402-2245-old/common/x86/x86inc-32.asm	2008-04-04 10:56:24.000000000 +0300
+++ x264-snapshot-20080402-2245/common/x86/x86inc-32.asm	2008-04-04 11:01:45.000000000 +0300
@@ -116,3 +116,7 @@
     %define picesp esp
 %endif
 
+
+%ifidn __OUTPUT_FORMAT__,elf
+        section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
