--- klamav-0.42-source/klamav-0.42/src/klammail/defaults.h.org      2008-01-28 02:05:26.000000000 +0200
+++ klamav-0.42-source/klamav-0.42/src/klammail/defaults.h  2008-01-28 02:08:49.000000000 +0200
@@ -15,7 +15,7 @@
  *  along with this program; if not, write to the Free Software
  *  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
-#define CONFDIR "/usr/local/etc"
+#define CONFDIR "/etc"
 #define DEFAULT_CFG CONFDIR"/clamav.conf"
 #ifdef CLAMAVUSER
 #define UNPUSER CLAMAVUSER
--- klamav-0.42-source/klamav-0.42/src/klammail/clamdmail.c.org     2008-01-28 02:03:12.000000000 +0200
+++ klamav-0.42-source/klamav-0.42/src/klammail/clamdmail.c 2008-01-28 02:05:15.000000000 +0200
@@ -185,7 +185,7 @@
                                     return 50;
                                 }
                         }else{
-                                if((ret = cl_loaddbdir(cl_retdbdir(), &engine, &no, CL_DB_STDOPT))) {
+                                if((ret = cl_loaddbdir(cl_retdbdir(), &engine, &no))) {
                                 printf("cl_loaddbdir: %s\n", cl_strerror(ret));
                                 close(fd);
                                 exit(2);
@@ -541,7 +541,7 @@
                         break;
                         }
         }else{
-                fprintf(tmp,"DatabaseDirectory /usr/local/share/clamav\n");
+                fprintf(tmp,"DatabaseDirectory /var/lib/clamav\n");
         }
         //fprintf(tmp,"ScanMail\n");
         if ((strstr(cl_retver(), "0.8")) || (strstr(cl_retver(), "0.7")))

