Index: kdewebdev/kxsldbg/Makefile.am
===================================================================
--- kdewebdev.orig/kxsldbg/Makefile.am
+++ kdewebdev/kxsldbg/Makefile.am
@@ -21,7 +21,7 @@ xsldbg_DEPENDENCIES  = kxsldbgpart/libxs
 xsldbg_LDADD = kxsldbgpart/libxsldbg/libxsldbg.la $(LIBXSLT_LIBS) $(LIBXML_LIBS) $(LIB_KDEUI)
 #
 # the library search path.
-xsldbg_LDFLAGS = $(all_libraries) $(XSLDBG_EXTRA_LIBS)
+xsldbg_LDFLAGS = /usr/kde/3.5/lib/libkdeui.so $(all_libraries) $(XSLDBG_EXTRA_LIBS)
 
 # These paths are KDE specific. Use them:
 # kde_appsdir         Where your application's menu entry (.desktop) should go to.
@@ -50,7 +50,7 @@ INCLUDES= $(LIBXML_CFLAGS)  $(LIBXSLT_CF
 METASOURCES = AUTO
 
 # the application source, library search path, and link libraries
-kxsldbg_LDFLAGS = $(KDE_RPATH) $(all_libraries)
+kxsldbg_LDFLAGS = /usr/kde/3.5/lib/libkdeui.so $(KDE_RPATH) $(all_libraries)
 
 # this is where the shell's XML-GUI resource file goes
 shellrcdir   = $(kde_datadir)/kxsldbg
