Index: kdewebdev/quanta/src/Makefile.am
===================================================================
--- kdewebdev.orig/quanta/src/Makefile.am
+++ kdewebdev/quanta/src/Makefile.am
@@ -64,7 +64,7 @@ AM_CPPFLAGS = -I$(top_srcdir)/quanta/dia
 quanta_METASOURCES = AUTO
 
 # the library search path.
-quanta_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+quanta_LDFLAGS = /usr/kde/3.5/lib/libkatepartinterfaces.so $(all_libraries) $(KDE_RPATH)
 
 # this is where the kdelnk file will go
 xdg_apps_DATA = quanta.desktop
