diff -Naur cvblob/cvBlob/CMakeLists.txt cvblob-patched//cvBlob/CMakeLists.txt
--- cvblob/cvBlob/CMakeLists.txt	2011-01-04 14:10:43.000000000 +0200
+++ cvblob-patched//cvBlob/CMakeLists.txt	2012-01-17 21:40:47.923497309 +0200
@@ -33,6 +33,10 @@
 )
 
 add_library(cvblob SHARED ${cvBlob_CVBLOB})
+target_link_libraries(cvblob ${OpenCV_LIBS})
+
+set_target_properties(cvblob PROPERTIES VERSION "0.10.3" SOVERSION
+    "0")
 
 install(FILES cvblob.h DESTINATION include)
 
