diff -ur mozilla/security/coreconf/headers.mk nss-fixed/mozilla/security/coreconf/headers.mk
--- mozilla/security/coreconf/headers.mk	2004-04-25 18:02:17.000000000 +0300
+++ mozilla/security/coreconf/headers.mk	2005-12-31 23:48:13.000000000 +0200
@@ -44,7 +44,7 @@
 #  (xp) include paths
 #
 
-INCLUDES  += -I$(SOURCE_MDHEADERS_DIR) 
+INCLUDES  += -I$(SOURCE_MDHEADERS_DIR) -I/usr/include/nspr -I$(DIST)/include/dbm
 
 ifneq ($(OS_TARGET),WIN16)
     INCLUDES  += -I$(SOURCE_XPHEADERS_DIR)
