Index: subversion/libsvn_subr/opt.c
===================================================================
--- subversion/libsvn_subr/opt.c	(revision 28645)
+++ subversion/libsvn_subr/opt.c	(working copy)
@@ -565,7 +565,7 @@
     {
       revision->kind = svn_opt_revision_base;
     }
-  else if (strcasecmp(word, "committed") == 0)
+  else if ( strcasecmp(word, "committed") == 0 || strcasecmp(word, "COMMITTED") == 0)
     {
       revision->kind = svn_opt_revision_committed;
     }
