--- kerry/src/kerryapp.cpp	2007-01-15 16:20:25.000000000 +0200
+++ kerry/src/kerryapp.cpp.new	2007-03-04 23:19:32.000000000 +0200
@@ -122,8 +122,7 @@
   globalKeys = new KGlobalAccel(this);
   globalKeys->insert( "Program:kerry", i18n("Kerry Beagle Search") );
 
-  KShortcut showDialogShortcut = KShortcut(ALT+Key_Space);
-  showDialogShortcut.append( KKey( Key_F12 ) );
+  KShortcut showDialogShortcut = KShortcut(Key_F9);
   globalKeys->insert( "Show Kerry Dialog", i18n("Show Search Dialog"), QString::null, showDialogShortcut, showDialogShortcut, hitListWindow, SLOT(showSearchDialog()) );
   globalKeys->insert( "Search Primary Selection with Kerry", i18n("Search Primary Selection"), QString::null, CTRL+ALT+Key_Space, CTRL+ALT+Key_Space, this, SLOT(searchPrimarySelection()) );
   configChanged();
