diff -ur nasm-0.98.39.orig/output/outieee.c nasm-0.98.39/output/outieee.c
--- nasm-0.98.39.orig/output/outieee.c	2005-01-16 00:16:08.000000000 +0200
+++ nasm-0.98.39/output/outieee.c	2006-03-30 14:26:57.000000000 +0300
@@ -1120,7 +1120,7 @@
     va_list ap;
 
     va_start(ap, format);
-    vsprintf(buffer, format, ap);
+    vsnprintf(buffer, sizeof(buffer), format, ap); 
     l = strlen(buffer);
     for (i = 0; i < l; i++)
         if ((buffer[i] & 0xff) > 31)
