diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/browser/components/migration/src/nsDogbertProfileMigrator.cpp mozilla/browser/components/migration/src/nsDogbertProfileMigrator.cpp
--- mozilla-orig/browser/components/migration/src/nsDogbertProfileMigrator.cpp	2008-07-30 20:25:14.000000000 +0300
+++ mozilla/browser/components/migration/src/nsDogbertProfileMigrator.cpp	2008-07-30 20:28:48.000000000 +0300
@@ -65,7 +65,8 @@
 #elif defined(CCHMAXPATH)
 #define MAXPATHLEN CCHMAXPATH
 #else
-#define MAXPATHLEN 1024
+#  include <limits.h>
+#  define MAXPATHLEN PATH_MAX
 #endif
 #endif
 
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/browser/components/migration/src/nsProfileMigrator.cpp mozilla/browser/components/migration/src/nsProfileMigrator.cpp
--- mozilla-orig/browser/components/migration/src/nsProfileMigrator.cpp	2008-07-30 20:25:14.000000000 +0300
+++ mozilla/browser/components/migration/src/nsProfileMigrator.cpp	2008-07-30 20:28:02.000000000 +0300
@@ -73,7 +73,8 @@
 #elif defined(CCHMAXPATH)
 #define MAXPATHLEN CCHMAXPATH
 #else
-#define MAXPATHLEN 1024
+#include <limits.h>
+#define MAXPATHLEN PATH_MAX
 #endif
 #endif
 
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/config/pathsub.h mozilla/config/pathsub.h
--- mozilla-orig/config/pathsub.h	2008-07-30 20:25:16.000000000 +0300
+++ mozilla/config/pathsub.h	2008-07-30 20:50:49.000000000 +0300
@@ -46,7 +46,7 @@
 #include <sys/types.h>
 
 #ifndef PATH_MAX
-#define PATH_MAX 1024
+#error  "PATH_MAX is not defined!"
 #endif
 
 /*
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/dbm/include/mcom_db.h mozilla/dbm/include/mcom_db.h
--- mozilla-orig/dbm/include/mcom_db.h	2008-07-30 20:25:49.000000000 +0300
+++ mozilla/dbm/include/mcom_db.h	2008-07-30 20:47:05.000000000 +0300
@@ -175,7 +175,8 @@
 #endif  /* __DBINTERFACE_PRIVATE */
 
 #ifdef SCO
-#define MAXPATHLEN 	1024              
+#include <limits.h>
+#define MAXPATHLEN PATH_MAX
 #endif
 
 #include <fcntl.h>
@@ -185,7 +186,8 @@
 #include <io.h>
 
 #ifndef XP_OS2 
-#define MAXPATHLEN 	1024               
+#include <limits.h>
+#define MAXPATHLEN PATH_MAX
 #endif
 
 #ifdef XP_OS2_VACPP
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/js/src/jsfile.c mozilla/js/src/jsfile.c
--- mozilla-orig/js/src/jsfile.c	2008-07-30 20:25:42.000000000 +0300
+++ mozilla/js/src/jsfile.c	2008-07-30 20:49:04.000000000 +0300
@@ -105,7 +105,8 @@
 #define utfstring               "binary"
 #define unicodestring           "unicode"
 
-#define MAX_PATH_LENGTH         1024
+#include <limits.h>
+#define MAX_PATH_LENGTH         PATH_MAX
 #define MODE_SIZE               256
 #define NUMBER_SIZE             32
 #define MAX_LINE_LENGTH         256
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/mail/components/migration/src/nsDogbertProfileMigrator.cpp mozilla/mail/components/migration/src/nsDogbertProfileMigrator.cpp
--- mozilla-orig/mail/components/migration/src/nsDogbertProfileMigrator.cpp	2008-07-30 20:25:57.000000000 +0300
+++ mozilla/mail/components/migration/src/nsDogbertProfileMigrator.cpp	2008-07-30 20:37:32.000000000 +0300
@@ -71,7 +71,8 @@
 #elif defined(CCHMAXPATH)
 #define MAXPATHLEN CCHMAXPATH
 #else
-#define MAXPATHLEN 1024
+#include <limits.h>
+#define MAXPATHLEN PATH_MAX
 #endif
 #endif
 
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/mail/components/migration/src/nsProfileMigrator.cpp mozilla/mail/components/migration/src/nsProfileMigrator.cpp
--- mozilla-orig/mail/components/migration/src/nsProfileMigrator.cpp	2008-07-30 20:25:57.000000000 +0300
+++ mozilla/mail/components/migration/src/nsProfileMigrator.cpp	2008-07-30 20:35:45.000000000 +0300
@@ -64,7 +64,8 @@
 #elif defined(CCHMAXPATH)
 #define MAXPATHLEN CCHMAXPATH
 #else
-#define MAXPATHLEN 1024
+#include <limits.h>
+#define MAXPATHLEN PATH_MAX
 #endif
 #endif
 
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/mailnews/import/comm4x/src/nsComm4xProfile.cpp mozilla/mailnews/import/comm4x/src/nsComm4xProfile.cpp
--- mozilla-orig/mailnews/import/comm4x/src/nsComm4xProfile.cpp	2008-07-30 20:25:30.000000000 +0300
+++ mozilla/mailnews/import/comm4x/src/nsComm4xProfile.cpp	2008-07-30 20:37:53.000000000 +0300
@@ -70,7 +70,8 @@
 #elif defined(CCHMAXPATH)
 #define MAXPATHLEN CCHMAXPATH
 #else
-#define MAXPATHLEN 1024
+#include <limits.h>
+#define MAXPATHLEN PATH_MAX
 #endif
 #endif
 
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/modules/libjar/nsZipArchive.cpp mozilla/modules/libjar/nsZipArchive.cpp
--- mozilla-orig/modules/libjar/nsZipArchive.cpp	2008-07-30 20:25:53.000000000 +0300
+++ mozilla/modules/libjar/nsZipArchive.cpp	2008-07-30 20:51:37.000000000 +0300
@@ -124,7 +124,7 @@
 #    define S_IFLNK  0120000
 #  endif
 #  ifndef PATH_MAX
-#    define PATH_MAX 1024
+#    include <limits.h>
 #  endif
 #endif  /* XP_UNIX */
 
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/nsprpub/config/pathsub.h mozilla/nsprpub/config/pathsub.h
--- mozilla-orig/nsprpub/config/pathsub.h	2008-07-30 20:25:42.000000000 +0300
+++ mozilla/nsprpub/config/pathsub.h	2008-07-30 20:52:35.000000000 +0300
@@ -50,7 +50,7 @@
 #endif
 
 #ifndef PATH_MAX
-#define PATH_MAX 1024
+#error  "PATH_MAX is not defined!"
 #endif
 
 /*
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/security/coreconf/nsinstall/pathsub.h mozilla/security/coreconf/nsinstall/pathsub.h
--- mozilla-orig/security/coreconf/nsinstall/pathsub.h	2008-07-30 20:25:14.000000000 +0300
+++ mozilla/security/coreconf/nsinstall/pathsub.h	2008-07-30 20:52:56.000000000 +0300
@@ -49,7 +49,7 @@
 #endif
 
 #ifndef PATH_MAX
-#define PATH_MAX 1024
+#error  "PATH_MAX is not defined!"
 #endif
 
 /*
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/suite/profile/migration/src/nsProfileMigrator.cpp mozilla/suite/profile/migration/src/nsProfileMigrator.cpp
--- mozilla-orig/suite/profile/migration/src/nsProfileMigrator.cpp	2008-07-30 20:25:56.000000000 +0300
+++ mozilla/suite/profile/migration/src/nsProfileMigrator.cpp	2008-07-30 20:38:15.000000000 +0300
@@ -64,7 +64,8 @@
 #elif defined(CCHMAXPATH)
 #define MAXPATHLEN CCHMAXPATH
 #else
-#define MAXPATHLEN 1024
+#include <limits.h>
+#define MAXPATHLEN PATH_MAX
 #endif
 #endif
 
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/toolkit/mozapps/update/src/updater/updater.cpp mozilla/toolkit/mozapps/update/src/updater/updater.cpp
--- mozilla-orig/toolkit/mozapps/update/src/updater/updater.cpp	2008-07-30 20:25:39.000000000 +0300
+++ mozilla/toolkit/mozapps/update/src/updater/updater.cpp	2008-07-30 20:40:58.000000000 +0300
@@ -124,7 +124,8 @@
 # elif defined(CCHMAXPATH)
 #  define MAXPATHLEN CCHMAXPATH
 # else
-#  define MAXPATHLEN 1024
+#include <limits.h>
+#define MAXPATHLEN PATH_MAX
 # endif
 #endif
 
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/toolkit/xre/nsAppRunner.h mozilla/toolkit/xre/nsAppRunner.h
--- mozilla-orig/toolkit/xre/nsAppRunner.h	2008-07-30 20:25:38.000000000 +0300
+++ mozilla/toolkit/xre/nsAppRunner.h	2008-07-30 20:38:44.000000000 +0300
@@ -48,7 +48,8 @@
 #elif defined(CCHMAXPATH)
 #define MAXPATHLEN CCHMAXPATH
 #else
-#define MAXPATHLEN 1024
+#include <limits.h>
+#define MAXPATHLEN PATH_MAX
 #endif
 #endif
 
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/xpcom/build/nsXPCOMPrivate.h mozilla/xpcom/build/nsXPCOMPrivate.h
--- mozilla-orig/xpcom/build/nsXPCOMPrivate.h	2008-07-30 20:25:48.000000000 +0300
+++ mozilla/xpcom/build/nsXPCOMPrivate.h	2008-07-30 20:41:38.000000000 +0300
@@ -272,7 +272,8 @@
 #elif defined(CCHMAXPATH)
 #define MAXPATHLEN CCHMAXPATH
 #else
-#define MAXPATHLEN 1024
+#include <limits.h>
+#define MAXPATHLEN PATH_MAX
 #endif
 #endif
 
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/xpcom/io/SpecialSystemDirectory.cpp mozilla/xpcom/io/SpecialSystemDirectory.cpp
--- mozilla-orig/xpcom/io/SpecialSystemDirectory.cpp	2008-07-30 20:25:48.000000000 +0300
+++ mozilla/xpcom/io/SpecialSystemDirectory.cpp	2008-07-30 20:41:57.000000000 +0300
@@ -107,7 +107,8 @@
 #elif defined(CCHMAXPATH)
 #define MAXPATHLEN CCHMAXPATH
 #else
-#define MAXPATHLEN 1024
+#include <limits.h>
+#define MAXPATHLEN PATH_MAX
 #endif
 #endif
 
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/xpcom/obsolete/nsFileSpecUnix.cpp mozilla/xpcom/obsolete/nsFileSpecUnix.cpp
--- mozilla-orig/xpcom/obsolete/nsFileSpecUnix.cpp	2008-07-30 20:25:48.000000000 +0300
+++ mozilla/xpcom/obsolete/nsFileSpecUnix.cpp	2008-07-30 20:47:33.000000000 +0300
@@ -80,7 +80,8 @@
 #endif
 
 #ifndef MAXPATHLEN
-#define MAXPATHLEN	1024  /* Guessing this is okay.  Works for SCO. */
+#include <limits.h>
+#define MAXPATHLEN PATH_MAX
 #endif
  
 #if defined(__QNX__)
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur mozilla-orig/xpcom/typelib/xpidl/xpidl_java.c mozilla/xpcom/typelib/xpidl/xpidl_java.c
--- mozilla-orig/xpcom/typelib/xpidl/xpidl_java.c	2008-07-30 20:25:48.000000000 +0300
+++ mozilla/xpcom/typelib/xpidl/xpidl_java.c	2008-07-30 20:49:37.000000000 +0300
@@ -44,6 +44,7 @@
 #include "xpidl.h"
 #include <ctype.h>
 #include <glib.h>
+#include <limits.h>
 
 #ifdef XP_WIN
 #include <windef.h>
