--- configure.orig	2008-07-30 18:47:31.000000000 +0300
+++ configure	2008-07-30 18:47:51.000000000 +0300
@@ -8227,7 +8227,7 @@
                       if ${CC-cc} -Werror -S conftest.c -o conftest.s >/dev/null 2>&1; then
                         if egrep '\.(hidden|private_extern).*foo_hidden' conftest.s >/dev/null; then
                           if ! egrep '\.(hidden|private_extern).*foo_default' conftest.s > /dev/null; then
-                            ac_cv_visibility_pragma=yes
+                            ac_cv_visibility_pragma=no
                           fi
                         fi
                       fi
