diff -Naur audiokonverter-5.8.2.orig/audioconvert4.desktop audiokonverter-5.8.2/audioconvert4.desktop
--- audiokonverter-5.8.2.orig/audioconvert4.desktop	2009-08-08 01:01:00.038337600 +0300
+++ audiokonverter-5.8.2/audioconvert4.desktop	2009-08-08 01:04:43.273338319 +0300
@@ -56,17 +56,17 @@
 
 [Desktop Action mp3]
 Name=MP3
-Icon=audio-basic
+Icon=audio-mpeg.png
 Exec=konsole -e audioconvert4 -t m %F
 
 [Desktop Action ogg]
 Name=OGG
-Icon=audio-basic
+Icon=audio-x-vorbis+ogg.png
 Exec=konsole -e audioconvert4 -t o %F
 
 [Desktop Action flac]
 Name=FLAC
-Icon=audio-basic
+Icon=audio-x-flac.png
 Exec=konsole -e audioconvert4 -t f %F
 
 #[Desktop Action aac]
@@ -76,15 +76,16 @@
 
 [Desktop Action ipod]
 Name=M4A for iPod (AAC)
+Name[tr]=iPod İçin M4A (AAC)
 Name[de]=M4A für iPod (AAC)
 Name[nl]=M4A voor iPod (AAC)
 Name[fr]=M4A pour iPod (AAC)
 Name[it]=M4A per iPod (AAC)
 Name[es]=M4A para iPod (AAC)
-Icon=audio-basic
+Icon=audio-aac.png
 Exec=konsole -e audioconvert4 -t i %F
 
 [Desktop Action wav]
 Name=WAV
-Icon=audio-basic
+Icon=audio-x-wav.png
 Exec=konsole -e anytowav4 %F
diff -Naur audiokonverter-5.8.2.orig/audiofrommovie4.desktop audiokonverter-5.8.2/audiofrommovie4.desktop
--- audiokonverter-5.8.2.orig/audiofrommovie4.desktop	2009-08-08 01:01:00.038337600 +0300
+++ audiokonverter-5.8.2/audiofrommovie4.desktop	2009-08-08 01:11:16.040337338 +0300
@@ -3,37 +3,40 @@
 ServiceTypes=KonqPopupMenu/Plugin
 MimeType=video/avi;video/mp4;video/mpeg;video/msvideo;video/ogg;video/quicktime;video/theora;video/vnd.rn-realvideo;application/vnd.rn-realvideo;video/x-matroska;video/x-mpeg;video/x-mpeg2;video/x-ms-asf;video/x-msvideo;video/x-ms-wmv;video/x-ogg;video/x-ogm;video/x-quicktime;video/x-theora;video/x-avi;video/x-flv
 Actions=show;dump;mp3;ogg;flac;ipod;wav
-X-KDE-Submenu=extract audio to
+X-KDE-Submenu=Extract Audio To
+X-KDE-Submenu[tr]=Sesi Şuraya Çıkar
 X-KDE-Submenu[de]=Tonspur umwandeln in
 X-KDE-Submenu[fr]=Extraire la piste audio au format...
 X-KDE-Submenu[es]=Extraer pista de audio en...
 Encoding=UTF8
 
 [Desktop Action show]
-Name=show properties only
+Name=Show Properties Only
+Name[tr]=Sadece Özellikleri Göster
 Name[de]=zeige nur Eigenschaften
-Icon=sound
+Icon=audio-mpeg.png
 Exec=movie2sound4 -t s %F
 
 [Desktop Action dump]
-Name=an audio file, copying only
+Name=An Audio File, Copying Only
+Name[tr]=Ses Dosyası, Sadece Kopyala
 Name[de]=eine Audio-Datei, als Kopie
-Icon=sound
+Icon=audio-mpeg.png
 Exec=konsole -e movie2sound4 -t d %F
 
 [Desktop Action mp3]
 Name=MP3
-Icon=sound
+Icon=audio-mpeg.png
 Exec=konsole -e movie2sound4 -t m %F
 
 [Desktop Action ogg]
 Name=OGG
-Icon=sound
+Icon=audio-x-vorbis+ogg.png
 Exec=konsole -e movie2sound4 -t o %F
 
 [Desktop Action flac]
 Name=FLAC
-Icon=sound
+Icon=audio-x-flac.png
 Exec=konsole -e movie2sound4 -t f %F
 
 #[Desktop Action aac]
@@ -43,15 +46,16 @@
 
 [Desktop Action ipod]
 Name=M4A for iPod (AAC)
+Name[tr]=iPod İçin M4A (AAC)
 Name[de]=M4A für iPod (AAC)
 Name[nl]=M4A voor iPod (AAC)
 Name[fr]=M4A pour iPod (AAC)
 Name[it]=M4A per iPod (AAC)
 Name[es]=M4A para iPod (AAC)
-Icon=sound
+Icon=audio-aac.png
 Exec=konsole -e movie2sound4 -t i %F
 
 [Desktop Action wav]
 Name=WAV
-Icon=sound
+Icon=audio-x-wav.png
 Exec=konsole -e movie2sound4 -t w %F
diff -Naur audiokonverter-5.8.2.orig/x-wavpack.desktop audiokonverter-5.8.2/x-wavpack.desktop
--- audiokonverter-5.8.2.orig/x-wavpack.desktop	2009-08-08 01:01:00.038337600 +0300
+++ audiokonverter-5.8.2/x-wavpack.desktop	2009-08-08 01:24:43.307463147 +0300
@@ -2,6 +2,6 @@
 Encoding=UTF-8
 Type=MimeType
 MimeType=audio/x-wavpack
-Icon=sound
+Icon=audio-x-wav.png
 Patterns=*.wv;*.WV;
 Comment=Wavpack Audio
