Index: data/kcm_systemmanager.desktop
===================================================================
--- data/kcm_systemmanager.desktop	(revision 27491)
+++ data/kcm_systemmanager.desktop	(revision 27508)
@@ -20,7 +20,7 @@
 Comment=System Manager
 Comment[sv]=Systemhanteraren
 Comment[es]=Administración del Sistema
-Comment[de]=System-Manager
+Comment[de]=System-Manager für grundlegende Einstellungen
 Comment[tr]=Sistem Yöneticisi
 Comment[nl]=Systeembeheer
 X-KDE-Keywords=python system managers
