--- src/finder.cpp.orig	2007-06-05 13:41:25.000000000 +0300
+++ src/finder.cpp	2007-09-20 12:08:14.000000000 +0300
@@ -25,7 +25,7 @@
 
 IFinder::IFinder() {
 #ifdef PREFIX
-	GET_CONFIG_VALUE("engine.path", std::string, path, RESOURCES_DIR "/private/data:" RESOURCES_DIR "/data");
+	GET_CONFIG_VALUE("engine.path", std::string, path, RESOURCES_DIR);
 #else
 	GET_CONFIG_VALUE("engine.path", std::string, path, "private/data:data");
 #endif
