--- src/core.cpp.orig	2011-06-02 16:25:55.723000061 +0300
+++ src/core.cpp	2011-06-02 16:33:50.614000063 +0300
@@ -1434,6 +1434,11 @@
 	if (!pref->vo.isEmpty()) {
 		proc->addArgument( "-vo");
 		proc->addArgument( pref->vo );
+                if(pref->vo == "vaapi") {
+                      proc->addArgument( "-va");
+                      proc->addArgument( pref->vo );
+                }
+
 	} else {
 		proc->addArgument("-vo");
 #ifdef Q_OS_WIN
