Index: kdeadmin-4.2.4/system-config-printer-kde/system-config-printer.ui
===================================================================
--- kdeadmin-4.2.4.orig/system-config-printer-kde/system-config-printer.ui
+++ kdeadmin-4.2.4/system-config-printer-kde/system-config-printer.ui
@@ -91,7 +91,7 @@ computer.</string>
          <item row="1" column="0">
           <widget class="QPushButton" name="btnNewPrinterNetwork">
            <property name="text">
-            <string>New Network Printer</string>
+            <string>New Printer</string>
            </property>
           </widget>
          </item>
@@ -104,8 +104,7 @@ computer.</string>
             </sizepolicy>
            </property>
            <property name="text">
-            <string>Add a new printer which is connected to your home
-network or on the Internet.</string>
+            <string>Add a new local or remote printer.</string>
            </property>
           </widget>
          </item>
