Index: florence-0.5.1/data/florence.desktop.in.in
===================================================================
--- florence-0.5.1.orig/data/florence.desktop.in.in
+++ florence-0.5.1/data/florence.desktop.in.in
@@ -1,10 +1,12 @@
 [Desktop Entry]
 Encoding=UTF-8
-_Name=Florence Virtual Keyboard
-_Comment=Florence Virtual Keyboard
+Name=Florence Virtual Keyboard
+Name[tr]=Florence Sanal Klavye
+Comment=Florence Virtual Keyboard
+Comment[tr]=Florence Sanal Klavye
 Exec=florence
 Terminal=false
 Type=Application
 Icon=@ICONDIR@/florence.svg
-Categories=Application;Accessibility;
+Categories=Qt;KDE;Utility;Accessibility;
 
