diff -Nur 2009_0302_RT2870_Linux_STA_v2.1.0.0.orig/os/linux/config.mk 2009_0302_RT2870_Linux_STA_v2.1.0.0/os/linux/config.mk
--- 2009_0302_RT2870_Linux_STA_v2.1.0.0.orig/os/linux/config.mk	2009-03-10 06:47:50.000000000 +0200
+++ 2009_0302_RT2870_Linux_STA_v2.1.0.0/os/linux/config.mk	2009-03-30 14:57:12.000000000 +0300
@@ -6,12 +6,12 @@
 
 #ifdef WPA_SUPPLICANT_SUPPORT
 # Support Wpa_Supplicant
-HAS_WPA_SUPPLICANT=n
+HAS_WPA_SUPPLICANT=y
 #endif // WPA_SUPPLICANT_SUPPORT //
 
 #ifdef NATIVE_WPA_SUPPLICANT_SUPPORT
 # Support Native WpaSupplicant for Network Maganger
-HAS_NATIVE_WPA_SUPPLICANT_SUPPORT=n
+HAS_NATIVE_WPA_SUPPLICANT_SUPPORT=y
 #endif // NATIVE_WPA_SUPPLICANT_SUPPORT //
 
 #Support Net interface block while Tx-Sw queue full
