diff -Nur 2009_0302_RT2870_Linux_STA_v2.1.0.0.orig/os/linux/usb_main_dev.c 2009_0302_RT2870_Linux_STA_v2.1.0.0/os/linux/usb_main_dev.c
--- 2009_0302_RT2870_Linux_STA_v2.1.0.0.orig/os/linux/usb_main_dev.c	2009-03-02 07:09:50.000000000 +0200
+++ 2009_0302_RT2870_Linux_STA_v2.1.0.0/os/linux/usb_main_dev.c	2009-03-30 15:03:05.000000000 +0300
@@ -105,6 +105,7 @@
 	{USB_DEVICE(0x1737,0x0071)}, /* Linksys WUSB600N */
 	{USB_DEVICE(0x0411,0x00e8)}, /* Buffalo WLI-UC-G300N*/
 	{USB_DEVICE(0x050d,0x815c)}, /* Belkin F5D8053 */
+    {USB_DEVICE(0x177f,0x0302)}, /* Sweex LW303 */
 #endif // RT2870 //
 	{ }/* Terminating entry */
 };
