diff -Nuar libglade-2.6.0.orig/Makefile.am libglade-2.6.0/Makefile.am
--- libglade-2.6.0.orig/Makefile.am	2007-03-31 18:59:51.000000000 +0300
+++ libglade-2.6.0/Makefile.am	2007-03-31 19:00:21.000000000 +0300
@@ -24,9 +24,7 @@
 
 if HAVE_XMLCATALOG
 install-data-local:
-	-$(XMLCATALOG) --noout --add "system" "http://glade.gnome.org/glade-2.0.dtd" $(xmldir)/glade-2.0.dtd $(XML_CATALOG_FILE)
 uninstall-local:
-	-$(XMLCATALOG) --noout --del $(xmldir)/glade-2.0.dtd $(XML_CATALOG_FILE)
 endif
 
 #small hack to get distcheck to work
