--- timidity/common.c
+++ timidity/common.c
@@ -104,7 +104,6 @@
   FILE *fp;
   PathList *plp;
   int l;
-  static int firsttime=1;
 
   if (!name || !(*name))
     {
@@ -113,10 +112,9 @@
     }
 
 #ifdef DEFAULT_PATH
-  if (firsttime && (pathlist==NULL)) {
+  if (pathlist==NULL) {
     /* Generate path list */
     add_to_pathlist(DEFAULT_PATH);
-    firsttime=0;
   }
 #endif
 
@@ -244,4 +242,5 @@
     free(plp);
     plp = next_plp;
   }
+  pathlist = NULL;
 }
