diff -Nur AssaultCube_v1.0.4-old/source/src/Makefile AssaultCube_v1.0.4/source/src/Makefile
--- AssaultCube_v1.0.4-old/source/src/Makefile	2009-08-30 16:31:18.298308784 +0300
+++ AssaultCube_v1.0.4/source/src/Makefile	2009-08-30 16:31:59.578286542 +0300
@@ -76,10 +76,10 @@
 	-$(MAKE) -C ../enet/ clean
 
 client: $(CLIENT_OBJS1) $(CLIENT_OBJS2)
-	$(CXX) $(CXXFLAGS) -o ac_client $(CLIENT_OBJS1) $(CLIENT_OBJS2) $(CLIENT_LIBS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o ac_client $(CLIENT_OBJS1) $(CLIENT_OBJS2) $(CLIENT_LIBS)
 
 server: libenet $(SERVER_FILES)
-	$(CXX) $(CXXFLAGSSERVER) -o ac_server -DSTANDALONE $(SERVER_FILES) $(SERVER_LIBS)
+	$(CXX) $(CXXFLAGSSERVER) $(LDFLAGS) -o ac_server -DSTANDALONE $(SERVER_FILES) $(SERVER_LIBS)
 
 install: all
 	cp ac_client ../../bin_unix/$(PLATFORM_PREFIX)_client
