--- configure.ac.orig	2008-01-29 00:47:14.000000000 +0200
+++ configure.ac	2008-04-09 22:54:18.000000000 +0300
@@ -271,7 +271,7 @@
 using_ffmpeg=no
 if test "$with_ffmpeg" != "no"; then
     using_ffmpeg=yes
-    AC_CHECK_HEADER(ffmpeg/avformat.h,
+    AC_CHECK_HEADER(libavformat/avformat.h,
         [AC_CHECK_LIB(avformat, av_open_input_file, FFMPEG_LIBS="$FFMPEG_LIBS -lavformat", using_ffmpeg=no)
         AC_CHECK_LIB(avutil, av_rescale_q, FFMPEG_LIBS="$FFMPEG_LIBS -lavutil", using_ffmpeg=no)
         AC_CHECK_LIB(avcodec, avcodec_decode_audio2, FFMPEG_LIBS="$FFMPEG_LIBS -lavcodec", using_ffmpeg=no)],
--- src/ffmpeg.c.orig	2008-01-08 05:36:41.000000000 +0200
+++ src/ffmpeg.c	2008-04-09 22:52:29.000000000 +0300
@@ -47,7 +47,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <ctype.h>
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 
 /* Private data for ffmpeg files */
 typedef struct ffmpeg
