diff -Naur sox-14.0.1.orig/configure.ac sox-14.0.1/configure.ac
--- sox-14.0.1.orig/configure.ac	2008-01-29 00:47:14.000000000 +0200
+++ sox-14.0.1/configure.ac	2008-02-02 12:05:12.000000000 +0200
@@ -412,7 +412,7 @@
 
 dnl Generate output files.
 AX_CREATE_STDINT_H(src/soxstdint.h)
-AC_CONFIG_FILES(Makefile src/Makefile libgsm/Makefile lpc10/Makefile)
+AC_CONFIG_FILES(Makefile src/Makefile)
 AC_OUTPUT
 
 if test "$found_libgsm" = "yes"; then
diff -Naur sox-14.0.1.orig/Makefile.am sox-14.0.1/Makefile.am
--- sox-14.0.1.orig/Makefile.am	2007-10-22 01:50:20.000000000 +0300
+++ sox-14.0.1/Makefile.am	2008-02-02 12:06:29.000000000 +0200
@@ -2,8 +2,8 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = lpc10 libgsm src
-DIST_SUBDIRS = lpc10 libgsm src
+SUBDIRS = src
+DIST_SUBDIRS = $(SUBDIRS)
 
 RM = rm -f
 
diff -Naur sox-14.0.1.orig/src/gsm.c sox-14.0.1/src/gsm.c
--- sox-14.0.1.orig/src/gsm.c	2007-11-04 18:28:29.000000000 +0200
+++ sox-14.0.1/src/gsm.c	2008-02-02 12:07:00.000000000 +0200
@@ -29,7 +29,7 @@
 #include "sox_i.h"
 
 #ifdef EXTERNAL_GSM
-#include <gsm/gsm.h>
+#include <gsm.h>
 #else
 #include "../libgsm/gsm.h"
 #endif
diff -Naur sox-14.0.1.orig/src/lpc10.c sox-14.0.1/src/lpc10.c
--- sox-14.0.1.orig/src/lpc10.c	2007-11-04 18:28:29.000000000 +0200
+++ sox-14.0.1/src/lpc10.c	2008-02-02 12:07:25.000000000 +0200
@@ -19,7 +19,7 @@
  * USA.  */
 
 #include "sox_i.h"
-#include "../lpc10/lpc10.h"
+#include <lpc10.h>
 
 /* Private data */
 typedef struct lpcpriv {
diff -Naur sox-14.0.1.orig/src/Makefile.am sox-14.0.1/src/Makefile.am
--- sox-14.0.1.orig/src/Makefile.am	2008-01-14 05:40:59.000000000 +0200
+++ sox-14.0.1/src/Makefile.am	2008-02-02 12:08:58.000000000 +0200
@@ -95,11 +95,11 @@
 libsox_fmt_dat_la_SOURCES = dat.c
 libsox_fmt_dat_la_LIBADD = libsox.la
 libsox_fmt_gsm_la_SOURCES = gsm.c
-libsox_fmt_gsm_la_LIBADD = libsox.la @GSM_LIBS@ @LIBGSM_LIBADD@
+libsox_fmt_gsm_la_LIBADD = libsox.la @GSM_LIBS@ -lgsm
 libsox_fmt_hcom_la_SOURCES = hcom.c
 libsox_fmt_hcom_la_LIBADD = libsox.la
 libsox_fmt_lpc10_la_SOURCES = lpc10.c
-libsox_fmt_lpc10_la_LIBADD = ../lpc10/liblpc10.la libsox.la
+libsox_fmt_lpc10_la_LIBADD = /usr/lib/liblpc10.la libsox.la -llpc10
 libsox_fmt_maud_la_SOURCES = maud.c
 libsox_fmt_maud_la_LIBADD = libsox.la
 libsox_fmt_prc_la_SOURCES = prc.c
@@ -121,7 +121,7 @@
 libsox_fmt_ima_la_SOURCES = ima-fmt.c
 libsox_fmt_ima_la_LIBADD = libsox.la
 libsox_fmt_wav_la_SOURCES = adpcm.c adpcm.h ima_rw.c ima_rw.h wav.c wav.h
-libsox_fmt_wav_la_LIBADD = libsox.la @GSM_LIBS@ @LIBGSM_LIBADD@
+libsox_fmt_wav_la_LIBADD = libsox.la @GSM_LIBS@ -lgsm
 libsox_fmt_wve_la_SOURCES = wve.c
 libsox_fmt_wve_la_LIBADD = libsox.la
 libsox_fmt_xa_la_SOURCES = xa.c
@@ -222,11 +222,11 @@
     dvms-fmt.c dat.c gsm.c hcom.c lpc10.c maud.c prc.c sf.c sfircam.h smp.c \
     sndrtool.c sphere.c tx16w.c voc.c vox-fmt.c ima-fmt.c adpcm.c adpcm.h \
     ima_rw.c ima_rw.h wav.c wav.h wve.c xa.c nulfile.c
-libsox_la_LIBADD = @GSM_LIBS@ @LIBGSM_LIBADD@
-libsox_la_LIBADD += ../lpc10/liblpc10.la
+libsox_la_LIBADD = @GSM_LIBS@ -lgsm
+libsox_la_LIBADD += /usr/lib/liblpc10.la -llpc10
 
-sox_LDADD += @GSM_LIBS@ @LIBGSM_LIBADD@
-sox_LDADD += ../lpc10/liblpc10.la
+sox_LDADD += @GSM_LIBS@ -lgsm
+sox_LDADD += /usr/lib/liblpc10.la -llpc10
 
 if HAVE_FLAC
     libsox_la_SOURCES += flac.c
diff -Naur sox-14.0.1.orig/src/wav.c sox-14.0.1/src/wav.c
--- sox-14.0.1.orig/src/wav.c	2007-11-04 18:28:31.000000000 +0200
+++ sox-14.0.1/src/wav.c	2008-02-02 12:09:12.000000000 +0200
@@ -24,7 +24,7 @@
 #include "ima_rw.h"
 #include "adpcm.h"
 #ifdef EXTERNAL_GSM
-#include <gsm/gsm.h>
+#include <gsm.h>
 #else
 #include "../libgsm/gsm.h"
 #endif
