diff -Nur hsfmodem-7.68.00.04full-old/modules/GPL/oscompat.h hsfmodem-7.68.00.04full/modules/GPL/oscompat.h
--- hsfmodem-7.68.00.04full-old/modules/GPL/oscompat.h	2007-12-05 18:21:34.626991735 +0200
+++ hsfmodem-7.68.00.04full/modules/GPL/oscompat.h	2007-12-05 18:23:36.163515374 +0200
@@ -70,7 +70,7 @@
 
 #include <linux/version.h>
 #ifdef FOUND_LINUX_CONFIG
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #endif
 #include <linux/types.h>
 
diff -Nur hsfmodem-7.68.00.04full-old/modules/GPL/serial_cnxt.c hsfmodem-7.68.00.04full/modules/GPL/serial_cnxt.c
--- hsfmodem-7.68.00.04full-old/modules/GPL/serial_cnxt.c	2007-12-05 18:21:34.625991887 +0200
+++ hsfmodem-7.68.00.04full/modules/GPL/serial_cnxt.c	2007-12-05 18:23:01.269820023 +0200
@@ -63,7 +63,7 @@
  */
 #include <linux/version.h>
 #ifdef FOUND_LINUX_CONFIG
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #endif
 #include <linux/module.h>
 #include <linux/errno.h>
diff -Nur hsfmodem-7.68.00.04full-old/modules/GPL/serial_core.c hsfmodem-7.68.00.04full/modules/GPL/serial_core.c
--- hsfmodem-7.68.00.04full-old/modules/GPL/serial_core.c	2007-12-05 18:21:34.625991887 +0200
+++ hsfmodem-7.68.00.04full/modules/GPL/serial_core.c	2007-12-05 18:23:15.396672414 +0200
@@ -29,7 +29,7 @@
  *
  */
 #include <linux/version.h>
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/module.h>
 #include <linux/errno.h>
 #include <linux/signal.h>
diff -Nur hsfmodem-7.68.00.04full-old/modules/GPL/serial_core.h hsfmodem-7.68.00.04full/modules/GPL/serial_core.h
--- hsfmodem-7.68.00.04full-old/modules/GPL/serial_core.h	2007-12-05 18:21:34.625991887 +0200
+++ hsfmodem-7.68.00.04full/modules/GPL/serial_core.h	2007-12-05 18:23:23.954371446 +0200
@@ -53,7 +53,7 @@
 
 #ifdef __KERNEL__
 
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/interrupt.h>
 #include <linux/serial.h>
 
