diff -ur vim70c.orig/runtime/plugin/zipPlugin.vim vim70c/runtime/plugin/zipPlugin.vim
--- vim70c.orig/runtime/plugin/zipPlugin.vim	2005-10-18 22:13:40.000000000 +0300
+++ vim70c/runtime/plugin/zipPlugin.vim	2006-04-04 02:26:14.000000000 +0300
@@ -35,6 +35,7 @@
  endif
 
  au BufReadCmd   *.zip		call zip#Browse(expand("<amatch>"))
+ au BufReadCmd   *.pisi		call zip#Browse(expand("<amatch>"))
 augroup END
 
 " ------------------------------------------------------------------------
