diff -Nur libvisual-plugins-0.4.0-old/plugins/actor/nastyfft/actor_nastyfft.c libvisual-plugins-0.4.0/plugins/actor/nastyfft/actor_nastyfft.c
--- libvisual-plugins-0.4.0-old/plugins/actor/nastyfft/actor_nastyfft.c	2008-06-16 05:52:46.000000000 +0300
+++ libvisual-plugins-0.4.0/plugins/actor/nastyfft/actor_nastyfft.c	2008-06-16 05:53:35.000000000 +0300
@@ -80,7 +80,7 @@
 		.vidoptions.depth = VISUAL_VIDEO_DEPTH_GL
 	}};
 
-	static const VisPluginInfo info[] = {{
+	static VisPluginInfo info[] = {{
 		.type = VISUAL_PLUGIN_TYPE_ACTOR,
 
 		.plugname = N_("nastyfft"),
