Index: data/usermanager.desktop
===================================================================
--- data/usermanager.desktop	(revision 27547)
+++ data/usermanager.desktop	(revision 27549)
@@ -5,18 +5,21 @@
 Name[es]=Usuarios
 Name[de]=Benutzerkonten
 Name[nl]=Gebruikerbeheer
+Name[fr]=Gestionnaire de L'utilisateur
 GenericName=User Manager
 GenericName[tr]=Kullanıcı Yöneticisi
 GenericName[sv]=Användarhanteraren
 GenericName[es]=Usuarios
 GenericName[de]=Benutzerkonten
 GenericName[nl]=Gebruikerbeheer
+GenericName[fr]=Gestionnaire de L'utilisateur
 Comment=An application for managing user accounts.
 Comment[tr]=Kullanıcı hesaplarını yönetmenizi sağlayan bir uygulama.
 Comment[sv]=Ett program för att hantera användarkonton.
 Comment[es]=Administración de cuentas de usuarios.
 Comment[de]=Benutzer hinzufügen und konfigurieren.
 Comment[nl]=Beheerder voor gebruiker van het systeem.
+Comment[fr]=Configuration de l'utilisateur
 Exec=user-manager
 Icon=system-users
 Type=Application
Index: data/kcm_usermanager.desktop
===================================================================
--- data/kcm_usermanager.desktop	(revision 27547)
+++ data/kcm_usermanager.desktop	(revision 27549)
@@ -17,10 +17,12 @@
 Name[es]=Usuarios
 Name[de]=Benutzerkonten
 Name[nl]=Gebruikerbeheer
+Name[fr]=Gestionnaire de L'utilisateur
 Comment=User Manager
 Comment[sv]=Användarhanteraren
 Comment[es]=Administrador de Usuarios
 Comment[de]=Benutzer hinzufügen und konfigurieren
 Comment[tr]=Kullanıcı Yöneticisi
 Comment[nl]=Beheerder voor gebruiker
+Comment[fr]=Configuration de l'utilisateur
 X-KDE-Keywords=python user managers
