diff -Nur lrzip-0.23-old/Makefile.in lrzip-0.23/Makefile.in
--- lrzip-0.23-old/Makefile.in	2009-05-06 01:04:36.000000000 +0300
+++ lrzip-0.23/Makefile.in	2009-05-06 01:05:12.000000000 +0300
@@ -122,7 +122,7 @@
 	${INSTALLCMD} -m 644 $(DOCFILES) $(DESTDIR)${INSTALL_DOC}
 
 lrzip: $(OBJS)
-	$(CXX) $(CXXFLAGS) -o lrzip $(OBJS) $(LIBS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o lrzip $(OBJS) $(LIBS)
 
 static: $(OBJS)
 	$(CXX) $(CXXFLAGS) -static -o lrzip $(OBJS) $(LIBS)
