diff -Naur konversation-1.4.orig/data/scripts/media konversation-1.4/data/scripts/media
--- konversation-1.4.orig/data/scripts/media	2012-01-06 23:51:25.000000000 +0200
+++ konversation-1.4/data/scripts/media	2012-01-06 23:54:02.017000040 +0200
@@ -44,7 +44,7 @@
 # If you add a new player, you must add it here or it won't get checked when in audio-only or video-only modes.
 playerRankings = {
     'video' : ['kaffeine','kmplayer', 'kplayer', 'noatun', 'kdetv'],
-    'audio' : ['amarok', 'amarok1', 'juk', 'noatun', 'kscd', 'kaffeine', 'kmplayer', 'yammi', 'Audacious', 'xmms', 'MPD', 'LastFM']
+    'audio' : ['amarok', 'amarok1', 'juk', 'noatun', 'kscd', 'kaffeine', 'kmplayer', 'yammi', 'Audacious', 'Audacious2', 'xmms', 'MPD', 'LastFM']
 }
 
 ## Title, album and artist fields to be quoted depending on contents
@@ -423,6 +423,26 @@
         else:
             return ''
             
+class Audacious2Player(Player):
+    def __init__(self, display_name):
+        Player.__init__(self, display_name)
+
+    def isRunning(self):
+        self.running = not os.system('audtool2 current-song')
+        return self.running
+
+    def getData(self):
+        if self.isRunning():
+            # get the title of the currently playing track
+            title = os.popen('audtool2 current-song-tuple-data title').read().strip()
+            artist = os.popen('audtool2 current-song-tuple-data artist').read().strip()
+            album = os.popen('audtool2 current-song-tuple-data album').read().strip()
+
+            return (self.reEncodeString(title),self.reEncodeString(artist),self.reEncodeString(album))
+
+        else:
+            return ''
+            
 class LastFMPlayer(Player):
     def __init__(self):
         Player.__init__(self,"last.fm")
@@ -505,7 +525,7 @@
 Amarok2Player(),
 ClementinePlayer(),
 DCOPPlayer("Yammi","yammi", "YammiPlayer songTitle", "YammiPlayer songArtist", "YammiPlayer songAlbum"),
-AudaciousPlayer('Audacious'),
+AudaciousPlayer('Audacious'), Audacious2Player('Audacious2'),
 XmmsPlayer('XMMS'),
 MPD('MPD'),
 LastFMPlayer()
