diff -Naur konversation-1.4.orig/data/scripts/media konversation-1.4/data/scripts/media
--- konversation-1.4.orig/data/scripts/media	2012-01-06 22:35:01.031000019 +0200
+++ konversation-1.4/data/scripts/media	2012-01-06 22:53:58.925000020 +0200
@@ -44,7 +44,7 @@
 # If you add a new player, you must add it here or it won't get checked when in audio-only or video-only modes.
 playerRankings = {
     'video' : ['kaffeine','kmplayer', 'kplayer', 'noatun', 'kdetv'],
-    'audio' : ['amarok', 'amarok1', 'juk', 'noatun', 'kscd', 'kaffeine', 'kmplayer', 'yammi', 'Audacious', 'xmms', 'MPD']
+    'audio' : ['amarok', 'amarok1', 'juk', 'noatun', 'kscd', 'kaffeine', 'kmplayer', 'yammi', 'Audacious', 'xmms', 'MPD', 'LastFM']
 }
 
 ## Title, album and artist fields to be quoted depending on contents
@@ -386,7 +386,16 @@
                 return (self.reEncodeString(data), '', '')
         else:
             return ''
+            
+class LastFMPlayer(Player):
+    def __init__(self):
+        Player.__init__(self,"last.fm")
 
+    def getData(self):
+        title = subprocess.Popen(["qdbus","com.lastfm.DBusInterface","/","currentTrack"], stdout=subprocess.PIPE).communicate()[0].rstrip()
+        artist = subprocess.Popen(["qdbus","com.lastfm.DBusInterface","/","currentArtist"], stdout=subprocess.PIPE).communicate()[0].rstrip()
+        album = subprocess.Popen(["qdbus","com.lastfm.DBusInterface","/","currentAlbum"], stdout=subprocess.PIPE).communicate()[0].rstrip()
+        return (title,artist,album)
 
 def playing(playerList, mode=None):
     for i in playerList:
@@ -461,7 +470,8 @@
 DCOPPlayer("Yammi","yammi", "YammiPlayer songTitle", "YammiPlayer songArtist", "YammiPlayer songAlbum"),
 AudaciousPlayer('Audacious'),
 XmmsPlayer('XMMS'),
-MPD('MPD')
+MPD('MPD'),
+LastFMPlayer()
 ]
 