--- Makefile	2006-11-01 22:48:49.000000000 +0200
+++ Makefile	2007-02-22 20:31:34.000000000 +0200
@@ -26,13 +26,13 @@
 # NOTE:  If you have problems compiling due to IW_MODE_MONITOR not being
 #        defined then you need to update the wireless extension version
 #	 installed in your kernel, or comment this line out.
-# CONFIG_IPW3945_MONITOR=y
+CONFIG_IPW3945_MONITOR=y
 
 # If you are interested in using radiotap headers in monitor mode,
 # simply uncomment:
 #
 # NOTE:  To use RADIOTAP you must also enable MONITOR above.
-# CONFIG_IEEE80211_RADIOTAP=y
+CONFIG_IEEE80211_RADIOTAP=y
 
 # The above monitor mode provides standard monitor mode.  The following
 # will create a new interface (named raw%d) which will be sent all
