--- testsuite/functions.sh.orig	2008-09-24 15:55:25.000000000 +0300
+++ testsuite/functions.sh	2008-09-24 15:55:58.000000000 +0300
@@ -1,5 +1,5 @@
 #!/bin/bash
-CC="${CC:-gcc} ${LINKOPTS}"
+CC="${CC:-gcc} -Wl,--disable-new-dtags ${LINKOPTS}"
 CCLINK=${CCLINK:-${CC} -Wl,--dynamic-linker=`echo ./ld*.so.*[0-9]`}
 CXX="${CXX:-g++} ${LINKOPTS}"
 CXXLINK=${CXXLINK:-${CXX} -Wl,--dynamic-linker=`echo ./ld*.so.*[0-9]`}
