diff -Nuar clutter-qt-b423e97ba28f529e3c618740ef6370bcc6ad2ced.orig//configure.ac clutter-qt-b423e97ba28f529e3c618740ef6370bcc6ad2ced/configure.ac
--- clutter-qt-b423e97ba28f529e3c618740ef6370bcc6ad2ced.orig//configure.ac	2010-11-22 14:23:56.000000000 +0200
+++ clutter-qt-b423e97ba28f529e3c618740ef6370bcc6ad2ced/configure.ac	2011-05-18 17:01:41.586999387 +0300
@@ -90,7 +90,7 @@
 dnl ========================================================================
 
 if test -z "$with_moc" ; then
-        AC_PATH_PROG(QT_MOC, moc-qt4, no)
+        AC_PATH_PROG(QT_MOC, moc, no)
         if test x"$QT_MOC" = xno; then
                 AC_PATH_PROG(QT_MOC, moc, no)
         fi
