diff -Naur mt-daapd-0.2.4.2.orig/configure mt-daapd-0.2.4.2/configure
--- mt-daapd-0.2.4.2.orig/configure	2010-12-07 12:27:34.812000078 +0000
+++ mt-daapd-0.2.4.2/configure	2010-12-07 12:29:09.684000077 +0000
@@ -3869,6 +3869,8 @@
 STATIC_LIBS=no
 CPPFLAGS="${CPPFLAGS} -g"
 
+AM_MAINTAINER_MODE
+
 { echo "$as_me:$LINENO: checking for pthread_creat in -lc_r" >&5
 echo $ECHO_N "checking for pthread_creat in -lc_r... $ECHO_C" >&6; }
 if test "${ac_cv_lib_c_r_pthread_creat+set}" = set; then
