diff -ur libvirt-0.6.4.orig/configure.in libvirt-0.6.4/configure.in
--- libvirt-0.6.4.orig/configure.in	2009-05-29 18:00:58.000000000 +0300
+++ libvirt-0.6.4/configure.in	2009-06-07 22:26:35.170146660 +0300
@@ -1315,7 +1315,6 @@
     old_LDFLAGS=$LDFLAGS
     CFLAGS="$CFLAGS $HAL_CFLAGS"
     LDFLAGS="$LDFLAGS $HAL_LIBS"
-    AC_CHECK_FUNCS([libhal_get_all_devices],,[with_hal=no])
     AC_CHECK_FUNCS([dbus_watch_get_unix_fd])
     CFLAGS="$old_CFLAGS"
     LDFLAGS="$old_LDFLAGS"

