--- pokerclient2d/Makefile.in~	2008-05-23 12:03:05.000000000 +0300
+++ pokerclient2d/Makefile.in	2008-06-29 14:48:25.000000000 +0300
@@ -402,7 +402,7 @@
 #
 # Authors:
 #  Loic Dachary <loic@gnu.org>
-#  Henry Prêcheur <henry@precheur.org> (2004)
+#  Henry Prï¿½cheur <henry@precheur.org> (2004)
 #
 INCLUDES = -Wstrict-prototypes
 MAINTAINERCLEANFILES = \
@@ -430,7 +430,7 @@
 	khotkeys.py
 
 man_MANS = poker2d.6
-gamesdir = ${prefix}/games
+gamesdir = ${bindir}
 games_SCRIPTS = poker2d
 poker_interface_SOURCES = \
 	network.c \
