diff -Naur tests.orig/test-unittestgui.cpp tests/test-unittestgui.cpp
--- tests.orig/test-unittestgui.cpp	2008-03-20 16:47:27.000000000 +0200
+++ tests/test-unittestgui.cpp	2008-06-29 13:24:40.000000000 +0300
@@ -28,6 +28,7 @@
 
 #include "UnitTest++.h"
 #include <iostream>
+#include <cstdlib>
 #include <gtk/gtk.h>
 #include <glade/glade-build.h>
 
diff -Naur tests.orig/test-unittesthandhistory.cpp tests/test-unittesthandhistory.cpp
--- tests.orig/test-unittesthandhistory.cpp	2008-03-20 16:47:27.000000000 +0200
+++ tests/test-unittesthandhistory.cpp	2008-06-29 13:27:38.000000000 +0300
@@ -28,6 +28,7 @@
 
 #include "UnitTest++.h"
 #include <iostream>
+#include <cstdlib>
 
 #include <gtk/gtk.h>
 #include <glade/glade-build.h>
diff -Naur tests.orig/test-unittestlobby.cpp tests/test-unittestlobby.cpp
--- tests.orig/test-unittestlobby.cpp	2008-03-20 16:47:27.000000000 +0200
+++ tests/test-unittestlobby.cpp	2008-06-29 13:24:24.000000000 +0300
@@ -28,6 +28,7 @@
 
 #include "UnitTest++.h"
 #include <iostream>
+#include <cstdlib>
 
 #include <gtk/gtk.h>
 #include <glade/glade-build.h>
diff -Naur tests.orig/test-unittestoutfits.cpp tests/test-unittestoutfits.cpp
--- tests.orig/test-unittestoutfits.cpp	2008-03-20 16:47:27.000000000 +0200
+++ tests/test-unittestoutfits.cpp	2008-06-29 13:25:01.000000000 +0300
@@ -28,6 +28,7 @@
 
 #include "UnitTest++.h"
 #include <iostream>
+#include <cstdlib>
 
 #include <gtk/gtk.h>
 #include <glade/glade-build.h>
diff -Naur tests.orig/test-unittesttournaments.cpp tests/test-unittesttournaments.cpp
--- tests.orig/test-unittesttournaments.cpp	2008-03-20 16:47:27.000000000 +0200
+++ tests/test-unittesttournaments.cpp	2008-06-29 13:26:35.000000000 +0300
@@ -28,6 +28,7 @@
 
 #include "UnitTest++.h"
 #include <iostream>
+#include <cstdlib>
 
 #include <gtk/gtk.h>
 #include <glade/glade-build.h>
