Index: zeroinstall-injector-1.2/setup.py
===================================================================
--- zeroinstall-injector-1.2.orig/setup.py
+++ zeroinstall-injector-1.2/setup.py
@@ -119,7 +119,7 @@ setup(name="zeroinstall-injector",
       author_email="zero-install-devel@lists.sourceforge.net",
       url="http://0install.net",
       scripts=['0launch', '0alias', '0store', '0store-secure-add', '0desktop', '0install'],
-      data_files = [('man/man1', ['0launch.1', '0alias.1', '0store-secure-add.1', '0store.1', '0desktop.1', '0install.1']),
+      data_files = [('share/man/man1', ['0launch.1', '0alias.1', '0store-secure-add.1', '0store.1', '0desktop.1', '0install.1']),
 		    ('share/applications', ['share/applications/zeroinstall-add.desktop', 'share/applications/zeroinstall-manage.desktop']),
 		    ('share/desktop-directories', ['share/desktop-directories/zeroinstall.directory']),
 		    ('share/icons/hicolor/24x24/apps', ['share/icons/24x24/zeroinstall.png']),
