--- man/Makefile.in.orig	2008-09-28 14:18:13.000000000 +0300
+++ man/Makefile.in	2008-09-28 14:23:42.000000000 +0300
@@ -837,8 +837,7 @@
 @HAVE_DBUS_TRUE@		$(LN_S) avahi-resolve.1 avahi-resolve-address.1 && \
 @HAVE_DBUS_TRUE@		$(LN_S) avahi-browse.1 avahi-browse-domains.1 && \
 @HAVE_DBUS_TRUE@		$(LN_S) avahi-publish.1 avahi-publish-address.1 && \
-@HAVE_DBUS_TRUE@                $(LN_S) avahi-publish.1 avahi-publish-service.1 && \
-@HAVE_DBUS_TRUE@                $(LN_S) bssh.1 bvnc.1
+@HAVE_DBUS_TRUE@                $(LN_S) avahi-publish.1 avahi-publish-service.1
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
