Index: mumble-1.1.7/src/mumble/ALSAAudio.cpp
===================================================================
--- mumble-1.1.7.orig/src/mumble/ALSAAudio.cpp	2009-02-13 14:02:05.000000000 +0100
+++ mumble-1.1.7/src/mumble/ALSAAudio.cpp	2009-02-13 14:02:16.000000000 +0100
@@ -112,12 +112,6 @@
 const QList<audioDevice> ALSAAudioInputRegistrar::getDeviceChoices() {
 	QList<audioDevice> qlReturn;
 
-	// Refresh the list of cards each time it is requested
-	if (cards) {
-		delete cards;
-		cards = new ALSAEnumerator();
-	}
-
 	QStringList qlInputDevs = cards->qhInput.keys();
 	qSort(qlInputDevs);
 
@@ -152,12 +146,6 @@
 const QList<audioDevice> ALSAAudioOutputRegistrar::getDeviceChoices() {
 	QList<audioDevice> qlReturn;
 
-	// Refresh the list of cards each time it is requested
-	if (cards) {
-		delete cards;
-		cards = new ALSAEnumerator();
-	}
-
 	QStringList qlOutputDevs = cards->qhOutput.keys();
 	qSort(qlOutputDevs);
 
