--- plasma/theme.cpp.orig	2009-12-11 14:58:02.999943106 +0200
+++ plasma/theme.cpp	2009-12-11 14:59:49.488043538 +0200
@@ -71,7 +71,8 @@
           useNativeWidgetStyle(false)
     {
         generalFont = QApplication::font();
-        cacheTheme = cacheConfig().readEntry("CacheTheme", true);
+        bool livemedia = QFile::exists("/var/run/pardus/livemedia");
+        cacheTheme = cacheConfig().readEntry("CacheTheme", !livemedia);
 
 #ifdef Q_WS_X11
         Display *dpy = QX11Info::display();
