Index: kdelibs/kded/kbuildsycoca.cpp
===================================================================
--- kded/kbuildsycoca.cpp	(revision 1057324)
+++ kded/kbuildsycoca.cpp	(revision 1057325)
@@ -665,10 +665,11 @@
 
 #ifdef KBUILDSYCOCA_GUI
    KApplication k;
+   k.disableSessionManagement();
 #else
-   KApplication k(false);
+   QCoreApplication k(argc, argv);
+   KComponentData mainComponent(d);
 #endif
-   k.disableSessionManagement();
 
 #ifdef KBUILDSYCOCA_GUI
    silent = args->isSet("silent");
