diff -ur /var/tmp/pisi/kdebase-3.5.0-21/work/kdebase-3.5.0/kcontrol/background/bgdialog.cpp kdebase-3.5.0/kcontrol/background/bgdialog.cpp
--- /var/tmp/pisi/kdebase-3.5.0-21/work/kdebase-3.5.0/kcontrol/background/bgdialog.cpp	2005-10-10 18:03:42.000000000 +0300
+++ kdebase-3.5.0/kcontrol/background/bgdialog.cpp	2006-01-28 05:53:32.527528888 +0200
@@ -1251,7 +1251,7 @@
    //kcmshellrc or kcontrolrc
    KConfig* config = KGlobal::config();
    config->setGroup("KNewStuff");
-   config->writeEntry( "ProvidersUrl", "http://download.kde.org/khotnewstuff/wallpaper-providers.xml" );
+   config->writeEntry( "ProvidersUrl", "http://sanat.uludag.org.tr/kde/wallpaper-providers.xml" );
    config->writeEntry( "StandardResource", "wallpaper" );
    config->sync();
 
