diff -ur kdebase-3.5.4.orig/ksmserver/legacy.cpp kdebase-3.5.4/ksmserver/legacy.cpp
--- kdebase-3.5.4.orig/ksmserver/legacy.cpp	2006-09-24 01:51:26.000000000 +0300
+++ kdebase-3.5.4/ksmserver/legacy.cpp	2006-09-24 02:19:06.000000000 +0300
@@ -220,7 +220,7 @@
     for (WindowMap::ConstIterator it = legacyWindows.begin(); it != legacyWindows.end(); ++it) {
         if ( (*it).type != SM_ERROR) {
             if( excludeApps.contains( (*it).wmclass1.lower())
-                || excludeApps.contains( (*it).wmclass2.lower()))
+                || excludeApps.contains( (*it).wmclass2.lower()) || (*it).wmCommand[0] == "compiz" || (*it).wmCommand[0] == "beryl")
                 continue;
             if ( !(*it).wmCommand.isEmpty() && !(*it).wmClientMachine.isEmpty() ) {
                 count++;
