--- cups-1.2.12/cgi-bin/search.c.orig	2006-09-17 21:01:47.000000000 +0200
+++ cups-1.2.12/cgi-bin/search.c	2008-03-06 11:27:21.000000000 +0100
@@ -171,7 +171,9 @@ cgiCompileSearch(const char *query)	/* I
       * string + RE overhead...
       */
 
-      wlen = (sptr - s) + 4 * wlen + 2 * strlen(prefix) + 4;
+      wlen = (sptr - s) + 2 * 4 * wlen + 2 * strlen(prefix) + 11;
+      if (lword)
+        wlen += strlen(lword);
 
       if (wlen > slen)
       {
