Index: calibre/src/calibre/db/backend.py
===================================================================
--- calibre.orig/src/calibre/db/backend.py
+++ calibre/src/calibre/db/backend.py
@@ -142,7 +142,7 @@ class Connection(apsw.Connection): # {{{
 
         # Dummy functions for dynamically created filters
         self.conn.create_function('books_list_filter', 1, lambda x: 1)
-        self.conn.create_collation('icucollate', icu_collator)
+        self.conn.create_collation('ICUCOLLATE', icu_collator)
 
     def create_dynamic_filter(self, name):
         f = DynamicFilter(name)
Index: calibre/src/calibre/library/sqlite.py
===================================================================
--- calibre.orig/src/calibre/library/sqlite.py
+++ calibre/src/calibre/library/sqlite.py
@@ -238,7 +238,7 @@ class DBThread(Thread):
         self.conn.create_function('uuid4', 0, lambda : str(uuid.uuid4()))
         # Dummy functions for dynamically created filters
         self.conn.create_function('books_list_filter', 1, lambda x: 1)
-        self.conn.create_collation('icucollate', icu_collator)
+        self.conn.create_collation('ICUCOLLATE', icu_collator)
 
     def run(self):
         try:
