diff -ur pyparted-1.7.1.orig/partedmodule.c pyparted-1.7.1/partedmodule.c
--- pyparted-1.7.1.orig/partedmodule.c	2005-11-12 00:02:29.000000000 +0200
+++ pyparted-1.7.1/partedmodule.c	2006-07-12 16:52:49.000000000 +0300
@@ -358,7 +358,7 @@
         REGISTER_ENUM(PARTITION_EXTENDED);
         REGISTER_ENUM(PARTITION_FREESPACE);
         REGISTER_ENUM(PARTITION_METADATA);
-        REGISTER_ENUM(PARTITION_PROTECTED);
+        /* REGISTER_ENUM(PARTITION_PROTECTED); */
 
         REGISTER_ENUM(PARTITION_BOOT);
         REGISTER_ENUM(PARTITION_ROOT);
