diff -ur apparmor-profiles-2.0.1.orig/abstractions/fonts apparmor-profiles-2.0.1/abstractions/fonts
--- apparmor-profiles-2.0.1.orig/abstractions/fonts	2006-08-04 22:13:59.000000000 +0300
+++ apparmor-profiles-2.0.1/abstractions/fonts	2006-11-18 01:16:37.000000000 +0200
@@ -11,18 +11,18 @@
 
   /usr/share/AbiSuite/fonts/**          r,
 
-  /usr/X11R6/lib/X11/fonts/**           r,
-  /usr/X11R6/lib/modules/fonts/**       mr,
-  /usr/X11R6/lib64/X11/fonts/**         r,
-  /usr/X11R6/lib64/modules/fonts/**     mr,
+  /usr/lib/X11/fonts/**           r,
+  /usr/lib/modules/fonts/**       mr,
+  /usr/lib64/X11/fonts/**         r,
+  /usr/lib64/modules/fonts/**     mr,
 
-  /usr/X11R6/include/X11/fonts/**       r,
+  /usr/include/X11/fonts/**       r,
 
   /usr/share/fonts/**                   r,
 
   /etc/fonts/**                         r,
 
-  /opt/kde3/share/fonts/**              r,
+  /usr/kde/3.5/share/fonts/**              r,
 
   /usr/lib/openoffice/share/fonts/**    r,
   /usr/lib64/openoffice/share/fonts/**  r,
diff -ur apparmor-profiles-2.0.1.orig/abstractions/kde apparmor-profiles-2.0.1/abstractions/kde
--- apparmor-profiles-2.0.1.orig/abstractions/kde	2006-08-04 22:13:59.000000000 +0300
+++ apparmor-profiles-2.0.1/abstractions/kde	2006-11-18 01:16:37.000000000 +0200
@@ -17,9 +17,9 @@
 /etc/X11/qt_plugins_3.3rc r,
 /etc/X11/qtrc r,
 /etc/kde3rc r,
-/etc/opt/kde3/share/config/* r,
-/etc/opt/kde3/share/icons r,
-/etc/opt/kde3/share/icons/** r,
+/etc/usr/kde/3.5/share/config/* r,
+/etc/usr/kde/3.5/share/icons r,
+/etc/usr/kde/3.5/share/icons/** r,
 
 @{HOME}/.DCOPserver_* r,
 @{HOME}/.ICEauthority r,
@@ -27,25 +27,25 @@
 @{HOME}/.kde/share/config/kdeglobals rw,
 @{HOME}/.qt/** rw,
 
-/opt/kde3/lib64/kde3/plugins/styles r,
-/opt/kde3/lib64/kde3/plugins/styles/* mr,
-/opt/kde3/lib64/lib*so* mr,
-/opt/kde3/lib/kde3/plugins/styles r,
-/opt/kde3/lib/kde3/plugins/styles/* mr,
-/opt/kde3/lib/lib*so* mr,
+/usr/kde/3.5/lib64/kde3/plugins/styles r,
+/usr/kde/3.5/lib64/kde3/plugins/styles/* mr,
+/usr/kde/3.5/lib64/lib*so* mr,
+/usr/kde/3.5/lib/kde3/plugins/styles r,
+/usr/kde/3.5/lib/kde3/plugins/styles/* mr,
+/usr/kde/3.5/lib/lib*so* mr,
 
-/opt/kde3/share/config/kdeglobals r,
-/opt/kde3/share/icons r,
-/opt/kde3/share/icons/** r,
-/usr/X11R6/lib64/X11/XKeysymDB r,
-/usr/X11R6/lib64/X11/icons/** r,
-/usr/X11R6/lib/X11/XKeysymDB r,
-/usr/X11R6/lib/X11/icons/** r,
+/usr/kde/3.5/share/config/kdeglobals r,
+/usr/kde/3.5/share/icons r,
+/usr/kde/3.5/share/icons/** r,
+/usr/lib64/X11/XKeysymDB r,
+/usr/lib64/X11/icons/** r,
+/usr/lib/X11/XKeysymDB r,
+/usr/lib/X11/icons/** r,
 
-/usr/lib/qt3/lib64/lib*so* mr,
+/usr/qt/3/lib64/lib*so* mr,
 /usr/lib64/qt3/plugins/**  mr,
-/usr/lib/qt3/lib/lib*so* mr,
-/usr/lib/qt3/plugins/**  mr,
+/usr/qt/3/lib/lib*so* mr,
+/usr/qt/3/plugins/**  mr,
 
 /usr/share/YaST2/theme/** r,
 /usr/share/pixmaps r,
diff -ur apparmor-profiles-2.0.1.orig/abstractions/php5 apparmor-profiles-2.0.1/abstractions/php5
--- apparmor-profiles-2.0.1.orig/abstractions/php5	2006-08-04 22:13:59.000000000 +0300
+++ apparmor-profiles-2.0.1/abstractions/php5	2006-11-18 01:16:37.000000000 +0200
@@ -17,7 +17,7 @@
   /etc/php5/fastcgi/php.ini r,
 
   # Xlibs
-  /usr/X11R6/lib{64,}/lib*.so* mr,
+  /usr/lib{64,}/lib*.so* mr,
   # php extensions
   /usr/lib{64,}/php5/extensions/*.so mr,
 
diff -ur apparmor-profiles-2.0.1.orig/abstractions/python apparmor-profiles-2.0.1/abstractions/python
--- apparmor-profiles-2.0.1.orig/abstractions/python	2006-09-12 01:17:43.000000000 +0300
+++ apparmor-profiles-2.0.1/abstractions/python	2006-11-18 01:16:40.000000000 +0200
@@ -18,3 +18,4 @@
   /usr/local/lib64/python2.[45]/site-packages r,
   /usr/local/lib/python2.[45]/**.{egg,py,pyc,pth,so} mr,
   /usr/local/lib/python2.[45]/site-packages r,
+  /usr/lib/pardus/**.{py,pyc,pth,so} mr,
diff -ur apparmor-profiles-2.0.1.orig/abstractions/X apparmor-profiles-2.0.1/abstractions/X
--- apparmor-profiles-2.0.1.orig/abstractions/X	2006-08-04 22:13:59.000000000 +0300
+++ apparmor-profiles-2.0.1/abstractions/X	2006-11-18 01:16:37.000000000 +0200
@@ -20,5 +20,5 @@
   # the unix socket to use to connect to the display
   /tmp/.X11-unix/*           w,
   # The X tree changes and is large -- grant read access to the whole thing
-  /usr/X11R6/**                   r,
-  /usr/X11R6/**.so*               mr,
+  /usr/**                   r,
+  /usr/**.so*               mr,
diff -ur apparmor-profiles-2.0.1.orig/extras/opt.gnome.bin.evolution-2.4 apparmor-profiles-2.0.1/extras/opt.gnome.bin.evolution-2.4
--- apparmor-profiles-2.0.1.orig/extras/opt.gnome.bin.evolution-2.4	2006-08-04 22:13:59.000000000 +0300
+++ apparmor-profiles-2.0.1/extras/opt.gnome.bin.evolution-2.4	2006-11-18 01:16:37.000000000 +0200
@@ -69,10 +69,10 @@
   /etc/opt/gnome/gnome-vfs-2.0/modules r,
   /etc/opt/gnome/gnome-vfs-2.0/modules/*.conf r,
   /etc/opt/gnome/pango/*.modules r,
-  /etc/opt/kde3/share/applications r,
-  /etc/opt/kde3/share/applications/kde r,
-  /etc/opt/kde3/share/applications/kde/*.desktop r,
-  /etc/opt/kde3/share/applications/mimeinfo.cache r,
+  /etc/usr/kde/3.5/share/applications r,
+  /etc/usr/kde/3.5/share/applications/kde r,
+  /etc/usr/kde/3.5/share/applications/kde/*.desktop r,
+  /etc/usr/kde/3.5/share/applications/mimeinfo.cache r,
   /etc/rpc r,
   /etc/xdg/menus/*.menu r,
   /etc/xdg/menus/applications-merged r,
@@ -136,10 +136,10 @@
   /opt/gnome/lib/pango/** r,
   /opt/gnome/lib/pango/**.so mr,
   /opt/gnome/share/** r,
-  /opt/kde3/bin/kde-config Uxr,
-  /opt/kde3/bin/kghostview Uxr,
-  /opt/kde3/bin/ksimus Ux,
-  /opt/kde3/share/** r,
+  /usr/kde/3.5/bin/kde-config Uxr,
+  /usr/kde/3.5/bin/kghostview Uxr,
+  /usr/kde/3.5/bin/ksimus Ux,
+  /usr/kde/3.5/share/** r,
   /opt/mozilla/bin/mozilla.sh Pxr,
   /opt/mozilla/lib/** r,
   /opt/mozilla/lib/**.so mr,
@@ -153,7 +153,7 @@
   /tmp/.ICE-unix/* w,
   /tmp/gconfd-** r,
   /tmp/orbit** lrw,
-  /usr/X11R6/bin/gv Uxr,
+  /usr/bin/gv Uxr,
   /usr/bin/gdb Uxr,
   /usr/bin/gpg Uxr,
   /usr/bin/lpr Uxr,
@@ -182,7 +182,7 @@
   /usr/share/spamassassin/*.cf r,
   /usr/share/spamassassin/triplets.txt r,
   /usr/share/xml/docbook/schema/** r,
-  /usr/X11R6/lib/Acrobat7/Resource/Font r,
-  /usr/X11R6/lib/Acrobat7/Resource/Font/** r,
+  /usr/lib/Acrobat7/Resource/Font r,
+  /usr/lib/Acrobat7/Resource/Font/** r,
   /var/tmp r,
 }
diff -ur apparmor-profiles-2.0.1.orig/extras/opt.gnome.bin.gaim apparmor-profiles-2.0.1/extras/opt.gnome.bin.gaim
--- apparmor-profiles-2.0.1.orig/extras/opt.gnome.bin.gaim	2006-08-04 22:13:59.000000000 +0300
+++ apparmor-profiles-2.0.1/extras/opt.gnome.bin.gaim	2006-11-18 01:16:37.000000000 +0200
@@ -55,17 +55,17 @@
   /opt/gnome/share/pixmaps/gaim/** r,
   /opt/gnome/share/sounds/gaim/* r,
   /opt/gnome/share/themes/** r,
-  /opt/kde3/bin/kde-config mixr,
-  /opt/kde3/bin/kfmclient Ux,
-  /opt/kde3/bin/kcminit rUx,
+  /usr/kde/3.5/bin/kde-config mixr,
+  /usr/kde/3.5/bin/kfmclient Ux,
+  /usr/kde/3.5/bin/kcminit rUx,
   /opt/mozilla/lib/lib*so* mr,
   /opt/mozilla/lib64/lib*so* mr,
   /proc/cpuinfo r,
   /proc/stat r,
   /proc/*/cmdline r,
-  /usr/X11R6/lib/Acrobat*/Resource/Font/* r,
-  /usr/X11R6/lib/Acrobat*/Resource/Font/PFM/* r,
-  /usr/X11R6/lib/lib*so* mr,
+  /usr/lib/Acrobat*/Resource/Font/* r,
+  /usr/lib/Acrobat*/Resource/Font/PFM/* r,
+  /usr/lib/lib*so* mr,
   /usr/lib/ao/plugins-* r,
   /usr/lib/ao/plugins-*/lib*.so mr,
   /usr/lib/aspell-** mr,
diff -ur apparmor-profiles-2.0.1.orig/extras/opt.gnome.lib.bonobo.bonobo-activation-server apparmor-profiles-2.0.1/extras/opt.gnome.lib.bonobo.bonobo-activation-server
--- apparmor-profiles-2.0.1.orig/extras/opt.gnome.lib.bonobo.bonobo-activation-server	2006-08-04 22:13:59.000000000 +0300
+++ apparmor-profiles-2.0.1/extras/opt.gnome.lib.bonobo.bonobo-activation-server	2006-11-18 01:16:37.000000000 +0200
@@ -24,5 +24,5 @@
   /opt/gnome/lib/evolution-data-server-1.2/evolution-data-server-1.4 Px,
   /opt/gnome/lib/evolution/2.4/evolution-alarm-notify Ux,
   /opt/gnome/lib/lib*so* mr,
-  /usr/X11R6/lib/X11/locale/* r,
+  /usr/lib/X11/locale/* r,
 }
diff -ur apparmor-profiles-2.0.1.orig/extras/opt.gnome.lib.evolution-data-server-1.2.evolution-data-server-1.4 apparmor-profiles-2.0.1/extras/opt.gnome.lib.evolution-data-server-1.2.evolution-data-server-1.4
--- apparmor-profiles-2.0.1.orig/extras/opt.gnome.lib.evolution-data-server-1.2.evolution-data-server-1.4	2006-08-04 22:13:59.000000000 +0300
+++ apparmor-profiles-2.0.1/extras/opt.gnome.lib.evolution-data-server-1.2.evolution-data-server-1.4	2006-11-18 01:16:37.000000000 +0200
@@ -39,6 +39,6 @@
   /opt/gnome/share/evolution-data-server-*/** mr,
   /proc/meminfo r,
   /proc/stat r,
-  /usr/X11R6/lib/X11/locale/* r,
+  /usr/lib/X11/locale/* r,
 
 }
diff -ur apparmor-profiles-2.0.1.orig/extras/usr.bin.opera apparmor-profiles-2.0.1/extras/usr.bin.opera
--- apparmor-profiles-2.0.1.orig/extras/usr.bin.opera	2006-08-04 22:13:59.000000000 +0300
+++ apparmor-profiles-2.0.1/extras/usr.bin.opera	2006-11-18 01:16:37.000000000 +0200
@@ -55,13 +55,13 @@
   /opt/mozilla/lib/plugins/** mr,
   /opt/gnome/share/** r,
   /usr r,
-  /opt/kde3/bin/kfmclient rUx,
-  /opt/kde3/lib/kde3/plugins/integration/*.so mr,
-  /usr/X11R6/bin/acroread rPx,
-  /usr/X11R6/lib r,
-  /usr/X11R6/lib/Acrobat7/Browser/intellinux/nppdf.so mr,
-  /usr/X11R6/lib/Acrobat7/Resource/Font/** r,
-  /usr/X11R6/lib/X11/XErrorDB r,
+  /usr/kde/3.5/bin/kfmclient rUx,
+  /usr/kde/3.5/lib/kde3/plugins/integration/*.so mr,
+  /usr/bin/acroread rPx,
+  /usr/lib r,
+  /usr/lib/Acrobat7/Browser/intellinux/nppdf.so mr,
+  /usr/lib/Acrobat7/Resource/Font/** r,
+  /usr/lib/X11/XErrorDB r,
   /usr/bin/lp rUx,
   /usr/bin/opera r,
   /usr/lib r,
@@ -79,17 +79,17 @@
   /var/spool/cups/tmp/* lrw,
   /opt/gnome/bin/gnome-open rUx,
 # external source editors
-  /usr/X11R6/bin/xedit rUx,
+  /usr/bin/xedit rUx,
   /usr/bin/xterm rUx,
-  /usr/X11R6/bin/rxvt rUx,
-  /opt/kde3/bin/kate rUx,
-  /opt/kde3/bin/kwrite rUx,
+  /usr/bin/rxvt rUx,
+  /usr/kde/3.5/bin/kate rUx,
+  /usr/kde/3.5/bin/kwrite rUx,
   /usr/bin/vi rUx,
   /usr/bin/vim rUx,
   /usr/bin/emacs rUx,
   /usr/bin/pico rUx,
-  /opt/kde3/bin/kvim rUx,
-  /usr/X11R6/bin/xemacs rUx,
+  /usr/kde/3.5/bin/kvim rUx,
+  /usr/bin/xemacs rUx,
   /bin/vim rUx,
   /bin/vi rUx,
 }
diff -ur apparmor-profiles-2.0.1.orig/extras/usr.lib.firefox.firefox-bin apparmor-profiles-2.0.1/extras/usr.lib.firefox.firefox-bin
--- apparmor-profiles-2.0.1.orig/extras/usr.lib.firefox.firefox-bin	2006-08-04 22:13:59.000000000 +0300
+++ apparmor-profiles-2.0.1/extras/usr.lib.firefox.firefox-bin	2006-11-18 01:16:37.000000000 +0200
@@ -43,7 +43,7 @@
   /etc/opt/gnome/gnome-vfs-*/modules r,
   /etc/opt/gnome/gnome-vfs-*/modules/*.conf r,
   /etc/opt/gnome/pango/* r,
-  /etc/opt/kde3/share/applications/mimeinfo.cache r,
+  /etc/usr/kde/3.5/share/applications/mimeinfo.cache r,
   /etc/rpc r,
   /etc/sysconfig/clock r,
   /opt/gnome/lib/GConf/2/gconfd-2 Px,
@@ -60,19 +60,19 @@
   /opt/gnome/share/locale/** r,
   /opt/gnome/share/pixmaps r,
   /opt/gnome/share/themes/** r,
-  /opt/kde3/bin/kde-config mixr,
-  /opt/kde3/share/applications/**.desktop r,
-  /opt/kde3/share/applications/mimeinfo.cache r,
+  /usr/kde/3.5/bin/kde-config mixr,
+  /usr/kde/3.5/share/applications/**.desktop r,
+  /usr/kde/3.5/share/applications/mimeinfo.cache r,
   /proc/*/cmdline r,
   /proc/*/maps r,
   /proc/*/stat r,
   /proc/net/if_inet6 r,
   /proc/stat r,
   /proc/sys/vm/heap-stack-gap r,
-  /usr/X11R6/lib/Acrobat*/Browser/intellinux/nppdf.so mr,
-  /usr/X11R6/lib/Acrobat*/Resource/Font/** r,
-  /usr/X11R6/lib/lib*so* mr,
-  /usr/X11R6/lib64/lib*so* mr,
+  /usr/lib/Acrobat*/Browser/intellinux/nppdf.so mr,
+  /usr/lib/Acrobat*/Resource/Font/** r,
+  /usr/lib/lib*so* mr,
+  /usr/lib64/lib*so* mr,
   /usr/bin/m4 rmix,
   /usr/bin/mozplugger-helper rmix,
   /usr/java/jre*/plugin/*/ns*/libjavaplugin_oji.so mr,
diff -ur apparmor-profiles-2.0.1.orig/extras/usr.lib.firefox.mozilla-xremote-client apparmor-profiles-2.0.1/extras/usr.lib.firefox.mozilla-xremote-client
--- apparmor-profiles-2.0.1.orig/extras/usr.lib.firefox.mozilla-xremote-client	2006-08-04 22:13:59.000000000 +0300
+++ apparmor-profiles-2.0.1/extras/usr.lib.firefox.mozilla-xremote-client	2006-11-18 01:16:37.000000000 +0200
@@ -18,5 +18,5 @@
 
   /usr/lib/mozilla/lib*so* mr,
   /usr/lib/firefox/mozilla-xremote-client rmix,
-  /usr/X11R6/lib/lib*so* mr,
+  /usr/lib/lib*so* mr,
 }
diff -ur apparmor-profiles-2.0.1.orig/extras/usr.lib.RealPlayer10.realplay apparmor-profiles-2.0.1/extras/usr.lib.RealPlayer10.realplay
--- apparmor-profiles-2.0.1.orig/extras/usr.lib.RealPlayer10.realplay	2006-11-03 14:58:04.000000000 +0200
+++ apparmor-profiles-2.0.1/extras/usr.lib.RealPlayer10.realplay	2006-11-18 01:16:37.000000000 +0200
@@ -44,8 +44,8 @@
   @{HOME}/ r,
   @{HOME}/.realplayerrc rw,
 
-  /usr/X11R6/lib/Acrobat7/Resource/Font/* r,
-  /usr/X11R6/lib/Acrobat7/Resource/Font/PFM/* r,
+  /usr/lib/Acrobat7/Resource/Font/* r,
+  /usr/lib/Acrobat7/Resource/Font/PFM/* r,
   /usr/lib/RealPlayer10/** mr,
   /usr/lib/RealPlayer10/realplay.bin mixr,
   /usr/lib/jvm/java-1.4.2-sun-1.4.2.06/jre/lib/fonts/** r,
diff -ur apparmor-profiles-2.0.1.orig/extras/usr.NX.bin.nxclient apparmor-profiles-2.0.1/extras/usr.NX.bin.nxclient
--- apparmor-profiles-2.0.1.orig/extras/usr.NX.bin.nxclient	2006-08-04 22:13:59.000000000 +0300
+++ apparmor-profiles-2.0.1/extras/usr.NX.bin.nxclient	2006-11-18 01:16:37.000000000 +0200
@@ -27,7 +27,7 @@
   /usr/NX/bin/nxproxy  mix,
   /usr/NX/lib/**       mr,
   /usr/NX/share/**     r,
-  /usr/X11R6/bin/xauth rmix,
+  /usr/bin/xauth rmix,
 
   @{HOME}/.nx               rwl,
   @{HOME}/.nx/**            rwl,
diff -ur apparmor-profiles-2.0.1.orig/extras/usr.sbin.httpd2-prefork apparmor-profiles-2.0.1/extras/usr.sbin.httpd2-prefork
--- apparmor-profiles-2.0.1.orig/extras/usr.sbin.httpd2-prefork	2006-11-03 14:58:04.000000000 +0200
+++ apparmor-profiles-2.0.1/extras/usr.sbin.httpd2-prefork	2006-11-18 01:16:37.000000000 +0200
@@ -45,8 +45,8 @@
   /tmp/auth_ldap_cache.sem wl,
   /tmp/session_mm_apache0.sem wl,
   /tmp/session_mm_apache2handler0.sem wl,
-  /usr/X11R6/lib64/lib*.so* mr,
-  /usr/X11R6/lib/lib*.so* mr,
+  /usr/lib64/lib*.so* mr,
+  /usr/lib/lib*.so* mr,
   /usr/apache2/error/* r,
 
   /usr/lib64/apache2-leader/{lib,mod_}*.so* mr,
@@ -75,7 +75,7 @@
   /usr/lib/php4/*.so mr,
   /usr/lib/python[12].[0-9]/**.{py,pyc,pth,so} mr,
   /usr/lib/python[12].[0-9]/site-packages r,
-  /usr/lib/qt3/lib/lib*.so* mr,
+  /usr/qt/3/lib/lib*.so* mr,
 
   /usr/local/tomcat/conf/mod_jk.conf r,
   /usr/local/tomcat/conf/workers-ajp12.properties r,
diff -ur apparmor-profiles-2.0.1.orig/extras/usr.sbin.xinetd apparmor-profiles-2.0.1/extras/usr.sbin.xinetd
--- apparmor-profiles-2.0.1.orig/extras/usr.sbin.xinetd	2006-08-04 22:13:59.000000000 +0300
+++ apparmor-profiles-2.0.1/extras/usr.sbin.xinetd	2006-11-18 01:16:37.000000000 +0200
@@ -67,6 +67,6 @@
   /usr/sbin/swat                   Px,
   /usr/sbin/tcpd                   Px,
   /usr/sbin/vsftpd                 Px,
-  /usr/X11R6/bin/vnc_inetd_httpd   Px,
-  /usr/X11R6/bin/Xvnc              Px,
+  /usr/bin/vnc_inetd_httpd   Px,
+  /usr/bin/Xvnc              Px,
 }
diff -ur apparmor-profiles-2.0.1.orig/extras/usr.X11R6.bin.acroread apparmor-profiles-2.0.1/extras/usr.X11R6.bin.acroread
--- apparmor-profiles-2.0.1.orig/extras/usr.X11R6.bin.acroread	2006-08-04 22:13:59.000000000 +0300
+++ apparmor-profiles-2.0.1/extras/usr.X11R6.bin.acroread	2006-11-18 01:16:37.000000000 +0200
@@ -13,7 +13,7 @@
 
 #include <tunables/global>
 
-/usr/X11R6/bin/acroread {
+/usr/bin/acroread {
   #include <abstractions/base>
   #include <abstractions/bash>
   #include <abstractions/consoles>
@@ -55,16 +55,16 @@
   /opt/gnome/share/icons r,
   /opt/gnome/share/icons/hicolor/icon-theme.cache r,
   /opt/gnome/share/pixmaps r,
-  /usr/X11R6/bin/acroread r,
-  /usr/X11R6/bin/xterm rUx,
-  /usr/X11R6/lib/Acrobat7/** mixr,
+  /usr/bin/acroread r,
+  /usr/bin/xterm rUx,
+  /usr/lib/Acrobat7/** mixr,
   /usr/bin/cut mixr,
   /usr/bin/dirname mixr,
   /usr/bin/lp rUx,
   /usr/bin/mutt rUx,
   /usr/bin/nail rUx,
   /opt/gnome/bin/evolution-2.4 Ux,
-  /opt/kde3/bin/kmail Ux,
+  /usr/kde/3.5/bin/kmail Ux,
   /usr/bin/which mixr,
   /usr/lib/jvm/java-1.4.2-sun-1.4.2.06/jre/lib/fonts/** r,
   /usr/lib/ooo-2.0/share/fonts/** r,
diff -ur apparmor-profiles-2.0.1.orig/extras/usr.X11R6.bin.ethereal apparmor-profiles-2.0.1/extras/usr.X11R6.bin.ethereal
--- apparmor-profiles-2.0.1.orig/extras/usr.X11R6.bin.ethereal	2006-11-03 14:58:04.000000000 +0200
+++ apparmor-profiles-2.0.1/extras/usr.X11R6.bin.ethereal	2006-11-18 01:16:37.000000000 +0200
@@ -13,7 +13,7 @@
 
 #include <tunables/global>
 
-/usr/X11R6/bin/ethereal {
+/usr/bin/ethereal {
   #include <abstractions/base>
   #include <abstractions/bash>
   #include <abstractions/consoles>
@@ -39,12 +39,12 @@
   /opt/gnome/share/mime/* r,
   /opt/mozilla/bin/mozilla.sh rPx,
   /opt/MozillaFirefox/bin/firefox.sh rPx,
-  /usr/X11R6/bin/ethereal mixr,
-  /usr/X11R6/lib/Acrobat7/Resource/Font/PFM/fonts.cache-1 r,
-  /usr/X11R6/lib/Acrobat7/Resource/Font/fonts.cache-1 r,
-  /usr/X11R6/lib/ethereal/** mr,
-  /usr/X11R6/lib/lib*so* mr,
-  /usr/X11R6/share/ethereal/** r,
+  /usr/bin/ethereal mixr,
+  /usr/lib/Acrobat7/Resource/Font/PFM/fonts.cache-1 r,
+  /usr/lib/Acrobat7/Resource/Font/fonts.cache-1 r,
+  /usr/lib/ethereal/** mr,
+  /usr/lib/lib*so* mr,
+  /usr/share/ethereal/** r,
   /usr/lib/jvm/java-1.4.2-sun-1.4.2.06/jre/lib/fonts/** r,
   /usr/lib/ooo-2.0/share/fonts/** r,
   /usr/share/icons r,
diff -ur apparmor-profiles-2.0.1.orig/extras/usr.X11R6.bin.xfs apparmor-profiles-2.0.1/extras/usr.X11R6.bin.xfs
--- apparmor-profiles-2.0.1.orig/extras/usr.X11R6.bin.xfs	2006-08-04 22:13:59.000000000 +0300
+++ apparmor-profiles-2.0.1/extras/usr.X11R6.bin.xfs	2006-11-18 01:16:37.000000000 +0200
@@ -11,7 +11,7 @@
 
 #include <tunables/global>
 
-/usr/X11R6/bin/xfs {
+/usr/bin/xfs {
   #include <abstractions/base>
   #include <abstractions/nameservice>
 
@@ -21,7 +21,7 @@
   /etc/mtab				r,
   /proc/meminfo				r,
   /tmp/.font-unix/fs710[0-9]		wl,
-  /usr/X11R6/bin/xfs                    rmix,
-  /usr/X11R6/lib/lib*.so*               mr,
+  /usr/bin/xfs                    rmix,
+  /usr/lib/lib*.so*               mr,
   /var/run/xfs.pid			rw,
 }
diff -ur apparmor-profiles-2.0.1.orig/program-chunks-disabled/user-mail-executables apparmor-profiles-2.0.1/program-chunks-disabled/user-mail-executables
--- apparmor-profiles-2.0.1.orig/program-chunks-disabled/user-mail-executables	2006-04-13 00:35:41.000000000 +0300
+++ apparmor-profiles-2.0.1/program-chunks-disabled/user-mail-executables	2006-11-18 01:16:37.000000000 +0200
@@ -15,7 +15,7 @@
   # wishes to pipe a mail message through a shell command
   /home/*/.findquote                px,
   /bin/gzip                         px,
-  /usr/X11R6/bin/xloadimage         px,
+  /usr/bin/xloadimage         px,
   /usr/bin/ee                       px,
   /usr/bin/fetchmail                px,
   /usr/bin/htmlview                 px,
diff -ur apparmor-profiles-2.0.1.orig/program-chunks-disabled/user-mozilla apparmor-profiles-2.0.1/program-chunks-disabled/user-mozilla
--- apparmor-profiles-2.0.1.orig/program-chunks-disabled/user-mozilla	2006-04-13 00:35:41.000000000 +0300
+++ apparmor-profiles-2.0.1/program-chunks-disabled/user-mozilla	2006-11-18 01:16:37.000000000 +0200
@@ -22,8 +22,8 @@
 
   # perhaps you wish your users to be able to execute programs right
   # from mozilla?
-  /usr/X11R6/bin/gv                         px,
+  /usr/bin/gv                         px,
   /usr/bin/plugger                          px,
   /usr/bin/xpdf                             px,
-  /usr/X11R6/bin/xmessage                   px,
+  /usr/bin/xmessage                   px,
 
