diff -Nur libvisual-plugins-0.4.0-old/configure libvisual-plugins-0.4.0/configure
--- libvisual-plugins-0.4.0-old/configure	2008-06-16 06:41:44.000000000 +0300
+++ libvisual-plugins-0.4.0/configure	2008-06-16 06:42:14.000000000 +0300
@@ -25860,7 +25860,7 @@
   ENABLE_ANALYZER=yes
 fi;
 
-if test "$ENABLE_ANALYZER" = xyes; then
+if test "x$ENABLE_ANALYZER" = "xyes"; then
   build_actor_plugins="$build_actor_plugins lv_analyzer"
 fi
 
