diff -Naur PisiYap-0.1-orig/setup.py PisiYap-0.1/setup.py
--- PisiYap-0.1-orig/setup.py	2009-07-20 10:01:38.806536354 +0300
+++ PisiYap-0.1/setup.py	2009-07-20 10:03:00.485648300 +0300
@@ -18,11 +18,10 @@
         system('lrelease-qt4 lang/%s' % l)
         langs.append(('lang/%s' % l).replace('.ts', '.qm'))
 
-datas = [('kde/3.5/share/apps/konqueror/servicemenus', ['menu/PisiYap-kde3.desktop']),\
-         ('kde/4/share/kde4/services/ServiceMenus', ['menu/PisiYap-kde4.desktop']),\
+datas = [('kde/4/share/kde4/services/ServiceMenus', ['menu/PisiYap-kde4.desktop']),\
          ('share/applications', ['menu/PisiYap.desktop']),\
          ('share/icons/hicolor/scalable/apps', ['icons/pisiyap.svg']),\
-         ('lib/python2.5/site-packages/PisiYap/lang', langs),\
+         ('lib/python2.6/site-packages/PisiYap/lang', langs),\
          ('share/pixmaps',      ['icons/pisiyap.png'])]
 
 setup(name = "PisiPap",
