--- a/template/en/default/reports/duplicates-table.html.tmpl	(revision 27749)
+++ b/template/en/default/reports/duplicates-table.html.tmpl	(revision 27750)
@@ -56,8 +56,8 @@
                               { name => "component", description => "Component" },
                               { name => "bug_severity", description => "Severity" },
                               { name => "op_sys", description => "Op Sys" },
-                              { name => "target_milestone",
-                                description => "Target<br>Milestone" },
+                              #{ name => "target_milestone",
+                              #  description => "Target<br>Milestone" },
                               { name => "short_desc", description => "Summary" } ]
          %]
 
--- a/template/en/custom/bug/create/create.html.tmpl	(revision 27749)
+++ b/template/en/custom/bug/create/create.html.tmpl	(revision 27750)
@@ -415,7 +415,7 @@
     </td>
   </tr>
 
-  <!--tr>
+  <tr>
     <th>Default CC:</th>
     <td colspan="2">
       <div id="initial_cc">
@@ -429,7 +429,7 @@
   
   <tr>
     <td colspan="3">&nbsp;</td>
-  </tr-->
+  </tr>
 
 [% IF user.in_group(Param('timetrackinggroup')) %]
   <!--tr>
@@ -552,6 +552,8 @@
     </td>
     <td valign="top">
       <select name="reproducible">
+        <option name="null" value="--">
+        --
         <option name="AlwaysReproducible" value="Always">
           Happens every time.
         </option>
@@ -658,7 +660,7 @@
       <b>Severity</b>
     </td>
     <td valign="top">
-      <select name="bug_severity">
+      <select name="bug_severity" size="7">
         <option name="blocking" value="blocking">
           Blocking: The bug makes whole system unusable
         <option name="critical" value="critical">
