--- a/template/en/default/reports/duplicates-table.html.tmpl	(revision 27734)
+++ b/template/en/default/reports/duplicates-table.html.tmpl	(revision 27735)
@@ -133,7 +133,7 @@
 
           <td>[% bug.component FILTER html %]</td>
           <td><center>[% bug.bug_severity FILTER html %]</center></td>
-          <td><center>[% bug.op_sys FILTER html %]</center></td>
+          <!--td><center>[% bug.op_sys FILTER html %]</center></td-->
           <td><center>[% bug.target_milestone FILTER html %]</center></td>
           <td>[% bug.short_desc FILTER html %]</td>
         </tr>
--- a/template/en/default/search/form.html.tmpl	(revision 27734)
+++ b/template/en/default/search/form.html.tmpl	(revision 27735)
@@ -395,7 +395,7 @@
         </tr>
       </table>
     </td>
-    <td>
+    <!--td>
       <table>
         <tr>
           <th align="left">
@@ -407,7 +407,7 @@
                                     size => 7 } %]
         </tr>
       </table>
-    </td>
+    </td-->
   </tr>
 </table>
 
--- a/template/en/default/list/edit-multiple.html.tmpl	(revision 27734)
+++ b/template/en/default/list/edit-multiple.html.tmpl	(revision 27735)
@@ -114,7 +114,7 @@
   </tr>
 
   <tr>
-    <th>
+    <!--th>
       <label for="op_sys">
         <a href="page.cgi?id=fields.html#op_sys">OS</a>:
       </label>
@@ -122,7 +122,7 @@
     <td [% " colspan=\"3\"" IF !Param("usetargetmilestone") %]>
       [% PROCESS selectmenu menuname = "op_sys"
                             menuitems = op_sys %]
-    </td>
+    </td-->
 
     [% IF Param("usetargetmilestone") %]
       <th><label for="target_milestone">Target Milestone:</label></th>
--- a/template/en/default/bug/edit.html.tmpl	(revision 27734)
+++ b/template/en/default/bug/edit.html.tmpl	(revision 27735)
@@ -417,10 +417,10 @@
             bug = bug, field = select_fields.rep_platform,
             no_tds = 1, value = bug.rep_platform
             editable = bug.check_can_change_field('rep_platform', 0, 1) %]
-       [%+ INCLUDE bug/field.html.tmpl 
+       <!--[%+ INCLUDE bug/field.html.tmpl 
             bug = bug, field = select_fields.op_sys, 
             no_tds = 1, value = bug.op_sys
-            editable = bug.check_can_change_field('op_sys', 0, 1) %]
+            editable = bug.check_can_change_field('op_sys', 0, 1) %]-->
        <script type="text/javascript">
          assignToDefaultOnChange(['product', 'component']);
        </script>
@@ -1034,7 +1034,7 @@
 [%############################################################################%]
 
 [% BLOCK section_timetracking %]
-  <table class="bz_time_tracking_table">
+  <!--table class="bz_time_tracking_table">
     <tr>
       <th>
         <label for="estimated_time">Orig. Est.</label>
@@ -1101,7 +1101,7 @@
         blocking this [% terms.bug %])</a>
       </td>
     </tr>
-  </table> 
+  </table--> 
 [% END %]
 
 [%############################################################################%]
