--- a/template/en/default/global/choose-product.html.tmpl	(revision 27735)
+++ b/template/en/default/global/choose-product.html.tmpl	(revision 27744)
@@ -40,7 +40,7 @@
 [% DEFAULT title = "Choose a Product" %]
 [% PROCESS global/header.html.tmpl %]
 
-<table>
+<table align="center">
 
 [% FOREACH c = classifications %]
   [% IF c.object %]
@@ -52,14 +52,29 @@
 
   [% FOREACH p = c.products %]
     <tr>
-      <th align="right" valign="top">
+      <td valign="top">
+      <table>
+      <tr>
+      <td rowspan="3">
         <a href="[% target %]?product=[% p.name FILTER url_quote -%]
               [%- IF cloned_bug_id %]&amp;cloned_bug_id=[% cloned_bug_id FILTER url_quote %][% END -%] 
               [%- IF format %]&amp;format=[% format FILTER url_quote %][% END %]">
+              <img height="50" width="50" src="icons/[%p.id%].png" border="0">
+              </a>&nbsp;
+     </td>
+     </tr>
+     <tr>
+      <td>
+        <a href="[% target %]?product=[% p.name FILTER url_quote -%]
+              [%- IF cloned_bug_id %]&amp;cloned_bug_id=[% cloned_bug_id FILTER url_quote %][% END -%] 
+              [%- IF format %]&amp;format=[% format FILTER url_quote %][% END %]">
         [% p.name FILTER html FILTER no_break %]</a>:&nbsp;
-      </th>
-
-      <td valign="top">[% p.description FILTER html_light %]</td>
+      </td>
+      </tr>
+      <tr>
+      <td>[% p.description FILTER html_light %]</td>
+      </tr>
+      </table>
     </tr>
   [% END %]
 
