--- imagination.desktop.in.in.orig	2009-07-20 17:02:42.000000000 +0300
+++ imagination.desktop.in.in	2009-11-01 19:57:48.233967823 +0200
@@ -3,9 +3,11 @@
 Version=1.0
 _Name=Imagination
 _GenericName=A DVD slideshow maker
+GenericName[tr]=Slayt gösteri uygulaması
 _Comment=A lightweight DVD slideshow maker using GTK+2
+Comment[tr]=Hafif, GTK+2 tabanlı DVD slayt gösteri oluşturma uygulaması
 Exec=imagination %F
 Icon=imagination
 Terminal=false
-Categories=AudioVideo;
+Categories=GTK;Application;Graphics;
 StartupNotify=true
