From 771c707eb2bba21b6013327bb62859c37be08673 Mon Sep 17 00:00:00 2001
From: John Nielsen <lists@jnielsen.net>
Date: Fri, 17 Apr 2009 20:16:57 +0000
Subject: R128: fix zaphod dualhead after libpciaccess switch

fixes bug 21251
---
diff --git a/src/r128_driver.c b/src/r128_driver.c
index 8f360a1..e3d7bb3 100644
--- a/src/r128_driver.c
+++ b/src/r128_driver.c
@@ -234,6 +234,16 @@ static Bool R128MapMMIO(ScrnInfoPtr pScrn)
     if (info->FBDev) {
 	info->MMIO = fbdevHWMapMMIO(pScrn);
     } else {
+        /* If the primary screen has already mapped the MMIO region,
+           use its pointer instead of mapping it a second time. */
+        if (info->IsSecondary) {
+            DevUnion* pPriv = xf86GetEntityPrivate(pScrn->entityList[0], 
+                                                   getR128EntityIndex());
+            R128EntPtr pR128Ent = pPriv->ptr;
+            R128InfoPtr info0 = R128PTR(pR128Ent->pPrimaryScrn);
+            info->MMIO=info0->MMIO;
+            if (info->MMIO) return TRUE;
+        }
 #ifndef XSERVER_LIBPCIACCESS
 	info->MMIO = xf86MapPciMem(pScrn->scrnIndex,
 				   VIDMEM_MMIO | VIDMEM_READSIDEEFFECT,
--
cgit v0.8.2
