Index: xastir-1.9.6/src/map_geo.c
===================================================================
--- xastir-1.9.6.orig/src/map_geo.c
+++ xastir-1.9.6/src/map_geo.c
@@ -1972,9 +1972,9 @@ fprintf(stderr,"2 ");
                 if (debug_level & 16)
                     fprintf(stderr,"Color quantum is [0..255]\n");
 
-                my_colors[l].red   = temp_pack.red << 8;
-                my_colors[l].green = temp_pack.green << 8;
-                my_colors[l].blue  = temp_pack.blue << 8;
+                my_colors[l].red   = (int)temp_pack.red << 8;
+                my_colors[l].green = (int)temp_pack.green << 8;
+                my_colors[l].blue  = (int)temp_pack.blue << 8;
             }
 
             // Take care not to screw up the transparency value by
@@ -2320,11 +2320,11 @@ fprintf(stderr,"2 ");
                         trans_skip = 1; // possibly transparent
                         if (image->storage_class == PseudoClass) {
                             if ( trans_color_head && 
-                                    check_trans(my_colors[index_pack[l]],trans_color_head) ) {
+                                    check_trans(my_colors[(int)index_pack[l]],trans_color_head) ) {
                                 trans_skip = 1; // skip it
                             }
                             else {
-                                XSetForeground(XtDisplay(w), gc, my_colors[index_pack[l]].pixel);
+                                XSetForeground(XtDisplay(w), gc, my_colors[(int)index_pack[l]].pixel);
                                 trans_skip = 0; // draw it
                             }
                         } else {
Index: xastir-1.9.6/src/map_tiger.c
===================================================================
--- xastir-1.9.6.orig/src/map_tiger.c
+++ xastir-1.9.6/src/map_tiger.c
@@ -711,9 +711,9 @@ void draw_tiger_map (Widget w,
             else {  // QuantumDepth = 8
                 if (debug_level & 512)
                     fprintf(stderr,"Color quantum is [0..255]\n");
-                my_colors[l].red   = (temp_pack.red << 8) * raster_map_intensity;
-                my_colors[l].green = (temp_pack.green << 8) * raster_map_intensity;
-                my_colors[l].blue  = (temp_pack.blue << 8) * raster_map_intensity;
+                my_colors[l].red   = ((int)temp_pack.red << 8) * raster_map_intensity;
+                my_colors[l].green = ((int)temp_pack.green << 8) * raster_map_intensity;
+                my_colors[l].blue  = ((int)temp_pack.blue << 8) * raster_map_intensity;
             }
 
             // Get the color allocated on < 8bpp displays. pixel color is written to my_colors.pixel
@@ -922,7 +922,7 @@ void draw_tiger_map (Widget w,
                         // now copy a pixel from the map image to the screen
                         l = map_x + map_y * image->columns;
                         if (image->storage_class == PseudoClass) {
-                            XSetForeground(XtDisplay(w), gc, my_colors[index_pack[l]].pixel);
+                            XSetForeground(XtDisplay(w), gc, my_colors[(int)index_pack[l]].pixel);
                         }
                         else {
                             pack_pixel_bits(pixel_pack[l].red * raster_map_intensity,
Index: xastir-1.9.6/src/map_WMS.c
===================================================================
--- xastir-1.9.6.orig/src/map_WMS.c
+++ xastir-1.9.6/src/map_WMS.c
@@ -766,9 +766,9 @@ void draw_WMS_map (Widget w,
             else {  // QuantumDepth = 8
                 if (debug_level & 512)
                     fprintf(stderr,"Color quantum is [0..255]\n");
-                my_colors[l].red   = (temp_pack.red << 8) * raster_map_intensity;
-                my_colors[l].green = (temp_pack.green << 8) * raster_map_intensity;
-                my_colors[l].blue  = (temp_pack.blue << 8) * raster_map_intensity;
+                my_colors[l].red   = ((int)temp_pack.red << 8) * raster_map_intensity;
+                my_colors[l].green = ((int)temp_pack.green << 8) * raster_map_intensity;
+                my_colors[l].blue  = ((int)temp_pack.blue << 8) * raster_map_intensity;
             }
 
             // Get the color allocated on < 8bpp displays. pixel color is written to my_colors.pixel
@@ -978,11 +978,11 @@ void draw_WMS_map (Widget w,
                         trans_skip = 1; // possibily transparent
                         if (image->storage_class == PseudoClass) {
                             if ( c_trans_color_head &&
-                                    check_trans(my_colors[index_pack[l]],c_trans_color_head)) {
+                                    check_trans(my_colors[(int)index_pack[l]],c_trans_color_head)) {
                                 trans_skip = 1; // skip it
                             }
                             else {
-                                XSetForeground(XtDisplay(w), gc, my_colors[index_pack[l]].pixel);
+                                XSetForeground(XtDisplay(w), gc, my_colors[(int)index_pack[l]].pixel);
                                 trans_skip = 0; // draw it
                             }
                         }
