diff -ur dmidecode-2.8.orig/biosdecode.c dmidecode-2.8/biosdecode.c
--- dmidecode-2.8.orig/biosdecode.c	2005-10-03 23:14:43.000000000 +0300
+++ dmidecode-2.8/biosdecode.c	2006-07-16 14:25:21.000000000 +0300
@@ -504,7 +504,7 @@
 };
 
 /* Believe it or not, this is significantly faster than memcmp and strncmp */
-static inline int anchor_match(const struct bios_entry *entry, const char *p)
+static __inline int anchor_match(const struct bios_entry *entry, const char *p)
 {
 	size_t i;
 	
diff -ur dmidecode-2.8.orig/Makefile dmidecode-2.8/Makefile
--- dmidecode-2.8.orig/Makefile	2005-10-03 21:38:20.000000000 +0300
+++ dmidecode-2.8/Makefile	2006-07-16 14:25:56.000000000 +0300
@@ -22,7 +22,7 @@
 LDFLAGS =
 
 DESTDIR =
-prefix  = /usr/local
+prefix  = /usr/
 sbindir = $(prefix)/sbin
 mandir  = $(prefix)/share/man
 man8dir = $(mandir)/man8
