diff -Nur linux-uvc-old/Makefile linux-uvc/Makefile
--- linux-uvc-old/Makefile	2008-02-14 20:43:09.359215039 +0200
+++ linux-uvc/Makefile	2008-02-14 20:43:26.838557774 +0200
@@ -11,7 +11,7 @@
 
 uvcvideo:
 	@echo "Building USB Video Class driver..."
-	@sh svn-version.sh > version.h 2>/dev/null
+	# @sh svn-version.sh > version.h 2>/dev/null
 	@(cd $(KERNEL_DIR) && make -C $(KERNEL_DIR) SUBDIRS=$(PWD) CROSS_COMPILE=$(CROSS_COMPILE) modules)
 
 install:
