diff -Nur linux-uvc-old/uvc_compat.h linux-uvc/uvc_compat.h
--- linux-uvc-old/uvc_compat.h	2008-01-12 03:02:03.000000000 +0200
+++ linux-uvc/uvc_compat.h	2008-01-12 03:04:24.000000000 +0200
@@ -244,6 +244,8 @@
 	return (usb_endpoint_xfer_int(epd) && usb_endpoint_dir_in(epd));
 }
 
+#endif
+
 /*
  * USB auto suspend
  *
@@ -255,8 +257,6 @@
 static inline void usb_autopm_put_interface(struct usb_interface *intf)
 { }
 
-#endif
-
 #endif /* __KERNEL__ */
 
 #endif /* _UVC_COMPAT_H */
