diff -Naur plasma-widget-athkar-0.2.1.orig/plasma_applet_athkar.desktop plasma-widget-athkar-0.2.1/plasma_applet_athkar.desktop
--- plasma-widget-athkar-0.2.1.orig/plasma_applet_athkar.desktop	2009-12-21 00:23:03.451978446 +0200
+++ plasma-widget-athkar-0.2.1/plasma_applet_athkar.desktop	2009-12-21 00:27:11.424976708 +0200
@@ -2,6 +2,7 @@
 Name=Athkar
 Name[ar]=أذكار
 Comment=Islamic supplications
+Comment[tr]=İslami dualar
 Comment[ar]=الأذكار الإسلامية
 Comment[fr]=Islamique supplications
 Icon=/usr/share/kde4/apps/plasma-athkar/tesbih.png
