diff --git a/libmpdemux/demux_mkv.c b/libmpdemux/demux_mkv.c
index fad06ef..8d8adae 100644
--- a/libmpdemux/demux_mkv.c
+++ b/libmpdemux/demux_mkv.c
@@ -543,6 +543,15 @@ lzo_fail:
             }
           *size = dstlen;
         }
+      else if (track->encodings[i].comp_algo == 3)
+        {
+          *dest = malloc (*size + track->encodings[i].comp_settings_len);
+          memcpy(*dest, track->encodings[i].comp_settings,
+                 track->encodings[i].comp_settings_len);
+          memcpy(*dest + track->encodings[i].comp_settings_len, src, *size);
+          *size += track->encodings[i].comp_settings_len;
+          modified = 1;
+        }
     }
 
   return modified;
@@ -713,7 +721,7 @@ demux_mkv_read_trackencodings (demuxer_t *demuxer, mkv_track_t *track)
                                  MSGTR_MPDEMUX_MKV_UnknownContentEncoding, track->tnum);
                         }
 
-                      if (e.comp_algo != 0 && e.comp_algo != 2)
+                      if (e.comp_algo != 0 && e.comp_algo != 2 && e.comp_algo != 3)
                         {
                           mp_msg (MSGT_DEMUX, MSGL_WARN,
                                   MSGTR_MPDEMUX_MKV_UnknownCompression,
