diff -Nur mplayer-20071112-old/libvo/x11_common.c mplayer-20071112/libvo/x11_common.c
--- mplayer-20071112-old/libvo/x11_common.c	2007-11-12 19:20:03.509924493 +0200
+++ mplayer-20071112/libvo/x11_common.c	2007-11-12 19:24:04.309317421 +0200
@@ -1353,7 +1353,7 @@
 
     if (u_dwidth > img_width)
     {
-        XFillRectangle(mDisplay, vo_window, f_gc, 0, left_ov, left_ov2,
+        XFillRectangle(mDisplay, vo_window, f_gc, 0, left_ov, left_ov2 + 1,
                        img_height);
         XFillRectangle(mDisplay, vo_window, f_gc, u_dwidth - left_ov2 - 1,
                        left_ov, left_ov2 + 1, img_height);
