Index: mozilla/security/nss/cmd/manifest.mn
===================================================================
--- mozilla.orig/security/nss/cmd/manifest.mn
+++ mozilla/security/nss/cmd/manifest.mn
@@ -54,7 +54,6 @@ DIRS = lib  \
  dbtest \
  derdump  \
  digest  \
- fipstest  \
  makepqg  \
  multinit \
  ocspclnt  \
@@ -84,6 +83,7 @@ DIRS = lib  \
  $(NULL)
 
 TEMPORARILY_DONT_BUILD = \
+ fipstest  \
  $(NULL)
 
 # rsaperf  \
