diff -Naur nprofile-0.1.5.orig/include/nprofile/node.h nprofile-0.1.5/include/nprofile/node.h
--- nprofile-0.1.5.orig/include/nprofile/node.h	2006-06-20 15:16:42.000000000 +0300
+++ nprofile-0.1.5/include/nprofile/node.h	2008-06-29 19:56:04.000000000 +0300
@@ -25,6 +25,7 @@
 
 
 #ifdef USE_NPROFILE
+#include <cstring>
 #include <vector>
 #include <iostream>
 #include <sstream>
diff -Naur nprofile-0.1.5.orig/include/nprofile/output.h nprofile-0.1.5/include/nprofile/output.h
--- nprofile-0.1.5.orig/include/nprofile/output.h	2006-06-27 03:46:48.000000000 +0300
+++ nprofile-0.1.5/include/nprofile/output.h	2008-06-29 19:56:15.000000000 +0300
@@ -28,6 +28,7 @@
 #ifndef NPROFILE_OUTPUT_H
 #define NPROFILE_OUTPUT_H
 
+#include <cstring>
 #include <ostream>
 #include <assert.h>
 
