Index: openshot-1.1.1/openshot/windows/preferences.py
===================================================================
--- openshot-1.1.1.orig/openshot/windows/preferences.py
+++ openshot-1.1.1/openshot/windows/preferences.py
@@ -249,7 +249,7 @@ class Settings:
 	#values will be overwritten with the file values.
 	general = {
 		"imported_image_length" : "7",
-		"default_theme" : "blue_glass",
+		"default_theme" : "simple",
 		"default_profile" : "DV NTSC",
 		"project_file_type" : "binary",
 		"max_history_size" : "20",
@@ -395,4 +395,4 @@ def main():
 	frm_titles.run()
 
 if __name__ == "__main__":
-	main()
\ Dosya sonunda yenisatır yok.
+	main()
