Index: config.tcl
===================================================================
--- config.tcl	(revision 10933)
+++ config.tcl	(working copy)
@@ -155,8 +155,8 @@
 
 		::config::setKey encoding auto						;#ANSN encoding
 
-		::config::setKey textsize 2							;#User text size
-		::config::setKey mychatfont "{Helvetica} {} 000000"	;#User chat font
+		::config::setKey textsize 0							;#User text size
+		::config::setKey mychatfont "{DeJavu\ Sans} {} 000000"	;#User chat font
 		::config::setKey winchatsize "350x320"		;#Default chat window size
 		::config::setKey winchatoutheight "200"		;#Default chat window output height
 		::config::setKey savechatwinsize 1			;#Save chat window sizes when changed?
@@ -247,7 +247,7 @@
 		::config::setKey enablebanner 1		;#Show or Hide AMSN Banner (By default Show)
 		::config::setKey startontray 0		;#Start amsn on tray icon only (hide contact list)
 		::config::setKey storename 1			;#Store original nick in a variable when go to custom states to revert it when go back
-		::config::setKey strictfonts 0		;#Use strict fonts' size in _ALL_ AMSN's fonts (Disabled by default)
+		::config::setKey strictfonts 1		;#Use strict fonts' size in _ALL_ AMSN's fonts (Disabled by default)
 		::config::setKey sngdblclick 0		;#Use single or double click to open a message window (0 double, 1 single)
 		::config::setKey nogap 0			;#Remove the empty line between groups
 		::config::setKey removeempty 0		;#Remove empty groups from the contact list
@@ -552,7 +552,7 @@
 		} elseif { [OnWin] } {
 			setGlobalKey basefont [list Arial 10 normal]
 		} else {
-			setGlobalKey basefont [list Helvetica 11 normal]	;#AMSN base font
+			setGlobalKey basefont [list DeJavu\ Sans 10 normal]	;#AMSN base font
 		}
 	}
 
