Index: config.tcl
===================================================================
--- config.tcl	(revision 10918)
+++ config.tcl	(working copy)
@@ -543,7 +543,7 @@
 
 		setGlobalKey last_client_version ""
 		setGlobalKey language [detect_language "en"]	;#Default language
-		setGlobalKey skin "default"			;#AMSN skin
+		setGlobalKey skin "Sapphire-2.5"			;#AMSN skin
 		setGlobalKey disableprofiles 0 ;#Disable profiles (useful for cybercafes or similar)
 
 		#Specific configs for Mac OS X (Aqua) first, and for others systems after
