diff -Nur blobwars-1.09b2-old/makefile blobwars-1.09b2/makefile
--- blobwars-1.09b2-old/makefile	2008-05-21 19:08:22.000000000 +0300
+++ blobwars-1.09b2/makefile	2008-05-21 19:08:44.000000000 +0300
@@ -15,7 +15,7 @@
 CXXFLAGS += `sdl-config --cflags` -DVERSION=$(VERSION) -DRELEASE=$(RELEASE) -DUSEPAK=$(USEPAK)
 CXXFLAGS += -DPAKNAME=\"$(PAKNAME)\" -DPAKLOCATION=\"$(DATADIR)\" -DUNIX -DGAMEPLAYMANUAL=\"$(DOCDIR)index.html\" -Wall
 CXXFLAGS += -DLOCALEDIR=\"$(LOCALEDIR)\"
-CXXFLAGS += $(CFLAGS) -pg -Werror
+CXXFLAGS += $(CFLAGS)
 LIBS = `sdl-config --libs` -lSDL_mixer -lSDL_image -lSDL_ttf -lz
 
 OBJS += CAudio.o
