--- gfx/src/ps/nsType1.cpp~	2002-12-11 07:24:47.000000000 +0100
+++ gfx/src/ps/nsType1.cpp	2006-08-06 20:57:01.000000000 +0200
@@ -362,10 +362,10 @@
 }
 
 static FT_Outline_Funcs ft_outline_funcs = {
-  moveto,
-  lineto,
-  conicto,
-  cubicto,
+  (int (*)(const FT_Vector*, void*))moveto,
+  (int (*)(const FT_Vector*, void*))lineto,
+  (int (*)(const FT_Vector*, const FT_Vector*, void*))conicto,
+  (int (*)(const FT_Vector*, const FT_Vector*, const FT_Vector*, void*))cubicto,
   0,
   0
 };
