diff -Nur mpeg4ip-1.5.0.1-old/common/video/iso-mpeg4/configure.in mpeg4ip-1.5.0.1/common/video/iso-mpeg4/configure.in
--- mpeg4ip-1.5.0.1-old/common/video/iso-mpeg4/configure.in	2008-04-11 14:56:18.000000000 +0300
+++ mpeg4ip-1.5.0.1/common/video/iso-mpeg4/configure.in	2008-04-11 14:56:54.000000000 +0300
@@ -62,8 +62,8 @@
 [
 save_CFLAG="$CFLAGS"
 save_CXXFLAG="$CXXFLAGS"
-CFLAGS="$CFLAGS -Werror $1"
-CXXFLAGS="$CXXFLAGS -Werror $1"
+CFLAGS="$CFLAGS $1"
+CXXFLAGS="$CXXFLAGS $1"
    AC_MSG_CHECKING([whether $3 accepts $1])
    AC_COMPILE_IFELSE(
 [extern void f(void); void f(void){}],
@@ -112,7 +112,7 @@
 	[if test x$enableval = xno; then
 	   AC_MSG_NOTICE([compiler warnings will not be errors])
          else 
-           BASE_ERR="$BASE_ERR -Werror"
+           BASE_ERR="$BASE_ERR "
 	   AC_MSG_NOTICE([enabling compiler warnings as errors])
          fi
 	 ],
diff -Nur mpeg4ip-1.5.0.1-old/common/video/iso-mpeg4/src/Makefile.am mpeg4ip-1.5.0.1/common/video/iso-mpeg4/src/Makefile.am
--- mpeg4ip-1.5.0.1-old/common/video/iso-mpeg4/src/Makefile.am	2008-04-11 14:56:18.000000000 +0300
+++ mpeg4ip-1.5.0.1/common/video/iso-mpeg4/src/Makefile.am	2008-04-11 14:57:03.000000000 +0300
@@ -1,5 +1,5 @@
 
-AM_CXXFLAGS = -D__TRACE_AND_STATS_ -D__DOUBLE_PRECISION_ -D_REENTRANT -DNOCONTROLS -fexceptions -Wall -Werror -D_OBSS_ @BILLS_CPPWARNINGS@
+AM_CXXFLAGS = -D__TRACE_AND_STATS_ -D__DOUBLE_PRECISION_ -D_REENTRANT -DNOCONTROLS -fexceptions -Wall -D_OBSS_ @BILLS_CPPWARNINGS@
 
 INCLUDES = -I$(srcdir)/../include
 
diff -Nur mpeg4ip-1.5.0.1-old/common/video/iso-mpeg4/src/Makefile.in mpeg4ip-1.5.0.1/common/video/iso-mpeg4/src/Makefile.in
--- mpeg4ip-1.5.0.1-old/common/video/iso-mpeg4/src/Makefile.in	2008-04-11 14:56:18.000000000 +0300
+++ mpeg4ip-1.5.0.1/common/video/iso-mpeg4/src/Makefile.in	2008-04-11 14:57:13.000000000 +0300
@@ -255,7 +255,7 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-AM_CXXFLAGS = -D__TRACE_AND_STATS_ -D__DOUBLE_PRECISION_ -D_REENTRANT -DNOCONTROLS -fexceptions -Wall -Werror -D_OBSS_ @BILLS_CPPWARNINGS@
+AM_CXXFLAGS = -D__TRACE_AND_STATS_ -D__DOUBLE_PRECISION_ -D_REENTRANT -DNOCONTROLS -fexceptions -Wall -D_OBSS_ @BILLS_CPPWARNINGS@
 INCLUDES = -I$(srcdir)/../include
 lib_LTLIBRARIES = libmpeg4ip_mpeg4_iso.la
 mp4vdec_SOURCES = \
