Index: work/xorg-server-1.4.2/hw/xfree86/common/xf86RandR.c
===================================================================
--- xorg-server-1.4.2/hw/xfree86/common/xf86RandR.c
+++ xorg-server-1.4.2/hw/xfree86/common/xf86RandR.c
@@ -224,6 +224,9 @@ xf86RandRSetConfig (ScreenPtr		pScreen,
     Bool		    useVirtual = FALSE;
     Rotation		    oldRotation = randrp->rotation;
 
+    if (!scrp->vtSema)
+                return FALSE;
+
     miPointerPosition (&px, &py);
     for (mode = scrp->modes; ; mode = mode->next)
     {
