--- emr.desktop.in.orig	2008-01-23 12:44:06.000000000 +0200
+++ emr.desktop.in	2008-01-23 12:48:28.000000000 +0200
@@ -3,13 +3,13 @@
 Version=1.0
 Type=Application
 Exec=@BINDIR@/emr
-Icon=emr.png
+Icon=emr
 Path=@DATADIR@/emr
 Terminal=0
 
-Name=Excalibur: Morgana's Revenge
-GenericName=Game
-Comment=First-person shooter game
+Name=EMR
+GenericName=Excalibur: Morgana's Revenge
+GenericName[tr]=Excalibur: Morgana'nın İntikamı
 Categories=Game;ActionGame;
 
 # Copyright (C) 2006 Glen Ditchfield
