diff -ur kdebase-runtime-4.0.72/phonon/xine/xineengine.cpp kdebase-runtime-4.0.72-pulseaudio/phonon/xine/xineengine.cpp
--- kdebase-runtime-4.0.72/phonon/xine/xineengine.cpp	2008-04-23 17:51:26.000000000 +0200
+++ kdebase-runtime-4.0.72-pulseaudio/phonon/xine/xineengine.cpp	2008-05-06 17:28:52.000000000 +0200
@@ -371,9 +371,9 @@
                                 "in KDE2 and KDE3. Its use is discouraged.</p></html>"),
                             /*icon name */"audio-backend-arts", outputPlugins[i]);
                 } else if (0 == strcmp(outputPlugins[i], "pulseaudio")) {
-                    addAudioOutput(nextIndex++, 10, i18n("PulseAudio"),
+                    addAudioOutput(nextIndex++, 2000, i18n("PulseAudio"),
                             xine_get_audio_driver_plugin_description(xine(), outputPlugins[i]),
-                            /*icon name */"audio-backend-pulseaudio", outputPlugins[i], true /*isAdvanced*/);
+                            /*icon name */"audio-backend-pulseaudio", outputPlugins[i]);
                 } else if (0 == strcmp(outputPlugins[i], "esd")) {
                     addAudioOutput(nextIndex++, 8, i18n("Esound (ESD)"),
                             xine_get_audio_driver_plugin_description(xine(), outputPlugins[i]),
