Index: work/vmd-1.8.7/src/vmd.C
===================================================================
--- work.orig/vmd-1.8.7/src/vmd.C
+++ work/vmd-1.8.7/src/vmd.C
@@ -305,6 +305,7 @@ int VMDinitialize(int *argc, char **argv
 #endif
 
 #ifdef VMDFLTK
+  Fl::scheme("gtk+");
   // Do various special FLTK initialization stuff here
   if ((which_display != DISPLAY_TEXT)) {
     // Cause FLTK to to use 24-bit color for all windows if possible
