Index: work/vmd-1.8.7/scripts/vmd/vmdinit.tcl
===================================================================
--- work.orig/vmd-1.8.7/scripts/vmd/vmdinit.tcl
+++ work/vmd-1.8.7/scripts/vmd/vmdinit.tcl
@@ -221,7 +221,7 @@ if { ![info exists env(VMDHTMLVIEWER)] }
       # Open in the current window (default)
       #set env(VMDHTMLVIEWER) "mozilla -remote openURL(%s)" 
       # Open in a new window
-      set env(VMDHTMLVIEWER) "firefox -remote openURL(%s,new-window)"
+      set env(VMDHTMLVIEWER) "xdg-open %s"
       # Open a new instance of mozilla
       # set env(VMDHTMLVIEWER) "mozilla %s &" 
     }
