Index: elsa-0.0.4_20110911/data/elsa.conf
===================================================================
--- elsa-0.0.4_20110911.orig/data/elsa.conf
+++ elsa-0.0.4_20110911/data/elsa.conf
@@ -1,5 +1,5 @@
 group "Elsa_Config" struct {
-    value "session_path" string: "/bin:/usr/bin:/usr/local/bin";
+    value "session_path" string: "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin";
 
     /* X server path */
     value "xinit_path" string: "/usr/bin/X";
