Index: pisi/api.py
===================================================================
--- pisi/api.py	(revision 18254)
+++ pisi/api.py	(revision 18255)
@@ -265,15 +265,24 @@
 
     return order
 
-def configure_pending():
+def configure_pending(pkgs):
     # start with pending packages
     # configure them in reverse topological order of dependency
-    A = ctx.installdb.list_pending()
-    order = generate_pending_order(A)
+
+    packages = {}
+    if not pkgs:
+        packages = ctx.installdb.list_pending()
+    else:
+        lp = ctx.installdb.list_pending()
+        for pkg in pkgs:
+            if lp.has_key(pkg):
+                packages[pkg] = lp[pkg]
+
+    order = generate_pending_order(packages)
     try:
         for x in order:
             if ctx.installdb.is_installed(x):
-                pkginfo = A[x]
+                pkginfo = packages[x]
                 pkgname = pisi.util.package_name(x, pkginfo.version,
                                         pkginfo.release,
                                         False,
Index: pisi/cli/commands.py
===================================================================
--- pisi/cli/commands.py	(revision 18254)
+++ pisi/cli/commands.py	(revision 18255)
@@ -859,7 +859,7 @@
     def run(self):
 
         self.init()
-        pisi.api.configure_pending()
+        pisi.api.configure_pending(self.args)
         self.finalize()
 
 
