Index: pisi/build.py
===================================================================
--- pisi/build.py	(revision 18290)
+++ pisi/build.py	(revision 18291)
@@ -429,7 +429,7 @@
         os.chdir(self.srcDir)
 
         if func in self.actionLocals:
-            if ctx.get_option('ignore_sandbox'):
+            if ctx.get_option('ignore_sandbox') or not ctx.config.values.build.enablesandbox:
                 self.actionLocals[func]()
             else:
                 import catbox
Index: pisi/configfile.py
===================================================================
--- pisi/configfile.py	(revision 18290)
+++ pisi/configfile.py	(revision 18291)
@@ -24,6 +24,7 @@
 #[build]
 #host = i686-pc-linux-gnu
 #generateDebug = False
+#enableSandbox = True
 #jobs = "-j1"
 #CFLAGS= -mtune=i686 -march=i686 -O2 -pipe -fomit-frame-pointer
 #CXXFLAGS= -mtune=i686 -march=i686 -O2 -pipe -fomit-frame-pointer
@@ -75,6 +76,7 @@
     host = "i686-pc-linux-gnu"
     jobs = "-j1"
     generateDebug = False
+    enableSandbox = True
     cflags = "-mtune=i686 -O2 -pipe -fomit-frame-pointer"
     cxxflags = "-mtune=i686 -O2 -pipe -fomit-frame-pointer"
     ldflags = ""
