Index: pisi/fetcher.py
===================================================================
--- pisi/fetcher.py     (revision 17110)
+++ pisi/fetcher.py     (working copy)
@@ -237,7 +237,7 @@

         elif ctx.config.values.general.ftp_proxy and self.url.scheme() == "ftp":
             ftp_proxy = ctx.config.values.general.ftp_proxy
-            proxy_handler = urllib2.ProxyHandler({pisi.uri.URI(http_proxy): ftp_proxy})
+            proxy_handler = urllib2.ProxyHandler({pisi.uri.URI(ftp_proxy): ftp_proxy})

         if proxy_handler:
             ctx.ui.info(_("Proxy configuration has been found for '%s' protocol") % self.url.scheme())
