diff -ur digikam-0.9.1.orig/digikam/utilities/hotplug/digikam-download.desktop.in digikam-0.9.1/digikam/utilities/hotplug/digikam-download.desktop.in
--- digikam-0.9.1.orig/digikam/utilities/hotplug/digikam-download.desktop.in	2007-03-04 22:33:32.000000000 +0200
+++ digikam-0.9.1/digikam/utilities/hotplug/digikam-download.desktop.in	2007-03-07 01:10:06.777415159 +0200
@@ -4,6 +4,7 @@
 Name=Download Photos with digiKam
 Name[de]=Bilder mit digiKam herunterladen
 Name[fr]=télécharger photos avec digiKam
+Name[tr]=digiKam ile resimler göster
 
 [Desktop Entry]
 Actions=digiKam Download
diff -ur digikam-0.9.1.orig/digikam/utilities/hotplug/digikam-gphoto2-camera.desktop.in digikam-0.9.1/digikam/utilities/hotplug/digikam-gphoto2-camera.desktop.in
--- digikam-0.9.1.orig/digikam/utilities/hotplug/digikam-gphoto2-camera.desktop.in	2007-03-04 22:33:32.000000000 +0200
+++ digikam-0.9.1/digikam/utilities/hotplug/digikam-gphoto2-camera.desktop.in	2007-03-07 01:10:34.950117237 +0200
@@ -4,6 +4,7 @@
 Name=digiKam Detect and Download
 Name[de]=Bilder mit digiKam herunterladen
 Name[fr]=télécharger photos avec digiKam
+Name[tr]=digiKam ile resimleri göster
 
 [Desktop Entry]
 Actions=digiKam Detect and Download
diff -ur digikam-0.9.1.orig/digikam/utilities/hotplug/digikam-mount-and-download.desktop.in digikam-0.9.1/digikam/utilities/hotplug/digikam-mount-and-download.desktop.in
--- digikam-0.9.1.orig/digikam/utilities/hotplug/digikam-mount-and-download.desktop.in	2007-03-04 22:33:32.000000000 +0200
+++ digikam-0.9.1/digikam/utilities/hotplug/digikam-mount-and-download.desktop.in	2007-03-07 01:10:21.735133266 +0200
@@ -4,6 +4,7 @@
 Name=Download Photos with digiKam
 Name[de]=Bilder mit digiKam herunterladen
 Name[fr]=télécharger photos avec digiKam
+Name[tr]=digiKam ile resimleri indir
 
 [Desktop Entry]
 Actions=digiKam Mount and Download
