--- Io-2007-04-30/build/AddonBuilder.io.orig	2007-04-28 15:09:52.000000000 +0200
+++ Io-2007-04-30/build/AddonBuilder.io	2007-05-29 23:01:38.000000000 +0200
@@ -58,6 +58,7 @@
 	libSearchPaths := List clone
 	appendLibSearchPath := method(v, if(File clone setPath(v) exists, libSearchPaths appendIfAbsent(v)))
 	searchPrefixes foreach(searchPrefix, appendLibSearchPath(searchPrefix .. "/lib"))
+	searchPrefixes foreach(searchPrefix, appendLibSearchPath(searchPrefix .. "/lib/mysql"))
 
 	debs    := Map clone
 	ebuilds := Map clone
