--- Makefile.orig	2008-01-21 06:57:00.000000000 +0200
+++ Makefile	2008-02-02 15:34:50.000000000 +0200
@@ -111,18 +111,17 @@
 	umask 022
 	mkdir -p $(INSTALL_PREFIX)/bin || true
 	mkdir -p $(INSTALL_PREFIX)/lib || true
+	mkdir -p $(INSTALL_PREFIX)/include || true
 	rm -f $(INSTALL_PREFIX)/bin/io$(BINARY_SUFFIX)
 	cp _build/binaries/io$(BINARY_SUFFIX) $(INSTALL_PREFIX)/bin || true
 	chmod ugo+rx $(INSTALL_PREFIX)/bin/io$(BINARY_SUFFIX)
-	cp _build/binaries/io_static$(BINARY_SUFFIX) $(INSTALL_PREFIX)/bin || true
-	chmod ugo+rx $(INSTALL_PREFIX)/bin/io_static$(BINARY_SUFFIX)  || true
 	cp _build/dll/* $(INSTALL_PREFIX)/lib  || true
-	cp _build/dll/* $(INSTALL_PREFIX)/bin  || true
-	cp _build/lib/* $(INSTALL_PREFIX)/lib  || true
 	rm -rf $(INSTALL_PREFIX)/lib/io || true
 	mkdir -p $(INSTALL_PREFIX)/lib/io || true
 	cp -fR addons $(INSTALL_PREFIX)/lib/io
 	chmod -R ugo+rX $(INSTALL_PREFIX)/lib/io
+	mkdir -p $(INSTALL_PREFIX)/include/io || true
+	cp _build/headers/* $(INSTALL_PREFIX)/include/io  || true
 
 linkInstall:
 	mkdir -p $(INSTALL_PREFIX)/bin || true
