--- configure.orig	2006-04-05 21:36:45.000000000 +0300
+++ configure	2006-04-05 21:37:07.000000000 +0300
@@ -32356,7 +32356,7 @@
 	for LIBREADLINE in "-lreadline" "-lreadline $LIBCURSES" "-lreadline -ltermcap" "-lreadline -lncurses" "-lreadline -lcurses"
 	do
 		LIBS="$ac_save_LIBS $LIBREADLINE"
-		LIB_GPSIM="$LIBREADLINE -lgpsim -lgpsim_eXdbm -lgpsimcli"
+		LIB_GPSIM="-lgpsim -lgpsim_eXdbm -lgpsimcli $LIBREADLINE"
 		if test "$cross_compiling" = yes; then
 
 		# action if cross compiling
