diff -Naur griffith-0.9.10.orig/data/griffith.desktop griffith-0.9.10/data/griffith.desktop
--- griffith-0.9.10.orig/data/griffith.desktop	2009-06-23 20:30:24.661972754 +0300
+++ griffith-0.9.10/data/griffith.desktop	2009-06-23 20:57:52.700848216 +0300
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Name=Griffith
-GenericName=Griffith - Movie Collection Manager
+GenericName=Movie Collection Manager
+GenericName[tr]=Film Koleksiyonu Yöneticisi
 Comment=Movie Collection Manager
 Version=1.0
 Exec=griffith
