diff -ur gmailfs-0.7.2.orig/mount.gmailfs gmailfs-0.7.2/mount.gmailfs
--- gmailfs-0.7.2.orig/mount.gmailfs	2006-02-07 00:35:13.000000000 +0200
+++ gmailfs-0.7.2/mount.gmailfs	2006-03-05 23:55:48.000000000 +0200
@@ -18,7 +18,7 @@
 DEFAULT_GMAILFS_LOCATION     = '/usr/share/gmailfs/gmailfs.py'
 # delay (in seconds) after which we consider gmailfs could not mount, and
 # we abort.
-GMAILFS_MOUNTING_MAX_DELAY   = 12 
+GMAILFS_MOUNTING_MAX_DELAY   = 30
 # delay (in seconds) between two consecutive checks for encfs still being
 # mounted.                                
 ENCFS_MOUNTPOINT_CHECK_DELAY = 5  # in seconds
