diff -Nur blockout-old/BlockOut_GL/BlockOut.cpp blockout/BlockOut_GL/BlockOut.cpp
--- blockout-old/BlockOut_GL/BlockOut.cpp	2007-09-13 21:39:35.289926854 +0300
+++ blockout/BlockOut_GL/BlockOut.cpp	2007-09-13 21:40:29.527681462 +0300
@@ -120,6 +120,11 @@
           break;
         case 100: // Exit
           InvalidateDeviceObjects();
+          BOOL fs = theSetup.GetFullScreen();
+          if (fs) {
+            theSetup.SetFullScreen(!fs);
+            UpdateFullScreen();
+          }
           _exit(0);
           break;
       }
