--- uhinv-0.4.orig/src/util/Makefile.in
+++ uhinv-0.4/src/util/Makefile.in
@@ -361,7 +361,7 @@ maintainer-clean-generic clean mostlycle
 
 locations.h: Makefile
 	echo '/* This file is auto generated, do not edit! */' > locations.h
-	echo '#define LOCATION_PKGDATADIR "$(pkgdatadir)"' >> locations.h
+	echo '#define LOCATION_PKGDATADIR "/usr/share/misc"' >> locations.h
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
