--- mcs/class/System.Web.Services/System.Web.Services.Description/BasicProfileChecker.cs.orig	2008-09-02 18:44:46.000000000 +0300
+++ mcs/class/System.Web.Services/System.Web.Services.Description/BasicProfileChecker.cs	2008-11-22 10:27:20.000000000 +0200
@@ -177,7 +177,7 @@
 				foreach (OperationMessage om in op.Messages) {
 					Message msg = ctx.Services.GetMessage (om.Message);
 					foreach (MessagePart part in msg.Parts)
-						parts.Add (part,part);
+ 						parts [part] = part; // do not use Add() - there could be the same MessagePart instance.
 				}
 			}
 			
