Index: libavformat/dv1394.h
===================================================================
--- libavformat/dv1394.h	(revision 5411)
+++ libavformat/dv1394.h	(working copy)
@@ -207,52 +207,34 @@
 #define DV1394_NTSC_FRAME_SIZE (480 * DV1394_NTSC_PACKETS_PER_FRAME)
 #define DV1394_PAL_FRAME_SIZE  (480 * DV1394_PAL_PACKETS_PER_FRAME)
 
+/* Get the driver ready to transmit video.  pass a struct dv1394_init* as
+ * the parameter (see below), or NULL to get default parameters */
+#define DV1394_INIT                     _IOW('#', 0x06, struct dv1394_init)
 
-/* ioctl() commands */
+/* Stop transmitting video and free the ringbuffer */
+#define DV1394_SHUTDOWN         _IO ('#', 0x07)
 
-enum {
-        /* I don't like using 0 as a valid ioctl() */
-        DV1394_INVALID = 0,
+/* Submit N new frames to be transmitted, where the index of the first new
+ * frame is first_clear_buffer, and the index of the last new frame is
+ * (first_clear_buffer + N) % n_frames */
+#define DV1394_SUBMIT_FRAMES    _IO ('#', 0x08)
 
+/* Block until N buffers are clear (pass N as the parameter) Because we
+ * re-transmit the last frame on underrun, there will at most be n_frames
+ * - 1 clear frames at any time */
+#define DV1394_WAIT_FRAMES              _IO ('#', 0x09)
 
-        /* get the driver ready to transmit video.
-           pass a struct dv1394_init* as the parameter (see below),
-           or NULL to get default parameters */
-        DV1394_INIT,
+/* Capture new frames that have been received, where the index of the
+ * first new frame is first_clear_buffer, and the index of the last new
+ * frame is (first_clear_buffer + N) % n_frames */
+#define DV1394_RECEIVE_FRAMES   _IO ('#', 0x0a)
 
+/* Tell card to start receiving DMA */
+#define DV1394_START_RECEIVE    _IO ('#', 0x0b)
 
-        /* stop transmitting video and free the ringbuffer */
-        DV1394_SHUTDOWN,
+/* Pass a struct dv1394_status* as the parameter */
+#define DV1394_GET_STATUS               _IOR('#', 0x0c, struct dv1394_status)
 
-
-        /* submit N new frames to be transmitted, where
-           the index of the first new frame is first_clear_buffer,
-           and the index of the last new frame is
-           (first_clear_buffer + N) % n_frames */
-        DV1394_SUBMIT_FRAMES,
-
-
-        /* block until N buffers are clear (pass N as the parameter)
-           Because we re-transmit the last frame on underrun, there
-           will at most be n_frames - 1 clear frames at any time */
-        DV1394_WAIT_FRAMES,
-
-        /* capture new frames that have been received, where
-           the index of the first new frame is first_clear_buffer,
-           and the index of the last new frame is
-           (first_clear_buffer + N) % n_frames */
-        DV1394_RECEIVE_FRAMES,
-
-
-        DV1394_START_RECEIVE,
-
-
-        /* pass a struct dv1394_status* as the parameter (see below) */
-        DV1394_GET_STATUS,
-};
-
-
-
 enum pal_or_ntsc {
         DV1394_NTSC = 0,
         DV1394_PAL
