diff -Naur libgcrypt-1.4.5.orig/configure.ac libgcrypt-1.4.5/configure.ac
--- libgcrypt-1.4.5.orig/configure.ac	2010-02-05 10:41:42.000000000 +0200
+++ libgcrypt-1.4.5/configure.ac	2010-02-05 10:43:59.000000000 +0200
@@ -1133,7 +1133,6 @@
 src/libgcrypt.pc
 src/libgcrypt-config
 src/versioninfo.rc
-tests/Makefile
 ])
 AC_OUTPUT
 
diff -Naur libgcrypt-1.4.5.orig/Makefile.am libgcrypt-1.4.5/Makefile.am
--- libgcrypt-1.4.5.orig/Makefile.am	2010-02-05 10:41:42.000000000 +0200
+++ libgcrypt-1.4.5/Makefile.am	2010-02-05 10:42:55.000000000 +0200
@@ -22,8 +22,8 @@
 DISTCHECK_CONFIGURE_FLAGS = --enable-random-daemon \
   --enable-ciphers=arcfour:blowfish:cast5:des:aes:twofish:serpent:rfc2268:seed:camellia
 
-DIST_SUBDIRS = m4 mpi cipher random src doc tests
-SUBDIRS = mpi cipher random src doc tests
+DIST_SUBDIRS = m4 mpi cipher random src doc
+SUBDIRS = mpi cipher random src doc
 EXTRA_DIST = autogen.sh README.SVN
 DISTCLEANFILES = 
 
diff -Naur libgcrypt-1.4.5.orig/Makefile.in libgcrypt-1.4.5/Makefile.in
--- libgcrypt-1.4.5.orig/Makefile.in	2010-02-05 10:41:42.000000000 +0200
+++ libgcrypt-1.4.5/Makefile.in	2010-02-05 10:42:32.000000000 +0200
@@ -239,8 +239,8 @@
 DISTCHECK_CONFIGURE_FLAGS = --enable-random-daemon \
   --enable-ciphers=arcfour:blowfish:cast5:des:aes:twofish:serpent:rfc2268:seed:camellia
 
-DIST_SUBDIRS = m4 mpi cipher random src doc tests
-SUBDIRS = mpi cipher random src doc tests
+DIST_SUBDIRS = m4 mpi cipher random src doc
+SUBDIRS = mpi cipher random src doc
 EXTRA_DIST = autogen.sh README.SVN
 DISTCLEANFILES = 
 all: config.h
