Index: browser/components/flock/lucene/src/Makefile.in
===================================================================
--- flock/base/lucene/src/Makefile.in	(revision 13183)
+++ flock/base/lucene/src/Makefile.in	(working copy)
@@ -73,8 +73,8 @@
 include $(topsrcdir)/config/config.mk
 
 ## CLUCENE_DIR=$(topsrcdir)/../../../local
-INCLUDES += -I$(CLUCENE_DIR)/include -I$(CLUCENE_DIR)/lib
-LUCENE_LIB = $(CLUCENE_DIR)/lib/libclucene.$(LIB_SUFFIX)
+INCLUDES += -I/usr/include/CLucene
+LUCENE_LIB=/usr/lib/libclucene.so
 
 # separate libraries linked in.
 EXTRA_DSO_LDOPTS  = $(NSPR_LIBS)
