--- configure.in	2007-05-30 17:42:36.000000000 +0300
+++ configure.in	2007-05-30 17:44:29.000000000 +0300
@@ -2561,7 +2561,7 @@
                   ac_cv_visibility_hidden=no
                   if ${CC-cc} -Werror -S conftest.c -o conftest.s >/dev/null 2>&1; then
                     if egrep '\.(hidden|private_extern).*foo' conftest.s >/dev/null; then
-                      ac_cv_visibility_hidden=yes
+                      ac_cv_visibility_hidden=no
                     fi
                   fi
                   rm -f conftest.[cs]
