diff -Nur et131x-1.2.3-old/et131x_initpci.c et131x-1.2.3/et131x_initpci.c
--- et131x-1.2.3-old/et131x_initpci.c	2007-07-06 18:10:27.279574467 +0300
+++ et131x-1.2.3/et131x_initpci.c	2007-07-06 18:12:07.352361091 +0300
@@ -1672,7 +1672,7 @@
     adapter->task.routine = (void (*)(void *))et131x_isr_handler;
     adapter->task.data = adapter;
 #else
-    INIT_WORK( &adapter->task, et131x_isr_handler );
+    INIT_WORK( &adapter->task, (void (*)(void *))et131x_isr_handler, adapter );
 #endif
 
 
