diff -Nur goonies_r1-0-1-old/src/SFXManager.cpp goonies_r1-0-1/src/SFXManager.cpp
--- goonies_r1-0-1-old/src/SFXManager.cpp	2007-08-25 18:06:33.471637249 +0300
+++ goonies_r1-0-1/src/SFXManager.cpp	2007-08-25 18:14:18.065008321 +0300
@@ -69,7 +69,7 @@
 		char filename[256];
 		Mix_Chunk *t;
 
-		sprintf(filename,"sound/%s",name->get());		
+		sprintf(filename,"/usr/share/goonies/sound/%s",name->get());		
 		t=Sound_create_sound(filename);
 		n=new SFXManagerNode();
 		n->m_sfx=t;
@@ -124,7 +124,7 @@
 	// FIXME: unfortunately the entire game has a hardcoded use of the sound/ dir...
 	char path[256];
 	char *filename;
-	sprintf(path, "sound/%s", localpath);
+	sprintf(path, "/usr/share/goonies/sound/%s", localpath);
 	
 	#ifdef _WIN32
 		WIN32_FIND_DATA finfo;
