--- Makefile.in.old	2008-11-08 00:48:59.000000000 +0200
+++ Makefile.in	2008-11-08 00:49:42.000000000 +0200
@@ -202,7 +202,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = libxml++ docs scripts examples MSVC_Net2005
+SUBDIRS = libxml++ docs scripts MSVC_Net2005
 EXTRA_DIST = libxml++-2.6.pc.in libxml++.spec.in
 DIST_SUBDIRS = $(SUBDIRS) win32_msvc6
 
