Index: rtl8187se-linux-2.6-1023-1118-2008/rtl8185/Makefile
===================================================================
--- rtl8187se-linux-2.6-1023-1118-2008.orig/rtl8185/Makefile
+++ rtl8187se-linux-2.6-1023-1118-2008/rtl8185/Makefile
@@ -33,11 +33,11 @@ EXTRA_CFLAGS += -DENABLE_DOT11D
 
 #EXTRA_CFLAGS += -mhard-float -DCONFIG_FORCE_HARD_FLOAT=y
 
-r8180-objs := r8180_core.o r8180_sa2400.o r8180_93cx6.o r8180_wx.o r8180_max2820.o r8180_gct.o r8180_rtl8225.o r8180_rtl8255.o 		    r8180_rtl8225z2.o r8185b_init.o r8180_dm.o r8180_pm.o
+rtl8187se-objs := r8180_core.o r8180_sa2400.o r8180_93cx6.o r8180_wx.o r8180_max2820.o r8180_gct.o r8180_rtl8225.o r8180_rtl8255.o 		    r8180_rtl8225z2.o r8185b_init.o r8180_dm.o r8180_pm.o
 
-obj-m += r8180.o
+obj-m += rtl8187se.o
 
-MODULES_LST := r8180.ko 
+MODULES_LST := rtl8187se.ko
 
 INSTALL_PREFIX :=
 
