diff -Nur sn9c1xx-1.48-old/sn9c102_core.c sn9c1xx-1.48/sn9c102_core.c
--- sn9c1xx-1.48-old/sn9c102_core.c	2008-04-04 16:21:54.674934005 +0300
+++ sn9c1xx-1.48/sn9c102_core.c	2008-04-04 16:22:20.546933921 +0300
@@ -3299,7 +3299,6 @@
 	strcpy(cam->v4ldev->name, "SN9C1xx PC Camera");
 	cam->v4ldev->owner = THIS_MODULE;
 	cam->v4ldev->type = VID_TYPE_CAPTURE | VID_TYPE_SCALES;
-	cam->v4ldev->hardware = 0;
 	cam->v4ldev->fops = &sn9c102_fops;
 	cam->v4ldev->minor = video_nr[dev_nr];
 	cam->v4ldev->release = video_device_release;
