--- kgraphviewer/src/kgraphviewer.desktop.orig	2008-08-25 14:02:32.000000000 +0300
+++ kgraphviewer/src/kgraphviewer.desktop	2008-08-25 14:08:22.000000000 +0300
@@ -9,6 +9,9 @@
 Icon=kgraphviewer
 Type=Application
 DocPath=kgraphviewer/kgraphviewer.html
+Categories=Qt;KDE;Graphics;
+GenericName[tr]=KDE için .dot uzantılı dosya görüntüleyici
+GenericName[en]=A Graphiz dot-graph viewer for KDE
 Comment=A Graphviz dot graph viewer for KDE
 Comment[bg]=Програма за преглед на Graphviz-графики в KDE
 Comment[ca]=Un visualitzador de gràfic de punts Graphviz pel KDE
@@ -32,6 +35,7 @@
 Comment[sr]=Прегледач Graphviz графике за KDE
 Comment[sr@Latn]=Pregledač Graphviz grafike za KDE
 Comment[sv]=Ett visningsprogram av Graphviz dot för KDE
+Comment[tr]=KDE için .dot uzantılı dosya görüntüleyici
 Comment[uk]=Переглядач графіків Graphviz для KDE
 Comment[vi]=Bộ xem đồ thị chấm Graphviz cho KDE
 Terminal=false
