Index: boot-manager-2.9.11/code/bootmanager/main.py
===================================================================
--- boot-manager-2.9.11.orig/code/bootmanager/main.py
+++ boot-manager-2.9.11/code/bootmanager/main.py
@@ -198,7 +198,7 @@ class MainWidget(QtGui.QWidget, Ui_MainW
         self.comboFilter.addItem(kdecore.i18n("All Items"), QtCore.QVariant("all"))
 
         for name, (label, mandatory, optional) in self.systems.iteritems():
-            self.comboFilter.addItem(label, QtCore.QVariant(name))
+            self.comboFilter.addItem(label, QtCore.QVariant("%s" % name))
 
     def buildMenu(self):
         """
