--- comar/network.py	2007-08-24 16:13:53.000000000 +0300
+++ comar/network.py	2007-11-09 00:52:58.000000000 +0200
@@ -223,7 +223,9 @@
             return None
         for line in f:
             line = line.strip()
-            if line.startswith("DNSSERVERS='"):
+            if line.startswith("DNS="):
+                return line[4:].rstrip('\n').split(',')
+            elif line.startswith("DNSSERVERS='"):
                 return line[12:].rstrip('\n').rstrip("'").split(' ')
 
 
