--- src/kompose.desktop	2005-07-06 01:32:41.000000000 +0300
+++ kompose-yeni.dektop	2006-05-03 00:35:19.000000000 +0300
@@ -1,14 +1,17 @@
 [Desktop Entry]
 Comment=An Apple Exposé like Taskmanager
 Comment[en_US]=An Apple Exposé like Taskmanager
+Comment[tr]=KDE Tam Ekran Görev Yöneticisi
 Encoding=UTF-8
 Exec=kompose
 GenericName=Window Switcher
 GenericName[en_US]=
+GenericName[tr]=Tam Ekran Görev Yöneticisi
 Icon=kompose
 MimeType=
 Name=Komposé
 Name[en_US]=Komposé
+Name[tr]=Kompose
 Path=
 StartupNotify=true
 Terminal=false
@@ -17,3 +20,4 @@
 X-DCOP-ServiceType=
 X-KDE-SubstituteUID=false
 X-KDE-Username=
+Categories=Qt;KDE;Utility;
