--- fatrat-1.2.0_beta1/data/fatrat.desktop.orig	2012-03-01 21:54:01.075679000 +0200
+++ fatrat-1.2.0_beta1/data/fatrat.desktop	2012-03-01 21:54:45.642164000 +0200
@@ -8,6 +8,7 @@
 Comment[sr@ijekavianlatin]=Qt Menadžer preuzimanja/slanja
 Comment[zh_TW]=基於-Qt 下載/上傳 管理員
 Comment[zh_CN]=基于-Qt 下载/上传 管理员
+Comment[tr]=Qt tabanlı indirme/yükleme yöneticisi
 Exec=fatrat
 Name=FatRat
 Name[sr]=ФатРат
@@ -20,6 +21,7 @@
 GenericName[sr@ijekavianlatin]=Menadžer preuzimanja/slanja
 GenericName[zh_TW]=下載/上傳 管理員
 GenericName[zh_CN]=下载/上传 管理员
+GenericName[tr]=İndirme/Yükleme Yöneticisi
 Icon=fatrat
 StartupNotify=true
 Terminal=false
