Index: pardus-2.6/drivers/bluetooth/hci_usb.c
===================================================================
--- pardus-2.6.orig/drivers/bluetooth/hci_usb.c
+++ pardus-2.6/drivers/bluetooth/hci_usb.c
@@ -142,6 +142,9 @@ static struct usb_device_id blacklist_id
 	/* Frontline ComProbe Bluetooth Sniffer */
 	{ USB_DEVICE(0x16d3, 0x0002), .driver_info = HCI_SNIFFER },
 
+	/* Dell laptop with Broadcom chip */
+	{ USB_DEVICE(0x413c, 0x8126), .driver_info = HCI_RESET | HCI_WRONG_SCO_MTU },
+
 	{ }	/* Terminating entry */
 };
 
