Index: pardus-2.6/drivers/media/common/saa7146_video.c
===================================================================
--- pardus-2.6.orig/drivers/media/common/saa7146_video.c
+++ pardus-2.6/drivers/media/common/saa7146_video.c
@@ -1204,19 +1204,16 @@ int saa7146_video_do_ioctl(struct inode 
 		DEB_D(("VIDIOCGMBUF \n"));
 
 		q = &fh->video_q;
-		mutex_lock(&q->lock);
 		err = videobuf_mmap_setup(q,gbuffers,gbufsize,
 					  V4L2_MEMORY_MMAP);
-		if (err < 0) {
-			mutex_unlock(&q->lock);
+		if (err < 0)
 			return err;
-		}
+
 		memset(mbuf,0,sizeof(*mbuf));
 		mbuf->frames = gbuffers;
 		mbuf->size   = gbuffers * gbufsize;
 		for (i = 0; i < gbuffers; i++)
 			mbuf->offsets[i] = i * gbufsize;
-		mutex_unlock(&q->lock);
 		return 0;
 	}
 #endif
