Index: deadbeef-0.5.1/plugins/ffap/Makefile.am
===================================================================
--- deadbeef-0.5.1.orig/plugins/ffap/Makefile.am
+++ deadbeef-0.5.1/plugins/ffap/Makefile.am
@@ -2,7 +2,7 @@ if HAVE_FFAP
 ffapdir = $(libdir)/$(PACKAGE)
 pkglib_LTLIBRARIES = ffap.la
 ffap_la_SOURCES = ffap.c
-ffap_la_LDFLAGS = -module
+ffap_la_LDFLAGS = -module -lm
 
 AM_CFLAGS = $(CFLAGS) -std=c99
 endif
Index: deadbeef-0.5.1/plugins/m3u/Makefile.am
===================================================================
--- deadbeef-0.5.1.orig/plugins/m3u/Makefile.am
+++ deadbeef-0.5.1/plugins/m3u/Makefile.am
@@ -5,7 +5,7 @@ m3u_la_SOURCES = m3u.c
 
 m3u_la_LDFLAGS = -module
 
-m3u_la_LIBADD = $(LIBADD)
+m3u_la_LIBADD = $(LIBADD) -lm
 
 m3u_la_CFLAGS = $(CFLAGS) -std=c99
 endif
