diff -Naur gwenview-orig/lib/mimetypeutils.cpp gwenview/lib/mimetypeutils.cpp
--- gwenview-orig/lib/mimetypeutils.cpp	2009-05-29 00:21:13.132804043 +0300
+++ gwenview/lib/mimetypeutils.cpp	2009-05-29 00:22:29.391556049 +0300
@@ -84,7 +84,7 @@
 	static QStringList list;
 	if (list.isEmpty()) {
 		list.append("image/svg+xml");
-		// FIXME svgz
+		list.append("image/svg+xml-compressed");
 		resolveAliasInList(&list);
 	}
 	return list;
diff -Naur gwenview-orig/app/gwenview.desktop gwenview/app/gwenview.desktop
--- gwenview-orig/app/gwenview.desktop	2009-05-29 00:29:56.341804025 +0300
+++ gwenview/app/gwenview.desktop	2009-05-29 00:31:13.942804133 +0300
@@ -119,6 +119,6 @@
 Type=Application
 Categories=Qt;KDE;Graphics;
 # TODO: what is image/x-krl? Please register this mimetype at freedesktop.org if you need it
-MimeType=image/gif;image/jpeg;image/png;image/bmp;image/x-eps;image/x-ico;image/x-portable-bitmap;image/x-portable-pixmap;image/x-xbitmap;image/x-xpixmap;
+MimeType=image/gif;image/jpeg;image/png;image/bmp;image/x-eps;image/x-ico;image/x-portable-bitmap;image/x-portable-pixmap;image/x-xbitmap;image/x-xpixmap;image/svg+xml;image/svg+xml-compressed;
 X-DocPath=gwenview/index.html
 InitialPreference=12
