diff -Naur gwenview-orig/app/gwenviewui.rc gwenview/app/gwenviewui.rc
--- gwenview-orig/app/gwenviewui.rc	2009-05-29 00:36:10.776553784 +0300
+++ gwenview/app/gwenviewui.rc	2009-05-29 00:51:14.879554423 +0300
@@ -122,12 +122,18 @@
 	<Action name="view" />
 	<Action name="fullscreen" />
 	<Separator/>
+    <Action name="go_up"/>
 	<Action name="go_previous" />
 	<Action name="go_next" />
 	<Separator/>
 	<Action name="rotate_left" />
 	<Action name="rotate_right" />
 	<Separator/>
+    <Action name="view_actual_size"/>
+    <Action name="view_zoom_in"/>
+    <Action name="view_zoom_out"/>
+    <Action name="view_zoom_to_fit"/>
+    <Separator/>
 	<Merge/>
 	<Separator/>
 	<Action name="toggle_sidebar" />
