--- bluez-utils-3.7/scripts/bluetooth.default.org	2004-05-08 01:04:27.000000000 +0300
+++ bluez-utils-3.7/scripts/bluetooth.default	    2006-12-12 12:36:39.000000000 +0200
@@ -9,14 +9,17 @@
 # Start sdpd (allowed values are "true" and "false")
 SDPD_ENABLE=true
 
+# Bluetooth device discoverable or not
+DISCOVERABLE=false
+
 # Start hidd (allowed values are "true" and "false")
-HIDD_ENABLE=true
+HIDD_ENABLE=false
 
 # Arguments to hidd
 HIDD_OPTIONS=""
 
 # Run hid2hci (allowed values are "true" and "false")
-HID2HCI_ENABLE=true
+HID2HCI_ENABLE=false
 
 # Bind rfcomm devices (allowed values are "true" and "false")
 RFCOMM_ENABLE=true
