Index: pardus/localedata.py
===================================================================
--- pardus/localedata.py	(revision 19260)
+++ pardus/localedata.py	(revision 19269)
@@ -58,6 +58,12 @@
         locale              = "en_US.UTF-8"
     ),
 
+    "en_GB":    Language(
+        name                = _("English GB"),
+        locale              = "en_GB.UTF-8",
+        keymaps             = [Keymap("uk", "gb")]
+    ),
+
     "af":   Language(
         name                = _("Afrikaans"),
         locale              = "af_ZA.UTF-8",
@@ -82,12 +88,6 @@
         keymaps             = [Keymap("bg")]
     ),
 
-    "br":   Language(
-        name                = _("Brazilian"),
-        locale              = "pt_BR.UTF-8",
-        keymaps             = [Keymap("br-abnt2", "br")]
-    ),
-
     "ca":   Language(
         name                = _("Catalan"),
         locale              = "ca_ES.UTF-8",
@@ -214,6 +214,12 @@
         keymaps             = [Keymap("pt-latin1", "pt")]
     ),
 
+    "pt_BR":   Language(
+        name                = _("Brazilian"),
+        locale              = "pt_BR.UTF-8",
+        keymaps             = [Keymap("br-abnt2", "br")]
+    ),
+
     "ru":   Language(
         name                = _("Russian"),
         locale              = "ru_RU.UTF-8",
