--- glibc-2.7.orig/ports/sysdeps/unix/sysv/linux/arm/bits/stdio-lock.h	2010-03-08 15:09:08.307075144 +0200
+++ glibc-2.7/ports/sysdeps/unix/sysv/linux/arm/bits/stdio-lock.h	2010-03-08 15:09:49.394076359 +0200
@@ -50,6 +50,8 @@
   _IO_cleanup_region_start ((void (*) (void *)) _IO_funlockfile, (_fp));      \
   _IO_flockfile (_fp)
 
+# define _IO_acquire_lock_clear_flags2(_fp) _IO_acquire_lock (_fp)
+
 # define _IO_release_lock(_fp) \
   _IO_funlockfile (_fp);						      \
   _IO_cleanup_region_end (0)
