Index: caps/src/nsScriptSecurityManager.cpp
===================================================================
RCS file: /cvsroot/mozilla/caps/src/nsScriptSecurityManager.cpp,v
retrieving revision 1.266.2.16
diff -u -p -d -8 -r1.266.2.16 nsScriptSecurityManager.cpp
--- caps/src/nsScriptSecurityManager.cpp	20 Apr 2007 20:41:31 -0000	1.266.2.16
+++ caps/src/nsScriptSecurityManager.cpp	9 Jul 2007 01:21:27 -0000
@@ -1355,17 +1354,18 @@ nsScriptSecurityManager::CheckLoadURIWit
         { "aim",             AllowProtocol  },
         { "data",            AllowProtocol  },
         { "keyword",         DenyProtocol   },
         { "resource",        ChromeProtocol },
         { "gopher",          AllowProtocol  },
         { "datetime",        DenyProtocol   },
         { "finger",          AllowProtocol  },
         { "res",             DenyProtocol   },
-        { "x-jsd",           ChromeProtocol }
+        { "x-jsd",           ChromeProtocol },
+        { "wyciwyg",         DenyProtocol   }
     };
 
     NS_NAMED_LITERAL_STRING(errorTag, "CheckLoadURIError");
     for (unsigned i=0; i < sizeof(protocolList)/sizeof(protocolList[0]); i++)
     {
         if (targetScheme.LowerCaseEqualsASCII(protocolList[i].name))
         {
             switch (protocolList[i].action)
