diff -Naur i686.orig/src/wl/sys/wl_linux.c i686/src/wl/sys/wl_linux.c
--- i686.orig/src/wl/sys/wl_linux.c	2012-02-15 10:14:23.664000006 +0200
+++ i686/src/wl/sys/wl_linux.c	2012-02-15 10:23:52.221000007 +0200
@@ -385,7 +385,8 @@
 #endif
 	.ndo_get_stats = wl_get_stats,
 	.ndo_set_mac_address = wl_set_mac_address,
-	.ndo_set_multicast_list = wl_set_multicast_list,
+	//.ndo_set_multicast_list = wl_set_multicast_list,
+    .ndo_set_rx_mode = wl_set_multicast_list,
 	.ndo_do_ioctl = wl_ioctl
 };
 
