--- manager/AuthDialog.cpp.orig	2009-02-12 22:48:59.000000000 +0200
+++ manager/AuthDialog.cpp	2009-02-12 22:56:17.000000000 +0200
@@ -39,7 +39,7 @@
     setButtons(Ok | Cancel | Details);
 
     kDebug() << "Getting action message...";
-    QString actionMessage = QString::fromLocal8Bit(polkit_policy_file_entry_get_action_message(entry));
+    QString actionMessage = QString::fromUtf8(polkit_policy_file_entry_get_action_message(entry));
     if (actionMessage.isEmpty()) {
         kWarning() << "Could not get action message for action.";
         lblHeader->hide();
@@ -298,7 +298,7 @@
 
     app_label->setText(appname);
 
-    action_label->setText(polkit_policy_file_entry_get_action_description(entry));
+    action_label->setText(QString::fromUtf8(polkit_policy_file_entry_get_action_description(entry)));
     QString actionId = polkit_policy_file_entry_get_id(entry);
     action_label->setTipText(i18n("Click to edit %1", actionId));
     action_label->setUrl(actionId);
diff -Naur authorization-orig/pkitem.cpp authorization/pkitem.cpp
--- authorization-orig/pkitem.cpp	2009-02-12 22:57:32.000000000 +0200
+++ authorization/pkitem.cpp	2009-02-12 22:59:21.000000000 +0200
@@ -49,7 +49,7 @@
 {
     Q_UNUSED(column);
     if (datarole == Qt::DisplayRole) {
-        return polkit_policy_file_entry_get_action_description(m_entry);
+        return QString::fromUtf8(polkit_policy_file_entry_get_action_description(m_entry));
     } else if (datarole == Qt::DecorationRole) {
         return m_icon;
     } else if (datarole == PolkitKde::AuthorizationsModel::IdRole) {
diff -Naur authorization-orig/PkKAction.cpp authorization/PkKAction.cpp
--- authorization-orig/PkKAction.cpp	2009-02-12 22:57:32.000000000 +0200
+++ authorization/PkKAction.cpp	2009-02-12 22:58:54.000000000 +0200
@@ -112,7 +112,7 @@
     revokePB->setEnabled(false);
     //Action description
     QString actionId    = polkit_policy_file_entry_get_id(m_pfe);
-    QString description = polkit_policy_file_entry_get_action_description(m_pfe);
+    QString description = QString::fromUtf8(polkit_policy_file_entry_get_action_description(m_pfe));
     QString iconName    = polkit_policy_file_entry_get_action_icon_name(m_pfe);
     QString vendor      = polkit_policy_file_entry_get_action_vendor(m_pfe);
     QString vendorUrl   = polkit_policy_file_entry_get_action_vendor_url(m_pfe);
