Index: kffmpegthumbnailer-1.1.0/kffmpegthumbnailer.desktop
===================================================================
--- kffmpegthumbnailer-1.1.0.orig/kffmpegthumbnailer.desktop
+++ kffmpegthumbnailer-1.1.0/kffmpegthumbnailer.desktop
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Type=Service
 Name=Video Files (kffmpegthumbnailer)
+Name[tr]=Video Dosyaları (kffmpegthumbnailer)
 X-KDE-ServiceTypes=ThumbCreator
 MimeType=video/*;application/x-flash-video;application/vnd.ms-asf;application/vnd.rn-realmedia;
 X-KDE-Library=kffmpegthumbnailer
