Index: tasma/modules/tasmatv/CMakeLists.txt
===================================================================
--- tasma/modules/tasmatv/CMakeLists.txt	(revision 12667)
+++ tasma/modules/tasmatv/CMakeLists.txt	(working copy)
@@ -14,7 +14,7 @@
 KDE3_ADD_UI_FILES(tasmatv_sources tvconfigui.ui)
 
 KDE3_ADD_KPART(kcm_tasmatv ${tasmatv_sources})
-TARGET_LINK_LIBRARIES(kcm_tasmatv ${QT_AND_KDECORE_LIBS} kdeui pci)
+TARGET_LINK_LIBRARIES(kcm_tasmatv ${QT_QT_LIBRARY} kdecore kdeui pci)
 
 INSTALL_TARGETS(/lib/kde3 kcm_tasmatv)
 INSTALL_FILES(/share/applications/kde/ tasmatv.desktop)
