diff -Naur vacuum.orig/src/packages/linux/vacuum.desktop vacuum/src/packages/linux/vacuum.desktop
--- vacuum.orig/src/packages/linux/vacuum.desktop	2010-01-29 00:21:44.040078667 +0200
+++ vacuum/src/packages/linux/vacuum.desktop	2010-01-29 00:37:32.360831097 +0200
@@ -2,7 +2,9 @@
 Encoding=UTF-8
 Name=Vacuum IM
 GenericName=Jabber Client
+GenericName[tr]=Jabber İstemcisi
 Comment=Communicate over the Jabber network
+Comment[tr]=Jabber ağı üzerinden haberleşin
 Exec=vacuum
 Icon=vacuum
 StartupNotify=true
