Index: xterm-241/charproc.c
===================================================================
--- xterm-241.orig/charproc.c
+++ xterm-241/charproc.c
@@ -389,10 +389,10 @@ static XtActionsRec actionsList[] = {
 static XtResource resources[] =
 {
     Bres(XtNallowSendEvents, XtCAllowSendEvents, screen.allowSendEvent0, False),
-    Bres(XtNallowFontOps, XtCAllowFontOps, screen.allowFontOp0, True),
+    Bres(XtNallowFontOps, XtCAllowFontOps, screen.allowFontOp0, False),
     Bres(XtNallowTcapOps, XtCAllowTcapOps, screen.allowTcapOp0, False),
     Bres(XtNallowTitleOps, XtCAllowTitleOps, screen.allowTitleOp0, True),
-    Bres(XtNallowWindowOps, XtCAllowWindowOps, screen.allowWindowOp0, True),
+    Bres(XtNallowWindowOps, XtCAllowWindowOps, screen.allowWindowOp0, False),
     Bres(XtNaltIsNotMeta, XtCAltIsNotMeta, screen.alt_is_not_meta, False),
     Bres(XtNaltSendsEscape, XtCAltSendsEscape, screen.alt_sends_esc, False),
     Bres(XtNalwaysBoldMode, XtCAlwaysBoldMode, screen.always_bold_mode, False),
Index: xterm-241/xterm.man
===================================================================
--- xterm-241.orig/xterm.man
+++ xterm-241/xterm.man
@@ -1472,7 +1472,7 @@ The default is ``false.''
 .TP
 .B "allowFontOps (\fPclass\fB AllowFontOps)"
 Specifies whether control sequences that set/query the font should be allowed.
-The default is ``true.''
+The default is ``false.''
 .TP 8
 .B "allowSendEvents (\fPclass\fB AllowSendEvents)"
 Specifies whether or not synthetic key and button events (generated using
@@ -1514,7 +1514,7 @@ The default is ``true.''
 .B "allowWindowOps (\fPclass\fB AllowWindowOps)"
 Specifies whether extended window control sequences (as used in dtterm)
 should be allowed.
-The default is ``true.''
+The default is ``false.''
 .TP 8
 .B "altIsNotMeta (\fPclass\fB AltIsNotMeta\fP)"
 If ``true'', treat the Alt-key as if it were the Meta-key.
