--- src/modules/graphic/ssggraph/Makefile.orig	2010-01-15 13:17:05.324507492 +0200
+++ src/modules/graphic/ssggraph/Makefile	2010-01-15 13:17:15.983271836 +0200
@@ -24,7 +24,7 @@
 
 MODULEDIR = modules/graphic
 
-LIBS      = -lplibsl -lplibsm -lplibsg -lplibssg -lplibul -lplibssgaux -lopenal
+LIBS      = -lplibsl -lplibsm -lplibsg -lplibssg -lplibul -lplibssgaux -lopenal -lalut
 
 SHIPDIR   = config
 
