--- Make-config.in.orig	2011-06-06 12:12:12.034000021 +0300
+++ Make-config.in	2011-06-06 12:13:08.975000025 +0300
@@ -22,14 +22,14 @@
 srcdir = @srcdir@
 top_srcdir = ${SPEED_DREAMS_BASE}
 VPATH = @srcdir@
-prefix = @prefix@
-exec_prefix = @exec_prefix@
+prefix = /usr
+exec_prefix = /usr
 
 bindir = @bindir@
 libdir = @libdir@/speed-dreams
 datarootdir = @datarootdir@
-datadir = @datarootdir@/games/speed-dreams
-vardir = @localstatedir@/games/speed-dreams
+datadir = @datarootdir@/speed-dreams
+vardir = @localstatedir@/speed-dreams
 
 top_builddir = .
 
