--- kvirc-20090511/cmake/kvirc4.desktop.cmake.orig	2009-07-14 12:02:08.000000000 +0300
+++ kvirc-20090511/cmake/kvirc4.desktop.cmake	2009-07-14 12:02:43.000000000 +0300
@@ -1,9 +1,12 @@
 [Desktop Entry]
 Name=KVIrc 4
 GenericName=IRC Client
+GenericName[tr]=IRC İstemcisi
 Comment=Connect to Internet Relay Chat
+Comment[tr]=IRC'e bağlanın
 Exec=${KVIRC_BINARYNAME} -m %u
 Icon=kvirc
 Type=Application
 MimeType=application/x-kva;application/x-kvt;
 Categories=Qt;Network;IRCClient;
+
