diff -urN noip-2.1.9.patched/Makefile noip-2.1.9/Makefile
--- noip-2.1.9.patched/Makefile	2008-11-25 16:58:32.000000000 +0000
+++ noip-2.1.9/Makefile	2008-11-21 22:27:15.000000000 +0000
@@ -22,7 +22,7 @@
 # ARCH=sun
 
 ${TGT}: Makefile ${TGT}.c 
-	${CC} -Wall -g -D${ARCH} -DPREFIX=\"${PREFIX}\" ${TGT}.c -o ${TGT} ${LIBS}
+	${CC} ${CFLAGS} -D${ARCH} -DPREFIX=\"${PREFIX}\" ${TGT}.c -o ${TGT} ${LIBS}
 
 install: ${TGT} 
 	if [ ! -d ${BINDIR} ]; then mkdir -p ${BINDIR};fi
