Index: libsdl1.2-1.2.13/build-scripts/strip_fPIC.sh
===================================================================
--- libsdl1.2-1.2.13.orig/build-scripts/strip_fPIC.sh	2007-12-31 05:47:51.000000000 +0100
+++ libsdl1.2-1.2.13/build-scripts/strip_fPIC.sh	2008-01-02 03:19:59.000000000 +0100
@@ -5,8 +5,8 @@
 command=""
 while [ $# -gt 0 ]; do
     case "$1" in
-        -?PIC)
-            # Ignore -fPIC and -DPIC options
+        -fPIC)
+            # Ignore -fPIC option
             ;;
         -fno-common)
             # Ignore -fPIC and -DPIC options
