--- ode-0.8/Makefile.am.install	2007-02-07 22:15:35.000000000 +0100
+++ ode-0.8/Makefile.am	2007-02-14 15:17:36.000000000 +0100
@@ -25,13 +25,12 @@
 
 if USE_SONAME
 install-exec-hook:
-	ln -s $(libdir)/@ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
-	$(libdir)/libode.so
-	ln -s $(libdir)/@ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
-	$(libdir)/@ODE_SONAME@
-	ln -s $(libdir)/@ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
-	$(libdir)/@ODE_SONAME@.@ODE_REVISION@
-	/sbin/ldconfig
+	ln -s @ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
+	$(DESTDIR)/$(libdir)/libode.so
+	ln -s @ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
+	$(DESTDIR)/$(libdir)/@ODE_SONAME@
+	ln -s @ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
+	$(DESTDIR)/$(libdir)/@ODE_SONAME@.@ODE_REVISION@
 else
 install-exec-hook:
 endif
--- ode-0.8/Makefile.in.install	2007-02-11 16:40:20.000000000 +0100
+++ ode-0.8/Makefile.in	2007-02-14 15:17:43.000000000 +0100
@@ -660,13 +660,12 @@
 	rm -rf ${PACKAGE}-${VERSION}
 
 @USE_SONAME_TRUE@install-exec-hook:
-@USE_SONAME_TRUE@	ln -s $(libdir)/@ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
-@USE_SONAME_TRUE@	$(libdir)/libode.so
-@USE_SONAME_TRUE@	ln -s $(libdir)/@ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
-@USE_SONAME_TRUE@	$(libdir)/@ODE_SONAME@
-@USE_SONAME_TRUE@	ln -s $(libdir)/@ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
-@USE_SONAME_TRUE@	$(libdir)/@ODE_SONAME@.@ODE_REVISION@
-@USE_SONAME_TRUE@	/sbin/ldconfig
+@USE_SONAME_TRUE@	ln -s @ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
+@USE_SONAME_TRUE@	$(DESTDIR)/$(libdir)/libode.so
+@USE_SONAME_TRUE@	ln -s @ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
+@USE_SONAME_TRUE@	$(DESTDIR)/$(libdir)/@ODE_SONAME@
+@USE_SONAME_TRUE@	ln -s @ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
+@USE_SONAME_TRUE@	$(DESTDIR)/$(libdir)/@ODE_SONAME@.@ODE_REVISION@
 @USE_SONAME_FALSE@install-exec-hook:
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
