Index: xf86-video-ati-6.12.2/src/radeon_dri.c
===================================================================
--- xf86-video-ati-6.12.2.orig/src/radeon_dri.c
+++ xf86-video-ati-6.12.2/src/radeon_dri.c
@@ -1607,7 +1607,7 @@ Bool RADEONDRIGetVersion(ScrnInfoPtr pSc
 	goto fail;
     }
 
-    if (info->dri->pKernelDRMVersion->version_minor >= 30) {
+    if (info->dri->pKernelDRMVersion->version_minor >= 31) {
 	    struct drm_radeon_gem_info mminfo;
 
 	    if (!drmCommandWriteRead(fd, DRM_RADEON_GEM_INFO, &mminfo, sizeof(mminfo)))
