diff -Naur audiokonverter-5.8.1.orig/audiofrommovie4.desktop audiokonverter-5.8.1/audiofrommovie4.desktop
--- audiokonverter-5.8.1.orig/audiofrommovie4.desktop	2009-06-19 22:54:13.000000000 +0300
+++ audiokonverter-5.8.1/audiofrommovie4.desktop	2009-06-19 23:05:03.000000000 +0300
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Service
 ServiceTypes=KonqPopupMenu/Plugin
-MimeType=video/avi;video/mp4;video/mpeg;video/msvideo;video/ogg;video/quicktime;video/theora;video/vnd.rn-realvideo;application/vnd.rn-realvideo;video/x-matroska;video/x-mpeg;video/x-mpeg2;video/x-ms-asf;video/x-msvideo;video/x-ms-wmv;video/x-ogg;video/x-ogm;video/x-quicktime;video/x-theora;video/x-avi;video/x-flv
+MimeType=video/avi;video/mp4;video/mpeg;video/msvideo;video/ogg;video/quicktime;video/theora;video/vnd.rn-realvideo;application/vnd.rn-realvideo;video/x-matroska;video/x-mpeg;video/x-mpeg2;video/x-ms-asf;video/x-msvideo;video/x-ms-wmv;video/x-ogg;video/x-ogm;video/x-quicktime;video/x-theora;video/x-avi;video/x-flv;
 Actions=show;dump;mp3;ogg;flac;ipod;wav
 X-KDE-Submenu=extract audio to
 X-KDE-Submenu[de]=Tonspur umwandeln in
diff -Naur audiokonverter-5.8.1.orig/x-flv.desktop audiokonverter-5.8.1/x-flv.desktop
--- audiokonverter-5.8.1.orig/x-flv.desktop	2009-06-19 22:54:13.000000000 +0300
+++ audiokonverter-5.8.1/x-flv.desktop	2009-06-19 23:05:33.000000000 +0300
@@ -5,4 +5,4 @@
 Icon=video
 MimeType=video/x-flv
 Patterns=*.flv;*.FLV;
-Type=MimeType
+Type=MimeType;
diff -Naur audiokonverter-5.8.1.orig/x-wavpack.desktop audiokonverter-5.8.1/x-wavpack.desktop
--- audiokonverter-5.8.1.orig/x-wavpack.desktop	2009-06-19 22:54:13.000000000 +0300
+++ audiokonverter-5.8.1/x-wavpack.desktop	2009-06-19 23:05:55.000000000 +0300
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Type=MimeType
+Type=MimeType;
 MimeType=audio/x-wavpack
 Icon=sound
 Patterns=*.wv;*.WV;
