diff -Nur alsa-tools-1.0.16-old/usx2yloader/Makefile.am alsa-tools-1.0.16/usx2yloader/Makefile.am
--- alsa-tools-1.0.16-old/usx2yloader/Makefile.am	2008-04-08 16:39:28.000000000 +0300
+++ alsa-tools-1.0.16/usx2yloader/Makefile.am	2008-04-08 16:42:37.000000000 +0300
@@ -11,8 +11,8 @@
 
 EXTRA_DIST = depcomp tascam_fw.usermap tascam_fw.in tascam_fpga.in
 
-hotplug_files = tascam_fw.usermap tascam_fw tascam_fpga
-hotplugdir = /etc/hotplug/usb
+hotplug_files = tascam_fw tascam_fpga
+hotplugdir = /lib/udev
 hotplug_SCRIPTS = $(hotplug_files)
 
 
diff -Nur alsa-tools-1.0.16-old/usx2yloader/Makefile.in alsa-tools-1.0.16/usx2yloader/Makefile.in
--- alsa-tools-1.0.16-old/usx2yloader/Makefile.in	2008-04-08 16:39:28.000000000 +0300
+++ alsa-tools-1.0.16/usx2yloader/Makefile.in	2008-04-08 16:42:37.000000000 +0300
@@ -166,8 +166,8 @@
 AM_CFLAGS = -DDATAPATH=\"$(datadir)/alsa/firmware/$(MYNAME)\"
 usx2yloader_SOURCES = usx2yloader.c
 EXTRA_DIST = depcomp tascam_fw.usermap tascam_fw.in tascam_fpga.in
-hotplug_files = tascam_fw.usermap tascam_fw tascam_fpga
-hotplugdir = /etc/hotplug/usb
+hotplug_files = tascam_fw tascam_fpga
+hotplugdir = /lib/udev
 hotplug_SCRIPTS = $(hotplug_files)
 all: all-am
 
