diff -Nur lirc-0.8.3pre1-old/drivers/lirc_gpio/Makefile.am lirc-0.8.3pre1/drivers/lirc_gpio/Makefile.am
--- lirc-0.8.3pre1-old/drivers/lirc_gpio/Makefile.am	2008-03-22 22:58:37.000000000 +0200
+++ lirc-0.8.3pre1/drivers/lirc_gpio/Makefile.am	2008-03-22 23:00:08.000000000 +0200
@@ -8,6 +8,6 @@
 automake_dummy_SOURCES = lirc_gpio.c
 
 ## there is no *just* object file support in automake.  This is close enough
-module_DATA = lirc_gpio.o
+module_DATA = 
 
 include ../Makefile.common
