Index: mudur-2.1.1/bin/mudur.py
===================================================================
--- mudur-2.1.1.orig/bin/mudur.py
+++ mudur-2.1.1/bin/mudur.py
@@ -782,12 +782,12 @@ def setupUdev():
     devpath = "/lib/udev/devices"
     if os.path.exists(devpath):
         ui.info(_("Restoring saved device states"))
-        run_quiet(
-            "/bin/cp",
-            "--preserve=all", "--recursive",
-            "--update", "--no-dereference",
-            "%s/*" % (devpath), "/dev/"
-        )
+        for name in os.listdir(devpath):
+            run_quiet("/bin/cp",
+                      "--preserve=all",
+                      "--recursive",
+                      "--update",
+                      "%s/%s" % (devpath, name), "/dev/")
 
     # When these files are missing, lots of trouble happens
     # so we double check their existence
