Index: bin/mudur.py
===================================================================
--- bin/mudur.py	(revision 22072)
+++ bin/mudur.py	(revision 22073)
@@ -767,11 +767,6 @@
             ui.warn(_("Can't move persistent udev rules from /dev/.udev"))
             pass
 
-    # If any persistent rules exist in /etc/udev/rules.d, trigger udev
-    # for processing them.
-    if glob.glob("/etc/udev/rules.d/70-persistent-*"):
-        run("/sbin/udevadm", "trigger")
-
 def setupUdev():
 
     ui.info(_("Mounting /dev"))
