Index: bin/mudur.py
===================================================================
--- bin/mudur.py	(revision 21810)
+++ bin/mudur.py	(revision 21811)
@@ -952,8 +952,10 @@
         splash.verbose()
         ui.info(_("A full fsck has been forced"))
         t = run_full("/sbin/fsck", "-C", "-R", "-A", "-a", "-f")
-        # remove forcefsck file
-        os.unlink("/forcefsck")
+
+        # remove forcefsck file if it exists
+        if os.path.exists("/forcefsck"):
+            os.unlink("/forcefsck")
     else:
         t = run_full("/sbin/fsck", "-C", "-T", "-R", "-A", "-a")
 
