diff -Naur ember-0.5.0.orig/ember.in ember-0.5.0/ember.in
--- ember-0.5.0.orig/ember.in	2007-06-26 00:45:11.000000000 +0300
+++ ember-0.5.0/ember.in	2007-09-26 15:56:03.000000000 +0300
@@ -54,7 +54,7 @@
 bindir=${exec_prefix}/bin
 datadir=${prefix}/share/ember
 media_user_dir=${datadir}/media/user
-etcdir=${prefix}/etc/ember
+etcdir=/etc/ember
 
 LD_LIBRARY=$prefix/lib/ember:$LD_LIBRARY
 LD_LIBRARY_PATH=$prefix/lib/ember:$LD_LIBRARY_PATH
diff -Naur ember-0.5.0.orig/src/services/config/ConfigService.cpp ember-0.5.0/src/services/config/ConfigService.cpp
--- ember-0.5.0.orig/src/services/config/ConfigService.cpp	2007-06-18 01:09:06.000000000 +0300
+++ ember-0.5.0/src/services/config/ConfigService.cpp	2007-09-26 15:54:49.000000000 +0300
@@ -68,7 +68,7 @@
 	void ConfigService::setPrefix(const std::string& prefix)
 	{
 		mSharedDataDir = prefix + "/share/ember/";
-		mEtcDir = prefix + "/etc/ember/";
+		mEtcDir = "/etc/ember/";
 	}
 	
 	void ConfigService::setHomeDirectory(const std::string& path)
