--- src/services/config/ConfigService.cpp~	2008-04-13 13:14:07.000000000 +0300
+++ src/services/config/ConfigService.cpp	2008-06-05 21:13:55.000000000 +0300
@@ -26,6 +26,8 @@
 #include "services/logging/LoggingService.h"
 // #include "framework/binreloc.h"
 
+#include <cstdlib>
+
 #include <iostream>
 #include "framework/ConsoleBackend.h"
 #include "framework/Tokeniser.h"
--- src/services/logging/LoggingService.h~	2008-04-13 13:14:07.000000000 +0300
+++ src/services/logging/LoggingService.h	2008-06-05 21:15:57.000000000 +0300
@@ -29,6 +29,7 @@
 #include <ctime>
 #include <stdarg.h>
 
+#include <memory>
 
 namespace Ember
 {
--- src/services/sound/SoundService.cpp~	2008-04-13 13:14:07.000000000 +0300
+++ src/services/sound/SoundService.cpp	2008-06-05 21:17:51.000000000 +0300
@@ -47,6 +47,8 @@
 */
 #include <sstream>
 
+#include <cstring>
+
 #include "SoundService.h"
 
 
--- src/services/wfut/WfutService.h~	2008-04-13 13:14:07.000000000 +0300
+++ src/services/wfut/WfutService.h	2008-06-05 21:19:19.000000000 +0300
@@ -28,6 +28,8 @@
 #include <sigc++/object.h>
 #include <sigc++/signal.h>
 
+#include <memory>
+
 namespace Ember {
 
 
--- src/components/ogre/environment/caelum/include/LayeredClouds.h~	2008-04-15 01:45:07.000000000 +0300
+++ src/components/ogre/environment/caelum/include/LayeredClouds.h	2008-06-05 21:24:16.000000000 +0300
@@ -24,6 +24,8 @@
 #include "CaelumPrerequisites.h"
 #include "CameraBoundElement.h"
 
+#include <memory>
+
 namespace caelum {
 
 /** Class for layered clouds.
--- src/components/ogre/environment/pagedgeometry/include/ImpostorPage.h~	2008-04-13 13:14:06.000000000 +0300
+++ src/components/ogre/environment/pagedgeometry/include/ImpostorPage.h	2008-06-05 21:27:43.000000000 +0300
@@ -23,6 +23,8 @@
 #include <OgreTextureManager.h>
 #include <OgreRenderTexture.h>
 
+#include <memory>
+
 
 #define IMPOSTOR_YAW_ANGLES 8
 #define IMPOSTOR_PITCH_ANGLES 4
--- src/components/ogre/input/InputCommandMapper.cpp~	2008-04-13 13:14:06.000000000 +0300
+++ src/components/ogre/input/InputCommandMapper.cpp	2008-06-05 21:34:42.000000000 +0300
@@ -31,6 +31,8 @@
 #include "services/config/ConfigService.h"
 #include "services/logging/LoggingService.h"
 
+#include <algorithm>
+
 namespace EmberOgre {
 
 
--- src/components/ogre/model/mapping/Cases/AttributeComparers/NumericComparerWrapper.h~	2008-04-13 13:14:07.000000000 +0300
+++ src/components/ogre/model/mapping/Cases/AttributeComparers/NumericComparerWrapper.h	2008-06-05 21:38:38.000000000 +0300
@@ -25,6 +25,8 @@
 
 #include "AttributeComparerWrapper.h"
 
+#include <memory>
+
 namespace EmberOgre {
 
 namespace Model {
--- src/components/ogre/model/mapping/Cases/AttributeComparers/HeightComparerWrapper.h~	2008-04-13 13:14:07.000000000 +0300
+++ src/components/ogre/model/mapping/Cases/AttributeComparers/HeightComparerWrapper.h	2008-06-05 21:41:50.000000000 +0300
@@ -25,6 +25,8 @@
 
 #include "AttributeComparerWrapper.h"
 
+#include <memory>
+
 namespace Eris
 {
 class Entity;
--- src/components/ogre/model/mapping/Cases/AttributeComparers/StringComparerWrapper.h~	2008-04-13 13:14:07.000000000 +0300
+++ src/components/ogre/model/mapping/Cases/AttributeComparers/StringComparerWrapper.h	2008-06-05 21:45:12.000000000 +0300
@@ -27,6 +27,8 @@
 #include "StringComparer.h"
 #include "AttributeComparerWrapper.h"
 
+#include <memory>
+
 namespace EmberOgre {
 
 namespace Model {
--- src/components/ogre/widgets/adapters/atlas/NumberAdapter.cpp~	2008-04-13 13:14:07.000000000 +0300
+++ src/components/ogre/widgets/adapters/atlas/NumberAdapter.cpp	2008-06-05 21:52:03.000000000 +0300
@@ -27,6 +27,8 @@
 #include "NumberAdapter.h"
 #include "../../ColouredListItem.h"
 
+#include <cstdlib>
+
 namespace EmberOgre {
 
 namespace Gui {
--- src/components/ogre/AvatarLogger.h~	2008-04-13 13:14:06.000000000 +0300
+++ src/components/ogre/AvatarLogger.h	2008-06-05 22:12:16.000000000 +0300
@@ -29,6 +29,7 @@
 #include <string>
 #include "framework/ConsoleObject.h"
 
+#include <memory>
 
 namespace EmberOgre {
 
--- src/components/ogre/widgets/icons/IconRenderer.h~	2008-04-13 13:14:07.000000000 +0300
+++ src/components/ogre/widgets/icons/IconRenderer.h	2008-06-05 22:23:47.000000000 +0300
@@ -26,6 +26,8 @@
 #include <string>
 #include <Ogre.h>
 
+#include <memory>
+
 namespace EmberOgre {
 
 class SimpleRenderContext;
