Index: Python-2.5.4/Tools/scripts/idle
===================================================================
--- Python-2.5.4.orig/Tools/scripts/idle
+++ Python-2.5.4/Tools/scripts/idle
@@ -2,4 +2,9 @@
 
 from idlelib.PyShell import main
 if __name__ == '__main__':
+    # Workaround for the bug reported in Pardus Bugzilla #8604
+    # See http://www.python.org/doc/2.3.5/whatsnew/node18.html
+    import Tkinter
+    Tkinter.wantobjects = 0
+
     main()
