Index: kdelibs-4.1.1/kio/kio/dummyanalyzers/dummyanalyzers.cpp
===================================================================
--- kdelibs-4.1.1.orig/kio/kio/dummyanalyzers/dummyanalyzers.cpp
+++ kdelibs-4.1.1/kio/kio/dummyanalyzers/dummyanalyzers.cpp
@@ -40,7 +40,7 @@ public:
     bool checkHeader(const char*, int32_t) const {
         return false;
     }
-    char analyze(Strigi::AnalysisResult&, InputStream*) {
+    signed char analyze(Strigi::AnalysisResult&, InputStream*) {
         return -1;
     }
     const char* name() const { return "DummyEndAnalyzer"; }
