--- stk11xx-1.4.0-old/stk11xx-buf.c	2008-11-14 10:44:55.000000000 +0200
+++ stk11xx-1.4.0/stk11xx-buf.c	2008-11-28 22:20:39.000000000 +0200
@@ -45,7 +45,7 @@
 
 #include <linux/usb.h>
 #include <media/v4l2-common.h>
-#include <media/v4l2-ioctl.h>
+
 
 #include "stk11xx.h"
 
--- stk11xx-1.4.0-old/stk11xx-dev-6a31.c	2008-11-14 10:44:55.000000000 +0200
+++ stk11xx-1.4.0/stk11xx-dev-6a31.c	2008-11-28 22:21:33.000000000 +0200
@@ -41,7 +41,7 @@
 
 #include <linux/usb.h>
 #include <media/v4l2-common.h>
-#include <media/v4l2-ioctl.h>
+
 
 #include "stk11xx.h"
 #include "stk11xx-dev.h"
--- stk11xx-1.4.0-old/stk11xx-dev-6a33.c	2008-11-14 10:44:55.000000000 +0200
+++ stk11xx-1.4.0/stk11xx-dev-6a33.c	2008-11-28 22:22:05.000000000 +0200
@@ -41,7 +41,7 @@
 
 #include <linux/usb.h>
 #include <media/v4l2-common.h>
-#include <media/v4l2-ioctl.h>
+
 
 #include "stk11xx.h"
 #include "stk11xx-dev.h"
--- stk11xx-1.4.0-old/stk11xx-dev-6a51.c	2008-11-14 10:44:55.000000000 +0200
+++ stk11xx-1.4.0/stk11xx-dev-6a51.c	2008-11-28 22:22:24.000000000 +0200
@@ -41,7 +41,7 @@
 
 #include <linux/usb.h>
 #include <media/v4l2-common.h>
-#include <media/v4l2-ioctl.h>
+
 
 #include "stk11xx.h"
 #include "stk11xx-dev.h"
--- stk11xx-1.4.0-old/stk11xx-dev-6a54.c	2008-11-14 10:44:55.000000000 +0200
+++ stk11xx-1.4.0/stk11xx-dev-6a54.c	2008-11-28 22:23:28.000000000 +0200
@@ -41,7 +41,7 @@
 
 #include <linux/usb.h>
 #include <media/v4l2-common.h>
-#include <media/v4l2-ioctl.h>
+
 
 #include "stk11xx.h"
 #include "stk11xx-dev.h"
--- stk11xx-1.4.0-old/stk11xx-dev-6d51.c	2008-11-14 10:44:55.000000000 +0200
+++ stk11xx-1.4.0/stk11xx-dev-6d51.c	2008-11-28 22:23:44.000000000 +0200
@@ -41,7 +41,7 @@
 
 #include <linux/usb.h>
 #include <media/v4l2-common.h>
-#include <media/v4l2-ioctl.h>
+
 
 #include "stk11xx.h"
 #include "stk11xx-dev.h"
--- stk11xx-1.4.0-old/stk11xx-dev-a311.c	2008-11-14 10:44:55.000000000 +0200
+++ stk11xx-1.4.0/stk11xx-dev-a311.c	2008-11-28 22:24:09.000000000 +0200
@@ -41,7 +41,7 @@
 
 #include <linux/usb.h>
 #include <media/v4l2-common.h>
-#include <media/v4l2-ioctl.h>
+
 
 #include "stk11xx.h"
 #include "stk11xx-dev.h"
--- stk11xx-1.4.0-old/stk11xx-dev-a821.c	2008-11-14 10:44:55.000000000 +0200
+++ stk11xx-1.4.0/stk11xx-dev-a821.c	2008-11-28 22:24:54.000000000 +0200
@@ -41,7 +41,7 @@
 
 #include <linux/usb.h>
 #include <media/v4l2-common.h>
-#include <media/v4l2-ioctl.h>
+
 
 #include "stk11xx.h"
 #include "stk11xx-dev.h"
--- stk11xx-1.4.0-old/stk11xx-dev.c	2008-11-14 10:44:55.000000000 +0200
+++ stk11xx-1.4.0/stk11xx-dev.c	2008-11-28 22:25:38.000000000 +0200
@@ -41,7 +41,7 @@
 
 #include <linux/usb.h>
 #include <media/v4l2-common.h>
-#include <media/v4l2-ioctl.h>
+
 
 #include "stk11xx.h"
 #include "stk11xx-dev.h"
--- stk11xx-1.4.0-old/stk11xx-sysfs.c	2008-11-14 10:44:55.000000000 +0200
+++ stk11xx-1.4.0/stk11xx-sysfs.c	2008-11-28 22:26:27.000000000 +0200
@@ -47,7 +47,7 @@
 
 #include <linux/usb.h>
 #include <media/v4l2-common.h>
-#include <media/v4l2-ioctl.h>
+
 
 #include "stk11xx.h"
 
--- stk11xx-1.4.0-old/stk11xx-usb.c	2008-11-14 10:44:55.000000000 +0200
+++ stk11xx-1.4.0/stk11xx-usb.c	2008-11-28 22:27:02.000000000 +0200
@@ -44,7 +44,7 @@
 
 #include <linux/usb.h>
 #include <media/v4l2-common.h>
-#include <media/v4l2-ioctl.h>
+
 
 #include "stk11xx.h"
 
--- stk11xx-1.4.0-old/stk11xx-v4l.c	2008-11-14 10:44:55.000000000 +0200
+++ stk11xx-1.4.0/stk11xx-v4l.c	2008-11-28 22:29:00.000000000 +0200
@@ -47,7 +47,7 @@
 
 #include <linux/usb.h>
 #include <media/v4l2-common.h>
-#include <media/v4l2-ioctl.h>
+
 
 #include "stk11xx.h"
 
