diff -up radiotray-0.5.1/setup.py.orig radiotray-0.5.1/setup.py
--- radiotray-0.5.1/setup.py.orig	2010-04-14 10:28:43.939997558 -0400
+++ radiotray-0.5.1/setup.py	2010-04-14 10:31:33.992801367 -0400
@@ -85,15 +85,12 @@ setup(name='radiotray',
     license='GPL',
     scripts=['radiotray'],
     data_files = [
-      ('share/applications', glob.glob('data/images/*.png')),
-      ('share/applications', glob.glob('data/images/*.gif')),
       ('share/applications', glob.glob('data/*.desktop')),
       ('share/doc/radiotray-%s' % APPVERSION,
           ['AUTHORS', 'CONTRIBUTORS', 'COPYING', 'NEWS', 'README']),
       ('share/radiotray/images', glob.glob('data/images/*.png')),
       ('share/radiotray/images', glob.glob('data/images/*.gif')),
-      ('share/pixmaps', glob.glob('data/images/*.png')),
-      ('share/pixmaps', glob.glob('data/images/*.gif')),
+      ('share/pixmaps', glob.glob('data/images/radiotray.png')),
       ('share/radiotray', ['data/configBookmarks.glade', 'data/bookmarks.xml']),
     ],
     package_dir={'radiotray': 'src'},
