diff -Naur radiotray-0.3.orig/setup.py radiotray-0.3/setup.py
--- radiotray-0.3.orig/setup.py	2010-01-31 14:21:45.888045746 +0200
+++ radiotray-0.3/setup.py	2010-01-31 14:23:37.542041563 +0200
@@ -11,6 +11,6 @@
       description='Radio Tray is an online radio stream player',
       license='GPL',
       packages=['radiotraylib'],
-      data_files=[('share/radiotray',['configBookmarks.glade','COPYING','README','NEWS']), ('share/radiotray/icons',['icons/radiotray.png']), ('share/applications',['radiotray.desktop'])],
+      data_files=[('share/radiotray',['configBookmarks.glade']), ('share/radiotray/icons',['icons/radiotray.png']), ('share/applications',['radiotray.desktop'])],
       scripts=['radiotray']
      )
