diff -Naur kdebase-workspace-4.0.85-orig/startkde.cmake kdebase-workspace-4.0.85/startkde.cmake
--- kdebase-workspace-4.0.85-orig/startkde.cmake	2008-07-15 09:32:52.000000000 +0300
+++ kdebase-workspace-4.0.85/startkde.cmake	2008-07-15 09:33:21.000000000 +0300
@@ -327,6 +327,11 @@
     exit 1
 fi
 
+# Launch PulseAudio
+if [ -d /tmp/pulse-$USER ]; then
+    rm -rf /tmp/pulse-$USER
+fi
+eval `/usr/bin/pulseaudio -D`
 
 # Mark that full KDE session is running (e.g. Konqueror preloading works only
 # with full KDE running). The KDE_FULL_SESSION property can be detected by
