--- avidemux/ADM_libraries/ADM_utilities/prefs.cpp.orig	2009-03-09 01:37:19.000000000 +0200
+++ avidemux/ADM_libraries/ADM_utilities/prefs.cpp	2009-03-09 01:37:48.000000000 +0200
@@ -90,7 +90,7 @@
 #ifdef __WIN32
 	{"filters.subtitle.fontname",		FILENAME,"C:\\Windows\\Fonts\\arial.ttf",NULL, NULL, NULL },
 #else
-	{"filters.subtitle.fontname",		FILENAME,"/usr/share/fonts/truetype/DejaVuSans.ttf",NULL, NULL, NULL },
+	{"filters.subtitle.fontname",		FILENAME,"/usr/share/fonts/dejavu/DejaVuSans.ttf",NULL, NULL, NULL },
 #endif
 	{"filters.subtitle.charset",		STRING,"ISO-8859-1",NULL, NULL, NULL },
 	{"filters.subtitle.fontsize",		UINT,	"24",	NULL,	"1",	"576"	},
@@ -103,7 +103,7 @@
 #ifdef __WIN32
 	{"device.audiodevice",		STRING,"WIN32",NULL, NULL, NULL },
 #else
-	{"device.audiodevice",		STRING,"ALSA",	NULL, NULL, NULL },
+	{"device.audiodevice",		STRING,"ARTS",	NULL, NULL, NULL },
 #endif
 	{"device.audio.alsa_device",		STRING,"dmix",	NULL, NULL, NULL },
 	{"device.videodevice",		UINT,	"0",	NULL,	"0",	"10"	},
--- avidemux/prefs.in.orig	2009-03-09 01:35:25.000000000 +0200
+++ avidemux/prefs.in	2009-03-09 01:40:04.000000000 +0200
@@ -39,7 +39,7 @@
 #ifdef __WIN32
 filters.subtitle.fontname,		FILENAME, "C:\\Windows\\Fonts\\arial.ttf"
 #else
-filters.subtitle.fontname,		FILENAME, "/usr/share/fonts/truetype/DejaVuSans.ttf"
+filters.subtitle.fontname,		FILENAME, "/usr/share/fonts/dejavu/DejaVuSans.ttf"
 #endif
 filters.subtitle.charset,		STRING, "ISO-8859-1"
 filters.subtitle.fontsize,		UINT,	24,	1,	576
@@ -53,7 +53,7 @@
 #ifdef __WIN32
 device.audiodevice,			STRING,	WIN32
 #else
-device.audiodevice,			STRING,	ALSA
+device.audiodevice,			STRING,	ARTS
 #endif
 device.audio.alsa_device,               STRING, "dmix"
 device.videodevice,			UINT,	0,	0,	10
