diff -Naur q4wine-0.113.orig/q4wine.desktop q4wine-0.113/q4wine.desktop
--- q4wine-0.113.orig/q4wine.desktop	2009-10-09 04:15:32.007057043 +0300
+++ q4wine-0.113/q4wine.desktop	2009-10-09 04:17:55.794054932 +0300
@@ -2,12 +2,14 @@
 Categories=Qt;System;Emulator;
 Comment[uk]=Графічна оболонка для налаштування Wine
 Comment=GUI configuration tool for Wine
+Comment[tr]=Wine ayarları için Qt arayüzü
 Comment[en]=GUI configuration tool for Wine
 Comment[ru]=Графическая оболочка для настройки Wine
 Encoding=UTF-8
 Exec=q4wine
 GenericName[uk]=Графічна оболонка для налаштування Wine
 GenericName=GUI configuration tool for Wine
+GenericName[tr]=Wine Yapılandırma Aracı
 GenericName[en]=GUI configuration tool for Wine
 GenericName[ru]=Графическая оболочка для настройки Wine
 Icon=q4wine
