--- resource.c
+++ resource.c
@@ -388,10 +388,10 @@
 				si.loads[2]);
 			tprintf("totalram=%lu, freeram=%lu, ",
 				si.totalram, si.freeram);
-			tprintf("sharedram=%lu, bufferram=%lu} ",
+			tprintf("sharedram=%lu, bufferram=%lu, ",
 				si.sharedram, si.bufferram);
-			tprintf("totalswap=%lu, freeswap=%lu, procs=%hu}",
-				si.totalswap, si.freeswap, si.procs);
+			tprintf("totalswap=%lu, freeswap=%lu, procs=%hu, mem_unit=%u}",
+				si.totalswap, si.freeswap, si.procs, si.mem_unit);
 		}
 	}
 	return 0;
