diff -Naur mameexecutor-0.9b3.orig/mameexecutor.desktop mameexecutor-0.9b3/mameexecutor.desktop
--- mameexecutor-0.9b3.orig/mameexecutor.desktop	2009-01-18 17:39:47.000000000 +0200
+++ mameexecutor-0.9b3/mameexecutor.desktop	2009-01-25 19:42:02.000000000 +0200
@@ -1,9 +1,12 @@
 [Desktop Entry]
 Name=Mame Executor
+GenericName=SDLMame Frontent
+GenericName[tr]=SDLMame Arayüzü
 Exec=mameexecutor
 Icon=mameexecutor
 Type=Application
 Terminal=0
 Categories=Qt;KDE;Game;Emulator
 Comment=SDLMame Frontend For KDE4
+Comment[tr]=KDE4 için SDLMame arayüzü
 
