--- swig/python/Makefile.in.orig	2010-01-14 15:38:11.756370955 +0200
+++ swig/python/Makefile.in	2010-01-14 15:38:58.056380873 +0200
@@ -403,13 +403,6 @@
 	    $(pkgpythonPYTHON_INSTALL) "$$b$$p" "$(DESTDIR)$(pkgpythondir)/$$f"; \
 	  else :; fi; \
 	done; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pkgpythondir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pkgpythondir)" $$dlist; \
-	  fi; \
-	else :; fi
 
 uninstall-pkgpythonPYTHON:
 	@$(NORMAL_UNINSTALL)
