Index: kdeadmin/secpolicy/Makefile.am
===================================================================
--- kdeadmin.orig/secpolicy/Makefile.am
+++ kdeadmin/secpolicy/Makefile.am
@@ -3,8 +3,8 @@ METASOURCES = AUTO
 
 bin_PROGRAMS = secpolicy
 secpolicy_SOURCES = main.cpp secpolicywin.cpp pamview.cpp ppitem.cpp
-secpolicy_LDADD = $(LIB_KIO)
-secpolicy_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+secpolicy_LDADD = /usr/kde/3.5/lib/libkio.so $(LIB_KIO)
+secpolicy_LDFLAGS = /usr/kde/3.5/lib/libkio.so $(all_libraries) $(KDE_RPATH)
 
 noinst_HEADERS = secpolicywin.h pamview.h ppitem.h
 
