diff -ru gdcm-2.0.16_orig/Utilities/VTK/Applications/CMakeLists.txt gdcm-2.0.16/Utilities/VTK/Applications/CMakeLists.txt
--- gdcm-2.0.16_orig/Utilities/VTK/Applications/CMakeLists.txt	2010-08-18 12:25:43.000000000 +0200
+++ gdcm-2.0.16/Utilities/VTK/Applications/CMakeLists.txt	2010-11-19 20:30:06.447787551 +0100
@@ -29,7 +29,7 @@
   IF(GDCM_EXECUTABLE_PROPERTIES)
     SET_TARGET_PROPERTIES(${app} PROPERTIES ${GDCM_EXECUTABLE_PROPERTIES})
   ENDIF(GDCM_EXECUTABLE_PROPERTIES)
-  TARGET_LINK_LIBRARIES(${app} ${VTKGDCM_NAME})
+  TARGET_LINK_LIBRARIES(${app} ${VTKGDCM_NAME} gdcmMSFF)
   IF(VTK_USE_RENDERING)
     TARGET_LINK_LIBRARIES(${app} vtkRendering)
 
Only in gdcm-2.0.16/Utilities/VTK/Applications: CMakeLists.txt~
diff -ru gdcm-2.0.16_orig/Utilities/VTK/Examples/Cxx/CMakeLists.txt gdcm-2.0.16/Utilities/VTK/Examples/Cxx/CMakeLists.txt
--- gdcm-2.0.16_orig/Utilities/VTK/Examples/Cxx/CMakeLists.txt	2010-08-18 12:25:43.000000000 +0200
+++ gdcm-2.0.16/Utilities/VTK/Examples/Cxx/CMakeLists.txt	2010-11-19 20:30:09.457709578 +0100
@@ -40,7 +40,7 @@
 FOREACH(app ${GDCM_VTK_APPS})
   ADD_EXECUTABLE(${app} ${app}.cxx)
   #SET_TARGET_PROPERTIES(${app} PROPERTIES ${GDCM_EXECUTABLE_PROPERTIES})
-  TARGET_LINK_LIBRARIES(${app} ${VTKGDCM_NAME} ${vtkgdcm_LIBS})
+  TARGET_LINK_LIBRARIES(${app} ${VTKGDCM_NAME} ${vtkgdcm_LIBS} gdcmMSFF)
   IF( "${VTK_MAJOR_VERSION}.${VTK_MINOR_VERSION}" GREATER 5.0 )
     IF(VTK_USE_RENDERING)
       TARGET_LINK_LIBRARIES(${app} vtkWidgets vtkVolumeRendering)
Only in gdcm-2.0.16/Utilities/VTK/Examples/Cxx: CMakeLists.txt~
