--- CMake/UseCopyright.cmake	2009-11-12 23:09:48.000000000 +0100
+++ CMake/UseCopyright2.cmake	2010-02-17 13:24:09.000000000 +0100
@@ -18,8 +18,10 @@
   # need to raise an error if COPYRIGHT_MODULE_FILENAME is not set...
   IF(EXISTS ${COPYRIGHT_MODULE_FILENAME} )
     FOREACH(filename ${ARGN})
-      FILE(READ ${filename} content)
-      FILE(APPEND ${COPYRIGHT_MODULE_FILENAME} ${content})
+      IF(EXISTS ${filename} )
+         FILE(READ ${filename} content)
+         FILE(APPEND ${COPYRIGHT_MODULE_FILENAME} ${content})
+      ENDIF(EXISTS ${filename} )
     ENDFOREACH(filename)
   ENDIF(EXISTS ${COPYRIGHT_MODULE_FILENAME} )
 ENDMACRO(APPEND_COPYRIGHT)
