--- src/libFingerprint/fplib/src/OptFFT.cpp	2007-12-05 14:39:28.000000000 +0200
+++ src/libFingerprint/fplib/src/OptFFT.cpp	2007-12-11 15:19:17.000000000 +0200
@@ -30,6 +30,7 @@
 
 #include <cmath>
 #include <cassert>
+#include <cstdlib>
 #include <iostream>
 #include <memory.h> // for memcopy
 
--- src/libFingerprint/fplib/src/FingerprintExtractor.cpp	2007-12-05 14:39:28.000000000 +0200
+++ src/libFingerprint/fplib/src/FingerprintExtractor.cpp	2007-12-11 15:19:55.000000000 +0200
@@ -31,6 +31,7 @@
 #include <vector>
 #include <stdexcept>
 #include <cmath>
+#include <string.h>
 
 #include <samplerate.h> // libsamplerate
 
--- src/libFingerprint/MP3_Source_Qt.cpp	2007-12-05 14:39:28.000000000 +0200
+++ src/libFingerprint/MP3_Source_Qt.cpp	2007-12-11 16:26:29.000000000 +0200
@@ -20,9 +20,10 @@
 
 #include <iostream>
 #include <fstream>
-#include <limits>
 #include <sstream>
 #include <cassert>
+#include <climits>
+#include <cstdlib>
 #include <stdexcept>
 
 #include "MP3_Source_Qt.h"
