diff -Nur migration4-old/setup.py migration4/setup.py
--- migration4-old/setup.py	2009-07-17 19:46:29.695252018 +0300
+++ migration4/setup.py	2009-07-17 19:47:31.162002085 +0300
@@ -70,7 +70,7 @@
             kde_dir = "/usr/kde/4"
         bin_dir = os.path.join(kde_dir, "bin")
         locale_dir = os.path.join(kde_dir, "share/locale")
-        autostart_dir = os.path.join(kde_dir, "share/autostart")
+        autostart_dir = os.path.join(kde_dir, "share/applications/kde4")
         project_dir = os.path.join(kde_dir, "share/apps", about.appName)
         # Make directories
         makeDirs(bin_dir)
