diff -ur a/doc/examples/kde-desktop-icons b/doc/examples/kde-desktop-icons
--- a/doc/examples/kde-desktop-icons	2009-03-02 21:22:19.000000000 +0530
+++ b/doc/examples/kde-desktop-icons	2011-03-22 11:51:19.000000000 +0530
@@ -26,19 +26,19 @@
         sys.exit(1)
 
     if dev.startswith('usb'):
-        icon = 'usbpendrive_mount'
+        icon = 'drive-harddisk'
     elif dev.startswith('ata'):
-        icon = 'hdd_mount'
+        icon = 'drive-harddisk'
     elif dev.startswith('floppy'):
-        icon = '3floppy_mount'
+        icon = 'yast-floppy'
     elif dev.startswith('cdrom'):
-        icon = 'cdrom_mount'
+        icon = 'yast-cdrom'
     else:
-        icon = 'usbpendrive_mount'
+        icon = 'drive-harddisk'
 
     try:
         desktop_file = open(desktop_file_s, 'w')
-        desktop_file.write('[Desktop Entry]\nEncoding=UTF-8\nName=%s\nIcon=%s\nType=Link\nURL=%s\n' % (dev, icon, mountpoint))
+        desktop_file.write('[Desktop Entry]\nEncoding=UTF-8\nName=%s\nIcon=%s\nType=Link\nURL=file://%s\n' % (dev, icon, mountpoint))
         desktop_file.close()
     except IOError, e:
         print >>sys.stderr, 'unable to create desktop file:', e      
