diff -ur a/scripts/ltspfsmounter b/scripts/ltspfsmounter
--- a/scripts/ltspfsmounter	2009-03-02 21:22:19.000000000 +0530
+++ b/scripts/ltspfsmounter	2011-03-22 11:52:23.000000000 +0530
@@ -46,11 +46,11 @@
         print >>sys.stderr, "mount failed:", e
     try:
         call(lbmount_command)
+        run_hooks('add', os.path.join(mediaroot, dev))
         if os.access(hidden_mount, 0):
             os.rmdir(hidden_mount)
         if os.access(root, 0):
             os.rmdir(root)
-        run_hooks('add', os.path.join(mediaroot, dev))
     except OSError, e:
         print >>sys.stderr, "suid mount failed:", e
 
