diff -Nuar wired-0.4.orig/src/engine/AudioEngine.h wired-0.4/src/engine/AudioEngine.h
--- wired-0.4.orig/src/engine/AudioEngine.h	2007-02-25 12:52:11.000000000 +0200
+++ wired-0.4/src/engine/AudioEngine.h	2007-02-25 14:30:21.000000000 +0200
@@ -6,7 +6,7 @@
 
 #include <vector>
 
-#include "portaudio.h"
+#include <portaudio.h>
 #include <wx/thread.h>
 
 #include "Device.h"
diff -Nuar wired-0.4.orig/src/engine/Device.h wired-0.4/src/engine/Device.h
--- wired-0.4.orig/src/engine/Device.h	2007-02-25 12:52:11.000000000 +0200
+++ wired-0.4/src/engine/Device.h	2007-02-25 14:30:43.000000000 +0200
@@ -6,7 +6,7 @@
 
 #include <iostream>
 #include <vector>
-#include "portaudio.h"
+#include <portaudio.h>
 #include <wx/wx.h>
 
 #define MIN	0
diff -Nuar wired-0.4.orig/src/samplerate/WiredSampleRate.h wired-0.4/src/samplerate/WiredSampleRate.h
--- wired-0.4.orig/src/samplerate/WiredSampleRate.h	2007-02-25 12:52:11.000000000 +0200
+++ wired-0.4/src/samplerate/WiredSampleRate.h	2007-02-25 14:30:02.000000000 +0200
@@ -13,7 +13,7 @@
 #include <stdio.h>
 #include <sndfile.h>
 #include <samplerate.h>
-#include "portaudio.h"
+#include <portaudio.h>
 #include <wx/msgdlg.h>
 #include <wx/progdlg.h>
 #include <wx/choicdlg.h>
