Index: pilot-link-0.12.3/src/pilot-schlep.c
===================================================================
--- pilot-link-0.12.3.orig/src/pilot-schlep.c
+++ pilot-link-0.12.3/src/pilot-schlep.c
@@ -38,7 +38,7 @@ static int Fetch(int sd, char *filename)
 		fd;
 	pi_buffer_t *buffer;
 
-	fd = open(filename, O_WRONLY | O_CREAT | O_TRUNC);
+	fd = open(filename, O_WRONLY | O_CREAT | O_TRUNC, 0777);
 	if (fd < 0)
 		return -1;
 
