diff -Nur linux-2.6.18-old/drivers/media/video/zc0301/zc0301_sensor.h linux-2.6.18/drivers/media/video/zc0301/zc0301_sensor.h
--- linux-2.6.18-old/drivers/media/video/zc0301/zc0301_sensor.h	2006-12-04 11:56:32.269820428 +0200
+++ linux-2.6.18/drivers/media/video/zc0301/zc0301_sensor.h	2006-12-04 12:05:48.166311315 +0200
@@ -77,8 +77,6 @@
 	{ ZC0301_USB_DEVICE(0x055f, 0xd004, 0xff), }, /* TAS5130 */           \
 	{ ZC0301_USB_DEVICE(0x046d, 0x08ae, 0xff), }, /* PAS202 */            \
 	{ ZC0301_USB_DEVICE(0x0ac8, 0x0301, 0xff), },                         \
-	{ ZC0301_USB_DEVICE(0x0ac8, 0x301b, 0xff), }, /* PB-0330/HV7131 */    \
-	{ ZC0301_USB_DEVICE(0x0ac8, 0x303b, 0xff), }, /* PB-0330 */           \
 	{ ZC0301_USB_DEVICE(0x10fd, 0x0128, 0xff), }, /* TAS5130 */           \
 	{ ZC0301_USB_DEVICE(0x10fd, 0x8050, 0xff), }, /* TAS5130 */           \
 	{ ZC0301_USB_DEVICE(0x10fd, 0x804e, 0xff), }, /* TAS5130 */           \
