Index: aria2c-0.14.0/src/OptionHandlerFactory.cc
===================================================================
--- aria2c-0.14.0.orig/src/OptionHandlerFactory.cc
+++ aria2c-0.14.0/src/OptionHandlerFactory.cc
@@ -86,7 +86,7 @@ OptionHandlers OptionHandlerFactory::cre
   handlers.push_back(SH(new DefaultOptionHandler(PREF_DIR)));
   handlers.push_back(SH(new DefaultOptionHandler(PREF_OUT)));
   handlers.push_back(SH(new LogOptionHandler(PREF_LOG)));
-  handlers.push_back(SH(new NumberOptionHandler(PREF_SPLIT, 1, 16)));
+  handlers.push_back(SH(new NumberOptionHandler(PREF_SPLIT, 1, 500)));
   handlers.push_back(SH(new NumberOptionHandler(PREF_TIMEOUT, 1, 600)));
   handlers.push_back(SH(new NumberOptionHandler(PREF_MAX_TRIES, 0)));
   handlers.push_back(SH(new BooleanOptionHandler(PREF_FTP_PASV)));
Index: aria2c-0.14.0/src/HelpItemFactory.cc
===================================================================
--- aria2c-0.14.0.orig/src/HelpItemFactory.cc
+++ aria2c-0.14.0/src/HelpItemFactory.cc
@@ -71,7 +71,7 @@ TagContainerHandle HelpItemFactory::crea
   {
     HelpItemHandle item(new HelpItem(PREF_SPLIT, TEXT_SPLIT,
 				     op->get(PREF_SPLIT)));
-    item->setAvailableValues("1-16");
+    item->setAvailableValues("1-500");
     item->addTag(TAG_BASIC);
     item->addTag(TAG_HTTP);
     item->addTag(TAG_FTP);
