Yalnızca src-stormbaancoureur.orig/'da: images
diff -ur src-stormbaancoureur.orig/main.cxx src-stormbaancoureur/main.cxx
--- src-stormbaancoureur.orig/main.cxx	2007-10-13 20:39:41.000000000 +0300
+++ src-stormbaancoureur/main.cxx	2007-10-14 13:42:54.000000000 +0300
@@ -705,11 +705,11 @@
   fprintf(stderr,"OpenDE is (c) by Russel L. Smith\n");
 
   char *bindirname  = dirname(argv[0]);
-  if (!strcmp(bindirname,"."))
-    dirprefix="/usr/share/games/stormbaancoureur";
+  if (!strcmp(bindirname,"/usr/bin"))
+    dirprefix="/usr/share/stormbaancoureur";
   else
   {
-    dirprefix = dirname(bindirname) + std::string("/share/games/stormbaancoureur");
+    dirprefix = std::string("/usr/share/stormbaancoureur");
   }
   if (getenv("PLODE_DATADIR"))
     dirprefix = getenv("PLODE_DATADIR");
diff -ur src-stormbaancoureur.orig/Makefile src-stormbaancoureur/Makefile
--- src-stormbaancoureur.orig/Makefile	2007-10-14 13:33:47.000000000 +0300
+++ src-stormbaancoureur/Makefile	2007-10-14 13:25:47.000000000 +0300
@@ -93,15 +93,15 @@
 clean:
 	rm -f *.o stormbaancoureur
 
-GAMEDIR=$(DESTDIR)/usr/share/games/stormbaancoureur
+GAMEDIR=$(DESTDIR)/usr/share/stormbaancoureur
 install: stormbaancoureur
 	# Directories
-	mkdir -p $(DESTDIR)/usr/games/
+	mkdir -p $(DESTDIR)/usr/bin/
 	mkdir -p $(GAMEDIR)/images/
 	mkdir -p $(GAMEDIR)/sounds/
 	mkdir -p $(GAMEDIR)/models/
 	# Binary
-	install stormbaancoureur $(DESTDIR)/usr/games/stormbaancoureur
+	install stormbaancoureur $(DESTDIR)/usr/bin/stormbaancoureur
 	# Images
 	install -m 644 images/engine.tga $(GAMEDIR)/images/engine.tga
 	install -m 644 images/info_carpet_smooth.rgb $(GAMEDIR)/images/info_carpet_smooth.rgb
Yalnızca src-stormbaancoureur.orig/'da: models
Yalnızca src-stormbaancoureur.orig/'da: sounds
