diff -Nur yali4-2.0.4.orig/yali4-bin yali4-2.0.4/yali4-bin
--- yali4-2.0.4.orig/yali4-bin	2008-09-04 13:57:07.000000000 +0300
+++ yali4-2.0.4/yali4-bin	2008-09-05 10:18:48.000000000 +0300
@@ -23,7 +23,7 @@
         yali4.sysutils.umount(consts.target_dir + mpoint)
 
 def main():
-    if not options.firstBoot == True or exists("/etc/yali-is-firstboot"):
+    if not exists("/etc/yali-is-firstboot"):
         print "Trying to umount .."
         try_umount_target()
     runner = yali4.default_runner()
