Index: OpenKinect-libfreenect-dbfd4ce/cmake_modules/SetupDirectories.cmake
===================================================================
--- OpenKinect-libfreenect-dbfd4ce.orig/cmake_modules/SetupDirectories.cmake
+++ OpenKinect-libfreenect-dbfd4ce/cmake_modules/SetupDirectories.cmake
@@ -11,13 +11,7 @@ MESSAGE (STATUS "${PROJECT_NAME} will be
 STRING (TOLOWER ${PROJECT_NAME} projectNameLower)
 SET (PROJECT_INCLUDE_INSTALL_DIR "include/${projectNameLower}")
 
-IF (PROJECT_PROC_64BIT)
-    SET (LIB_SUFFIX "64" CACHE STRING "Suffix for library installation directory")
-ELSE (PROJECT_PROC_64BIT)
-    SET (LIB_SUFFIX "" CACHE STRING "Suffix for library installation directory")
-ENDIF (PROJECT_PROC_64BIT)
-
-SET (PROJECT_LIBRARY_INSTALL_DIR "lib${LIB_SUFFIX}")
+SET (PROJECT_LIBRARY_INSTALL_DIR "lib")
 
 MESSAGE (STATUS "Headers will be installed to ${CMAKE_INSTALL_PREFIX}/${PROJECT_INCLUDE_INSTALL_DIR}")
 MESSAGE (STATUS "Libraries will be installed to ${CMAKE_INSTALL_PREFIX}/${PROJECT_LIBRARY_INSTALL_DIR}")
