Index: OpenKinect-libfreenect-dbfd4ce/fakenect/CMakeLists.txt
===================================================================
--- OpenKinect-libfreenect-dbfd4ce.orig/fakenect/CMakeLists.txt
+++ OpenKinect-libfreenect-dbfd4ce/fakenect/CMakeLists.txt
@@ -14,5 +14,6 @@ install (TARGETS fakenect
 
 add_executable(record record.c)
 target_link_libraries(record freenect m)
+target_link_libraries(fakenect m)
 install (TARGETS record
-  DESTINATION bin)
\ Dosya sonunda yenisatır yok.
+  DESTINATION bin)
