--- frostwire.desktop	2007-12-17 08:50:18.000000000 +0200
+++ frostwire.desktop2	2007-12-20 06:55:49.000000000 +0200
@@ -1,9 +1,12 @@
 [Desktop Entry]
 Name=FrostWire
+Name[tr]=FrostWire
 GenericName=P2P Gnutella/Bittorrent client
+GenericName[tr]=P2P Gnutella/Bittorrent İstemcisi
 Comment=Search and share all kinds of files on the Gnutella network
-Exec=/usr/bin/frostwire
-Icon=frostwire.png
+Comment[tr]=Gnutella ağında dosya arama ve paylaşımı
+Exec=frostwire
+Icon=frostwire
 Terminal=false
 Type=Application
-Categories=Application;Network;
\ Dosya sonunda yenisatır yok.
+Categories=Application;Network;
