Index: svxlink-11.11.1/create_config.sh
===================================================================
--- svxlink-11.11.1.orig/create_config.sh
+++ svxlink-11.11.1/create_config.sh
@@ -123,26 +123,6 @@ else
   exit_error
 fi
 
-# Checking for speex
-info "--- Checking for speex..."
-if which pkg-config > /dev/null 2>&1; then
-  if pkg-config speex; then
-    ver=$(pkg-config speex --modversion)
-    ver_major=$(echo $ver | sed -r 's/^([0-9]+)\..*$/\1/')
-    ver_minor=$(echo $ver | sed -r 's/^([0-9]+)\.([0-9]+).*$/\2/')
-    info "$ver\n"
-    output "SPEEX_LIBPATH=$(pkg-config speex --libs-only-L)"
-    output "SPEEX_LIBS=$(pkg-config speex --libs-only-l)"
-    output "SPEEX_INCPATH=$(pkg-config speex --cflags-only-I)"
-    output "CFLAGS_DEFINES+=-DSPEEX_MAJOR=$ver_major -DSPEEX_MINOR=$ver_minor"
-    output "USE_SPEEX=1"
-  else
-    info "no (optional)\n"
-  fi
-else
-  info "no (optional)\n"
-fi
-
 # Checking for libgcrypt
 info "--- Checking for libgcrypt..."
 if which libgcrypt-config > /dev/null 2>&1; then
