diff -Naur blogtk-2.0.orig/data/blogtk.desktop blogtk-2.0/data/blogtk.desktop
--- blogtk-2.0.orig/data/blogtk.desktop	2009-09-23 17:46:56.523926085 +0300
+++ blogtk-2.0/data/blogtk.desktop	2009-09-23 18:12:16.780798648 +0300
@@ -1,10 +1,13 @@
 [Desktop Entry]
 Encoding=UTF-8
-Name=BloGTK 2.0
+Name=BloGTK
+GenericName=GTK Weblog Client
+GenericName[tr]=GTK Günlük Uygulaması
 Comment=GTK Weblog Client
+Comment[tr]=GTK Günlük Uygulaması
 Exec=blogtk2
 Terminal=false
 Type=Application
-Icon=blogtk-icon.png
+Icon=blogtk.png
 StartupNotify=true
 Categories=GNOME;Application;Network;
