diff -Nur -x '*.orig' -x '*~' hal-info-20080215/fdi/information/10freedesktop/20-video-quirk-pm-asus.fdi hal-info-20080215.new/fdi/information/10freedesktop/20-video-quirk-pm-asus.fdi
--- hal-info-20080215/fdi/information/10freedesktop/20-video-quirk-pm-asus.fdi	2008-02-15 23:22:56.000000000 +0100
+++ hal-info-20080215.new/fdi/information/10freedesktop/20-video-quirk-pm-asus.fdi	2008-02-28 16:38:04.000000000 +0100
@@ -5,10 +5,10 @@
       <match key="system.hardware.product" prefix="U5F ">
         <merge key="power_management.quirk.s3_bios" type="bool">true</merge>
       </match>
-      <match key="system.hardware.product" prefix_outof="K8N-E-Deluxe;L2000D;M6Ne;V6V">
+      <match key="system.hardware.product" prefix_outof="K8N-E-Deluxe;L2000D;M6Ne">
         <merge key="power_management.quirk.s3_mode" type="bool">true</merge>
       </match>
-      <match key="system.hardware.product" prefix_outof="M6VA;M6N ;M5N;M7A;S6F;S5N ;W3A;W5A ;M2N ;A2D ;F3F ">
+      <match key="system.hardware.product" prefix_outof="M6VA;M6N ;M5N;M7A;S6F;S5N ;W3A;W5A ;M2N ;A2D ;F3F ;V6V">
         <merge key="power_management.quirk.s3_bios" type="bool">true</merge>
         <merge key="power_management.quirk.s3_mode" type="bool">true</merge>
       </match>
