diff -Nur -x '*.orig' -x '*~' hal-info-20080215/fdi/information/10freedesktop/10-usb-music-players.fdi hal-info-20080215.new/fdi/information/10freedesktop/10-usb-music-players.fdi
--- hal-info-20080215/fdi/information/10freedesktop/10-usb-music-players.fdi	2008-02-28 16:57:10.000000000 +0100
+++ hal-info-20080215.new/fdi/information/10freedesktop/10-usb-music-players.fdi	2008-02-28 16:57:43.000000000 +0100
@@ -350,8 +350,8 @@
 
 	<!-- RCA -->
 	<match key="@storage.originating_device:usb.vendor_id" int="0x69b">
-	  <!-- Lyra RD1072, RD2212, MC2021C -->
-	  <match key="@storage.originating_device:usb.product_id" int_outof="0x713;0x718;0x767">
+	  <!-- Lyra RD1072, RD2212, MC2021C, TH1501 -->
+	  <match key="@storage.originating_device:usb.product_id" int_outof="0x713;0x718;0x767;0x756">
 	    <merge key="portable_audio_player.type" type="string">generic</merge>
 	    <merge key="portable_audio_player.access_method" type="string">storage</merge>
 	    <append key="portable_audio_player.output_formats" type="strlist">audio/x-ms-wma</append>
