diff -Nur -x '*.orig' -x '*~' hal-info-20080215/fdi/information/10freedesktop/10-usb-music-players.fdi hal-info-20080215.new/fdi/information/10freedesktop/10-usb-music-players.fdi
--- hal-info-20080215/fdi/information/10freedesktop/10-usb-music-players.fdi	2008-02-28 16:11:49.000000000 +0100
+++ hal-info-20080215.new/fdi/information/10freedesktop/10-usb-music-players.fdi	2008-02-28 16:14:37.000000000 +0100
@@ -774,6 +774,16 @@
           </match>
         </match>
 
+        <!-- Mobile Media Tech. -->
+        <match key="@storage.originating_device:usb.vendor_id" int="0x1011">
+          <!-- No-name device, sold as "Lavod" -->
+          <match key="@storage.originating_device:usb.product_id" int="0x0001">
+            <merge key="portable_audio_player.type" type="string">generic</merge>
+            <merge key="portable_audio_player.access_method" type="string">storage</merge>
+            <append key="portable_audio_player.input_formats" type="strlist">audio/x-wav</append>
+          </match>
+        </match>
+
       </match>
     </match>      
 
