diff -Nur -x '*.orig' -x '*~' hal-info-20080215/fdi/information/10freedesktop/10-usb-music-players.fdi hal-info-20080215.new/fdi/information/10freedesktop/10-usb-music-players.fdi
--- hal-info-20080215/fdi/information/10freedesktop/10-usb-music-players.fdi	2008-02-28 16:19:23.000000000 +0100
+++ hal-info-20080215.new/fdi/information/10freedesktop/10-usb-music-players.fdi	2008-02-28 16:20:25.000000000 +0100
@@ -247,6 +247,14 @@
 	    <append key="portable_audio_player.output_formats" type="strlist">application/ogg</append>
 	    <append key="portable_audio_player.input_formats" type="strlist">audio/mpeg</append>
 	  </match>	
+	  <!-- Samsung Yepp YP-K3 (Firmware_4.06KR)-->
+	  <match key="@storage.originating_device:usb.product_id" int="0x5080">
+	    <merge key="portable_audio_player.type" type="string">generic</merge>
+	    <merge key="portable_audio_player.access_method" type="string">storage</merge>
+	    <append key="portable_audio_player.output_formats" type="strlist">audio/x-ms-wma</append>
+	     <append key="portable_audio_player.output_formats" type="strlist">application/ogg</append>
+	    <append key="portable_audio_player.audio_folders" type="strlist">Music/</append>
+	  </match>
 	  <!-- Meizu M6 MiniPlayer -->
 	  <match key="@storage.originating_device:usb.product_id" int="0x5a0f">
 	    <merge key="portable_audio_player.type" type="string">generic</merge>
