diff -Nur kqemu-1.3.0pre9-old/kqemu-linux.c kqemu-1.3.0pre9/kqemu-linux.c
--- kqemu-1.3.0pre9-old/kqemu-linux.c	2006-10-09 11:40:43.000000000 +0300
+++ kqemu-1.3.0pre9/kqemu-linux.c	2006-10-09 11:41:51.000000000 +0300
@@ -52,7 +52,7 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)
 module_param(major, int, 0);
 #else
-MODULE_PARM(major,"i");
+module_param(major, int, 0);
 #endif
 
 /* Lock the page at virtual address 'user_addr' and return its
