--- configure.in	2007-03-05 20:41:56.000000000 +0200
+++ configure.in	2007-04-03 18:53:28.000000000 +0300
@@ -899,7 +899,7 @@
 if test "$with_colamd" = yes; then
   with_colamd=no
   AC_CHECK_HEADERS([suitesparse/colamd.h ufsparse/colamd.h colamd/colamd.h colamd.h], [
-    AC_CHECK_LIB(colamd, colamd, [COLAMD_LIBS="-lcolamd"; with_colamd=yes])
+    AC_CHECK_LIB(colamd, colamd, [COLAMD_LIBS="-lcolamd -lcamd"; with_colamd=yes])
     if test "$with_colamd" = yes; then
       AC_DEFINE(HAVE_COLAMD, 1, [Define if the COLAMD library is used.])
       TEXINFO_COLAMD="@set HAVE_COLAMD"
