--- src/Makefile.in~	2006-08-22 20:04:28.000000000 +0300
+++ src/Makefile.in	2006-09-18 15:06:46.000000000 +0300
@@ -313,7 +313,7 @@
 	$(OCTAVE_LIBS) \
 	$(LEXLIB) $(UMFPACK_LIBS) $(AMD_LIBS) $(COLAMD_LIBS) \
 	$(CHOLMOD_LIBS) $(CCOLAMD_LIBS) $(CXSPARSE_LIBS) $(BLAS_LIBS) \
-	$(FFTW_LIBS) $(LIBS) $(FLIBS)
+	$(FFTW_LIBS) $(LIBS) $(FLIBS) -lmpi -lmpi_cxx
 
 stmp-pic: pic
 	@if [ -f stmp-pic ]; then \
