Index: python/pykde4/sip/kio/global.sip
===================================================================
--- python/pykde4/sip/kio/global.sip	(revision 959020)
+++ python/pykde4/sip/kio/global.sip	(revision 959021)
@@ -243,8 +243,8 @@
     {
         int iserr = 0;
 
-        acpp = (QString *)sipForceConvertTo_QString (aelem, &iserr);
-        bcpp = (QString *)sipForceConvertTo_QString (belem, &iserr);
+        acpp = (QString *)sipForceConvertToInstance(aelem, sipClass_QString, NULL, 0, NULL, &iserr);
+        bcpp = (QString *)sipForceConvertToInstance(belem, sipClass_QString, NULL, 0, NULL, &iserr);
 
         if (iserr)
         {
Index: python/pykde4/sip/kio/kacl.sip
===================================================================
--- python/pykde4/sip/kio/kacl.sip	(revision 959020)
+++ python/pykde4/sip/kio/kacl.sip	(revision 959021)
@@ -144,8 +144,7 @@
     {
         PyObject *elem = PyList_GET_ITEM (sipPy, i);
         PyObject *pyp1 = PyTuple_GET_ITEM (elem, 0);
-        p1 = *(QString *)sipForceConvertTo_QString (pyp1, &iserr);
-
+        p1 = *(QString *)sipForceConvertToInstance(pyp1, sipClass_QString, NULL, 0, NULL, &iserr);
         if (iserr)
         {
             *sipIsErr = 1;
Index: python/pykde4/sip/phonon/backendcapabilities.sip
===================================================================
--- python/pykde4/sip/phonon/backendcapabilities.sip	(revision 959020)
+++ python/pykde4/sip/phonon/backendcapabilities.sip	(revision 959021)
@@ -26,7 +26,7 @@
 {
 namespace BackendCapabilities
 {
-class Notifier : QObject /NoDefaultCtors,NoQMetaObject/
+class Notifier : QObject /NoDefaultCtors,PyQt4NoQMetaObject/
 {
 
 signals:
Index: python/pykde4/sip/akonadi/messagethreadingattribute.sip
===================================================================
--- python/pykde4/sip/akonadi/messagethreadingattribute.sip	(revision 959020)
+++ python/pykde4/sip/akonadi/messagethreadingattribute.sip	(revision 959021)
@@ -30,7 +30,7 @@
 
 public:
                             MessageThreadingAttribute ();
-                            MessageThreadingAttribute (const Akonadi::MessageThreadingAttribute& other);
+//ig                            MessageThreadingAttribute (const Akonadi::MessageThreadingAttribute& other);
     QList<Akonadi::Entity::Id>         perfectParents () const;
     void                    setPerfectParents (const QList<Akonadi::Entity::Id>& parents);
     QList<Akonadi::Entity::Id>         unperfectParents () const;
Index: python/pykde4/sip/dnssd/servicebase.sip
===================================================================
--- python/pykde4/sip/dnssd/servicebase.sip	(revision 959020)
+++ python/pykde4/sip/dnssd/servicebase.sip	(revision 959021)
@@ -27,7 +27,7 @@
 %End
 
 public:
-    typedef KSharedPtr<DNSSD::ServiceBase> Ptr;
+//    typedef KSharedPtr<DNSSD::ServiceBase> Ptr;
 
     explicit                ServiceBase (const QString& name = QString(), const QString& type = QString(), const QString& domain = QString(), const QString& host = QString(), unsigned short port = 0);
     QString                 serviceName () const;
Index: python/pykde4/sip/kdecore/typedefs.sip
===================================================================
--- python/pykde4/sip/kdecore/typedefs.sip	(revision 959020)
+++ python/pykde4/sip/kdecore/typedefs.sip	(revision 959021)
@@ -244,7 +244,7 @@
         return PyInstance_Check(sipPy);
 
     int iserr = 0;
-    TYPE *cpp = (TYPE *)sipForceConvertTo_TYPE (sipPy, &iserr);
+    TYPE *cpp = (TYPE *)sipForceConvertToInstance(sipPy, sipClass_TYPE, NULL, 0, NULL, &iserr);
 
     if (iserr)
     {
Index: python/pykde4/sip/kdecore/kptydevice.sip
===================================================================
--- python/pykde4/sip/kdecore/kptydevice.sip	(revision 959020)
+++ python/pykde4/sip/kdecore/kptydevice.sip	(revision 959021)
@@ -46,8 +46,8 @@
     void                 readEof ();
 
 protected:
-    virtual qint64       readData (char*, qint64);
-    virtual qint64       readLineData (char*, qint64);
+//ig    virtual qint64       readData (char*, qint64);
+//ig    virtual qint64       readLineData (char*, qint64);
     virtual qint64       writeData (const char*, qint64);
 
 };  // class KPtyDevice
Index: python/pykde4/sip/plasma/abstractrunner.sip
===================================================================
--- python/pykde4/sip/plasma/abstractrunner.sip	(revision 959020)
+++ python/pykde4/sip/plasma/abstractrunner.sip	(revision 959021)
@@ -129,7 +129,7 @@
     void                    setHasRunOptions (bool hasRunOptions);
     void                    setSpeed (Plasma::AbstractRunner::Speed newSpeed);
     void                    setPriority (Plasma::AbstractRunner::Priority newPriority);
-    KService::List          serviceQuery (const QString& serviceType, const QString& constraint = QString()) const;
+//FIXME    KService::List          serviceQuery (const QString& serviceType, const QString& constraint = QString()) const;
     virtual QList<QAction*>  actionsForMatch (const Plasma::QueryMatch& match);
     QAction*                addAction (const QString& id, const QIcon& icon, const QString& text);
     void                    addAction (const QString& id, QAction* action);
