diff -uNr KGtk-0.9.4/kdialogd4/kdialogd.cpp KGtk-0.9.4-mod/kdialogd4/kdialogd.cpp
--- KGtk-0.9.4/kdialogd4/kdialogd.cpp	2007-10-31 22:33:19.000000000 +0200
+++ KGtk-0.9.4-mod/kdialogd4/kdialogd.cpp	2008-01-29 09:57:57.000000000 +0200
@@ -11,6 +11,7 @@
 #include <klocale.h>
 #include <kconfig.h>
 #include <kurlcombobox.h>
+#include <stdint.h>
 #ifdef USE_KWIN
 #include <kwindowsystem.h>
 #else
@@ -382,7 +383,7 @@
     // * the dir select dialog doesnt seem to set the QDialog result parameter
     //   when it is accepted - so for this reason if ok is clicked we store an
     //   'accepted' value there, and check for that after the dialog is finished.
-    kDebug() << "finished " << (int)itsDlg << itsAccepted << (itsDlg ? QDialog::Accepted==itsDlg->result() : false);
+    kDebug() << "finished " << (intptr_t)itsDlg << itsAccepted << (itsDlg ? QDialog::Accepted==itsDlg->result() : false);
 
     if(itsDlg && !(itsAccepted || QDialog::Accepted==itsDlg->result()))
         cancel();
