Index: KGtk-0.10.0/kdialogd4/kdialogd.cpp
===================================================================
--- KGtk-0.10.0.orig/kdialogd4/kdialogd.cpp
+++ KGtk-0.10.0/kdialogd4/kdialogd.cpp
@@ -178,7 +178,7 @@ KDialogD::KDialogD(QObject *parent)
             f.close();
         }
         if(!theirConfig)
-            theirConfig=new KConfig("kdialogd4rc"); // , KConfig::OnlyLocal);
+            theirConfig=new KConfig("kdialogd4rc", KConfig::NoGlobals);
 
         connect(new QSocketNotifier(itsFd, QSocketNotifier::Read, this),
                 SIGNAL(activated(int)), this, SLOT(newConnection()));
