diff -uNr KGtk-0.9.4/common/common.h KGtk-0.9.4-mod/common/common.h
--- KGtk-0.9.4/common/common.h	2007-10-31 22:33:19.000000000 +0200
+++ KGtk-0.9.4-mod/common/common.h	2008-01-28 17:39:09.000000000 +0200
@@ -165,7 +165,7 @@
 {
     do
     {
-        lockFd=open(getLockName(), O_WRONLY | O_CREAT | O_EXCL);
+        lockFd=open(getLockName(), O_WRONLY | O_CREAT | O_EXCL, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP);
         if (lockFd<0 && errno==EEXIST)
         {
             /* Hmm, lock file already exists. Is it stale? */
