Index: xf86-video-intel-2.4.3/src/i830_quirks.c
===================================================================
--- xf86-video-intel-2.4.3/src/i830_quirks.c.orig
+++ xf86-video-intel-2.4.3/src/i830_quirks.c
@@ -286,6 +286,8 @@ static i830_quirk i830_quirk_list[] = {
 
     /* Lenovo Napa TV (use dmi)*/
     { PCI_CHIP_I945_GM, 0x17aa, SUBSYS_ANY, quirk_lenovo_tv_dmi },
+    /* Lenovo T60/R60 needs pipe A force quirk (See LP: #243405) */
+    { PCI_CHIP_I945_GM, 0x17aa, 0x201a, quirk_pipea_force },
     /* Lenovo 3000 v200 */
     { PCI_CHIP_I965_GM, 0x17aa, 0x3c18, quirk_ignore_tv },
 
