--- src-QT4/kdiff3.desktop.orig	2009-03-09 14:51:10.000000000 +0100
+++ src-QT4/kdiff3.desktop	2009-03-09 14:57:01.000000000 +0100
@@ -33,6 +33,7 @@
 Exec=kdiff3 %i %m -caption "%c"
 Icon=kdiff3
 Type=Application
+Categories=Qt;KDE;Development;
 DocPath=kdiff3/index.html
 Comment=A File And Directory Comparison And Merge Tool
 Comment[bg]=Инструмент за сравняване и сливане на файлове и директории
