diff -Naur paprefs-0.9.6-orig/src/paprefs.desktop.in paprefs-0.9.6/src/paprefs.desktop.in
--- paprefs-0.9.6-orig/src/paprefs.desktop.in	2008-06-07 20:52:06.000000000 +0300
+++ paprefs-0.9.6/src/paprefs.desktop.in	2008-06-07 20:52:45.000000000 +0300
@@ -5,7 +5,7 @@
 _GenericName=Sound Server Preferences
 _Comment=View and modify the configuration of the local sound server
 Exec=paprefs
-Icon=preferences-desktop
+Icon=artsbuilderexecute
 StartupNotify=true
 Type=Application
 Categories=Settings;
diff -Naur paprefs-0.9.6-orig/src/paprefs.glade paprefs-0.9.6/src/paprefs.glade
--- paprefs-0.9.6-orig/src/paprefs.glade	2008-06-07 20:52:06.000000000 +0300
+++ paprefs-0.9.6/src/paprefs.glade	2008-06-07 20:55:15.000000000 +0300
@@ -10,7 +10,7 @@
   <property name="modal">False</property>
   <property name="resizable">False</property>
   <property name="destroy_with_parent">False</property>
-  <property name="icon_name">preferences-desktop</property>
+  <property name="icon_name">artsbuilderexecute</property>
   <property name="decorated">True</property>
   <property name="skip_taskbar_hint">False</property>
   <property name="skip_pager_hint">False</property>
@@ -42,7 +42,7 @@
 		<widget class="GtkImage" id="image19">
 		  <property name="visible">True</property>
 		  <property name="icon_size">6</property>
-		  <property name="icon_name">preferences-desktop</property>
+          <property name="icon_name">artsbuilderexecute</property>
 		  <property name="xalign">0.5</property>
 		  <property name="yalign">0.5</property>
 		  <property name="xpad">0</property>
