--- qcommon/files.c~	2008-09-19 22:28:29.000000000 +0300
+++ qcommon/files.c	2008-10-25 14:03:24.000000000 +0300
@@ -19,7 +19,7 @@
 */
 
 #include "qcommon.h"
-#include "../unix/glob.h"
+#include "../unix/glob.h"
 
 // define this to dissalow any data but the demo pak file
 //#define	NO_ADDONS
@@ -608,7 +608,7 @@
 ================
 FS_AddHomeAsGameDirectory
 
-Use ~/.codered/dir as fs_gamedir
+Use ~/.alienarena/dir as fs_gamedir
 ================
 */
 void FS_AddHomeAsGameDirectory (char *dir)
@@ -626,7 +626,7 @@
 		homedir = getenv("HOME");
 		if (!homedir)
 			return;
-		len = snprintf(gdir,sizeof(gdir),"%s/.codered/%s/", homedir, dir);
+		len = snprintf(gdir,sizeof(gdir),"%s/.alienarena/%s/", homedir, dir);
 	}
 
 	Com_Printf("using %s for writing\n",gdir);
