diff -Naur qalculate_applet-0.5.1-orig/CMakeLists.txt qalculate_applet-0.5.1/CMakeLists.txt
--- qalculate_applet-0.5.1-orig/CMakeLists.txt	2009-07-23 16:14:49.208571794 +0300
+++ qalculate_applet-0.5.1/CMakeLists.txt	2009-07-23 16:15:44.115572429 +0300
@@ -34,8 +34,8 @@
 		  
 install(FILES
 		  icons/128x128/qalculate-applet.png
-		  DESTINATION ${SHARE_INSTALL_PREFIX}/icons/oxygen/128x128/apps/)		
+		  DESTINATION ${ICON_INSTALL_DIR}/oxygen/128x128/apps/)		
 		  
 install(FILES
 		  icons/48x48/qalculate-applet.png
-		  DESTINATION ${SHARE_INSTALL_PREFIX}/icons/oxygen/48x48/apps/)	
+		  DESTINATION ${ICON_INSTALL_DIR}/oxygen/48x48/apps/)	
