--- sqliteman.desktop	2010-06-19 22:13:06.000000000 +0300
+++ sqliteman.desktop	2011-01-30 08:48:07.797000019 +0200
@@ -1,14 +1,16 @@
 [Desktop Entry]
 Type=Application
 Name=Sqliteman
-GenericName=Sqlite admin tool
+GenericName=Sqlite Admin Tool
 Comment=Administer and develop your Sqlite3 databases
 Icon=sqliteman
 Exec=sqliteman %f
 Terminal=false
 StartupNotify=true
 MimeType=application/x-sqlite3;
+Categories=Qt;Utility;Database
 
 # Translations
 GenericName[cs]=Sqlite administrace
+GenericName[tr]=Sqlite Yönetim Aracı
 
