Index: trunk/src/common/pkcs11_lib.c
===================================================================
--- trunk/src/common/pkcs11_lib.c	(revision 453)
+++ trunk/src/common/pkcs11_lib.c	(revision 477)
@@ -1445,5 +1445,6 @@
   DBG("logout user");
   rv = h->fl->C_Logout(h->session);
-  if (rv != CKR_OK && rv != CKR_USER_NOT_LOGGED_IN) {
+  if (rv != CKR_OK && rv != CKR_USER_NOT_LOGGED_IN
+	  && rv != CKR_FUNCTION_NOT_SUPPORTED) {
     set_error("C_Logout() failed: 0x%08lX", rv);
     return -1;
@@ -1451,5 +1452,5 @@
   DBG("closing the PKCS #11 session");
   rv = h->fl->C_CloseSession(h->session);
-  if (rv != CKR_OK) {
+  if (rv != CKR_OK && rv != CKR_FUNCTION_NOT_SUPPORTED) {
     set_error("C_CloseSession() failed: 0x%08lX", rv);
     return -1;
