Index: pardus-2.6/drivers/media/video/bt8xx/bttv-driver.c
===================================================================
--- pardus-2.6.orig/drivers/media/video/bt8xx/bttv-driver.c
+++ pardus-2.6/drivers/media/video/bt8xx/bttv-driver.c
@@ -2525,17 +2525,15 @@ static int bttv_do_ioctl(struct inode *i
 		struct video_mbuf *mbuf = arg;
 		unsigned int i;
 
-		mutex_lock(&fh->cap.lock);
 		retval = videobuf_mmap_setup(&fh->cap,gbuffers,gbufsize,
 					     V4L2_MEMORY_MMAP);
 		if (retval < 0)
-			goto fh_unlock_and_return;
+			return retval;
 		memset(mbuf,0,sizeof(*mbuf));
 		mbuf->frames = gbuffers;
 		mbuf->size   = gbuffers * gbufsize;
 		for (i = 0; i < gbuffers; i++)
 			mbuf->offsets[i] = i * gbufsize;
-		mutex_unlock(&fh->cap.lock);
 		return 0;
 	}
 	case VIDIOCMCAPTURE:
