--- DevIL-1.6.7/src-IL/src/Makefile.in~	2004-06-22 18:47:25.000000000 +0200
+++ DevIL-1.6.7/src-IL/src/Makefile.in	2007-01-19 11:35:04.000000000 +0100
@@ -312,7 +312,7 @@ libIL_la_SOURCES = il_alloc.c     il_doo
 
 AM_CFLAGS = @LIBIL_CFLAGS@
 libIL_la_LDFLAGS = -no-undefined @VERSIONINFO@
-libIL_la_LIBADD = @LIBPNG@ @LIBJPG@ @LIBTIFF@ @LIBGIF@ @LIBMNG@ @LIBLCMS@
+libIL_la_LIBADD = @LIBPNG@ @LIBJPG@ @LIBTIFF@ -lgif @LIBMNG@ @LIBLCMS@
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/src-IL/include
 all: all-am
 
