--- DevIL-1.6.7/configure~	2007-01-19 11:42:41.000000000 +0100
+++ DevIL-1.6.7/configure	2007-01-19 11:43:29.000000000 +0100
@@ -25811,7 +25811,7 @@ fi
     LIBALLEGRO=""
     HALLEGRO=""
   else
-    LIBALLEGRO=`allegro-config --libs`
+    LIBALLEGRO=`allegro-config --libs | sed 's/-lalleg_unsharable//'`
     HALLEGRO=`allegro-config --cflags`
   fi
 fi
--- DevIL-1.6.7/configure.in~	2004-06-28 09:39:33.000000000 +0200
+++ DevIL-1.6.7/configure.in	2007-01-19 11:43:42.000000000 +0100
@@ -627,7 +627,7 @@ if test "$use_allegro" = "yes" ; then
     LIBALLEGRO=""
     HALLEGRO=""
   else
-    LIBALLEGRO=`allegro-config --libs`
+    LIBALLEGRO=`allegro-config --libs | sed 's/-lalleg_unsharable//'`
     HALLEGRO=`allegro-config --cflags`
   fi
 fi
