--- include/IL/il.h.orig	2008-04-06 17:11:51.000000000 +0300
+++ include/IL/il.h	2008-04-06 18:37:08.000000000 +0300
@@ -494,15 +494,15 @@
 ILAPI ILvoid		ILAPIENTRY ilBindImage(ILuint Image);
 ILAPI ILboolean		ILAPIENTRY ilBlit(ILuint Source, ILint DestX, ILint DestY, ILint DestZ, ILuint SrcX, ILuint SrcY, ILuint SrcZ, ILuint Width, ILuint Height, ILuint Depth);
 ILAPI ILvoid		ILAPIENTRY ilClearColour(ILclampf Red, ILclampf Green, ILclampf Blue, ILclampf Alpha);
-ILAPI ILboolean		ILAPIENTRY ilClearImage(ILvoid);
-ILAPI ILuint		ILAPIENTRY ilCloneCurImage(ILvoid);
+ILAPI ILboolean		ILAPIENTRY ilClearImage(void);
+ILAPI ILuint		ILAPIENTRY ilCloneCurImage(void);
 ILAPI ILboolean		ILAPIENTRY ilCompressFunc(ILenum Mode);
 ILAPI ILboolean		ILAPIENTRY ilConvertImage(ILenum DestFormat, ILenum DestType);
 ILAPI ILboolean		ILAPIENTRY ilConvertPal(ILenum DestFormat);
 ILAPI ILboolean		ILAPIENTRY ilCopyImage(ILuint Src);
 ILAPI ILuint		ILAPIENTRY ilCopyPixels(ILuint XOff, ILuint YOff, ILuint ZOff, ILuint Width, ILuint Height, ILuint Depth, ILenum Format, ILenum Type, ILvoid *Data);
 ILAPI ILuint		ILAPIENTRY ilCreateSubImage(ILenum Type, ILuint Num);
-ILAPI ILboolean		ILAPIENTRY ilDefaultImage(ILvoid);
+ILAPI ILboolean		ILAPIENTRY ilDefaultImage(void);
 ILAPI ILvoid		ILAPIENTRY ilDeleteImages(ILsizei Num, const ILuint *Images);
 ILAPI ILboolean		ILAPIENTRY ilDisable(ILenum Mode);
 ILAPI ILboolean		ILAPIENTRY ilEnable(ILenum Mode);
@@ -513,16 +513,16 @@
 ILAPI ILvoid            ILAPIENTRY ilSetAlpha( ILdouble AlphaValue );
 ILAPI ILboolean		ILAPIENTRY ilGetBoolean(ILenum Mode);
 ILAPI ILvoid		ILAPIENTRY ilGetBooleanv(ILenum Mode, ILboolean *Param);
-ILAPI ILubyte*		ILAPIENTRY ilGetData(ILvoid);
+ILAPI ILubyte*		ILAPIENTRY ilGetData(void);
 ILAPI ILuint		ILAPIENTRY ilGetDXTCData(ILvoid *Buffer, ILuint BufferSize, ILenum DXTCFormat);
-ILAPI ILenum		ILAPIENTRY ilGetError(ILvoid);
+ILAPI ILenum		ILAPIENTRY ilGetError(void);
 ILAPI ILint             ILAPIENTRY ilGetInteger(ILenum Mode);
 ILAPI ILvoid		ILAPIENTRY ilGetIntegerv(ILenum Mode, ILint *Param);
-ILAPI ILuint		ILAPIENTRY ilGetLumpPos(ILvoid);
-ILAPI ILubyte*		ILAPIENTRY ilGetPalette(ILvoid);
+ILAPI ILuint		ILAPIENTRY ilGetLumpPos(void);
+ILAPI ILubyte*		ILAPIENTRY ilGetPalette(void);
 ILAPI const ILstring    ILAPIENTRY ilGetString(ILenum StringName);
 ILAPI ILvoid		ILAPIENTRY ilHint(ILenum Target, ILenum Mode);
-ILAPI ILvoid		ILAPIENTRY ilInit(ILvoid);
+ILAPI ILvoid		ILAPIENTRY ilInit(void);
 ILAPI ILboolean		ILAPIENTRY ilIsDisabled(ILenum Mode);
 ILAPI ILboolean		ILAPIENTRY ilIsEnabled(ILenum Mode);
 ILAPI ILboolean		ILAPIENTRY ilIsImage(ILuint Image);
@@ -537,7 +537,7 @@
 ILAPI ILboolean		ILAPIENTRY ilLoadPal(const ILstring FileName);
 ILAPI ILboolean		ILAPIENTRY ilOriginFunc(ILenum Mode);
 ILAPI ILboolean		ILAPIENTRY ilOverlayImage(ILuint Source, ILint XCoord, ILint YCoord, ILint ZCoord);
-ILAPI ILvoid		ILAPIENTRY ilPopAttrib(ILvoid);
+ILAPI ILvoid		ILAPIENTRY ilPopAttrib(void);
 ILAPI ILvoid		ILAPIENTRY ilPushAttrib(ILuint Bits);
 ILAPI ILvoid		ILAPIENTRY ilRegisterFormat(ILenum Format);
 ILAPI ILboolean		ILAPIENTRY ilRegisterLoad(const ILstring Ext, IL_LOADPROC Load);
@@ -549,9 +549,9 @@
 ILAPI ILvoid		ILAPIENTRY ilRegisterType(ILenum Type);
 ILAPI ILboolean		ILAPIENTRY ilRemoveLoad(const ILstring Ext);
 ILAPI ILboolean		ILAPIENTRY ilRemoveSave(const ILstring Ext);
-ILAPI ILvoid		ILAPIENTRY ilResetMemory(ILvoid);
-ILAPI ILvoid		ILAPIENTRY ilResetRead(ILvoid);
-ILAPI ILvoid		ILAPIENTRY ilResetWrite(ILvoid);
+ILAPI ILvoid		ILAPIENTRY ilResetMemory(void);
+ILAPI ILvoid		ILAPIENTRY ilResetRead(void);
+ILAPI ILvoid		ILAPIENTRY ilResetWrite(void);
 ILAPI ILboolean		ILAPIENTRY ilSave(ILenum Type, const ILstring FileName);
 ILAPI ILuint		ILAPIENTRY ilSaveF(ILenum Type, ILHANDLE File);
 ILAPI ILboolean		ILAPIENTRY ilSaveImage(const ILstring FileName);
@@ -565,7 +565,7 @@
 ILAPI ILvoid		ILAPIENTRY ilSetRead(fOpenRProc, fCloseRProc, fEofProc, fGetcProc, fReadProc, fSeekRProc, fTellRProc);
 ILAPI ILvoid		ILAPIENTRY ilSetString(ILenum Mode, const char *String);
 ILAPI ILvoid		ILAPIENTRY ilSetWrite(fOpenWProc, fCloseWProc, fPutcProc, fSeekWProc, fTellWProc, fWriteProc);
-ILAPI ILvoid		ILAPIENTRY ilShutDown(ILvoid);
+ILAPI ILvoid		ILAPIENTRY ilShutDown(void);
 ILAPI ILboolean		ILAPIENTRY ilTexImage(ILuint Width, ILuint Height, ILuint Depth, ILubyte Bpp, ILenum Format, ILenum Type, ILvoid *Data);
 ILAPI ILboolean		ILAPIENTRY ilTypeFunc(ILenum Mode);
 
--- include/IL/ilu.h.orig	2008-04-06 18:39:28.000000000 +0300
+++ include/IL/ilu.h	2008-04-08 23:18:28.000000000 +0300
@@ -132,36 +132,36 @@
 
 
 // ImageLib Utility Functions
-ILAPI ILboolean			ILAPIENTRY iluAlienify(ILvoid);
+ILAPI ILboolean			ILAPIENTRY iluAlienify(void);
 ILAPI ILboolean			ILAPIENTRY iluBlurAvg(ILuint Iter);
 ILAPI ILboolean			ILAPIENTRY iluBlurGaussian(ILuint Iter);
-ILAPI ILboolean			ILAPIENTRY iluBuildMipmaps(ILvoid);
-ILAPI ILuint			ILAPIENTRY iluColoursUsed(ILvoid);
+ILAPI ILboolean			ILAPIENTRY iluBuildMipmaps(void);
+ILAPI ILuint			ILAPIENTRY iluColoursUsed(void);
 ILAPI ILboolean			ILAPIENTRY iluCompareImage(ILuint Comp);
 ILAPI ILboolean			ILAPIENTRY iluContrast(ILfloat Contrast);
 ILAPI ILboolean			ILAPIENTRY iluCrop(ILuint XOff, ILuint YOff, ILuint ZOff, ILuint Width, ILuint Height, ILuint Depth);
 ILAPI ILvoid			ILAPIENTRY iluDeleteImage(ILuint Id);
-ILAPI ILboolean			ILAPIENTRY iluEdgeDetectE(ILvoid);
-ILAPI ILboolean			ILAPIENTRY iluEdgeDetectP(ILvoid);
-ILAPI ILboolean			ILAPIENTRY iluEdgeDetectS(ILvoid);
-ILAPI ILboolean			ILAPIENTRY iluEmboss(ILvoid);
+ILAPI ILboolean			ILAPIENTRY iluEdgeDetectE(void);
+ILAPI ILboolean			ILAPIENTRY iluEdgeDetectP(void);
+ILAPI ILboolean			ILAPIENTRY iluEdgeDetectS(void);
+ILAPI ILboolean			ILAPIENTRY iluEmboss(void);
 ILAPI ILboolean			ILAPIENTRY iluEnlargeCanvas(ILuint Width, ILuint Height, ILuint Depth);
 ILAPI ILboolean			ILAPIENTRY iluEnlargeImage(ILfloat XDim, ILfloat YDim, ILfloat ZDim);
-ILAPI ILboolean			ILAPIENTRY iluEqualize(ILvoid);
+ILAPI ILboolean			ILAPIENTRY iluEqualize(void);
 ILAPI const ILstring	ILAPIENTRY iluErrorString(ILenum Error);
-ILAPI ILboolean			ILAPIENTRY iluFlipImage(ILvoid);
+ILAPI ILboolean			ILAPIENTRY iluFlipImage(void);
 ILAPI ILboolean			ILAPIENTRY iluGammaCorrect(ILfloat Gamma);
-ILAPI ILuint			ILAPIENTRY iluGenImage(ILvoid);
+ILAPI ILuint			ILAPIENTRY iluGenImage(void);
 ILAPI ILvoid			ILAPIENTRY iluGetImageInfo(ILinfo *Info);
 ILAPI ILint				ILAPIENTRY iluGetInteger(ILenum Mode);
 ILAPI ILvoid			ILAPIENTRY iluGetIntegerv(ILenum Mode, ILint *Param);
 ILAPI const ILstring	ILAPIENTRY iluGetString(ILenum StringName);
 ILAPI ILvoid			ILAPIENTRY iluImageParameter(ILenum PName, ILenum Param);
-ILAPI ILvoid			ILAPIENTRY iluInit(ILvoid);
-ILAPI ILboolean			ILAPIENTRY iluInvertAlpha(ILvoid);
+ILAPI ILvoid			ILAPIENTRY iluInit(void);
+ILAPI ILboolean			ILAPIENTRY iluInvertAlpha(void);
 ILAPI ILuint			ILAPIENTRY iluLoadImage(const ILstring FileName);
-ILAPI ILboolean			ILAPIENTRY iluMirror(ILvoid);
-ILAPI ILboolean			ILAPIENTRY iluNegative(ILvoid);
+ILAPI ILboolean			ILAPIENTRY iluMirror(void);
+ILAPI ILboolean			ILAPIENTRY iluNegative(void);
 ILAPI ILboolean			ILAPIENTRY iluNoisify(ILclampf Tolerance);
 ILAPI ILboolean			ILAPIENTRY iluPixelize(ILuint PixSize);
 ILAPI ILvoid			ILAPIENTRY iluRegionfv(ILpointf *Points, ILuint n);
@@ -174,7 +174,7 @@
 ILAPI ILboolean			ILAPIENTRY iluScale(ILuint Width, ILuint Height, ILuint Depth);
 ILAPI ILboolean			ILAPIENTRY iluScaleColours(ILfloat r, ILfloat g, ILfloat b);
 ILAPI ILboolean			ILAPIENTRY iluSharpen(ILfloat Factor, ILuint Iter);
-ILAPI ILboolean			ILAPIENTRY iluSwapColours(ILvoid);
+ILAPI ILboolean			ILAPIENTRY iluSwapColours(void);
 ILAPI ILboolean			ILAPIENTRY iluWave(ILfloat Angle);
 
 #define iluColorsUsed	iluColoursUsed
