--- Makefile.orig	2006-11-13 14:43:14.029340674 +0200
+++ Makefile	2006-11-13 14:47:56.105458586 +0200
@@ -74,8 +74,7 @@
 INSTALL := $(shell which install)
 DEPMOD  := $(shell which depmod || which /sbin/depmod || echo "true You should now run depmod")
 
-# Get VERSION_CODE (from version.h in kernel source directory)
-VERSION_CODE := $(shell awk '/[ 	]*\#[ 	]*define[ 	]*LINUX_VERSION_CODE[ 	]*/ { print $$3 }' $(LINUX_DIR)/include/linux/version.h|tail -n 1)
+VERSION_CODE := 132608
 
 ifeq ($(shell if [ $(VERSION_CODE) -ge 132608 ]; then echo y; fi),y)
 MODULE_NAME := quickcam.ko	# for 2.6.x
