3c1e3896344063273715b332b1c0534deb9b286c

Index: 2008-devel/drivers/ata/libata-acpi.c
===================================================================
--- 2008-devel.orig/drivers/ata/libata-acpi.c
+++ 2008-devel/drivers/ata/libata-acpi.c
@@ -195,6 +195,10 @@ static void ata_acpi_handle_hotplug(stru
 		/* This device does not support hotplug */
 		return;
 
+	if (event == ACPI_NOTIFY_BUS_CHECK ||
+	    event == ACPI_NOTIFY_DEVICE_CHECK)
+		status = acpi_evaluate_integer(handle, "_STA", NULL, &sta);
+
 	spin_lock_irqsave(ap->lock, flags);
 
 	switch (event) {
@@ -202,7 +206,6 @@ static void ata_acpi_handle_hotplug(stru
 	case ACPI_NOTIFY_DEVICE_CHECK:
 		ata_ehi_push_desc(ehi, "ACPI event");
 
-		status = acpi_evaluate_integer(handle, "_STA", NULL, &sta);
 		if (ACPI_FAILURE(status)) {
 			ata_port_printk(ap, KERN_ERR,
 				"acpi: failed to determine bay status (0x%x)\n",
