
@DPATCH@

diff -Naur file-4.20.orig/magic/Magdir/commands file-4.20/magic/Magdir/commands
--- file-4.20.orig/magic/Magdir/commands	2007-01-19 19:28:01.000000000 +0000
+++ file-4.20/magic/Magdir/commands	2007-03-27 12:23:14.000000000 +0000
@@ -28,7 +28,7 @@
 0	string/b	#!\ /bin/awk		awk script text executable
 0	string/b	#!\ /usr/bin/awk	awk script text executable
 # update to distinguish from *.vcf files
-0	regex		BEGIN[[:space:]]*[{]	awk script text
+#0	regex		BEGIN[[:space:]]*[{]	awk script text
 
 # AT&T Bell Labs' Plan 9 shell
 0	string/b	#!\ /bin/rc	Plan 9 rc shell script text executable
