--- xfig.3.2.5-alpha5.orig/main.c	2006-04-05 15:29:50.000000000 +0300
+++ xfig.3.2.5-alpha5/main.c	2006-04-05 15:37:12.000000000 +0300
@@ -251,9 +251,9 @@
     {"pageborder", "Color",   XtRString, sizeof(char *),
       XtOffset(appresPtr, pageborder), XtRString, (caddr_t) "lightblue"},
     {"browser", "Browser", XtRString, sizeof(char *),
-      XtOffset(appresPtr, browser), XtRString, (caddr_t) "netscape"},
+      XtOffset(appresPtr, browser), XtRString, (caddr_t) "firefox"},
     {"pdfviewer", "Viewer", XtRString, sizeof(char *),
-      XtOffset(appresPtr, pdf_viewer), XtRString, (caddr_t) "acroread"},
+      XtOffset(appresPtr, pdf_viewer), XtRString, (caddr_t) "kpdf"},
     {"spinner_delay", "spinnerDelay",   XtRInt, sizeof(int),
       XtOffset(appresPtr, spinner_delay), XtRImmediate, (caddr_t) 500},
     {"spinner_rate", "spinnerRate",   XtRInt, sizeof(int),
