--- configure.orig	2006-12-20 23:16:57.000000000 +0200
+++ configure	2007-11-24 13:00:53.000000000 +0200
@@ -4506,7 +4506,7 @@
 
 if test "x$LUAPFX" != x ; then
     LUA_CFLAGS="-I$LUAPFX"
-    LUA_LIBS="-L$LUAPFX -llua -llualib"
+    LUA_LIBS="-L$LUAPFX -llua"
 else
     for ac_prog in lua-config lua-config50 lua-config5.0
 do
@@ -4870,13 +4870,13 @@
 fi
 
 
-            { echo "$as_me:$LINENO: checking for luaopen_base in -llualib" >&5
-echo $ECHO_N "checking for luaopen_base in -llualib... $ECHO_C" >&6; }
+            { echo "$as_me:$LINENO: checking for luaopen_base in -llua" >&5
+echo $ECHO_N "checking for luaopen_base in -llua... $ECHO_C" >&6; }
 if test "${ac_cv_lib_lualib_luaopen_base+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-llualib $LUA_LIBS $LIBS"
+LIBS="-llua $LUA_LIBS $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -4948,7 +4948,7 @@
 { echo "$as_me:$LINENO: result: $ac_cv_lib_lualib_luaopen_base" >&5
 echo "${ECHO_T}$ac_cv_lib_lualib_luaopen_base" >&6; }
 if test $ac_cv_lib_lualib_luaopen_base = yes; then
-  LUA_LIBS="$LUA_LIBS -llualib"
+  LUA_LIBS="$LUA_LIBS -llua"
 else
   { echo "$as_me:$LINENO: checking for luaopen_base in -llualib50" >&5
 echo $ECHO_N "checking for luaopen_base in -llualib50... $ECHO_C" >&6; }
@@ -5380,13 +5380,13 @@
 fi
 
 
-            { echo "$as_me:$LINENO: checking for luaopen_base in -llualib" >&5
-echo $ECHO_N "checking for luaopen_base in -llualib... $ECHO_C" >&6; }
+            { echo "$as_me:$LINENO: checking for luaopen_base in -llua" >&5
+echo $ECHO_N "checking for luaopen_base in -llua... $ECHO_C" >&6; }
 if test "${ac_cv_lib_lualib_luaopen_base+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-llualib $LUA_LIBS $LIBS"
+LIBS="-llua $LUA_LIBS $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -5458,7 +5458,7 @@
 { echo "$as_me:$LINENO: result: $ac_cv_lib_lualib_luaopen_base" >&5
 echo "${ECHO_T}$ac_cv_lib_lualib_luaopen_base" >&6; }
 if test $ac_cv_lib_lualib_luaopen_base = yes; then
-  LUA_LIBS="$LUA_LIBS -llualib"
+  LUA_LIBS="$LUA_LIBS -llua"
 else
   { echo "$as_me:$LINENO: checking for luaopen_base in -llualib50" >&5
 echo $ECHO_N "checking for luaopen_base in -llualib50... $ECHO_C" >&6; }
