diff -Nur hplip-3.9.4b-old/prnt/hpijs/hpijs.cpp hplip-3.9.4b/prnt/hpijs/hpijs.cpp
--- hplip-3.9.4b-old/prnt/hpijs/hpijs.cpp	2009-05-04 16:56:44.000000000 +0300
+++ hplip-3.9.4b/prnt/hpijs/hpijs.cpp	2009-05-04 16:57:10.000000000 +0300
@@ -594,11 +594,11 @@
 			case WARN_LOW_INK_YELLOW:
 			case WARN_LOW_INK_MULTIPLE_PENS:
                         {
-			   BUG ("STATE: marker-supply-low-warning\n");
+			   BUG ("STATE: +marker-supply-low-warning\n");
                            break;
                         }
 			default:
-			   BUG ("STATE: -marker-supply-low-warning");
+			   BUG ("STATE: -marker-supply-low-warning\n");
 		}
     }
 
