diff -up hplip-2.8.7/prnt/backend/hp.c.dbus hplip-2.8.7/prnt/backend/hp.c
--- hplip-2.8.7/prnt/backend/hp.c.dbus	2008-12-11 10:44:30.000000000 +0000
+++ hplip-2.8.7/prnt/backend/hp.c	2008-12-11 10:46:15.000000000 +0000
@@ -489,6 +489,9 @@ static int device_event(const char *dev,
     DBusMessage * msg = NULL;
     int id = atoi(jobid);
 
+    if (dbus_conn == NULL)
+	return 0;
+
     msg = dbus_message_new_signal(DBUS_PATH, DBUS_INTERFACE, "Event");
 
     if (NULL == msg)
