diff -Nur a/src/qp_fswrap.cpp b/src/qp_fswrap.cpp
--- a/src/qp_fswrap.cpp	2005-07-07 00:29:54.000000000 +0300
+++ b/src/qp_fswrap.cpp	2006-07-13 10:21:10.000000000 +0300
@@ -1002,7 +1002,7 @@
 /*---FAT32 WRAPPER---------------------------------------------------------------*/
 QString QP_FSFat32::_get_label(PedPartition *part) 
 {
-    char buffer[PED_SECTOR_SIZE];
+    char buffer[PED_SECTOR_SIZE_DEFAULT];
     char label[12];
     
     if (!QP_FSWrap::read_sector(part, 0, 1, buffer)) 
