diff -ur xorg-proto-7.2.orig/inputproto-1.4.2.1/XI.h xorg-proto-7.2/inputproto-1.4.2/XI.h
--- xorg-proto-7.2.orig/inputproto-1.4.2.1/XI.h	2007-04-24 15:40:11.000000000 +0300
+++ xorg-proto-7.2/inputproto-1.4.2.1/XI.h	2007-07-14 13:52:15.000000000 +0300
@@ -215,6 +215,13 @@
 #define Relative                0
 #define Absolute                1
 
+/* Merged from Metrolink tree for XINPUT stuff */
+#define TS_Raw                  57
+#define TS_Scaled               58
+#define SendCoreEvents          59
+#define DontSendCoreEvents      60
+/* End of merged section       */
+
 #define ProximityState          (1L << 1)
 #define InProximity             (0L << 1)
 #define OutOfProximity          (1L << 1)
diff -ur xorg-proto-7.2.orig/inputproto-1.4.2.1/XIproto.h xorg-proto-7.2/inputproto-1.4.2/XIproto.h
--- xorg-proto-7.2.orig/inputproto-1.4.2.1/XIproto.h	2007-04-24 14:46:15.000000000 +0300
+++ xorg-proto-7.2/inputproto-1.4.2.1/XIproto.h	2007-07-14 13:52:15.000000000 +0300
@@ -1386,6 +1386,8 @@
      CARD32         button_threshold;
 } xDeviceAbsCalibCtl;
 
+#define xDeviceTSCalibrationCtl xDeviceAbsCalibCtl
+
 typedef struct {
      CARD16         control B16;
      CARD16         length B16;
