Index: 2ManDVD/mainfrm.cpp
===================================================================
--- 2ManDVD.orig/mainfrm.cpp
+++ 2ManDVD/mainfrm.cpp
@@ -299,18 +299,18 @@ mainfrm::mainfrm(QWidget *parent) : QMai
         QTextCodec::setCodecForCStrings(QTextCodec::codecForName("UTF-8"));
         QTextCodec::setCodecForTr(QTextCodec::codecForName("UTF-8"));
 
-	qDebug() << "LOCALE : " << QLocale::system().name().left(2) << " " << qApp->applicationDirPath();
+	qDebug() << "LOCALE : " << QLocale::system().name().left(2) << " " << QString("/usr/share/2mandvd/locale");
 	if ( QLocale::system().name().left(2) != "fr" )
 	{
 		QTranslator *qtTranslator = new QTranslator(this);
 
-		if (qtTranslator->load("2mandvd_" + QLocale::system().name().left(2), qApp->applicationDirPath()) == true )
+		if (qtTranslator->load("2mandvd_" + QLocale::system().name().left(2), QString("/usr/share/2mandvd/locale")) == true )
 		{
 			qApp->installTranslator(qtTranslator);
 		}
 		else
 		{
-			if (qtTranslator->load("2mandvd_en", qApp->applicationDirPath()) == true )
+			if (qtTranslator->load("2mandvd_en",QString("/usr/share/2mandvd/locale")) == true )
 			{
 				qApp->installTranslator(qtTranslator);
 			}
