Index: artha-1.0.2/data/artha.desktop
===================================================================
--- artha-1.0.2.orig/data/artha.desktop
+++ artha-1.0.2/data/artha.desktop
@@ -2,10 +2,12 @@
 Version=1.0
 Name=Artha
 GenericName=Thesaurus
+GenericName[tr]=Eşanlamlılar Sözlüğü
 Comment=Handy thesaurus based on WordNet
+Comment[tr]=Wordnet tabanlı kullanışlı eşanlamlılar sözlüğü
 Exec=artha
 Icon=artha
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=Utility;Languages;Literature;
+Categories=Utility;Languages;Literature;Education;
