Index: pykde4-4.7.1/CMakeLists.txt
===================================================================
--- pykde4-4.7.1.orig/CMakeLists.txt
+++ pykde4-4.7.1/CMakeLists.txt
@@ -73,6 +73,7 @@ INCLUDE_DIRECTORIES(
     ${QT_QTWEBKIT_INCLUDE_DIR}
     ${KDE4_INCLUDE_DIR}
     ${KDE4_INCLUDE_DIR}/solid
+    ${QT_INCLUDE_DIR}/phonon
     ${KDE4_INCLUDE_DIR}/phonon
     ${KDE4_INCLUDE_DIR}/kio
     ${KDE4_INCLUDE_DIR}/kdeprint
