diff -Naur kde-l10n-fr-4.1.1-orig/docs/CMakeLists.txt kde-l10n-fr-4.1.1/docs/CMakeLists.txt
--- kde-l10n-fr-4.1.1-orig/docs/CMakeLists.txt	2008-09-05 13:56:45.000000000 +0300
+++ kde-l10n-fr-4.1.1/docs/CMakeLists.txt	2008-09-05 13:56:58.000000000 +0300
@@ -5,7 +5,7 @@
 add_subdirectory( kdebase-runtime )
 add_subdirectory( kdebase-workspace )
 add_subdirectory( kdeedu )
-add_subdirectory( kdegames )
+#add_subdirectory( kdegames )
 add_subdirectory( kdegraphics )
 add_subdirectory( kdelibs )
 add_subdirectory( kdemultimedia )
