diff -Naur mysql-5.0.88.orig/strings/longlong2str-x86.s mysql-5.0.88/strings/longlong2str-x86.s
--- mysql-5.0.88.orig/strings/longlong2str-x86.s	2009-12-12 22:17:18.000000000 +0200
+++ mysql-5.0.88/strings/longlong2str-x86.s	2009-12-12 22:18:59.000000000 +0200
@@ -214,3 +214,4 @@
 
 .L10end:
 	.size	 longlong10_to_str,.L10end-longlong10_to_str
+	.section	.note.GNU-stack,"",@progbits
diff -Naur mysql-5.0.88.orig/strings/my_strtoll10-x86.s mysql-5.0.88/strings/my_strtoll10-x86.s
--- mysql-5.0.88.orig/strings/my_strtoll10-x86.s	2009-12-12 22:17:18.000000000 +0200
+++ mysql-5.0.88/strings/my_strtoll10-x86.s	2009-12-12 22:19:23.000000000 +0200
@@ -416,3 +416,5 @@
         .comm   end_ptr,120,32
         .comm   error,120,32
 	.ident	"Monty"
+
+	.section	.note.GNU-stack,"",@progbits
diff -Naur mysql-5.0.88.orig/strings/strings-x86.s mysql-5.0.88/strings/strings-x86.s
--- mysql-5.0.88.orig/strings/strings-x86.s	2009-12-12 22:17:18.000000000 +0200
+++ mysql-5.0.88/strings/strings-x86.s	2009-12-12 22:20:23.000000000 +0200
@@ -414,3 +414,5 @@
 	ret
 .strxmov_end:
 	.size	 strxmov,.strxmov_end-strxmov
+
+	.section	.note.GNU-stack,"",@progbits
