diff -ur gtk-qt-engine.orig/CMakeLists.txt gtk-qt-engine/CMakeLists.txt
--- gtk-qt-engine.orig/CMakeLists.txt	2006-10-19 14:43:58.000000000 +0300
+++ gtk-qt-engine/CMakeLists.txt	2006-11-25 19:51:22.000000000 +0200
@@ -71,7 +71,7 @@
 
 SET(INSTALL_PATH_KCONTROL_MODULES "${KDE3_LIB_DIR}" CACHE PATH "The directory in which to install KControl plugins (eg. /usr/lib)")
 MESSAGE(STATUS "Installing the KControl plugins to: ${INSTALL_PATH_KCONTROL_MODULES}")
-
+SET(CMAKE_INSTALL_PREFIX "$ENV{CMAKE_INSTALL_PREFIX}")
 MESSAGE(STATUS "Installing everything else to: ${CMAKE_INSTALL_PREFIX}")
 
 # Setup definitions and directories to be used in all subdirectories
@@ -91,4 +91,4 @@
   IMMEDIATE @ONLY)
 
 ADD_CUSTOM_TARGET(uninstall
-  "${CMAKE_COMMAND}" -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
\ Dosya sonunda yenisatır yok.
+  "${CMAKE_COMMAND}" -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
