diff -Nur mplayer-20070524-old/configure mplayer-20070524/configure
--- mplayer-20070524-old/configure	2007-05-24 13:59:24.852526780 +0300
+++ mplayer-20070524/configure	2007-05-24 14:00:18.539365139 +0300
@@ -7383,7 +7383,7 @@
 # to use its own copy of the library)
 echocheck "DVD support (libdvdnav)"
 if test "$_dvdnav" = auto ; then
-  $_dvdnavconfig --version --minilibs >> $TMPLOG 2>&1 || _dvdnav=no
+  $_dvdnavconfig --version >> $TMPLOG 2>&1 || _dvdnav=no
 fi
 if test "$_dvdnav" = auto ; then
   cat > $TMPC <<EOF
@@ -7405,7 +7405,7 @@
   _dvdnav_version=`$_dvdnavconfig --version | sed "s/\.//g"`
   _def_dvdnav_version="#define DVDNAVVERSION $_dvdnav_version"
   _inc_extra="$_inc_extra `$_dvdnavconfig --cflags`"
-  _ld_extra="$_ld_extra `$_dvdnavconfig --minilibs`"
+  _ld_extra="$_ld_extra `$_dvdnavconfig --libs`"
   _inputmodules="dvdnav $_inputmodules"
 else
   _def_dvdnav='#undef USE_DVDNAV'
