diff -Nur mplayer-20070916-old/etc/codecs.conf mplayer-20070916/etc/codecs.conf
--- mplayer-20070916-old/etc/codecs.conf	2007-09-17 02:04:24.859575746 +0300
+++ mplayer-20070916/etc/codecs.conf	2007-09-17 02:18:41.956277242 +0300
@@ -3200,3 +3200,11 @@
   fourcc "MP3 "      ; used in .nsv files
   fourcc "LAME"      ; used in mythtv .nuv files
   driver hwmpa
+
+audiocodec ffeac3
+    info "FFmpeg EAC3"
+    status buggy
+    format 0x2002
+    driver ffmpeg
+    dll "E-AC3"
+
