diff -Nur mplayer-20070203-old/configure mplayer-20070203/configure
--- mplayer-20070203-old/configure	2007-02-04 13:07:25.133702545 +0200
+++ mplayer-20070203/configure	2007-02-04 13:08:06.101474497 +0200
@@ -1380,29 +1380,6 @@
 EOF
     die "Bad gcc version"
   fi
-else
-cat <<EOF
-
-******************************************************************************
-
-Hmm. You really want to compile MPlayer with an *UNSUPPORTED* C compiler?
-Ok. You know. Do it.
-
-DO NOT SEND BUGREPORTS OR COMPLAIN, it's *YOUR* compiler's fault!
-Get ready for mysterious crashes, no-picture bugs, strange noises... REALLY!
-Lame which is used by mencoder produces weird errors, too.
-
-If you have any problem, install a GCC 2.95.x or 3.x version and try again.
-If the problem _still_ exists, then read DOCS/HTML/en/bugreports.html !
-
-  *** DO NOT SEND BUG REPORTS OR COMPLAIN it's *YOUR* compiler's fault! ***
-
-******************************************************************************
-
-EOF
-
-read _answer
-
 fi
 echocheck "host cc"
 test "$_host_cc" || _host_cc=$_cc
