--- klipper/klipperrc.desktop	2008/05/19 14:36:11	1.1
+++ klipper/klipperrc.desktop	2008/05/19 14:39:55
@@ -276,7 +276,7 @@
 Enabled=true
 
 [Action_1/Command_2]
-Commandline=ps x |grep -q '[m]ozilla' && mozilla -remote openURL(%s, new-window) || mozilla %s
+Commandline=ps x |grep -q '[m]ozilla' && "mozilla -remote openURL(%s, new-window)" || mozilla %s
 Icon=mozilla
 Description=Open with &Mozilla
 Description[af]=Maak oop met Mozilla
@@ -423,7 +423,7 @@
 Enabled=true
 
 [Action_1/Command_4]
-Commandline=ps x |grep -q '[f]irefox' && firefox -remote openURL(%s, new-window) || firefox %s
+Commandline=ps x |grep -q '[f]irefox' && "firefox -remote openURL(%s, new-window)" || firefox "%s"
 Icon=firefox
 Description=Open with &Firefox
 Description[af]=Maak oop met Firefox
