Index: src/contactlist.cpp
===================================================================
--- src/contactlist.cpp	(revision 950963)
+++ src/contactlist.cpp	(revision 950964)
@@ -335,7 +335,7 @@
     QList<Contact *> rcontacts;
     QHash<uint, Contact *>::const_iterator i;
 
-    for (i = d->contacts.constBegin(); i != d->contacts.end(); i++)
+    for (i = d->contacts.constBegin(); i != d->contacts.constEnd(); i++)
         if (i.value()->subscriptionStatus() == Contact::Subscribed)
             rcontacts << i.value();
 
@@ -347,7 +347,7 @@
     QList<Contact *> rcontacts;
     QHash<uint, Contact *>::const_iterator i;
 
-    for (i = d->contacts.constBegin(); i != d->contacts.end(); i++)
+    for (i = d->contacts.constBegin(); i != d->contacts.constEnd(); i++)
         if (i.value()->authorizationStatus() == Contact::Authorized)
             rcontacts << i.value();
 
@@ -359,7 +359,7 @@
     QList<Contact *> rcontacts;
     QHash<uint, Contact *>::const_iterator i;
 
-    for (i = d->contacts.constBegin(); i != d->contacts.end(); i++)
+    for (i = d->contacts.constBegin(); i != d->contacts.constEnd(); i++)
         if (i.value()->isBlocked())
             rcontacts << i.value();
 
@@ -371,7 +371,7 @@
     QList<Contact *> rcontacts;
     QHash<uint, Contact *>::const_iterator i;
 
-    for (i = d->contacts.constBegin(); i != d->contacts.end(); i++)
+    for (i = d->contacts.constBegin(); i != d->contacts.constEnd(); i++)
         if (i.value()->hidingFrom())
             rcontacts << i.value();
 
