Index: pyexiv2-0.3.0/doc/SConscript
===================================================================
--- pyexiv2-0.3.0.orig/doc/SConscript
+++ pyexiv2-0.3.0/doc/SConscript
@@ -20,7 +20,9 @@ output = [os.path.join(outdir, doc + '.h
           for doc in sphinx.builder.get_outdated_docs()]
 
 def build_doc(target, source, env):
-    sphinx.build(all_files=False, filenames=[])
+    # The 'all_files' parameter was renamed to 'force_all'
+    # in sphinx 1.0
+    sphinx.build(force_all=False, filenames=[])
     return sphinx.statuscode
 
 env = Environment()
