--- src/miracle/Makefile~	2008-08-06 09:32:06.000000000 +0200
+++ src/miracle/Makefile	2008-08-06 09:36:30.000000000 +0200
@@ -30,7 +30,7 @@
 
 CFLAGS += -I.. $(RDYNAMIC)
 
-LDFLAGS += -L../valerie -lvalerie -L../framework -lmlt 
+LDFLAGS += -L../valerie -lvalerie -L../framework -lmlt -pthread
 
 SRCS := $(OBJS:.o=.c)
 
--- src/modules/plus/Makefile~	2008-01-23 11:04:08.000000000 +0100
+++ src/modules/plus/Makefile	2008-08-06 09:52:33.000000000 +0200
@@ -11,7 +11,7 @@
 
 CFLAGS += -I../..
 
-LDFLAGS+=-L../../framework -lmlt
+LDFLAGS+=-L../../framework -lmlt -lm
 
 SRCS := $(OBJS:.o=.c)
 
--- src/modules/jackrack/Makefile~	2008-01-23 11:04:13.000000000 +0100
+++ src/modules/jackrack/Makefile	2008-08-06 09:59:59.000000000 +0200
@@ -21,7 +21,7 @@
 LDFLAGS += `xml2-config --libs`
 LDFLAGS += `pkg-config glib-2.0 --libs`
 
-LDFLAGS+=-L../../framework -lmlt
+LDFLAGS+=-L../../framework -lmlt -ldl
 
 SRCS := $(OBJS:.o=.c)
 
--- src/modules/core/Makefile~	2008-01-23 11:04:09.000000000 +0100
+++ src/modules/core/Makefile	2008-08-06 10:07:10.000000000 +0200
@@ -31,7 +31,7 @@
 
 CFLAGS += -I../..
 
-LDFLAGS+=-L../../framework -lmlt
+LDFLAGS+=-L../../framework -lmlt -lm -pthread
 
 SRCS := $(OBJS:.o=.c)
 
--- src/modules/normalize/Makefile~	2008-01-23 11:04:11.000000000 +0100
+++ src/modules/normalize/Makefile	2008-08-06 10:14:11.000000000 +0200
@@ -7,7 +7,7 @@
 
 CFLAGS += -I../..
 
-LDFLAGS+=-L../../framework -lmlt
+LDFLAGS+=-L../../framework -lmlt -lm
 
 SRCS := $(OBJS:.o=.c)
 
--- src/modules/sdl/Makefile~	2008-01-23 11:04:07.000000000 +0100
+++ src/modules/sdl/Makefile	2008-08-06 10:21:01.000000000 +0200
@@ -20,7 +20,7 @@
 
 LDFLAGS +=`sdl-config --libs`
 
-LDFLAGS +=-L../../framework -lmlt
+LDFLAGS +=-L../../framework -lmlt -pthread
 
 ifeq ($(WITH_SDL_IMAGE),1)
 OBJS += producer_sdl_image.o
--- src/modules/dv/Makefile~	2008-01-23 11:04:13.000000000 +0100
+++ src/modules/dv/Makefile	2008-08-06 10:28:06.000000000 +0200
@@ -10,7 +10,7 @@
 
 LDFLAGS += `pkg-config --libs libdv`
 
-LDFLAGS+=-L../../framework -lmlt
+LDFLAGS+=-L../../framework -lmlt -pthread
 
 SRCS := $(OBJS:.o=.c)
 
--- src/modules/avformat/Makefile.underlink	2008-05-09 03:12:55.000000000 +0200
+++ src/modules/avformat/Makefile	2008-08-06 10:58:28.000000000 +0200
@@ -17,7 +17,7 @@
 
 LDFLAGS+=-L../../framework
 
-LDFLAGS+=-lavformat$(AVFORMAT_SUFFIX) -lavcodec$(AVFORMAT_SUFFIX) -lavutil$(AVFORMAT_SUFFIX) -lavdevice$(AVFORMAT_SUFFIX) $(EXTRA_LIBS) -lmlt
+LDFLAGS+=-lavformat$(AVFORMAT_SUFFIX) -lavcodec$(AVFORMAT_SUFFIX) -lavutil$(AVFORMAT_SUFFIX) -lavdevice$(AVFORMAT_SUFFIX) $(EXTRA_LIBS) -lmlt -pthread
 
 ifdef SWSCALE
 	CFLAGS+=-DSWSCALE
--- src/modules/effectv/Makefile~	2008-02-04 08:24:19.000000000 +0100
+++ src/modules/effectv/Makefile	2008-08-06 11:05:04.000000000 +0200
@@ -9,7 +9,7 @@
 
 CFLAGS += -I../..
 
-LDFLAGS+=-L../../framework -lmlt
+LDFLAGS+=-L../../framework -lmlt -lm
 
 SRCS := $(OBJS:.o=.c)
 
