--- kgraphviewer/src/part/dotgrammar.cpp.orig	2008-12-16 16:32:50.000000000 +0200
+++ kgraphviewer/src/part/dotgrammar.cpp	2008-12-16 16:33:03.000000000 +0200
@@ -1095,7 +1095,7 @@
 bool DotGraphParsingHelper::parse(const std::string& str)
 {
   DotGrammar g;
-  return boost::spirit::parse(str.c_str(), g, (+space_p|comment_p("/*", "*/"))).full;
+  return boost::spirit::parse(str.c_str(), g >> end_p, (+space_p|comment_p("/*", "*/"))).full;
 }
 
 void DotGraphParsingHelper::finalactions()
