Index: libXt-1.0.8/src/Makefile.in
===================================================================
--- libXt-1.0.8.orig/src/Makefile.in
+++ libXt-1.0.8/src/Makefile.in
@@ -719,7 +719,7 @@ all-local: $(BUILT_FILES)
 $(BUILT_HEADER): $(BUILT_SOURCE)
 
 $(BUILT_SOURCE):
-	$(top_builddir)/util/makestrs -i $(top_srcdir) $(STRINGSABIOPTIONS) < $(STRING_LIST) > StringDefs.c
+	$(top_builddir)/util/hostmakestrs -i $(top_srcdir) $(STRINGSABIOPTIONS) < $(STRING_LIST) > StringDefs.c
 	mkdir -p $(top_builddir)/src
 	mkdir -p $(top_builddir)/include/X11
 	cp StringDefs.h $(top_builddir)/include/X11
