--- DevIL-1.6.7/src-ILU/src/ilu_region.c~	2003-04-06 13:44:30.000000000 +0200
+++ DevIL-1.6.7/src-ILU/src/ilu_region.c	2007-01-19 11:48:06.000000000 +0100
@@ -12,7 +12,7 @@
 
 
 #include "ilu_region.h"
-
+#include "ilu_internal.h"
 
 ILpointi	*RegionPointsi = NULL;
 ILpointf	*RegionPointsf = NULL;
--- DevIL-1.6.7/include/IL/ilu_region.h~	2003-08-12 05:04:25.000000000 +0200
+++ DevIL-1.6.7/include/IL/ilu_region.h	2007-01-19 11:46:23.000000000 +0100
@@ -13,7 +13,7 @@
 #ifndef ILU_REGION_H
 #define ILU_REGION_H
 
-#include "ilu_internal.h"
+#include <IL/il.h>
 
 
 typedef struct Edge
@@ -24,4 +24,4 @@ typedef struct Edge
 } Edge;
 
 
-#endif//ILU_REGION_H
\ No newline at end of file
+#endif//ILU_REGION_H
