Index: work/xorg-server-1.4.2/hw/xfree86/parser/Flags.c
===================================================================
--- work.orig/xorg-server-1.4.2/hw/xfree86/parser/Flags.c
+++ work/xorg-server-1.4.2/hw/xfree86/parser/Flags.c
@@ -63,6 +63,7 @@
 #include "xf86tokens.h"
 #include "Configint.h"
 #include <math.h>
+#include <X11/Xfuncproto.h>
 
 extern LexRec val;
 
@@ -330,7 +331,7 @@ xf86findOption (XF86OptionPtr list, cons
  * returned.  If the option is not found, a NULL is returned.
  */
 
-char *
+_X_EXPORT char *
 xf86findOptionValue (XF86OptionPtr list, const char *name)
 {
 	XF86OptionPtr p = xf86findOption (list, name);
Index: work/xorg-server-1.4.2/hw/xfree86/parser/scan.c
===================================================================
--- work.orig/xorg-server-1.4.2/hw/xfree86/parser/scan.c
+++ work/xorg-server-1.4.2/hw/xfree86/parser/scan.c
@@ -64,6 +64,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <stdarg.h>
+#include <X11/Xfuncproto.h>
 
 #if !defined(X_NOT_POSIX)
 #if defined(_POSIX_SOURCE)
@@ -948,7 +949,7 @@ StringToToken (char *str, xf86ConfigSymT
  * Compare two names.  The characters '_', ' ', and '\t' are ignored
  * in the comparison.
  */
-int
+_X_EXPORT int
 xf86nameCompare (const char *s1, const char *s2)
 {
 	char c1, c2;
