diff -ur liblrdf-0.4.0.orig/examples/showdefaults.c liblrdf-0.4.0/examples/showdefaults.c
--- liblrdf-0.4.0.orig/examples/showdefaults.c	2004-11-09 05:12:32.000000000 +0200
+++ liblrdf-0.4.0/examples/showdefaults.c	2006-04-19 14:33:26.000000000 +0300
@@ -1,6 +1,6 @@
 #include <stdio.h>
 #include <stdlib.h>
-#include <ladspa.h>
+#include <ladspa/ladspa.h>
 
 #include "lrdf.h"
 
diff -ur liblrdf-0.4.0.orig/examples/showtaxonomy.c liblrdf-0.4.0/examples/showtaxonomy.c
--- liblrdf-0.4.0.orig/examples/showtaxonomy.c	2004-11-09 05:12:32.000000000 +0200
+++ liblrdf-0.4.0/examples/showtaxonomy.c	2006-04-19 14:33:30.000000000 +0300
@@ -1,7 +1,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <ladspa.h>
+#include <ladspa/ladspa.h>
 
 #include "lrdf.h"
 
