diff -ur paprefs-0.9.6.orig/src/paprefs.cc paprefs-0.9.6/src/paprefs.cc
--- paprefs-0.9.6.orig/src/paprefs.cc	2007-10-29 23:18:29.000000000 +0200
+++ paprefs-0.9.6/src/paprefs.cc	2008-05-06 15:25:12.000000000 +0300
@@ -29,6 +29,8 @@
 #include <libglademm.h>
 #include <gconfmm.h>
 
+#include <libintl.h>
+
 #define PA_GCONF_ROOT "/system/pulseaudio"
 #define PA_GCONF_PATH_MODULES PA_GCONF_ROOT"/modules"
 
