--- cdrecord/cdrecord.c~	2006-09-05 17:47:08.000000000 +0300
+++ cdrecord/cdrecord.c	    2006-09-05 17:47:36.000000000 +0300
@@ -4048,7 +4048,7 @@
 	scsi_start_stop_unit(scgp, 1, 0, immed);
 	wait_unit_ready(scgp, 120);
 	scgp->silent++;
-	rezero_unit(scgp);	/* Is this needed? Not supported by some drives */
+	//rezero_unit(scgp);	/* Is this needed? Not supported by some drives */
 	scgp->silent--;
 	test_unit_ready(scgp);
 	scsi_start_stop_unit(scgp, 1, 0, immed);
