--- cdrtools/cdrecord/cdrecord.c	2006-12-03 21:10:14.000000000 +0200
+++ cdrtools/cdrecord/cdrecord.c	2006-12-04 10:25:59.189505584 +0200
@@ -393,7 +393,7 @@
 	 * prefer a model with many proprietary differing programs
 	 * instead of cooperating with the program authors.
 	 */
-	linuxcheck();	/* For version 1.343 of cdrecord.c */
+	/*linuxcheck();	 For version 1.343 of cdrecord.c */
 
 	if (flags & F_VERSION)
 		exit(0);
