--- config.tcl.orig	2008-06-15 18:02:39.000000000 +0300
+++ config.tcl	2008-06-15 18:03:32.000000000 +0300
@@ -82,10 +82,10 @@
 			::config::setKey usesnack 0
 		} elseif { [OnUnix] } {
 			::config::setKey soundcommand "play \$sound"
-			::config::setKey browser "mozilla \$url"
+			::config::setKey browser "firefox \$url"
 			::config::setKey notifyXoffset 0
 			::config::setKey notifyYoffset 0
-			::config::setKey filemanager "my_filemanager open \$location"
+			::config::setKey filemanager "xdg-open \$location"
 			::config::setKey openfilecommand ""
 			::config::setKey usesnack 0
 		} elseif { [OnWin] } {
