--- gv-3.6.1.orig/src/Makefile.in
+++ gv-3.6.1/src/Makefile.in
@@ -233,7 +233,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 dist_pkglib_DATA = gv_system.ad gv_user.ad gv_class.ad gv_spartan.dat gv_user_res.dat gv_copyright.dat
-appdefaultsdir = $(pkglibdir)
+appdefaultsdir = /etc/X11/app-defaults
 appdefaults_DATA = GV
 EXTRA_DIST = ad2c gv_font_res.dat gv_layout_res.dat gv_misc_res.dat Aaa_bison.yacc \
            gv_current.xbm gv_doc.xbm gv_empty.xbm gv_even.xbm gv_icon.xbm gv_odd.xbm \
@@ -805,7 +805,7 @@
 	@echo "GV.useBackingPixmap:	$(USE_BACKING_PIXMAP)"		>> $(srcdir)/gv_make_res.dat
 	@echo "GV*dirs:		Home\n\\"				>> $(srcdir)/gv_make_res.dat
 	@echo "			Tmp\n\\"				>> $(srcdir)/gv_make_res.dat
-	@echo "			/usr/doc\n\\"				>> $(srcdir)/gv_make_res.dat
+	@echo "			/usr/share/doc\n\\"			>> $(srcdir)/gv_make_res.dat
 	@echo "			/usr/local/doc"				>> $(srcdir)/gv_make_res.dat
 	@echo "GV*filter:		no .*"				>> $(srcdir)/gv_make_res.dat
 	@echo "GV*filters:		None\n\\"			>> $(srcdir)/gv_make_res.dat
