diff -Naur cuberok-0.0.11.orig/src/player_ffmpeg.h cuberok-0.0.11/src/player_ffmpeg.h
--- cuberok-0.0.11.orig/src/player_ffmpeg.h	2009-07-16 14:00:27.000000000 +0300
+++ cuberok-0.0.11/src/player_ffmpeg.h	2009-07-16 14:02:38.000000000 +0300
@@ -27,8 +27,8 @@
 #include <libavcodec/avcodec.h>
 #include <libavformat/avformat.h>
 #else
-#include <avcodec.h>
-#include <avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 #endif
 }
 
