--- transport.hxx	2006-09-23 14:29:07.000000000 +0300
+++ transport.hxx	2006-09-25 21:08:15.000000000 +0300
@@ -284,9 +284,9 @@
 #if !KERNEL_BROKEN
 		if (sg_io.masked_status&CHECK_CONDITION)
 #endif
-		{   ret = ERRCODE(sg_io.sbp);
+		{   ret = ERRCODE(_sense.u);
 		    if (ret==0) ret=-1;
-		    else	CREAM_ON_ERRNO(sg_io.sbp);
+		    else	CREAM_ON_ERRNO(_sense.u);
 		}
 	    }
 	    return ret;
