--- tools/build/v2/tools/gcc.jam	2008-05-20 13:45:04.000000000 +0200
+++ tools/build/v2/tools/gcc.jam.oden	2008-05-20 13:49:29.000000000 +0200
@@ -284,7 +284,7 @@ flags gcc.compile PCH_FILE <pch>on : <pc
 
 # Declare flags and action for compilation
 flags gcc.compile OPTIONS <optimization>off   : -O0 ;
-flags gcc.compile OPTIONS <optimization>speed : -O3 ;
+flags gcc.compile OPTIONS <optimization>speed : PARDUS_FLAGS ;
 flags gcc.compile OPTIONS <optimization>space : -Os ;
 
 flags gcc.compile OPTIONS <inlining>off  : -fno-inline ;
