diff -ru urlgrabber-3.0.0-orig/urlgrabber/progress.py urlgrabber-3.0.0/urlgrabber/progress.py
--- urlgrabber-3.0.0-orig/urlgrabber/progress.py	2008-10-14 23:46:24.000000000 -0400
+++ urlgrabber-3.0.0/urlgrabber/progress.py	2008-10-14 23:57:53.000000000 -0400
@@ -258,6 +258,8 @@
             # Make text grow a bit before we start growing the bar too
             blen = 4 + tl.rest_split(8 + 8 + 4)
             bar  = '='*int(blen * frac)
+            if (blen * frac) - int(blen * frac) >= 0.5:
+                bar += '-'
             ui_bar  = tl.add(' [%-*.*s]' % (blen, blen, bar))
             out = '\r%-*.*s%s%s%s%s%s%s%s' % (tl.rest(), tl.rest(), text,
                                               ui_sofar_pc, ui_pc, ui_bar,
Only in urlgrabber-3.0.0/urlgrabber: progress.py~
