Index: sudo-1.6.9p22/find_path.c
===================================================================
--- sudo-1.6.9p22.orig/find_path.c
+++ sudo-1.6.9p22/find_path.c
@@ -131,7 +131,7 @@ find_path(infile, outfile, sbp, path)
     if (!result && checkdot) {
 	len = snprintf(command, sizeof(command), "./%s", infile);
 	if (len <= 0 || len >= sizeof(command))
-	    errorx(1, "%s: File name too long", infile);
+	    errx(1, "%s: File name too long", infile);
 	result = sudo_goodpath(command, sbp);
 	if (result && def_ignore_dot)
 	    return(NOT_FOUND_DOT);
