--- source/qcommon/files.c.orig	2007-05-26 11:32:02.000000000 +0300
+++ source/qcommon/files.c	2007-05-26 21:21:56.000000000 +0300
@@ -633,7 +633,7 @@
 ================
 FS_AddHomeAsGameDirectory
  
-Use ~/.codered/dir as fs_gamedir
+Use ~/.alienarena2007/dir as fs_gamedir
 ================
 */
 void FS_AddHomeAsGameDirectory (char *dir)
@@ -642,7 +642,7 @@
 	char *homedir=getenv("HOME");
 	if(homedir)
 	{
-		int len = snprintf(gdir,sizeof(gdir),"%s/.codered/%s/", homedir, dir);
+		int len = snprintf(gdir,sizeof(gdir),"%s/.alienarena2007/%s/", homedir, dir);
 		Com_Printf("using %s for writing\n",gdir);
 		FS_CreatePath (gdir);
 

