--- source/client/snd_dma.c.orig	2007-05-26 11:32:01.000000000 +0300
+++ source/client/snd_dma.c	2007-05-27 20:47:37.000000000 +0300
@@ -125,7 +125,7 @@
 	else
 	{
 		s_volume = Cvar_Get ("s_volume", "0.7", CVAR_ARCHIVE);
-		s_khz = Cvar_Get ("s_khz", "22", CVAR_ARCHIVE); //default sound sampling rate will be 22000 in windows
+		s_khz = Cvar_Get ("s_khz", "48", CVAR_ARCHIVE); //default sound sampling rate will be 48000 in windows
 
 		s_loadas8bit = Cvar_Get ("s_loadas8bit", "0", CVAR_ARCHIVE);
 		s_mixahead = Cvar_Get ("s_mixahead", "0.2", CVAR_ARCHIVE);
