Index: kdenetwork/ktalkd/ktalkdlg/Makefile.am
===================================================================
--- kdenetwork.orig/ktalkd/ktalkdlg/Makefile.am
+++ kdenetwork/ktalkd/ktalkdlg/Makefile.am
@@ -7,7 +7,7 @@ bin_PROGRAMS	 =	ktalkdlg
 ktalkdlg_SOURCES =	ktalkdlg.cpp
 
 INCLUDES	=	$(all_includes)
-ktalkdlg_LDFLAGS=	$(all_libraries) $(KDE_RPATH)
+ktalkdlg_LDFLAGS=	/usr/kde/3.5/lib/libkdeui.so $(all_libraries) $(KDE_RPATH)
 ktalkdlg_LDADD	=	$(LIB_KDEUI)
 
 #for extra warnings during compilation :
