Index: gdata-sharp/Makefile
===================================================================
--- gdata-sharp.orig/Makefile	2011-02-10 10:31:46.295372901 +0800
+++ gdata-sharp/Makefile	2011-02-10 10:31:47.572670311 +0800
@@ -137,7 +137,7 @@
 	for i in $(PKGCONFIG_FILES) ; do rm "$(DESTDIR)$(PKGCONFIGDIR)/$$i" ; done
 
 tests: $(ALLLIBS)
-	$(CSC) $(DEFINES) -target:library -out:unittests.dll $(ALLREFS) -r:nunit.framework.dll $(test_sources)
+	$(CSC) $(DEFINES) -target:library -out:unittests.dll $(ALLREFS) -pkg:mono-nunit $(test_sources)
 
 gbase_customertool_sources = samples/gbase/example.cs \
 	samples/gbase/customertool.cs	\
