Index: gdata-sharp/misc/gdata-sharp-acl.pc.in
===================================================================
--- gdata-sharp.orig/misc/gdata-sharp-acl.pc.in	2011-02-10 10:32:08.446311299 +0800
+++ gdata-sharp/misc/gdata-sharp-acl.pc.in	2011-02-11 02:19:49.650791230 +0800
@@ -6,4 +6,4 @@
 Version: @VERSION@
 
 Requires: gdata-sharp-core
-Libs: -r:${libdir}/mono/GData-Sharp/Google.GData.AccessControl.dll
+Libs: -r:${libdir}/cli/Google.GData.AccessControl-1.7/Google.GData.AccessControl.dll
Index: gdata-sharp/misc/gdata-sharp-apps.pc.in
===================================================================
--- gdata-sharp.orig/misc/gdata-sharp-apps.pc.in	2011-02-10 10:32:08.506340944 +0800
+++ gdata-sharp/misc/gdata-sharp-apps.pc.in	2011-02-11 02:19:49.650791230 +0800
@@ -6,4 +6,4 @@
 Version: @VERSION@
 
 Requires: gdata-sharp-core
-Libs: -r:${libdir}/mono/GData-Sharp/Google.GData.Apps.dll
+Libs: -r:${libdir}/cli/Google.GData.Apps-1.7/Google.GData.Apps.dll
Index: gdata-sharp/misc/gdata-sharp-base.pc.in
===================================================================
--- gdata-sharp.orig/misc/gdata-sharp-base.pc.in	2011-02-10 10:32:08.482996082 +0800
+++ gdata-sharp/misc/gdata-sharp-base.pc.in	2011-02-11 02:19:49.650791230 +0800
@@ -6,4 +6,4 @@
 Version: @VERSION@
 
 Requires: gdata-sharp-core
-Libs: -r:${libdir}/mono/GData-Sharp/Google.GData.GoogleBase.dll
+Libs: -r:${libdir}/cli/Google.GData.GoogleBase-1.7/Google.GData.GoogleBase.dll
Index: gdata-sharp/misc/gdata-sharp-blogger.pc.in
===================================================================
--- gdata-sharp.orig/misc/gdata-sharp-blogger.pc.in	2011-02-10 10:32:08.513010905 +0800
+++ gdata-sharp/misc/gdata-sharp-blogger.pc.in	2011-02-11 02:19:49.650791230 +0800
@@ -6,4 +6,4 @@
 Version: @VERSION@
 
 Requires: gdata-sharp-core
-Libs: -r:${libdir}/mono/GData-Sharp/Google.GData.Blogger.dll
+Libs: -r:${libdir}/cli/Google.GData.Blogger-1.7/Google.GData.Blogger.dll
Index: gdata-sharp/misc/gdata-sharp-calendar.pc.in
===================================================================
--- gdata-sharp.orig/misc/gdata-sharp-calendar.pc.in	2011-02-10 10:32:08.452981261 +0800
+++ gdata-sharp/misc/gdata-sharp-calendar.pc.in	2011-02-11 02:19:49.650791230 +0800
@@ -6,4 +6,4 @@
 Version: @VERSION@
 
 Requires: gdata-sharp-core gdata-sharp-acl
-Libs: -r:${libdir}/mono/GData-Sharp/Google.GData.Calendar.dll
+Libs: -r:${libdir}/cli/Google.GData.Calendar-1.7/Google.GData.Calendar.dll
Index: gdata-sharp/misc/gdata-sharp-codesearch.pc.in
===================================================================
--- gdata-sharp.orig/misc/gdata-sharp-codesearch.pc.in	2011-02-10 10:32:08.526350825 +0800
+++ gdata-sharp/misc/gdata-sharp-codesearch.pc.in	2011-02-11 02:19:49.650791230 +0800
@@ -6,4 +6,4 @@
 Version: @VERSION@
 
 Requires: gdata-sharp-core
-Libs: -r:${libdir}/mono/GData-Sharp/Google.GData.CodeSearch.dll
+Libs: -r:${libdir}/cli/Google.GData.CodeSearch-1.7/Google.GData.CodeSearch.dll
Index: gdata-sharp/misc/gdata-sharp-contacts.pc.in
===================================================================
--- gdata-sharp.orig/misc/gdata-sharp-contacts.pc.in	2011-02-10 10:32:08.466321182 +0800
+++ gdata-sharp/misc/gdata-sharp-contacts.pc.in	2011-02-11 02:19:49.650791230 +0800
@@ -6,4 +6,4 @@
 Version: @VERSION@
 
 Requires: gdata-sharp-core
-Libs: -r:${libdir}/mono/GData-Sharp/Google.GData.Contacts.dll
+Libs: -r:${libdir}/cli/Google.GData.Contacts-1.7/Google.GData.Contacts.dll
Index: gdata-sharp/misc/gdata-sharp-core.pc.in
===================================================================
--- gdata-sharp.orig/misc/gdata-sharp-core.pc.in	2011-02-10 10:32:08.496336002 +0800
+++ gdata-sharp/misc/gdata-sharp-core.pc.in	2011-02-11 02:22:37.516571682 +0800
@@ -5,4 +5,4 @@
 Description: CLI bindings to Google's online services (Core libraries)
 Version: @VERSION@
 
-Libs: -r:${libdir}/mono/GData-Sharp/Google.GData.Client.dll -r:${libdir}/mono/GData-Sharp/Google.GData.Extensions.dll
+Libs: -r:${libdir}/cli/Google.GData.Client-1.7/Google.GData.Client.dll -r:${libdir}/cli/Google.GData.Extensions-1.7/Google.GData.Extensions.dll
Index: gdata-sharp/misc/gdata-sharp-documents.pc.in
===================================================================
--- gdata-sharp.orig/misc/gdata-sharp-documents.pc.in	2011-02-10 10:32:08.406291539 +0800
+++ gdata-sharp/misc/gdata-sharp-documents.pc.in	2011-02-11 02:19:49.654124543 +0800
@@ -6,4 +6,4 @@
 Version: @VERSION@
 
 Requires: gdata-sharp-core
-Libs: -r:${libdir}/mono/GData-Sharp/Google.GData.Documents.dll
+Libs: -r:${libdir}/cli/Google.GData.Documents-1.7/Google.GData.Documents.dll
Index: gdata-sharp/misc/gdata-sharp-health.pc.in
===================================================================
--- gdata-sharp.orig/misc/gdata-sharp-health.pc.in	2011-02-10 10:32:08.436306359 +0800
+++ gdata-sharp/misc/gdata-sharp-health.pc.in	2011-02-11 02:19:49.654124543 +0800
@@ -6,4 +6,4 @@
 Version: @VERSION@
 
 Requires: gdata-sharp-core
-Libs: -r:${libdir}/mono/GData-Sharp/Google.GData.Health.dll
+Libs: -r:${libdir}/cli/Google.GData.Health-1.7/Google.GData.Health.dll
Index: gdata-sharp/misc/gdata-sharp-photos.pc.in
===================================================================
--- gdata-sharp.orig/misc/gdata-sharp-photos.pc.in	2011-02-10 10:32:08.422966439 +0800
+++ gdata-sharp/misc/gdata-sharp-photos.pc.in	2011-02-11 02:19:49.654124543 +0800
@@ -6,4 +6,4 @@
 Version: @VERSION@
 
 Requires: gdata-sharp-core
-Libs: -r:${libdir}/mono/GData-Sharp/Google.GData.Photos.dll
+Libs: -r:${libdir}/cli/Google.GData.Photos-1.7/Google.GData.Photos.dll
Index: gdata-sharp/misc/gdata-sharp-spreadsheets.pc.in
===================================================================
--- gdata-sharp.orig/misc/gdata-sharp-spreadsheets.pc.in	2011-02-10 10:32:08.476326122 +0800
+++ gdata-sharp/misc/gdata-sharp-spreadsheets.pc.in	2011-02-11 02:19:49.654124543 +0800
@@ -6,4 +6,4 @@
 Version: @VERSION@
 
 Requires: gdata-sharp-core
-Libs: -r:${libdir}/mono/GData-Sharp/Google.GData.Spreadsheets.dll
+Libs: -r:${libdir}/cli/Google.GData.Spreadsheets-1.7/Google.GData.Spreadsheets.dll
Index: gdata-sharp/misc/gdata-sharp-youtube.pc.in
===================================================================
--- gdata-sharp.orig/misc/gdata-sharp-youtube.pc.in	2011-02-10 10:32:08.416296479 +0800
+++ gdata-sharp/misc/gdata-sharp-youtube.pc.in	2011-02-11 02:19:49.654124543 +0800
@@ -6,4 +6,4 @@
 Version: @VERSION@
 
 Requires: gdata-sharp-core
-Libs: -r:${libdir}/mono/GData-Sharp/Google.GData.YouTube.dll
+Libs: -r:${libdir}/cli/Google.GData.YouTube-1.7/Google.GData.YouTube.dll
Index: gdata-sharp/misc/gdata-sharp-analytics.pc.in
===================================================================
--- gdata-sharp.orig/misc/gdata-sharp-analytics.pc.in	2011-02-11 02:21:44.810105798 +0800
+++ gdata-sharp/misc/gdata-sharp-analytics.pc.in	2011-02-11 02:22:01.066711564 +0800
@@ -6,4 +6,4 @@
 Version: @VERSION@
 
 Requires: gdata-sharp-core
-Libs: -r:${libdir}/mono/GData-Sharp/Google.GData.Analytics.dll
+Libs: -r:${libdir}/cli/Google.GData.Analytics.dll
