Index: src/common.inc
===================================================================
--- src/common.inc	(revision 24881)
+++ src/common.inc	(working copy)
@@ -399,6 +399,10 @@
   syslinux {
     video.modes.list xmenu.video .xm_current get get .vm_mode get
 
+    % Make sure we boot as livecd even 
+    % FIXME: this can be defined in config
+    "mudur=livecd " bc.cmd dup length add sprintf
+
     dup 0 ge {
       0x200 add "vga=0x%x " bc.cmd dup length add sprintf
     } {
@@ -407,6 +411,15 @@
       } if
     } ifelse
 
+    v_impaired 1 ge {
+      "braille=1 " bc.cmd dup length add sprintf
+    } if
+
+    v_impaired 2 ge {
+      "linemode=1 " bc.cmd dup length add sprintf
+    } if
+  } if
+
     config.lang {
       %config.lang "en_US" ne {
       %  % config.lang "lang=%s " bc.cmd dup length add sprintf
@@ -415,60 +428,112 @@
       %  config.lang "mudur=livecd,language:en " bc.cmd dup length add sprintf
       %} ifelse
 
-      config.lang "ca_ES" eq {
-        "mudur=livecd,language:ca " bc.cmd dup length add sprintf
-      } if
+      syslinux {
+        config.lang "ca_ES" eq {
+          "mudur=language:ca " bc.cmd dup length add sprintf
+        } if
 
-      config.lang "de_DE" eq {
-        "mudur=livecd,language:de " bc.cmd dup length add sprintf
-      } if
+        config.lang "de_DE" eq {
+          "mudur=language:de " bc.cmd dup length add sprintf
+        } if
 
-      config.lang "en_US" eq {
-        "mudur=livecd,language:en " bc.cmd dup length add sprintf
-      } if
+        config.lang "en_US" eq {
+          "mudur=language:en " bc.cmd dup length add sprintf
+        } if
 
-      config.lang "es_ES" eq {
-        "mudur=livecd,language:es " bc.cmd dup length add sprintf
-      } if
+        config.lang "es_ES" eq {
+          "mudur=language:es " bc.cmd dup length add sprintf
+        } if
 
-      config.lang "fr_FR" eq {
-        "mudur=livecd,language:fr " bc.cmd dup length add sprintf
-      } if
+        config.lang "fr_FR" eq {
+          "mudur=language:fr " bc.cmd dup length add sprintf
+        } if
 
-      config.lang "it_IT" eq {
-        "mudur=livecd,language:it " bc.cmd dup length add sprintf
-      } if
+        config.lang "it_IT" eq {
+          "mudur=language:it " bc.cmd dup length add sprintf
+        } if
 
-      config.lang "nl_NL" eq {
-        "mudur=livecd,language:nl " bc.cmd dup length add sprintf
-      } if
+        config.lang "nl_NL" eq {
+          "mudur=language:nl " bc.cmd dup length add sprintf
+        } if
 
-      config.lang "pl_PL" eq {
-        "mudur=livecd,language:pl " bc.cmd dup length add sprintf
-      } if
+        config.lang "pl_PL" eq {
+          "mudur=language:pl " bc.cmd dup length add sprintf
+        } if
 
-      config.lang "pt_BR" eq {
-        "mudur=livecd,language:pt_BR " bc.cmd dup length add sprintf
-      } if
+        config.lang "pt_BR" eq {
+          "mudur=language:pt_BR " bc.cmd dup length add sprintf
+        } if
 
-      config.lang "sv_SE" eq {
-        "mudur=livecd,language:sv " bc.cmd dup length add sprintf
-      } if
+        config.lang "sv_SE" eq {
+          "mudur=language:sv " bc.cmd dup length add sprintf
+        } if
 
-      config.lang "tr_TR" eq {
-        "mudur=livecd,language:tr " bc.cmd dup length add sprintf
-      } if
+        config.lang "tr_TR" eq {
+          "mudur=language:tr " bc.cmd dup length add sprintf
+        } if
 
-    } if
+      } {
 
-    v_impaired 1 ge {
-      "braille=1 " bc.cmd dup length add sprintf
-    } if
+        "lang" findfile dup {
+          /tmp over length 1 add 2 max string def
+          tmp exch {
+            dup ' ' eq over '\n' eq or { pop pop exit } if
+            over exch 0 exch put
+            1 add
+          } forall
+        } if
 
-    v_impaired 2 ge {
-      "linemode=1 " bc.cmd dup length add sprintf
+        config.lang tmp ne {
+          config.lang "ca_ES" eq {
+            "mudur=language:ca " bc.cmd dup length add sprintf
+          } if
+
+          config.lang "de_DE" eq {
+            "mudur=language:de " bc.cmd dup length add sprintf
+          } if
+
+          config.lang "en_US" eq {
+            "mudur=language:en " bc.cmd dup length add sprintf
+          } if
+
+          config.lang "es_ES" eq {
+            "mudur=language:es " bc.cmd dup length add sprintf
+          } if
+
+          config.lang "fr_FR" eq {
+            "mudur=language:fr " bc.cmd dup length add sprintf
+          } if
+
+          config.lang "it_IT" eq {
+            "mudur=language:it " bc.cmd dup length add sprintf
+          } if
+
+          config.lang "nl_NL" eq {
+            "mudur=language:nl " bc.cmd dup length add sprintf
+          } if
+
+          config.lang "pl_PL" eq {
+            "mudur=language:pl " bc.cmd dup length add sprintf
+          } if
+
+          config.lang "pt_BR" eq {
+            "mudur=language:pt_BR " bc.cmd dup length add sprintf
+          } if
+
+          config.lang "sv_SE" eq {
+            "mudur=language:sv " bc.cmd dup length add sprintf
+          } if
+
+          config.lang "tr_TR" eq {
+            "mudur=language:tr " bc.cmd dup length add sprintf
+          } if
+
+        } if
+
+      } ifelse
+
     } if
-  } if
 
   xmenu.dud {
     xmenu.dud .xm_current get .dud_url eq {
