Index: kdebase-workspace-4.2.1/libs/ksysguard/processui/ProcessWidgetUI.ui
===================================================================
--- kdebase-workspace-4.2.1.orig/libs/ksysguard/processui/ProcessWidgetUI.ui	2009-03-26 18:35:16.000000000 -0400
+++ kdebase-workspace-4.2.1/libs/ksysguard/processui/ProcessWidgetUI.ui	2009-03-26 18:36:02.000000000 -0400
@@ -45,9 +45,6 @@
      </item>
      <item>
       <widget class="KLineEdit" name="txtFilter" >
-       <property name="styleSheet" >
-        <string>QLineEdit { padding-right: 16; }</string>
-       </property>
        <property name="urlDropsEnabled" >
         <bool>false</bool>
        </property>
