Index: kdebase-workspace-4.2.0/plasma/dataengines/soliddevice/soliddeviceengine.cpp
===================================================================
--- kdebase-workspace-4.2.0.orig/plasma/dataengines/soliddevice/soliddeviceengine.cpp	2009-02-24 10:53:26.000000000 -0500
+++ kdebase-workspace-4.2.0/plasma/dataengines/soliddevice/soliddeviceengine.cpp	2009-02-24 10:53:56.000000000 -0500
@@ -517,7 +517,7 @@
     if (statvfs(path,&fs_obj) < 0) {
         return -1;
     } else {
-        return (qlonglong)fs_obj.f_bfree*(qlonglong)fs_obj.f_frsize;
+        return (qlonglong)fs_obj.f_bavail*(qlonglong)fs_obj.f_frsize;
     }
 #elif defined(HAVE_STATFS) && !defined(USE_SOLARIS)
     struct statfs fs_obj;
