Index: kdebase-workspace-4.2.1a/plasma/applets/tasks/abstracttaskitem.cpp
===================================================================
--- kdebase-workspace-4.2.1a.orig/plasma/applets/tasks/abstracttaskitem.cpp	2009-03-05 12:34:28.000000000 -0500
+++ kdebase-workspace-4.2.1a/plasma/applets/tasks/abstracttaskitem.cpp	2009-03-05 12:43:13.000000000 -0500
@@ -286,6 +286,11 @@
     QMimeData* mimeData = new QMimeData();
     setAdditionalMimeData(mimeData);
 
+    if (mimeData->formats().isEmpty()) {
+        delete mimeData;
+        return;
+    }
+
     QDrag *drag = new QDrag(event->widget());
     drag->setMimeData(mimeData);
     drag->setPixmap(m_icon.pixmap(20));
Index: kdebase-workspace-4.2.1a/plasma/applets/tasks/windowtaskitem.cpp
===================================================================
--- kdebase-workspace-4.2.1a.orig/plasma/applets/tasks/windowtaskitem.cpp	2009-03-05 12:34:36.000000000 -0500
+++ kdebase-workspace-4.2.1a/plasma/applets/tasks/windowtaskitem.cpp	2009-03-05 12:45:01.000000000 -0500
@@ -248,10 +248,9 @@
     }
 }
 
-
 TaskManager::TaskPtr WindowTaskItem::windowTask() const
 {
-    return m_task->task();
+    return m_task ? m_task->task() : TaskManager::TaskPtr();
 }
 
 void WindowTaskItem::contextMenuEvent(QGraphicsSceneContextMenuEvent *e)
@@ -277,8 +276,6 @@
     delete a;
 }
 
-
-
 bool WindowTaskItem::isWindowItem() const
 {
     return true;
@@ -290,12 +287,15 @@
         //kDebug() << "no task set";
         return false;
     }
+
     return m_task->isActive();
 }
 
 void WindowTaskItem::setAdditionalMimeData(QMimeData* mimeData)
 {
-    m_task->addMimeData(mimeData);
+    if (m_task) {
+        m_task->addMimeData(mimeData);
+    }
 }
 
 #include "windowtaskitem.moc"
