--- kdebase-workspace-4.2.0/plasma/applets/kickoff/core/leavemodel.cpp	2008-12-10 16:13:23.000000000 +0000
+++ kdebase-workspace-4.2.0/plasma/applets/kickoff/core/leavemodel.cpp	2009-02-16 19:27:44.000000000 +0000
@@ -74,13 +74,13 @@
         item->setIcon(KIcon("system-suspend"));
         item->setData(i18n("Pause without logging out"), Kickoff::SubTitleRole);
     } else if (basename == "suspenddisk") {
-        item->setText(i18n("Suspend to Disk"));
+        item->setText(i18n("Hibernate"));
         item->setIcon(KIcon("system-suspend-hibernate"));
-        item->setData(i18n("Pause without logging out"), Kickoff::SubTitleRole);
+        item->setData(i18n("Suspend to Disk"), Kickoff::SubTitleRole);
     } else if (basename == "suspendram") {
-        item->setText(i18n("Suspend to RAM"));
-        item->setIcon(KIcon("system-suspend-hibernate"));
-        item->setData(i18n("Pause without logging out"), Kickoff::SubTitleRole);
+        item->setText(i18n("Sleep"));
+        item->setIcon(KIcon("system-suspend"));
+        item->setData(i18n("Suspend to RAM"), Kickoff::SubTitleRole);
     } else {
         item->setText(basename);
         item->setData(url, Kickoff::SubTitleRole);
