diff --git a/plasma/generic/dataengines/soliddevice/hddtemp.cpp b/plasma/generic/dataengines/soliddevice/hddtemp.cpp
index 9b74ac5..567137e 100644
--- a/plasma/generic/dataengines/soliddevice/hddtemp.cpp
+++ b/plasma/generic/dataengines/soliddevice/hddtemp.cpp
@@ -21,30 +21,39 @@
 
 #include <QTcpSocket>
 
+#include <QTimerEvent>
+
 #include <KDebug>
 
 HddTemp::HddTemp(QObject* parent)
     : QObject(parent),
       m_failCount(0),
-      m_timer(0)
+      m_cacheValid(false)
 {
     updateData();
-    m_timer = new QTimer(this);
-    connect(m_timer, SIGNAL(timeout()), this, SLOT(updateData()));
-    m_timer->start(10000);
 }
 
 HddTemp::~HddTemp()
 {
 }
 
-QStringList HddTemp::sources() const
+QStringList HddTemp::sources()
 {
+    updateData();
     return m_data.keys();
 }
 
+void HddTemp::timerEvent(QTimerEvent *event)
+{
+    killTimer(event->timerId());
+    m_cacheValid = false;
+}
+
 bool HddTemp::updateData()
 {
+    if (m_cacheValid) {
+	return true;
+    }
     if (m_failCount > 4) {
         return false;
     }
@@ -80,6 +89,9 @@ bool HddTemp::updateData()
         m_data[list[i]].append(list[i + 3]);
         i += 5;
     }
+    m_cacheValid = true;
+    startTimer(0);
+
     return true;
 }
 
diff --git a/plasma/generic/dataengines/soliddevice/hddtemp.h b/plasma/generic/dataengines/soliddevice/hddtemp.h
index d03d59b..ea372ea 100644
--- a/plasma/generic/dataengines/soliddevice/hddtemp.h
+++ b/plasma/generic/dataengines/soliddevice/hddtemp.h
@@ -37,16 +37,17 @@ class HddTemp : public QObject
         
         HddTemp(QObject *parent=0);
         ~HddTemp();
-        QStringList sources() const;
+        QStringList sources();
         QVariant data(const QString source, const DataType type) const;
 
-    private Q_SLOTS:
-        bool updateData();
-        
+    protected:
+        void timerEvent(QTimerEvent *event);
+
     private:
         int m_failCount;
+        bool m_cacheValid;
         QMap<QString, QList<QVariant> > m_data;
-        QTimer *m_timer;
+        bool updateData();
 };
 
 
