--- modules/demux/live555.cpp.orig      2008-09-06 13:40:28.000000000 +0300
+++ modules/demux/live555.cpp   2008-09-06 13:41:02.000000000 +0300
@@ -104,7 +104,7 @@
         add_shortcut( "sdp" );
         set_capability( "access_demux", 0 );
         set_callbacks( Open, Close );
-        add_bool( "rtsp-tcp", VLC_TRUE, NULL,
+        add_bool( "rtsp-tcp", true, NULL,
                   N_("Use RTP over RTSP (TCP)"),
                   N_("Use RTP over RTSP (TCP)"), true );
         add_integer( "rtp-client-port", -1, NULL,
