Index: kdegames/lskat/lskatproc/Makefile.am
===================================================================
--- kdegames.orig/lskat/lskatproc/Makefile.am
+++ kdegames/lskat/lskatproc/Makefile.am
@@ -10,4 +10,4 @@ INCLUDES= $(all_includes)
 METASOURCES = AUTO
 
 # the library search path. 
-lskatproc_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+lskatproc_LDFLAGS = /usr/kde/3.5/lib/libkio.so $(all_libraries) $(KDE_RPATH)
