diff -Nur AstroMenaceSourceCode-old/AstroMenaceSource/Main.cpp AstroMenaceSourceCode/AstroMenaceSource/Main.cpp
--- AstroMenaceSourceCode-old/AstroMenaceSource/Main.cpp	2007-09-25 18:58:48.029366820 +0300
+++ AstroMenaceSourceCode/AstroMenaceSource/Main.cpp	2007-09-25 19:06:22.463282364 +0300
@@ -329,11 +329,8 @@
 	}
 	if (!dirpresent)
 	{
-		strcpy(ProgrammDir, argv[0]);
-		char* s = strrchr(ProgrammDir,'/');
-		if (s) s[0]=0x0;
-		const char *Fi = "/";
-		strcat( ProgrammDir, Fi );
+		const char *Fi = "/usr/share/AstroMenace/";
+		strcpy( ProgrammDir, Fi );
 	}
 
 
