--- org/gudy/azureus2/platform/PlatformManagerPluginDelegate.java.orig	2009-09-09 16:46:34.541012339 +0300
+++ org/gudy/azureus2/platform/PlatformManagerPluginDelegate.java	2009-09-09 16:47:04.355886978 +0300
@@ -47,9 +47,6 @@
 		if ( platformType == PlatformManager.PT_WINDOWS ){
 			org.gudy.azureus2.platform.win32.PlatformManagerUpdateChecker plugin = new org.gudy.azureus2.platform.win32.PlatformManagerUpdateChecker();
 			plugin.initialize(pluginInterface);
-		}else if ( platformType == PlatformManager.PT_MACOSX ){
-			org.gudy.azureus2.platform.macosx.PlatformManagerUpdateChecker plugin = new org.gudy.azureus2.platform.macosx.PlatformManagerUpdateChecker();
-			plugin.initialize(pluginInterface);
 		}else if ( platformType == PlatformManager.PT_UNIX ){
 			PlatformManagerUnixPlugin plugin = new PlatformManagerUnixPlugin();
 			plugin.initialize(pluginInterface);
