diff -ur virt-manager-0.5.3.orig/src/virt-manager.py.in virt-manager-0.5.3/src/virt-manager.py.in
--- virt-manager-0.5.3.orig/src/virt-manager.py.in	2008-01-11 03:17:51.000000000 +0200
+++ virt-manager-0.5.3/src/virt-manager.py.in	2008-02-15 17:36:53.000000000 +0200
@@ -248,6 +248,9 @@
 
     gtk.window_set_default_icon_from_file(icon_dir + "/" + appname + "-icon.svg")
 
+    if options.uri != None and (options.uri.upper() == "XEN" or options.uri.upper().startswith("XEN:")):
+        options.uri = "xen:///"
+
     if options.show and options.uri==None:
         raise OptionValueError("can't use --show-* options without --connect")
 
