--- quassel-0.3.1/src/qtui/coreconnectdlg.cpp.orig	2008-11-25 17:42:37.000000000 +0200
+++ quassel-0.3.1/src/qtui/coreconnectdlg.cpp	2008-11-25 17:42:51.000000000 +0200
@@ -227,6 +227,7 @@
     accountSettings.storeAccountData(_internalAccountId, data);
   }
   clientSyncer->useInternalCore(_internalAccountId);
+  ui.loginButtonBox->setStandardButtons(QDialogButtonBox::Cancel);
   startSync();
 }
 
@@ -425,7 +426,8 @@
   ui.user->setEnabled(true);
   ui.password->setEnabled(true);
   ui.rememberPasswd->setEnabled(true);
-  ui.loginButtonBox->button(QDialogButtonBox::Ok)->setEnabled(true);
+  if(ui.loginButtonBox->standardButtons() & QDialogButtonBox::Ok) // in mono mode we don't show an Ok Button
+    ui.loginButtonBox->button(QDialogButtonBox::Ok)->setEnabled(true);
 }
 
 void CoreConnectDlg::coreSessionProgress(quint32 val, quint32 max) {
