Index: ktranslator-0.4/ktranslator/plugins/babylon/libbab.cc
===================================================================
--- ktranslator-0.4.orig/ktranslator/plugins/babylon/libbab.cc
+++ ktranslator-0.4/ktranslator/plugins/babylon/libbab.cc
@@ -26,6 +26,7 @@
 #include <iostream>
 #include <iomanip>
 #include <fstream>
+#include <cstring>
 #include <string>
 #include <list>
 #include <cctype>
