Index: konversation/src/config/preferences.cpp
===================================================================
--- konversation/src/config/preferences.cpp	(revision 501817)
+++ konversation/src/config/preferences.cpp	(working copy)
@@ -77,7 +77,7 @@
     serverGroup->addServer(server);
     serverGroup->setIdentityId(mIdentity->id());
     Konversation::ChannelSettings channel;
-    channel.setName("#kde");
+    channel.setName("#pardus");
     serverGroup->addChannel(channel);
     serverGroup->setExpanded(false);
     mServerGroupList.append(serverGroup);
Index: konversation/src/config/konversation.kcfg
===================================================================
--- konversation/src/config/konversation.kcfg	(revision 501817)
+++ konversation/src/config/konversation.kcfg	(working copy)
@@ -163,12 +163,12 @@
       <whatsthis></whatsthis>
     </entry>
     <entry key="ShowNicknameBox" type="Bool">
-      <default>true</default>
+      <default>false</default>
       <label></label>
       <whatsthis></whatsthis>
     </entry>
     <entry key="UseBoldNicks" type="Bool">
-      <default>false</default>
+      <default>true</default>
       <label></label>
       <whatsthis></whatsthis>
     </entry>
@@ -387,12 +387,12 @@
   </group>
   <group name="OSD">
     <entry key="UseOSD" type="Bool">
-      <default>false</default>
+      <default>true</default>
       <label></label>
       <whatsthis></whatsthis>
     </entry>
     <entry key="ShowOwnNick" type="Bool" name="OSDShowOwnNick">
-      <default>false</default>
+      <default>true</default>
       <label></label>
       <whatsthis></whatsthis>
     </entry>
@@ -402,7 +402,7 @@
       <whatsthis></whatsthis>
     </entry>
     <entry key="ShowQuery" type="Bool" name="OSDShowQuery">
-      <default>false</default>
+      <default>true</default>
       <label></label>
       <whatsthis></whatsthis>
     </entry>
@@ -752,7 +752,7 @@
   </group>
   <group name="Themes">
     <entry key="IconTheme" type="String">
-      <default>default</default>
+      <default>smiling</default>
       <label></label>
       <whatsthis></whatsthis>
     </entry>
