diff -uNr scintilla.orig/gtk/makefile scintilla/gtk/makefile
--- scintilla.orig/gtk/makefile	2010-09-04 05:27:14.000000000 -0500
+++ scintilla/gtk/makefile	2010-12-04 09:13:52.349929002 -0500
@@ -39,9 +39,9 @@
 endif
 
 ifdef DEBUG
-CXXFLAGS=-DDEBUG -g $(CXXBASEFLAGS) $(THREADFLAGS)
+CXXFLAGS=-DDEBUG $(OPTFLAGS) $(CXXBASEFLAGS) $(THREADFLAGS)
 else
-CXXFLAGS=-DNDEBUG -Os $(CXXBASEFLAGS) $(THREADFLAGS)
+CXXFLAGS=-DNDEBUG $(OPTFLAGS) $(CXXBASEFLAGS) $(THREADFLAGS)
 endif
 
 CONFIGFLAGS:=$(shell pkg-config --cflags gtk+-2.0)
