diff -uNr scite.orig/gtk/makefile scite/gtk/makefile
--- scite.orig/gtk/makefile	2010-09-04 05:27:07.000000000 -0500
+++ scite/gtk/makefile	2010-12-04 09:16:36.813929020 -0500
@@ -45,9 +45,9 @@
 CXXBASEFLAGS=-W -Wall -Wno-char-subscripts -pedantic -DGTK -DSCI_LEXER -DPIXMAP_PATH=\"$(pixmapdir)\" -DSYSCONF_PATH=\"$(SYSCONF_PATH)\" $(INCLUDEDIRS) $(DEPRECATED)
 
 ifdef DEBUG
-CXXTFLAGS=-DDEBUG -g $(CXXBASEFLAGS)
+CXXTFLAGS=-DDEBUG $(OPTFLAGS) $(CXXBASEFLAGS)
 else
-CXXTFLAGS=-DNDEBUG -Os $(CXXBASEFLAGS)
+CXXTFLAGS=-DNDEBUG $(OPTFLAGS) $(CXXBASEFLAGS)
 endif
 
 ifndef NO_LUA
