--- scite.orig/gtk/SciTE.desktop	2011-05-17 15:05:53.013000025 +0300
+++ scite/gtk/SciTE.desktop	2011-05-17 14:43:15.553000031 +0300
@@ -4,10 +4,13 @@
 Name=SciTE Text Editor
 Name[es]=Editor de textos SciTE
 Name[fr]=Éditeur de texte SciTE
+Name[tr]=Scite Metin Düzenleyici
 Comment=Edit your source files
 Comment[es]=Edita especializado para programación
 Comment[fr]=Éditer des fichiers sources
+Comment[tr]=Kaynak dosyalarını düzenle
 GenericName=Text Editor
+GenericName[tr]=Metin Düzenleyici
 Type=Application
 Exec=SciTE %F
 Icon=Sci48M.png
