diff -Naur plasma-orig/shells/desktop/desktopcorona.cpp plasma/shells/desktop/desktopcorona.cpp
--- plasma-orig/shells/desktop/desktopcorona.cpp	2009-06-23 19:06:03.668340022 +0300
+++ plasma/shells/desktop/desktopcorona.cpp	2009-06-23 19:17:06.423341250 +0300
@@ -209,6 +209,7 @@
         applet->setGlobalShortcut(KShortcut("Alt+F1"));
     }
 
+    loadDefaultApplet("showdesktop", panel);
     loadDefaultApplet("notifier", panel);
     loadDefaultApplet("pager", panel);
     loadDefaultApplet("tasks", panel);
@@ -224,6 +225,7 @@
     }
     engines->unloadEngine("powermanagement");
 
+    loadDefaultApplet("nm-applet", panel);
     loadDefaultApplet("digital-clock", panel);
     emit containmentAdded(panel);
 
