Index: kwin/compositingprefs.cpp
===================================================================
--- kwin/compositingprefs.cpp	(revision 980437)
+++ kwin/compositingprefs.cpp	(revision 980438)
@@ -240,7 +240,7 @@
     kDebug( 1212 ) << "GL version is" << mGLVersion;
     kDebug( 1212 ) << "XGL:" << ( mXgl ? "yes" : "no" );
 
-    if( mGLRenderer.startsWith( "Mesa DRI Intel" ))
+    if( mGLRenderer.startsWith( "Mesa DRI Intel" ) || mGLRenderer.startsWith( "Mesa DRI Mobile Intel" ))
         {
         mDriver = "intel";
         QStringList words = mGLRenderer.split(" ");
