--- kdebase-workspace-4.2.3/libs/kephal/configurations/xml/xmlconfigurations.cpp.orig	2008-12-04 10:34:51.000000000 +0200
+++ kdebase-workspace-4.2.3/libs/kephal/configurations/xml/xmlconfigurations.cpp	2009-05-21 00:34:46.206703501 +0300
@@ -27,6 +27,8 @@
 #include <QDebug>
 #include <QDir>
 
+#include <kdebug.h>
+
 
 namespace Kephal {
 
@@ -163,7 +165,14 @@
         m_awaitingConfirm(false)
     {
         QDir dir = QDir::home();
-        dir.cd(".local");
+        if (!dir.cd(".local"))
+        {
+            kDebug() << QDir::homePath() + "/.local directory not found, creating now.";
+            if (!dir.mkdir(".local"))
+                kWarning() << "Error during creation of " << QDir::homePath() + "/.local directory.";
+
+            dir.cd(".local");
+        }
         m_configPath = dir.filePath("screen-configurations.xml");
         
         m_externalConfiguration = new ExternalConfiguration(this);
