diff -Naur kdebase-workspace-4.2.3-orig/plasma/applets/kickoff/core/favoritesmodel.cpp kdebase-workspace-4.2.3/plasma/applets/kickoff/core/favoritesmodel.cpp
--- kdebase-workspace-4.2.3-orig/plasma/applets/kickoff/core/favoritesmodel.cpp	2009-05-10 19:26:11.236238413 +0300
+++ kdebase-workspace-4.2.3/plasma/applets/kickoff/core/favoritesmodel.cpp	2009-05-10 19:28:55.783239866 +0300
@@ -91,12 +91,12 @@
     }
     static QList<QString> defaultFavorites() {
         QList<QString> applications;
-        applications << "konqbrowser" << "kmail" << "systemsettings" << "dolphin";
+        applications << "mozillafirefox" << "Kontact" << "kopete" << "systemsettings" << "dolphin" << "amarok";
 
         QList<QString> desktopFiles;
 
         foreach(const QString& application, applications) {
-            KService::Ptr service = KService::serviceByStorageId("kde4-" + application + ".desktop");
+            KService::Ptr service = KService::serviceByStorageId(application + ".desktop");
             if (service) {
                 desktopFiles << service->entryPath();
             }
