--- a/ksmserver/shutdowndlg.cpp	(revision 985680)
+++ b/ksmserver/shutdowndlg.cpp	(revision 985681)
@@ -606,13 +606,7 @@
     if( KWindowSystem::compositingActive()) {
         clearMask();
     } else {
-        QBitmap mask(size());
-        mask.fill(Qt::color0);
-
-        QPainter p(&mask);
-        m_svg->resize(size());
-        m_svg->paint(&p, QRect(0, 0, width(), height()), "background");
-        setMask(mask);
+        setMask(m_svg->mask());
     }
 
     KDialog::centerOnScreen(this);
