--- a/ksmserver/shutdowndlg.cpp	(revision 973993)
+++ b/ksmserver/shutdowndlg.cpp	(revision 973994)
@@ -211,6 +211,8 @@
 
     p.save();
 
+    m_glowSvg->resize();
+
     if (m_glowOpacity > 0) {
         p.setOpacity(m_glowOpacity); // fade in
         m_glowSvg->paint(&p, QRect(0, 0, width(), height()), m_smallButton ? "button-small-hover" : "button-hover");
@@ -218,7 +220,6 @@
         m_glowSvg->paint(&p, QRect(0, 0, width(), height()), m_smallButton ? "button-small-normal" : "button-normal");
         p.setOpacity(1.0);
     } else {
-        m_glowSvg->resize();
         m_glowSvg->paint(&p, QRect(0, 0, width(), height()), m_smallButton ? "button-small-normal" : "button-normal");
     }
 
@@ -567,6 +568,7 @@
         QRect r = layout()->geometry();
         r.setWidth(m_pictureWidth);
 
+	m_svg->resize();
         m_svg->resize(m_svg->elementRect("picture").size());
         QPixmap picture = m_svg->pixmap("picture");
         m_svg->resize();
