--- kdm/config.def.orig	2009-05-08 18:24:41.509507815 +0300
+++ kdm/config.def	2009-05-08 18:25:37.150483607 +0300
@@ -1243,10 +1243,10 @@
  None/BO_NONE: no boot manager
  Grub/BO_GRUB: Grub boot manager
  Lilo/BO_LILO: Lilo boot manager (Linux on i386 &amp; x86-64 only)
-Default: None
+Default: Grub
 User: core
 User: greeter
-Instance: #Grub
+Instance: Grub
 Merge: kdm:UseLilo(P_UseLilo)
 Comment: &
 Description:
@@ -1448,7 +1448,7 @@
 
 Key: TerminateServer
 Type: bool
-Default: false
+Default: true
 User: core
 Instance: #:*/!
 Merge: xdm
@@ -2145,7 +2145,7 @@
 # This needs to come _in front_ of the font settings to be effective!
 Key: AntiAliasing
 Type: bool
-Default: false
+Default: true
 User: greeter
 Instance: #*/!
 Comment: &
@@ -2154,30 +2154,30 @@
 
 Key: GreetFont
 Type: string
-Default: "Serif,20,5,0,50,0"
-CDefault: "Serif,20,bold"
+Default: "DejaVu Sans,20,5,0,50,0"
+CDefault: "DejaVu Sans,20,bold"
 User: greeter:font
-Instance: #*/"Serif,20,5,0,50,0"
+Instance: #*/"DejaVu Sans,20,5,0,50,0"
 Comment: &
 Description:
  The font for the greeter headline.
 
 Key: StdFont
 Type: string
-Default: "Sans Serif,10,5,0,50,0"
-CDefault: "Sans Serif,10"
+Default: "DejaVu Sans,10,5,0,50,0"
+CDefault: "DejaVu Sans,10"
 User: greeter(normalFont):font
-Instance: #*/"Sans Serif,10,5,0,50,0"
+Instance: #*/"DejaVu Sans,10,5,0,50,0"
 Comment: &
 Description:
  The normal font used in the greeter.
 
 Key: FailFont
 Type: string
-Default: "Sans Serif,10,5,0,75,0"
-CDefault: "Sans Serif,10,bold"
+Default: "DejaVu Sans,10,5,0,75,0"
+CDefault: "DejaVu Sans,10,bold"
 User: greeter:font
-Instance: #*/"Sans Serif,10,5,0,75,0"
+Instance: #*/"DejaVu Sans,10,5,0,75,0"
 Comment: &
 Description:
  The font used for the <quote>Login Failed</quote> message.
@@ -2318,9 +2318,9 @@
  PreferAdmin/FACE_PREFER_ADMIN: prefer &lt;<option>FaceDir</option>&gt;, fallback on $<envar>HOME</envar>
  PreferUser/FACE_PREFER_USER: ... and the other way round
  UserOnly/FACE_USER_ONLY: from the user's <filename>$<envar>HOME</envar>/.face[.icon]</filename>
-Default: AdminOnly
+Default: PreferUser
 User: greeter
-Instance: #*/PreferUser
+Instance: */PreferUser
 Comment:
  Specify, where the users' pictures should be taken from.
 Description:
@@ -2646,7 +2646,7 @@
 
 Key: UseTheme
 Type: bool
-Default: false
+Default: true
 User: greeter
 Instance: */true
 Comment: &
--- kdm/config.def.orig	2009-07-03 17:03:58.833436709 +0300
+++ kdm/config.def	2009-07-03 17:04:19.651418959 +0300
@@ -2655,9 +2655,9 @@
 
 Key: Theme
 Type: string
-Default: ""
+Default: "pardus-air"
 User: greeter
-Instance: */KDMDATA "/themes/oxygen"
+Instance: */KDMDATA "/themes/pardus-air"
 Comment: &
 Description:
  The theme to use for the greeter. Can point to either a directory or an XML
