Index: plasma/shells/desktop/panelview.cpp
===================================================================
--- plasma/shells/desktop/panelview.cpp.orig
+++ plasma/shells/desktop/panelview.cpp
@@ -1144,7 +1144,7 @@ void PanelView::leaveEvent(QEvent *event
         } else {
             createUnhideTrigger();
         }
-    } else if (m_visibilityMode == AutoHide && !m_editting) {
+    } else if (containment() && m_visibilityMode == AutoHide && !m_editting) {
         // even if we dont have a popup, we'll start a timer, so
         // that the panel stays if the mouse only leaves for a
         // few ms
@@ -1382,6 +1382,10 @@ void PanelView::panelDeleted()
         c.deleteGroup();
     }
 
+    delete m_mousePollTimer;
+    m_mousePollTimer = 0;
+    m_strutsTimer->stop();
+
     deleteLater();
 }
 
