Index: workspace/powerdevil/daemon/PowerDevilDaemon.cpp
===================================================================
--- workspace/powerdevil/daemon/PowerDevilDaemon.cpp	(revision 976023)
+++ workspace/powerdevil/daemon/PowerDevilDaemon.cpp	(revision 976024)
@@ -845,7 +845,7 @@
 
 void PowerDevilDaemon::shutdownDialog()
 {
-    d->ksmServerIface->logout((int)KWorkSpace::ShutdownConfirmYes, (int)KWorkSpace::ShutdownTypeNone,
+    d->ksmServerIface->logout((int)KWorkSpace::ShutdownConfirmYes, (int)KWorkSpace::ShutdownTypeDefault,
                               (int)KWorkSpace::ShutdownModeDefault);
 }
 
