Index: trunk/KDE/kdebase/workspace/kwin/effects/shadow/shadow.cpp
===================================================================
--- kwin/effects/shadow.cpp	(revision 959836)
+++ kwin/effects/shadow.cpp	(revision 959837)
@@ -667,6 +667,8 @@
         shader->setUniform("opacity", opacity);
         shader->setUniform("saturation", saturation);
         shader->setUniform("brightness", brightness);
+        shader->setUniform("textureWidth", 1.0f);
+        shader->setUniform("textureHeight", 1.0f);
         }
     texture->bind();
     texture->enableNormalizedTexCoords();
