Index: plasma/wallpapers/image/image.cpp
===================================================================
--- plasma/wallpapers/image/image.cpp	(revision 944081)
+++ plasma/wallpapers/image/image.cpp	(revision 944082)
@@ -548,21 +543,15 @@
 
 void Image::updateBackground(int token, const QImage &img)
 {
-
     if (m_rendererToken == token) {
-
         m_oldPixmap = m_pixmap;
-        if (m_oldPixmap.isNull()) {
-            m_oldPixmap = QPixmap(img.size());
-            m_oldPixmap.fill(m_color);
-        }
         m_oldFadedPixmap = m_oldPixmap;
-
         m_pixmap = QPixmap::fromImage(img);
 
-        Plasma::Animator::self()->customAnimation(254, 1500, Plasma::Animator::LinearCurve, this, "updateFadedImage");
-        
-        suspendStartup(false);
+        if (!m_oldPixmap.isNull()) {
+            Plasma::Animator::self()->customAnimation(254, 1500, Plasma::Animator::LinearCurve, this, "updateFadedImage");
+            suspendStartup(false);
+        }
     }
 }
 
@@ -591,7 +580,6 @@
 
 void Image::updateFadedImage(qreal frame)
 {
-
     //If we are done, delete the pixmaps and don't draw.
     if (frame == 1) {
         m_oldFadedPixmap = QPixmap();
@@ -599,7 +587,7 @@
         emit update(boundingRect());
         return;
     }
-   
+
     //Create the faded image.
     m_oldFadedPixmap.fill(Qt::transparent);
 
