Index: KDE/kdebase/workspace/plasma/wallpapers/image/image.cpp
===================================================================
--- KDE/kdebase/workspace/plasma/wallpapers/image/image.cpp	(revision 938023)
+++ KDE/kdebase/workspace/plasma/wallpapers/image/image.cpp	(revision 938024)
@@ -25,6 +25,7 @@
 #include <KStandardDirs>
 
 #include <Plasma/Theme>
+#include <Plasma/Animator>
 #include "backgroundlistmodel.h"
 #include "backgrounddelegate.h"
 #include "ksmserver_interface.h"
@@ -233,6 +234,12 @@
     // bitmapBackground already has the size of the viewport)
     painter->drawPixmap(exposedRect, m_pixmap, exposedRect);
 
+    if (!m_oldFadedPixmap.isNull()) {
+        // Put old faded image on top.
+        painter->setCompositionMode(QPainter::CompositionMode_SourceAtop);
+        painter->drawPixmap(exposedRect, m_oldFadedPixmap, exposedRect);
+    }
+
     // restore transformation and composition mode
     painter->restore();
 }
@@ -527,9 +534,20 @@
 
 void Image::updateBackground(int token, const QImage &img)
 {
+
     if (m_rendererToken == token) {
+
+        m_oldPixmap = m_pixmap;
+        if (m_oldPixmap.isNull()) {
+            m_oldPixmap = QPixmap(img.size());
+            m_oldPixmap.fill(m_color);
+        }
+        m_oldFadedPixmap = m_oldPixmap;
+
         m_pixmap = QPixmap::fromImage(img);
-        emit update(boundingRect());
+
+        Plasma::Animator::self()->customAnimation(254, 1500, Plasma::Animator::LinearCurve, this, "updateFadedImage");
+        
         suspendStartup(false);
     }
 }
@@ -557,4 +575,31 @@
     }
 }
 
+void Image::updateFadedImage(qreal frame)
+{
+
+    //If we are done, delete the pixmaps and don't draw.
+    if (frame == 1) {
+        m_oldFadedPixmap = QPixmap();
+        m_oldPixmap = QPixmap();
+        emit update(boundingRect());
+        return;
+    }
+   
+    //Create the faded image.
+    m_oldFadedPixmap.fill(Qt::transparent);
+
+    QPainter p;
+    p.begin(&m_oldFadedPixmap);
+    p.drawPixmap(0, 0, m_oldPixmap);
+
+    p.setCompositionMode(QPainter::CompositionMode_DestinationIn);  
+    p.fillRect(m_oldFadedPixmap.rect(), QColor(0, 0, 0, 254 * (1-frame)));//255*((150 - frame)/150)));
+
+    p.end();
+
+    emit update(boundingRect());
+
+}
+
 #include "image.moc"
Index: KDE/kdebase/workspace/plasma/wallpapers/image/image.h
===================================================================
--- KDE/kdebase/workspace/plasma/wallpapers/image/image.h	(revision 938023)
+++ KDE/kdebase/workspace/plasma/wallpapers/image/image.h	(revision 938024)
@@ -49,6 +49,7 @@
         void showFileDialog();
         void updateScreenshot(QPersistentModelIndex index);
         void removeBackground(const QString &path);
+        void updateFadedImage(qreal frame);
 
     protected:
         void init(const KConfigGroup &config);
@@ -76,6 +77,8 @@
         QList<Background *> m_slideshowBackgrounds;
         QTimer m_timer;
         QPixmap m_pixmap;
+        QPixmap m_oldPixmap;
+        QPixmap m_oldFadedPixmap;
         int m_currentSlide;
         qreal m_ratio;
         BackgroundListModel *m_model;
