diff -ur xorg-input-7.2_rc3.orig/xf86-input-keyboard-1.1.1/src/kbd.c xorg-input-7.2_rc3/xf86-input-keyboard-1.1.1/src/kbd.c
--- xorg-input-7.2_rc3.orig/xf86-input-keyboard-1.1.1/src/kbd.c	2006-10-12 01:23:44.000000000 +0300
+++ xorg-input-7.2_rc3/xf86-input-keyboard-1.1.1/src/kbd.c	2007-02-05 20:55:12.696646328 +0200
@@ -389,7 +389,22 @@
    InputInfoPtr pInfo = (InputInfoPtr) device->public.devicePrivate;
    KbdDevPtr pKbd = (KbdDevPtr) pInfo->private;
 
-   if ( ctrl->leds & XCOMP ) {
+   if ( ctrl->leds & XLED1) {
+       pKbd->keyLeds |= CAPSFLAG;
+   } else {
+       pKbd->keyLeds &= ~CAPSFLAG;
+   }
+   if ( ctrl->leds & XLED2) {
+       pKbd->keyLeds |= NUMFLAG;
+   } else {
+       pKbd->keyLeds &= ~NUMFLAG;
+   }
+   if ( ctrl->leds & XLED3) {
+       pKbd->keyLeds |= SCROLLFLAG;
+   } else {
+       pKbd->keyLeds &= ~SCROLLFLAG;
+   }
+   if ( ctrl->leds & (XCOMP|XLED4) ) {
        pKbd->keyLeds |= COMPOSEFLAG;
    } else {
        pKbd->keyLeds &= ~COMPOSEFLAG;
Yalnızca xorg-input-7.2_rc3/xf86-input-keyboard-1.1.1/src'da: kbd.c.orig
