--- src/NetworkState.cpp.orig	2007-11-12 14:27:55.000000000 +0200
+++ src/NetworkState.cpp	2008-04-22 20:09:37.000000000 +0300
@@ -19,6 +19,8 @@
 
 #include <sstream>
 
+#include <algorithm>
+
 #include "NetworkState.h"
 #include "NetworkMessage.h"
 #include "NetworkGame.h"
--- src/ReplayRecorder.cpp.orig	2007-11-12 14:27:55.000000000 +0200
+++ src/ReplayRecorder.cpp	2008-04-22 20:09:02.000000000 +0300
@@ -27,6 +27,8 @@
 #include <boost/crc.hpp>
 #include <SDL/SDL.h>
 
+#include <algorithm>
+
 ChecksumException::ChecksumException(std::string filename, uint32_t expected, uint32_t real)
 {
 	std::stringstream errorstr;
--- src/State.cpp.orig	2007-11-12 14:27:55.000000000 +0200
+++ src/State.cpp	2008-04-22 20:09:26.000000000 +0300
@@ -31,6 +31,8 @@
 #include <physfs.h>
 #include <sstream>
 
+#include <algorithm>
+
 State::State()
 {
 
