diff -Naur viking-1.1.orig/src/viking.desktop.in viking-1.1/src/viking.desktop.in
--- viking-1.1.orig/src/viking.desktop.in	2010-01-14 22:19:52.000000000 +0200
+++ viking-1.1/src/viking.desktop.in	2011-05-08 11:57:21.303000019 +0300
@@ -1,8 +1,10 @@
 [Desktop Entry]
 Version=1.0
-_Name=Viking
-_GenericName=GPS Data Manager
-_Comment=Map Based GPS Data Manager (live data capable).
+Name=Viking
+GenericName=GPS Data Manager
+GenericName[tr]=GPS Veri yöneticisi
+Comment=Map Based GPS Data Manager (live data capable).
+Comment[tr]=Geçerli haritada GPS verilerini yönetin.
 Exec=viking %F
 Icon=viking
 Terminal=false
