Index: linux-2.6.25/drivers/net/tun.c
===================================================================
--- linux-2.6.25.orig/drivers/net/tun.c
+++ linux-2.6.25/drivers/net/tun.c
@@ -501,7 +501,7 @@ static int tun_set_iff(struct file *file
 		if (((tun->owner != -1 &&
 		      current->euid != tun->owner) ||
 		     (tun->group != -1 &&
-		      current->egid != tun->group)) &&
+			  !in_egroup_p(tun->group))) &&
 		     !capable(CAP_NET_ADMIN))
 			return -EPERM;
 	}
