--- xkbdata-1.0.1/symbols/inet	2007-05-09 21:51:50.000000000 +0200
+++ xkbdata-1.0.1/symbols/inet	2007-05-09 21:53:17.000000000 +0200
@@ -1138,6 +1138,24 @@
     key <I66>   {       [ XF86Go                ]       };
 };
 
+// Logitech Cordless Desktop EX110
+partial alphanumeric_keys
+xkb_symbols "logicdex" {
+
+    key <I20>	{	[ XF86AudioMute		]	};
+    key <I2E>	{	[ XF86AudioLowerVolume	]	};
+    key <I30>	{	[ XF86AudioRaiseVolume	]	};
+    key <I22>	{	[ XF86AudioPlay, XF86AudioPause ] };
+    key <I24>	{	[ XF86AudioStop		]	};
+    key <I10>	{	[ XF86AudioPrev		]	};
+    key <I19>	{	[ XF86AudioNext		]	};
+
+    key <I32>	{	[ XF86HomePage		]	};
+    key <I6C>	{	[ XF86Mail		]	};
+    key <I21>	{	[ XF86Calculator	]	};
+    key <I6A>	{	[ XF86Back		]	};
+};
+
 // Logitech Cordless Desktop Pro
 partial alphanumeric_keys
 xkb_symbols "logicdp" {
--- xkbdata-1.0.1/rules/xorg.lst	2006-10-05 11:53:24.000000000 +0300
+++ xkbdata-1.0.1/rules/xorg.lst	2007-06-12 06:31:43.000000000 +0300
@@ -73,6 +73,7 @@
   logicdpa2       Logitech Cordless Desktop Pro (alternate option2)
   logicdo         Logitech Cordless Desktop Optical
   logicfn         Logitech Cordless Freedom/Desktop Navigator
+  logicdex        Logitech Cordless Desktop EX110
   logicdn         Logitech Cordless Desktop Navigator
   logidak         Logitech Deluxe Access Keyboard
   logiitc         Logitech iTouch Cordless Keyboard (model Y-RB6)
--- xkbdata-1.0.1/rules/xorg.xml	2006-10-05 11:53:24.000000000 +0300
+++ xkbdata-1.0.1/rules/xorg.xml	2007-06-12 06:34:17.000000000 +0300
@@ -1445,6 +1445,29 @@
     </model>
     <model>
       <configItem>
+        <name>logicdex</name>
+        <description>Logitech Cordless Desktop EX110</description>
+        <description xml:lang="az">Logitech Cordless Desktop EX110</description>
+        <description xml:lang="bg">Logitech Cordless Desktop EX110</description>
+        <description xml:lang="cs">Logitech Cordless Desktop EX110</description>
+        <description xml:lang="da">Logitech Cordless Desktop EX110</description>
+        <description xml:lang="en_GB">Logitech Cordless Desktop EX110</description>
+        <description xml:lang="fr">Logitech Cordless Desktop EX110</description>
+        <description xml:lang="hu">Logitech Cordless Desktop EX110</description>
+        <description xml:lang="nl">Logitech Cordless Desktop EX110</description>
+        <description xml:lang="ru">Logitech Cordless Desktop EX110</description>
+        <description xml:lang="sk">Logitech Cordless Desktop EX110</description>
+        <description xml:lang="sq">Logitech Cordless Desktop EX110</description>
+        <description xml:lang="sr">Logitech Cordless Desktop EX110</description>
+        <description xml:lang="sv">Logitech Cordless Desktop EX110</description>
+        <description xml:lang="tr">Logitech Cordless Desktop EX110</description>
+        <description xml:lang="uk">Logitech Cordless Desktop EX110</description>
+        <description xml:lang="vi">Logitech Cordless Desktop EX110</description>
+        <description xml:lang="zh_CN">Logitech Cordless Desktop EX110</description>
+      </configItem>
+    </model> 
+    <model>
+      <configItem>
         <name>logicdp</name>
         <description>Logitech Cordless Desktop Pro</description>
         <description xml:lang="af">Logitech Cordless Desktop Pro</description>
--- xkbdata-1.0.1/rules/xorg	2005-08-12 13:50:44.000000000 +0300
+++ xkbdata-1.0.1/rules/xorg	2007-06-12 06:56:42.000000000 +0300
@@ -34,7 +34,7 @@
               hp500fa hp5xx hp5185 \
               honeywell_euroboard \
               rapidaccess rapidaccess2 rapidaccess2a \
-              ltcd logiaccess logicdp logicdpa logicdit logicink logiciink \
+              ltcd logiaccess logicdex logicdp logicdpa logicdit logicink logiciink \
               logiinkse logiinkseusb logiitc logiik itouch logiultrax \
               mx1998 mx2500 mx2750 \
               microsoftinet microsoftpro microsoftprousb microsoftprooem microsoftprose \
