--- acx_config.h.org	2007-01-04 22:50:11.000000000 +0200
+++ acx_config.h	2007-01-04 22:50:46.000000000 +0200
@@ -7,7 +7,7 @@
 /* set to 0 if you don't want any debugging code to be compiled in */
 /* set to 1 if you want some debugging */
 /* set to 2 if you want extensive debug log */
-#define ACX_DEBUG 2
+#define ACX_DEBUG 0
 #define ACX_DEFAULT_MSG (L_ASSOC|L_INIT)
 
 /* assume 32bit I/O width
