--- build/contents/build_contents.py~	2009-02-22 19:59:55.000000000 +0200
+++ build/contents/build_contents.py	2009-03-12 21:56:48.000000000 +0200
@@ -8,7 +8,7 @@
     NVDXT_EXECUTABLE = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', 'tools', 'dxt', 'nvdxt')
 else:
     ARCH = os.uname()[-1]
-    NVDXT_EXECUTABLE = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', 'tools', 'dxt', 'nvcompress_'+ARCH)
+    NVDXT_EXECUTABLE = "/usr/bin/nvcompress"
     if not os.path.isfile(NVDXT_EXECUTABLE):
         print "tool exetubale file not found: %s" % NVDXT_EXECUTABLE
         print "please download and compile this: http://code.google.com/p/nvidia-texture-tools/"
@@ -119,11 +119,6 @@
       #print entry
       counter += packDirectory(entry)
   print "done with all. processed %d files in %0.2f seconds." % (counter,  time.time()-ts)
-  wait=True
-  if len(sys.argv) > 1 and sys.argv[1] == 'nowait':
-    wait=False
-  if wait:
-    raw_input("Press enter to continue.")
 
 if __name__ == '__main__':
   main()
