--- streams/build_streams.py~	2009-03-12 22:01:40.000000000 +0200
+++ streams/build_streams.py	2009-03-12 22:01:41.000000000 +0200
@@ -8,7 +8,7 @@
 if platform.system() == 'Windows':
     NVDXT_EXECUTABLE = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', 'tools', 'dxt', 'nvdxt')
 else:
-    NVDXT_EXECUTABLE = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', 'tools', 'dxt', 'nvcompress_'+ARCH)
+    NVDXT_EXECUTABLE = "/usr/bin/nvcompress"
     if not os.path.isfile(NVDXT_EXECUTABLE):
         print "tool exetubale file not found: %s" % NVDXT_EXECUTABLE
         print "please download and compile this: http://code.google.com/p/nvidia-texture-tools/"
@@ -299,11 +299,6 @@
     centerPrint('done')
     clearTempDir()
     print "done with all. processed %d directories in %0.2f seconds." % (counter,  time.time()-ts)
-    wait=True
-    if len(sys.argv) > 1 and sys.argv[1] == 'nowait':
-      wait=False
-    if wait:
-      raw_input("Press enter to continue.")
 
 if __name__ == '__main__':
     main()
