diff -Naur 0.36.1.orig/build/dependencies/hydrax/Hydrax/Prerequisites.h 0.36.1/build/dependencies/hydrax/Hydrax/Prerequisites.h
--- 0.36.1.orig/build/dependencies/hydrax/Hydrax/Prerequisites.h	2009-03-12 16:11:52.000000000 +0200
+++ 0.36.1/build/dependencies/hydrax/Hydrax/Prerequisites.h	2009-03-12 16:12:50.000000000 +0200
@@ -26,7 +26,7 @@
 #define _Hydrax_Prerequisites_H_
 
 /// Include external headers
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 
 /// Define the dll export qualifier if compiling for Windows
 #if OGRE_PLATFORM == OGRE_PLATFORM_WIN32
diff -Naur 0.36.1.orig/build/dependencies/mygui/include/MyGUI_Common.h 0.36.1/build/dependencies/mygui/include/MyGUI_Common.h
--- 0.36.1.orig/build/dependencies/mygui/include/MyGUI_Common.h	2009-03-12 16:11:52.000000000 +0200
+++ 0.36.1/build/dependencies/mygui/include/MyGUI_Common.h	2009-03-12 16:12:50.000000000 +0200
@@ -20,7 +20,7 @@
 #include "MyGUI_Version.h"
 #include "MyGUI_WidgetStyle.h"
 
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 
 // for debugging
 #if MYGUI_COMPILER == MYGUI_COMPILER_MSVC
diff -Naur 0.36.1.orig/build/dependencies/mygui/include/MyGUI_PluginManager.h 0.36.1/build/dependencies/mygui/include/MyGUI_PluginManager.h
--- 0.36.1.orig/build/dependencies/mygui/include/MyGUI_PluginManager.h	2009-03-12 16:11:52.000000000 +0200
+++ 0.36.1/build/dependencies/mygui/include/MyGUI_PluginManager.h	2009-03-12 16:12:50.000000000 +0200
@@ -14,7 +14,7 @@
 #include "MyGUI_Version.h"
 #include "MyGUI_ResourceManager.h"
 
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 
 #include "MyGUI_LastHeader.h"
 
diff -Naur 0.36.1.orig/build/dependencies/mygui/include/MyGUI_Precompiled.h 0.36.1/build/dependencies/mygui/include/MyGUI_Precompiled.h
--- 0.36.1.orig/build/dependencies/mygui/include/MyGUI_Precompiled.h	2009-03-12 16:11:52.000000000 +0200
+++ 0.36.1/build/dependencies/mygui/include/MyGUI_Precompiled.h	2009-03-12 16:12:50.000000000 +0200
@@ -28,7 +28,7 @@
 #include "MyGUI_Convert.h"
 #include "MyGUI_XmlDocument.h"
 #include "MyGUI_CastWidget.h"
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include "MyGUI_LastHeader.h"
 
 #endif // __MYGUI_PRECOMPILED_H__
diff -Naur 0.36.1.orig/build/dependencies/ogrenew/ReferenceApplication/ReferenceAppLayer/include/OgreRefAppPrerequisites.h 0.36.1/build/dependencies/ogrenew/ReferenceApplication/ReferenceAppLayer/include/OgreRefAppPrerequisites.h
--- 0.36.1.orig/build/dependencies/ogrenew/ReferenceApplication/ReferenceAppLayer/include/OgreRefAppPrerequisites.h	2009-03-12 16:12:29.000000000 +0200
+++ 0.36.1/build/dependencies/ogrenew/ReferenceApplication/ReferenceAppLayer/include/OgreRefAppPrerequisites.h	2009-03-12 16:12:52.000000000 +0200
@@ -36,7 +36,7 @@
 #include <ode/odecpp_collision.h>
 
 // Include main application-facing Ogre header
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 
 
 // To save us some typing
diff -Naur 0.36.1.orig/build/dependencies/ogrenew/Samples/Compositor/src/Compositor.cpp 0.36.1/build/dependencies/ogrenew/Samples/Compositor/src/Compositor.cpp
--- 0.36.1.orig/build/dependencies/ogrenew/Samples/Compositor/src/Compositor.cpp	2009-03-12 16:12:05.000000000 +0200
+++ 0.36.1/build/dependencies/ogrenew/Samples/Compositor/src/Compositor.cpp	2009-03-12 16:12:59.000000000 +0200
@@ -26,7 +26,7 @@
             added gui framework to demo
 */
 
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 
 #if OGRE_PLATFORM == OGRE_PLATFORM_WIN32
 #define WIN32_LEAN_AND_MEAN
diff -Naur 0.36.1.orig/build/dependencies/ogrenew/Samples/Compositor/src/CompositorDemo_FrameListener.cpp 0.36.1/build/dependencies/ogrenew/Samples/Compositor/src/CompositorDemo_FrameListener.cpp
--- 0.36.1.orig/build/dependencies/ogrenew/Samples/Compositor/src/CompositorDemo_FrameListener.cpp	2009-03-12 16:12:05.000000000 +0200
+++ 0.36.1/build/dependencies/ogrenew/Samples/Compositor/src/CompositorDemo_FrameListener.cpp	2009-03-12 16:12:59.000000000 +0200
@@ -12,7 +12,7 @@
 -----------------------------------------------------------------------------
 */
 
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include <OgreTimer.h>
 #include <CEGUI/CEGUIPropertyHelper.h>
 
diff -Naur 0.36.1.orig/build/dependencies/ogrenew/Tools/3dsmaxExport/LEXIExporter/LexiExport/Sources/LexiStdAfx.h 0.36.1/build/dependencies/ogrenew/Tools/3dsmaxExport/LEXIExporter/LexiExport/Sources/LexiStdAfx.h
--- 0.36.1.orig/build/dependencies/ogrenew/Tools/3dsmaxExport/LEXIExporter/LexiExport/Sources/LexiStdAfx.h	2009-03-12 16:11:54.000000000 +0200
+++ 0.36.1/build/dependencies/ogrenew/Tools/3dsmaxExport/LEXIExporter/LexiExport/Sources/LexiStdAfx.h	2009-03-12 16:12:51.000000000 +0200
@@ -58,7 +58,7 @@
 
 //
 
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include <OgreNoMemoryMacros.h>
 
 #ifdef _DEBUG
diff -Naur 0.36.1.orig/build/dependencies/ogrenew/Tools/VRMLConverter/src/stdafx.h 0.36.1/build/dependencies/ogrenew/Tools/VRMLConverter/src/stdafx.h
--- 0.36.1.orig/build/dependencies/ogrenew/Tools/VRMLConverter/src/stdafx.h	2009-03-12 16:11:54.000000000 +0200
+++ 0.36.1/build/dependencies/ogrenew/Tools/VRMLConverter/src/stdafx.h	2009-03-12 16:12:51.000000000 +0200
@@ -4,7 +4,7 @@
 #include <vrmllib/nodes.h>
 #include <vrmllib/file.h>
 
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 
 using namespace Ogre;
 using namespace vrmllib;
diff -Naur 0.36.1.orig/build/LayoutEditor/source/Base/BaseManager.h 0.36.1/build/LayoutEditor/source/Base/BaseManager.h
--- 0.36.1.orig/build/LayoutEditor/source/Base/BaseManager.h	2009-03-12 16:11:51.000000000 +0200
+++ 0.36.1/build/LayoutEditor/source/Base/BaseManager.h	2009-03-12 16:12:50.000000000 +0200
@@ -7,7 +7,7 @@
 #ifndef __BASE_MANAGER_H__
 #define __BASE_MANAGER_H__
 
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include <OIS.h>
 #include <MyGUI.h>
 #include "StatisticInfo.h"
diff -Naur 0.36.1.orig/build/LayoutEditor/source/precompiled.h 0.36.1/build/LayoutEditor/source/precompiled.h
--- 0.36.1.orig/build/LayoutEditor/source/precompiled.h	2009-03-12 16:11:51.000000000 +0200
+++ 0.36.1/build/LayoutEditor/source/precompiled.h	2009-03-12 16:12:50.000000000 +0200
@@ -1,4 +1,4 @@
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include <OIS.h>
 #include <MyGUI.h>
 #include <vector>
diff -Naur 0.36.1.orig/build/main/source/BeamStats.h 0.36.1/build/main/source/BeamStats.h
--- 0.36.1.orig/build/main/source/BeamStats.h	2009-03-12 16:11:51.000000000 +0200
+++ 0.36.1/build/main/source/BeamStats.h	2009-03-12 16:12:50.000000000 +0200
@@ -22,7 +22,7 @@
 #ifndef __BeamStats_H__
 #define __BeamStats_H__
 
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include <map>
 
 class Beam;
diff -Naur 0.36.1.orig/build/main/source/CacheSystem.h 0.36.1/build/main/source/CacheSystem.h
--- 0.36.1.orig/build/main/source/CacheSystem.h	2009-03-12 16:11:51.000000000 +0200
+++ 0.36.1/build/main/source/CacheSystem.h	2009-03-12 16:12:50.000000000 +0200
@@ -22,7 +22,7 @@
 #ifndef __CacheSystem_H__
 #define __CacheSystem_H__
 
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include <vector>
 #include <map>
 #include "sha1.h"
diff -Naur 0.36.1.orig/build/main/source/Character.h 0.36.1/build/main/source/Character.h
--- 0.36.1.orig/build/main/source/Character.h	2009-03-12 16:11:51.000000000 +0200
+++ 0.36.1/build/main/source/Character.h	2009-03-12 16:12:50.000000000 +0200
@@ -21,7 +21,7 @@
 #ifndef CHARACTER_H_
 #define CHARACTER_H_
 
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include <OgreVector3.h>
 
 class Water;
diff -Naur 0.36.1.orig/build/main/source/DynamicLines.cpp 0.36.1/build/main/source/DynamicLines.cpp
--- 0.36.1.orig/build/main/source/DynamicLines.cpp	2009-03-12 16:11:51.000000000 +0200
+++ 0.36.1/build/main/source/DynamicLines.cpp	2009-03-12 16:12:57.000000000 +0200
@@ -18,7 +18,7 @@
 along with Rigs of Rods.  If not, see <http://www.gnu.org/licenses/>.
 */
 #include "DynamicLines.h"
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include <cassert>
 #include <cmath>
 
diff -Naur 0.36.1.orig/build/main/source/gui_loader.h 0.36.1/build/main/source/gui_loader.h
--- 0.36.1.orig/build/main/source/gui_loader.h	2009-03-12 16:11:51.000000000 +0200
+++ 0.36.1/build/main/source/gui_loader.h	2009-03-12 16:12:50.000000000 +0200
@@ -22,7 +22,7 @@
 #define GUI_LOADER_H
 
 #include <MyGUI.h>
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 
 #define UILOADER GUI_Loader::Instance()
 #define UI_PROGRESSBAR_AUTOTRACK -1
diff -Naur 0.36.1.orig/build/main/source/InputEngine.cpp 0.36.1/build/main/source/InputEngine.cpp
--- 0.36.1.orig/build/main/source/InputEngine.cpp	2009-03-12 16:11:51.000000000 +0200
+++ 0.36.1/build/main/source/InputEngine.cpp	2009-03-12 16:12:57.000000000 +0200
@@ -22,7 +22,7 @@
 */
 #include "InputEngine.h"
 
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include <OgreStringConverter.h>
 #include <OgreException.h>
 #include <OgreWindowEventUtilities.h>
diff -Naur 0.36.1.orig/build/main/source/MapControl.h 0.36.1/build/main/source/MapControl.h
--- 0.36.1.orig/build/main/source/MapControl.h	2009-03-12 16:11:51.000000000 +0200
+++ 0.36.1/build/main/source/MapControl.h	2009-03-12 16:12:50.000000000 +0200
@@ -23,7 +23,7 @@
 
 //#define OGREPLUGIN
 
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include "OgreTextAreaOverlayElement.h"
 #include <vector>
 
diff -Naur 0.36.1.orig/build/main/source/ProceduralManager.h 0.36.1/build/main/source/ProceduralManager.h
--- 0.36.1.orig/build/main/source/ProceduralManager.h	2009-03-12 16:11:51.000000000 +0200
+++ 0.36.1/build/main/source/ProceduralManager.h	2009-03-12 16:12:50.000000000 +0200
@@ -24,7 +24,7 @@
 #define __ProceduralManager_H__
 
 
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include <vector>
 
 class Road2;
diff -Naur 0.36.1.orig/build/main/source/pythonBinding.cpp 0.36.1/build/main/source/pythonBinding.cpp
--- 0.36.1.orig/build/main/source/pythonBinding.cpp	2009-03-12 16:11:51.000000000 +0200
+++ 0.36.1/build/main/source/pythonBinding.cpp	2009-03-12 16:12:57.000000000 +0200
@@ -26,7 +26,7 @@
 #include "collisions.h"
 #include "Beam.h"
 
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include <boost/python.hpp>
 #include <frameobject.h> // for python's _frame object (used in traceback function)
 
diff -Naur 0.36.1.orig/build/main/source/ResourceBuffer.h 0.36.1/build/main/source/ResourceBuffer.h
--- 0.36.1.orig/build/main/source/ResourceBuffer.h	2009-03-12 16:11:51.000000000 +0200
+++ 0.36.1/build/main/source/ResourceBuffer.h	2009-03-12 16:12:50.000000000 +0200
@@ -20,7 +20,7 @@
 #ifndef RESOURCEBUFFER_H_
 #define RESOURCEBUFFER_H_
 
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include <OgreResource.h>
 
 class ResourceBuffer : public Ogre::ManualResourceLoader
diff -Naur 0.36.1.orig/build/main/source/Skidmark.cpp 0.36.1/build/main/source/Skidmark.cpp
--- 0.36.1.orig/build/main/source/Skidmark.cpp	2009-03-12 16:11:51.000000000 +0200
+++ 0.36.1/build/main/source/Skidmark.cpp	2009-03-12 16:12:57.000000000 +0200
@@ -19,7 +19,7 @@
 */
 
 #include "Skidmark.h"
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include <cassert>
 #include <cmath>
 
diff -Naur 0.36.1.orig/build/main/source/tsmheightfinder.cpp 0.36.1/build/main/source/tsmheightfinder.cpp
--- 0.36.1.orig/build/main/source/tsmheightfinder.cpp	2009-03-12 16:11:51.000000000 +0200
+++ 0.36.1/build/main/source/tsmheightfinder.cpp	2009-03-12 16:12:58.000000000 +0200
@@ -19,7 +19,7 @@
 */
 #include "heightfinder.h"
 //#include <OgreTerrainSceneManager.h>
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include <OgreConfigFile.h>
 #include <OgreLogManager.h>
 #include <OgreStringConverter.h>
diff -Naur 0.36.1.orig/build/main/unused/etm.h 0.36.1/build/main/unused/etm.h
--- 0.36.1.orig/build/main/unused/etm.h	2009-03-12 16:11:51.000000000 +0200
+++ 0.36.1/build/main/unused/etm.h	2009-03-12 16:12:50.000000000 +0200
@@ -1,7 +1,7 @@
 #ifndef ETM_H_
 #define ETM_H_
 
-#include <Ogre.h>
+#include <OGRE/Ogre.h>
 #include "ETTerrainManager.h"
 #include "ETTerrainInfo.h"
 #include "ETBrush.h"
