--- build/dependencies/CMakeLists.txt~	2009-03-12 15:22:46.000000000 +0200
+++ build/dependencies/CMakeLists.txt	2009-03-12 15:22:46.000000000 +0200
@@ -1,17 +1,8 @@
 project(RoR_Dependencies)
 
 # non - ogre dependent
-if(ROR_BUILD_LUASCRIPT)
- add_subdirectory("lua-5.1.2")
-endif(ROR_BUILD_LUASCRIPT)
 
 add_subdirectory("mysocketw")
-add_subdirectory("OIS-HEAD")
-#add_subdirectory("openal-soft-1.5.304")
-add_subdirectory("openal-soft-1.6.372")
-
-set(OGRE_SET_DISABLE_FREEIMAGE 1)
-add_subdirectory("ogrenew")
 
 # OGRE dependent
 if(CAELUM_BUILD_NEW)
--- build/configurator/CMakeLists.txt~	2009-03-12 15:31:20.000000000 +0200
+++ build/configurator/CMakeLists.txt	2009-03-12 15:31:20.000000000 +0200
@@ -43,7 +43,7 @@
 
   add_executable(rorconfig ${config_source} ${config_headers} ${config_res})
   TARGET_LINK_LIBRARIES(rorconfig ${wxWidgets_LIBRARIES} )
-  target_link_libraries(rorconfig OgreMain ois)
+  target_link_libraries(rorconfig OgreMain OIS)
 
   IF(WIN32)
     SET_TARGET_PROPERTIES(rorconfig PROPERTIES COMPILE_FLAGS "")
--- build/main/CMakeLists.txt~	2009-03-12 15:36:45.000000000 +0200
+++ build/main/CMakeLists.txt	2009-03-12 15:36:45.000000000 +0200
@@ -98,7 +98,7 @@
 add_executable(RoR ${ror_src} source/icon.rc)
 
 # add dependencies
-add_dependencies(RoR OgreMain caelum mysocketw ois mygui mofilereader)
+add_dependencies(RoR OgreMain caelum mysocketw OIS mygui mofilereader)
 
 # optional features
 if(ROR_BUILD_HYDRAX)
@@ -132,7 +132,7 @@
   caelum
   mysocketw
   OgreMain
-  ois
+  OIS
   mygui
   mofilereader
 )
--- build/dependencies/CMakeLists.txt~	2009-03-12 16:34:47.000000000 +0200
+++ build/dependencies/CMakeLists.txt	2009-03-12 16:34:47.000000000 +0200
@@ -24,5 +24,3 @@
 if(ROR_BUILD_NEWNET)
 add_subdirectory("raknet")
 endif(ROR_BUILD_NEWNET)
-
-add_subdirectory("mygui")
\ Dosya sonunda yenisatır yok.
