Index: metamonitor-0.4.5/src/mconfiguration.cpp
===================================================================
--- metamonitor-0.4.5.orig/src/mconfiguration.cpp
+++ metamonitor-0.4.5/src/mconfiguration.cpp
@@ -46,7 +46,7 @@ void MConfiguration::read()
 	cfg->setGroup("General Options");
 	groupMessages	= cfg->readBoolEntry( "GroupMessages", true );
 	pausePopups		= cfg->readBoolEntry( "PausePopups", false );
-	logFile 		= cfg->readEntry( "LogFile", "/var/log/everything/current" );
+	logFile 		= cfg->readEntry( "LogFile", "/var/log/messages" );
 	updateInterval	= cfg->readNumEntry( "UpdateInterval", 2000 );
 	popupsLifetime	= cfg->readNumEntry( "PopupsLifetime", 4000 );
 	defaultRegExp   = (LogDaemonType)cfg->readNumEntry( "DefaultRegExpType", MetaLog );
