diff -ur alsa-utils-1.0.16.orig/configure.in alsa-utils-1.0.16/configure.in
--- alsa-utils-1.0.16.orig/configure.in	2008-05-02 15:36:19.000000000 +0300
+++ alsa-utils-1.0.16/configure.in	2008-05-02 15:36:34.000000000 +0300
@@ -101,8 +101,6 @@
 
 AC_OUTPUT(Makefile alsactl/Makefile alsamixer/Makefile amidi/Makefile amixer/Makefile \
 	  m4/Makefile po/Makefile.in \
-	  alsaconf/alsaconf alsaconf/Makefile \
-	  alsaconf/po/Makefile \
 	  salsa/Makefile \
 	  aplay/Makefile include/Makefile iecset/Makefile utils/Makefile \
 	  utils/alsa-utils.spec seq/Makefile seq/aconnect/Makefile \
diff -ur alsa-utils-1.0.16.orig/Makefile.am alsa-utils-1.0.16/Makefile.am
--- alsa-utils-1.0.16.orig/Makefile.am	2008-05-02 15:36:19.000000000 +0300
+++ alsa-utils-1.0.16/Makefile.am	2008-05-02 15:36:34.000000000 +0300
@@ -4,7 +4,7 @@
 else
 ALSAMIXER_DIR=
 endif
-SUBDIRS= include alsactl alsaconf $(ALSAMIXER_DIR) amidi amixer aplay iecset seq speaker-test utils m4 po salsa
+SUBDIRS= include alsactl $(ALSAMIXER_DIR) amidi amixer aplay iecset seq speaker-test utils m4 po salsa
 EXTRA_DIST= config.rpath  config.rpath mkinstalldirs ChangeLog INSTALL TODO README configure gitcompile depcomp
 AUTOMAKE_OPTIONS=foreign
 ACLOCAL_AMFLAGS = -I m4
