--- default.rules.org	2008-03-15 18:41:28.000000000 +0200
+++ default.rules	2008-03-15 18:42:06.000000000 +0200
@@ -183,11 +183,3 @@
 	$(INSTALL) $(install_includes) $(DESTDIR)$(includedir)
 endif
 	@true
-ifneq ($(install_manuals),)
-define inst_man
-@mkdir -p $(DESTDIR)$(mandir)/man$(suffix $(1))
-	cp -f $(1) $(DESTDIR)$(mandir)/man$(suffix $(1))
-	
-endef
-	$(foreach man,$(install_manuals),$(call inst_man,$(man)))
-endif
