Index: linux-2.6.18/drivers/usb/storage/usb.c
===================================================================
--- linux-2.6.18.orig/drivers/usb/storage/usb.c
+++ linux-2.6.18/drivers/usb/storage/usb.c
@@ -104,7 +104,7 @@ MODULE_AUTHOR("Matthew Dharm <mdharm-usb
 MODULE_DESCRIPTION("USB Mass Storage driver for Linux");
 MODULE_LICENSE("GPL");
 
-static unsigned int delay_use = 5;
+static unsigned int delay_use = 0;
 module_param(delay_use, uint, S_IRUGO | S_IWUSR);
 MODULE_PARM_DESC(delay_use, "seconds to delay before using a new device");
 
