Index: data/servicemanager.desktop
===================================================================
--- data/servicemanager.desktop	(revision 27545)
+++ data/servicemanager.desktop	(revision 27546)
@@ -5,18 +5,21 @@
 Name[es]=Servicios
 Name[de]=Dienste
 Name[nl]=Dienstbeheer
+Name[fr]=Gestionnaire de Service
 GenericName=Service Manager
 GenericName[tr]=Servis Yöneticisi
 GenericName[sv]=Tjänstehanteraren
 GenericName[es]=Administrador de Servicios
 GenericName[de]=Dienste-Manager
 GenericName[nl]=Dienstbeheer
+GenericName[fr]=Gestionnaire de Service
 Comment=An application for managing system services.
 Comment[tr]=Sistem servislerini yönetmenizi sağlayan bir uygulama.
 Comment[sv]=Ett program för att hantera systemtjänster.
 Comment[es]=Administración de servicios del sistema
 Comment[de]=Verwaltung der Systemdienste
 Comment[nl]=Programma voor beheer van systeemdiensten.
+Comment[fr]=Configuration du service
 Exec=service-manager
 Icon=flag-yellow
 Type=Application
Index: data/kcm_servicemanager.desktop
===================================================================
--- data/kcm_servicemanager.desktop	(revision 27545)
+++ data/kcm_servicemanager.desktop	(revision 27546)
@@ -18,10 +18,12 @@
 Name[es]=Servicios
 Name[de]=Dienste
 Name[nl]=Dienstbeheer
+Name[fr]=Gestionnaire de Service
 Comment=Service Manager
 Comment[sv]=Tjänstehanteraren
 Comment[es]=Administrador de Servicios
 Comment[de]=Dienste-Manager
 Comment[tr]=Servis Yöneticisi
 Comment[nl]=Beheerder voor diensten
+Comment[fr]=Configuration du service
 X-KDE-Keywords=python service managers
