Index: kdecore/kernel/kstandarddirs.cpp
===================================================================
--- kdecore/kernel/kstandarddirs.cpp	(revision 757828)
+++ kdecore/kernel/kstandarddirs.cpp	(working copy)
@@ -1470,7 +1470,7 @@
 #ifdef Q_WS_MACX
         localKdeDir =  QDir::homePath() + "/Library/Preferences/KDE/";
 #else
-        localKdeDir =  QDir::homePath() + "/.kde/";
+        localKdeDir =  QDir::homePath() + "/.kde4/";
 #endif
     }
 
Index: kinit/lnusertemp.c
===================================================================
--- kinit/lnusertemp.c	(revision 758243)
+++ kinit/lnusertemp.c	(working copy)
@@ -123,7 +123,7 @@
 
   if (!kde_home || !kde_home[0])
   {
-     kde_home = "~/.kde/";
+     kde_home = "~/.kde4/";
   }
 
   if (kde_home[0] == '~')
@@ -270,17 +270,17 @@
 
   if (strcmp(argv[1], "tmp") == 0)
   {
-    tmp_prefix = (char *)malloc(strlen(tmp)+strlen("/kde-")+1);
+    tmp_prefix = (char *)malloc(strlen(tmp)+strlen("/kde4-")+1);
     strcpy(tmp_prefix, tmp);
-    strcat(tmp_prefix, "/kde-"); 
+    strcat(tmp_prefix, "/kde4-"); 
     
     kde_prefix = "/tmp-"; 
   }
   else if (strcmp(argv[1], "socket") == 0)
   {
-    tmp_prefix = (char *)malloc(strlen(tmp)+strlen("/ksocket-")+1);
+    tmp_prefix = (char *)malloc(strlen(tmp)+strlen("/ksocket4-")+1);
     strcpy(tmp_prefix, tmp );
-    strcat(tmp_prefix, "/ksocket-" );
+    strcat(tmp_prefix, "/ksocket4-" );
 
     kde_prefix = "/socket-"; 
   }
@@ -290,9 +290,9 @@
     if (!tmp || !tmp[0]) 
       tmp = "/var/tmp";
 
-    tmp_prefix = (char *)malloc(strlen(tmp)+strlen("/kdecache-")+1);
+    tmp_prefix = (char *)malloc(strlen(tmp)+strlen("/kde4cache-")+1);
     strcpy(tmp_prefix, tmp );
-    strcat(tmp_prefix, "/kdecache-" );
+    strcat(tmp_prefix, "/kde4cache-" );
 
     kde_prefix = "/cache-"; 
   }
Index: kinit/wrapper.c
===================================================================
--- kinit/wrapper.c	(revision 758243)
+++ kinit/wrapper.c	(working copy)
@@ -142,7 +142,7 @@
 
   if (!kde_home || !kde_home[0])
   {
-     kde_home = "~/.kde/";
+     kde_home = "~/.kde4/";
   }
 
   if (kde_home[0] == '~')
