diff -Naur 55242-semantik-0.5.1c-orig/semantik-0.5.1c/wscript 55242-semantik-0.5.1c/semantik-0.5.1c/wscript
--- semantik-0.5.1c/wscript	2007-08-13 00:14:34.000000000 +0300
+++ semantik-0.5.1c/wscript	2007-08-13 02:03:39.000000000 +0300
@@ -157,9 +157,9 @@
 	opt.add_option('--exe', action='store_true', default=False, help='execute after the compilation (developers)')
 
 def shutdown():
-	if Params.g_commands['install']:
-		try: os.popen('/sbin/ldconfig')
-		except: pass
+	#if Params.g_commands['install']:
+		#try: os.popen('/sbin/ldconfig')
+		#except: pass
 	if Params.g_options.exe:
 		#os.popen('export LD_LIBRARY_PATH=build/default/:$LD_LIBRARY_PATH; PATH=plugins:$PATH build/default/src/semantik')
 		os.popen('LD_LIBRARY_PATH=build/default/:$LD_LIBRARY_PATH build/default/src/semantik')
