--- config/config.in.orig	2008-01-25 10:16:52.000000000 +0200
+++ config/config.in	2008-01-25 10:20:49.000000000 +0200
@@ -56,12 +56,12 @@
 ##
 ## OPTIMISE=4 will turn off DEBUG
 
-OPTIMISE=3
+OPTIMISE=2
 WARN=1
 # VERBOSE=1
 #DEBUG=1
 # PROFILE=gprof
-#SHARED=2
+SHARED=2
 
 ## Directory specific selections which override the above
 
@@ -94,7 +94,7 @@
 ## It may not work under all systems, so may be optionally omitted.
 INCLUDE_MODULES += EDITLINE
 
-TERMCAPLIB = @TERMCAPLIB@
+TERMCAPLIB = -lncurses
 # speech recognition
 #INCLUDE_MODULES += ASR
 
@@ -165,8 +165,8 @@
 CONFIG_WRAPPER_LANGUAGES = PYTHON PERL5
 
 # Language specific includes should be set to correct site paths
-CONFIG_PYTHON_INCLUDES= -I/usr/include/python2.2/
-CONFIG_PERL_INCLUDES= -I/usr/lib/perl5/5.8.3/i386-linux-thread-multi/CORE/
+CONFIG_PYTHON_INCLUDES= -I/usr/include/python2.4/
+CONFIG_PERL_INCLUDES= -I/usr/lib/perl5/5.8.8/i686-linux/
 #OR OLD#CONFIG_PERL_INCLUDES= -I/usr/lib/perl5/5.6.1/i386-linux/CORE/
 ##
 ###################################################################
