--- config/compilers/gcc_defaults.mak.orig	2006-08-05 18:37:59.000000000 +0300
+++ config/compilers/gcc_defaults.mak	2008-05-18 11:44:33.000000000 +0300
@@ -76,9 +76,9 @@
 PROFILE_gprof_CXXFLAGS  = -pg
 PROFILE_gprof_LINKFLAGS = -pg
 
-SHARED_CCFLAGS  = -fPIC -fno-shared-data
-SHARED_CXXFLAGS  = -fPIC -fno-shared-data
-SHARED_LINKFLAGS = -fno-shared-data
+SHARED_CCFLAGS  = -fPIC
+SHARED_CXXFLAGS  = -fPIC
+SHARED_LINKFLAGS = 
 
 ifndef GCC_MAKE_SHARED_LIB
     MAKE_SHARED_LIB = $(CXX) -shared -fno-shared-data -o XXX
--- base_class/EST_TSimpleMatrix.cc.orig	2008-05-18 11:49:38.000000000 +0300
+++ base_class/EST_TSimpleMatrix.cc	2008-05-18 11:50:03.000000000 +0300
@@ -44,6 +44,7 @@
 #include "EST_TVector.h"
 #include <fstream>
 #include <iostream>
+#include <cstring>
 #include "EST_cutils.h"
 
 template<class T> 
--- base_class/EST_TSimpleVector.cc.orig	2008-05-18 11:52:56.000000000 +0300
+++ base_class/EST_TSimpleVector.cc	2008-05-18 11:53:51.000000000 +0300
@@ -43,6 +43,7 @@
 #include "EST_TSimpleVector.h"
 #include "EST_matrix_support.h"
 #include <fstream>
+#include <cstring>
 #include "EST_cutils.h"
 
 template<class T> void EST_TSimpleVector<T>::copy(const EST_TSimpleVector<T> &a)
