Index: speech_tools/base_class/EST_Pathname_unix.cc
================================================================================
--- speech_tools/base_class/EST_Pathname_unix.cc
+++ speech_tools/base_class/EST_Pathname_unix.cc
@@ -148,9 +148,9 @@
 EST_Pathname operator + (const char *p, const EST_Pathname addition) 
 {return EST_Pathname::append(p, addition); }
 
-EST_Pathname &operator += (EST_Pathname p, const EST_Pathname addition)
+EST_Pathname &operator += (EST_Pathname& p, const EST_Pathname addition)
 { p = EST_Pathname::append(p, addition); return p; }
-EST_Pathname &operator += (EST_Pathname p, const EST_String addition)
+EST_Pathname &operator += (EST_Pathname& p, const EST_String addition)
 { p = EST_Pathname::append(p, addition); return p; }
 
 EST_Pathname EST_Pathname::append(EST_Pathname directory, EST_Pathname addition)
