Index: kdeutils/kcharselect/Makefile.am
===================================================================
--- kdeutils.orig/kcharselect/Makefile.am
+++ kdeutils/kcharselect/Makefile.am
@@ -1,9 +1,9 @@
 INCLUDES = $(all_includes)
-LDADD	= $(LIB_KFILE)
+LDADD	= /usr/kde/3.5/lib/libkio.so $(LIB_KFILE)
 bin_PROGRAMS = kcharselect
 
 kcharselect_SOURCES = kcharselectdia.cc main.cc
-kcharselect_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+kcharselect_LDFLAGS = /usr/kde/3.5/lib/libkio.so $(all_libraries) $(KDE_RPATH)
 kcharselect_METASOURCES = kcharselectdia.moc
 
 updatedir = $(kde_datadir)/kconf_update
