diff -Naur openmolar-0.1.6.orig/bin/openmolar.desktop openmolar-0.1.6/bin/openmolar.desktop
--- openmolar-0.1.6.orig/bin/openmolar.desktop	2009-10-27 19:35:33.503903929 +0200
+++ openmolar-0.1.6/bin/openmolar.desktop	2009-10-27 19:40:09.211459606 +0200
@@ -2,8 +2,10 @@
 [Desktop Entry]
 Version=1.0
 Name=OpenMolar
-GenericName=OpenMolar
+GenericName=Dental Practice Management Suite
+GenericName[tr]=Diş Hekimliği Uygulamaları
 Comment=Dental Practice Management Software
+Comment[tr]=Diş Hekimliği Uygulamaları Takip Yazılımı
 Exec=openmolar
 Icon=openmolar
 Terminal=false
