--- modules/pam_console/50-default.perms	2006-11-10 19:22:05.725138683 +0200
+++ modules/pam_console/50-default.perms	2006-11-10 19:23:27.317691228 +0200
@@ -1,10 +1,10 @@
 # device classes -- these are shell-style globs
 <floppy>=/dev/fd[0-1]* \
 	 /dev/floppy* /mnt/floppy*
-<sound>=/dev/dsp* /dev/audio* /dev/midi* \
-	/dev/mixer* /dev/sequencer* \
-	/dev/sound/* /dev/beep \
-	/dev/snd/* /dev/adsp*
+#<sound>=/dev/dsp* /dev/audio* /dev/midi* \
+#	/dev/mixer* /dev/sequencer* \
+#	/dev/sound/* /dev/beep \
+#	/dev/snd/* /dev/adsp*
 <cdrom>=/dev/cdrom* /dev/cdroms/* /dev/cdwriter* /mnt/cdrom*
 <pilot>=/dev/pilot
 <jaz>=/mnt/jaz*
@@ -34,7 +34,7 @@
 
 # permission definitions
 <console>  0660 <floppy>     0660 root.floppy
-<console>  0600 <sound>      0600 root
+#<console>  0600 <sound>      0600 root
 <console>  0600 <cdrom>      0660 root.disk
 <console>  0600 <pilot>      0660 root.uucp
 <console>  0600 <jaz>        0660 root.disk
