--- trousers-0.3.1.orig/src/tspi/tspi_asn1.c	2007-07-17 22:50:43.000000000 +0300
+++ trousers-0.3.1/src/tspi/tspi_asn1.c	2008-04-17 12:43:10.000000000 +0300
@@ -23,9 +23,19 @@
 #include "trousers/tss.h"
 #include "trousers/trousers.h"
 #include "trousers_types.h"
+#include "tsplog.h"
 
 #define TSS_OPENSSL_ASN1_ERROR	(0xffffffff)
 
+#if (OPENSSL_VERSION_NUMBER >= 0x0090800FL)
+#define OPENSSL_COMPAT_CONST const
+#else
+#define OPENSSL_COMPAT_CONST
+#endif
+
+#define OPENSSL_COMPAT_ASN1_SEQUENCE(tname) \
+	static const ASN1_TEMPLATE tname##_seq_tt[] 
+
 typedef struct tdTSS_BLOB {
 	ASN1_INTEGER *		structVersion;
 	ASN1_INTEGER *		blobType;
@@ -33,7 +43,7 @@
 	ASN1_OCTET_STRING *	blob;
 } TSS_BLOB;
 
-ASN1_SEQUENCE(TSS_BLOB) = {
+OPENSSL_COMPAT_ASN1_SEQUENCE(TSS_BLOB) = {
 	ASN1_SIMPLE(TSS_BLOB, structVersion, ASN1_INTEGER),
 	ASN1_SIMPLE(TSS_BLOB, blobType, ASN1_INTEGER),
 	ASN1_SIMPLE(TSS_BLOB, blobLength, ASN1_INTEGER),
@@ -184,7 +194,8 @@
 			BYTE *rawBlob)			/* out */
 {
 	TSS_BLOB *tssBlob = NULL;
-	const BYTE *encBlob = berBlob;
+	OPENSSL_COMPAT_CONST BYTE *encBlob = berBlob;
+
 	UINT32 encBlobLen = berBlobSize;
 	UINT32 decStructVersion, decBlobType, decBlobSize;
 
