--- setup.py	2006-02-28 20:06:43.000000000 +0200
+++ setup.py.fix	2006-03-09 22:33:04.000000000 +0200
@@ -28,16 +28,16 @@
 # it.  It makes very nice antialiased output and also supports alpha
 # blending
 BUILD_AGG          = 1
-BUILD_GTKAGG       = 'auto'
-BUILD_GTK          = 'auto'
+BUILD_GTKAGG       = 0
+BUILD_GTK          = 0
 
 # build TK GUI with Agg renderer ; requires Tkinter Python extension
 # and Tk includes
 # Use False or 0 if you don't want to build
-BUILD_TKAGG        = 'auto'
+BUILD_TKAGG        = 0
 
 # build wxPython GUI with Agg renderer ; requires wxPython package
-BUILD_WXAGG        = 'auto'
+BUILD_WXAGG        = 0
 
 
 # build a small extension to manage the focus on win32 platforms.
