diff -ur kaffeine-0.8.6.orig/kaffeine/src/kaffeine.desktop kaffeine-0.8.6/kaffeine/src/kaffeine.desktop
--- kaffeine-0.8.6.orig/kaffeine/src/kaffeine.desktop	2008-01-21 14:27:01.000000000 +0200
+++ kaffeine-0.8.6/kaffeine/src/kaffeine.desktop	2008-01-21 14:27:45.000000000 +0200
@@ -42,7 +42,7 @@
 GenericName[sv]=Mediaspelare
 GenericName[tg]=Навозишгари Медиа
 GenericName[th]=โปรแกรมเล่นสื่อ
-GenericName[tr]=Oynatıcı
+GenericName[tr]=Çokluortam Oynatıcısı
 GenericName[uk]=Програвач медіа
 GenericName[xx]=xxMedia Playerxx
 GenericName[zh_CN]=媒体播放器
