diff -ur cvs-shadow/src/usermod.c shadow/src/usermod.c
--- cvs-shadow/src/usermod.c	2006-11-14 16:40:54.000000000 +0100
+++ shadow/src/usermod.c	2007-01-13 14:59:02.000000000 +0100
@@ -905,7 +905,7 @@
 		 */
 		int c;
 		static struct option long_options[] = {
-			{"append", required_argument, NULL, 'a'},
+			{"append", no_argument, NULL, 'a'},
 			{"comment", required_argument, NULL, 'c'},
 			{"home", required_argument, NULL, 'd'},
 			{"expiredate", required_argument, NULL, 'e'},
