Index: libs/plasma/containment.cpp
===================================================================
--- libs/plasma/containment.cpp	(revision 760763)
+++ libs/plasma/containment.cpp	(working copy)
@@ -182,11 +182,13 @@
             Plasma::Widget *addWidgetTool = addToolBoxTool("addwidgets", "list-add", i18n("Add Widgets"));
             connect(addWidgetTool, SIGNAL(clicked()), this, SIGNAL(showAddWidgets()));
 
+#if 0
             Plasma::Widget *zoomInTool = addToolBoxTool("zoomIn", "zoom-in", i18n("Zoom In"));
             connect(zoomInTool, SIGNAL(clicked()), this, SIGNAL(zoomIn()));
 
             Plasma::Widget *zoomOutTool = addToolBoxTool("zoomOut", "zoom-out", i18n("Zoom Out"));
             connect(zoomOutTool, SIGNAL(clicked()), this, SIGNAL(zoomOut()));
+#endif
         }
     } else {
         delete d->toolbox;
