--- startkde.cmake	2008-01-13 17:55:39.000000000 +0200
+++ startkde.cmake	2008-01-13 17:56:57.000000000 +0200
@@ -30,6 +30,9 @@
 MALLOC_CHECK_=2 
 export MALLOC_CHECK_
 
+# Make GTK+ apps use our colors if set
+export GTK2_RC_FILES=~/.gtkrc-2.0:~/.gtkrc-2.0-kde:~/.kde4/share/config/gtkrc-2.0
+
 # in case we have been started with full pathname spec without being in PATH
 bindir=`echo "$0" | sed -n 's,^\(/.*\)/[^/][^/]*$,\1,p'`
 if [ -n "$bindir" ]; then
