Index: plasma/applets/lock_logout/lockout.cpp
===================================================================
--- plasma/applets/lock_logout/lockout.cpp	(revision 763694)
+++ plasma/applets/lock_logout/lockout.cpp	(working copy)
@@ -46,13 +46,15 @@
     layout->setSpacing(0);
 
     Plasma::Icon *icon_lock = new Plasma::Icon(KIcon("system-lock-screen"), "", this);
-    layout->addItem(icon_lock);
     connect(icon_lock, SIGNAL(clicked()), this, SLOT(clickLock()));
 
     Plasma::Icon *icon_logout = new Plasma::Icon(KIcon("system-log-out"), "", this);
     layout->addItem(icon_logout);
     connect(icon_logout, SIGNAL(clicked()), this, SLOT(clickLogout()));
 
+    // TODO make order configurable
+    layout->addItem(icon_lock);
+
     //It seems the layout geometry must be calculated by hand for the first time
     layout->setGeometry(QRectF(0,0, contentSize().width(), contentSize().height()));
 }
