--- kdm/config.def	2008-01-05 01:55:45.000000000 +0200
+++ kdm/config.def	2008-01-11 21:54:11.000000000 +0200
@@ -2318,7 +2318,7 @@
  PreferAdmin/FACE_PREFER_ADMIN: prefer &lt;<option>FaceDir</option>&gt;, fallback on $<envar>HOME</envar>
  PreferUser/FACE_PREFER_USER: ... and the other way round
  UserOnly/FACE_USER_ONLY: from the user's <filename>$<envar>HOME</envar>/.face[.icon]</filename>
-Default: AdminOnly
+Default: PreferUser
 User: greeter
 Instance: #*/PreferUser
 Comment:
@@ -2646,7 +2646,7 @@
 
 Key: UseTheme
 Type: bool
-Default: false
+Default: true
 User: greeter
 Instance: #*/true
 Comment: &
@@ -2655,7 +2655,7 @@
 
 Key: Theme
 Type: string
-Default: ""
+Default: "circles"
 User: greeter
 Instance: */KDMDATA "/themes/circles"
 Comment: &
