--- kobodeluxe-0.4pre10.orig/kobo.cpp
+++ kobodeluxe-0.4pre10/kobo.cpp
@@ -323,21 +323,21 @@
 	 * Graphics data
 	 */
 	/* Current dir; from within the build tree */
-	fmap.addpath("GFX", "./data");
+	//fmap.addpath("GFX", "./data");
 	/* Real data dir */
 	fmap.addpath("GFX", "DATA>>gfx");
 	/* Current dir */
-	fmap.addpath("GFX", "./gfx");
+	//fmap.addpath("GFX", "./gfx");
 
 	/*
 	 * Sound data
 	 */
 	/* Current dir; from within the build tree */
-	fmap.addpath("SFX", "./data");
+	//fmap.addpath("SFX", "./data");
 	/* Real data dir */
 	fmap.addpath("SFX", "DATA>>sfx");
 	/* Current dir */
-	fmap.addpath("SFX", "./sfx");
+	//fmap.addpath("SFX", "./sfx");
 
 	/*
 	 * Score files (user and global)
@@ -355,7 +355,7 @@
 	/* System local */
 	fmap.addpath("CONFIG", SYSCONF_DIR);
 	/* In current dir (last resort) */
-	fmap.addpath("CONFIG", "./");
+	//fmap.addpath("CONFIG", "./");
 }
 
 
