diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur PolicyKit-kde-4.2.0-orig/manager/AuthDialog.cpp PolicyKit-kde-4.2.0/manager/AuthDialog.cpp
--- PolicyKit-kde-4.2.0-orig/manager/AuthDialog.cpp	2009-02-12 23:47:36.000000000 +0200
+++ PolicyKit-kde-4.2.0/manager/AuthDialog.cpp	2009-02-12 23:50:34.000000000 +0200
@@ -136,8 +136,6 @@
     }
 
     if (requiresAdmin) {
-        // Check to see if we have the application name
-        if (m_appname.isEmpty()) {
             // Check to see if the authentication is provided through group of admin users
             if (adminUsers.count()) {
                 lblContent->setText(i18n("An application is attempting to perform an action that requires privileges."
@@ -147,24 +145,10 @@
                 lblContent->setText(i18n("An application is attempting to perform an action that requires privileges."
                                          " Authentication as the super user is required to perform this action."));
             }
-        } else {
-            // Check to see if the authentication is provided through group of admin users
-            if (adminUsers.count()) {
-                lblContent->setText(i18n("The application %1 is attempting to perform an action that requires privileges."
-                                         " Authentication as one of the users below is required to perform this action.", m_appname));
-                createUserCB(adminUsers);
-            } else {
-                lblContent->setText(i18n("The application %1 is attempting to perform an action that requires privileges."
-                                         " Authentication as the super user is required to perform this action.", m_appname));
-            }
-        }
-    } else {
-        if (m_appname.isEmpty())
+    }
+    else {
             lblContent->setText(i18n("An application is attempting to perform an action that requires privileges."
                                      " Authentication is required to perform this action."));
-        else
-            lblContent->setText(i18n("The application %1 is attempting to perform an action that requires privileges."
-                                     " Authentication is required to perform this action.", m_appname));
     }
 }
 
