--- Makefile.in~	2006-06-15 12:31:30.000000000 +0300
+++ Makefile.in	2006-09-14 23:19:34.000000000 +0300
@@ -153,9 +153,9 @@
 			rm -Rf $(dstdir)/$(PACKAGE)/$$p; \
 		done
 
-	ln -sf $(dstdir)/$(PACKAGE)/amsn $(slnkdir)/amsn
-	ln -sf $(dstdir)/$(PACKAGE)/amsn-remote $(slnkdir)/amsn-remote
-	ln -sf $(dstdir)/$(PACKAGE)/amsn-remote-CLI $(slnkdir)/amsn-remote-CLI
+	ln -sf /usr/share/amsn/amsn $(slnkdir)/amsn
+	ln -sf /usr/share/amsn/amsn-remote $(slnkdir)/amsn-remote
+	ln -sf /usr/share/amsn/amsn-remote-CLI $(slnkdir)/amsn-remote-CLI
 
 	chmod -R +rx $(dstdir)/$(PACKAGE)
 	mkdir -p $(dstdir)/icons/hicolor
