diff -Nur k3b-orig/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp k3b/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp
--- k3b-orig/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp	2008-07-18 14:27:32.000000000 +0300
+++ k3b/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp	2008-07-18 14:28:21.000000000 +0300
@@ -21,8 +21,8 @@
 #include <kdebug.h>
 
 extern "C" {
-#include <ffmpeg/avcodec.h>
-//#include <libavcodec/avcodec.h>
+//#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 }
 
 #include <math.h>
diff -Nur k3b-orig/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp k3b/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
--- k3b-orig/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2008-07-18 14:27:32.000000000 +0300
+++ k3b/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2008-07-18 14:28:05.000000000 +0300
@@ -18,10 +18,10 @@
 #include "k3bffmpegwrapper.h"
 
 extern "C" {
-#include <ffmpeg/avcodec.h>
-#include <ffmpeg/avformat.h>
-// #include <libavcodec/avcodec.h>
-// #include <libavformat/avformat.h>
+//#include <ffmpeg/avcodec.h>
+//#include <ffmpeg/avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 }
 
 #include <string.h>
