diff -Naur xastir-1.9.8/acinclude.m4 xastir-1.9.8-mod/acinclude.m4
--- xastir-1.9.8/acinclude.m4	2010-01-31 04:12:23.000000000 +0200
+++ xastir-1.9.8-mod/acinclude.m4	2010-04-14 08:35:39.000000000 +0300
@@ -291,7 +291,7 @@
 # Remove the -lgdal from what gdal-config --libs returns, because AC_CHECK_LIB
 # will put it into LIBS for us.
 #
-   LDFLAGS="$LDFLAGS `${GDAL_BIN} --libs | sed -e 's/-lgdal[^ ]*//'`"
+   LDFLAGS="$LDFLAGS `${GDAL_BIN} --libs | sed -e 's/-lgdal[^ ]*//'` -lhdf5_hl -lhdf5"
    AC_CHECK_HEADERS(gdal.h, [AC_CHECK_LIB(gdal, GDALAllRegister,
                     [use_gdal="yes"
                      LIBS="$LIBS -lgdal"
