Index: bouml_4.22.2/Makefile
===================================================================
--- bouml_4.22.2.orig/Makefile
+++ bouml_4.22.2/Makefile
@@ -1,24 +1,26 @@
+PREFIX = /var/pisi/bouml-4.22.2-1/install
+
 # The directory where all the bouml files will be installed, you
 # can change it for instance because you don't have root permission
-BOUML_LIB = /usr/lib/bouml
+BOUML_LIB = $(PREFIX)/usr/lib/bouml
 
 # The directory where the bouml shell script will be installed, you
 # can change it for instance because you don't have root permission
-BOUML_DIR = /usr/bin
+BOUML_DIR = $(PREFIX)/usr/bin
 
 # The directory containing bouml.desktop
 # The copy is not done if you comment the definition
-BOUML_DESKTOP_DIR = /usr/share/applications
+BOUML_DESKTOP_DIR = $(PREFIX)/usr/share/applications
 
 # The directories containing the bouml icons are
 # <BOUML_ICONS_PREFIX_DIR>/<size>x<size>/apps
 # The copy is not done if you comment the definition
-BOUML_ICONS_PREFIX_DIR = /usr/share/icons/hicolor
+BOUML_ICONS_PREFIX_DIR = $(PREFIX)/usr/share/icons/hicolor
 
 # On several Unix systems, the icons used inside the
 # menus must be stored inside the pixmaps' directory.
 # The copy is done if you uncomment the definition.
-#BOUML_UNIX_PIXMAPS_DIR = /usr/share/pixmaps
+#BOUML_UNIX_PIXMAPS_DIR = $(PREFIX)/usr/share/pixmaps
 
 # for packaging purpose, useless by default
 # DESTDIR = 
