diff -Naur 2ManDVD.orig/mainfrm.cpp 2ManDVD/mainfrm.cpp
--- 2ManDVD.orig/mainfrm.cpp	2009-08-10 19:27:08.689839770 +0300
+++ 2ManDVD/mainfrm.cpp	2009-08-10 19:34:00.081840487 +0300
@@ -250,18 +250,18 @@
 	splash.show();
 	splash.showMessage("Loading interface",spal,Qt::white);
 	//
-	qDebug() << "LOCALE : " << QLocale::system().name().left(2) << " " << qApp->applicationDirPath();
+	qDebug() << "LOCALE : " << QLocale::system().name().left(2) << " " << QString("/usr/share/2mandvd/locale");
 	if ( QLocale::system().name().left(2) != "fr" )
 	{
 		QTranslator *qtTranslator = new QTranslator(this);
 
-		if (qtTranslator->load("2mandvd_" + QLocale::system().name().left(2), qApp->applicationDirPath()) == true )
+		if (qtTranslator->load("2mandvd_" + QLocale::system().name().left(2), QString("/usr/share/2mandvd/locale")) == true )
 		{
 			qApp->installTranslator(qtTranslator);
 		}
 		else
 		{
-			if (qtTranslator->load("2mandvd_en", qApp->applicationDirPath()) == true )
+			if (qtTranslator->load("2mandvd_en", QString("/usr/share/2mandvd/locale")) == true )
 			{
 				qApp->installTranslator(qtTranslator);
 			}
