--- libk3b/core/k3bglobals.cpp	2006-08-23 10:32:16.000000000 +0300
+++ libk3b/core/k3bglobals.cpp	2006-11-24 14:35:11.201839539 +0200
@@ -345,12 +345,7 @@
 
 QString K3b::externalBinDeviceParameter( K3bDevice::Device* dev, const K3bExternalBin* bin )
 {
-  if( dev->interfaceType() == K3bDevice::SCSI )
-    return dev->busTargetLun();
-  else if( (plainAtapiSupport() && bin->hasFeature("plain-atapi") ) )
     return dev->blockDeviceName();
-  else
-    return QString("ATAPI:%1").arg(dev->blockDeviceName());
 }
 
 
