Index: keepnote-0.7.5/desktop/keepnote.desktop
===================================================================
--- keepnote-0.7.5.orig/desktop/keepnote.desktop
+++ keepnote-0.7.5/desktop/keepnote.desktop
@@ -4,11 +4,13 @@ Type=Application
 Name=KeepNote
 GenericName=Note-taking Application
 GenericName[sk]=Program na správu poznámok
+GenericName[tr]=Not Alma Uygulaması
 Comment=Note-taking Application
 Comment[sk]=Program na správu poznámok
+Comment[tr]=Not Alma Uygulaması
 Icon=keepnote
 Exec=keepnote %f
-Categories=Office
+Categories=Office;
 #MimeType=
 Terminal=false
 StartupNotify=false
