--- a/src/container.cpp
+++ b/src/container.cpp
@@ -518,9 +518,9 @@
              this,                            SLOT( gotoProfile() ) );
     connect( ui.actionSettings,             SIGNAL( triggered() ),
              this,                            SLOT( showSettingsDialog() ) );
+  #ifndef Q_WS_X11
     connect( ui.actionGetPlugin,            SIGNAL( triggered() ),
              this,                            SLOT( getPlugin() ) );
-  #ifndef Q_WS_X11
     connect( ui.actionCheckForUpdates,      SIGNAL( triggered() ),
              this,                            SLOT( checkForUpdates() ) );
   #endif
--- a/src/controlinterface.cpp
+++ b/src/controlinterface.cpp
@@ -82,6 +82,7 @@
                 m_parent->restoreWindow();
             }
 
+#ifndef Q_WS_X11
             if ( requests[i].contains( "container://checkScrobblerCache" ) )
             {
                 const QString user = requests[i].split( "checkScrobblerCache/" ).at( 1 );
@@ -97,6 +98,7 @@
                 LOGL( 3, "Calling container for media device addition" );
                 m_parent->addMediaDevice( requests[i].split( "addMediaDevice/" ).at( 1 ) );
             }
+#endif
 
         }
     }
--- a/src/mediadevices/mediadevice.cpp
+++ b/src/mediadevices/mediadevice.cpp
@@ -20,9 +20,11 @@
  ***************************************************************************/
 
 #include <QtGui>
+#ifndef LINUX
 #include <QSqlQuery>
 #include <QSqlResult>
 #include <QSqlError>
+#endif
 
 #include "mediadevice.h"
 #include "containerutils.h"
@@ -61,7 +63,9 @@
 
 MediaDevices::~MediaDevices()
 {
+#ifndef LINUX
     m_db.close();
+#endif
 
     delete m_http;
 }
--- a/src/mediadevices/mediadevice.h
+++ b/src/mediadevices/mediadevice.h
@@ -23,7 +23,9 @@
 #define MEDIADEVICE_H
 
 #include "interfaces/MediaDeviceInterface.h"
+#ifndef LINUX
 #include <QSqlDatabase>
+#endif
 #include <QFile>
 #include "http.h"
 
@@ -58,7 +60,9 @@
         void onUploadProgress( int done, int total );
 
     private:
+#ifndef LINUX
         QSqlDatabase m_db;
+#endif
         MediaDeviceInterface* m_gpod;
         QString m_savePath;
 
--- a/src/settingsdialog.cpp
+++ b/src/settingsdialog.cpp
@@ -68,6 +68,9 @@
 
     QWidget* scrobWidget = new QWidget();
     ui_scrobbling.setupUi( scrobWidget );
+#ifdef Q_WS_X11
+    ui_scrobbling.groupBox->hide();
+#endif
     ui.pageStack->addWidget( scrobWidget );
 
     QWidget* connWidget = new QWidget();
@@ -77,7 +80,9 @@
     QWidget* mediadeviceWidget = new QWidget();
     ui_mediadevices.setupUi( mediadeviceWidget );
     ui_mediadevices.deviceWidget->header()->setResizeMode( QHeaderView::ResizeToContents );
+#ifndef Q_WS_X11
     ui.pageStack->addWidget( mediadeviceWidget );
+#endif
 
     // Add icons to user icon dropdown
     QPixmap pixmap( dataPath( "icons/user_red.png" ) );
@@ -125,9 +130,13 @@
     //pixmap.scaled( 48, 48, Qt::IgnoreAspectRatio, Qt::SmoothTransformation );
     ui.pageList->item( 3 )->setIcon( pixmap );
 
+#ifndef Q_WS_X11
     pixmap.load( dataPath( "/icons/options_mediadevices.png" ) );
     //pixmap.scaled( 48, 48, Qt::IgnoreAspectRatio, Qt::SmoothTransformation );
     ui.pageList->item( 4 )->setIcon( pixmap );
+#else
+    delete ui.pageList->takeItem( 4 );
+#endif
 
     #ifdef HIDE_RADIO
         // Need to disable this as the language choice is hardcoded at startup
@@ -207,7 +216,9 @@
     populateRadio();
     populateScrobbling();
     populateConnection();
+#ifndef Q_WS_X11
     populateMediaDevices();
+#endif
     foreach (ExtensionInterface *i, extensions)
         i->populateSettings();
 
--- a/src/settingsdialog_scrobbling.ui
+++ b/src/settingsdialog_scrobbling.ui
@@ -192,6 +192,19 @@
      </layout>
     </widget>
    </item>
+   <item>
+    <spacer>
+     <property name="orientation" >
+      <enum>Qt::Vertical</enum>
+     </property>
+     <property name="sizeHint" >
+      <size>
+       <width>20</width>
+       <height>0</height>
+      </size>
+     </property>
+    </spacer>
+   </item>
   </layout>
  </widget>
  <customwidgets>
--- a/src/src.pro
+++ b/src/src.pro
@@ -206,6 +206,8 @@
 }
 
 linux* {
+    QT -= sql
+
     FORMS += wizarddialog_mac.ui \
              wizardshell_mac.ui \
              searchextension/search_win.ui
