--- src/export/ExportMP3.cpp.orig	2006-02-18 19:19:43.000000000 +0200
+++ src/export/ExportMP3.cpp	2006-02-18 19:18:40.000000000 +0200
@@ -96,7 +96,7 @@
 
 bool MP3Exporter::FindLibrary(wxWindow *parent)
 {
-   mLibPath = gPrefs->Read("/MP3/MP3LibPath", "");
+   mLibPath = gPrefs->Read("/MP3/MP3LibPath", "/usr/lib/libmp3lame.so");
 
    if (mLibPath=="" || !::wxFileExists(FILENAME(mLibPath))) {
    
