--- sdricoh_cs-0.1.3/sdricoh_cs.c.org	2008-05-05 15:13:41.000000000 +0300
+++ sdricoh_cs-0.1.3/sdricoh_cs.c	2008-05-05 16:00:40.000000000 +0300
@@ -328,10 +328,10 @@
 			cmd->error = 0;
 			mmc_request_done(mmc, mrq);
 			return;
-		case MMC_SEND_EXT_CSD:
-			cmd->error = -EINVAL;
-			mmc_request_done(mmc, mrq);
-			return;
+		/* case MMC_SEND_EXT_CSD: */
+		/*	cmd->error = -EINVAL; */
+		/*	mmc_request_done(mmc, mrq); */
+		/*	return; */
 		/* untested commands */
 		default:
 			INFO("untested command %i\n",cmd->opcode);
