Index: etc/policy/tr.org.pardus.comar.net.link.policy
===================================================================
--- etc/policy/tr.org.pardus.comar.net.link.policy	(revision 19344)
+++ etc/policy/tr.org.pardus.comar.net.link.policy	(revision 19345)
@@ -41,8 +41,8 @@
         <description>Set connection state</description>
         <description xml:lang="tr">Bağlantı durumunu değiştir</description>
         <description xml:lang="fr">Établir l'état de la connexion</description>
-        <description xml:lang="pl">Ustawianie stanu połączenia</description
-	<description xml:lang="de">Verbindungszustand einstellen</description>>
+        <description xml:lang="pl">Ustawianie stanu połączenia</description>
+	<description xml:lang="de">Verbindungszustand einstellen</description>
         <message>System policy prevents changing connection state</message>
         <message xml:lang="tr">Sistem politikası bağlantı durumunu değiştirmeyi engelliyor</message>
         <message xml:lang="fr">Le mode de gestion du système ne permet pas d'établir l'état de la connexion</message>
