--- common/config_file.c.orig	2008-05-09 16:05:06.000000000 +0300
+++ common/config_file.c	2008-05-09 16:03:36.000000000 +0300
@@ -1123,7 +1123,7 @@
 
 static const char * getSystemConfigFile()
 {
-    static const char * constFiles[]={ "/etc/qt4/"QTC_FILE, "/etc/qt3/"QTC_FILE, "/etc/qt/"QTC_FILE, NULL };
+    static const char * constFiles[]={ "/etc/qt4/"QTC_FILE, "/etc/qt3/"QTC_FILE, "/etc/qt/"QTC_FILE, "/usr/kde/3.5/share/apps/QtCurve/Pardus.qtcurve", NULL };
 
     int i;
 
