Index: samba-3.3.12/source/smbd/service.c
===================================================================
--- samba-3.3.12.orig/source/smbd/service.c
+++ samba-3.3.12/source/smbd/service.c
@@ -409,7 +409,7 @@ int find_service(fstring service)
 		if ((iPrinterService = lp_servicenumber(PRINTERS_NAME)) < 0) {
 			iPrinterService = load_registry_service(PRINTERS_NAME);
 		}
-		if (iPrinterService) {
+		if (iPrinterService >= 0) {
 			DEBUG(3,("checking whether %s is a valid printer name...\n", service));
 			if (pcap_printername_ok(service)) {
 				DEBUG(3,("%s is a valid printer name\n", service));
