--- configure.bak	2010-09-06 11:03:06.229834854 +0300
+++ configure	2010-09-06 11:04:12.042586012 +0300
@@ -3176,7 +3176,7 @@
 /* end confdefs.h.  */
 
 int
-main ()
+main (void)
 {
 
   ;
@@ -3332,7 +3332,7 @@
 /* end confdefs.h.  */
 
 int
-main ()
+main (void)
 {
 
   ;
@@ -3382,7 +3382,7 @@
 /* end confdefs.h.  */
 
 int
-main ()
+main (void)
 {
 #ifndef __GNUC__
        choke me
@@ -3444,7 +3444,7 @@
 /* end confdefs.h.  */
 
 int
-main ()
+main (void)
 {
 
   ;
@@ -3551,7 +3551,7 @@
 int argc;
 char **argv;
 int
-main ()
+main (void)
 {
 return f (e, argv, 0) != argv[0]  ||  f (e, argv, 1) != argv[1];
   ;
@@ -3662,7 +3662,7 @@
 $ac_declaration
 #include <stdlib.h>
 int
-main ()
+main (void)
 {
 exit (42);
   ;
@@ -3707,7 +3707,7 @@
 /* end confdefs.h.  */
 $ac_declaration
 int
-main ()
+main (void)
 {
 exit (42);
   ;
@@ -3889,7 +3889,7 @@
 /* end confdefs.h.  */
 
 int
-main ()
+main (void)
 {
 #ifndef __GNUC__
        choke me
@@ -3951,7 +3951,7 @@
 /* end confdefs.h.  */
 
 int
-main ()
+main (void)
 {
 
   ;
@@ -4023,7 +4023,7 @@
 $ac_declaration
 #include <stdlib.h>
 int
-main ()
+main (void)
 {
 exit (42);
   ;
@@ -4068,7 +4068,7 @@
 /* end confdefs.h.  */
 $ac_declaration
 int
-main ()
+main (void)
 {
 exit (42);
   ;
@@ -4489,7 +4489,7 @@
 /* end confdefs.h.  */
 #include "gmp.h"
 int
-main ()
+main (void)
 {
 
   #if __GNU_MP_VERSION < 4 || (__GNU_MP_VERSION == 4 && __GNU_MP_VERSION_MINOR < 1)
@@ -4547,7 +4547,7 @@
 #include <gmp.h>
     #include <mpfr.h>
 int
-main ()
+main (void)
 {
 
     #if MPFR_VERSION < MPFR_VERSION_NUM(2,2,1)
@@ -4597,7 +4597,7 @@
 #include <gmp.h>
     #include <mpfr.h>
 int
-main ()
+main (void)
 {
 
     #if MPFR_VERSION < MPFR_VERSION_NUM(2,3,0)
@@ -12021,7 +12021,7 @@
 #endif
 
 int
-main ()
+main (void)
 {
 
   ;
