--- a/configure	2010-09-06 09:25:46.806614806 +0300
+++ b/configure	2010-09-06 09:26:03.755232886 +0300
@@ -5402,7 +5402,7 @@
 # executable--we might be using a cross compiler--we only care whether it
 # can be created.  At this point the main configure script has set CC.
 we_are_ok=no
-echo "int main () { return 0; }" > conftest.c
+echo "int main (void) { return 0; }" > conftest.c
 ${CC} -o conftest ${CFLAGS} ${CPPFLAGS} conftest.c
 if test $? = 0 ; then
   if test -s conftest || test -s conftest.exe ; then
