diff -r d0f2345d8718 -r 7bb16cbb3179 stubdom/grub/mini-os.c
--- a/stubdom/grub/mini-os.c	Mon Apr 12 18:01:56 2010 +0100
+++ b/stubdom/grub/mini-os.c	Mon Apr 12 18:02:19 2010 +0100
@@ -651,12 +651,11 @@
     return 0;
 }
 
+static DECLARE_MUTEX_LOCKED(kbd_sem);
 static void kbd_thread(void *p)
 {
-    struct semaphore *sem = p;
-
     kbd_dev = init_kbdfront(NULL, 1);
-    up(sem);
+    up(&kbd_sem);
 }
 
 struct fbfront_dev *fb_open(void *fb, int width, int height, int depth)
@@ -665,10 +664,9 @@
     int linesize = width * (depth / 8);
     int memsize = linesize * height;
     int numpages = (memsize + PAGE_SIZE - 1) / PAGE_SIZE;
-    DECLARE_MUTEX_LOCKED(sem);
     int i;
 
-    create_thread("kbdfront", kbd_thread, &sem);
+    create_thread("kbdfront", kbd_thread, &kbd_sem);
 
     mfns = malloc(numpages * sizeof(*mfns));
     for (i = 0; i < numpages; i++) {
@@ -681,7 +679,7 @@
     if (!fb_dev)
         return NULL;
 
-    down(&sem);
+    down(&kbd_sem);
     if (!kbd_dev)
         return NULL;
 

