diff -Nur dvgrab-2.0-old/filehandler.cc dvgrab-2.0/filehandler.cc
--- dvgrab-2.0-old/filehandler.cc	2006-09-27 20:51:48.974059999 +0300
+++ dvgrab-2.0/filehandler.cc	2006-09-27 20:52:37.187730407 +0300
@@ -907,7 +907,7 @@
 	if ( fd )
 	{
 		struct stat file_status;
-		fstat( fileno( fd->stream ), &file_status );
+		fstat( lqt_fileno( fd ), &file_status );
 		return file_status.st_size;
 	}
 	else
