Index: contents/code/config.py
===================================================================
--- systemservices-0.1/contents/code/config.py	(revision 25846)
+++ systemservices-0.1/contents/code/config.py	(revision 25847)
@@ -16,7 +16,7 @@
         self.config = config
 
         # Get enabled services from config
-        self.enabledServices = str(self.config.readEntry("services", '')).split(',')
+        self.enabledServices = self.config.readEntry("services", QVariant('')).toString().split(',')
 
         # Show descriptions if showDesc is checked
         self.showDesc.setChecked(self.config.readEntry("showdesc", QVariant(True)).toBool())
