Index: wiimotedev-project-1.3.0/subprojects/manager/CMakeLists.txt
===================================================================
--- wiimotedev-project-1.3.0.orig/subprojects/manager/CMakeLists.txt
+++ wiimotedev-project-1.3.0/subprojects/manager/CMakeLists.txt
@@ -57,11 +57,11 @@ add_executable(qwiimotedev ${qwiimotedev
 add_dependencies(qwiimotedev qtsingleapplication)
 
 target_link_libraries(qwiimotedev
+  ${QTSINGLEAPPLICATION_LIBRARIES}
   ${QT_QTCORE_LIBRARY}
   ${QT_QTGUI_LIBRARY}
   ${QT_QTNETWORK_LIBRARY}
   ${QT_QTDBUS_LIBRARY}
-  ${QTSINGLEAPPLICATION_LIBRARIES}
 )
 
 install(TARGETS qwiimotedev DESTINATION ${WIIMOTEDEV_BIN_DIR})
