--- /dev/null	2007-09-19 22:37:42.720175399 +0300
+++ lua-5.0.3/etc/lua.pc	2007-09-20 00:35:00.000000000 +0300
@@ -0,0 +1,13 @@
+prefix=/usr
+exec_prefix=\${prefix}
+includedir=\${prefix}/include
+libdir=\${exec_prefix}/$(get_libdir)
+interpreter=\${exec_prefix}/bin/lua
+compiler=\${exec_prefix}/bin/luac
+version=
+
+Name: Lua
+Description: An extension programming language
+Version: ${version}
+Cflags: -I\${includedir}
+Libs: -L\${libdir} -llua -llualib -ldl -lm
