--- Makefile	2007-02-21 09:16:30.000000000 +0200
+++ Makefile	2007-06-01 03:42:58.000000000 +0300
@@ -70,7 +70,6 @@
 else
 	install $(INSTFLAGS) -m 755 bin/gcc/libgpac.$(DYN_LIB_SUFFIX) $(DESTDIR)$(prefix)/lib/libgpac-$(VERSION).$(DYN_LIB_SUFFIX)
 	ln -sf libgpac-$(VERSION).$(DYN_LIB_SUFFIX) $(DESTDIR)$(prefix)/lib/libgpac.$(DYN_LIB_SUFFIX)
-	ldconfig || true
 endif
 endif
 	install -d "$(DESTDIR)$(mandir)"
