Index: vnc_unixsrc/vncserver
===================================================================
--- vnc_unixsrc.orig/vncserver
+++ vnc_unixsrc/vncserver
@@ -48,10 +48,7 @@ ReadConfiguration();
 
 $defaultXStartup
     = ("#!/bin/sh\n\n".
-       "xrdb \$HOME/.Xresources\n".
-       "xsetroot -solid grey\n".
-       "xterm -geometry 80x24+10+10 -ls -title \"\$VNCDESKTOP Desktop\" &\n".
-       "twm &\n");
+       "/usr/lib/X11/xdm/Xsession");
 
 $xauthorityFile = "$ENV{XAUTHORITY}";
 
