--- vnc_unixsrc.orig/vncviewer/rfbproto.c
+++ vnc_unixsrc/vncviewer/rfbproto.c
@@ -240,8 +240,9 @@
     return False;
   }
 
-  if (server_major == 3 && server_minor >= 8) {
+  if (server_major == 3 && server_minor >= 8 && server_minor != 130) {
     /* the server supports protocol 3.8 or higher version */
+    /* but not the non-standard version 3.130 of XF4VNC */
     protocolMinorVersion = 8;
   } else if (server_major == 3 && server_minor == 7) {
     /* the server supports protocol 3.7 */
