--- vnc_unixsrc.orig/vncviewer/misc.c
+++ vnc_unixsrc/vncviewer/misc.c
@@ -24,6 +24,7 @@
 #include <vncviewer.h>
 #include <signal.h>
 #include <fcntl.h>
+#include <X11/Xproto.h>
 
 static void CleanupSignalHandler(int sig);
 static int CleanupXErrorHandler(Display *dpy, XErrorEvent *error);
@@ -292,6 +293,8 @@
 static int
 CleanupXErrorHandler(Display *dpy, XErrorEvent *error)
 {
+  if( error->request_code == X_SetInputFocus )
+      return 0;
   fprintf(stderr,"CleanupXErrorHandler called\n");
   Cleanup();
   return (*defaultXErrorHandler)(dpy, error);
