--- vnc_unixsrc.orig/vncpasswd/vncpasswd.c
+++ vnc_unixsrc/vncpasswd/vncpasswd.c
@@ -128,15 +128,15 @@
 
   /* Actually write the passwords. */
   if (!vncEncryptAndStorePasswd2(passwd1, passwd2_ptr, passwdFile)) {
-    memset(passwd1, 0, strlen(passwd1));
-    memset(passwd2, 0, strlen(passwd2));
+    memset(passwd1, 0, sizeof(passwd1));
+    memset(passwd2, 0, sizeof(passwd2));
     fprintf(stderr, "Cannot write password file %s\n", passwdFile);
     exit(1);
   }
 
   /* Zero the memory. */
-  memset(passwd1, 0, strlen(passwd1));
-  memset(passwd2, 0, strlen(passwd2));
+  memset(passwd1, 0, sizeof(passwd1));
+  memset(passwd2, 0, sizeof(passwd2));
   return 0;
 }
 
