Index: ui/item.ui
===================================================================
--- ui/item.ui	(revision 30788)
+++ ui/item.ui	(working copy)
@@ -20,7 +20,7 @@
       <string>Active or deactivate module.</string>
      </property>
      <property name="text">
-      <string notr="true"/>
+      <string/>
      </property>
     </widget>
    </item>
@@ -88,7 +88,7 @@
       <string>Edit</string>
      </property>
      <property name="text">
-      <string notr="true"/>
+      <string/>
      </property>
     </widget>
    </item>
@@ -98,7 +98,7 @@
       <string>Delete</string>
      </property>
      <property name="text">
-      <string notr="true"/>
+      <string/>
      </property>
     </widget>
    </item>
