--- CMakeLists.txt.orig	2010-12-12 13:25:25.938503775 +0100
+++ CMakeLists.txt	2010-12-12 13:27:53.657504969 +0100
@@ -52,7 +52,9 @@
 add_subdirectory(app_templates)
 add_subdirectory(documentation)
 add_subdirectory(providers)
+if (${KDE_VERSION} VERSION_GREATER 4.4.50)
 add_subdirectory(utils)
+endif (${KDE_VERSION} VERSION_GREATER 4.4.50)
 #macro_optional_add_subdirectory(doc)
 
 macro_display_feature_log()
