Index: qtmindmap-qtmindmap/other/qtmindmap.desktop
===================================================================
--- qtmindmap-qtmindmap.orig/other/qtmindmap.desktop
+++ qtmindmap-qtmindmap/other/qtmindmap.desktop
@@ -1,9 +1,13 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=QtMindMap
+GenericName=Mind-Mapping Software
+GenericName[tr]=Zihin Haritalama Yazılımı
+Comment=Mind-mapping software written in Qt4
+Comment[tr]=Qt4'de yazılmış bir zihin haritalama yazılımı
 TryExec=qtmindmap
 Exec=qtmindmap
-Icon=qtmindmap
+Icon=/usr/share/pixmaps/qtmindmap.svg
 Terminal=false
 Type=Application
 Categories=Application;Office;
