Index: applets/lancelot/app/src/launcher/LancelotApplet.cpp
===================================================================
--- applets/lancelot/app/src/launcher/LancelotApplet.cpp	(revision 1082192)
+++ applets/lancelot/app/src/launcher/LancelotApplet.cpp	(revision 1082193)
@@ -197,7 +197,7 @@
 {
     KConfigGroup kcg = config();
     d->showCategories =  (kcg.readEntry("show", "main") != "main");
-    d->mainIcon = kcg.readEntry("icon", "lancelot");
+    d->mainIcon = kcg.readEntry("icon", "kde");
     d->clickActivation = (kcg.readEntry("activate", "click") == "click");
     d->categsHide = kcg.readEntry("hiddenCategories", QStringList());
 }
