diff -Nur mplayer-20061209-old/libavcodec/dsputil.c mplayer-20061209/libavcodec/dsputil.c
--- mplayer-20061209-old/libavcodec/dsputil.c	2006-12-12 18:03:49.000000000 +0200
+++ mplayer-20061209/libavcodec/dsputil.c	2006-12-12 18:04:39.000000000 +0200
@@ -3807,7 +3807,7 @@
 
     if((int)&aligned & 15){
         if(!did_fail){
-#if defined(HAVE_MMX) || defined(HAVE_ALTIVEC)
+#if 0
             av_log(NULL, AV_LOG_ERROR,
                 "Compiler did not align stack variables, libavcodec has been misscompiled\n"
                 "and will possible be very slow or may crash, this is not a bug in the\n"
