diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kdebindings-4.1.0-orig/ruby/qtruby/bin/CMakeLists.txt kdebindings-4.1.0/ruby/qtruby/bin/CMakeLists.txt
--- kdebindings-4.1.0-orig/ruby/qtruby/bin/CMakeLists.txt	2008-08-20 16:52:43.000000000 +0300
+++ kdebindings-4.1.0/ruby/qtruby/bin/CMakeLists.txt	2008-08-20 16:52:21.000000000 +0300
@@ -1,8 +1 @@
 install( PROGRAMS rbqtapi DESTINATION ${CMAKE_INSTALL_PREFIX}/bin )
-
-install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink \"${CMAKE_INSTALL_PREFIX}/bin/rbqtapi\"  \"${CMAKE_INSTALL_PREFIX}/bin/rbqt3api\" )" )
-install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink \"${CMAKE_INSTALL_PREFIX}/bin/rbqtapi\"  \"${CMAKE_INSTALL_PREFIX}/bin/rbqt4api\" )" )
-install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink \"${CMAKE_INSTALL_PREFIX}/bin/rbqtapi\"  \"${CMAKE_INSTALL_PREFIX}/bin/rbkdeapi\" )" )
-install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink \"${CMAKE_INSTALL_PREFIX}/bin/rbqtapi\"  \"${CMAKE_INSTALL_PREFIX}/bin/rbkde4api\" )" )
-install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink \"${CMAKE_INSTALL_PREFIX}/bin/rbqtapi\"  \"${CMAKE_INSTALL_PREFIX}/bin/rbplasmaapi\" )" )
-
