diff -Naur apps/konsole/data/profiles/Shell.profile apps/konsole-orig/data/profiles/Shell.profile
--- apps/konsole/data/profiles/Shell.profile	2009-08-24 11:29:31.331015933 +0300
+++ apps/konsole-orig/data/profiles/Shell.profile	2009-08-24 11:29:07.491012618 +0300
@@ -138,7 +138,7 @@
 RemoteTabTitleFormat=%h : %u
 
 [Appearance]
-ColorScheme=DarkPastels
+ColorScheme=Linux
 
 [Scrolling]
 ScrollMode=FixedLineCount
