diff -up attica-0.1.2/lib/cmake/libattica.pc.cmake.pkgconfig attica-0.1.2/lib/cmake/libattica.pc.cmake
--- attica-0.1.2/lib/cmake/libattica.pc.cmake.pkgconfig	2010-01-14 07:00:16.000000000 -0600
+++ attica-0.1.2/lib/cmake/libattica.pc.cmake	2010-02-09 15:06:28.501736013 -0600
@@ -6,7 +6,7 @@ includedir=${CMAKE_INSTALL_PREFIX}/inclu
 Name: libattica
 Description: Qt library to access Open Collaboration Services
 #Requires:
-Version: ${ATTICA_VERSION_MAJOR}.${ATTICA_VERSION_MINOR}.${ATTICA_VERSION_PATCH}
+Version: ${CMAKE_LIBATTICA_VERSION_MAJOR}.${CMAKE_LIBATTICA_VERSION_MINOR}.${CMAKE_LIBATTICA_VERSION_PATCH}
 Libs: -L${LIB_DESTINATION} -lattica
 Cflags: -I${CMAKE_INSTALL_PREFIX}/include
 
