diff -ur mingetty-1.00.orig/mingetty.c mingetty-1.00/mingetty.c
--- mingetty-1.00.orig/mingetty.c	2006-05-10 10:58:00.000000000 +0300
+++ mingetty-1.00/mingetty.c	2006-05-10 11:00:23.000000000 +0300
@@ -71,7 +71,7 @@
 #include <sys/utsname.h>
 #include <time.h>
 
-#define LOGIN _(" login: ")	/* login prompt */
+//#define LOGIN _(" login: ")	/* login prompt */
 
 /* name of this program (argv[0]) */
 static char *progname;
@@ -360,7 +360,7 @@
 	}
 #endif
 	write (1, hn, strlen (hn));
-	write (1, LOGIN, strlen (LOGIN));
+	write (1, _(" login: "), strlen (_(" login: ")));
 }
 
 /* get_logname - get user name, establish speed, erase, kill, eol */
Yalnızca mingetty-1.00'da: mingetty.c~
