--- src/rtppacket.cpp~	2007-01-31 17:07:15.000000000 +0200
+++ src/rtppacket.cpp	2008-07-04 23:21:48.000000000 +0300
@@ -30,6 +30,8 @@
 
 */
 
+#include <cstring>
+
 #include "rtppacket.h"
 #include "rtpstructs.h"
 #include "rtpdefines.h"
--- src/rtcpcompoundpacketbuilder.cpp~	2007-01-31 17:07:15.000000000 +0200
+++ src/rtcpcompoundpacketbuilder.cpp	2008-07-04 23:25:19.000000000 +0300
@@ -30,6 +30,8 @@
 
 */
 
+#include <cstring>
+
 #include "rtcpcompoundpacketbuilder.h"
 #include "rtcpsrpacket.h"
 #include "rtcprrpacket.h"
