diff -Naur kstreamripper.orig/data/kstreamripper.desktop kstreamripper/data/kstreamripper.desktop
--- kstreamripper.orig/data/kstreamripper.desktop	2010-03-20 23:11:55.278923438 +0200
+++ kstreamripper/data/kstreamripper.desktop	2010-03-20 23:13:00.521923048 +0200
@@ -23,9 +23,13 @@
 Exec=kstreamripper
 #
 Name=KStreamRipper
-GenericName=Recorder for Internet radios
+GenericName=Recorder For Internet Radios
 Comment=Splits a stream in its tracks!
 #
+Name[tr]=KStreamRipper
+GenericName[tr]=İnternet Radyosu Kaydedici
+Comment[tr]=Bir yayın akışını istediğiniz şekilde kaydedin
+#
 Name[de]=KStreamRipper
 GenericName[de]=Aufnahmeprogramm für Internetradios
 Comment[de]=Schneidet Streams automatisch in einzelne Stücke.
