diff -Naur sdcc.orig/sdas/as6808/Makefile.in sdcc/sdas/as6808/Makefile.in
--- sdcc.orig/sdas/as6808/Makefile.in	2010-08-17 10:11:27.000000000 +0300
+++ sdcc/sdas/as6808/Makefile.in	2011-07-07 00:30:51.217000023 +0300
@@ -36,7 +36,8 @@
 
 CPPFLAGS        = @CPPFLAGS@ -DSDCDB -DNOICE -DINDEXLIB -I. -I$(srcdir) -I$(srcdir)/../asxxsrc -I$(srcdir)/../../support/Util
 CFLAGS          = @CFLAGS@ @WALL_FLAG@ $(CPPFLAGS)
-LDFLAGS         = @LDFLAGS@ -lm
+LDFLAGS         = @LDFLAGS@
+LIBS            = -lm
 
 OBJDIR = obj
 
@@ -65,7 +66,7 @@
 all: checkconf $(ASX)
 
 $(ASX): $(ASOBJECTS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ASOBJECTS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ASOBJECTS) $(LIBS)
 
 # Compiling and installing everything and running test
 # ----------------------------------------------------
diff -Naur sdcc.orig/sdas/as8051/Makefile.in sdcc/sdas/as8051/Makefile.in
--- sdcc.orig/sdas/as8051/Makefile.in	2010-08-17 10:11:27.000000000 +0300
+++ sdcc/sdas/as8051/Makefile.in	2011-07-07 00:31:49.528000023 +0300
@@ -36,7 +36,8 @@
 
 CPPFLAGS        = @CPPFLAGS@ -DSDCDB -DNOICE -DINDEXLIB -I. -I$(srcdir) -I$(srcdir)/../asxxsrc -I$(srcdir)/../../support/Util
 CFLAGS          = @CFLAGS@ @WALL_FLAG@ $(CPPFLAGS)
-LDFLAGS         = @LDFLAGS@ -lm
+LDFLAGS         = @LDFLAGS@
+LIBS            = -lm
 
 OBJDIR = obj
 
@@ -65,7 +66,7 @@
 all: checkconf $(ASX)
 
 $(ASX): $(ASOBJECTS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ASOBJECTS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ASOBJECTS) $(LIBS)
 
 # Compiling and installing everything and running test
 # ----------------------------------------------------
diff -Naur sdcc.orig/sdas/asgb/Makefile.in sdcc/sdas/asgb/Makefile.in
--- sdcc.orig/sdas/asgb/Makefile.in	2010-08-17 10:11:27.000000000 +0300
+++ sdcc/sdas/asgb/Makefile.in	2011-07-07 00:32:37.657000023 +0300
@@ -36,7 +36,8 @@
 
 CPPFLAGS        = @CPPFLAGS@ -DSDCDB -DNOICE -DINDEXLIB -I. -I$(srcdir) -I$(srcdir)/../asxxsrc -I$(srcdir)/../../support/Util
 CFLAGS          = @CFLAGS@ @WALL_FLAG@ $(CPPFLAGS)
-LDFLAGS         = @LDFLAGS@ -lm
+LDFLAGS         = @LDFLAGS@ 
+LIBS            = -lm
 
 OBJDIR = obj
 
@@ -65,7 +66,7 @@
 all: checkconf $(ASX)
 
 $(ASX): $(ASOBJECTS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ASOBJECTS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ASOBJECTS) $(LIBS)
 
 # Compiling and installing everything and running test
 # ----------------------------------------------------
diff -Naur sdcc.orig/sdas/asz80/Makefile.in sdcc/sdas/asz80/Makefile.in
--- sdcc.orig/sdas/asz80/Makefile.in	2010-08-17 10:11:27.000000000 +0300
+++ sdcc/sdas/asz80/Makefile.in	2011-07-07 00:33:35.989000023 +0300
@@ -36,7 +36,8 @@
 
 CPPFLAGS        = @CPPFLAGS@ -DSDCDB -DNOICE -DINDEXLIB -I. -I$(srcdir) -I$(srcdir)/../asxxsrc -I$(srcdir)/../../support/Util
 CFLAGS          = @CFLAGS@ @WALL_FLAG@ $(CPPFLAGS)
-LDFLAGS         = @LDFLAGS@ -lm
+LDFLAGS         = @LDFLAGS@
+LIBS            = -lm
 
 OBJDIR = obj
 
@@ -65,7 +66,7 @@
 all: checkconf $(ASX)
 
 $(ASX): $(ASOBJECTS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ASOBJECTS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ASOBJECTS) $(LIBS)
 
 # Compiling and installing everything and running test
 # ----------------------------------------------------
