diff -Naur TaskCoach-0.71.3.orig/build.in/fedora/taskcoach.desktop TaskCoach-0.71.3/build.in/fedora/taskcoach.desktop
--- TaskCoach-0.71.3.orig/build.in/fedora/taskcoach.desktop	2008-06-29 22:00:45.000000000 +0300
+++ TaskCoach-0.71.3/build.in/fedora/taskcoach.desktop	2008-11-09 23:01:05.000000000 +0200
@@ -1,10 +1,13 @@
 [Desktop Entry]
 Name=Task Coach
 GenericName=Task Coach
+GenericName[tr]=Görev Yöneticisi
 Comment=Your friendly task manager
+Comment[tr]=Kullanışlı bir görev yöneticisi
 Exec=taskcoach.py %f
 Icon=taskcoach
 Terminal=false
 Type=Application
 Categories=Office;Application;
+MimeType=application/x-taskcoach;

