diff -Nur clutter-gtk-old/clutter-gtk.pc.in clutter-gtk/clutter-gtk.pc.in
--- clutter-gtk-old/clutter-gtk.pc.in	2009-11-07 00:30:33.339245746 +0200
+++ clutter-gtk/clutter-gtk.pc.in	2009-11-07 00:34:54.678248503 +0200
@@ -9,6 +9,6 @@
 Name: clutter-gtk
 Description: GTK+ integration for Clutter
 Version: @VERSION@
 Libs: -L${libdir} -lclutter-gtk-${apiversion}
-Cflags: -I${includedir}/clutter-gtk-${apiversion}
+Cflags: -I${includedir}/clutter-${clutterapiversion}
 Requires: @CLUTTER_GTK_REQUIRES@
