--- cdrtools-2.01.01/cdrecord/cdrecord.c	2007-04-04 13:19:23.000000000 +0300
+++ cdrtools-2.01.01/cdrecord/cdrecord.c	2007-04-09 11:37:26.000000000 +0300
@@ -358,7 +358,7 @@
 #	define	CLONE_TITLE	""
 #endif
 	if ((flags & F_MSINFO) == 0 || lverbose || flags & F_VERSION) {
-		printf("Cdrecord%s%s %s (%s-%s-%s) Copyright (C) 1995-2007 Jörg Schilling\n",
+		printf("Cdrecord%s%s %s (%s-%s-%s) Copyright (C) 1995-2007 Joerg Schilling\n",
 								PRODVD_TITLE,
 								CLONE_TITLE,
 								cdr_version,
--- cdrtools-2.01.01/readcd/readcd.c	2007-02-26 17:46:48.000000000 +0200
+++ cdrtools-2.01.01/readcd/readcd.c	2007-04-09 11:38:23.000000000 +0300
@@ -306,7 +306,7 @@
 	if (help)
 		usage(0);
 	if (pversion) {
-		printf("readcd %s (%s-%s-%s) Copyright (C) 1987, 1995-2006 Jörg Schilling\n",
+		printf("readcd %s (%s-%s-%s) Copyright (C) 1987, 1995-2006 Joerg Schilling\n",
 								cdr_version,
 								HOST_CPU, HOST_VENDOR, HOST_OS);
 		exit(0);
