Index: aria2c-0.16.0/src/OptionHandlerFactory.cc
===================================================================
--- aria2c-0.16.0.orig/src/OptionHandlerFactory.cc
+++ aria2c-0.16.0/src/OptionHandlerFactory.cc
@@ -391,7 +391,7 @@ OptionHandlers OptionHandlerFactory::cre
 				   (PREF_SPLIT,
 				    TEXT_SPLIT,
 				    "5",
-				    1, 16));
+				    1, 500));
     op->addTag(TAG_BASIC);
     op->addTag(TAG_FTP);
     op->addTag(TAG_HTTP);
