diff -Naur audex.orig/audex.desktop audex/audex.desktop
--- audex.orig/audex.desktop	2009-11-21 02:45:39.050457460 +0200
+++ audex/audex.desktop	2009-11-21 02:46:32.996323762 +0200
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Name=Audex
 GenericName=CDDA Extraction
+GenericName[tr]=CDDA Kayıt Aracı
 GenericName[de]=Audio-CDs auslesen
 GenericName[it]=Estrazione e codifica CDDA
 Exec=audex
@@ -12,5 +13,6 @@
 X-DCOP-ServiceType=Unique
 X-DCOP-ServiceName=audex
 Comment=Rip audio data from your audio cd's
+Comment[tr]=Ses cd'lerini kaydedin
 Comment[de]=Rippt Audio-CDs
 Comment[it]=Estrae e codifica le canzoni dai tuoi CD audio
