--- src/gnokii_contact_format.c.orig	2007-03-27 13:49:51.000000000 +0200
+++ src/gnokii_contact_format.c	2007-09-07 00:57:32.000000000 +0200
@@ -141,8 +141,6 @@
 			case GN_PHONEBOOK_ENTRY_Date:
 			case GN_PHONEBOOK_ENTRY_LogoSwitch:	
 			case GN_PHONEBOOK_ENTRY_RingtoneAdv:	
-			// TODO support new CVS entries
-			/*	
 			case GN_PHONEBOOK_ENTRY_Location:
 			case GN_PHONEBOOK_ENTRY_Image:
 			case GN_PHONEBOOK_ENTRY_UserID:
@@ -161,7 +159,6 @@
 			case GN_PHONEBOOK_ENTRY_Company:
 			case GN_PHONEBOOK_ENTRY_Nickname:
 			case GN_PHONEBOOK_ENTRY_Birthday:
-			*/
 				break;
 		}
 
