Index: util-linux-2.20.1/libblkid/src/superblocks/ntfs.c
===================================================================
--- util-linux-2.20.1.orig/libblkid/src/superblocks/ntfs.c
+++ util-linux-2.20.1/libblkid/src/superblocks/ntfs.c
@@ -160,7 +160,7 @@ static int probe_ntfs(blkid_probe pr, co
 
 const struct blkid_idinfo ntfs_idinfo =
 {
-	.name		= "ntfs",
+	.name		= "ntfs-3g",
 	.usage		= BLKID_USAGE_FILESYSTEM,
 	.probefunc	= probe_ntfs,
 	.magics		=
