--- xgnokii/xgnokii.desktop.orig	2007-08-29 10:37:45.000000000 +0300
+++ xgnokii/xgnokii.desktop	2007-08-29 10:41:56.000000000 +0300
@@ -1,8 +1,11 @@
 [Desktop Entry]
 Version=0.9.4
-Name=Xgnokii Mobile Phone Tool
+Name=Xgnokii
+Name[tr]=Xgnokii
 GenericName=Mobile Phone Tool
+GenericName[tr]=Cep Telefonu Aracı
 Comment=Mobile phone tool suite
+Comment[tr]=Cep telefonu araç takımı
 Icon=xgnokii
 Exec=xgnokii
 Terminal=false
