Index: kfile/kfilepreviewgenerator.cpp
===================================================================
--- kfile/kfilepreviewgenerator.cpp	(revision 1105732)
+++ kfile/kfilepreviewgenerator.cpp	(working copy)
@@ -357,7 +357,7 @@
     QTimer* m_iconUpdateTimer;
     QTimer* m_scrollAreaTimer;
     QList<KJob*> m_previewJobs;
-    KDirModel* m_dirModel;
+    QPointer<KDirModel> m_dirModel;
     QAbstractProxyModel* m_proxyModel;
 
     /**
@@ -648,6 +648,8 @@
 
 void KFilePreviewGenerator::Private::updateCutItems()
 {
+    if (!m_dirModel) return; // see bug #196681
+
     DataChangeObtainer obt(this);
     clearCutItemsCache();
 
