--- src/ressources.cc.orig	2005-03-01 16:15:08.000000000 +0200
+++ src/ressources.cc	2007-10-07 17:25:27.000000000 +0300
@@ -31,11 +31,11 @@
 
 //...............................................................................
 
-char* ressources::fnamescore = "/var/lib/games/tecnoballz.hi";
+char* ressources::fnamescore = "/var/lib/games/tecnoballz/tecnoballz.hi";
 const char* ressources::folderlist[] =
 {	"/",		// Normally unused, except when running from the source directory...
 	DATADIR,
-	"./TecnoballZ/",
+	"/usr/share/tecnoballz/",
 	0		// Special value meaning "$(PREFIX)/share/games/tecnoballz/"
 			// Also marks end of list
 };
