diff -ur xorg-input-7.2.orig/xf86-input-keyboard-1.2.2/src/kbd.c xorg-input-7.2/xf86-input-keyboard-1.1.1/src/kbd.c
--- xorg-input-7.2.orig/xf86-input-keyboard-1.2.2/src/kbd.c	2007-04-19 03:08:02.826243000 +0300
+++ xorg-input-7.2/xf86-input-keyboard-1.2.2/src/kbd.c	2007-04-19 03:09:06.316557227 +0300
@@ -545,7 +545,6 @@
      * passing on parts of the VT switch sequence.
      */
     if (pInfo->fd >= 0) {
-	sleep(1);
 	xf86FlushInput(pInfo->fd);
 	AddEnabledDevice(pInfo->fd);
     }
Yalnızca xorg-input-7.2/xf86-input-keyboard-1.1.1/src'da: kbd.c.orig
