diff -ur kdebase-3.5.5.orig/kicker/kicker/ui/default-favs kdebase-3.5.5/kicker/kicker/ui/default-favs
--- kdebase-3.5.5.orig/kicker/kicker/ui/default-favs	2006-11-09 12:12:20.000000000 +0200
+++ kdebase-3.5.5/kicker/kicker/ui/default-favs	2006-11-09 12:15:17.000000000 +0200
@@ -1,9 +1,8 @@
 writer.desktop
 kde-konsole.desktop
 kde-Home.desktop
-kde-KControl.desktop
 MozillaFirefox.desktop
 kde-Kontact.desktop
 kde-amarok.desktop
-kde-digikam.desktop
+gimp.desktop
 kde-Help.desktop
