Index: kcontrol/kcontrol/main.cpp
===================================================================
--- kcontrol/kcontrol/main.cpp	(revision 600625)
+++ kcontrol/kcontrol/main.cpp	(working copy)
@@ -104,7 +104,7 @@
     KCONTROL_VERSION, I18N_NOOP("The KDE Control Center"), KAboutData::License_GPL,
     I18N_NOOP("(c) 1998-2004, The KDE Control Center Developers"));
 
-  KAboutData aboutKInfoCenter( "kcminfo", I18N_NOOP("KDE Info Center"),
+  KAboutData aboutKInfoCenter( "kinfocenter", I18N_NOOP("KDE Info Center"),
     KCONTROL_VERSION, I18N_NOOP("The KDE Info Center"), KAboutData::License_GPL,
     I18N_NOOP("(c) 1998-2004, The KDE Control Center Developers"));
 
