--- ./kicker/kicker/ui/service_mnu.cpp	2006/11/09 13:49:50	1.85
+++ ./kicker/kicker/ui/service_mnu.cpp	2006/11/09 13:58:16
@@ -426,7 +426,7 @@
             {
                 if (comment.find(serviceName, 0, true) == -1)
                 {
-                    serviceName = QString("%1 %2").arg(serviceName, comment);
+                    serviceName = QString("%1 (%2)").arg(serviceName, comment);
                 }
                 else
                 {
