diff -Naur teeworlds-0.4.2-src.orig/datasrc/data.ds teeworlds-0.4.2-src/datasrc/data.ds
--- teeworlds-0.4.2-src.orig/datasrc/data.ds	2008-04-05 16:13:02.000000000 +0300
+++ teeworlds-0.4.2-src/datasrc/data.ds	2008-04-19 13:05:31.000000000 +0300
@@ -1,228 +1,228 @@
 sounds {
 	gun_fire {
-		"data/audio/wp_gun_fire-01.wv"
-		"data/audio/wp_gun_fire-02.wv"
-		"data/audio/wp_gun_fire-03.wv"
+		"/usr/share/teeworlds/audio/wp_gun_fire-01.wv"
+		"/usr/share/teeworlds/audio/wp_gun_fire-02.wv"
+		"/usr/share/teeworlds/audio/wp_gun_fire-03.wv"
 	}
 
 	shotgun_fire {
-		"data/audio/wp_shotty_fire-01.wv"
-		"data/audio/wp_shotty_fire-02.wv"
-		"data/audio/wp_shotty_fire-03.wv"
+		"/usr/share/teeworlds/audio/wp_shotty_fire-01.wv"
+		"/usr/share/teeworlds/audio/wp_shotty_fire-02.wv"
+		"/usr/share/teeworlds/audio/wp_shotty_fire-03.wv"
 	}
 	
 	grenade_fire {
-		"data/audio/wp_flump_launch-01.wv"
-		"data/audio/wp_flump_launch-02.wv"
-		"data/audio/wp_flump_launch-03.wv"
+		"/usr/share/teeworlds/audio/wp_flump_launch-01.wv"
+		"/usr/share/teeworlds/audio/wp_flump_launch-02.wv"
+		"/usr/share/teeworlds/audio/wp_flump_launch-03.wv"
 	}
 	
 	hammer_fire {
-		"data/audio/wp_hammer_swing-01.wv"
-		"data/audio/wp_hammer_swing-02.wv"
-		"data/audio/wp_hammer_swing-03.wv"
+		"/usr/share/teeworlds/audio/wp_hammer_swing-01.wv"
+		"/usr/share/teeworlds/audio/wp_hammer_swing-02.wv"
+		"/usr/share/teeworlds/audio/wp_hammer_swing-03.wv"
 	}
 
 	hammer_hit {
-		"data/audio/wp_hammer_hit-01.wv"
-		"data/audio/wp_hammer_hit-02.wv"
-		"data/audio/wp_hammer_hit-03.wv"
+		"/usr/share/teeworlds/audio/wp_hammer_hit-01.wv"
+		"/usr/share/teeworlds/audio/wp_hammer_hit-02.wv"
+		"/usr/share/teeworlds/audio/wp_hammer_hit-03.wv"
 	}
 	
 	ninja_fire { 
-		"data/audio/wp_ninja_attack-01.wv"
-		"data/audio/wp_ninja_attack-02.wv"
-		"data/audio/wp_ninja_attack-03.wv"
+		"/usr/share/teeworlds/audio/wp_ninja_attack-01.wv"
+		"/usr/share/teeworlds/audio/wp_ninja_attack-02.wv"
+		"/usr/share/teeworlds/audio/wp_ninja_attack-03.wv"
 	}
 
 	grenade_explode {
-		"data/audio/wp_flump_explo-01.wv"
-		"data/audio/wp_flump_explo-02.wv"
-		"data/audio/wp_flump_explo-03.wv"
+		"/usr/share/teeworlds/audio/wp_flump_explo-01.wv"
+		"/usr/share/teeworlds/audio/wp_flump_explo-02.wv"
+		"/usr/share/teeworlds/audio/wp_flump_explo-03.wv"
 	}
 	
 	ninja_hit {
-		"data/audio/wp_ninja_hit-01.wv"
-		"data/audio/wp_ninja_hit-02.wv"
-		"data/audio/wp_ninja_hit-03.wv"
+		"/usr/share/teeworlds/audio/wp_ninja_hit-01.wv"
+		"/usr/share/teeworlds/audio/wp_ninja_hit-02.wv"
+		"/usr/share/teeworlds/audio/wp_ninja_hit-03.wv"
 	}
 	
 	rifle_fire {
-		"data/audio/wp_rifle_fire-01.wv"
-		"data/audio/wp_rifle_fire-02.wv"
-		"data/audio/wp_rifle_fire-03.wv"
+		"/usr/share/teeworlds/audio/wp_rifle_fire-01.wv"
+		"/usr/share/teeworlds/audio/wp_rifle_fire-02.wv"
+		"/usr/share/teeworlds/audio/wp_rifle_fire-03.wv"
 	}
 
 	rifle_bounce {
-		"data/audio/wp_rifle_bnce-01.wv"
-		"data/audio/wp_rifle_bnce-02.wv"
-		"data/audio/wp_rifle_bnce-03.wv"
+		"/usr/share/teeworlds/audio/wp_rifle_bnce-01.wv"
+		"/usr/share/teeworlds/audio/wp_rifle_bnce-02.wv"
+		"/usr/share/teeworlds/audio/wp_rifle_bnce-03.wv"
 	}
 
 	weapon_switch { 
-		"data/audio/wp_switch-01.wv"
-		"data/audio/wp_switch-02.wv"
-		"data/audio/wp_switch-03.wv"
+		"/usr/share/teeworlds/audio/wp_switch-01.wv"
+		"/usr/share/teeworlds/audio/wp_switch-02.wv"
+		"/usr/share/teeworlds/audio/wp_switch-03.wv"
 	}
 
 	player_pain_short {
-		"data/audio/vo_teefault_pain_short-01.wv"
-		"data/audio/vo_teefault_pain_short-02.wv"
-		"data/audio/vo_teefault_pain_short-03.wv"
-		"data/audio/vo_teefault_pain_short-04.wv"
-		"data/audio/vo_teefault_pain_short-05.wv"
-		"data/audio/vo_teefault_pain_short-06.wv"
-		"data/audio/vo_teefault_pain_short-07.wv"
-		"data/audio/vo_teefault_pain_short-08.wv"
-		"data/audio/vo_teefault_pain_short-09.wv"
-		"data/audio/vo_teefault_pain_short-10.wv"
-		"data/audio/vo_teefault_pain_short-11.wv"
-		"data/audio/vo_teefault_pain_short-12.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_pain_short-01.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_pain_short-02.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_pain_short-03.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_pain_short-04.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_pain_short-05.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_pain_short-06.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_pain_short-07.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_pain_short-08.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_pain_short-09.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_pain_short-10.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_pain_short-11.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_pain_short-12.wv"
 	}
 
 	player_pain_long {
-		"data/audio/vo_teefault_pain_long-01.wv"
-		"data/audio/vo_teefault_pain_long-02.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_pain_long-01.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_pain_long-02.wv"
 	}
 
 	body_land {
-		"data/audio/foley_land-01.wv"
-		"data/audio/foley_land-02.wv"
-		"data/audio/foley_land-03.wv"
-		"data/audio/foley_land-04.wv"
+		"/usr/share/teeworlds/audio/foley_land-01.wv"
+		"/usr/share/teeworlds/audio/foley_land-02.wv"
+		"/usr/share/teeworlds/audio/foley_land-03.wv"
+		"/usr/share/teeworlds/audio/foley_land-04.wv"
 	}
 	
 	player_airjump {
-		"data/audio/foley_dbljump-01.wv"
-		"data/audio/foley_dbljump-02.wv"
-		"data/audio/foley_dbljump-03.wv"
+		"/usr/share/teeworlds/audio/foley_dbljump-01.wv"
+		"/usr/share/teeworlds/audio/foley_dbljump-02.wv"
+		"/usr/share/teeworlds/audio/foley_dbljump-03.wv"
 	}
 
 	player_jump {
-		"data/audio/foley_foot_left-01.wv"
-		"data/audio/foley_foot_left-02.wv"
-		"data/audio/foley_foot_left-03.wv"
-		"data/audio/foley_foot_left-04.wv"
-		"data/audio/foley_foot_right-01.wv"
-		"data/audio/foley_foot_right-02.wv"
-		"data/audio/foley_foot_right-03.wv"
-		"data/audio/foley_foot_right-04.wv"
+		"/usr/share/teeworlds/audio/foley_foot_left-01.wv"
+		"/usr/share/teeworlds/audio/foley_foot_left-02.wv"
+		"/usr/share/teeworlds/audio/foley_foot_left-03.wv"
+		"/usr/share/teeworlds/audio/foley_foot_left-04.wv"
+		"/usr/share/teeworlds/audio/foley_foot_right-01.wv"
+		"/usr/share/teeworlds/audio/foley_foot_right-02.wv"
+		"/usr/share/teeworlds/audio/foley_foot_right-03.wv"
+		"/usr/share/teeworlds/audio/foley_foot_right-04.wv"
 	}
 
 	player_die {
-		"data/audio/foley_body_splat-02.wv"
-		"data/audio/foley_body_splat-03.wv"
-		"data/audio/foley_body_splat-04.wv"
+		"/usr/share/teeworlds/audio/foley_body_splat-02.wv"
+		"/usr/share/teeworlds/audio/foley_body_splat-03.wv"
+		"/usr/share/teeworlds/audio/foley_body_splat-04.wv"
 	}
 	
 	player_spawn {
-		"data/audio/vo_teefault_spawn-01.wv"
-		"data/audio/vo_teefault_spawn-02.wv"
-		"data/audio/vo_teefault_spawn-03.wv"
-		"data/audio/vo_teefault_spawn-04.wv"
-		"data/audio/vo_teefault_spawn-05.wv"
-		"data/audio/vo_teefault_spawn-06.wv"
-		"data/audio/vo_teefault_spawn-07.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_spawn-01.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_spawn-02.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_spawn-03.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_spawn-04.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_spawn-05.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_spawn-06.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_spawn-07.wv"
 	}
 	
 	player_skid {
-		"data/audio/sfx_skid-01.wv"
-		"data/audio/sfx_skid-02.wv"
-		"data/audio/sfx_skid-03.wv"
-		"data/audio/sfx_skid-04.wv"
+		"/usr/share/teeworlds/audio/sfx_skid-01.wv"
+		"/usr/share/teeworlds/audio/sfx_skid-02.wv"
+		"/usr/share/teeworlds/audio/sfx_skid-03.wv"
+		"/usr/share/teeworlds/audio/sfx_skid-04.wv"
 	}
 
 	tee_cry {
-		"data/audio/vo_teefault_cry-01.wv"
-		"data/audio/vo_teefault_cry-02.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_cry-01.wv"
+		"/usr/share/teeworlds/audio/vo_teefault_cry-02.wv"
 	}
 
 	hook_loop { 
-		"data/audio/hook_loop-01.wv"
-		"data/audio/hook_loop-02.wv"
+		"/usr/share/teeworlds/audio/hook_loop-01.wv"
+		"/usr/share/teeworlds/audio/hook_loop-02.wv"
 	}
 	
 	hook_attach_ground {
-		"data/audio/hook_attach-01.wv"
-		"data/audio/hook_attach-02.wv"
-		"data/audio/hook_attach-03.wv"
+		"/usr/share/teeworlds/audio/hook_attach-01.wv"
+		"/usr/share/teeworlds/audio/hook_attach-02.wv"
+		"/usr/share/teeworlds/audio/hook_attach-03.wv"
 	}
 	
 	hook_attach_player {
-		"data/audio/foley_body_impact-01.wv"
-		"data/audio/foley_body_impact-02.wv"
-		"data/audio/foley_body_impact-03.wv"
+		"/usr/share/teeworlds/audio/foley_body_impact-01.wv"
+		"/usr/share/teeworlds/audio/foley_body_impact-02.wv"
+		"/usr/share/teeworlds/audio/foley_body_impact-03.wv"
 	}
 		
 	pickup_health {
-		"data/audio/sfx_pickup_hrt-01.wv"
-		"data/audio/sfx_pickup_hrt-02.wv"
+		"/usr/share/teeworlds/audio/sfx_pickup_hrt-01.wv"
+		"/usr/share/teeworlds/audio/sfx_pickup_hrt-02.wv"
 	}
 	
 	pickup_armor {
-		"data/audio/sfx_pickup_arm-01.wv"
-		"data/audio/sfx_pickup_arm-02.wv"
-		"data/audio/sfx_pickup_arm-03.wv"
-		"data/audio/sfx_pickup_arm-04.wv"
+		"/usr/share/teeworlds/audio/sfx_pickup_arm-01.wv"
+		"/usr/share/teeworlds/audio/sfx_pickup_arm-02.wv"
+		"/usr/share/teeworlds/audio/sfx_pickup_arm-03.wv"
+		"/usr/share/teeworlds/audio/sfx_pickup_arm-04.wv"
 	}
 	
 	pickup_grenade {
-		"data/audio/sfx_pickup_launcher.wv"
+		"/usr/share/teeworlds/audio/sfx_pickup_launcher.wv"
 	}
 	
 	pickup_shotgun {
-		"data/audio/sfx_pickup_sg.wv"
+		"/usr/share/teeworlds/audio/sfx_pickup_sg.wv"
 	}
 	
 	pickup_ninja {
-		"data/audio/sfx_pickup_ninja.wv"
+		"/usr/share/teeworlds/audio/sfx_pickup_ninja.wv"
 	}
 		
 	weapon_spawn {
-		"data/audio/sfx_spawn_wpn-01.wv"
-		"data/audio/sfx_spawn_wpn-02.wv"
-		"data/audio/sfx_spawn_wpn-03.wv"
+		"/usr/share/teeworlds/audio/sfx_spawn_wpn-01.wv"
+		"/usr/share/teeworlds/audio/sfx_spawn_wpn-02.wv"
+		"/usr/share/teeworlds/audio/sfx_spawn_wpn-03.wv"
 	}
 
 	weapon_noammo {
-		"data/audio/wp_noammo-01.wv"
-		"data/audio/wp_noammo-02.wv"
-		"data/audio/wp_noammo-03.wv"
-		"data/audio/wp_noammo-04.wv"
-		"data/audio/wp_noammo-05.wv"
+		"/usr/share/teeworlds/audio/wp_noammo-01.wv"
+		"/usr/share/teeworlds/audio/wp_noammo-02.wv"
+		"/usr/share/teeworlds/audio/wp_noammo-03.wv"
+		"/usr/share/teeworlds/audio/wp_noammo-04.wv"
+		"/usr/share/teeworlds/audio/wp_noammo-05.wv"
 	}
 
 	hit {
-		"data/audio/sfx_hit_weak-01.wv"
-		"data/audio/sfx_hit_weak-02.wv"
+		"/usr/share/teeworlds/audio/sfx_hit_weak-01.wv"
+		"/usr/share/teeworlds/audio/sfx_hit_weak-02.wv"
 	}
 
 	chat_server {
-		"data/audio/sfx_msg-server.wv"
+		"/usr/share/teeworlds/audio/sfx_msg-server.wv"
 	}
 	
 	chat_client {
-		"data/audio/sfx_msg-client.wv"
+		"/usr/share/teeworlds/audio/sfx_msg-client.wv"
 	}
 
 	ctf_drop {
-		"data/audio/sfx_ctf_drop.wv"
+		"/usr/share/teeworlds/audio/sfx_ctf_drop.wv"
 	}
 
 	ctf_return {
-		"data/audio/sfx_ctf_rtn.wv"
+		"/usr/share/teeworlds/audio/sfx_ctf_rtn.wv"
 	}
 	
 	ctf_grab_pl {
-		"data/audio/sfx_ctf_grab_pl.wv"
+		"/usr/share/teeworlds/audio/sfx_ctf_grab_pl.wv"
 	}
 
 	ctf_grab_en {
-		"data/audio/sfx_ctf_grab_en.wv"
+		"/usr/share/teeworlds/audio/sfx_ctf_grab_en.wv"
 	}
 
 	ctf_capture {
-		"data/audio/sfx_ctf_cap_pl.wv"
+		"/usr/share/teeworlds/audio/sfx_ctf_cap_pl.wv"
 	}
 }
 
@@ -233,34 +233,34 @@
 	}
 
 	game {
-		filename "data/game.png"
+		filename "/usr/share/teeworlds/game.png"
 	}
 
 	particles {
-		filename "data/particles.png"
+		filename "/usr/share/teeworlds/particles.png"
 	}
 
 	cursor {
-		filename "data/gui_cursor.png"
+		filename "/usr/share/teeworlds/gui_cursor.png"
 	}
 
 	banner {
-		filename "data/gui_logo.png"
+		filename "/usr/share/teeworlds/gui_logo.png"
 	}
 
 	emoticons {
-		filename "data/emoticons.png"
+		filename "/usr/share/teeworlds/emoticons.png"
 	}
 	
 	browseicons {
-		filename "data/browse_icons.png"
+		filename "/usr/share/teeworlds/browse_icons.png"
 	}
 
 	console_bg {
-		filename "data/console.png"
+		filename "/usr/share/teeworlds/console.png"
 	}
 	console_bar {
-		filename "data/console_bar.png"
+		filename "/usr/share/teeworlds/console_bar.png"
 	}
 }
 
diff -Naur teeworlds-0.4.2-src.orig/src/engine/client/ec_client.c teeworlds-0.4.2-src/src/engine/client/ec_client.c
--- teeworlds-0.4.2-src.orig/src/engine/client/ec_client.c	2008-04-05 16:13:02.000000000 +0300
+++ teeworlds-0.4.2-src/src/engine/client/ec_client.c	2008-04-19 13:05:22.000000000 +0300
@@ -518,7 +518,7 @@
 
 static int client_load_data()
 {
-	debug_font = gfx_load_texture("data/debug_font.png", IMG_AUTO, TEXLOAD_NORESAMPLE);
+	debug_font = gfx_load_texture("/usr/share/teeworlds/debug_font.png", IMG_AUTO, TEXLOAD_NORESAMPLE);
 	return 1;
 }
 
@@ -641,7 +641,7 @@
 	client_set_state(CLIENTSTATE_LOADING);
 	
 	/* try the normal maps folder */
-	str_format(buf, sizeof(buf), "data/maps/%s.map", mapname);
+	str_format(buf, sizeof(buf), "/usr/share/teeworlds/maps/%s.map", mapname);
 	error = client_load_map(buf, wanted_crc);
 	if(!error)
 		return error;
diff -Naur teeworlds-0.4.2-src.orig/src/engine/e_map.c teeworlds-0.4.2-src/src/engine/e_map.c
--- teeworlds-0.4.2-src.orig/src/engine/e_map.c	2008-04-05 16:13:02.000000000 +0300
+++ teeworlds-0.4.2-src/src/engine/e_map.c	2008-04-19 13:05:22.000000000 +0300
@@ -53,7 +53,7 @@
 int map_load(const char *mapname)
 {
 	char buf[512];
-	str_format(buf, sizeof(buf), "data/maps/%s.map", mapname);
+	str_format(buf, sizeof(buf), "/usr/share/teeworlds/maps/%s.map", mapname);
 	map = datafile_load(buf);
 	return map != 0;
 }
diff -Naur teeworlds-0.4.2-src.orig/src/engine/external/portaudio/src/hostapi/asihpi/pa_linux_asihpi.c teeworlds-0.4.2-src/src/engine/external/portaudio/src/hostapi/asihpi/pa_linux_asihpi.c
--- teeworlds-0.4.2-src.orig/src/engine/external/portaudio/src/hostapi/asihpi/pa_linux_asihpi.c	2008-04-05 16:13:02.000000000 +0300
+++ teeworlds-0.4.2-src/src/engine/external/portaudio/src/hostapi/asihpi/pa_linux_asihpi.c	2008-04-19 13:05:21.000000000 +0300
@@ -241,7 +241,7 @@
 /** Enable background bus mastering (BBM) for buffer transfers, if available (see HPI docs) */
 #define PA_ASIHPI_USE_BBM_ 1
 /** Minimum number of frames in HPI buffer (for either data or available space).
- If buffer contains less data/space, it indicates xrun or completion. */
+ If buffer contains less /usr/share/teeworlds/space, it indicates xrun or completion. */
 #define PA_ASIHPI_MIN_FRAMES_ 1152
 /** Minimum polling interval in milliseconds, which determines minimum host buffer size */
 #define PA_ASIHPI_MIN_POLLING_INTERVAL_ 10
@@ -1176,7 +1176,7 @@
 /* ---------------------------- Stream Interface ---------------------------- */
 
 /** Obtain HPI stream information.
- This obtains info such as stream state and available data/space in buffers. It also
+ This obtains info such as stream state and available /usr/share/teeworlds/space in buffers. It also
  estimates whether an underflow or overflow occurred.
 
  @param streamComp Pointer to stream component (input or output) to query
diff -Naur teeworlds-0.4.2-src.orig/src/engine/server/es_server.c teeworlds-0.4.2-src/src/engine/server/es_server.c
--- teeworlds-0.4.2-src.orig/src/engine/server/es_server.c	2008-04-05 16:13:02.000000000 +0300
+++ teeworlds-0.4.2-src/src/engine/server/es_server.c	2008-04-19 13:05:21.000000000 +0300
@@ -871,7 +871,7 @@
 {
 	DATAFILE *df;
 	char buf[512];
-	str_format(buf, sizeof(buf), "data/maps/%s.map", mapname);
+	str_format(buf, sizeof(buf), "/usr/share/teeworlds/maps/%s.map", mapname);
 	df = datafile_load(buf);
 	if(!df)
 		return 0;
diff -Naur teeworlds-0.4.2-src.orig/src/game/client/gc_hooks.cpp teeworlds-0.4.2-src/src/game/client/gc_hooks.cpp
--- teeworlds-0.4.2-src.orig/src/game/client/gc_hooks.cpp	2008-04-05 16:13:02.000000000 +0300
+++ teeworlds-0.4.2-src/src/game/client/gc_hooks.cpp	2008-04-19 13:05:26.000000000 +0300
@@ -68,7 +68,7 @@
 	
 	
 	int before = gfx_memory_usage();
-	font_set_load(&default_font, "data/fonts/default_font%d.tfnt", "data/fonts/default_font%d.png", "data/fonts/default_font%d_b.png", 14, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 36);
+	font_set_load(&default_font, "/usr/share/teeworlds/fonts/default_font%d.tfnt", "/usr/share/teeworlds/fonts/default_font%d.png", "/usr/share/teeworlds/fonts/default_font%d_b.png", 14, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 36);
 	dbg_msg("font", "gfx memory used for font textures: %d", gfx_memory_usage()-before);
 	
 	gfx_text_set_default_font(&default_font);
diff -Naur teeworlds-0.4.2-src.orig/src/game/client/gc_map_image.cpp teeworlds-0.4.2-src/src/game/client/gc_map_image.cpp
--- teeworlds-0.4.2-src.orig/src/game/client/gc_map_image.cpp	2008-04-05 16:13:02.000000000 +0300
+++ teeworlds-0.4.2-src/src/game/client/gc_map_image.cpp	2008-04-19 13:05:26.000000000 +0300
@@ -110,7 +110,7 @@
 		{
 			char buf[256];
 			char *name = (char *)map_get_data(img->image_name);
-			str_format(buf, sizeof(buf), "data/mapres/%s.png", name);
+			str_format(buf, sizeof(buf), "/usr/share/teeworlds/mapres/%s.png", name);
 			map_textures[i] = gfx_load_texture(buf, IMG_AUTO, 0);
 		}
 		else
diff -Naur teeworlds-0.4.2-src.orig/src/game/client/gc_skin.cpp teeworlds-0.4.2-src/src/game/client/gc_skin.cpp
--- teeworlds-0.4.2-src.orig/src/game/client/gc_skin.cpp	2008-04-05 16:13:02.000000000 +0300
+++ teeworlds-0.4.2-src/src/game/client/gc_skin.cpp	2008-04-19 13:05:26.000000000 +0300
@@ -24,7 +24,7 @@
 		return;
 		
 	char buf[512];
-	str_format(buf, sizeof(buf), "data/skins/%s", name);
+	str_format(buf, sizeof(buf), "/usr/share/teeworlds/skins/%s", name);
 	IMAGE_INFO info;
 	if(!gfx_load_png(&info, buf))
 	{
@@ -118,7 +118,7 @@
 void skin_init()
 {
 	// load skins
-	fs_listdir("data/skins", skinscan, 0);
+	fs_listdir("/usr/share/teeworlds/skins", skinscan, 0);
 }
 
 int skin_num()
diff -Naur teeworlds-0.4.2-src.orig/src/game/editor/ed_editor.cpp teeworlds-0.4.2-src/src/game/editor/ed_editor.cpp
--- teeworlds-0.4.2-src.orig/src/game/editor/ed_editor.cpp	2008-04-05 16:13:02.000000000 +0300
+++ teeworlds-0.4.2-src/src/game/editor/ed_editor.cpp	2008-04-19 13:05:26.000000000 +0300
@@ -1537,7 +1537,7 @@
 	ui_hsplit_t(&view, 12.0f, &slot, &view);
 	if(do_editor_button(&replace_button, "Replace", 0, &slot, draw_editor_button_menuitem, 0, "Replaces the image with a new one"))
 	{
-		editor.invoke_file_dialog("Replace Image", "Replace", "data/mapres/", "", replace_image);
+		editor.invoke_file_dialog("Replace Image", "Replace", "/usr/share/teeworlds/mapres/", "", replace_image);
 		return 1;
 	}
 
@@ -1618,7 +1618,7 @@
 	ui_hsplit_t(&toolbox, 10.0f, &slot, &toolbox);
 	ui_hsplit_t(&toolbox, 12.0f, &slot, &toolbox);
 	if(do_editor_button(&new_image_button, "Add", 0, &slot, draw_editor_button, 0, "Load a new image to use in the map"))
-		editor.invoke_file_dialog("Add Image", "Add", "data/mapres/", "", add_image);
+		editor.invoke_file_dialog("Add Image", "Add", "/usr/share/teeworlds/mapres/", "", add_image);
 }
 
 
@@ -2158,7 +2158,7 @@
 	ui_hsplit_t(&view, 12.0f, &slot, &view);
 	if(do_editor_button(&open_button, "Open", 0, &slot, draw_editor_button_menuitem, 0, "Opens a map for editing"))
 	{
-		editor.invoke_file_dialog("Open Map", "Open", "data/maps/", "", callback_open_map);
+		editor.invoke_file_dialog("Open Map", "Open", "/usr/share/teeworlds/maps/", "", callback_open_map);
 		return 1;
 	}
 
@@ -2166,7 +2166,7 @@
 	ui_hsplit_t(&view, 12.0f, &slot, &view);
 	if(do_editor_button(&append_button, "Append", 0, &slot, draw_editor_button_menuitem, 0, "Opens a map and adds everything from that map to the current one"))
 	{
-		editor.invoke_file_dialog("Append Map", "Append", "data/maps/", "", callback_append_map);
+		editor.invoke_file_dialog("Append Map", "Append", "/usr/share/teeworlds/maps/", "", callback_append_map);
 		return 1;
 	}
 
@@ -2181,7 +2181,7 @@
 	ui_hsplit_t(&view, 12.0f, &slot, &view);
 	if(do_editor_button(&save_as_button, "Save As", 0, &slot, draw_editor_button_menuitem, 0, "Saves the current map under a new name"))
 	{
-		editor.invoke_file_dialog("Save Map", "Save", "data/maps/", "", callback_save_map);
+		editor.invoke_file_dialog("Save Map", "Save", "/usr/share/teeworlds/maps/", "", callback_save_map);
 		return 1;
 	}
 		
@@ -2390,10 +2390,10 @@
 
 extern "C" void editor_init()
 {
-	checker_texture = gfx_load_texture("data/editor/checker.png", IMG_AUTO, 0);
-	background_texture = gfx_load_texture("data/editor/background.png", IMG_AUTO, 0);
-	cursor_texture = gfx_load_texture("data/editor/cursor.png", IMG_AUTO, 0);
-	entities_texture = gfx_load_texture("data/editor/entities.png", IMG_AUTO, 0);
+	checker_texture = gfx_load_texture("/usr/share/teeworlds/editor/checker.png", IMG_AUTO, 0);
+	background_texture = gfx_load_texture("/usr/share/teeworlds/editor/background.png", IMG_AUTO, 0);
+	cursor_texture = gfx_load_texture("/usr/share/teeworlds/editor/cursor.png", IMG_AUTO, 0);
+	entities_texture = gfx_load_texture("/usr/share/teeworlds/editor/entities.png", IMG_AUTO, 0);
 	
 	tileset_picker.make_palette();
 	tileset_picker.readonly = true;
@@ -2466,13 +2466,13 @@
 		editor.gui_active = !editor.gui_active;
 
 	if(inp_key_down(KEY_F5))
-		editor.save("data/maps/debug_test2.map");
+		editor.save("/usr/share/teeworlds/maps/debug_test2.map");
 
 	if(inp_key_down(KEY_F6))
-		editor.load("data/maps/debug_test2.map");
+		editor.load("/usr/share/teeworlds/maps/debug_test2.map");
 	
 	if(inp_key_down(KEY_F8))
-		editor.load("data/maps/debug_test.map");
+		editor.load("/usr/share/teeworlds/maps/debug_test.map");
 	
 	editor.render();
 	
diff -Naur teeworlds-0.4.2-src.orig/src/game/editor/ed_io.cpp teeworlds-0.4.2-src/src/game/editor/ed_io.cpp
--- teeworlds-0.4.2-src.orig/src/game/editor/ed_io.cpp	2008-04-05 16:13:02.000000000 +0300
+++ teeworlds-0.4.2-src/src/game/editor/ed_io.cpp	2008-04-19 13:05:26.000000000 +0300
@@ -398,7 +398,7 @@
 				if(item->external)
 				{
 					char buf[256];
-					sprintf(buf, "data/mapres/%s.png", name);
+					sprintf(buf, "/usr/share/teeworlds/mapres/%s.png", name);
 					
 					// load external
 					IMAGE imginfo;
