diff -Nur teewars-0.3.3-src-old/default.bam teewars-0.3.3-src/default.bam
--- teewars-0.3.3-src-old/default.bam	2008-01-25 23:56:39.927498710 +0200
+++ teewars-0.3.3-src/default.bam	2008-01-25 23:57:11.426710103 +0200
@@ -190,9 +190,9 @@
    		if platform == "macosx" then
 			engine_settings.cc.flags = "-Wall"
 		else
-			engine_settings.cc.flags = "-Wall -pedantic-errors"
+			engine_settings.cc.flags = "pardusCFLAGS"
 		end
-		engine_settings.linker.flags = ""
+		engine_settings.linker.flags = "pardusLDFLAGS"
 	end
 
 	-- server
