diff -Nur teewars-0.3.3-src-old/src/engine/client/ec_snd.c teewars-0.3.3-src/src/engine/client/ec_snd.c
--- teewars-0.3.3-src-old/src/engine/client/ec_snd.c	2008-01-25 23:38:45.358857975 +0200
+++ teewars-0.3.3-src/src/engine/client/ec_snd.c	2008-01-25 23:40:53.522374158 +0200
@@ -372,13 +372,6 @@
 }
 
 
-static FILE *file = NULL;
-
-static int read_data(void *buffer, int size)
-{
-	return fread(buffer, 1, size, file);	
-}
-
 int snd_load_wv(const char *filename)
 {
 	SAMPLE *snd;
@@ -394,19 +387,12 @@
 	if(!sound_enabled)
 		return 1;
 
-	file = fopen(filename, "rb"); /* TODO: use system.h stuff for this */
-	if(!file)
-	{
-		dbg_msg("sound/wv", "failed to open %s", filename);
-		return -1;
-	}
-
 	sid = snd_alloc_id();
 	if(sid < 0)
 		return -1;
 	snd = &samples[sid];
 
-	context = WavpackOpenFileInput(read_data, error);
+	context = WavpackOpenFileInput(filename, error, OPEN_2CH_MAX, 0);
 	if (context)
 	{
 		int samples = WavpackGetNumSamples(context);
@@ -461,8 +447,6 @@
 		dbg_msg("sound/wv", "failed to open %s: %s", filename, error);
 	}
 
-	fclose(file);
-	file = NULL;
 
 	if(config.debug)
 		dbg_msg("sound/wv", "loaded %s", filename);
