diff -Naur teeworlds-0.4.2-src.orig/scripts/gen_keys.py teeworlds-0.4.2-src/scripts/gen_keys.py
--- teeworlds-0.4.2-src.orig/scripts/gen_keys.py	2008-04-05 16:13:02.000000000 +0300
+++ teeworlds-0.4.2-src/scripts/gen_keys.py	2008-06-30 20:11:35.000000000 +0300
@@ -1,7 +1,7 @@
 import sys, os
 
 # load header
-glfw = "src/engine/external/glfw/include/GL/glfw.h"
+glfw = "/usr/include/GL/glfw.h"
 lines = [line.strip() for line in file(glfw).readlines()]
 
 # genereate keys.h file
diff -Naur teeworlds-0.4.2-src.orig/src/engine/client/ec_gfx.c teeworlds-0.4.2-src/src/engine/client/ec_gfx.c
--- teeworlds-0.4.2-src.orig/src/engine/client/ec_gfx.c	2008-04-05 16:13:02.000000000 +0300
+++ teeworlds-0.4.2-src/src/engine/client/ec_gfx.c	2008-06-30 20:16:40.000000000 +0300
@@ -1,5 +1,5 @@
 /* copyright (c) 2007 magnus auvinen, see licence.txt for more info */
-#include <engine/external/glfw/include/GL/glfw.h>
+#include <GL/glfw.h>
 #include <engine/external/pnglite/pnglite.h>
 
 #include <engine/e_system.h>
diff -Naur teeworlds-0.4.2-src.orig/src/engine/client/ec_inp.c teeworlds-0.4.2-src/src/engine/client/ec_inp.c
--- teeworlds-0.4.2-src.orig/src/engine/client/ec_inp.c	2008-04-05 16:13:02.000000000 +0300
+++ teeworlds-0.4.2-src/src/engine/client/ec_inp.c	2008-06-30 20:12:18.000000000 +0300
@@ -1,6 +1,6 @@
 /* copyright (c) 2007 magnus auvinen, see licence.txt for more info */
 #include <string.h>
-#include <engine/external/glfw/include/GL/glfw.h>
+#include <GL/glfw.h>
 
 #include <engine/e_system.h>
 #include <engine/e_client_interface.h>
diff -Naur teeworlds-0.4.2-src.orig/src/engine/client/ec_snd.c teeworlds-0.4.2-src/src/engine/client/ec_snd.c
--- teeworlds-0.4.2-src.orig/src/engine/client/ec_snd.c	2008-04-05 16:13:02.000000000 +0300
+++ teeworlds-0.4.2-src/src/engine/client/ec_snd.c	2008-06-30 20:12:42.000000000 +0300
@@ -3,8 +3,8 @@
 #include <engine/e_client_interface.h>
 #include <engine/e_config.h>
 
-#include <engine/external/portaudio/include/portaudio.h>
-#include <engine/external/wavpack/wavpack.h>
+#include <portaudio.h>
+#include <wavpack/wavpack.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
diff -Naur teeworlds-0.4.2-src.orig/src/engine/e_datafile.c teeworlds-0.4.2-src/src/engine/e_datafile.c
--- teeworlds-0.4.2-src.orig/src/engine/e_datafile.c	2008-04-05 16:13:02.000000000 +0300
+++ teeworlds-0.4.2-src/src/engine/e_datafile.c	2008-06-30 20:13:09.000000000 +0300
@@ -1,7 +1,7 @@
 /* copyright (c) 2007 magnus auvinen, see licence.txt for more info */
 #include "e_system.h"
 #include "e_datafile.h"
-#include "external/zlib/zlib.h"
+#include <zlib.h>
 
 static const int DEBUG=0;
 
diff -Naur teeworlds-0.4.2-src.orig/src/engine/external/pnglite/pnglite.c teeworlds-0.4.2-src/src/engine/external/pnglite/pnglite.c
--- teeworlds-0.4.2-src.orig/src/engine/external/pnglite/pnglite.c	2008-04-05 16:13:02.000000000 +0300
+++ teeworlds-0.4.2-src/src/engine/external/pnglite/pnglite.c	2008-06-30 20:14:12.000000000 +0300
@@ -5,7 +5,7 @@
 #define USE_ZLIB 1
 
 #if USE_ZLIB
-#include "../zlib/zlib.h"
+#include <zlib.h>
 #else
 #include "zlite.h"
 #endif
