Index: kmail/accountmanager.cpp
===================================================================
--- kmail/accountmanager.cpp	(revision 857910)
+++ kmail/accountmanager.cpp	(revision 857911)
@@ -7,6 +7,7 @@
 #include "accountmanager.h"
 
 #include "kmaccount.h"
+#include "kmacctfolder.h"
 #include "kmacctmaildir.h"
 #include "kmacctlocal.h"
 #include "popaccount.h"
@@ -239,6 +240,11 @@
 {
   if ( account ) {
     mAcctList.append( account );
+    // init folder's account list
+    KMAcctFolder *folder = static_cast<KMAcctFolder*>( account->folder() );
+    if ( folder && !folder->hasAccounts() ) {
+      folder->addAccount( account );
+    }
     emit accountAdded( account );
     account->installTimer();
   }
Index: kmail/kmfoldertree.cpp
===================================================================
--- kmail/kmfoldertree.cpp	(revision 857910)
+++ kmail/kmfoldertree.cpp	(revision 857911)
@@ -432,6 +432,9 @@
   connect(kmkernel->acctMgr(), SIGNAL(accountRemoved(KMAccount*)),
           this, SLOT(slotAccountRemoved(KMAccount*)));
 
+  connect(kmkernel->acctMgr(), SIGNAL(accountAdded(KMAccount*)),
+          this, SLOT(slotUnhideLocalInbox()));
+
   connect(kmkernel->searchFolderMgr(), SIGNAL(folderRemoved(KMFolder*)),
           this, SLOT(slotFolderRemoved(KMFolder*)));
 
