--- Makefile.am.orig	2008-04-24 15:16:50.000000000 +0300
+++ Makefile.am	2008-05-12 22:03:05.000000000 +0300
@@ -728,7 +728,7 @@
 	-lz
 if BOOSTSYSTEM
   vegastrike_LDADD = $(vegastrike_LDADD_1) \
-    -lboost_python-st
+    -lboost_python
 else
   vegastrike_LDADD = $(vegastrike_LDADD_1)
 endif  
@@ -739,7 +739,7 @@
 	-lz
 if BOOSTSYSTEM
   vegaserver_LDADD = $(vegaserver_LDADD_1) \
-    -lboost_python-st
+    -lboost_python
 else
   vegaserver_LDADD = $(vegaserver_LDADD_1)
 endif  
