Index: vegastrike-0.5~svn12102/src/xml_serializer.h
===================================================================
--- vegastrike-0.5~svn12102.orig/src/xml_serializer.h	2004-04-26 04:24:52.000000000 +0200
+++ vegastrike-0.5~svn12102/src/xml_serializer.h	2008-04-05 00:55:33.000000000 +0200
@@ -12,7 +12,7 @@
 	unsigned int *ui;
     float *f;
     void * p;
-    char * c;
+    signed char * c;
    // short * s;
     bool * b;
     double * d;
@@ -30,7 +30,7 @@
   XMLType (float myhardfloat) {w.hardfloat = myhardfloat;}
   XMLType (float  *myfloat) {w.f =myfloat;}
   XMLType (void * myvoid) {w.p=myvoid;}
-  XMLType (char * mychar) {w.c=mychar;}
+  XMLType (signed char * mychar) {w.c=mychar;}
   //XMLType (short * mychar) {w.s=mychar;} // removing useless shorts - use integers instead
   //XMLType (unsigned short * mychar) {w.us=mychar;} // removing useless shorts - use integers instead
   XMLType (unsigned char * mychar) {w.uc=mychar;}
