diff -Naur vegastrike-0.5.0.orşg/src/gfx/vid_file.cpp vegastrike-0.5.0/src/gfx/vid_file.cpp
--- vegastrike-0.5.0.orşg/src/gfx/vid_file.cpp	2009-03-11 19:09:19.000000000 +0200
+++ vegastrike-0.5.0/src/gfx/vid_file.cpp	2009-03-11 19:11:55.000000000 +0200
@@ -21,9 +21,9 @@
     #else
         #define DEPRECATED_IMG_CONVERT 1
     #endif
-    #include <ffmpeg/avcodec.h>
-    #include <ffmpeg/avformat.h>
-    #include <ffmpeg/avio.h>
+    #include <libavcodec/avcodec.h>
+    #include <libavformat/avformat.h>
+    #include <libavformat/avio.h>
 }
 
 #endif
