Push further joystick limits for unusual joysticks.

Index: vegastrike-0.5~svn12102/src/in_joystick.h
===================================================================
--- vegastrike-0.5~svn12102.orig/src/in_joystick.h	2005-12-17 03:21:05.000000000 +0100
+++ vegastrike-0.5~svn12102/src/in_joystick.h	2008-04-05 15:55:21.000000000 +0200
@@ -34,7 +34,7 @@
 //#include "glob.h"
 //#include "dbg.h"
 #include "in_kb.h"
-#define NUMJBUTTONS 32
+#define NUMJBUTTONS 48
 
 class JoyStick;
 
@@ -47,7 +47,7 @@
 
 const int MAX_JOYSTICKS=16;
 const int MOUSE_JOYSTICK = MAX_JOYSTICKS-1;
-const int MAX_BUTTONS=48;
+const int MAX_BUTTONS=64;
 const int MAX_DIGITAL_HATSWITCHES=4;
 const int MAX_DIGITAL_VALUES=9;
 
