diff -Naur vegastrike-0.5.0.orig/Makefile.am vegastrike-0.5.0/Makefile.am
--- vegastrike-0.5.0.orig/Makefile.am	2009-03-10 23:08:56.000000000 +0200
+++ vegastrike-0.5.0/Makefile.am	2009-03-10 23:08:01.000000000 +0200
@@ -486,7 +486,7 @@
   mesher \
   mesh_xml \
   tempgen \
-  replace \
+  vsreplace \
   trisort \
   vegaserver \
   vsrextract \
@@ -1050,7 +1050,7 @@
   objconv/trisort.cpp \
   objconv/trisort.h
 
-replace_SOURCES = \
+vsreplace_SOURCES = \
   objconv/replace.cpp
 
 asteroidgen_SOURCES = \
