--- a/bin/fingerprint-gui/fingerprint-gui.desktop.orig	2011-01-30 04:00:49.183000073 +0200
+++ b/bin/fingerprint-gui/fingerprint-gui.desktop	2011-01-30 04:00:49.192000073 +0200
@@ -6,7 +6,7 @@
 Name=Fingerprint GUI
 GenericName=Fingerprint Tool
 Comment=Tool for fingerprint enrollment and verification
-Icon=gnome-user-interface.png
+Icon=user-identity.png
 TryExec=fingerprint-gui
 Exec=fingerprint-gui --debug
-Categories=GNOME;GTK;Settings;HardwareSettings;
+Categories=Qt;KDE;Utility;X-KDE-Utilities-Desktop;
