--- a/src/MainWindowImpl.cpp.orig	2011-01-30 03:17:39.142000074 +0200
+++ a/src/MainWindowImpl.cpp	2011-01-30 03:17:39.154000074 +0200
@@ -7,7 +7,7 @@
  * Copyright (C) 2008-2010  Wolfgang Ullrich
  */
 
-#include <qt4/QtCore/qlocale.h>
+#include <QtCore/qlocale.h>
 
 
 /*

--- a/src/PluginDialog.cpp.orig	2011-01-30 03:23:38.606000073 +0200
+++ a/src/PluginDialog.cpp	2011-01-30 03:23:38.615000073 +0200
@@ -25,7 +25,7 @@
 
 #include <sys/poll.h>
 #include <string>
-#include <qt4/QtCore/qthread.h>
+#include <QtCore/qthread.h>
 
 #include "../include/PluginDialog.h"
 
--- a/src/FingerprintPlugin.cpp.orig	2011-01-30 03:25:49.508000073 +0200
+++ b/src/FingerprintPlugin.cpp	2011-01-30 03:25:49.513000073 +0200
@@ -25,7 +25,7 @@
 
 #include <syslog.h>
 #include <string>
-#include <qt4/QtGui/qwidget.h>
+#include <QtGui/qwidget.h>
 
 #include "PluginDialog.h"
 
--- a/src/PolkitListener.cpp.orig	2011-01-30 03:27:48.368000072 +0200
+++ b/src/PolkitListener.cpp	2011-01-30 03:27:48.374000072 +0200
@@ -32,7 +32,7 @@
 #include <polkit-qt-1/PolkitQt1/Subject>
 #include <polkit-qt-1/PolkitQt1/Identity>
 #include <polkit-qt-1/PolkitQt1/Details>
-#include <qt4/QtGui/qdialogbuttonbox.h>
+#include <QtGui/qdialogbuttonbox.h>
 
 #include "../include/Globals.h"
 #include "../include/PolkitListener.h"
