diff -uNr linux-2.6.16.1/drivers/usb/storage/usb.c linux-2.6.16.1-imac/drivers/usb/storage/usb.c
--- linux-2.6.16.1/drivers/usb/storage/usb.c	2006-03-20 05:53:29.000000000 +0000
+++ linux-2.6.16.1-imac/drivers/usb/storage/usb.c	2006-04-01 23:51:38.000000000 +0000
@@ -104,7 +104,7 @@
 MODULE_DESCRIPTION("USB Mass Storage driver for Linux");
 MODULE_LICENSE("GPL");
 
-static unsigned int delay_use = 5;
+static unsigned int delay_use = 0;
 module_param(delay_use, uint, S_IRUGO | S_IWUSR);
 MODULE_PARM_DESC(delay_use, "seconds to delay before using a new device");
 
