diff -Naur QtCurve-Gtk2-0.62.8-orig/style/qt_settings.c QtCurve-Gtk2-0.62.8/style/qt_settings.c
--- QtCurve-Gtk2-0.62.8-orig/style/qt_settings.c	2009-05-13 18:51:33.358629095 +0300
+++ QtCurve-Gtk2-0.62.8/style/qt_settings.c	2009-05-13 18:55:00.501355118 +0300
@@ -194,7 +194,7 @@
 #endif
 
 #define DEFAULT_KDE_FONT      "Sans Serif"
-#define DEFAULT_KDE_FONT_SIZE 10.0
+#define DEFAULT_KDE_FONT_SIZE 9.0
 #define MAX_LINE_LEN 1024
 
 struct QtData qtSettings;
