diff -Nur eric4-4.1.5-old/eric/eric4.py eric4-4.1.5/eric/eric4.py
--- eric4-4.1.5-old/eric/eric4.py	2008-06-22 01:58:33.000000000 +0300
+++ eric4-4.1.5/eric/eric4.py	2008-06-22 02:45:46.000000000 +0300
@@ -149,6 +149,9 @@
     global args, mainWindow, splash
     
     sys.excepthook = excepthook
+
+    # FIXME: Fix i/I stuff in module load
+    os.environ["LC_ALL"] = "en_US.UTF-8"
     
     options = [\
         ("--nosplash", "don't show the splash screen"),
