From b5f33108fea77280c9acd219954f20401f760faf Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Mon, 27 Oct 2008 17:54:36 -0400
Subject: [PATCH] Adjust the nv34 panel tweak for Toshiba Tecra M2.

---
 src/nv_dac.c |   17 +++++++++++++++--
 1 files changed, 15 insertions(+), 2 deletions(-)

diff --git xf86-video-nv-2.1.12/src/nv_dac.c xf86-video-nv-2.1.12/src/nv_dac.c
index b1049cc..a256d0a 100644
--- xf86-video-nv-2.1.12/src/nv_dac.c
+++ xf86-video-nv-2.1.12/src/nv_dac.c
@@ -42,8 +42,21 @@ NVDACPanelTweaks(NVPtr pNv, NVRegPtr state)
           we can do is experiment and apply hacks. */
 
        if(((pNv->Chipset & 0xffff) == 0x0328) && (state->bpp == 32)) {
-          /* At least one NV34 laptop needs this workaround. */
-          tweak = -1;
+#if XSERVER_LIBPCIACCESS
+          if (((pNv->PciInfo->subvendor_id & 0xffff) == 0x1179) &&
+              ((pNv->PciInfo->subdevice_id & 0xffff) == 0x0020))
+#else
+          if (((pNv->PciInfo->subsysVendor & 0xffff) == 0x1179) &&
+              ((pNv->PciInfo->subsysCard & 0xffff) == 0x0020))
+#endif
+          {
+
+             /* Toshiba Tecra M2 */
+             tweak = 1;
+          } else {
+             /* At least one NV34 laptop needs this workaround. */
+             tweak = -1;
+          }
        }
 
        if((pNv->Chipset & 0xfff0) == 0x0310) {
-- 
1.6.0.6

