diff -Nur hsfmodem-7.47.00.03full-old/scripts/cnxtconfig.in hsfmodem-7.47.00.03full/scripts/cnxtconfig.in
--- hsfmodem-7.47.00.03full-old/scripts/cnxtconfig.in	2006-09-22 17:08:42.854701824 +0300
+++ hsfmodem-7.47.00.03full/scripts/cnxtconfig.in	2006-09-22 17:09:58.352224456 +0300
@@ -1838,7 +1838,7 @@
 
 	if [ -d /etc/udev/rules.d ]; then
 		rm -f /etc/udev/rules.d/@CNXTTARGET@.rules
-		echo 'KERNEL=="ttyS@CNXTSERDEV@0", SYMLINK="modem"' > /etc/udev/rules.d/@CNXTTARGET@.rules
+		echo 'KERNEL="ttyS@CNXTSERDEV@0", SYMLINK="modem", GROUP="dialout"' > /etc/udev/rules.d/@CNXTTARGET@.rules
 	fi
 }
 
