Index: dbus-glib-0.86/dbus/examples/Makefile.in
===================================================================
--- dbus-glib-0.86.orig/dbus/examples/Makefile.in
+++ dbus-glib-0.86/dbus/examples/Makefile.in
@@ -162,7 +162,7 @@ CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DATADIRNAME = @DATADIRNAME@
-DBUS_BINDING_TOOL = @DBUS_BINDING_TOOL@
+DBUS_BINDING_TOOL = dbus-binding-tool
 DBUS_CFLAGS = @DBUS_CFLAGS@
 DBUS_DAEMONDIR = @DBUS_DAEMONDIR@
 DBUS_GLIB_CFLAGS = @DBUS_GLIB_CFLAGS@
Index: dbus-glib-0.86/dbus/examples/statemachine/Makefile.in
===================================================================
--- dbus-glib-0.86.orig/dbus/examples/statemachine/Makefile.in
+++ dbus-glib-0.86/dbus/examples/statemachine/Makefile.in
@@ -107,7 +107,7 @@ CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DATADIRNAME = @DATADIRNAME@
-DBUS_BINDING_TOOL = @DBUS_BINDING_TOOL@
+DBUS_BINDING_TOOL = dbus-binding-tool
 DBUS_CFLAGS = @DBUS_CFLAGS@
 DBUS_DAEMONDIR = @DBUS_DAEMONDIR@
 DBUS_GLIB_CFLAGS = @DBUS_GLIB_CFLAGS@
Index: dbus-glib-0.86/tools/Makefile.in
===================================================================
--- dbus-glib-0.86.orig/tools/Makefile.in
+++ dbus-glib-0.86/tools/Makefile.in
@@ -522,7 +522,7 @@ uninstall-am: uninstall-nodist_libdbus_g
 
 
 dbus-glib-bindings.h: $(top_srcdir)/dbus-bus-introspect.xml $(top_builddir)/dbus/dbus-binding-tool$(EXEEXT)
-	$(DBUS_BINDING_TOOL) --mode=glib-client --prefix=dbus_bus --output=dbus-glib-bindings.h $(top_srcdir)/dbus-bus-introspect.xml
+	dbus-binding-tool --mode=glib-client --prefix=dbus_bus --output=dbus-glib-bindings.h $(top_srcdir)/dbus-bus-introspect.xml
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
