--- digikam/utilities/hotplug/digikam-download.desktop.in	2006-12-05 21:52:32.000000000 +0200
+++ digikam/utilities/hotplug/digikam-download.desktop.in	2006-12-12 11:18:47.009094128 +0200
@@ -2,6 +2,7 @@
 Exec=@digikamhelper_dir@/digikam-camera storage %u
 Icon=digikam
 Name=digiKam Download
+Name[tr]=digiKam ile resimler göster
 
 [Desktop Entry]
 Actions=digiKam Download
--- digikam/utilities/hotplug/digikam-mount-and-download.desktop.in	2006-12-05 21:52:32.000000000 +0200
+++ digikam/utilities/hotplug/digikam-mount-and-download.desktop.in	2006-12-12 11:19:18.392323152 +0200
@@ -2,6 +2,7 @@
 Exec=@digikamhelper_dir@/digikam-camera storage %u
 Icon=digikam
 Name=digiKam Mount and Download
+Name[tr]=digiKam ile resimleri indir
 
 [Desktop Entry]
 Actions=digiKam Mount and Download
--- digikam/utilities/hotplug/digikam-gphoto2-camera.desktop.in	2006-12-12 10:18:40.416379560 +0200
+++ digikam/utilities/hotplug/digikam-gphoto2-camera.desktop.in	2006-12-12 11:19:04.543428504 +0200
@@ -2,6 +2,7 @@
 Exec=@digikamhelper_dir@/digikam-camera detect %u
 Icon=digikam
 Name=digiKam Detect and Download
+Name[tr]=digiKam ile resimleri göster
 
 [Desktop Entry]
 Actions=digiKam Detect and Download
