diff -Naur waheela.orig/misc/waheela.desktop waheela/misc/waheela.desktop
--- waheela.orig/misc/waheela.desktop	2010-02-13 15:57:01.725272688 +0200
+++ waheela/misc/waheela.desktop	2010-02-13 16:07:12.459274494 +0200
@@ -1,7 +1,8 @@
 [Desktop Entry]
 Name=Waheela
 Version=0.2.1
-GenericName=Amarok fullscreen player
+GenericName=Amarok Fullscreen Player
+GenericName[tr]=Tam ekran Amarok Oynatıcı
 Type=Application
 Exec=waheela
 Icon=waheela
