--- mdadm-3.0-devel3/super-ddf.c.orig	2009-03-10 01:39:41.000000000 -0400
+++ mdadm-3.0-devel3/super-ddf.c	2009-03-17 15:22:41.000000000 -0400
@@ -1062,9 +1062,9 @@
 			       map_num(ddf_sec_level, vc->srl) ?: "-unknown-");
 		}
 		printf("  Device Size[%d] : %llu\n", n,
-		       __be64_to_cpu(vc->blocks)/2);
+		       (unsigned long long)__be64_to_cpu(vc->blocks)/2);
 		printf("   Array Size[%d] : %llu\n", n,
-		       __be64_to_cpu(vc->array_blocks)/2);
+		       (unsigned long long)__be64_to_cpu(vc->array_blocks)/2);
 	}
 }
 
@@ -1111,7 +1111,8 @@
 		//printf("\n");
 		printf("       %3d    %08x  ", i,
 		       __be32_to_cpu(pd->refnum));
-		printf("%lluK ",  __be64_to_cpu(pd->config_size)>>1);
+		printf("%lluK ", 
+		       (unsigned long long)__be64_to_cpu(pd->config_size)>>1);
 		for (dl = sb->dlist; dl ; dl = dl->next) {
 			if (dl->disk.refnum == pd->refnum) {
 				char *dv = map_dev(dl->major, dl->minor, 0);
