--- file-4.21/magic/Magdir/revision.svn	2007-04-09 18:33:07.000000000 +0200
+++ file-4.21/magic/Magdir/revision	2007-06-03 22:03:52.000000000 +0200
@@ -4,6 +4,11 @@
 # From Hendrik Scholz <hendrik@scholz.net>
 0	string	/1\ :pserver:	cvs password text file
 
+# Subversion (SVN) dumps
+# Uwe Zeisberger <zeisberg@informatik.uni-freiburg.de>
+0       string  SVN-fs-dump-format-version:     Subversion dumpfile
+>28     string  >\0                             (version: %s)
+
 # Conary changesets
 # From: Jonathan Smith <smithj@rpath.com>
 0	belong	0xea3f81bb	Conary changeset data
