diff -Nur mc-4.6.1-old/lib/mc.ext.in mc-4.6.1/lib/mc.ext.in
--- mc-4.6.1-old/lib/mc.ext.in	2006-12-12 13:04:59.000000000 +0200
+++ mc-4.6.1/lib/mc.ext.in	2006-12-12 13:07:13.000000000 +0200
@@ -343,7 +343,7 @@
 ### Sound files ###
 
 regex/\.([wW][aA][vV]|[sS][nN][dD]|[vV][oO][cC]|[aA][uU]|[sS][mM][pP]|[aA][iI][fF][fF]|[sS][nN][dD])$
-       Open=if [ "$DISPLAY" = "" ]; then play %f; else (xmms -e %f 1>/dev/null 2>&1 &); fi
+       Open=if [ "$DISPLAY" = "" ]; then play %f; else (gmplayer %f 1>/dev/null 2>&1 &); fi
 
 regex/\.([mM][oO][dD]|[sS]3[mM]|[xX][mM]|[iI][tT]|[mM][tT][mM]|669|[sS][tT][mM]|[uU][lL][tT]|[fF][aA][rR])$
        Open=mikmod %f
@@ -353,11 +353,11 @@
        Open=vplay -s 22 %f
 
 regex/\.([mM][pP]3)$
-	Open=if [ "$DISPLAY" = "" ]; then mpg123 %f; else (xmms %f &); fi
+	Open=if [ "$DISPLAY" = "" ]; then mpg123 %f; else (gmplayer %f &); fi
 	View=%view{ascii} mpg123 -vtn1 %f 2>&1 | sed -n '/^Title/,/^Comment/p;/^MPEG/,/^Audio/p'
 
 regex/\.([oO][gG][gG])$
-	Open=if [ "$DISPLAY" = "" ]; then ogg123 %f; else (xmms %f &); fi
+	Open=if [ "$DISPLAY" = "" ]; then ogg123 %f; else (gmplayer %f &); fi
 	View=%view{ascii} ogginfo %s
 
 regex/\.([mM][iI][dD][iI]?|[rR][mM][iI][dD]?)$
@@ -371,7 +371,7 @@
 ### Play lists ###
 
 regex/\.([mM]3[uU]|[pP][lL][sS])$
-	Open=if [ -z "$DISPLAY" ]; then mplayer -vo null -playlist %f; else (xmms -p %f &); fi
+	Open=if [ -z "$DISPLAY" ]; then mplayer -vo null -playlist %f; else (gmplayer -playlist %f &); fi
 
 
 ### Video ###
