Index: src/support.c
===================================================================
--- src/support.c	(revision 1088)
+++ src/support.c	(revision 1089)
@@ -1049,6 +1049,7 @@
     av[ac++] = g_strdup_printf("null");
     av[ac++] = g_strdup_printf("-frames");
     av[ac++] = g_strdup_printf("0");
+    av[ac++] = g_strdup_printf("-noidx");
     av[ac++] = g_strdup_printf("-identify");
     av[ac++] = g_strdup_printf("-nocache");
     if (idledata->device != NULL) {
