--- ghostscript-8.60.dfsg.4~/lib/pdf_main.ps	2007-06-14 10:51:40.000000000 +0100
+++ ghostscript-8.60.dfsg.4/lib/pdf_main.ps	2007-06-14 11:17:37.000000000 +0100
@@ -79,7 +79,7 @@
   } loop
   exch pop
   {
-    dup 1023 string .peekstring pop dup length 400 ge {
+    dup 127 string .peekstring pop dup length 400 ge {
       % "1024 string" exceeds current %stdin buffer
       % Valid PDF file cannot be smaller than 400 bytes.
       (%PDF-1.) search {
