--- code/tools/asm/cmdlib.c.orig	2005-10-29 03:53:08.000000000 +0300
+++ code/tools/asm/cmdlib.c	2007-10-20 16:27:11.000000000 +0300
@@ -312,7 +312,7 @@
 char *ExpandPath (const char *path)
 {
 	static char full[1024];
-	if (!qdir)
+	if (!strlen(qdir))
 		Error ("ExpandPath called without qdir set");
 	if (path[0] == '/' || path[0] == '\\' || path[1] == ':') {
 		strcpy( full, path );
@@ -325,7 +325,7 @@
 char *ExpandGamePath (const char *path)
 {
 	static char full[1024];
-	if (!qdir)
+	if (!strlen(qdir))
 		Error ("ExpandGamePath called without qdir set");
 	if (path[0] == '/' || path[0] == '\\' || path[1] == ':') {
 		strcpy( full, path );
