diff -Nur ffmpeg-old/libavformat/mov.c ffmpeg/libavformat/mov.c
--- ffmpeg-old/libavformat/mov.c	2008-08-20 11:15:21.000000000 +0300
+++ ffmpeg/libavformat/mov.c	2008-08-20 11:19:03.000000000 +0300
@@ -323,6 +323,8 @@
         st->codec->codec_type = CODEC_TYPE_AUDIO;
     else if(type == MKTAG('m','1','a',' '))
         st->codec->codec_id = CODEC_ID_MP2;
+    else if(type == MKTAG('m','1','v',' '))
+        st->codec->codec_id = CODEC_ID_MPEG1VIDEO;
     else if(type == MKTAG('s','u','b','p')) {
         st->codec->codec_type = CODEC_TYPE_SUBTITLE;
     }
@@ -995,6 +997,9 @@
         st->codec->sample_rate= 8000;
         st->codec->channels= 1; /* really needed */
         break;
+    case CODEC_ID_MPEG1VIDEO:
+        st->codec->codec_type = CODEC_TYPE_VIDEO;
+        break;
     case CODEC_ID_MP2:
     case CODEC_ID_MP3:
         st->codec->codec_type = CODEC_TYPE_AUDIO; /* force type after stsd for m1a hdlr */
