--- mplayer-old/mplayer.c	2009-01-24 02:40:38.954320617 +1300
+++ mplayer/mplayer.c	2009-01-24 02:42:08.954210575 +1300
@@ -2296,9 +2296,9 @@
 	}
 	if (sh_video->last_pts == MP_NOPTS_VALUE)
 	    sh_video->last_pts= sh_video->pts;
-	else if (sh_video->last_pts >= sh_video->pts) {
+	else if (sh_video->last_pts > sh_video->pts) {
 	    sh_video->last_pts = sh_video->pts;
-	    mp_msg(MSGT_CPLAYER, MSGL_INFO, "pts value <= previous\n");
+	    mp_msg(MSGT_CPLAYER, MSGL_INFO, "pts value < previous\n");
 	}
 	frame_time = sh_video->pts - sh_video->last_pts;
 	sh_video->last_pts = sh_video->pts;

