Index: help/help_mp-tr.h
===================================================================
--- help/help_mp-tr.h	(revision 28227)
+++ help/help_mp-tr.h	(working copy)
@@ -56,7 +56,7 @@
 // mplayer.c:
 
 #define MSGTR_Exiting "\nÇıkılıyor...\n"
-#define MSGTR_ExitingHow "\n(%s) Çıkılıyor... \n"
+#define MSGTR_ExitingHow "\nÇıkılıyor... (%s)\n"
 #define MSGTR_Exit_quit "Çık"
 #define MSGTR_Exit_eof "Dosya sonu"
 #define MSGTR_Exit_error "Önemli Hata"
