Index: help/help_mp-tr.h
===================================================================
--- help/help_mp-tr.h	(revision 28227)
+++ help/help_mp-tr.h	(working copy)
@@ -1518,7 +1518,7 @@
 #define MSGTR_MPDEMUX_NW_UnknownAF "Bilinmeyen adres sınıfı %d\n"
 #define MSGTR_MPDEMUX_NW_ResolvingHostForAF "%s çözülüyor ( %s için )...\n"
 #define MSGTR_MPDEMUX_NW_CantResolv "%s için ad çözülemedi: %s\n"
-#define MSGTR_MPDEMUX_NW_ConnectingToServer "%s[%s] sunucusuna bağlanılıyor: %d...\n"
+#define MSGTR_MPDEMUX_NW_ConnectingToServer "Bağlantı kuruluyor %s[%s]: %d...\n"
 #define MSGTR_MPDEMUX_NW_CantConnect2Server "%s ile sunucuya bağlanılamadı\n"
 #define MSGTR_MPDEMUX_NW_SelectFailed "Seçim başarısız.\n"
 #define MSGTR_MPDEMUX_NW_ConnTimeout "Bağlantı zamanaşımı.\n"
