--- configure.in	2007-04-25 11:38:16.000000000 +0300
+++ configure.in	2007-04-25 22:42:09.000000000 +0300
@@ -29,11 +29,6 @@
 SCROLLKEEPER_BUILD_REQUIRED=0.3.5
 AC_SUBST(SCROLLKEEPER_BUILD_REQUIRED)
 
-AC_PATH_PROG(SK_CONFIG,scrollkeeper-config,no)
-if test x$SK_CONFIG = xno; then
-  AC_MSG_ERROR(Couldn't find scrollkeeper-config. Please install the scrollkeeper package: http://scrollkeeper.sourceforge.net)
-fi
-
 # here follows a hack to get the docs installed in a separate, recursive install-doc target
 MK=''; AC_SUBST(MK)
 #########################################################
