diff -Nur hal-0.5.9.1.orig/privileges/hal-power-cpufreq.privilege hal-0.5.9.1/privileges/hal-power-cpufreq.privilege
--- hal-0.5.9.1.orig/privileges/hal-power-cpufreq.privilege	2007-04-03 19:38:50.000000000 +0300
+++ hal-0.5.9.1/privileges/hal-power-cpufreq.privilege	2007-07-09 16:01:47.000000000 +0300
@@ -4,8 +4,8 @@
 
 [Privilege]
 RequiredPrivileges=
-SufficientPrivileges=desktop-console
-Allow=uid:root
+SufficientPrivileges=
+Allow=gid:power
 Deny=
 CanObtain=True
 CanGrant=True
diff -Nur hal-0.5.9.1.orig/privileges/hal-power-hibernate.privilege hal-0.5.9.1/privileges/hal-power-hibernate.privilege
--- hal-0.5.9.1.orig/privileges/hal-power-hibernate.privilege	2007-04-03 19:38:50.000000000 +0300
+++ hal-0.5.9.1/privileges/hal-power-hibernate.privilege	2007-07-09 16:01:59.000000000 +0300
@@ -3,8 +3,8 @@
 
 [Privilege]
 RequiredPrivileges=
-SufficientPrivileges=desktop-console
-Allow=uid:root
+SufficientPrivileges=
+Allow=gid:power
 Deny=
 CanObtain=True
 CanGrant=True
diff -Nur hal-0.5.9.1.orig/privileges/hal-power-poweroff.privilege hal-0.5.9.1/privileges/hal-power-poweroff.privilege
--- hal-0.5.9.1.orig/privileges/hal-power-poweroff.privilege	2007-04-03 19:38:50.000000000 +0300
+++ hal-0.5.9.1/privileges/hal-power-poweroff.privilege	2007-07-09 16:02:10.000000000 +0300
@@ -2,9 +2,9 @@
 # This privilege specifies who is allowed to power off the system.
 
 [Privilege]
-RequiredPrivileges=desktop-console
+RequiredPrivileges=
 SufficientPrivileges=
-Allow=uid:__all__
+Allow=gid:power
 Deny=
 CanObtain=True
 CanGrant=True
diff -Nur hal-0.5.9.1.orig/privileges/hal-power-reboot.privilege hal-0.5.9.1/privileges/hal-power-reboot.privilege
--- hal-0.5.9.1.orig/privileges/hal-power-reboot.privilege	2007-04-03 19:38:50.000000000 +0300
+++ hal-0.5.9.1/privileges/hal-power-reboot.privilege	2007-07-09 16:02:17.000000000 +0300
@@ -2,9 +2,9 @@
 # This privilege specifies who is allowed to reboot the system.
 
 [Privilege]
-RequiredPrivileges=desktop-console
+RequiredPrivileges=
 SufficientPrivileges=
-Allow=uid:__all__
+Allow=gid:power
 Deny=
 CanObtain=True
 CanGrant=True
diff -Nur hal-0.5.9.1.orig/privileges/hal-power-standby.privilege hal-0.5.9.1/privileges/hal-power-standby.privilege
--- hal-0.5.9.1.orig/privileges/hal-power-standby.privilege	2007-07-09 16:00:53.000000000 +0300
+++ hal-0.5.9.1/privileges/hal-power-standby.privilege	2007-07-09 16:02:23.000000000 +0300
@@ -2,9 +2,9 @@
 # This privilege specifies who is allowed to suspend the system.
 
 [Privilege]
-RequiredPrivileges=desktop-console
+RequiredPrivileges=
 SufficientPrivileges=
-Allow=uid:__all__
+Allow=gid:power
 Deny=
 CanObtain=True
 CanGrant=True
diff -Nur hal-0.5.9.1.orig/privileges/hal-power-suspend.privilege hal-0.5.9.1/privileges/hal-power-suspend.privilege
--- hal-0.5.9.1.orig/privileges/hal-power-suspend.privilege	2007-04-03 19:38:50.000000000 +0300
+++ hal-0.5.9.1/privileges/hal-power-suspend.privilege	2007-07-09 16:02:26.000000000 +0300
@@ -3,8 +3,8 @@
 
 [Privilege]
 RequiredPrivileges=
-SufficientPrivileges=desktop-console
-Allow=uid:root
+SufficientPrivileges=
+Allow=gid:power
 Deny=
 CanObtain=True
 CanGrant=True
diff -Nur hal-0.5.9.1.orig/privileges/hal-storage-fixed-mount-all-options.privilege hal-0.5.9.1/privileges/hal-storage-fixed-mount-all-options.privilege
--- hal-0.5.9.1.orig/privileges/hal-storage-fixed-mount-all-options.privilege	2007-04-03 19:38:50.000000000 +0300
+++ hal-0.5.9.1/privileges/hal-storage-fixed-mount-all-options.privilege	2007-07-09 16:03:32.000000000 +0300
@@ -5,9 +5,9 @@
 # uid=<userid>).
 
 [Privilege]
-RequiredPrivileges=desktop-console
+RequiredPrivileges=
 SufficientPrivileges=
-Allow=
+Allow=gid:disk
 Deny=
 CanObtain=True
 CanGrant=True
diff -Nur hal-0.5.9.1.orig/privileges/hal-storage-fixed-mount.privilege hal-0.5.9.1/privileges/hal-storage-fixed-mount.privilege
--- hal-0.5.9.1.orig/privileges/hal-storage-fixed-mount.privilege	2007-04-03 19:38:50.000000000 +0300
+++ hal-0.5.9.1/privileges/hal-storage-fixed-mount.privilege	2007-07-09 16:02:43.000000000 +0300
@@ -7,9 +7,9 @@
 #
 
 [Privilege]
-RequiredPrivileges=desktop-console
+RequiredPrivileges=
 SufficientPrivileges=
-Allow=
+Allow=gid:disk
 Deny=
 CanObtain=True
 CanGrant=True
diff -Nur hal-0.5.9.1.orig/privileges/hal-storage-removable-mount-all-options.privilege hal-0.5.9.1/privileges/hal-storage-removable-mount-all-options.privilege
--- hal-0.5.9.1.orig/privileges/hal-storage-removable-mount-all-options.privilege	2007-04-03 19:38:50.000000000 +0300
+++ hal-0.5.9.1/privileges/hal-storage-removable-mount-all-options.privilege	2007-07-09 16:02:47.000000000 +0300
@@ -5,9 +5,9 @@
 # uid=<userid>).
 
 [Privilege]
-RequiredPrivileges=desktop-console
+RequiredPrivileges=
 SufficientPrivileges=
-Allow=
+Allow=gid:removable
 Deny=
 CanObtain=True
 CanGrant=True
diff -Nur hal-0.5.9.1.orig/privileges/hal-storage-removable-mount.privilege hal-0.5.9.1/privileges/hal-storage-removable-mount.privilege
--- hal-0.5.9.1.orig/privileges/hal-storage-removable-mount.privilege	2007-04-03 19:38:50.000000000 +0300
+++ hal-0.5.9.1/privileges/hal-storage-removable-mount.privilege	2007-07-09 16:03:40.000000000 +0300
@@ -6,9 +6,9 @@
 # the privilege hal-removable-fixed-mount-all-options.privilege.
 
 [Privilege]
-RequiredPrivileges=desktop-console
+RequiredPrivileges=
 SufficientPrivileges=
-Allow=uid:__all__
+Allow=gid:removable
 Deny=
 CanObtain=True
 CanGrant=True
