Index: mozilla/security/coreconf/Linux.mk
===================================================================
RCS file: /cvsroot/mozilla/security/coreconf/Linux.mk,v
retrieving revision 1.27
diff -u -r1.27 Linux.mk
--- mozilla/security/coreconf/Linux.mk	16 Sep 2005 17:09:23 -0000	1.27
+++ mozilla/security/coreconf/Linux.mk	25 Nov 2005 22:08:04 -0000
@@ -148,7 +148,7 @@
 endif
 
 ifdef BUILD_OPT
-	OPTIMIZER	= -O2
+	OPTIMIZER	= -fno-strict-aliasing
 endif
 
 ifeq ($(USE_PTHREADS),1)
