Index: pisi-1.1.1/pisi/api.py
===================================================================
--- pisi-1.1.1/pisi/api.py	(revision 12780)
+++ pisi-1.1.1/pisi/api.py	(working copy)
@@ -339,7 +339,10 @@
     return metadata, files
 
 def search_package_terms(terms, repo = pisi.itembyrepodb.all):
+    return search_in_packages(terms, ctx.packagedb.list_packages(repo), repo)
 
+def search_in_packages(terms, packages, repo = pisi.itembyrepodb.all):
+
     def search(package, term):
         term = unicode(term).lower()
         if term in unicode(package.name).lower() or \
@@ -348,7 +351,7 @@
             return True
 
     found = []
-    for name in ctx.packagedb.list_packages(repo):
+    for name in packages:
         pkg = ctx.packagedb.get_package(name, repo)
         if terms == filter(lambda x:search(pkg, x), terms):
             found.append(name)
