diff -Nur file-integration/x-basket-archive.desktop file-integration/x-basket-archive.desktop
--- file-integration/x-basket-archive.desktop	2007-02-06 11:15:07.000000000 +0200
+++ file-integration/x-basket-archive.desktop	2007-03-22 04:55:34.000000000 +0200
@@ -12,5 +12,6 @@
 Comment[nn]=Korgarkiv
 Comment[pt]=Ficheiro de cestos
 Comment[ru]=Контейнер Basket
+Comment[tr]=Basket Arşivi
 Icon=baskets
 Patterns=*.baskets
diff -Nur file-integration/x-basket-template.desktop file-integration/x-basket-template.desktop
--- file-integration/x-basket-template.desktop	2007-02-06 11:15:06.000000000 +0200
+++ file-integration/x-basket-template.desktop	2007-03-22 04:56:09.000000000 +0200
@@ -12,5 +12,6 @@
 Comment[nn]=Korgmal
 Comment[pt]=Modelo de cesto
 Comment[ru]=Шаблон корзины
+Comment[tr]=Basket Şablonu
 Icon=baskett
 Patterns=*.baskett
diff -Nur kontact_plugin/basket.desktop kontact_plugin/basket.desktop
--- kontact_plugin/basket.desktop	2007-03-18 18:54:52.000000000 +0200
+++ kontact_plugin/basket.desktop	2007-03-22 04:53:46.000000000 +0200
@@ -27,6 +27,7 @@
 Comment[nn]=Korgnotat-tillegg for Kontakt
 Comment[pt]=Plugin do Bloco de Notas BasKet para o Kontact
 Comment[ru]=Модуль Альбом заметок Basket для Kontact
+Comment[tr]=Kontact için Basket Not Tutucu eklentisi
 
 Name=Baskets
 Name[de]=Körbe
@@ -38,3 +39,4 @@
 Name[nn]=Korger
 Name[pt]=Cestos
 Name[ru]=Корзины
+Name[tr]=Sepetler
\ Dosya sonunda yenisatır yok.
diff -Nur kontact_plugin/basket_v4.desktop kontact_plugin/basket_v4.desktop
--- kontact_plugin/basket_v4.desktop	2007-03-18 18:54:59.000000000 +0200
+++ kontact_plugin/basket_v4.desktop	2007-03-22 04:54:29.000000000 +0200
@@ -27,6 +27,7 @@
 Comment[nn]=Korgnotat-tillegg for Kontakt
 Comment[pt]=Plugin do Bloco de Notas BasKet para o Kontact
 Comment[ru]=Модуль Альбом заметок Basket для Kontact
+Comment[tr]=Kontact için Basket Not Tutucu eklentisi
 
 Name=Baskets
 Name[de]=Körbe
@@ -38,3 +39,4 @@
 Name[nn]=Korger
 Name[pt]=Cestos
 Name[ru]=Корзины
+Name[tr]=Sepetler
diff -Nur src/basket_config_general.desktop src/basket_config_general.desktop
--- src/basket_config_general.desktop	2007-03-01 15:19:37.000000000 +0200
+++ src/basket_config_general.desktop	2007-03-22 05:01:59.000000000 +0200
@@ -24,6 +24,7 @@
 Name[pt]=Geral
 Name[ru]=Общие
 Name[tr]=Genel
+
 Comment=General
 Comment[de]=Allgemein
 Comment[es]=General
@@ -34,4 +35,4 @@
 Comment[nn]=Generelt
 Comment[pt]=Geral
 Comment[ru]=Общие
-Comment[de]=Genel
\ Dosya sonunda yenisatır yok.
+Comment[tr]=Genel
\ Dosya sonunda yenisatır yok.
