diff -Nur pwc-v4l2-20080322-042701-old/pwc-if.c pwc-v4l2-20080322-042701/pwc-if.c
--- pwc-v4l2-20080322-042701-old/pwc-if.c	2008-03-22 19:35:27.000000000 +0200
+++ pwc-v4l2-20080322-042701/pwc-if.c	2008-03-22 19:36:55.000000000 +0200
@@ -64,7 +64,6 @@
 #include <linux/vmalloc.h>
 #include <linux/version.h>
 #include <asm/io.h>
-#include <linux/moduleparam.h>
 
 #include "pwc.h"
 #include "pwc-kiara.h"
@@ -167,7 +166,6 @@
 	.owner =	THIS_MODULE,
 	.name =		"Philips Webcam",	/* Filled in later */
 	.type =		VID_TYPE_CAPTURE,
-	.hardware =	VID_HARDWARE_PWC,
 	.release =	video_device_release,
 	.fops =         &pwc_fops,
 	.minor =        -1,
