--- Makefile.in~	2007-05-14 15:50:54.000000000 +0300
+++ Makefile.in	2008-07-04 23:30:15.000000000 +0300
@@ -191,7 +191,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-SUBDIRS = src tools doc examples pkgconfig
+SUBDIRS = src tools doc pkgconfig
 EXTRA_DIST = LICENSE.MIT ChangeLog README.TXT jrtplib.dsw jrtplib.dsp jrtplib.vcproj \
 	     jrtplib.sln jrtplib_wce.vcproj jrtplib_wce.sln jrtplib_wce5.sln \
 	     jrtplib_wce5.vcproj Doxyfile
