diff -Naur irrlicht-1.4.orig/examples/01.HelloWorld/main.cpp irrlicht-1.4/examples/01.HelloWorld/main.cpp
--- irrlicht-1.4.orig/examples/01.HelloWorld/main.cpp	2007-11-17 07:38:54.000000000 +0200
+++ irrlicht-1.4/examples/01.HelloWorld/main.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -145,7 +145,7 @@
 	By the way, that cool Quake 2 model called sydney was modelled
 	by Brian Collins.
 	*/
-	IAnimatedMesh* mesh = smgr->getMesh("../../media/sydney.md2");
+	IAnimatedMesh* mesh = smgr->getMesh("/usr/share/irrlicht/sydney.md2");
 	IAnimatedMeshSceneNode* node = smgr->addAnimatedMeshSceneNode( mesh );
 
 	/*
@@ -160,7 +160,7 @@
 	{
 		node->setMaterialFlag(EMF_LIGHTING, false);
 		node->setMD2Animation ( scene::EMAT_STAND );
-		node->setMaterialTexture( 0, driver->getTexture("../../media/sydney.bmp") );
+		node->setMaterialTexture( 0, driver->getTexture("/usr/share/irrlicht/sydney.bmp") );
 	}
 
 	/*
diff -Naur irrlicht-1.4.orig/examples/02.Quake3Map/main.cpp irrlicht-1.4/examples/02.Quake3Map/main.cpp
--- irrlicht-1.4.orig/examples/02.Quake3Map/main.cpp	2007-11-17 07:38:54.000000000 +0200
+++ irrlicht-1.4/examples/02.Quake3Map/main.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -92,7 +92,7 @@
 	we are able to read from the files in that archive as they would
 	directly be stored on disk.
 	*/
-	device->getFileSystem()->addZipFileArchive("../../media/map-20kdm2.pk3");
+	device->getFileSystem()->addZipFileArchive("/usr/share/irrlicht/map-20kdm2.pk3");
 
 	/* 
 	Now we can load the mesh by calling getMesh(). We get a pointer returned
diff -Naur irrlicht-1.4.orig/examples/04.Movement/main.cpp irrlicht-1.4/examples/04.Movement/main.cpp
--- irrlicht-1.4.orig/examples/04.Movement/main.cpp	2007-11-17 07:38:56.000000000 +0200
+++ irrlicht-1.4/examples/04.Movement/main.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -119,7 +119,7 @@
 	*/
 	node = smgr->addSphereSceneNode();
 	node->setPosition(core::vector3df(0,0,30));
-	node->setMaterialTexture(0, driver->getTexture("../../media/wall.bmp"));
+	node->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/wall.bmp"));
 	node->setMaterialFlag(video::EMF_LIGHTING, false);
 
 
@@ -136,7 +136,7 @@
 
 	if (n)
 	{
-		n->setMaterialTexture(0, driver->getTexture("../../media/t351sml.jpg"));
+		n->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/t351sml.jpg"));
 		n->setMaterialFlag(video::EMF_LIGHTING, false);
 		scene::ISceneNodeAnimator* anim =
 			smgr->createFlyCircleAnimator(core::vector3df(0,0,30), 20.0f);
@@ -148,7 +148,7 @@
 	The last scene node we add to show possibilities of scene node animators is 
 	a md2 model, which uses a 'fly straight' animator to run between to points.
 	*/
-	scene::IAnimatedMeshSceneNode* anms = smgr->addAnimatedMeshSceneNode(smgr->getMesh("../../media/sydney.md2"));
+	scene::IAnimatedMeshSceneNode* anms = smgr->addAnimatedMeshSceneNode(smgr->getMesh("/usr/share/irrlicht/sydney.md2"));
 
 	if (anms)
 	{
@@ -175,7 +175,7 @@
 		anms->setMD2Animation(scene::EMAT_RUN);
 
 		anms->setRotation(core::vector3df(0,180.0f,0));
-		anms->setMaterialTexture(0, driver->getTexture("../../media/sydney.bmp"));
+		anms->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/sydney.bmp"));
 
 	}
 
@@ -192,7 +192,7 @@
 	Add a colorful irrlicht logo
 	*/
 	device->getGUIEnvironment()->addImage(
-		driver->getTexture("../../media/irrlichtlogoalpha2.tga"),
+		driver->getTexture("/usr/share/irrlicht/irrlichtlogoalpha2.tga"),
 		core::position2d<s32>(10,10));
 
 	/*
diff -Naur irrlicht-1.4.orig/examples/05.UserInterface/main.cpp irrlicht-1.4/examples/05.UserInterface/main.cpp
--- irrlicht-1.4.orig/examples/05.UserInterface/main.cpp	2007-11-17 07:38:54.000000000 +0200
+++ irrlicht-1.4/examples/05.UserInterface/main.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -188,7 +188,7 @@
 	*/
 
 	IGUISkin* skin = env->getSkin();
-	IGUIFont* font = env->getFont("../../media/fonthaettenschweiler.bmp");
+	IGUIFont* font = env->getFont("/usr/share/irrlicht/fonthaettenschweiler.bmp");
 	if (font)
 		skin->setFont(font);
 
@@ -225,7 +225,7 @@
 	env->addEditBox(L"Editable Text", rect<s32>(350, 80, 550, 100));
 
 	// add the engine logo
-	env->addImage(driver->getTexture("../../media/irrlichtlogo2.png"),
+	env->addImage(driver->getTexture("/usr/share/irrlicht/irrlichtlogo2.png"),
 			position2d<int>(10,10));
 
 
diff -Naur irrlicht-1.4.orig/examples/06.2DGraphics/main.cpp irrlicht-1.4/examples/06.2DGraphics/main.cpp
--- irrlicht-1.4.orig/examples/06.2DGraphics/main.cpp	2007-11-17 07:38:56.000000000 +0200
+++ irrlicht-1.4/examples/06.2DGraphics/main.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -69,7 +69,7 @@
 	e.g. all black pixels transparent. Please note, that makeColorKeyTexture
 	just creates an alpha channel based on the color. 
 */
-	video::ITexture* images = driver->getTexture("../../media/2ddemo.bmp");
+	video::ITexture* images = driver->getTexture("/usr/share/irrlicht/2ddemo.bmp");
 	driver->makeColorKeyTexture(images, core::position2d<s32>(0,0));
 
 	
@@ -81,7 +81,7 @@
 	images of the red imps (little flying creatures) in the texture.
 */
 	gui::IGUIFont* font = device->getGUIEnvironment()->getBuiltInFont();
-	gui::IGUIFont* font2 = device->getGUIEnvironment()->getFont("../../media/fonthaettenschweiler.bmp");
+	gui::IGUIFont* font2 = device->getGUIEnvironment()->getFont("/usr/share/irrlicht/fonthaettenschweiler.bmp");
 
 	core::rect<s32> imp1(349,15,385,78);
 	core::rect<s32> imp2(387,15,423,78);
diff -Naur irrlicht-1.4.orig/examples/07.Collision/main.cpp irrlicht-1.4/examples/07.Collision/main.cpp
--- irrlicht-1.4.orig/examples/07.Collision/main.cpp	2007-11-17 07:38:54.000000000 +0200
+++ irrlicht-1.4/examples/07.Collision/main.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -58,7 +58,7 @@
 	scene::ISceneManager* smgr = device->getSceneManager();
 
 	
-	device->getFileSystem()->addZipFileArchive("../../media/map-20kdm2.pk3");
+	device->getFileSystem()->addZipFileArchive("/usr/share/irrlicht/map-20kdm2.pk3");
 
 	
 	scene::IAnimatedMesh* q3levelmesh = smgr->getMesh("20kdm2.bsp");
@@ -152,7 +152,7 @@
 
 	scene::IBillboardSceneNode * bill = smgr->addBillboardSceneNode();
 	bill->setMaterialType(video::EMT_TRANSPARENT_ADD_COLOR );
-	bill->setMaterialTexture(0, driver->getTexture("../../media/particle.bmp"));
+	bill->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/particle.bmp"));
 	bill->setMaterialFlag(video::EMF_LIGHTING, false);
 	bill->setMaterialFlag(video::EMF_ZBUFFER, false);
 	bill->setSize(core::dimension2d<f32>(20.0f, 20.0f));
@@ -160,11 +160,11 @@
 	// add 3 animated faeries.
 
 	video::SMaterial material;
-	material.setTexture(0, driver->getTexture("../../media/faerie2.bmp"));
+	material.setTexture(0, driver->getTexture("/usr/share/irrlicht/faerie2.bmp"));
 	material.Lighting = true;
 
 	scene::IAnimatedMeshSceneNode* node = 0;
-	scene::IAnimatedMesh* faerie = smgr->getMesh("../../media/faerie.md2");
+	scene::IAnimatedMesh* faerie = smgr->getMesh("/usr/share/irrlicht/faerie.md2");
 
 	if (faerie)
 	{
diff -Naur irrlicht-1.4.orig/examples/08.SpecialFX/main.cpp irrlicht-1.4/examples/08.SpecialFX/main.cpp
--- irrlicht-1.4.orig/examples/08.SpecialFX/main.cpp	2007-11-17 07:38:54.000000000 +0200
+++ irrlicht-1.4/examples/08.SpecialFX/main.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -74,7 +74,7 @@
 	*/
 
 	scene::IAnimatedMesh* mesh = smgr->getMesh(
-		"../../media/room.3ds");
+		"/usr/share/irrlicht/room.3ds");
 
 	smgr->getMeshManipulator()->makePlanarTextureMapping(
 		mesh->getMesh(0), 0.004f);
@@ -82,7 +82,7 @@
 	scene::ISceneNode* node = 0;
 
 	node = smgr->addAnimatedMeshSceneNode(mesh);
-	node->setMaterialTexture(0,	driver->getTexture("../../media/wall.jpg"));
+	node->setMaterialTexture(0,	driver->getTexture("/usr/share/irrlicht/wall.jpg"));
 	node->getMaterial(0).SpecularColor.set(0,0,0,0);
 
 	/*
@@ -106,8 +106,8 @@
 	node = smgr->addWaterSurfaceSceneNode(mesh->getMesh(0), 3.0f, 300.0f, 30.0f);
 	node->setPosition(core::vector3df(0,7,0));
 
-	node->setMaterialTexture(0, driver->getTexture("../../media/stones.jpg"));
-	node->setMaterialTexture(1, driver->getTexture("../../media/water.jpg"));
+	node->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/stones.jpg"));
+	node->setMaterialTexture(1, driver->getTexture("/usr/share/irrlicht/water.jpg"));
 
 	node->setMaterialType(video::EMT_REFLECTION_2_LAYER);
 
@@ -132,7 +132,7 @@
 	node = smgr->addBillboardSceneNode(node, core::dimension2d<f32>(50, 50));
 	node->setMaterialFlag(video::EMF_LIGHTING, false);
 	node->setMaterialType(video::EMT_TRANSPARENT_ADD_COLOR);
-	node->setMaterialTexture(0, driver->getTexture("../../media/particlewhite.bmp"));
+	node->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/particlewhite.bmp"));
 
 	/*
 	The next special effect is a lot more interesting: A particle system. The particle
@@ -188,7 +188,7 @@
 
 	ps->setMaterialFlag(video::EMF_LIGHTING, false);
 	ps->setMaterialFlag(video::EMF_ZWRITE_ENABLE, false);
-	ps->setMaterialTexture(0, driver->getTexture("../../media/fire.bmp"));
+	ps->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/fire.bmp"));
 	ps->setMaterialType(video::EMT_TRANSPARENT_VERTEX_ALPHA);
 
 	/*
@@ -207,7 +207,7 @@
 
 	// add animated character
 
-	mesh = smgr->getMesh("../../media/dwarf.x");
+	mesh = smgr->getMesh("/usr/share/irrlicht/dwarf.x");
 	scene::IAnimatedMeshSceneNode* anode = 0;
 
 	anode = smgr->addAnimatedMeshSceneNode(mesh);
diff -Naur irrlicht-1.4.orig/examples/09.Meshviewer/main.cpp irrlicht-1.4/examples/09.Meshviewer/main.cpp
--- irrlicht-1.4.orig/examples/09.Meshviewer/main.cpp	2007-11-17 07:38:54.000000000 +0200
+++ irrlicht-1.4/examples/09.Meshviewer/main.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -468,7 +468,7 @@
 	smgr->addLightSceneNode();
 	smgr->addLightSceneNode(0, core::vector3df(50,-50,100), video::SColorf(1.0f,1.0f,1.0f),20000);
 	// add our media directory as "search path"
-	Device->getFileSystem()->addFolderFileArchive ( "../../media/" );
+	Device->getFileSystem()->addFolderFileArchive ( "/usr/share/irrlicht/" );
 
 	/*
 		The next step is to read the configuration file. It is stored in the xml
diff -Naur irrlicht-1.4.orig/examples/10.Shaders/main.cpp irrlicht-1.4/examples/10.Shaders/main.cpp
--- irrlicht-1.4.orig/examples/10.Shaders/main.cpp	2007-11-17 07:38:56.000000000 +0200
+++ irrlicht-1.4/examples/10.Shaders/main.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -174,32 +174,32 @@
 	switch(driverType)
 	{
 	case video::EDT_DIRECT3D8:
-		psFileName = "../../media/d3d8.psh";
-		vsFileName = "../../media/d3d8.vsh";
+		psFileName = "/usr/share/irrlicht/d3d8.psh";
+		vsFileName = "/usr/share/irrlicht/d3d8.vsh";
 		break;
 	case video::EDT_DIRECT3D9:
 		if (UseHighLevelShaders)
 		{
-			psFileName = "../../media/d3d9.hlsl";
+			psFileName = "/usr/share/irrlicht/d3d9.hlsl";
 			vsFileName = psFileName; // both shaders are in the same file
 		}
 		else
 		{
-			psFileName = "../../media/d3d9.psh";
-			vsFileName = "../../media/d3d9.vsh";
+			psFileName = "/usr/share/irrlicht/d3d9.psh";
+			vsFileName = "/usr/share/irrlicht/d3d9.vsh";
 		}
 		break;
 
 	case video::EDT_OPENGL:
 		if (UseHighLevelShaders)
 		{
-			psFileName = "../../media/opengl.frag";
-			vsFileName = "../../media/opengl.vert";
+			psFileName = "/usr/share/irrlicht/opengl.frag";
+			vsFileName = "/usr/share/irrlicht/opengl.vert";
 		}
 		else
 		{
-			psFileName = "../../media/opengl.psh";
-			vsFileName = "../../media/opengl.vsh";
+			psFileName = "/usr/share/irrlicht/opengl.psh";
+			vsFileName = "/usr/share/irrlicht/opengl.vsh";
 		}
 		break;
 	}
@@ -303,7 +303,7 @@
 
 	scene::ISceneNode* node = smgr->addCubeSceneNode(50);
 	node->setPosition(core::vector3df(0,0,0));
-	node->setMaterialTexture(0, driver->getTexture("../../media/wall.bmp"));
+	node->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/wall.bmp"));
 	node->setMaterialFlag(video::EMF_LIGHTING, false);
 	node->setMaterialType((video::E_MATERIAL_TYPE)newMaterialType1);
 
@@ -324,7 +324,7 @@
 
 	node = smgr->addCubeSceneNode(50);
 	node->setPosition(core::vector3df(0,-10,50));
-	node->setMaterialTexture(0, driver->getTexture("../../media/wall.bmp"));
+	node->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/wall.bmp"));
 	node->setMaterialFlag(video::EMF_LIGHTING, false);
 	node->setMaterialType((video::E_MATERIAL_TYPE)newMaterialType2);
 
@@ -345,7 +345,7 @@
 
 	node = smgr->addCubeSceneNode(50);
 	node->setPosition(core::vector3df(0,50,25));
-	node->setMaterialTexture(0, driver->getTexture("../../media/wall.bmp"));
+	node->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/wall.bmp"));
 	node->setMaterialFlag(video::EMF_LIGHTING, false);
 	smgr->addTextSceneNode(gui->getBuiltInFont(), L"NO SHADER",
 		video::SColor(255,255,255,255), node);
@@ -361,12 +361,12 @@
 	driver->setTextureCreationFlag(video::ETCF_CREATE_MIP_MAPS, false);
 
 	smgr->addSkyBoxSceneNode(
-		driver->getTexture("../../media/irrlicht2_up.jpg"),
-		driver->getTexture("../../media/irrlicht2_dn.jpg"),
-		driver->getTexture("../../media/irrlicht2_lf.jpg"),
-		driver->getTexture("../../media/irrlicht2_rt.jpg"),
-		driver->getTexture("../../media/irrlicht2_ft.jpg"),
-		driver->getTexture("../../media/irrlicht2_bk.jpg"));
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_up.jpg"),
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_dn.jpg"),
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_lf.jpg"),
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_rt.jpg"),
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_ft.jpg"),
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_bk.jpg"));
 
 	driver->setTextureCreationFlag(video::ETCF_CREATE_MIP_MAPS, true);
 
diff -Naur irrlicht-1.4.orig/examples/11.PerPixelLighting/main.cpp irrlicht-1.4/examples/11.PerPixelLighting/main.cpp
--- irrlicht-1.4.orig/examples/11.PerPixelLighting/main.cpp	2007-11-17 07:38:54.000000000 +0200
+++ irrlicht-1.4/examples/11.PerPixelLighting/main.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -36,7 +36,7 @@
 
 		// set a nicer font
 		gui::IGUISkin* skin = env->getSkin();
-		gui::IGUIFont* font = env->getFont("../../media/fonthaettenschweiler.bmp");
+		gui::IGUIFont* font = env->getFont("/usr/share/irrlicht/fonthaettenschweiler.bmp");
 		if (font)
 			skin->setFont(font);
 
@@ -200,7 +200,7 @@
 	driver->setTextureCreationFlag(video::ETCF_ALWAYS_32_BIT, true);
 
 	// add irrlicht logo
-	env->addImage(driver->getTexture("../../media/irrlichtlogo2.png"),
+	env->addImage(driver->getTexture("/usr/share/irrlicht/irrlichtlogo2.png"),
 		core::position2d<s32>(10,10));
 
 	// add camera
@@ -232,7 +232,7 @@
 	*/
 
 	scene::IAnimatedMesh* roomMesh = smgr->getMesh(
-		"../../media/room.3ds");
+		"/usr/share/irrlicht/room.3ds");
 	scene::ISceneNode* room = 0;
 
 	if (roomMesh)
@@ -254,8 +254,8 @@
 		If you set it to a bigger value, the map will look more rocky.
 		*/
 
-		video::ITexture* colorMap = driver->getTexture("../../media/rockwall.bmp");
-		video::ITexture* normalMap = driver->getTexture("../../media/rockwall_height.bmp");
+		video::ITexture* colorMap = driver->getTexture("/usr/share/irrlicht/rockwall.bmp");
+		video::ITexture* normalMap = driver->getTexture("/usr/share/irrlicht/rockwall_height.bmp");
 
 		driver->makeNormalMapTexture(normalMap, 9.0f);
 
@@ -299,7 +299,7 @@
 
 	// add earth sphere
 
-	scene::IAnimatedMesh* earthMesh = smgr->getMesh("../../media/earth.x");
+	scene::IAnimatedMesh* earthMesh = smgr->getMesh("/usr/share/irrlicht/earth.x");
 	if (earthMesh)
 	{
 		//perform various task with the mesh manipulator
@@ -322,7 +322,7 @@
 		sphere->setPosition(core::vector3df(-70,130,45));
 
 		// load heightmap, create normal map from it and set it
-		video::ITexture* earthNormalMap = driver->getTexture("../../media/earthbump.bmp");
+		video::ITexture* earthNormalMap = driver->getTexture("/usr/share/irrlicht/earthbump.bmp");
 		driver->makeNormalMapTexture(earthNormalMap, 20.0f);
 		sphere->setMaterialTexture(1, earthNormalMap);
 
@@ -366,7 +366,7 @@
 	bill->setMaterialFlag(video::EMF_LIGHTING, false);
 	bill->setMaterialFlag(video::EMF_ZWRITE_ENABLE, false);
 	bill->setMaterialType(video::EMT_TRANSPARENT_ADD_COLOR);
-	bill->setMaterialTexture(0, driver->getTexture("../../media/particlered.bmp"));
+	bill->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/particlered.bmp"));
 
 	/*
 	Now the same again, with the second light. The difference is that we add a particle
@@ -395,7 +395,7 @@
 	bill->setMaterialFlag(video::EMF_LIGHTING, false);
 	bill->setMaterialFlag(video::EMF_ZWRITE_ENABLE, false);
 	bill->setMaterialType(video::EMT_TRANSPARENT_ADD_COLOR);
-	bill->setMaterialTexture(0, driver->getTexture("../../media/particlewhite.bmp"));
+	bill->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/particlewhite.bmp"));
 
 	// add particle system
 	scene::IParticleSystemSceneNode* ps =
@@ -421,7 +421,7 @@
 	// adjust some material settings
 	ps->setMaterialFlag(video::EMF_LIGHTING, false);
 	ps->setMaterialFlag(video::EMF_ZWRITE_ENABLE, false);
-	ps->setMaterialTexture(0, driver->getTexture("../../media/fireball.bmp"));
+	ps->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/fireball.bmp"));
 	ps->setMaterialType(video::EMT_TRANSPARENT_VERTEX_ALPHA);
 
 
diff -Naur irrlicht-1.4.orig/examples/12.TerrainRendering/main.cpp irrlicht-1.4/examples/12.TerrainRendering/main.cpp
--- irrlicht-1.4.orig/examples/12.TerrainRendering/main.cpp	2007-11-17 07:38:54.000000000 +0200
+++ irrlicht-1.4/examples/12.TerrainRendering/main.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -110,11 +110,11 @@
 	driver->setTextureCreationFlag(video::ETCF_ALWAYS_32_BIT, true);
 
 	// add irrlicht logo
-	env->addImage(driver->getTexture("../../media/irrlichtlogo2.png"),
+	env->addImage(driver->getTexture("/usr/share/irrlicht/irrlichtlogo2.png"),
 		core::position2d<s32>(10,10));
 
 	//set other font
-	env->getSkin()->setFont(env->getFont("../../media/fontlucida.png"));
+	env->getSkin()->setFont(env->getFont("/usr/share/irrlicht/fontlucida.png"));
 
 	// add some help text
 	gui::IGUIStaticText* text = env->addStaticText(
@@ -148,7 +148,7 @@
 
 	// add terrain scene node
 	scene::ITerrainSceneNode* terrain = smgr->addTerrainSceneNode( 
-		"../../media/terrain-heightmap.bmp",
+		"/usr/share/irrlicht/terrain-heightmap.bmp",
 		0,										// parent node
 		-1,										// node id
 		core::vector3df(0.f, 0.f, 0.f),			// position
@@ -162,8 +162,8 @@
 
 	terrain->setMaterialFlag(video::EMF_LIGHTING, false);
 
-	terrain->setMaterialTexture(0, driver->getTexture("../../media/terrain-texture.jpg"));
-	terrain->setMaterialTexture(1, driver->getTexture("../../media/detailmap3.jpg"));
+	terrain->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/terrain-texture.jpg"));
+	terrain->setMaterialTexture(1, driver->getTexture("/usr/share/irrlicht/detailmap3.jpg"));
 	
 	terrain->setMaterialType(video::EMT_DETAIL_MAP);
 
@@ -207,12 +207,12 @@
 	driver->setTextureCreationFlag(video::ETCF_CREATE_MIP_MAPS, false);
 
 	smgr->addSkyBoxSceneNode(
-		driver->getTexture("../../media/irrlicht2_up.jpg"),
-		driver->getTexture("../../media/irrlicht2_dn.jpg"),
-		driver->getTexture("../../media/irrlicht2_lf.jpg"),
-		driver->getTexture("../../media/irrlicht2_rt.jpg"),
-		driver->getTexture("../../media/irrlicht2_ft.jpg"),
-		driver->getTexture("../../media/irrlicht2_bk.jpg"));
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_up.jpg"),
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_dn.jpg"),
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_lf.jpg"),
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_rt.jpg"),
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_ft.jpg"),
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_bk.jpg"));
 
 	driver->setTextureCreationFlag(video::ETCF_CREATE_MIP_MAPS, true);
 
diff -Naur irrlicht-1.4.orig/examples/13.RenderToTexture/main.cpp irrlicht-1.4/examples/13.RenderToTexture/main.cpp
--- irrlicht-1.4.orig/examples/13.RenderToTexture/main.cpp	2007-11-17 07:38:54.000000000 +0200
+++ irrlicht-1.4/examples/13.RenderToTexture/main.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -63,11 +63,11 @@
 	// load and display animated fairy mesh
 
 	scene::IAnimatedMeshSceneNode* fairy = smgr->addAnimatedMeshSceneNode(
-		smgr->getMesh("../../media/faerie.md2"));
+		smgr->getMesh("/usr/share/irrlicht/faerie.md2"));
 
 	if (fairy)
 	{
-		fairy->setMaterialTexture(0, driver->getTexture("../../media/faerie2.bmp")); // set diffuse texture
+		fairy->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/faerie2.bmp")); // set diffuse texture
 		fairy->setMaterialFlag(video::EMF_LIGHTING, true); // enable dynamic lighting
 		fairy->getMaterial(0).Shininess = 20.0f; // set size of specular highlights
 		fairy->setPosition(core::vector3df(-10,0,-100));
@@ -143,7 +143,7 @@
 	{
 		// create problem text
 		gui::IGUISkin* skin = env->getSkin();
-		gui::IGUIFont* font = env->getFont("../../media/fonthaettenschweiler.bmp");
+		gui::IGUIFont* font = env->getFont("/usr/share/irrlicht/fonthaettenschweiler.bmp");
 		if (font)
 			skin->setFont(font);
 
diff -Naur irrlicht-1.4.orig/examples/14.Win32Window/main.cpp irrlicht-1.4/examples/14.Win32Window/main.cpp
--- irrlicht-1.4.orig/examples/14.Win32Window/main.cpp	2007-11-17 07:38:56.000000000 +0200
+++ irrlicht-1.4/examples/14.Win32Window/main.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -121,18 +121,18 @@
 
 	scene::ISceneNode* cube = smgr->addCubeSceneNode(20);
 
-	cube->setMaterialTexture(0, driver->getTexture("../../media/wall.bmp"));
-	cube->setMaterialTexture(1, driver->getTexture("../../media/water.jpg"));
+	cube->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/wall.bmp"));
+	cube->setMaterialTexture(1, driver->getTexture("/usr/share/irrlicht/water.jpg"));
 	cube->setMaterialFlag( video::EMF_LIGHTING, false );
 	cube->setMaterialType( video::EMT_REFLECTION_2_LAYER );
 
 	smgr->addSkyBoxSceneNode(
-	driver->getTexture("../../media/irrlicht2_up.jpg"),
-	driver->getTexture("../../media/irrlicht2_dn.jpg"),
-	driver->getTexture("../../media/irrlicht2_lf.jpg"),
-	driver->getTexture("../../media/irrlicht2_rt.jpg"),
-	driver->getTexture("../../media/irrlicht2_ft.jpg"),
-	driver->getTexture("../../media/irrlicht2_bk.jpg"));
+	driver->getTexture("/usr/share/irrlicht/irrlicht2_up.jpg"),
+	driver->getTexture("/usr/share/irrlicht/irrlicht2_dn.jpg"),
+	driver->getTexture("/usr/share/irrlicht/irrlicht2_lf.jpg"),
+	driver->getTexture("/usr/share/irrlicht/irrlicht2_rt.jpg"),
+	driver->getTexture("/usr/share/irrlicht/irrlicht2_ft.jpg"),
+	driver->getTexture("/usr/share/irrlicht/irrlicht2_bk.jpg"));
 
 	// show and execute dialog
 
diff -Naur irrlicht-1.4.orig/examples/15.LoadIrrFile/main.cpp irrlicht-1.4/examples/15.LoadIrrFile/main.cpp
--- irrlicht-1.4.orig/examples/15.LoadIrrFile/main.cpp	2007-11-17 07:38:54.000000000 +0200
+++ irrlicht-1.4/examples/15.LoadIrrFile/main.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -65,7 +65,7 @@
 
 	// load the scene
 
-	smgr->loadScene("../../media/example.irr");
+	smgr->loadScene("/usr/share/irrlicht/example.irr");
 
 	/* 
 	That was it already. Now add a camera and draw the scene
diff -Naur irrlicht-1.4.orig/examples/16.Quake3MapShader/main.cpp irrlicht-1.4/examples/16.Quake3MapShader/main.cpp
--- irrlicht-1.4.orig/examples/16.Quake3MapShader/main.cpp	2007-11-17 07:38:54.000000000 +0200
+++ irrlicht-1.4/examples/16.Quake3MapShader/main.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -137,7 +137,7 @@
 	scene::ISceneManager* smgr = device->getSceneManager();
 
 	//! add our private media directory to the file system
-	device->getFileSystem()->addFolderFileArchive("../../media/");
+	device->getFileSystem()->addFolderFileArchive("/usr/share/irrlicht/");
 
 	/*
 	To display the Quake 3 map, we first need to load it. Quake 3 maps
@@ -146,7 +146,7 @@
 	we are able to read from the files in that archive as they would
 	directly be stored on disk.
 	*/
-	device->getFileSystem()->addZipFileArchive("../../media/map-20kdm2.pk3");
+	device->getFileSystem()->addZipFileArchive("/usr/share/irrlicht/map-20kdm2.pk3");
 	//device->getFileSystem()->addFolderFileArchive("/baseq3/");
 	
 	/* 
diff -Naur irrlicht-1.4.orig/examples/Demo/CDemo.cpp irrlicht-1.4/examples/Demo/CDemo.cpp
--- irrlicht-1.4.orig/examples/Demo/CDemo.cpp	2007-11-24 10:39:24.000000000 +0200
+++ irrlicht-1.4/examples/Demo/CDemo.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -48,9 +48,9 @@
 	device = createDevice(driverType,resolution, 32, fullscreen, shadows, vsync, this);
 
 	device->getFileSystem()->addZipFileArchive("irrlicht.dat");
-	device->getFileSystem()->addZipFileArchive("../../media/irrlicht.dat");
+	device->getFileSystem()->addZipFileArchive("/usr/share/irrlicht/irrlicht.dat");
 	device->getFileSystem()->addZipFileArchive("map-20kdm2.pk3");
-	device->getFileSystem()->addZipFileArchive("../../media/map-20kdm2.pk3");
+	device->getFileSystem()->addZipFileArchive("/usr/share/irrlicht/map-20kdm2.pk3");
 
 	video::IVideoDriver* driver = device->getVideoDriver();
 	scene::ISceneManager* smgr = device->getSceneManager();
@@ -399,13 +399,13 @@
 	// load sydney model and create 2 instances
 
 	scene::IAnimatedMesh* mesh = 0;
-	mesh = sm->getMesh("../../media/sydney.md2");
+	mesh = sm->getMesh("/usr/share/irrlicht/sydney.md2");
 	if (mesh)
 	{
 		model1 = sm->addAnimatedMeshSceneNode(mesh);
 		if (model1)
 		{
-			model1->setMaterialTexture(0, driver->getTexture("../../media/spheremap.jpg"));
+			model1->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/spheremap.jpg"));
 			model1->setPosition(core::vector3df(100,40,-80));
 			model1->setScale(core::vector3df(2,2,2));
 			model1->setMD2Animation(scene::EMAT_STAND);
@@ -421,7 +421,7 @@
 			model2->setPosition(core::vector3df(180,15,-60));
 			model2->setScale(core::vector3df(2,2,2));
 			model2->setMD2Animation(scene::EMAT_RUN);
-			model2->setMaterialTexture(0, device->getVideoDriver()->getTexture("../../media/sydney.bmp"));
+			model2->setMaterialTexture(0, device->getVideoDriver()->getTexture("/usr/share/irrlicht/sydney.bmp"));
 			model2->setMaterialFlag(video::EMF_LIGHTING, true);
 			model2->addShadowVolumeSceneNode();
 			model2->setAutomaticCulling ( scene::EAC_BOX );
@@ -433,12 +433,12 @@
 	// create sky box
 	driver->setTextureCreationFlag(video::ETCF_CREATE_MIP_MAPS, false);
 	skyboxNode = sm->addSkyBoxSceneNode(
-		driver->getTexture("../../media/irrlicht2_up.jpg"),
-		driver->getTexture("../../media/irrlicht2_dn.jpg"),
-		driver->getTexture("../../media/irrlicht2_lf.jpg"),
-		driver->getTexture("../../media/irrlicht2_rt.jpg"),
-		driver->getTexture("../../media/irrlicht2_ft.jpg"),
-		driver->getTexture("../../media/irrlicht2_bk.jpg"));
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_up.jpg"),
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_dn.jpg"),
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_lf.jpg"),
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_rt.jpg"),
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_ft.jpg"),
+		driver->getTexture("/usr/share/irrlicht/irrlicht2_bk.jpg"));
 	driver->setTextureCreationFlag(video::ETCF_CREATE_MIP_MAPS, true);
 
 	//driver->setTextureCreationFlag(video::ETCF_CREATE_MIP_MAPS, true);
@@ -463,7 +463,7 @@
 	for (s32 g=1; g<8; ++g)
 	{
 		char tmp[64];
-		sprintf(tmp, "../../media/portal%d.bmp", g);
+		sprintf(tmp, "/usr/share/irrlicht/portal%d.bmp", g);
 		video::ITexture* t = driver->getTexture(tmp);
 		textures.push_back(t);
 	}
@@ -479,7 +479,7 @@
 		bill = sm->addBillboardSceneNode(0, core::dimension2d<f32>(100,100),
 			waypoint[r]+ core::vector3df(0,20,0));
 		bill->setMaterialFlag(video::EMF_LIGHTING, false);
-		bill->setMaterialTexture(0, driver->getTexture("../../media/portal1.bmp"));
+		bill->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/portal1.bmp"));
 		bill->setMaterialType(video::EMT_TRANSPARENT_ADD_COLOR);
 		bill->addAnimator(anim);
 	}
@@ -502,7 +502,7 @@
 	bill = device->getSceneManager()->addBillboardSceneNode(
 		light, core::dimension2d<f32>(40,40));
 	bill->setMaterialFlag(video::EMF_LIGHTING, false);
-	bill->setMaterialTexture(0, driver->getTexture("../../media/particlewhite.bmp"));
+	bill->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/particlewhite.bmp"));
 	bill->setMaterialType(video::EMT_TRANSPARENT_ADD_COLOR);
 
 	// create meta triangle selector with all triangles selectors in it.
@@ -530,7 +530,7 @@
 	paf->drop();
 
 	campFire->setMaterialFlag(video::EMF_LIGHTING, false);
-	campFire->setMaterialTexture(0, driver->getTexture("../../media/fireball.bmp"));
+	campFire->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/fireball.bmp"));
 	campFire->setMaterialType(video::EMT_TRANSPARENT_VERTEX_ALPHA);
 
 	// load music
@@ -564,7 +564,7 @@
 	inOutFader->setColor(backColor,	video::SColor ( 0, 230, 230, 230 ));
 
 	// irrlicht logo
-	device->getGUIEnvironment()->addImage(device->getVideoDriver()->getTexture("../../media/irrlichtlogo2.png"),
+	device->getGUIEnvironment()->addImage(device->getVideoDriver()->getTexture("/usr/share/irrlicht/irrlichtlogo2.png"),
 		core::position2d<s32>(5,5));
 
 	// loading text
@@ -581,7 +581,7 @@
 	// load bigger font
 
 	device->getGUIEnvironment()->getSkin()->setFont(
-		device->getGUIEnvironment()->getFont("../../media/fonthaettenschweiler.bmp"));
+		device->getGUIEnvironment()->getFont("/usr/share/irrlicht/fonthaettenschweiler.bmp"));
 
 	// set new font color
 
@@ -643,7 +643,7 @@
 		core::dimension2d<f32>(25,25), start);
 
 	node->setMaterialFlag(video::EMF_LIGHTING, false);
-	node->setMaterialTexture(0, device->getVideoDriver()->getTexture("../../media/fireball.bmp"));
+	node->setMaterialTexture(0, device->getVideoDriver()->getTexture("/usr/share/irrlicht/fireball.bmp"));
 	node->setMaterialType(video::EMT_TRANSPARENT_ADD_COLOR);
 
 	f32 length = (f32)(end - start).getLength();
@@ -710,7 +710,7 @@
 			paf->drop();
 
 			pas->setMaterialFlag(video::EMF_LIGHTING, false);
-			pas->setMaterialTexture(0, device->getVideoDriver()->getTexture("../../media/smoke.bmp"));
+			pas->setMaterialTexture(0, device->getVideoDriver()->getTexture("/usr/share/irrlicht/smoke.bmp"));
 			pas->setMaterialType(video::EMT_TRANSPARENT_VERTEX_ALPHA);
 
 			scene::ISceneNodeAnimator* anim = sm->createDeleteAnimator(2000);
@@ -756,7 +756,7 @@
 
 	// play music
 
-	irrklang::ISound* snd = irrKlang->play2D("../../media/IrrlichtTheme.ogg", true, false, true);
+	irrklang::ISound* snd = irrKlang->play2D("/usr/share/irrlicht/IrrlichtTheme.ogg", true, false, true);
 	if ( !snd )
 		snd = irrKlang->play2D("IrrlichtTheme.ogg", true, false, true);
 
@@ -768,8 +768,8 @@
 
 	// preload both sound effects
 
-	ballSound = irrKlang->getSoundSource("../../media/ball.wav");
-	impactSound = irrKlang->getSoundSource("../../media/impact.wav");
+	ballSound = irrKlang->getSoundSource("/usr/share/irrlicht/ball.wav");
+	impactSound = irrKlang->getSoundSource("/usr/share/irrlicht/impact.wav");
 }
 #endif
 
@@ -786,12 +786,12 @@
 	if (Mix_OpenAudio(22050, AUDIO_S16, 2, 128))
 		return;
 
-	stream = Mix_LoadMUS("../../media/IrrlichtTheme.ogg");
+	stream = Mix_LoadMUS("/usr/share/irrlicht/IrrlichtTheme.ogg");
 	if (stream)
 		Mix_PlayMusic(stream, -1);
 
-	ballSound = Mix_LoadWAV("../../media/ball.wav");
-	impactSound = Mix_LoadWAV("../../media/impact.wav");
+	ballSound = Mix_LoadWAV("/usr/share/irrlicht/ball.wav");
+	impactSound = Mix_LoadWAV("/usr/share/irrlicht/impact.wav");
 }
 
 void CDemo::playSound(Mix_Chunk *sample)
diff -Naur irrlicht-1.4.orig/examples/Demo/CMainMenu.cpp irrlicht-1.4/examples/Demo/CMainMenu.cpp
--- irrlicht-1.4.orig/examples/Demo/CMainMenu.cpp	2007-11-17 07:38:54.000000000 +0200
+++ irrlicht-1.4/examples/Demo/CMainMenu.cpp	2007-12-08 14:42:09.000000000 +0200
@@ -82,7 +82,7 @@
 		core::dimension2d<s32>(512, 384), 16, false, false, false, this);
 
 	device->getFileSystem()->addZipFileArchive("irrlicht.dat");
-	device->getFileSystem()->addZipFileArchive("../../media/irrlicht.dat");
+	device->getFileSystem()->addZipFileArchive("/usr/share/irrlicht/irrlicht.dat");
 
 	video::IVideoDriver* driver = device->getVideoDriver();
 	scene::ISceneManager* smgr = device->getSceneManager();
@@ -98,7 +98,7 @@
 	newskin->drop();
 
 	// load font
-	gui::IGUIFont* font = guienv->getFont("../../media/fonthaettenschweiler.bmp");
+	gui::IGUIFont* font = guienv->getFont("/usr/share/irrlicht/fonthaettenschweiler.bmp");
 	if (font)
 		guienv->getSkin()->setFont(font);
 
@@ -167,12 +167,12 @@
 
 	// add md2 model
 
-	scene::IAnimatedMesh* mesh = smgr->getMesh("../../media/faerie.md2");
+	scene::IAnimatedMesh* mesh = smgr->getMesh("/usr/share/irrlicht/faerie.md2");
 	scene::IAnimatedMeshSceneNode* modelNode = smgr->addAnimatedMeshSceneNode(mesh);
 	if (modelNode)
 	{
 		modelNode->setPosition ( core::vector3df ( 0.f, 0.f, -5.f ) );
-		modelNode->setMaterialTexture(0, driver->getTexture("../../media/faerie2.bmp"));
+		modelNode->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/faerie2.bmp"));
 		modelNode->setMaterialFlag(video::EMF_LIGHTING, true);
 		modelNode->getMaterial(0).Shininess = 28.f;
 		modelNode->getMaterial(0).NormalizeNormals = true;
@@ -206,7 +206,7 @@
 	bill = smgr->addBillboardSceneNode(light1, core::dimension2d<f32>(10, 10));
 	bill->setMaterialFlag(video::EMF_LIGHTING, false);
 	bill->setMaterialType(video::EMT_TRANSPARENT_ADD_COLOR);
-	bill->setMaterialTexture(0, driver->getTexture("../../media/particlered.bmp"));
+	bill->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/particlered.bmp"));
 
 #if 1
 	// add light 2 (nearly red)
@@ -229,7 +229,7 @@
 	bill = smgr->addBillboardSceneNode(light2, core::dimension2d<f32>(10, 10));
 	bill->setMaterialFlag(video::EMF_LIGHTING, false);
 	bill->setMaterialType(video::EMT_TRANSPARENT_ADD_COLOR);
-	bill->setMaterialTexture(0, driver->getTexture("../../media/particlered.bmp"));
+	bill->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/particlered.bmp"));
 
 	// add light 3 (nearly blue)
 	scene::ILightSceneNode* light3 =
@@ -251,7 +251,7 @@
 
 	bill->setMaterialFlag(video::EMF_LIGHTING, false);
 	bill->setMaterialType(video::EMT_TRANSPARENT_ADD_COLOR);
-	bill->setMaterialTexture(0, driver->getTexture("../../media/portal1.bmp"));
+	bill->setMaterialTexture(0, driver->getTexture("/usr/share/irrlicht/portal1.bmp"));
 #endif
 
 	// create a fixed camera
@@ -262,10 +262,10 @@
 	bool oldMipMapState = driver->getTextureCreationFlag(video::ETCF_CREATE_MIP_MAPS);
 	driver->setTextureCreationFlag(video::ETCF_CREATE_MIP_MAPS, false);
 
-	guienv->addImage(driver->getTexture("../../media/irrlichtlogo2.png"),
+	guienv->addImage(driver->getTexture("/usr/share/irrlicht/irrlichtlogo2.png"),
 		core::position2d<s32>(5,5));
 
-	video::ITexture* irrlichtBack = driver->getTexture("../../media/demoback.jpg");
+	video::ITexture* irrlichtBack = driver->getTexture("/usr/share/irrlicht/demoback.jpg");
 
 	driver->setTextureCreationFlag(video::ETCF_CREATE_MIP_MAPS, oldMipMapState);
 
