diff -p -up mono-2.0/mono/profiler/Makefile.am.orig mono-2.0/mono/profiler/Makefile.am
--- mono-2.0/mono/profiler/Makefile.am.orig	2008-09-02 17:42:42.000000000 +0200
+++ mono-2.0/mono/profiler/Makefile.am	2008-10-06 09:51:40.000000000 +0200
@@ -17,9 +17,9 @@ if HAVE_OPROFILE
 endif
 
 libmono_profiler_cov_la_SOURCES = mono-cov.c
-libmono_profiler_cov_la_LIBADD = $(top_builddir)/mono/mini/libmono.la
+libmono_profiler_cov_la_LIBADD = $(top_builddir)/mono/mini/libmono.la $(GLIB_LIBS)
 libmono_profiler_aot_la_SOURCES = mono-profiler-aot.c
-libmono_profiler_aot_la_LIBADD = $(top_builddir)/mono/mini/libmono.la
+libmono_profiler_aot_la_LIBADD = $(top_builddir)/mono/mini/libmono.la $(GLIB_LIBS)
 libmono_profiler_logging_la_SOURCES = mono-profiler-logging.c
-libmono_profiler_logging_la_LIBADD = $(top_builddir)/mono/mini/libmono.la
+libmono_profiler_logging_la_LIBADD = $(top_builddir)/mono/mini/libmono.la $(GLIB_LIBS)
 
