Index: ffmpeg.c
===================================================================
--- ffmpeg.c	(revision 7378)
+++ ffmpeg.c	(working copy)
@@ -3811,13 +3811,13 @@
 static void show_banner(void)
 {
     fprintf(stderr, "FFmpeg version " FFMPEG_VERSION ", Copyright (c) 2000-2007 Fabrice Bellard, et al.\n");
-    fprintf(stderr, "  configuration: " FFMPEG_CONFIGURATION "\n");
-    fprintf(stderr, "  libavutil version: " AV_STRINGIFY(LIBAVUTIL_VERSION) "\n");
-    fprintf(stderr, "  libavcodec version: " AV_STRINGIFY(LIBAVCODEC_VERSION) "\n");
-    fprintf(stderr, "  libavformat version: " AV_STRINGIFY(LIBAVFORMAT_VERSION) "\n");
-    fprintf(stderr, "  built on " __DATE__ " " __TIME__);
+    //fprintf(stderr, "  configuration: " FFMPEG_CONFIGURATION "\n");
+    //fprintf(stderr, "  libavutil version: " AV_STRINGIFY(LIBAVUTIL_VERSION) "\n");
+    //fprintf(stderr, "  libavcodec version: " AV_STRINGIFY(LIBAVCODEC_VERSION) "\n");
+    //fprintf(stderr, "  libavformat version: " AV_STRINGIFY(LIBAVFORMAT_VERSION) "\n");
+    //fprintf(stderr, "  built on " __DATE__ " " __TIME__);
 #ifdef __GNUC__
-    fprintf(stderr, ", gcc: " __VERSION__ "\n");
+    //fprintf(stderr, ", gcc: " __VERSION__ "\n");
 #else
     fprintf(stderr, ", using a non-gcc compiler\n");
 #endif
