diff -Nur warsow_0.41_src-old/source/qcommon/files.c warsow_0.41_src/source/qcommon/files.c
--- warsow_0.41_src-old/source/qcommon/files.c	2008-02-16 01:55:32.863761266 +0200
+++ warsow_0.41_src/source/qcommon/files.c	2008-02-16 01:56:56.370066378 +0200
@@ -3225,7 +3225,7 @@
 	// set basepaths
 	//
 	fs_cdpath = Cvar_Get( "fs_cdpath", "", CVAR_NOSET );
-	fs_basepath = Cvar_Get( "fs_basepath", ".", CVAR_NOSET );
+	fs_basepath = Cvar_Get( "fs_basepath", "/usr/share/warsow", CVAR_NOSET );
 	if( Sys_FS_GetHomeDirectory() != NULL )
 		fs_usehomedir = Cvar_Get( "fs_usehomedir", "1", CVAR_NOSET );
 
