diff -ur attr-2.4.38.orig/include/builddefs.in attr-2.4.38/include/builddefs.in
--- attr-2.4.38.orig/include/builddefs.in	2007-01-25 06:32:46.000000000 +0200
+++ attr-2.4.38/include/builddefs.in	2007-07-15 18:06:37.000000000 +0300
@@ -9,6 +9,7 @@
 
 DEBUG = @debug_build@
 OPTIMIZER = @opt_build@
+LDFLAGS_OPT = @LDFLAGS@
 MALLOCLIB = @malloc_lib@
 LOADERFLAGS = @LDFLAGS@
 
diff -ur attr-2.4.38.orig/include/buildmacros attr-2.4.38/include/buildmacros
--- attr-2.4.38.orig/include/buildmacros	2007-07-15 18:05:35.000000000 +0300
+++ attr-2.4.38/include/buildmacros	2007-07-15 18:06:16.000000000 +0300
@@ -10,6 +10,8 @@
 # during the "dist" phase (packaging).
 
 LDFLAGS += $(LOADERFLAGS) $(LLDFLAGS)
+LDFLAGS += $(LDFLAGS_OPT) $(LLDFLAGS)
+LTLDFLAGS += $(LDFLAGS_OPT)
 LDLIBS = $(LLDLIBS) $(PLDLIBS) $(MALLOCLIB)
 
 MAKEOPTS = --no-print-directory
