diff -Nuar pyclutter-gtk-0.10.0.orig//configure.ac pyclutter-gtk-0.10.0/configure.ac
--- pyclutter-gtk-0.10.0.orig//configure.ac	2010-01-26 19:57:00.000000000 +0200
+++ pyclutter-gtk-0.10.0/configure.ac	2011-04-28 17:31:39.314000043 +0300
@@ -9,7 +9,7 @@
 m4_define(pycluttergtk_version, pycluttergtk_version_major.pycluttergtk_version_minor.pycluttergtk_version_micro)
 
 m4_define(clutter_required_version,      1.0.0)
-m4_define(cluttergtk_required_version,   0.10.2)
+m4_define(cluttergtk_required_version,   0.11.4)
 m4_define(pygtk_required_version,        2.8.0)
 m4_define(pyclutter_required_version,    1.0.0)
 
@@ -102,7 +102,7 @@
 )
 
 dnl check for clutter-gtk
-PKG_CHECK_MODULES(CLUTTERGTK, clutter-gtk-0.10 >= cluttergtk_required_version)
+PKG_CHECK_MODULES(CLUTTERGTK, clutter-gtk-0.12 >= cluttergtk_required_version)
 AC_SUBST(CLUTTERGTK_CFLAGS)
 AC_SUBST(CLUTTERGTK_LIBS)
 AS_IF([test -n "$export_dynamic"],
