Index: kid3-1.0/kid3/scriptinterface.h
===================================================================
--- kid3-1.0.orig/kid3/scriptinterface.h
+++ kid3-1.0/kid3/scriptinterface.h
@@ -210,13 +210,6 @@ public slots:
 	void convertToId3v24();
 #endif
 
-#if defined HAVE_TAGLIB && defined HAVE_ID3LIB
-	/**
-	 * Convert ID3v2.4 tags to ID3v2.3.
-	 */
-	void convertToId3v23();
-#endif
-
 	/**
 	 * Get path of directory.
 	 *
