diff -Nur ufoai-2.2-source-old/src/common/files.c ufoai-2.2-source/src/common/files.c
--- ufoai-2.2-source-old/src/common/files.c	2008-01-15 02:48:38.946036250 +0200
+++ ufoai-2.2-source/src/common/files.c	2008-01-15 02:50:47.330518842 +0200
@@ -889,7 +889,7 @@
 
 	/* basedir <path> */
 	/* allows the game to run from outside the data tree */
-	fs_basedir = Cvar_Get("fs_basedir", ".", CVAR_NOSET, "Allows the game to run from outside the data tree");
+	fs_basedir = Cvar_Get("fs_basedir", "/usr/share/ufoai", CVAR_NOSET, "Allows the game to run from outside the data tree");
 
 	/* start up with base by default */
 	FS_AddGameDirectory(va("%s/" BASEDIRNAME, fs_basedir->string));
