diff -Nur ufoai-2.2-source-old/src/client/cl_language.c ufoai-2.2-source/src/client/cl_language.c
--- ufoai-2.2-source-old/src/client/cl_language.c	2008-01-15 02:48:38.942036858 +0200
+++ ufoai-2.2-source/src/client/cl_language.c	2008-01-15 02:56:13.800887802 +0200
@@ -133,7 +133,7 @@
 	if (*fs_i18ndir->string)
 		Q_strncpyz(languagePath, fs_i18ndir->string, sizeof(languagePath));
 	else
-		Com_sprintf(languagePath, sizeof(languagePath), "%s/"BASEDIRNAME"/i18n/", FS_GetCwd());
+		Com_sprintf(languagePath, sizeof(languagePath), "%s/"BASEDIRNAME"/i18n/", "/usr/share/ufoai");
 	Com_DPrintf(DEBUG_CLIENT, "CL_LanguageTest()... using mo files from '%s'\n", languagePath);
 	Q_strcat(languagePath, localeID, sizeof(languagePath));
 	Q_strcat(languagePath, "/LC_MESSAGES/ufoai.mo", sizeof(languagePath));
diff -Nur ufoai-2.2-source-old/src/client/cl_main.c ufoai-2.2-source/src/client/cl_main.c
--- ufoai-2.2-source-old/src/client/cl_main.c	2008-01-15 02:48:38.939037314 +0200
+++ ufoai-2.2-source/src/client/cl_main.c	2008-01-15 02:55:56.523514362 +0200
@@ -2497,7 +2497,7 @@
 	if (*fs_i18ndir->string)
 		Q_strncpyz(languagePath, fs_i18ndir->string, sizeof(languagePath));
 	else
-		Com_sprintf(languagePath, sizeof(languagePath), "%s/"BASEDIRNAME"/i18n/", FS_GetCwd());
+		Com_sprintf(languagePath, sizeof(languagePath), "%s/"BASEDIRNAME"/i18n/", "/usr/share/ufoai");
 	Com_DPrintf(DEBUG_CLIENT, "...using mo files from %s\n", languagePath);
 	bindtextdomain(TEXT_DOMAIN, languagePath);
 	bind_textdomain_codeset(TEXT_DOMAIN, "UTF-8");
