diff -Nur ufoai-2.2-source-old/src/ports/unix/unix_main.c ufoai-2.2-source/src/ports/unix/unix_main.c
--- ufoai-2.2-source-old/src/ports/unix/unix_main.c	2008-01-15 02:48:38.947036098 +0200
+++ ufoai-2.2-source/src/ports/unix/unix_main.c	2008-01-15 02:52:02.279124922 +0200
@@ -290,7 +290,7 @@
 	char libName[MAX_OSPATH];
 	char libDir[MAX_OSPATH];
 
-	s_libdir = Cvar_Get("s_libdir", "", CVAR_ARCHIVE, "Library dir for graphic, sound and game libraries");
+	s_libdir = Cvar_Get("s_libdir", "/usr/share/ufoai", CVAR_ARCHIVE, "Library dir for graphic, sound and game libraries");
 
 	/* try path given via cvar */
 	if (*s_libdir->string) {
