--- setup.py	2007-12-04 11:29:39.000000000 +0200
+++ setup.py	2007-12-04 11:31:55.000000000 +0200
@@ -58,12 +58,6 @@
 if not os.path.exists('build/sage'):
     os.makedirs('build/sage')
 
-sage_link = SITE_PACKAGES + '/sage'
-if not os.path.islink(sage_link) or not os.path.exists(sage_link):
-    os.system('rm -rf "%s"'%sage_link)
-    os.system('cd %s; ln -sf ../../../../devel/sage/build/sage .'%SITE_PACKAGES)
-
-
 def is_older(file1, file2):
     """
     Return True if either file2 does not exist or is older than file1.
