diff -up stellarium-0.10.0/data/CMakeLists.txt.fp stellarium-0.10.0/data/CMakeLists.txt
--- stellarium-0.10.0/data/CMakeLists.txt.fp	2009-01-04 19:20:51.000000000 +0100
+++ stellarium-0.10.0/data/CMakeLists.txt	2009-01-04 19:21:25.000000000 +0100
@@ -1,6 +1,6 @@
 ########### install files ###############
 
-INSTALL(FILES stellarium.ico ssystem.ini zone.tab base_locations.txt DejaVuSans.ttf default_config.ini DejaVuSansMono.ttf constellations_boundaries.dat fontmap.dat iso639-1.utf8 countryCodes.dat DESTINATION share/${PACKAGE}/data)
+INSTALL(FILES stellarium.ico ssystem.ini zone.tab base_locations.txt default_config.ini constellations_boundaries.dat fontmap.dat iso639-1.utf8 countryCodes.dat DESTINATION share/${PACKAGE}/data)
 
 IF(MACOSX)
 INSTALL(FILES Icon.icns DESTINATION share )
 
