Index: konversation/scripts/media
===================================================================
--- konversation/scripts/media.orig
+++ konversation/scripts/media
@@ -48,7 +48,7 @@ formatVariables={'audio': _('is listenin
 ## If you add a new player, you must add it here or it won't get checked when in audio-only or video-only modes.
 playerRankings= {
     'video' :['kaffeine','kmplayer', 'kplayer', 'noatun', 'kdetv'],
-    'audio' :['amarok', 'MPD' 'juk', 'noatun', 'kscd', 'kaffeine', 'kmplayer', 'Audacious', 'xmms', 'yammi']
+    'audio' :['amarok', 'MPD' 'juk', 'noatun', 'kscd', 'kaffeine', 'kmplayer', 'Audacious', 'xmms', 'yammi', 'LastFM']
 }
 
 ## Title, album and artist fields to be quoted depending on contents
@@ -381,6 +381,16 @@ class AudaciousPlayer(Player):
             return ''
 
 
+class LastFMPlayer(Player):
+    def __init__(self):
+        Player.__init__(self,"last.fm")
+
+    def getData(self):
+        title = subprocess.Popen(["qdbus","com.lastfm.DBusInterface","/","currentTrack"], stdout=subprocess.PIPE).communicate()[0].rstrip()
+        artist = subprocess.Popen(["qdbus","com.lastfm.DBusInterface","/","currentArtist"], stdout=subprocess.PIPE).communicate()[0].rstrip()
+        album = subprocess.Popen(["qdbus","com.lastfm.DBusInterface","/","currentAlbum"], stdout=subprocess.PIPE).communicate()[0].rstrip()
+        return (title,artist,album)
+
 def playing(playerList, mode=None):
     for i in playerList:
         s=i.get(mode)
@@ -445,7 +455,8 @@ DCOPPlayer("KsCD","kscd","CDPlayer curre
 DCOPPlayer("kdetv","kdetv","KdetvIface channelName",playerType='video'),
 AudaciousPlayer('Audacious'), XmmsPlayer('XMMS'),
 DCOPPlayer("Yammi","yammi","YammiPlayer songTitle","YammiPlayer songArtist","YammiPlayer songAlbum"),
-MPD('MPD')
+MPD('MPD'),
+LastFMPlayer()
 ]
 
 # Get rid of players that didn't get subclassed so they don't appear in the available players list
