Index: yokadi-0.13.0/setup.py
===================================================================
--- yokadi-0.13.0.orig/setup.py
+++ yokadi-0.13.0/setup.py
@@ -17,11 +17,11 @@ root=abspath(dirname(__file__))
 
 # Additional files
 data_files=[]
-data_files.append(["share/yokadi",
+data_files.append(["share/doc/yokadi",
                    ["version", "README.markdown", "NEWS", "LICENSE"]])
 
 # Doc
-data_files.append(["share/yokadi/doc",
+data_files.append(["share/doc/yokadi",
                    ["doc/%s" % f for f in os.listdir(join(root, "doc"))]])
 
 # Man
@@ -29,7 +29,7 @@ data_files.append(["share/man/man1",
                    ["man/%s" % f for f in os.listdir(join(root, "man"))]])
 
 # Update scripts
-data_files.append(["share/yokadi/update",
+data_files.append(["share/doc/yokadi/update",
                    ["update/%s" % f for f in os.listdir(join(root, "update"))]])
 
 # Scripts
