Index: kdebase-workspace-4.4.5/plasma/generic/dataengines/CMakeLists.txt
===================================================================
--- kdebase-workspace-4.4.5.orig/plasma/generic/dataengines/CMakeLists.txt
+++ kdebase-workspace-4.4.5/plasma/generic/dataengines/CMakeLists.txt
@@ -4,7 +4,7 @@ add_subdirectory(dict)
 add_subdirectory(executable)
 add_subdirectory(favicons)
 add_subdirectory(filebrowser)
-add_subdirectory(geolocation)
+#add_subdirectory(geolocation)
 add_subdirectory(hotplug)
 add_subdirectory(keystate)
 add_subdirectory(network)
