diff -Nur ode-0.9-old/ode/demo/demo_basket.cpp ode-0.9/ode/demo/demo_basket.cpp
--- ode-0.9-old/ode/demo/demo_basket.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_basket.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -210,7 +210,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     fn.path_to_textures = argv[1];
 
diff -Nur ode-0.9-old/ode/demo/demo_boxstack.cpp ode-0.9/ode/demo/demo_boxstack.cpp
--- ode-0.9-old/ode/demo/demo_boxstack.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_boxstack.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -539,7 +539,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_buggy.cpp ode-0.9/ode/demo/demo_buggy.cpp
--- ode-0.9-old/ode/demo/demo_buggy.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_buggy.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -219,7 +219,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_chain1.c ode-0.9/ode/demo/demo_chain1.c
--- ode-0.9-old/ode/demo/demo_chain1.c	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_chain1.c	2008-04-20 23:35:55.000000000 +0300
@@ -129,7 +129,7 @@
   fn.step = &simLoop;
   fn.command = 0;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_chain2.cpp ode-0.9/ode/demo/demo_chain2.cpp
--- ode-0.9-old/ode/demo/demo_chain2.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_chain2.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -122,7 +122,7 @@
   fn.step = &simLoop;
   fn.command = 0;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_collision.cpp ode-0.9/ode/demo/demo_collision.cpp
--- ode-0.9-old/ode/demo/demo_collision.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_collision.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -1284,7 +1284,7 @@
     fn.step = &simLoop;
     fn.command = &command;
     fn.stop = 0;
-    fn.path_to_textures = "../../drawstuff/textures";
+    fn.path_to_textures = "/usr/share/ode/examples";
 
     dsSetSphereQuality (3);
     dsSetCapsuleQuality (8);
diff -Nur ode-0.9-old/ode/demo/demo_convex_cd.cpp ode-0.9/ode/demo/demo_convex_cd.cpp
--- ode-0.9-old/ode/demo/demo_convex_cd.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_convex_cd.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -180,7 +180,7 @@
   fn.step = &simLoop;
   fn.command = command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";	// uses default
+  fn.path_to_textures = "/usr/share/ode/examples";	// uses default
   world = dWorldCreate();
   space = dHashSpaceCreate (0);
   contactgroup = dJointGroupCreate (0);
diff -Nur ode-0.9-old/ode/demo/demo_crash.cpp ode-0.9/ode/demo/demo_crash.cpp
--- ode-0.9-old/ode/demo/demo_crash.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_crash.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -609,7 +609,7 @@
 	fn.step = &simLoop;
 	fn.command = &command;
 	fn.stop = 0;
-	fn.path_to_textures = "../../drawstuff/textures";
+	fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_cyl.cpp ode-0.9/ode/demo/demo_cyl.cpp
--- ode-0.9-old/ode/demo/demo_cyl.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_cyl.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -230,7 +230,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_cylvssphere.cpp ode-0.9/ode/demo/demo_cylvssphere.cpp
--- ode-0.9-old/ode/demo/demo_cylvssphere.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_cylvssphere.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -183,7 +183,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     fn.path_to_textures = argv[1];
 
diff -Nur ode-0.9-old/ode/demo/demo_feedback.cpp ode-0.9/ode/demo/demo_feedback.cpp
--- ode-0.9-old/ode/demo/demo_feedback.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_feedback.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -224,7 +224,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
   {
     fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_friction.cpp ode-0.9/ode/demo/demo_friction.cpp
--- ode-0.9-old/ode/demo/demo_friction.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_friction.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -167,7 +167,7 @@
   fn.step = &simLoop;
   fn.command = 0;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_heightfield.cpp ode-0.9/ode/demo/demo_heightfield.cpp
--- ode-0.9-old/ode/demo/demo_heightfield.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_heightfield.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -2061,7 +2061,7 @@
 	fn.step = &simLoop;
 	fn.command = &command;
 	fn.stop = 0;
-	fn.path_to_textures = "../../drawstuff/textures";
+	fn.path_to_textures = "/usr/share/ode/examples";
 	if(argc==2)
 	{
 		fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_hinge.cpp ode-0.9/ode/demo/demo_hinge.cpp
--- ode-0.9-old/ode/demo/demo_hinge.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_hinge.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -125,7 +125,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_I.cpp ode-0.9/ode/demo/demo_I.cpp
--- ode-0.9-old/ode/demo/demo_I.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_I.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -235,7 +235,7 @@
   fn.step = &simLoop;
   fn.command = 0;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_jointPR.cpp ode-0.9/ode/demo/demo_jointPR.cpp
--- ode-0.9-old/ode/demo/demo_jointPR.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_jointPR.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -35,7 +35,7 @@
 #include <math.h>
 
 
-#define DRAWSTUFF_TEXTURE_PATH "../../drawstuff/textures"
+#define DRAWSTUFF_TEXTURE_PATH "/usr/share/ode/examples"
 
 
 #ifdef _MSC_VER
diff -Nur ode-0.9-old/ode/demo/demo_joints.cpp ode-0.9/ode/demo/demo_joints.cpp
--- ode-0.9-old/ode/demo/demo_joints.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_joints.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -1032,7 +1032,7 @@
   if (cmd_path_to_textures)
     fn.path_to_textures = cmd_path_to_textures;
   else
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
 
   // run simulation
   if (cmd_graphics) {
diff -Nur ode-0.9-old/ode/demo/demo_motor.cpp ode-0.9/ode/demo/demo_motor.cpp
--- ode-0.9-old/ode/demo/demo_motor.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_motor.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -140,7 +140,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc>=2)
   {
     fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_moving_trimesh.cpp ode-0.9/ode/demo/demo_moving_trimesh.cpp
--- ode-0.9-old/ode/demo/demo_moving_trimesh.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_moving_trimesh.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -1895,7 +1895,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_plane2d.cpp ode-0.9/ode/demo/demo_plane2d.cpp
--- ode-0.9-old/ode/demo/demo_plane2d.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_plane2d.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -258,7 +258,7 @@
         drawstuff_functions.step = &cb_sim_step;
         drawstuff_functions.command = 0;
         drawstuff_functions.stop = 0;
-        drawstuff_functions.path_to_textures = "../../drawstuff/textures";
+        drawstuff_functions.path_to_textures = "/usr/share/ode/examples";
 
         dsSimulationLoop (argc, argv, 352,288,&drawstuff_functions);
     }
diff -Nur ode-0.9-old/ode/demo/demo_slider.cpp ode-0.9/ode/demo/demo_slider.cpp
--- ode-0.9-old/ode/demo/demo_slider.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_slider.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -136,7 +136,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_space.cpp ode-0.9/ode/demo/demo_space.cpp
--- ode-0.9-old/ode/demo/demo_space.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_space.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -201,7 +201,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_space_stress.cpp ode-0.9/ode/demo/demo_space_stress.cpp
--- ode-0.9-old/ode/demo/demo_space_stress.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_space_stress.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -396,7 +396,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_step.cpp ode-0.9/ode/demo/demo_step.cpp
--- ode-0.9-old/ode/demo/demo_step.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_step.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -174,7 +174,7 @@
   fn.step = &simLoop;
   fn.command = 0;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
diff -Nur ode-0.9-old/ode/demo/demo_trimesh.cpp ode-0.9/ode/demo/demo_trimesh.cpp
--- ode-0.9-old/ode/demo/demo_trimesh.cpp	2008-04-20 23:34:29.000000000 +0300
+++ ode-0.9/ode/demo/demo_trimesh.cpp	2008-04-20 23:35:55.000000000 +0300
@@ -446,7 +446,7 @@
   fn.step = &simLoop;
   fn.command = &command;
   fn.stop = 0;
-  fn.path_to_textures = "../../drawstuff/textures";
+  fn.path_to_textures = "/usr/share/ode/examples";
   if(argc==2)
     {
         fn.path_to_textures = argv[1];
