Index: opencv-1.0.0/Makefile.am
===================================================================
--- opencv-1.0.0.orig/Makefile.am
+++ opencv-1.0.0/Makefile.am
@@ -47,11 +47,6 @@ changed:
 install-lib:
 	cd src && $(MAKE) install-libLTLIBRARIES
 
-# Run ldconfig after installing the library:
-install-hook:
-	-ldconfig
-
-
 FRAMEWORK_NAME = OpenCV
 FRAMEWORK_VERSION = A
 FRAMEWORK_CURRENT_VERSION = 2
