Index: opencv-1.0.0/otherlibs/highgui/cvcap_ffmpeg.cpp
===================================================================
--- opencv-1.0.0.orig/otherlibs/highgui/cvcap_ffmpeg.cpp
+++ opencv-1.0.0/otherlibs/highgui/cvcap_ffmpeg.cpp
@@ -390,7 +390,7 @@ typedef struct CvAVI_FFMPEG_Writer
  * the following function is a modified version of code
  * found in ffmpeg-0.4.9-pre1/output_example.c
  */
-static AVFrame * icv_alloc_picture_FFMPEG(int pix_fmt, int width, int height, bool alloc)
+static AVFrame * icv_alloc_picture_FFMPEG(PixelFormat pix_fmt, int width, int height, bool alloc)
 {
 	AVFrame * picture;
 	uint8_t * picture_buf;
