*** xen-3.0.2/tools/examples/xend-config.sxp.orig	2006-06-07 12:53:18.000000000 +0200
--- xen-3.0.2/tools/examples/xend-config.sxp	2006-07-11 15:43:47.000000000 +0200
***************
*** 15,27 ****
  #(loglevel DEBUG)
  
  #(xend-http-server no)
! #(xend-unix-server no)
  #(xend-tcp-xmlrpc-server no)
  #(xend-unix-xmlrpc-server yes)
  #(xend-relocation-server no)
  (xend-relocation-server yes)
  
! #(xend-unix-path /var/lib/xend/xend-socket)
  
  # Port xend should use for the HTTP interface, if xend-http-server is set.
  #(xend-port            8000)
--- 15,27 ----
  #(loglevel DEBUG)
  
  #(xend-http-server no)
! (xend-unix-server yes)
  #(xend-tcp-xmlrpc-server no)
  #(xend-unix-xmlrpc-server yes)
  #(xend-relocation-server no)
  (xend-relocation-server yes)
  
! (xend-unix-path /var/lib/xend/xend-socket)
  
  # Port xend should use for the HTTP interface, if xend-http-server is set.
  #(xend-port            8000)
