*** xen-3.0.2/tools/examples/xend-config.sxp.orig	2006-07-17 16:09:53.000000000 -0400
--- xen-3.0.2/tools/examples/xend-config.sxp	2006-07-17 16:12:27.000000000 -0400
***************
*** 19,25 ****
  #(xend-tcp-xmlrpc-server no)
  #(xend-unix-xmlrpc-server yes)
  #(xend-relocation-server no)
! (xend-relocation-server yes)
  
  (xend-unix-path /var/lib/xend/xend-socket)
  
--- 19,27 ----
  #(xend-tcp-xmlrpc-server no)
  #(xend-unix-xmlrpc-server yes)
  #(xend-relocation-server no)
! # The relocation server should be kept desactivated unless using a trusted
! # network, the domain virtual memory will be exchanged in raw form without
! # encryption of the communication. See also xend-relocation-hosts-allow option
  
  (xend-unix-path /var/lib/xend/xend-socket)
  
