Index: libcap-2.16/pam_cap/Makefile
===================================================================
--- libcap-2.16.orig/pam_cap/Makefile
+++ libcap-2.16/pam_cap/Makefile
@@ -14,7 +14,7 @@ install: all
 	install -m 0755 pam_cap.so $(LIBDIR)/security
 
 pam_cap.so: pam_cap.o
-	$(LD) $(LDFLAGS) -o pam_cap.so $< $(LDLIBS)
+	$(LD) $(LDFLAGS) -o pam_cap.so $< $(LDLIBS) -lpam
 
 pam_cap.o: pam_cap.c
 	$(CC) $(CFLAGS) -c $< -o $@
