diff -ur linux-2.6.15.orig/include/linux/joystick.h linux-2.6.15/include/linux/joystick.h
--- linux-2.6.15.orig/include/linux/joystick.h	2006-01-03 05:21:10.000000000 +0200
+++ linux-2.6.15/include/linux/joystick.h	2006-04-21 14:57:11.000000000 +0300
@@ -110,6 +110,9 @@
 #define JS_GET_ALL		7
 #define JS_SET_ALL		8
 
+typedef signed int int32_t;
+typedef signed long long int64_t;
+
 struct JS_DATA_TYPE {
 	int32_t buttons;
 	int32_t x;
