diff -Nur et131x-1.2.3-old/et131x_isr.h et131x-1.2.3/et131x_isr.h
--- et131x-1.2.3-old/et131x_isr.h	2007-07-06 18:10:27.275575075 +0300
+++ et131x-1.2.3/et131x_isr.h	2007-07-06 18:16:18.985107099 +0300
@@ -89,7 +89,7 @@
 irqreturn_t et131x_isr( int irq, void *dev_id, struct pt_regs *regs );
 #endif
 
-void et131x_isr_handler( struct work_struct *work );
+void et131x_isr_handler( ET131X_ADAPTER *adapter );
 
 
 #endif  /* __ET131X_ISR_H__ */
