diff -Nur audacity-src-1.2.6.orig/configure.in audacity-src-1.2.6/configure.in
--- audacity-src-1.2.6.orig/configure.in	2006-11-15 06:12:13.000000000 +0200
+++ audacity-src-1.2.6/configure.in	2007-05-19 21:37:33.483739742 +0300
@@ -95,9 +95,7 @@
 dnl Required sublibraries
 dnl
 
-BUILDLIBS="$BUILDLIBS expat/expat.a"
-EXTRAOBJS="$EXTRAOBJS ../lib-src/expat/expat.a"
-CXXFLAGS="$CXXFLAGS -I../lib-src/expat"
+LIBS="$LIBS -lexpat"
 
 dnl TODO enable/disable allegro
 BUILDLIBS="$BUILDLIBS allegro/allegro.a"
@@ -862,7 +860,7 @@
 esac
 
 AC_CONFIG_FILES([Makefile src/Makefile lib-src/Makefile \
-					  lib-src/allegro/Makefile lib-src/expat/Makefile \
+					  lib-src/allegro/Makefile \
 					  lib-src/libnyquist/Makefile \
 					  locale/Makefile ])
 
diff -Nur audacity-src-1.2.6.orig/src/xml/XMLFileReader.h audacity-src-1.2.6/src/xml/XMLFileReader.h
--- audacity-src-1.2.6.orig/src/xml/XMLFileReader.h	2006-11-15 06:12:10.000000000 +0200
+++ audacity-src-1.2.6/src/xml/XMLFileReader.h	2007-05-19 21:37:16.091383781 +0300
@@ -8,7 +8,7 @@
 
 **********************************************************************/
 
-#include "xmlparse/xmlparse.h"
+#include <expat.h>
 
 #include "XMLTagHandler.h"
 
