Index: src/plugins/welcome/communitywelcomepagewidget.cpp
===================================================================
--- src/plugins/welcome/communitywelcomepagewidget.cpp.orig
+++ src/plugins/welcome/communitywelcomepagewidget.cpp
@@ -53,7 +53,9 @@ CommunityWelcomePageWidget::CommunityWel
     connect(m_rssFetcher, SIGNAL(newsItemReady(QString, QString, QString)),
         ui->newsTreeWidget, SLOT(slotAddNewsItem(QString, QString, QString)));
     //: Add localized feed here only if one exists
-    m_rssFetcher->fetch(QUrl(tr("http://labs.trolltech.com/blogs/feed")));
+
+    // Disable rss fetch at startup
+    // m_rssFetcher->fetch(QUrl(tr("http://labs.trolltech.com/blogs/feed")));
 
     QList<QPair<QString, QString> > sites;
     sites << qMakePair(tr("Qt Home"), QString(QLatin1String("http://qt.nokia.com")));
