diff -Naur kdelibs-4.2.3-orig/kdeui/kernel/kglobalsettings.cpp kdelibs-4.2.3/kdeui/kernel/kglobalsettings.cpp
--- kdelibs-4.2.3-orig/kdeui/kernel/kglobalsettings.cpp	2009-05-10 21:56:55.975487969 +0300
+++ kdelibs-4.2.3/kdeui/kernel/kglobalsettings.cpp	2009-05-10 21:59:57.455239120 +0300
@@ -387,13 +387,13 @@
     { GeneralId, "toolBarFont", DefaultMacFont, 11, -1, QFont::SansSerif },
     { GeneralId, "menuFont",    DefaultMacFont, 13, -1, QFont::SansSerif },
 #else
-    { GeneralId, "font",        DefaultFont, 10, -1, QFont::SansSerif },
-    { GeneralId, "fixed",       "Monospace", 10, -1, QFont::TypeWriter },
+    { GeneralId, "font",        DefaultFont, 9, -1, QFont::SansSerif },
+    { GeneralId, "fixed",       "Monospace", 9, -1, QFont::TypeWriter },
     { GeneralId, "toolBarFont", DefaultFont,  8, -1, QFont::SansSerif },
-    { GeneralId, "menuFont",    DefaultFont, 10, -1, QFont::SansSerif },
+    { GeneralId, "menuFont",    DefaultFont, 9, -1, QFont::SansSerif },
 #endif
     { "WM",      "activeFont",           DefaultFont,  9, QFont::Bold, QFont::SansSerif },// inconsistency
-    { GeneralId, "taskbarFont",          DefaultFont, 10, -1, QFont::SansSerif },
+    { GeneralId, "taskbarFont",          DefaultFont, 9, -1, QFont::SansSerif },
     { GeneralId, "smallestReadableFont", DefaultFont,  8, -1, QFont::SansSerif }
 };
 
