diff -Naur kdecore-orig/localization/kcharsets.cpp kdecore/localization/kcharsets.cpp
--- kdecore-orig/localization/kcharsets.cpp	2009-07-15 10:23:45.254784748 +0300
+++ kdecore/localization/kcharsets.cpp	2009-07-15 10:24:42.451400133 +0300
@@ -724,7 +724,7 @@
     if ( codec )
         return codec;
     else
-        return QTextCodec::codecForName( "iso8859-1" );
+        return QTextCodec::codecForName( "iso-8859-1" );
 }
 
 QTextCodec* KCharsets::codecForName(const QString &n, bool &ok) const
@@ -739,7 +739,7 @@
     else
     {
         ok = false;
-        return QTextCodec::codecForName( "iso8859-1" );
+        return QTextCodec::codecForName( "iso-8859-1" );
     }
 }
 
