--- plasma/delegate.cpp.orig	2009-05-29 19:25:06.291318233 +0300
+++ plasma/delegate.cpp	2009-05-29 19:26:15.236535393 +0300
@@ -363,13 +363,8 @@
         if (column == 0) {
             //clip right (or left for rtl languages) to make the connection with the next column
             if (columns > 1) {
-                if (option.direction == Qt::LeftToRight) {
                     painter->setClipRect(option.rect);
                     highlightRect.adjust(0, 0, roundedRadius, 0);
-                } else {
-                    painter->setClipRect(option.rect);
-                    highlightRect.adjust(-roundedRadius, 0, 0, 0);
-                }
             }
 
             QLinearGradient gradient(highlightRect.topLeft(), highlightRect.topRight());
@@ -390,13 +385,8 @@
 
         //last column, clip left (right for rtl)
         } else if (column == columns-1) {
-            if (option.direction == Qt::LeftToRight) {
                 painter->setClipRect(option.rect);
                 highlightRect.adjust(-roundedRadius, 0, 0, 0);
-            } else {
-                painter->setClipRect(option.rect);
-                highlightRect.adjust(0, 0, +roundedRadius, 0);
-            }
 
         //column < columns-1; clip both ways
         } else {
