--- plasma/abstractrunner.cpp.gokcen	2009-03-26 17:44:22.000000000 +0300
+++ plasma/abstractrunner.cpp	2009-05-29 20:50:52.626416985 +0300
@@ -51,7 +51,8 @@
         runnerDescription(service),
         runner(r),
         fastRuns(0),
-        package(0)
+        package(0),
+        hasRunOptions(false)
     {
         if (runnerDescription.isValid()) {
             const QString api = runnerDescription.property("X-Plasma-API").toString();
@@ -84,8 +85,6 @@
         package = 0;
     }
 
-    bool hasRunOptions;
-    bool hasConfig;
     AbstractRunner::Priority priority;
     AbstractRunner::Speed speed;
     RunnerContext::Types blackListed;
@@ -96,6 +95,7 @@
     int fastRuns;
     Package *package;
     QHash<QString, QAction*> actions;
+    bool hasRunOptions;
 };
 
 K_GLOBAL_STATIC(QMutex, s_bigLock)
