--- plasma/widgets/pushbutton.cpp	2009/04/24 07:40:52	958514
+++ plasma/widgets/pushbutton.cpp	2009/05/10 14:46:46	966173
@@ -300,7 +300,7 @@
     }
 
     //if is under mouse draw the animated glow overlay
-    if (!nativeWidget()->isDown() && isEnabled() && acceptHoverEvents()) {
+    if (!nativeWidget()->isDown() && !nativeWidget()->isChecked() && isEnabled() && acceptHoverEvents()) {
         if (d->animId != -1) {
             QPixmap normalPix = d->background->framePixmap();
             d->background->setElementPrefix("active");
