--- src/historygui.ui.orig	2008-09-13 02:42:48.000000000 +0300
+++ src/historygui.ui	2008-09-13 02:27:48.000000000 +0300
@@ -48,6 +48,9 @@
                     <property name="text">
                         <string>Help</string>
                     </property>
+                    <property name="flat">
+                        <bool>true</bool>
+                    </property>
                 </widget>
                 <spacer>
                     <property name="name">
@@ -73,6 +76,9 @@
                     <property name="text">
                         <string>New</string>
                     </property>
+                    <property name="flat">
+                        <bool>true</bool>
+                    </property>
                 </widget>
                 <widget class="QPushButton">
                     <property name="name">
@@ -84,6 +90,9 @@
                     <property name="text">
                         <string>Restore</string>
                     </property>
+                    <property name="flat">
+                        <bool>true</bool>
+                    </property>
                 </widget>
             </hbox>
         </widget>
@@ -219,6 +228,9 @@
                         <property name="frameShape">
                             <enum>MenuBarPanel</enum>
                         </property>
+                        <property name="selectionMode">
+                            <enum>Extended</enum>
+                        </property>
                     </widget>
                 </grid>
             </widget>
