Index: kfingermanager.desktop
===================================================================
--- kfingermanager.desktop
+++ kfingermanager.desktop	2010-10-07 19:08:18.900607751 -0300
@@ -8,7 +8,7 @@
 X-KDE-Library=kcm_kfingermanager
 X-KDE-ParentApp=kcontrol
 
-X-KDE-System-Settings-Parent-Category=about-me
+X-KDE-System-Settings-Parent-Category=account-details
 X-KDE-Weight=50
 
 Name=Fingerprint Manager
