--- cube.xml.in.orig	2009-07-15 01:47:54.746537147 +0300
+++ metadata/cube.xml.in	2009-07-15 01:48:58.084609894 +0300
@@ -102,7 +102,7 @@
 		    <option name="scale_image" type="bool">
 			<_short>Scale image</_short>
 			<_long>Scale images to cover top face of cube</_long>
-			<default>false</default>
+			<default>true</default>
 		    </option>
 		    <option name="images" type="list">
 			<_short>Image files</_short>
@@ -110,7 +110,7 @@
 			<type>string</type>
 			<hints>file;image;</hints>
 			<default>
-			    <value>freedesktop</value>
+			    <value>pardus-top.png</value>
 			</default>
 		    </option>
 		    <option name="adjust_image" type="bool">
