diff -up ManiaDrive-1.2-src/raydium/config.h~ ManiaDrive-1.2-src/raydium/config.h
--- ManiaDrive-1.2-src/raydium/config.h~	2006-08-08 18:58:51.000000000 +0200
+++ ManiaDrive-1.2-src/raydium/config.h	2009-02-15 21:47:11.000000000 +0100
@@ -17,7 +17,7 @@
 #ifndef NO_ODE_SUPPORT
 // Enable ODE physics support
 #define ODE_SUPPORT
-#define ODE_QUICKSTEP
+//#define ODE_QUICKSTEP
 #define ODE_PREDICTION
 //#define ODE_NETWORK_GHOSTS
 #endif
