diff -Nur mc-4.6.2-pre1-old/lib/mc.ext.in mc-4.6.2-pre1/lib/mc.ext.in
--- mc-4.6.2-pre1-old/lib/mc.ext.in	2008-01-26 18:52:08.864191228 +0200
+++ mc-4.6.2-pre1/lib/mc.ext.in	2008-01-26 18:54:50.280652197 +0200
@@ -207,7 +207,7 @@
 	View=%view{ascii} isoinfo -l -i %f
 
 # 7zip archives (they are not man pages)
-shell/.7z
+regex/\.(7z|7Z)$
 	Open=%cd %p#u7z
 	View=%view{ascii} 7za l %f 2>/dev/null
 
