diff -ur pm-utils-0.20.0.20061031.orig/pm/functions pm-utils-0.20.0.20061031/pm/functions
--- pm-utils-0.20.0.20061031.orig/pm/functions	2007-01-13 03:08:48.918193000 +0200
+++ pm-utils-0.20.0.20061031/pm/functions	2007-01-13 03:11:51.711404969 +0200
@@ -1,6 +1,7 @@
 #!/bin/bash
 
 export PATH=/sbin:/usr/sbin:/bin:/usr/bin
+export LC_ALL=C
 
 # default values go here
 HIBERNATE_RESUME_POST_VIDEO=no
@@ -229,18 +230,30 @@
 
 stopservice()
 {
-	service "$1" status 2>/dev/null | grep -c -q running
-	if [ "$?" == "0" -a -x "/etc/init.d/$1" ]; then
+	if [ -f /etc/pardus-release ]; then
+		service "$1" info | grep -c -q running
+	else
+		service "$1" status 2>/dev/null | grep -c -q running
+	fi
+
+	if [ "$?" == "0" ]; then
 		echo "export ${1}_SERVICE_ACTIVATE=yes" >> /var/run/pm-suspend
-		"/etc/init.d/$1" stop >/dev/null 2>&1
+		if [ -x "/etc/init.d/$1" ]; then
+			"/etc/init.d/$1" stop 2>&1
+		elif [ -f /etc/pardus-release ]; then
+			service "$1" stop 2>&1
+		fi
 	fi
 }
 
 restartservice()
 {
-	if [ "x$(eval echo \$${1}_SERVICE_ACTIVATE)" == "xyes" \
-			-a -x "/etc/init.d/$1" ]; then
-		"/etc/init.d/$1" start >/dev/null 2>&1
+	if [ "x$(eval echo \$${1}_SERVICE_ACTIVATE)" == "xyes" ]; then
+		if [ -x "/etc/init.d/$1" ]; then
+			"/etc/init.d/$1" start 2>&1
+		elif [ -f /etc/pardus-release ]; then
+			service "$1" start 2>&1
+		fi
 	fi
 }
 
