Index: kdepim/wizards/Makefile.am
===================================================================
--- kdepim.orig/wizards/Makefile.am
+++ kdepim/wizards/Makefile.am
@@ -19,7 +19,7 @@ kde_module_LTLIBRARIES = libegroupwarewi
                          libscalixwizard.la
 
 
-groupwarewizard_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+groupwarewizard_LDFLAGS = $(all_libraries) 
 groupwarewizard_LDADD = libegroupwarewizard.la libsloxwizard.la $(LIB_KDEUI) \
                         libkolabwizard.la \
                         libexchangewizard.la \
@@ -39,7 +39,7 @@ libkolabwizard_la_SOURCES = kolabwizard.
   kolabkmailchanges.cpp
 
 kolabwizard_LDADD = libkolabwizard.la $(LIB_KDEUI)
-kolabwizard_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+kolabwizard_LDFLAGS = $(all_libraries) 
 kolabwizard_SOURCES = kolabmain.cpp
 
 # eGroupware
@@ -53,7 +53,7 @@ libegroupwarewizard_la_LIBADD = $(top_bu
                                 $(LIB_KDEUI)
 
 egroupwarewizard_LDADD = libegroupwarewizard.la
-egroupwarewizard_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+egroupwarewizard_LDFLAGS = $(all_libraries) 
 egroupwarewizard_SOURCES = egroupwaremain.cpp
 
 # SLOX
@@ -64,7 +64,7 @@ libsloxwizard_la_LIBADD = $(top_builddir
                           $(top_builddir)/libkdepim/libkdepim.la
 
 sloxwizard_LDADD = libsloxwizard.la
-sloxwizard_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+sloxwizard_LDFLAGS = $(all_libraries) 
 sloxwizard_SOURCES = sloxmain.cpp
 
 # GroupWise
@@ -80,7 +80,7 @@ libgroupwisewizard_la_COMPILE_FIRST = $(
 
 
 groupwisewizard_LDADD = libgroupwisewizard.la
-groupwisewizard_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+groupwisewizard_LDFLAGS = $(all_libraries) 
 groupwisewizard_SOURCES = groupwisemain.cpp
 
 # Exchange
@@ -91,7 +91,7 @@ libexchangewizard_la_LIBADD = $(top_buil
                               $(top_builddir)/libkdepim/libkdepim.la
 
 exchangewizard_LDADD = libexchangewizard.la
-exchangewizard_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+exchangewizard_LDFLAGS = $(all_libraries) 
 exchangewizard_SOURCES = exchangemain.cpp
 
 # Scalix
@@ -106,7 +106,7 @@ libscalixwizard_la_SOURCES = scalixwizar
   scalixkmailchanges.cpp
 
 scalixwizard_LDADD = libscalixwizard.la $(LIB_KDEUI)
-scalixwizard_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+scalixwizard_LDFLAGS = $(all_libraries) 
 scalixwizard_SOURCES = scalixmain.cpp
 
 noinst_HEADERS = egroupwarewizard.h kmailchanges.h kolabwizard.h sloxwizard.h \
