Index: kdepim/kaddressbook/Makefile.am
===================================================================
--- kdepim.orig/kaddressbook/Makefile.am
+++ kdepim/kaddressbook/Makefile.am
@@ -17,13 +17,13 @@ INCLUDES= -I$(top_srcdir)/kaddressbook/i
 bin_PROGRAMS = kaddressbook
 
 kaddressbook_SOURCES = main.cpp kaddressbookmain.cpp
-kaddressbook_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+kaddressbook_LDFLAGS = $(all_libraries) 
 kaddressbook_LDADD = libkaddressbook.la
 
 kde_module_LTLIBRARIES = libkaddressbookpart.la
 
 libkaddressbookpart_la_SOURCES = kaddressbook_part.cpp
-libkaddressbookpart_la_LDFLAGS = -module $(KDE_PLUGIN) $(KDE_RPATH) $(all_libraries) -no-undefined
+libkaddressbookpart_la_LDFLAGS = -module $(KDE_PLUGIN)  $(all_libraries) -no-undefined
 libkaddressbookpart_la_LIBADD = libkaddressbook.la
 
 lib_LTLIBRARIES = libkaddressbook.la
@@ -40,7 +40,7 @@ libkaddressbook_la_SOURCES = addresseeco
     xxportmanager.cpp xxportselectdialog.cpp contacteditorwidgetmanager.cpp \
     simpleaddresseeeditor.cpp advancedcustomfields.cpp customfieldswidget.cpp \
     freebusywidget.cpp searchmanager.cpp imeditwidget.cpp kabtools.cpp distributionlistpicker.cpp distributionlisteditor.cpp distributionlistentryview.cpp
-libkaddressbook_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined
+libkaddressbook_la_LDFLAGS = $(all_libraries)  -no-undefined
 libkaddressbook_la_LIBADD = $(top_builddir)/kaddressbook/printing/libprinter.la \
                             $(top_builddir)/kaddressbook/interfaces/libkabinterfaces.la \
                             $(top_builddir)/libkdepim/libkdepim.la \
