Index: kdepim/kmail/Makefile.am
===================================================================
--- kdepim.orig/kmail/Makefile.am
+++ kdepim/kmail/Makefile.am
@@ -35,7 +35,7 @@ libkmailprivate_la_LIBADD  = \
         $(LIB_KHTML) $(LIB_KSPELL) $(LIB_KABC)
 
 kde_module_LTLIBRARIES = kcm_kmail.la libkmailpart.la libkmail_bodypartformatter_application_octetstream.la
-libkmailpart_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version -no-undefined
+libkmailpart_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
 libkmailpart_la_LIBADD  = libkmailprivate.la
 
 kcm_kmail_la_SOURCES = kcm_kmail.cpp
@@ -49,7 +49,7 @@ linkdir = $(kde_datadir)/kmail/plugins/b
 link_DATA = application_octetstream.desktop
 
 bin_PROGRAMS = kmail
-kmail_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+kmail_LDFLAGS = $(all_libraries)
 kmail_LDADD   = libkmailprivate.la
 
 libkmailprivate_la_SOURCES = kmmessage.cpp kmmainwin.cpp configuredialog.cpp \
