Index: kdepim/kabc/kabcdistlistupdater/Makefile.am
===================================================================
--- kdepim.orig/kabc/kabcdistlistupdater/Makefile.am
+++ kdepim/kabc/kabcdistlistupdater/Makefile.am
@@ -4,7 +4,7 @@ METASOURCES = AUTO
 
 bin_PROGRAMS = kabcdistlistupdater
 
-kabcdistlistupdater_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+kabcdistlistupdater_LDFLAGS = $(all_libraries)
 kabcdistlistupdater_LDADD = $(LIB_KABC) $(top_builddir)/libkdepim/libkdepim.la
 kabcdistlistupdater_SOURCES = kabcdistlistupdater.cpp
 
