Index: kdepim/korganizer/korgac/Makefile.am
===================================================================
--- kdepim.orig/korganizer/korgac/Makefile.am
+++ kdepim/korganizer/korgac/Makefile.am
@@ -2,7 +2,7 @@ INCLUDES= -I$(top_srcdir) -I$(top_srcdir
 
 bin_PROGRAMS = korgac
 
-korgac_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+korgac_LDFLAGS = $(all_libraries)
 korgac_LDADD   = $(LIB_KDEUI) \
                  $(top_builddir)/korganizer/libkorganizer_eventviewer.la
 korgac_SOURCES = korgacmain.cpp alarmdialog.cpp \
@@ -11,7 +11,7 @@ korgac_SOURCES = korgacmain.cpp alarmdia
 
 check_PROGRAMS = testalarmdlg
 
-testalarmdlg_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+testalarmdlg_LDFLAGS = $(all_libraries)
 testalarmdlg_LDADD = $(LIB_KDEUI) $(top_builddir)/libkcal/libkcal.la \
                      $(top_builddir)/korganizer/libkorganizer_eventviewer.la
 testalarmdlg_SOURCES = testalarmdlg.cpp alarmdialog.cpp
