Index: dependencies/tqtinterface/qtinterface/Makefile.am
===================================================================
--- dependencies.orig/tqtinterface/qtinterface/Makefile.am
+++ dependencies/tqtinterface/qtinterface/Makefile.am
@@ -78,7 +78,7 @@ tqxml.cpp \
 private/tqucomextra_p.cpp \
 \
 tqlistiterator.cpp
-libtqt_la_LIBADD = $(LIB_QT)
+libtqt_la_LIBADD = $(QT_LDFLAGS) $(LIB_QT)
 libtqt_la_LDFLAGS = $(KDE_RPATH) -version-info 6:0:2 -no-undefined $(all_libraries)
 
 include_HEADERS = tqt.h tqaccel.h tqaccessible.h tqaction.h tqapplication.h tqasyncio.h \
