diff -pruN kdebase-3.5.9-orig/kcontrol/ebrowsing/plugins/localdomain/Makefile.am kdebase-3.5.9/kcontrol/ebrowsing/plugins/localdomain/Makefile.am
--- kdebase-3.5.9-orig/kcontrol/ebrowsing/plugins/localdomain/Makefile.am	2008-03-23 23:32:54.000000000 +0200
+++ kdebase-3.5.9/kcontrol/ebrowsing/plugins/localdomain/Makefile.am	2008-03-23 23:38:28.000000000 +0200
@@ -5,7 +5,7 @@ AM_LDFLAGS   = $(all_libraries)
 
 kde_module_LTLIBRARIES = liblocaldomainurifilter.la
 
-liblocaldomainurifilter_la_LDFLAGS = -module $(KDE_PLUGIN)
+liblocaldomainurifilter_la_LDFLAGS = -module $(AM_LDFLAGS) $(KDE_PLUGIN)
 liblocaldomainurifilter_la_LIBADD = $(LIB_KIO) $(LIB_KDEUI) $(LIB_KDECORE)
 
 liblocaldomainurifilter_la_SOURCES = localdomainurifilter.cpp localdomainurifilter.skel
diff -pruN kdebase-3.5.9-orig/kcontrol/ebrowsing/plugins/shorturi/Makefile.am kdebase-3.5.9/kcontrol/ebrowsing/plugins/shorturi/Makefile.am
--- kdebase-3.5.9-orig/kcontrol/ebrowsing/plugins/shorturi/Makefile.am	2008-03-23 23:32:55.000000000 +0200
+++ kdebase-3.5.9/kcontrol/ebrowsing/plugins/shorturi/Makefile.am	2008-03-23 23:37:44.000000000 +0200
@@ -5,7 +5,7 @@ AM_LDFLAGS   = $(all_libraries)
 
 kde_module_LTLIBRARIES = libkshorturifilter.la
 
-libkshorturifilter_la_LDFLAGS = -module $(KDE_PLUGIN)
+libkshorturifilter_la_LDFLAGS = -module $(AM_LDFLAGS) $(KDE_PLUGIN)
 libkshorturifilter_la_LIBADD = $(LIB_KSYCOCA) $(LIB_KDEUI)
 
 libkshorturifilter_la_SOURCES = kshorturifilter.cpp kshorturifilter.skel
diff -pruN kdebase-3.5.9-orig/kioslave/floppy/Makefile.am kdebase-3.5.9/kioslave/floppy/Makefile.am
--- kdebase-3.5.9-orig/kioslave/floppy/Makefile.am	2008-03-23 23:32:51.000000000 +0200
+++ kdebase-3.5.9/kioslave/floppy/Makefile.am	2008-03-23 23:33:44.000000000 +0200
@@ -9,7 +9,7 @@ kde_module_LTLIBRARIES = kio_floppy.la
 
 kio_floppy_la_SOURCES = kio_floppy.cpp program.cpp
 kio_floppy_la_LIBADD = $(LIB_KIO)
-kio_floppy_la_LDFLAGS = -module $(KDE_PLUGIN)
+kio_floppy_la_LDFLAGS = -module $(AM_LDFLAGS) $(KDE_PLUGIN)
 
 kdelnk_DATA = floppy.protocol
 kdelnkdir = $(kde_servicesdir)
diff -pruN kdebase-3.5.9-orig/kioslave/smb/Makefile.am kdebase-3.5.9/kioslave/smb/Makefile.am
--- kdebase-3.5.9-orig/kioslave/smb/Makefile.am	2008-03-23 23:32:51.000000000 +0200
+++ kdebase-3.5.9/kioslave/smb/Makefile.am	2008-03-23 23:34:31.000000000 +0200
@@ -16,7 +16,7 @@ kio_smb_la_SOURCES = kio_smb.cpp \
 
 kio_smb_la_LIBADD = -lkio -lsmbclient $(SMBCLIENT_EXTRA_LIBS)
 
-kio_smb_la_LDFLAGS = -module $(KDE_PLUGIN)
+kio_smb_la_LDFLAGS = -module $(AM_LDFLAGS) $(KDE_PLUGIN)
 
 noinst_HEADERS = kio_smb.h kio_smb_internal.h
 
