Index: kioslave/media/libmediacommon/notifiersettings.cpp
===================================================================
--- kioslave/media/libmediacommon/notifiersettings.cpp	(revision 605180)
+++ kioslave/media/libmediacommon/notifiersettings.cpp	(working copy)
@@ -194,22 +194,22 @@
 
 	m_idMap.clear();
 	m_autoMimetypesMap.clear();
-	
-	NotifierOpenAction *open = new NotifierOpenAction();
-	m_actions.append( open );
-	m_idMap[ open->id() ] = open;
 
 	QValueList<NotifierServiceAction*> services = listServices();
 
 	QValueList<NotifierServiceAction*>::iterator serv_it = services.begin();
 	QValueList<NotifierServiceAction*>::iterator serv_end = services.end();
-	
+
 	for ( ; serv_it!=serv_end; ++serv_it )
 	{
 		m_actions.append( *serv_it );
 		m_idMap[ (*serv_it)->id() ] = *serv_it;
 	}
-	
+
+        NotifierOpenAction *open = new NotifierOpenAction();
+        m_actions.append( open );
+        m_idMap[ open->id() ] = open;
+
 	NotifierNothingAction *nothing = new NotifierNothingAction();
 	m_actions.append( nothing );
 	m_idMap[ nothing->id() ] = nothing;
