--- openct-0.6.14-latest-orig/etc/openct.fdi	2008-07-13 11:41:24.000000000 +0300
+++ openct-0.6.14-latest/etc/openct.fdi	2007-08-23 10:09:41.000000000 +0300
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?> <!-- -*- SGML -*- -->
 <deviceinfo version="0.2">
   <device>
-    <match key="info.bus" string="usb">
+    <match key="info.subsystem" string="usb">
 
       <match key="usb.vendor_id" int="0x0973">
 	<match key="usb.product_id" int="0x0001">
