--- bindings/python/Makefile.in.orig	2007-11-10 13:16:18.000000000 +0200
+++ bindings/python/Makefile.in	2008-09-05 23:43:39.000000000 +0300
@@ -443,13 +443,6 @@
 	    $(gpodPYTHON_INSTALL) "$$b$$p" "$(DESTDIR)$(gpoddir)/$$f"; \
 	  else :; fi; \
 	done; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(gpoddir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(gpoddir)" $$dlist; \
-	  fi; \
-	else :; fi
 
 uninstall-gpodPYTHON:
 	@$(NORMAL_UNINSTALL)
