diff -up dvgrab-3.1/dvframe.cc.gcc43~ dvgrab-3.1/dvframe.cc
--- dvgrab-3.1/dvframe.cc.gcc43~	2008-02-13 16:15:39.000000000 +0900
+++ dvgrab-3.1/dvframe.cc	2008-02-13 16:15:39.000000000 +0900
@@ -25,6 +25,7 @@
     \file dvframe.cc
 */
 
+#include <string.h>
 #include "dvframe.h"
 
 VideoInfo::VideoInfo() : width( 0 ), height( 0 ), isPAL( false )
diff -up dvgrab-3.1/avi.cc.gcc43~ dvgrab-3.1/avi.cc
--- dvgrab-3.1/avi.cc.gcc43~	2008-02-13 16:15:35.000000000 +0900
+++ dvgrab-3.1/avi.cc	2008-02-13 16:15:35.000000000 +0900
@@ -17,6 +17,7 @@
 * Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 */
 
+#include <string.h>
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
diff -up dvgrab-3.1/main.cc.gcc43~ dvgrab-3.1/main.cc
--- dvgrab-3.1/main.cc.gcc43~	2008-02-13 16:16:00.000000000 +0900
+++ dvgrab-3.1/main.cc	2008-02-13 16:16:00.000000000 +0900
@@ -23,6 +23,7 @@
     contains the main logic
 */
 
+#include <string.h>
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
diff -up dvgrab-3.1/filehandler.cc.gcc43~ dvgrab-3.1/filehandler.cc
--- dvgrab-3.1/filehandler.cc.gcc43~	2008-02-13 16:15:45.000000000 +0900
+++ dvgrab-3.1/filehandler.cc	2008-02-13 16:15:45.000000000 +0900
@@ -19,6 +19,7 @@
 * Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 */
 
+#include <string.h>
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
diff -up dvgrab-3.1/hdvframe.cc.gcc43~ dvgrab-3.1/hdvframe.cc
--- dvgrab-3.1/hdvframe.cc.gcc43~	2008-02-13 16:15:51.000000000 +0900
+++ dvgrab-3.1/hdvframe.cc	2008-02-13 16:15:51.000000000 +0900
@@ -18,6 +18,7 @@
 * Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 */
 
+#include <string.h>
 #include "hdvframe.h"
 
 HDVFrame::HDVFrame( HDVStreamParams *p )
diff -up dvgrab-3.1/ieee1394io.cc.gcc43~ dvgrab-3.1/ieee1394io.cc
--- dvgrab-3.1/ieee1394io.cc.gcc43~	2008-02-13 16:15:57.000000000 +0900
+++ dvgrab-3.1/ieee1394io.cc	2008-02-13 16:15:57.000000000 +0900
@@ -43,6 +43,8 @@
  
  */
 
+#include <string.h>
+#include <typeinfo>
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
diff -up dvgrab-3.1/v4l2reader.cc.gcc43~ dvgrab-3.1/v4l2reader.cc
--- dvgrab-3.1/v4l2reader.cc.gcc43~	2008-02-13 16:16:05.000000000 +0900
+++ dvgrab-3.1/v4l2reader.cc	2008-02-13 16:16:05.000000000 +0900
@@ -18,6 +18,7 @@
 */
 
 
+#include <string.h>
 #include "v4l2reader.h"
 
 #ifdef HAVE_LINUX_VIDEODEV2_H
diff -up dvgrab-3.1/dvgrab.cc.gcc43~ dvgrab-3.1/dvgrab.cc
--- dvgrab-3.1/dvgrab.cc.gcc43~	2008-02-13 16:15:40.000000000 +0900
+++ dvgrab-3.1/dvgrab.cc	2008-02-13 16:15:40.000000000 +0900
@@ -19,6 +19,7 @@
 * Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 */
 
+#include <string.h>
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
diff -up dvgrab-3.1/iec13818-1.cc.gcc43~ dvgrab-3.1/iec13818-1.cc
--- dvgrab-3.1/iec13818-1.cc.gcc43~	2008-02-13 16:15:53.000000000 +0900
+++ dvgrab-3.1/iec13818-1.cc	2008-02-13 16:15:53.000000000 +0900
@@ -17,6 +17,7 @@
 * Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 */
 
+#include <string.h>
 #include <string>
 using std::string;
 
