--- synergy-1.3.1/lib/platform/CXWindowsClipboard.cpp~	2008-06-02 16:43:10.000000000 +0200
+++ synergy-1.3.1/lib/platform/CXWindowsClipboard.cpp	2008-06-02 16:43:10.000000000 +0200
@@ -1193,7 +1193,7 @@ CXWindowsClipboard::wasOwnedAtTime(::Tim
 	// compare time to range
 	Time duration = lost - m_timeOwned;
 	Time when     = time - m_timeOwned;
-	return (/*when >= 0 &&*/ when < duration);
+	return (/*when >= 0 &&*/ when <= duration);
 }
 
 Atom
