diff -pruN inkscape-0.46-orig/src/dom/io/uristream.cpp inkscape-0.46/src/dom/io/uristream.cpp
--- inkscape-0.46-orig/src/dom/io/uristream.cpp	2008-04-02 15:25:39.000000000 +0300
+++ inkscape-0.46/src/dom/io/uristream.cpp	2008-04-02 15:45:56.000000000 +0300
@@ -41,7 +41,7 @@
 
 
 #include "uristream.h"
-
+#include <cstring>
 
 
 namespace org
diff -pruN inkscape-0.46-orig/src/libcola/straightener.cpp inkscape-0.46/src/libcola/straightener.cpp
--- inkscape-0.46-orig/src/libcola/straightener.cpp	2008-04-02 15:25:39.000000000 +0300
+++ inkscape-0.46/src/libcola/straightener.cpp	2008-04-02 15:38:38.000000000 +0300
@@ -21,6 +21,7 @@
 #include <iostream>
 #include <cmath>
 #include <cstdlib>
+#include <algorithm>
 
 using std::set;
 using std::vector;
diff -pruN inkscape-0.46-orig/src/libvpsc/generate-constraints.cpp inkscape-0.46/src/libvpsc/generate-constraints.cpp
--- inkscape-0.46-orig/src/libvpsc/generate-constraints.cpp	2008-04-02 15:25:39.000000000 +0300
+++ inkscape-0.46/src/libvpsc/generate-constraints.cpp	2008-04-02 15:40:26.000000000 +0300
@@ -15,6 +15,7 @@
 #include <cstdlib>
 #include "generate-constraints.h"
 #include "constraint.h"
+#include <algorithm>
 
 #include "isnan.h" /* Include last */
 
diff -pruN inkscape-0.46-orig/src/livarot/int-line.cpp inkscape-0.46/src/livarot/int-line.cpp
--- inkscape-0.46-orig/src/livarot/int-line.cpp	2008-04-02 15:25:39.000000000 +0300
+++ inkscape-0.46/src/livarot/int-line.cpp	2008-04-02 15:41:26.000000000 +0300
@@ -15,6 +15,7 @@
 #include <string>
 #include <cstdlib>
 #include <cstdio>
+#include <algorithm>
 #include "livarot/int-line.h"
 #include "livarot/float-line.h"
 #include "livarot/BitLigne.h"
