--- src/game.cpp.orig	2007-09-19 14:09:30.000000000 +0300
+++ src/game.cpp	2007-09-22 18:19:39.000000000 +0300
@@ -867,7 +867,7 @@
 
 void IGame::loadPlugins() {
 	IFinder::FindResult path;
-	Finder->findAll(path, "../" + sdlx::Module::mangle("bt_objects"));
+	Finder->findAll(path, "../../lib/" + sdlx::Module::mangle("bt_objects"));
 	if (path.empty()) {
 		std::vector<std::string> dirs;
 		Finder->getPath(dirs);
