Index: libavcodec/wmadec.c
===================================================================
--- libavcodec/wmadec.c	(revision 7375)
+++ libavcodec/wmadec.c	(working copy)
@@ -910,7 +910,7 @@
                 ptr += run;
                 if (ptr >= eptr)
                 {
-                    av_log(NULL, AV_LOG_ERROR, "overflow in spectral RLE, ignoring\n");
+                    //av_log(NULL, AV_LOG_ERROR, "overflow in spectral RLE, ignoring\n");
                     break;
                 }
                 *ptr++ = level;
