Index: yagf-0.8.7/src/mainform.cpp
===================================================================
--- yagf-0.8.7.orig/src/mainform.cpp
+++ yagf-0.8.7/src/mainform.cpp
@@ -587,7 +587,7 @@ void MainForm::fillLanguagesBox()
     selectLangsBox->addItem(trUtf8("Serbian"), QVariant("srp"));
     selectLangsBox->addItem(trUtf8("Slovenian"), QVariant("slo"));
     selectLangsBox->addItem(trUtf8("Slovakian (tesseract only)"), QVariant("slk"));
-    selectLangsBox->addItem(trUtf8("Turkish (tesseract only)"), QVariant("tur"));
+    selectLangsBox->addItem(trUtf8("Turkish"), QVariant("tur"));
     selectLangsBox->addItem(trUtf8("Ukrainian"), QVariant("ukr"));
     selectLangsBox->addItem(trUtf8("Russian-French"), QVariant("rus_fra"));
     selectLangsBox->addItem(trUtf8("Russian-German"), QVariant("rus_ger"));
@@ -681,6 +681,9 @@ QString MainForm::selectDefaultLanguageN
         case QLocale::Slovenian:
             name = "slo";
             break;
+        case QLocale::Turkish:
+            name = "tur";
+            break;
         case QLocale::Ukrainian:
             name = "ukr";
         default:
