Index: yagf-0.8.6-Source/YAGF.desktop
===================================================================
--- yagf-0.8.6-Source.orig/YAGF.desktop
+++ yagf-0.8.6-Source/YAGF.desktop
@@ -11,13 +11,16 @@ Name[ru]=YAGF
 Name[lt]=YAGF
 Name[pl]=YAGF
 Name[de]=YAGF
+Name[tr]=yagf
 GenericName=OCR text recognition
 GenericName[ru]=Распознавание текста
 GenericName[lt]=ENTER YOUR TRANSLATION HERE
 GenericName[pl]=ENTER YOUR TRANSLATION HERE
 GenericName[de]=OCR Texterkennung
+GenericName[tr]=OKT metin tanıma
 Comment=Graphical frontend for Cuneiform
 Comment[ru]=Графическая оболочка для Cuneiform
 Comment[lt]=Grafinė teksto atpažinimo programos Cuneiform sąsaja
 Comment[pl]=Grafinė teksto atpažinimo programos Cuneiform sąsaja
-Comment[de]=Graphische Benutzeroberfläche zu Cuneiform
\ Dosya sonunda yenisatır yok.
+Comment[de]=Graphische Benutzeroberfläche zu Cuneiform
+Comment[tr]=Cuneiform için grafiksel önyüz
