Index: yagf-0.8.6-Source/CMakeLists.txt
===================================================================
--- yagf-0.8.6-Source.orig/CMakeLists.txt
+++ yagf-0.8.6-Source/CMakeLists.txt
@@ -36,7 +36,7 @@ if (CMAKE_SIZEOF_VOID_P EQUAL 8)
         set (LIB_PATH_SUFFIX 64)
         set_property (GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS TRUE)
 endif(CMAKE_SIZEOF_VOID_P EQUAL 8)
-install_targets(/lib${LIB_PATH_SUFFIX}/yagf xspreload)
+install_targets(/lib/yagf xspreload)
 install(FILES ${QM_FILES} DESTINATION share/yagf/translations/)
 #install(FILES COPYING DESCRIPTION README AUTHORS ChangeLog DESTINATION share/yagf/)
 install(FILES yagf.png DESTINATION share/pixmaps/)
