Index: yagf-0.8.6-Source/CMakeLists.txt
===================================================================
--- yagf-0.8.6-Source.orig/CMakeLists.txt
+++ yagf-0.8.6-Source/CMakeLists.txt
@@ -19,6 +19,7 @@ include_directories( ${CMAKE_BINARY_DIR}
 add_executable(yagf ${SOURCES} ${UI_HEADERS} ${MOC_SRCS} ${yagf_RCC_SRCS} ${QM_FILES})
 add_library(xspreload SHARED ${INT_SOURCES})
 target_link_libraries(yagf ${QT_LIBRARIES} ${ASPELL_LIBRARIES})
+target_link_libraries(xspreload dl)
 add_custom_target(translations DEPENDS ${QM_FILES})
 qt4_add_translation(yagf_ru.qm src/mainform.cpp ${MOC_SRCS} ${UI_HEADERS} src/yagf_ru.ts)
 qt4_add_translation(yagf_lt.qm src/mainform.cpp ${MOC_SRCS} ${UI_HEADERS} src/yagf_lt.ts)
