Index: sqlite-3.5.9/src/util.c
===================================================================
--- sqlite-3.5.9.orig/src/util.c
+++ sqlite-3.5.9/src/util.c
@@ -14,7 +14,7 @@
 ** This file contains functions for allocating memory, comparing
 ** strings, and stuff like that.
 **
-** $Id: util.c,v 1.229 2008/05/13 16:41:50 drh Exp $
+** $Id: util.c,v 1.229 2008/07/11 16:41:50 drh Exp $
 */
 #include "sqliteInt.h"
 #include <stdarg.h>
@@ -35,7 +35,8 @@ int sqlite3IsNaN(double x){
   **      rules/specifications for math functions.
   */
   volatile double y = x;
-  return x!=y;
+  volatile double z = y;
+  return y!=z;
 }
 
 /*
