--- skstream/skserver.cpp~	2006-10-12 04:10:12.000000000 +0300
+++ skstream/skserver.cpp	2008-06-05 20:33:02.000000000 +0300
@@ -41,6 +41,7 @@
 
 #include <iostream>
 #include <cstdio>
+#include <cstring>
 
 #ifdef _WIN32
 #define SHUT_RD SD_RECEIVE
--- skstream/skstream.cpp~	2006-10-12 04:10:12.000000000 +0300
+++ skstream/skstream.cpp	2008-06-05 20:34:34.000000000 +0300
@@ -44,6 +44,8 @@
 #include <cstdio>
 #include <cassert>
 
+#include <cstring>
+
 #ifdef _WIN32
 #define SHUT_RD SD_RECEIVE
 #define SHUT_WR SD_SEND
