diff -ur xorg-server-7.3.orig/xorg-server-1.4.2/composite/compalloc.c xorg-server-7.3/xorg-server-1.4.2/composite/compalloc.c
--- xorg-server-7.3.orig/xorg-server-1.4.2/composite/compalloc.c	2007-09-04 21:19:44.000000000 +0300
+++ xorg-server-7.3/xorg-server-1.4.2/composite/compalloc.c	2007-09-17 17:35:19.000000000 +0300
@@ -478,6 +478,7 @@
 	 * Copy bits from the parent into the new pixmap so that it will
 	 * have "reasonable" contents in case for background None areas.
 	 */
+#if 0
 	if (pGC)
 	{
 	    XID val = IncludeInferiors;
@@ -528,6 +529,7 @@
 	if (pDstPicture)
 	    FreePicture (pDstPicture, 0);
+#endif
     }
     return pPixmap;
 }
 
