--- blobAndConquer-1.03/src/defs.h.old	2008-11-16 17:51:46.000000000 +0200
+++ blobAndConquer-1.03/src/defs.h	2008-11-16 17:52:07.000000000 +0200
@@ -339,11 +339,11 @@
 
 /* ############# debug ################## */
 
-//#if USEPAK
-//	#define debug(x)
-//#else
+#if USEPAK
+	#define debug(x)
+#else
 	#define debug(x) {printf("[DEBUG (%d)] ", SDL_GetTicks()); printf x; fflush(stdout);}
-//#endif
+#endif
 
 #ifndef IGNORE_FLAGTOKEN_ERRORS
 	#define IGNORE_FLAGTOKEN_ERRORS 0
