diff -ur pyparted-1.8.6.orig/Makefile pyparted-1.8.6/Makefile
--- pyparted-1.8.6.orig/Makefile	2007-05-20 03:23:43.331612000 +0300
+++ pyparted-1.8.6/Makefile	2007-05-20 03:24:25.825130087 +0300
@@ -34,7 +34,7 @@
 # Build with libparted (parted-1.8.3 and higher have a pkg-config file)
 LDFLAGS += $(shell pkg-config --libs libparted)
 
-libdir ?= $(DESTDIR)$(shell rpm --eval "%{_libdir}")/python$(PYVER)/site-packages
+libdir ?= $(DESTDIR)/usr/lib/python2.4/site-packages
 
 all: partedmodule.so
 
