diff -Nur mpeg4ip-1.6-old/lib/mp4v2/Makefile.in mpeg4ip-1.6/lib/mp4v2/Makefile.in
--- mpeg4ip-1.6-old/lib/mp4v2/Makefile.in	2010-02-12 16:16:04.690903072 +0200
+++ mpeg4ip-1.6/lib/mp4v2/Makefile.in	2010-02-12 16:16:33.178903639 +0200
@@ -289,6 +289,7 @@
 AM_CXXFLAGS = @BILLS_CPPWARNINGS@
 lib_LTLIBRARIES = libmp4v2.la 
 include_HEADERS = mp4.h 
+libmp4v2_la_LDFLAGS = -version-number 1:9:1
 libmp4v2_la_SOURCES = \
 	3gp.cpp \
 	atom_amr.cpp \
