Index: network.py
===================================================================
--- network.py	(revision 10728)
+++ network.py	(working copy)
@@ -258,7 +258,7 @@
             data = "/usr/share/misc/pci.ids"
         else:
             data = "/usr/share/misc/usb.ids"
-        return idsQuery(data, vendor, device)
+        return idsQuery(data, vendor, device) + " (%s)" % dev
     if devuid.startswith("logic:"):
         return devuid.split(":", 1)[1]
     return devuid
