diff -Nurd xchat-2.0.8-virgin/src/common/cfgfiles.c xchat-2.0.8/src/common/cfgfiles.c
--- xchat-2.0.8-virgin/src/common/cfgfiles.c	2004-01-30 04:24:41.000000000 -0500
+++ xchat-2.0.8/src/common/cfgfiles.c	2004-04-19 14:58:37.428918632 -0400
@@ -616,7 +616,7 @@
 	strcpy (prefs.stamp_format, "[%H:%M] ");
 	strcpy (prefs.timestamp_log_format, "%b %d %H:%M:%S ");
 	strcpy (prefs.logmask, "%n-%c.log");
-	strcpy (prefs.nick_suffix, ",");
+	strcpy (prefs.nick_suffix, ":");
 	strcpy (prefs.cmdchar, "/");
 	strcpy (prefs.nick1, username);
 	strcpy (prefs.nick2, username);
