Index: src/QIComplete/tree.cpp
===================================================================
--- src/QIComplete/tree.cpp	(revision 322)
+++ src/QIComplete/tree.cpp	(revision 323)
@@ -8,6 +8,7 @@
  ********************************************************************************************************/
 #include <QRegExp>
 
+#include <stdlib.h>
 #include "tree.h"
 #include "parse.h"
 #include "readtags.h"
