Index: resources/translations/QDevelop_Turkish.ts
===================================================================
--- resources/translations/QDevelop_Turkish.ts	(revision 285)
+++ resources/translations/QDevelop_Turkish.ts	(working copy)
@@ -329,12 +329,12 @@
     <message>
         <location filename="../../ui/addnewclassmethod.ui" line="13"/>
         <source>Add New Method</source>
-        <translation>Yeni Yöntem Ekle</translation>
+        <translation>Yeni Metod Ekle</translation>
     </message>
     <message>
         <location filename="../../ui/addnewclassmethod.ui" line="41"/>
         <source>Method Name:</source>
-        <translation>Yöntem Adı:</translation>
+        <translation>Metod Adı:</translation>
     </message>
     <message>
         <location filename="../../ui/addnewclassmethod.ui" line="48"/>
@@ -500,7 +500,7 @@
     <message>
         <location filename="../../ui/addnewclassvariable.ui" line="25"/>
         <source>Create get/set methods</source>
-        <translation>get/set yöntemleri oluştur</translation>
+        <translation>get/set metodları oluştur</translation>
     </message>
     <message>
         <location filename="../../ui/addnewclassvariable.ui" line="56"/>
@@ -510,12 +510,12 @@
     <message>
         <location filename="../../ui/addnewclassvariable.ui" line="73"/>
         <source>set method</source>
-        <translation>set yöntemi</translation>
+        <translation>set metodu</translation>
     </message>
     <message>
         <location filename="../../ui/addnewclassvariable.ui" line="87"/>
         <source>get method</source>
-        <translation>get yöntemi</translation>
+        <translation>get metodu</translation>
     </message>
     <message>
         <location filename="../../ui/addnewclassvariable.ui" line="131"/>
@@ -727,7 +727,7 @@
     <message>
         <location filename="../../ui/addsetget.ui" line="13"/>
         <source>Add get/set methods</source>
-        <translation>get/set yöntemleri ekle</translation>
+        <translation>get/set metodları ekle</translation>
     </message>
     <message>
         <location filename="../../ui/addsetget.ui" line="59"/>
@@ -752,12 +752,12 @@
     <message>
         <location filename="../../ui/addsetget.ui" line="123"/>
         <source>set method</source>
-        <translation>set yöntemi</translation>
+        <translation>set metodu</translation>
     </message>
     <message>
         <location filename="../../ui/addsetget.ui" line="140"/>
         <source>get method</source>
-        <translation>get yöntemi</translation>
+        <translation>get metodu</translation>
     </message>
 </context>
 <context>
@@ -855,7 +855,7 @@
     <message>
         <location filename="../../src/editor.cpp" line="101"/>
         <source>Refresh classes and methods lists</source>
-        <translation>Sınıf ve yöntem listelerini yenile</translation>
+        <translation>Sınıf ve metod listelerini yenile</translation>
     </message>
     <message>
         <location filename="../../src/editor.cpp" line="245"/>
@@ -878,7 +878,7 @@
 was modified outside editor.
 
 </source>
-        <translation>\&quot;%1\&quot; dosyası 
+        <translation>&quot;%1&quot; dosyası 
 editör dışında bir yerde değiştirilmiş.</translation>
     </message>
     <message>
@@ -2135,7 +2135,7 @@
         <location filename="../../src/mainimpl.cpp" line="446"/>
         <source>The file &quot;%1&quot;
  already exists in directory.</source>
-        <translation>\&quot;%1\&quot; dosyası 
+        <translation>&quot;%1&quot; dosyası 
  dizin içinde zaten mevcut.</translation>
     </message>
     <message>
@@ -2162,7 +2162,7 @@
         <location filename="../../src/mainimpl.cpp" line="2241"/>
         <source>The variable &quot;%1&quot;
  already exists.</source>
-        <translation>\&quot;%1\&quot; değişkeni 
+        <translation>&quot;%1&quot; değişkeni 
  zaten mevcut.</translation>
     </message>
     <message>
@@ -2409,7 +2409,7 @@
     <message>
         <location filename="../../src/newprojectimpl.cpp" line="120"/>
         <source>The directory &quot;%1&quot; cannot be created</source>
-        <translation>\&quot;%1\&quot; dizini oluşturulamadı</translation>
+        <translation>&quot;%1&quot; dizini oluşturulamadı</translation>
     </message>
     <message>
         <location filename="../../src/newprojectimpl.cpp" line="129"/>
@@ -2648,7 +2648,7 @@
     <message>
         <location filename="../../ui/options.ui" line="880"/>
         <source>Function/method names :</source>
-        <translation>İşlev/yöntem adları :</translation>
+        <translation>İşlev/metod adları :</translation>
     </message>
     <message>
         <location filename="../../ui/options.ui" line="887"/>
@@ -2854,7 +2854,7 @@
     <message>
         <location filename="../../src/parametersimpl.cpp" line="114"/>
         <source>The variable &quot;%1&quot; already exists.</source>
-        <translation>\&quot;%1\&quot; değişkeni zaten mevcut.</translation>
+        <translation>&quot;%1&quot; değişkeni zaten mevcut.</translation>
     </message>
 </context>
 <context>
@@ -2972,17 +2972,17 @@
     <message>
         <location filename="../../src/projectmanager.cpp" line="1249"/>
         <source>The file  &quot;%1&quot; </source>
-        <translation>\&quot;%1\&quot; dosyası </translation>
+        <translation>&quot;%1&quot; dosyası </translation>
     </message>
     <message>
         <location filename="../../src/projectmanager.cpp" line="1325"/>
         <source>Unable to rename the file to &quot;%1&quot;</source>
-        <translation>Dosya \&quot;%1\&quot; olarak yeniden adlandırılamadı</translation>
+        <translation>Dosya &quot;%1&quot; olarak yeniden adlandırılamadı</translation>
     </message>
     <message>
         <location filename="../../src/projectmanager.cpp" line="2015"/>
         <source>The only available version for &quot;%1&quot; is %2</source>
-        <translation>\&quot;%1\&quot; için tek uygun sürüm %2</translation>
+        <translation>&quot;%1&quot; için tek uygun sürüm %2</translation>
     </message>
     <message>
         <location filename="../../src/projectmanager.cpp" line="2016"/>
@@ -2997,7 +2997,7 @@
     <message>
         <location filename="../../src/projectmanager.cpp" line="1347"/>
         <source>Do you want to remove &quot;%1&quot; from the project ?</source>
-        <translation>\&quot;%1\&quot; adlı öğeyi projeden silmek istiyor musunuz?</translation>
+        <translation>&quot;%1&quot; adlı öğeyi projeden silmek istiyor musunuz?</translation>
     </message>
 </context>
 <context>
@@ -3763,12 +3763,12 @@
     <message>
         <location filename="../../src/textEdit.cpp" line="435"/>
         <source>The file &quot;%1&quot; could not be loaded.</source>
-        <translation>\&quot;%1\&quot; dosyası yüklenemedi.</translation>
+        <translation>&quot;%1&quot; dosyası yüklenemedi.</translation>
     </message>
     <message>
         <location filename="../../src/textEdit.cpp" line="665"/>
         <source>Save &quot;%1&quot;</source>
-        <translation>\&quot;%1\&quot; Dosyasını Kaydet</translation>
+        <translation>&quot;%1&quot; Dosyasını Kaydet</translation>
     </message>
     <message>
         <location filename="../../src/textEdit.cpp" line="684"/>
@@ -3889,7 +3889,7 @@
     <message>
         <location filename="../../src/treeclasses.cpp" line="489"/>
         <source>Add Method...</source>
-        <translation>Yöntem Ekle...</translation>
+        <translation>Metod Ekle...</translation>
     </message>
     <message>
         <location filename="../../src/treeclasses.cpp" line="490"/>
@@ -3899,7 +3899,7 @@
     <message>
         <location filename="../../src/treeclasses.cpp" line="494"/>
         <source>Add get/set Methods...</source>
-        <translation>get/set Yöntemleri Ekle...</translation>
+        <translation>get/set Metodları Ekle...</translation>
     </message>
     <message>
         <location filename="../../src/treeclasses.cpp" line="501"/>
