--- libkdepim/Makefile.am
+++ libkdepim/Makefile.am	2005/02/07 16:51:07
@@ -45,7 +45,7 @@
                        $(LIB_KIO) $(LIB_KABC) $(LIB_KPARTS) $(LIB_KIMPROXY) $(LIB_POLL) -lqui
 #               $top_builddir)/libkdepim/resources/libkpimresources.la
 
-include_HEADERS = kdepimmacros.h
+include_HEADERS = kdepimmacros.h kpimprefs.h
 
 AM_CXXFLAGS = -DQT_PLUGIN
 kde_widget_LTLIBRARIES = kdepimwidgets.la
--- networkstatus/Makefile.am
+++ networkstatus/Makefile.am	2005/02/07 16:52:10
@@ -3,7 +3,7 @@
 METASOURCES = AUTO
 
 kde_module_LTLIBRARIES = kded_networkstatus.la 
-noinst_LTLIBRARIES = libnetworkstatus.la libconnectionmanager.la
+lib_LTLIBRARIES = libnetworkstatus.la libconnectionmanager.la
 
 libnetworkstatus_la_LIBADD = $(LIB_KDECORE)
 libnetworkstatus_la_LDFLAGS = $(all_libraries)
@@ -24,8 +24,10 @@
 servicesdir = $(kde_servicesdir)/kded
 services_DATA = networkstatus.desktop
 
-noinst_HEADERS = serviceiface.h networkstatuscommon.h serviceifaceimpl.h \
-	network.h clientifaceimpl.h provideriface.h testservice.h connectionmanager.h
+noinst_HEADERS = serviceifaceimpl.h \
+	network.h clientifaceimpl.h testservice.h connectionmanager.h
+
+include_HEADERS = serviceiface.h provideriface.h networkstatuscommon.h
 
 bin_PROGRAMS = networkstatustestservice
 
