diff -ur dbus-0.62.orig/bus/signals.c dbus-0.62/bus/signals.c
--- dbus-0.62.orig/bus/signals.c	2005-11-22 22:37:00.000000000 +0200
+++ dbus-0.62/bus/signals.c	2006-12-13 00:18:16.000000000 +0200
@@ -1067,6 +1067,9 @@
   if (a->flags != b->flags)
     return FALSE;
 
+  if (a->matches_go_to != b->matches_go_to)
+    return FALSE;
+
   if ((a->flags & BUS_MATCH_MESSAGE_TYPE) &&
       a->message_type != b->message_type)
     return FALSE;
