--- Doxyfile.in
+++ Doxyfile.in	2005/01/20 14:15:58
@@ -49,8 +49,8 @@
 #---------------------------------------------------------------------------
 # configuration options related to the input files
 #---------------------------------------------------------------------------
-INPUT                  = dbus glib
-FILE_PATTERNS          = *.c *.h
+INPUT                  = dbus glib mono qt gcj
+FILE_PATTERNS          = *.c *.h *.cs *.cpp *.java
 RECURSIVE              = YES
 #EXCLUDE                = test
 
