diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kdelibs-4.2.2-orig/kdeui/kernel/kglobalsettings.h kdelibs-4.2.2/kdeui/kernel/kglobalsettings.h
--- kdelibs-4.2.2-orig/kdeui/kernel/kglobalsettings.h	2009-04-12 06:44:57.000000000 +0300
+++ kdelibs-4.2.2/kdeui/kernel/kglobalsettings.h	2009-04-12 07:15:01.000000000 +0300
@@ -24,7 +24,7 @@
 #include <QtCore/QString>
 #include <QtGui/QPalette>
 
-#define KDE_DEFAULT_SINGLECLICK true
+#define KDE_DEFAULT_SINGLECLICK false
 #define KDE_DEFAULT_SMOOTHSCROLL true
 #define KDE_DEFAULT_INSERTTEAROFFHANDLES 0
 #define KDE_DEFAULT_AUTOSELECTDELAY -1
