--- data/desktop/xlog.desktop.orig	2011-07-11 14:28:27.926006925 +0300
+++ data/desktop/xlog.desktop	2011-07-11 14:33:13.306006925 +0300
@@ -1,12 +1,13 @@
 [Desktop Entry]
 Name=Xlog
-Comment=Amateur Radio logging program
-Comment[nl]=Log programma voor zendamateurs
-Comment[pl]=Dziennik łączności dla radioamatorów
+GenericName=Amateur Radio logging program
+GenericName[tr]=Amatör Telsiz QSO Kayıt Uygulaması
+GenericName[nl]=Log programma voor zendamateurs
+GenericName[pl]=Dziennik łączności dla radioamatorów
 TryExec=xlog
 Exec=xlog %F
-Icon=xlog-icon.png
+Icon=xlog-icon
 Terminal=false
 Type=Application
-Categories=Utility;Database;HamRadio;GTK
-MimeType=text/x-xlog
+Categories=Network;HamRadio;
+MimeType=text/x-xlog;
