diff -up kdenetwork-4.2.0/kopete/protocols/jabber/jabbercontact.cpp.handle-enc-message kdenetwork-4.2.0/kopete/protocols/jabber/jabbercontact.cpp
--- kdenetwork-4.2.0/kopete/protocols/jabber/jabbercontact.cpp.handle-enc-message	2008-11-19 11:19:43.000000000 +0100
+++ kdenetwork-4.2.0/kopete/protocols/jabber/jabbercontact.cpp	2009-01-26 14:12:35.000000000 +0100
@@ -51,6 +51,7 @@
 #include "kopetechatsessionmanager.h"
 #include "kopeteaccountmanager.h"
 #include "kopetemetacontact.h"
+#include "kopetepluginmanager.h"
 #include "jabberprotocol.h"
 #include "jabberaccount.h"
 #include "jabberclient.h"
@@ -392,9 +393,15 @@ void JabberContact::handleIncomingMessag
 		QString body = message.body ();
 		if( !message.xencrypted().isEmpty() )
 		{
-			body = QString ("-----BEGIN PGP MESSAGE-----\n\n") + message.xencrypted () + QString ("\n-----END PGP MESSAGE-----\n");
+			kDebug ( JABBER_DEBUG_GLOBAL ) << "Received encrypted message";
+			if (Kopete::PluginManager::self()->plugin("kopete_cryptography"))
+			{
+				kDebug( JABBER_DEBUG_GLOBAL ) << "Kopete cryptography plugin loaded";
+				body = QString ("-----BEGIN PGP MESSAGE-----\n\n") + message.xencrypted () + QString ("\n-----END PGP MESSAGE-----\n");
+			}
 		}
-		else if( message.containsHTML() )
+
+		if( message.containsHTML() )
 		{
 			kDebug ( JABBER_DEBUG_GLOBAL ) << "Received a xHTML message";
 			newMessage = new Kopete::Message ( this, contactList );
diff -up kdenetwork-4.2.0/kopete/protocols/jabber/jabbergroupcontact.cpp.handle-enc-message kdenetwork-4.2.0/kopete/protocols/jabber/jabbergroupcontact.cpp
--- kdenetwork-4.2.0/kopete/protocols/jabber/jabbergroupcontact.cpp.handle-enc-message	2008-10-09 11:48:21.000000000 +0200
+++ kdenetwork-4.2.0/kopete/protocols/jabber/jabbergroupcontact.cpp	2009-01-26 14:12:35.000000000 +0100
@@ -32,6 +32,7 @@
 #include "jabbergroupmembercontact.h"
 #include "jabbercontactpool.h"
 #include "kopetemetacontact.h"
+#include "kopetepluginmanager.h"
 #include "xmpp_tasks.h"
 
 /**
@@ -181,7 +182,11 @@ void JabberGroupContact::handleIncomingM
 
 		if( !message.xencrypted().isEmpty () )
 		{
-			body = QString ("-----BEGIN PGP MESSAGE-----\n\n") + message.xencrypted () + QString ("\n-----END PGP MESSAGE-----\n");
+			if (Kopete::PluginManager::self()->plugin("kopete_cryptography"))
+			{
+				kDebug( JABBER_DEBUG_GLOBAL ) << "Kopete cryptography plugin loaded";
+				body = QString ("-----BEGIN PGP MESSAGE-----\n\n") + message.xencrypted () + QString ("\n-----END PGP MESSAGE-----\n");
+			}
 		}
 
 		// locate the originating contact
