--- Makefile	2004-03-05 15:29:56.000000000 +0200
@@ -3,15 +3,15 @@
 #
 
 # Where does the fortune program go?
-FORTDIR=$(prefix)/usr/games
+FORTDIR=$(prefix)/usr/bin
 # Where do the data files (fortunes, or cookies) go?
-COOKIEDIR=$(prefix)/usr/share/games/fortunes
+COOKIEDIR=$(prefix)/usr/share/fortunes
 # Offensive ones?
 OCOOKIEDIR=$(COOKIEDIR)/off
 # The ones with html tags?
 WCOOKIEDIR=$(COOKIEDIR)/html
 # Where do local data files go?
-LOCALDIR=$(prefix)/usr/local/share/games/fortunes
+LOCALDIR=$(prefix)/usr/share/fortunes
 # Offensive ones?
 LOCALODIR=$(LOCALDIR)/off
 # With HTML tags?
