diff -ur pavucontrol-0.9.6.orig/src/pavucontrol.desktop pavucontrol-0.9.6/src/pavucontrol.desktop
--- pavucontrol-0.9.6.orig/src/pavucontrol.desktop	2006-06-21 02:44:05.000000000 +0300
+++ pavucontrol-0.9.6/src/pavucontrol.desktop	2008-05-06 15:03:13.000000000 +0300
@@ -5,7 +5,7 @@
 GenericName=Volume Control
 Comment=Adjust the volume level
 Exec=pavucontrol
-Icon=multimedia-volume-control
+Icon=artsenvironment
 StartupNotify=true
 Type=Application
 Categories=Application;AudioVideo;Audio;
diff -ur pavucontrol-0.9.6.orig/src/pavucontrol.glade pavucontrol-0.9.6/src/pavucontrol.glade
--- pavucontrol-0.9.6.orig/src/pavucontrol.glade	2007-10-30 16:28:19.000000000 +0200
+++ pavucontrol-0.9.6/src/pavucontrol.glade	2008-05-06 15:04:15.000000000 +0300
@@ -6,7 +6,7 @@
     <property name="title" translatable="yes">PulseAudio Volume Control</property>
     <property name="default_width">500</property>
     <property name="default_height">500</property>
-    <property name="icon_name">multimedia-volume-control</property>
+    <property name="icon_name">artsenvironment</property>
     <child>
       <widget class="GtkVBox" id="vbox3">
         <property name="visible">True</property>
@@ -22,7 +22,7 @@
                   <widget class="GtkImage" id="image19">
                     <property name="visible">True</property>
                     <property name="icon_size">6</property>
-                    <property name="icon_name">multimedia-volume-control</property>
+                    <property name="icon_name">artsenvironment</property>
                   </widget>
                   <packing>
                     <property name="expand">False</property>
@@ -545,7 +545,7 @@
                               <widget class="GtkImage" id="image18">
                                 <property name="visible">True</property>
                                 <property name="icon_size">1</property>
-                                <property name="icon_name">stock_lock</property>
+                                <property name="icon_name">lock</property>
                               </widget>
                             </child>
                           </widget>
@@ -566,7 +566,7 @@
                               <widget class="GtkImage" id="image20">
                                 <property name="visible">True</property>
                                 <property name="icon_size">1</property>
-                                <property name="icon_name">audio-volume-muted</property>
+                                <property name="icon_name">kmixdocked_mute</property>
                               </widget>
                             </child>
                           </widget>
