diff -Nur foo2zjs-old/foo2zjs-wrapper.in foo2zjs/foo2zjs-wrapper.in
--- foo2zjs-old/foo2zjs-wrapper.in	2006-09-21 22:41:23.682294576 +0300
+++ foo2zjs/foo2zjs-wrapper.in	2006-09-21 22:48:57.360325032 +0300
@@ -339,7 +339,7 @@
 case "$PAPER" in
 1|letter)	PAPER=1;	paper=letter;    XDIM="10200"; YDIM="6600"
 		case "$MODEL" in
-		0)	set_clipping 204 102	204 106;;
+		0)	set_clipping 108 94	108 98;;
 		1)	set_clipping 192 96	192 96;;
 		# 1)	set_clipping 96 96	288 96;;
 		esac
