Index: kbluetooth-RC1/src/kbluetooth.desktop
===================================================================
--- kbluetooth-RC1.orig/src/kbluetooth.desktop
+++ kbluetooth-RC1/src/kbluetooth.desktop
@@ -19,7 +19,7 @@ GenericName[pt]=Bluetooth do KDE
 GenericName[ro]=KDE Bluetooth
 GenericName[ru]=KDE Bluetooth
 GenericName[sv]=KDE Blåtand
-GenericName[tr]=KDE Bluetooth
+GenericName[tr]=KDE Bluetooth Altyapısı
 GenericName[uk]=Bluetooth для KDE
 GenericName[x-test]=xxKDE Bluetoothxx
 X-DBUS-StartupType=Unique
