Index: kbluetooth4-0.3_20090504/src/obexsession/obexsessionfiletransfer.cpp
===================================================================
--- kbluetooth4-0.3_20090504.orig/src/obexsession/obexsessionfiletransfer.cpp
+++ kbluetooth4-0.3_20090504/src/obexsession/obexsessionfiletransfer.cpp
@@ -64,5 +64,7 @@ void ObexSessionFileTransfer::slotTransf
 void ObexSessionFileTransfer::slotTransferCompleted() {
 	kDebug() << "Transfer completed";
 	setProcessedAmount(Bytes, m_totalFileSize);
+    m_dbusSession->call("Disconnect");
+    m_dbusSession->call("Close");
 	emitResult();
 }
