--- src/sm_mainview.py.orig	2007-07-09 15:34:31.000000000 +0300
+++ src/sm_mainview.py	2008-03-23 22:43:54.000000000 +0200
@@ -120,6 +120,18 @@
         self.connect(self.radioAutoRun, SIGNAL('clicked()'), self.slotOn)
         self.connect(self.radioNoAutoRun, SIGNAL('clicked()'), self.slotOff)
 
+        self.connect(self.listServices, SIGNAL('doubleClicked(QListViewItem*)'), self.slotDoubleClicked)
+
+    # start or stop service when the item is double clicked
+    def slotDoubleClicked(self, item):
+        # if it's not started
+        if not item.state:
+            self.comar.call_package('System.Service.start', item.package, id=SERVICE_START)
+            self.updateItemStatus(item)
+        else:
+            self.comar.call_package('System.Service.stop', item.package, id=SERVICE_STOP)
+            self.updateItemStatus(item)
+
     def slotComar(self, sock):
         reply = self.comar.read_cmd()
         if reply.command == 'notify':
