--- data/fldigi.desktop.orig	2011-07-11 14:06:13.393006924 +0300
+++ data/fldigi.desktop	2011-07-11 14:07:52.094006925 +0300
@@ -1,7 +1,9 @@
 [Desktop Entry]
 Name=Fldigi
 GenericName=Amateur Radio Digital Modem
+GenericName[tr]=Amatör Telsiz Dijital Modem
 Comment=Amateur Radio Sound Card Communications
+Comment[tr]=Amatör Telsiz Ses Kartı ile Haberleşme
 Exec=fldigi
 Icon=fldigi
 Terminal=false
