From 78e06ee5680e9d73598031d96dd1784ea600a290 Mon Sep 17 00:00:00 2001
From: Brian Paul <brian.paul@tungstengraphics.com>
Date: Fri, 5 Sep 2008 08:07:16 -0600
Subject: [PATCH] mesa: replace MALLOC w/ CALLOC to fix memory error in glPushClientAttrib()

---
 src/mesa/main/attrib.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git Mesa-7.0.4/src/mesa/main/attrib.c Mesa-7.0.4/src/mesa/main/attrib.c
index 9440826..c2d63f3 100644
--- Mesa-7.0.4/src/mesa/main/attrib.c
+++ Mesa-7.0.4/src/mesa/main/attrib.c
@@ -1334,7 +1334,7 @@ _mesa_PushClientAttrib(GLbitfield mask)
       newnode->next = head;
       head = newnode;
       /* unpacking attribs */
-      attr = MALLOC_STRUCT( gl_pixelstore_attrib );
+      attr = CALLOC_STRUCT( gl_pixelstore_attrib );
       copy_pixelstore(ctx, attr, &ctx->Unpack);
       newnode = new_attrib_node( GL_CLIENT_UNPACK_BIT );
       newnode->data = attr;
-- 
1.6.0.2

