--- libstage/CMakeLists.txt.orig	2011-06-08 15:16:17.912000064 +0300
+++ libstage/CMakeLists.txt	2011-06-08 15:17:47.659000064 +0300
@@ -85,9 +85,9 @@
 # to explicitly link on Linux
 
 target_link_libraries( stagebinary stage )
-set_target_properties( stagebinary PROPERTIES LINK_FLAGS "${FLTK_LDFLAGS}" )
 
 IF(PROJECT_OS_LINUX)
+  set_target_properties( stagebinary PROPERTIES LINK_FLAGS "${FLTK_LDFLAGS}" )
   target_link_libraries( stagebinary stage pthread )
 ENDIF(PROJECT_OS_LINUX)
 
--- libstage/CMakeLists.txt.orig	2011-06-08 16:18:40.355000064 +0300
+++ libstage/CMakeLists.txt	2011-06-08 16:20:25.074000064 +0300
@@ -60,7 +60,7 @@
            LINK_FLAGS "${FLTK_LDFLAGS}"	  
 )
 
-target_link_libraries( stage ${LTDL_LIB} )
+target_link_libraries( stage ${LTDL_LIB} ${FLTK_LDFLAGS} )  
 
 set( stagebinarySrcs main.cc )
 set_source_files_properties( ${stagebinarySrcs} PROPERTIES COMPILE_FLAGS "${FLTK_CFLAGS}" )
