--- Objects/stringobject.c.orig	2008-04-12 16:44:13.000000000 +0300
+++ Objects/stringobject.c	2008-04-12 16:46:27.000000000 +0300
@@ -52,6 +52,11 @@
 PyString_FromStringAndSize(const char *str, int size)
 {
 	register PyStringObject *op;
+    if (size < 0) {
+        PyErr_SetString(PyExc_SystemError,
+                "Negative size passed to PyString_FromStringAndSize");
+        return NULL;
+    }
 	if (size == 0 && (op = nullstring) != NULL) {
 #ifdef COUNT_ALLOCS
 		null_strings++;
