diff -ur ufraw-0.7.orig/dcraw.c ufraw-0.7/dcraw.c
--- ufraw-0.7.orig/dcraw.c	2006-04-30 22:35:19.000000000 +0300
+++ ufraw-0.7/dcraw.c	2006-04-30 22:35:33.000000000 +0300
@@ -44,7 +44,7 @@
 #include <jpeglib.h>
 #endif
 #ifndef NO_LCMS
-#include <lcms.h>
+#include <lcms/lcms.h>
 #endif
 
 #ifdef __CYGWIN__
Yalnızca ufraw-0.7'da: dcraw.c~
diff -ur ufraw-0.7.orig/ufraw_developer.c ufraw-0.7/ufraw_developer.c
--- ufraw-0.7.orig/ufraw_developer.c	2006-04-30 22:35:19.000000000 +0300
+++ ufraw-0.7/ufraw_developer.c	2006-04-30 22:35:46.000000000 +0300
@@ -14,7 +14,7 @@
 
 #include <stdio.h>
 #include <math.h>
-#include <lcms.h>
+#include <lcms/lcms.h>
 #include <glib.h>
 #include "nikon_curve.h"
 #include "ufraw.h"
Yalnızca ufraw-0.7'da: ufraw_developer.c~
