--- a/src/output-plugins/spo_platypus.c	2009-07-19 22:07:50.547283102 +0200
+++ b/src/output-plugins/spo_platypus.c	2009-07-19 22:28:59.837282220 +0200
@@ -298,14 +298,14 @@
             break;
 #ifdef SUP_IP6
         case UNIFIED2_IDS_EVENT_IPV6:
-            inet_ntop(AF_INET6, &((Unified2IDSEvent6_legacy *)event)->ip_source, sip6, INET6_ADDRSTRLEN);
-            inet_ntop(AF_INET6, &((Unified2IDSEvent6_legacy *)event)->ip_destination, dip6, INET6_ADDRSTRLEN);
+            inet_ntop(AF_INET6, &((Unified2IDSEventIPv6_legacy *)event)->ip_source, sip6, INET6_ADDRSTRLEN);
+            inet_ntop(AF_INET6, &((Unified2IDSEventIPv6_legacy *)event)->ip_destination, dip6, INET6_ADDRSTRLEN);
 
             SnortSnprintfAppend(evt_msg, MAX_MSG_LEN, "6|%s|%u|%s|%u|%u|",
                 sip6,
-                ntohs(((Unified2IDSEvent6_legacy *)event)->sport_itype),
+                ntohs(((Unified2IDSEventIPv6_legacy *)event)->sport_itype),
                 dip6,
-                ntohs(((Unified2IDSEvent6_legacy *)event)->dport_icode),
+                ntohs(((Unified2IDSEventIPv6_legacy *)event)->sport_itype),
-                ((Unified2IDSEvent6_legacy *)event)->protocol);
+                ((Unified2IDSEventIPv6_legacy *)event)->protocol);
             break;
 #endif
