--- SDL-1.2.11/sdl.pc.in.no_lpthread	2006-06-20 07:49:00.000000000 +0200
+++ SDL-1.2.11/sdl.pc.in	2006-07-24 18:20:46.000000000 +0200
@@ -10,5 +10,5 @@ Description: Simple DirectMedia Layer is
 Version: @SDL_VERSION@
 Requires:
 Conflicts:
-Libs: -L${libdir} @SDL_RLD_FLAGS@ @SDL_LIBS@
+Libs: -L${libdir} @SDL_RLD_FLAGS@ -lSDL
 Cflags: -I${includedir}/SDL @SDL_CFLAGS@
--- SDL-1.2.11/sdl-config.in.no_lpthread	2006-07-24 18:20:46.000000000 +0200
+++ SDL-1.2.11/sdl-config.in	2006-07-24 18:22:21.000000000 +0200
@@ -44,7 +44,7 @@ while test $# -gt 0; do
       echo -I@includedir@/SDL @SDL_CFLAGS@
       ;;
 @ENABLE_SHARED_TRUE@    --libs)
-@ENABLE_SHARED_TRUE@      echo -L@libdir@ @SDL_LIBS@
+@ENABLE_SHARED_TRUE@      echo -L@libdir@ -lSDL
 @ENABLE_SHARED_TRUE@      ;;
 @ENABLE_STATIC_TRUE@@ENABLE_SHARED_TRUE@    --static-libs)
 @ENABLE_STATIC_TRUE@@ENABLE_SHARED_FALSE@    --libs|--static-libs)
