diff -Nur glest_source_2.0.0-old/source/shared_lib/include/graphics/particle.h glest_source_2.0.0/source/shared_lib/include/graphics/particle.h
--- glest_source_2.0.0-old/source/shared_lib/include/graphics/particle.h	2006-04-25 14:07:53.000000000 +0300
+++ glest_source_2.0.0/source/shared_lib/include/graphics/particle.h	2006-04-25 14:08:49.000000000 +0300
@@ -310,7 +310,7 @@
 	void setTrajectoryFrequency(float trajectoryFrequency)	{this->trajectoryFrequency= trajectoryFrequency;}
 	void setPath(Vec3f startPos, Vec3f endPos);
 
-	static Trajectory ProjectileParticleSystem::strToTrajectory(const string &str);
+	static Trajectory strToTrajectory(const string &str);
 };
 
 // =====================================================
