--- wxmaxima.desktop.orig	2009-01-06 01:02:24.000000000 +0200
+++ wxmaxima.desktop	2009-01-06 01:35:18.000000000 +0200
@@ -1,9 +1,12 @@
 [Desktop Entry]
 Name=wxMaxima
+Name[tr]=wxMaxima
 Comment=Perform symbolic and numeric calculations using Maxima
+Comment[tr]=Maxima grafik arayüzü
 Exec=wxmaxima
-Icon=wxmaxima.png
-Terminal=0
+Icon=wxmaxima
+Terminal=false
+StartupNotify=false
 Type=Application
 Encoding=UTF-8
-Categories=Application;Utility;X-Red-Hat-Base;X-Red-Hat-Base-Only;
\ Dosya sonunda yenisatır yok.
+Categories=Application;Science;Mathematics;
