From 7aca689ce14d314b5c8c72c8df76f53f76ab467c Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@redhat.com>
Date: Fri, 14 Nov 2008 10:32:50 +1000
Subject: [PATCH] Add XF86Suspend, XF86Hibernate to KeysymDB.

---
 src/XKeysymDB |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/XKeysymDB b/src/XKeysymDB
index e6fec07..5e66882 100644
--- a/src/XKeysymDB
+++ b/src/XKeysymDB
@@ -351,6 +351,8 @@ XF86Red			:1008FFA3
 XF86Green		:1008FFA4
 XF86Yellow		:1008FFA5
 XF86Blue             	:1008FFA6
+XF86Suspend		:1008FFA7
+XF86Hibernate		:1008FFA8
 
 ! XFree86 special action keys
 XF86_Switch_VT_1	:1008FE01
-- 
1.6.0.3

