--- m4/qt.m4	2006-05-02 07:03:46.000000000 +0300
+++ m4/qt.m4	2007-02-24 18:45:55.000000000 +0200
@@ -99,8 +99,8 @@
 [
 have_qt4=no
 dnl Search paths for Qt4 - not much real experience with this yet.
-qt4_incdirs="/usr/local/qt/include /usr/include/qt4 /usr/include/qt /usr/include /usr/X11R6/include/X11/qt /usr/X11R6/include/qt /usr/lib/qt/include /usr/lib/qt4/include"
-qt4_libdirs="/usr/local/qt/lib /usr/lib/qt /usr/lib /usr/X11R6/lib/X11/qt /usr/X11R6/lib/qt /usr/lib/qt/lib /usr/lib/qt4/lib"
+qt4_incdirs="/usr/qt/4/include /usr/local/qt/include /usr/include/qt4 /usr/include/qt /usr/include /usr/X11R6/include/X11/qt /usr/X11R6/include/qt /usr/lib/qt/include /usr/lib/qt4/include"
+qt4_libdirs="/usr/qt/4/lib /usr/local/qt/lib /usr/lib/qt /usr/lib /usr/X11R6/lib/X11/qt /usr/X11R6/lib/qt /usr/lib/qt/lib /usr/lib/qt4/lib"
 
 if test -n "$QTDIR" ; then
     qt4_incdirs="$QTDIR/include $qt4_incdirs"
