https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=206813


--- gdb-6.5-8-src/gdb/valops.c-orig	2006-09-20 21:09:56.000000000 +0200
+++ gdb-6.5-8-src/gdb/valops.c	2006-09-20 21:16:19.000000000 +0200
@@ -1847,7 +1847,9 @@
   else
     {
       const char *qualified_name = SYMBOL_CPLUS_DEMANGLED_NAME (fsym);
-      func_name	= cp_func_name (qualified_name);
+
+      if (qualified_name != NULL)
+	func_name = cp_func_name (qualified_name);
 
       /* If the name is NULL this must be a C-style function.
          Just return the same symbol. */
