diff -ur amarok-1.4.9.1.orig/amarok/src/mediadevice/ipod/ipodmediadevice.cpp amarok-1.4.9.1/amarok/src/mediadevice/ipod/ipodmediadevice.cpp
--- amarok-1.4.9.1.orig/amarok/src/mediadevice/ipod/ipodmediadevice.cpp	2008-04-10 23:16:37.000000000 +0300
+++ amarok-1.4.9.1/amarok/src/mediadevice/ipod/ipodmediadevice.cpp	2008-04-15 18:07:40.000000000 +0300
@@ -72,6 +72,12 @@
 #define mediatype unk208
 #endif
 
+#ifndef HAVE_LIBGPOD_060
+#define ITDB_MEDIATYPE_AUDIO 0x01
+#define ITDB_MEDIATYPE_MOVIE 0x02
+#define ITDB_MEDIATYPE_PODCAST 0x04
+#define ITDB_MEDIATYPE_AUDIOBOOK 0x08
+#endif
 
 #include "metadata/audible/taglib_audiblefile.h"
 
