Index: git-1.6.0.1/gitk-git/gitk
===================================================================
--- git-1.6.0.1.orig/gitk-git/gitk
+++ git-1.6.0.1/gitk-git/gitk
@@ -9785,9 +9785,9 @@ if {$tclencoding == {}} {
     puts stderr "Warning: encoding $gitencoding is not supported by Tcl/Tk"
 }
 
-set mainfont {Helvetica 9}
-set textfont {Courier 9}
-set uifont {Helvetica 9 bold}
+set mainfont {{DejaVu Sans} 9}
+set textfont {{DejaVu Sans Mono} 9}
+set uifont {{DejaVu Sans} 9 bold}
 set tabstop 8
 set findmergefiles 0
 set maxgraphpct 50
