--- ppd/Makefile.old	2008-02-27 16:36:32.000000000 +0200
+++ ppd/Makefile	2008-02-27 16:36:40.000000000 +0200
@@ -65,23 +65,14 @@
 	install -d -m 755 ${CUPSPPD}/samsung
 	for filename in ${SAMSUNG}; do \
 		install -m 644 $$filename.${DRIVERSEXT} ${CUPSPPD}/samsung;\
-		for lang in ${LANGUAGES}; do \
-			install -m 644 $$filename$$lang.${DRIVERSEXT} ${CUPSPPD}/samsung;\
-		done; \
 	done; \
 	install -d -m 755 ${CUPSPPD}/xerox
 	for filename in ${XEROX}; do \
 		install -m 644 $$filename.${DRIVERSEXT} ${CUPSPPD}/xerox;\
-		for lang in ${LANGUAGES}; do \
-			install -m 644 $$filename$$lang.${DRIVERSEXT} ${CUPSPPD}/xerox;\
-		done; \
 	done; \
 	install -d -m 755 ${CUPSPPD}/dell
 	for filename in ${DELL}; do \
 		install -m 644 $$filename.${DRIVERSEXT} ${CUPSPPD}/dell;\
-		for lang in ${LANGUAGES}; do \
-			install -m 644 $$filename$$lang.${DRIVERSEXT} ${CUPSPPD}/dell;\
-		done; \
 	done;
 
 .PHONY: clean distclean
