Index: kdebase-3.5.10/konqueror/listview/konq_listview.cc
===================================================================
--- kdebase-3.5.10.orig/konqueror/listview/konq_listview.cc
+++ kdebase-3.5.10/konqueror/listview/konq_listview.cc
@@ -173,9 +173,8 @@ void ListViewBrowserExtension::rename()
          le->setSelection(0, txt.length()-pattern.stripWhiteSpace().length()+1);
      else
      {
-         int lastDot = txt.findRev('.');
-         if (lastDot > 0)
-             le->setSelection(0, lastDot);
+         // If the extension can't be determined, just highlight all (Fixes #6280)
+         le->setSelection(0, txt.length());
      }
   }
 }
