Index: bin/service.py
===================================================================
--- bin/service.py      (revision 15333)
+++ bin/service.py      (working copy)
@@ -37,7 +37,7 @@
         else:
             print _("Service doesn't provide this operation.")
     else:
-        print _("%(script)s error: %(data)s") % {"script": reply.script, "data": reply.data)
+        print _("%(script)s error: %(data)s") % {"script": reply.script, "data": reply.data}

 def collect(c):
     reply = c.read_cmd()
