diff -ur amarok-1.4.2.orig/amarok/src/amarokrc amarok-1.4.2/amarok/src/amarokrc
--- amarok-1.4.2.orig/amarok/src/amarokrc	2006-08-21 22:49:32.000000000 +0300
+++ amarok-1.4.2/amarok/src/amarokrc	2006-08-28 21:28:06.000000000 +0300
@@ -11,4 +11,4 @@
 [KNewStuff]
 TargetDir=amarok/themes
 Uncompress=application/x-gzip
-ProvidersUrl=http://amarok.kde.org/knewstuff/amarokthemes-providers.xml
+ProvidersUrl=http://sanat.pardus.org.tr/kde/amarokthemes-providers.xml
diff -ur amarok-1.4.2.orig/amarok/src/Options2.ui.h amarok-1.4.2/amarok/src/Options2.ui.h
--- amarok-1.4.2.orig/amarok/src/Options2.ui.h	2006-08-21 22:49:32.000000000 +0300
+++ amarok-1.4.2/amarok/src/Options2.ui.h	2006-08-28 21:28:33.000000000 +0300
@@ -118,7 +118,7 @@
     // you have to do this by hand when providing your own Engine
     KNS::ProviderLoader *p = new KNS::ProviderLoader( this );
     connect( p, SIGNAL( providersLoaded(Provider::List*) ), d, SLOT( slotProviders(Provider::List *) ) );
-    p->load( "amarok/theme", "http://amarok.kde.org/knewstuff/amarokthemes-providers.xml" );
+    p->load( "amarok/theme", "http://sanat.pardus.org.tr/kde/amarokthemes-providers.xml" );
 
     connect( d, SIGNAL( finished() ), d, SLOT( delayedDestruct() ) );
     connect( d, SIGNAL( finished() ), this, SLOT( updateStyleComboBox() ) );
