diff -ur konversation-1.0.1.orig/admin/cvs.sh konversation-1.0.1/admin/cvs.sh
--- konversation-1.0.1.orig/admin/cvs.sh	2006-10-06 19:43:29.000000000 +0300
+++ konversation-1.0.1/admin/cvs.sh	2008-05-21 01:00:24.000000000 +0300
@@ -316,7 +316,7 @@
    fi
 fi
 if test -z "$VERSION" || test "$VERSION" = "@VERSION@"; then
-     VERSION="\"3.5.9\""
+     VERSION="\"3.5.10\""
 fi
 if test -z "$modulename" || test "$modulename" = "@MODULENAME@"; then
    modulename=`pwd`; 
