--- nmm-0.9.1/configure.in	2005-11-06 22:08:55.000000000 +0200
+++ nmm-fixed/configure.in	2006-02-23 20:22:36.000000000 +0200
@@ -881,7 +881,7 @@
 	dnl live.com libs
 	NMM_CHECK_EXTPACKAGE(live,
                      [live/liveMedia.hh live/BasicUsageEnvironment.hh live/GroupsockHelper.hh],
-                     [liveMedia],
+                     [liveMedia BasicUsageEnvironment UsageEnvironment groupsock],
                      ,
                      RTPSTRATEGY,
 	[libliveMedia is part of the LIVE.COM Streaming Media project and can be found
