--- a/src/libLastFmTools/draglabel.cpp
+++ b/src/libLastFmTools/draglabel.cpp
@@ -560,7 +560,7 @@
         if ( index != -1 )
         {
             //QToolTip::showText( helpEvent->globalPos(), m_items[index].m_tooltip, this, m_hfwLayout.at( index ) );
-            QToolTip::showText( helpEvent->globalPos(), m_items[index].m_tooltip );
+            QToolTip::showText( helpEvent->globalPos(), m_items[index].m_tooltip, this );
         }
         else
         {
