Index: ffmpeg-gui/ffmpeg-gui.desktop
===================================================================
--- ffmpeg-gui.orig/ffmpeg-gui.desktop
+++ ffmpeg-gui/ffmpeg-gui.desktop
@@ -1,12 +1,15 @@
 [Desktop Entry]
 Name=ffmpeg-gui
+Name[tr]=ffmpeg-gui
 GenericName=ffmpeg frontend
+GenericName[tr]=ffmpeg arayüzü
 X-GNOME-FullName=ffmpeg-gui
 Comment=frontend for ffmpeg
+Comment[tr]=ffmpeg için yazılmış bir arayüz
 Categories=AudioVideo;Qt;
 Exec=/usr/bin/ffmpeg-gui
 X-KDE-StartupNotify=true
 StartupNotify=true
-Icon=icon128.png
+Icon=ffmpeg-gui.png
 Terminal=false
 Type=Application
