diff -Nur scummvm-0.13.1-old/Makefile.common scummvm-0.13.1/Makefile.common
--- scummvm-0.13.1-old/Makefile.common	2009-07-22 14:04:38.912530632 +0300
+++ scummvm-0.13.1/Makefile.common	2009-07-22 14:04:47.391905304 +0300
@@ -103,7 +103,7 @@
 
 ifdef HAVE_NASM
 %.o: %.asm
-	$(NASM) -O1 $(NASMFLAGS) -g -o $*.o $(<)
+	$(NASM) -O1 $(NASMFLAGS)  -o $*.o $(<)
 endif
 
 # Include the dependency tracking files.
