diff -Naur scummvm-1.0.0/config.log scummvm-1.0.0-mod/config.log
--- scummvm-1.0.0/config.log	1970-01-01 02:00:00.000000000 +0200
+++ scummvm-1.0.0-mod/config.log	2009-11-25 21:53:23.000000000 +0200
@@ -0,0 +1 @@
+Configure run on Wed Nov 25 21:53:23 EET 2009
diff -Naur scummvm-1.0.0/ports.mk scummvm-1.0.0-mod/ports.mk
--- scummvm-1.0.0/ports.mk	2009-11-07 02:10:17.000000000 +0200
+++ scummvm-1.0.0-mod/ports.mk	2009-11-25 22:07:53.000000000 +0200
@@ -10,7 +10,7 @@
 #
 install: all
 	$(INSTALL) -d "$(DESTDIR)$(BINDIR)"
-	$(INSTALL) -c -s -m 755 "./$(EXECUTABLE)" "$(DESTDIR)$(BINDIR)/$(EXECUTABLE)"
+	$(INSTALL) -c -m 755 "./$(EXECUTABLE)" "$(DESTDIR)$(BINDIR)/$(EXECUTABLE)"
 	$(INSTALL) -d "$(DESTDIR)$(MANDIR)/man6/"
 	$(INSTALL) -c -m 644 "$(srcdir)/dists/scummvm.6" "$(DESTDIR)$(MANDIR)/man6/scummvm.6"
 	$(INSTALL) -d "$(DESTDIR)$(PREFIX)/share/pixmaps/"
