--- src/db-artwork-writer.c	11 Nov 2006 13:44:29 -0000	1.25
+++ src/db-artwork-writer.c	24 Feb 2007 12:38:38 -0000	1.26
@@ -957,7 +957,7 @@
 		mhfd->unknown2 = get_gint32 (2, buffer->byte_order);
 		break;
 	case DB_TYPE_ITUNES:
-		mhfd->unknown2 = get_gint32 (1, buffer->byte_order);
+		mhfd->unknown2 = get_gint32 (2, buffer->byte_order);
 		break;
 	}
 	mhfd->next_id = get_gint32 (id_max, buffer->byte_order);
