diff -Nur transcode-1.1.0alpha7-old/configure.in transcode-1.1.0alpha7/configure.in
--- transcode-1.1.0alpha7-old/configure.in	2008-04-10 16:03:48.000000000 +0300
+++ transcode-1.1.0alpha7/configure.in	2008-04-10 16:05:42.000000000 +0300
@@ -1201,8 +1201,8 @@
 dnl
 IMAGEMAGICK_EXTRA_LIBS="$IMAGEMAGICK_EXTRA_LIBS -lz -lm"
 dnl MagickWand libraries/headers always (?) are a superset of MagickCore ones.
-TC_PKG_CHECK(imagemagick, no, IMAGEMAGICK, Wand-config, [magick/api.h],
- Magick, InitializeMagick, Wand, [http://www.imagemagick.org/])
+TC_PKG_CHECK(imagemagick, no, IMAGEMAGICK, Wand-config, [magick/api.h], 
+ MagickCore, InitializeMagick, Wand, [http://www.imagemagick.org/])
 TC_PKG_HAVE(imagemagick, IMAGEMAGICK)
 if test x"$have_imagemagick" = x"yes" ; then
   AC_CHECK_HEADER([wand/magick-wand.h],
