Index: kdegraphics/kooka/Makefile.am
===================================================================
--- kdegraphics.orig/kooka/Makefile.am
+++ kdegraphics/kooka/Makefile.am
@@ -16,8 +16,8 @@ kooka_SOURCES =	main.cpp kooka.cpp kooka
 		kocrocrad.cpp
 # 		pagesetup.cpp 
 
-kooka_LDADD   = $(LIB_KFILE) -lkdeprint -lktexteditor $(LIBTIFF) $(top_builddir)/libkscan/libkscan.la $(KADMOS_LIB) $(LIB_KSPELL)
-kooka_LDFLAGS = $(KDE_RPATH) $(all_libraries)
+kooka_LDADD   = -lkdeprint -lktexteditor $(LIBTIFF) $(top_builddir)/libkscan/libkscan.la $(KADMOS_LIB) $(LIB_KSPELL)
+kooka_LDFLAGS = /usr/kde/3.5/lib/libkio.so /usr/kde/3.5/lib/libkdeprint.so /usr/kde/3.5/lib/libktexteditor.so /usr/kde/3.5/lib/libkspell.so $(KDE_RPATH) $(all_libraries)
 
 INCLUDES         = -I$(top_srcdir)/libkscan $(all_includes) $(LIBSANE_INCLUDES) $(KADMOS_INC)
 
