diff -Naur rapid-down.orig/data/rapid-down.desktop rapid-down/data/rapid-down.desktop
--- rapid-down.orig/data/rapid-down.desktop	2009-06-29 19:38:42.300026866 +0300
+++ rapid-down/data/rapid-down.desktop	2009-06-29 19:39:30.344027286 +0300
@@ -2,7 +2,8 @@
 Type=Application
 Name=Rapidshare Downloader
 GenericName=Rapidshare File Download
-Comment=Download file from YouTube
+GenericName[tr]=Rapidshare İndirme Yöneticisi
+Comment=Download file from Rapidshare.Com
 Icon=rapid-down
 Exec=rapid-down
 Terminal=false
