--- python/pykde4/sip/plasma/applet.sip.old	2008-12-11 01:52:33.000000000 +0200
+++ python/pykde4/sip/plasma/applet.sip	2008-12-11 01:56:03.000000000 +0200
@@ -33,6 +33,14 @@
     {
         setHasConfigurationInterface(hasInterface);
     }
+    static inline void static_public_setConfigurationRequired (Applet *applet, bool needsConfiguring, const QString& reason = QString()) {
+        ((AppletProtectedThunk *) applet ) -> public_setConfigurationRequired(needsConfiguring, reason);
+    };
+
+    inline void public_setConfigurationRequired(bool needsConfiguring, const QString& reason = QString())
+    {
+        setConfigurationRequired(needsConfiguring, reason);
+    }
 };
 }
 
@@ -164,6 +172,7 @@
 {
 public:
     static void static_public_setHasConfigurationInterface (Plasma::Applet *applet, bool hasInterface);
+    static void static_public_setConfigurationRequired (Plasma::Applet *applet, bool needsConfiguring, const QString& reason = QString());
 };
 
 };   // Plasma
