Index: heimdal-1.2.1/appl/gssmask/Makefile.am
===================================================================
--- heimdal-1.2.1.orig/appl/gssmask/Makefile.am
+++ heimdal-1.2.1/appl/gssmask/Makefile.am
@@ -8,5 +8,5 @@ gssmask_SOURCES = gssmask.c common.c com
 
 gssmaestro_SOURCES = gssmaestro.c common.c common.h protocol.h
 
-LDADD = $(top_builddir)/lib/gssapi/libgssapi.la $(LIB_roken)
+LDADD = $(top_builddir)/lib/gssapi/libgssapi.la $(LIB_roken) $(LIB_krb5)
 
