diff -Naur vlc-0.8.6i.orig/modules/access/rtsp/real.c vlc-0.8.6i/modules/access/rtsp/real.c
--- vlc-0.8.6i.orig/modules/access/rtsp/real.c	2008-07-08 23:59:23.000000000 +0300
+++ vlc-0.8.6i/modules/access/rtsp/real.c	2009-07-30 09:04:26.000000000 +0300
@@ -598,6 +598,7 @@
 
   int n;
   rmff_dump_pheader(ph, *buffer);
+  if (ph->length<12) return 0;
   n=rtsp_read_data(rtsp_session, *buffer + 12, ph->length - 12);
   return (n <= 0) ? 0 : n+12;
 }
