--- wordwarvi.c	2009-01-18 20:02:56.000000000 +0200
+++ wordwarvi.c.modified	2009-03-31 09:21:54.000000000 +0300
@@ -11454,7 +11454,7 @@
 	srandom(level.random_seed);
 	generate_terrain(&terrain);
 
-	add_buildings(&terrain);/* Some FreeBSD users report that */
+	/*add_buildings(&terrain); Some FreeBSD users report that */
 				/*add_buildings() causes crashes. */
 				/* Commenting this out on FreeBSD */
 				/* may help, but, no buildings. */
