diff -Naur rtl8187se-linux-2.6-1023-1118-2008-orig/rtl8185/r8180_core.c rtl8187se-linux-2.6-1023-1118-2008/rtl8185/r8180_core.c
--- rtl8187se-linux-2.6-1023-1118-2008-orig/rtl8185/r8180_core.c	2008-12-10 20:06:12.000000000 +0200
+++ rtl8187se-linux-2.6-1023-1118-2008/rtl8185/r8180_core.c	2008-12-10 20:07:38.000000000 +0200
@@ -103,6 +103,7 @@
                 .driver_data = 0,
         },
+#if 0
         {
                 .vendor = PCI_VENDOR_ID_BELKIN,
                 .device = 0x6001,
                 .subvendor = PCI_ANY_ID,
@@ -130,6 +131,7 @@
 		.subdevice = PCI_ANY_ID,
 		.driver_data = 4,
 	},
+#endif
         {
                 .vendor = 0,
                 .device = 0,
