diff -ur apparmor-parser-2.0.1.orig/rc.apparmor.slackware apparmor-parser-2.0.1/rc.apparmor.slackware
--- apparmor-parser-2.0.1.orig/rc.apparmor.slackware	2006-04-12 07:09:10.000000000 +0300
+++ apparmor-parser-2.0.1/rc.apparmor.slackware	2007-02-09 14:21:01.537624336 +0200
@@ -51,6 +51,11 @@
 	echo ": Failed."
 }
 
+function sd_log_skipped_msg() {
+	[ -n "$1" ] && echo -n $1
+	echo -e	"$rc_skipped"
+}
+
 function usage() {
     echo "Usage: $0 {start|stop|restart|try-restart|reload|force-reload|status|kill}"
 }
