--- build.xml	2008-01-14 14:00:34.000000000 +0200
+++ build.xml2	2008-01-14 14:10:48.000000000 +0200
@@ -80,7 +80,7 @@
   <!-- Xalan Java directories -->  
 <!--  <property name="bin.dir"                value="./bin"/> -->
   <property name="build.dir"              value="./build"/> 
-  <property name="lib.dir"                value="./lib"/>   
+  <property name="lib.dir"                value="/usr/share/java"/>   
   <property name="samples.dir"            value="./samples"/>   
   <property name="src.dir"                value="./src"/>  
   <property name="tools.dir"              value="./tools"/> 
