diff -Naur silvertree-0.2.1.orig/Makefile.in silvertree-0.2.1/Makefile.in
--- silvertree-0.2.1.orig/Makefile.in	2007-11-17 08:01:56.000000000 +0200
+++ silvertree-0.2.1/Makefile.in	2007-11-18 19:50:03.000000000 +0200
@@ -545,14 +545,20 @@
 model-sources/sketchup/cave-entrace.skp \
 model-sources/sketchup/tower.skp \
 \
+data/bandit-fort.cfg \
 data/big-cave.cfg \
 data/cave-small.cfg \
 data/character_generators.cfg \
+data/formulatest.cfg \
+data/fort.cfg \
+data/island.cfg \
 data/items.cfg \
+data/lighting.cfg \
 data/rules.cfg \
 data/scenario.cfg \
 data/skills.cfg \
 data/terrain.cfg \
+data/town-balmoro.cfg \
 data/town-felson.cfg \
 data/town-isolt.cfg \
 data/town-telfa.cfg \
