diff -Naur silvertree-0.2.1.orig/Makefile.in silvertree-0.2.1/Makefile.in
--- silvertree-0.2.1.orig/Makefile.in	2007-11-17 08:01:56.000000000 +0200
+++ silvertree-0.2.1/Makefile.in	2007-11-18 18:03:37.000000000 +0200
@@ -572,7 +572,6 @@
 cave-entrance.dae \
 colorCube.dae \
 fountain1.dae \
-FreeSans.ttf \
 house2.3d \
 house.3d \
 male-base.dae \
diff -Naur silvertree-0.2.1.orig/src/font.cpp silvertree-0.2.1/src/font.cpp
--- silvertree-0.2.1.orig/src/font.cpp	2007-11-17 08:01:44.000000000 +0200
+++ silvertree-0.2.1/src/font.cpp	2007-11-18 18:03:00.000000000 +0200
@@ -60,7 +60,7 @@
 	if(i != fonts.end()) {
 		return i->second;
 	} else {
-		font_ptr new_font(TTF_OpenFont("FreeSans.ttf",size),
+		font_ptr new_font(TTF_OpenFont("/usr/share/fonts/dejavu/DejaVuSerif.ttf",size),
 		                  release_font());
 		if(!new_font) {
 			std::cerr << "ttf open failed: " << TTF_GetError() << "\n";
