diff -Nuar pmount-0.9.23.orig/etc/Makefile.in pmount-0.9.23/etc/Makefile.in
--- pmount-0.9.23.orig/etc/Makefile.in	2011-08-09 15:51:40.988000104 +0300
+++ pmount-0.9.23/etc/Makefile.in	2011-08-09 15:53:00.961000104 +0300
@@ -376,8 +376,8 @@
 
 
 install-data-hook:
-	test -z $(DESTDIR)/etc || mkdir -p -- $(DESTDIR)/etc
-	$(INSTALL_DATA) $(srcdir)/pmount.allow $(DESTDIR)/etc/pmount.allow
+	test -z $(INSTALL_DIR)/etc || mkdir -p -- $(INSTALL_DIR)/etc
+	$(INSTALL_DATA) $(srcdir)/pmount.allow $(INSTALL_DIR)/etc/pmount.allow
 
 uninstall-hook:
 	rm -f $(DESTDIR)/etc/pmount.allow
diff -Nuar pmount-0.9.23.orig/src/Makefile.in pmount-0.9.23/src/Makefile.in
--- pmount-0.9.23.orig/src/Makefile.in	2011-08-09 15:51:40.989000106 +0300
+++ pmount-0.9.23/src/Makefile.in	2011-08-09 15:52:13.315000110 +0300
@@ -605,8 +605,8 @@
 
 install-data-hook:
 	test -z $(INSTALL_DIR) || mkdir -p -- $(INSTALL_DIR)
-	$(INSTALL_DATA) -o root -g root -m 4755 -D $(INSTALL_SRC)/pmount $(INSTALL_DIR)/pmount
-	$(INSTALL_DATA) -o root -g root -m 4755 -D $(INSTALL_SRC)/pumount $(INSTALL_DIR)/pumount
+	$(INSTALL_DATA) -m 4755 -D $(INSTALL_SRC)/pmount $(INSTALL_DIR)/pmount
+	$(INSTALL_DATA) -m 4755 -D $(INSTALL_SRC)/pumount $(INSTALL_DIR)/pumount
 
 uninstall-hook:
 	rm -f $(INSTALL_DIR)/pmount
