diff -p -up pmount-0.9.17/src/Makefile.am.no-user-change-during-install pmount-0.9.17/src/Makefile.am
--- pmount-0.9.17/src/Makefile.am.no-user-change-during-install	2008-01-21 17:39:37.000000000 +0000
+++ pmount-0.9.17/src/Makefile.am	2008-01-21 17:39:51.000000000 +0000
@@ -36,8 +36,8 @@ INSTALL_SRC = $(top_builddir)/src
 
 install-data-hook:
 	test -z $(INSTALL_DIR) || mkdir -p -- $(INSTALL_DIR)
-	$(INSTALL_DATA) -o root -g root -m 4755 -D $(INSTALL_SRC)/pmount $(INSTALL_DIR)/pmount
-	$(INSTALL_DATA) -o root -g root -m 4755 -D $(INSTALL_SRC)/pumount $(INSTALL_DIR)/pumount
+	$(INSTALL_DATA) -m 4755 -D $(INSTALL_SRC)/pmount $(INSTALL_DIR)/pmount
+	$(INSTALL_DATA) -m 4755 -D $(INSTALL_SRC)/pumount $(INSTALL_DIR)/pumount
 
 uninstall-hook:
 	rm -f $(INSTALL_DIR)/pmount
diff -p -up pmount-0.9.17/Makefile.am.no-user-change-during-install pmount-0.9.17/Makefile.am
