diff -uNBwp Lib/site.py.orig Lib/site.py
--- Lib/site.py.orig	2010-12-26 20:54:29.000000000 +0100
+++ Lib/site.py	2011-02-16 14:50:25.423000254 +0100
@@ -289,6 +289,7 @@ def getsitepackages():
                                         "python" + sys.version[:3],
                                         "site-packages"))
             sitepackages.append(os.path.join(prefix, "lib", "site-python"))
+            sitepackages.insert(0, os.path.join(prefix, "lib", "pardus"))
         else:
             sitepackages.append(prefix)
             sitepackages.append(os.path.join(prefix, "lib", "site-packages"))
