diff -uNBwp Lib/decimal.py.orig Lib/decimal.py
--- Lib/decimal.py.orig	2011-02-02 00:54:43.000000000 +0100
+++ Lib/decimal.py	2011-02-16 15:03:45.052999195 +0100
@@ -154,6 +154,12 @@ ROUND_UP = 'ROUND_UP'
 ROUND_HALF_DOWN = 'ROUND_HALF_DOWN'
 ROUND_05UP = 'ROUND_05UP'

+import string
+
+def ascii_upper(s):
+    trans_table = str.maketrans(string.ascii_lowercase, string.ascii_uppercase)
+    return s.translate(trans_table)
+
 # Errors

 class DecimalException(ArithmeticError):
@@ -3770,7 +3776,7 @@ rounding_functions = [name for name in D
                                     if name.startswith('_round_')]
 for name in rounding_functions:
     # name is like _round_half_even, goes to the global ROUND_HALF_EVEN value.
-    globalname = name[1:].upper()
+    globalname = ascii_upper(name[1:])
     val = globals()[globalname]
     Decimal._pick_rounding_function[val] = name

