--- /var/pisi/faac-1.25-4/work/faac/configure.in	2006-08-13 17:17:26.000000000 +0300
+++ configure.in	2007-01-23 21:14:07.686074533 +0200
@@ -66,6 +66,4 @@
 	 #include <netinet/in.h>])
 AC_CHECK_MEMBERS(fpos_t.__pos,,, [#include <stdio.h>])
 
-AC_OUTPUT(common/Makefile common/mp4v2/Makefile \
-          libfaac/Makefile frontend/Makefile \
-          include/Makefile Makefile)
+AC_OUTPUT(common/Makefile common/mp4v2/Makefile libfaac/Makefile frontend/Makefile include/Makefile Makefile)
