--- Makefile.in	2005-10-07 17:03:22.000000000 +0300
+++ Makefile.in.new	2007-08-23 17:57:04.000000000 +0300
@@ -832,19 +832,19 @@
 
 
 install-data-hook: 
-	-export GNOME_MAGIC=/etc/gnome-vfs-mime-magic ; \
-	if [ -f $$GNOME_MAGIC ]; then \
-	if ! grep "TRUEVISION" $$GNOME_MAGIC; then \
-	echo -e "0\tstring\t\tTRUEVISION SCENE\t\t\tapplication/x-truevision" >> $$GNOME_MAGIC; \
-	fi \
-	fi ; \
+#	-export GNOME_MAGIC=/etc/gnome-vfs-mime-magic ; \
+#	if [ -f $$GNOME_MAGIC ]; then \
+#	if ! grep "TRUEVISION" $$GNOME_MAGIC; then \
+#	echo -e "0\tstring\t\tTRUEVISION SCENE\t\t\tapplication/x-truevision" >> $$GNOME_MAGIC; \
+#	fi \
+#	fi ; 
 	export DEFAULTS=$(prefix)/share/applications/defaults.list ; \
 	if [ -f $$DEFAULTS ]; then \
 	if ! grep "truevision" $$DEFAULTS; then \
 	echo -e "application/x-truevision=truevision.desktop" >> $$DEFAULTS; \
 	fi \
-	fi ; \
-	update-mime-database "$(prefix)/share/mime"
+	fi ;
+#	update-mime-database "$(prefix)/share/mime"
 
 #	$(truevisionmaterials_DATA)
 #	$(truevisionpython_DATA)
