diff -ur elfutils-0.124.orig/src/Makefile.am elfutils-0.124/src/Makefile.am
--- elfutils-0.124.orig/src/Makefile.am	2007-01-10 16:53:32.687364000 +0200
+++ elfutils-0.124/src/Makefile.am	2007-01-10 16:54:24.260524428 +0200
@@ -33,7 +33,6 @@
 AM_CFLAGS =
 endif
 AM_CFLAGS += -Wall -Wshadow -std=gnu99 $(native_ld_cflags) \
-	     $(if $($(*F)_no_Werror),,-Werror) \
 	     $(if $($(*F)_no_Wunused),,-Wunused $(WEXTRA)) \
 	     $(if $($(*F)_no_Wformat),,-Wformat=2)
 
diff -ur elfutils-0.124.orig/src/Makefile.in elfutils-0.124/src/Makefile.in
--- elfutils-0.124.orig/src/Makefile.in	2007-01-10 16:53:32.699362000 +0200
+++ elfutils-0.124/src/Makefile.in	2007-01-10 16:54:38.676332892 +0200
@@ -287,12 +287,10 @@
 target_alias = @target_alias@
 @MUDFLAP_FALSE@AM_CFLAGS = -Wall -Wshadow -std=gnu99 \
 @MUDFLAP_FALSE@	$(native_ld_cflags) $(if \
-@MUDFLAP_FALSE@	$($(*F)_no_Werror),,-Werror) $(if \
 @MUDFLAP_FALSE@	$($(*F)_no_Wunused),,-Wunused $(WEXTRA)) $(if \
 @MUDFLAP_FALSE@	$($(*F)_no_Wformat),,-Wformat=2)
 @MUDFLAP_TRUE@AM_CFLAGS = -fmudflap -Wall -Wshadow -std=gnu99 \
 @MUDFLAP_TRUE@	$(native_ld_cflags) $(if \
-@MUDFLAP_TRUE@	$($(*F)_no_Werror),,-Werror) $(if \
 @MUDFLAP_TRUE@	$($(*F)_no_Wunused),,-Wunused $(WEXTRA)) $(if \
 @MUDFLAP_TRUE@	$($(*F)_no_Wformat),,-Wformat=2)
 INCLUDES = -I$(srcdir) -I$(srcdir)/../libelf -I$(srcdir)/../libebl \
