Index: kmetronome-0.10.0/kmetronome.desktop
===================================================================
--- kmetronome-0.10.0.orig/kmetronome.desktop
+++ kmetronome-0.10.0/kmetronome.desktop
@@ -5,4 +5,5 @@ Icon=kmetronome
 Type=Application
 Comment=KDE MIDI Metronome Using ALSA Sequencer
 Comment[es]=Metrónomo MIDI de KDE utilizando el secuenciador ALSA
+Comment[tr]=ALSA Ardışımlayıcı Kullanan KDE MIDI Metronom Uygulaması
 Categories=AudioVideo;Audio;Midi;Education;Music;
