Index: startkde
===================================================================
--- startkde	(revision 668197)
+++ startkde	(revision 668198)
@@ -342,7 +342,7 @@
 
 # We set LD_BIND_NOW to increase the efficiency of kdeinit.
 # kdeinit unsets this variable before loading applications.
-LD_BIND_NOW=true start_kdeinit --new-startup +kcminit_startup
+LD_BIND_NOW=true start_kdeinit_wrapper --new-startup +kcminit_startup
 if test $? -ne 0; then
   # Startup error
   echo 'startkde: Could not start kdeinit. Check your installation.'  1>&2
