Index: xmms2-0.7DrNo/wscript
===================================================================
--- xmms2-0.7DrNo.orig/wscript
+++ xmms2-0.7DrNo/wscript
@@ -487,8 +487,6 @@ def shutdown():
         ldconfig = '/sbin/ldconfig'
         if os.path.isfile(ldconfig):
             libprefix = Utils.subst_vars('${PREFIX}/lib', Build.bld.env)
-            try: Utils.cmd_output(ldconfig + ' ' + libprefix)
-            except: pass
 
     if Options.options.run_tests:
         os.system(os.path.join(blddir, "default/tests/test_xmmstypes"))
