--- configure.in.orig	2008-08-07 17:51:47.000000000 +0300
+++ configure.in	2008-08-07 17:52:17.000000000 +0300
@@ -31,8 +31,8 @@
 AC_GNU_SOURCE
 AC_CANONICAL_HOST
 AC_PROG_CC
-AC_AIX
-AC_MINIX
+
+
 AC_EXEEXT
 AC_PROG_RANLIB
 AC_SYS_LARGEFILE
@@ -86,7 +86,7 @@
 fi
 
 if test "x$GFTP_TEXT" = "x" -a $found_glib20 = 0 ; then
-  AM_PATH_GLIB(1.2.3, , AC_MSG_ERROR(gFTP needs GLIB 1.2.3 or higher))
+  AC_DEFINE(dummy, 1, [dummy])
 fi
 
 EXTRA_LIBS=""
@@ -192,7 +192,7 @@
 PTHREAD_CFLAGS=""
 PTHREAD_LIBS=""
 
-# I don't have PKG_CHECK_MODULES or AM_PATH_GLIB check for gthread because
+# I don't have PKG_CHECK_MODULES or AC_DEFINE(dummy, 1, [dummy])
 # the text port doesn't need to be compiled against the thread libraries
 GTHREAD_LIBS=""
 
@@ -202,7 +202,7 @@
   fi
 
   if test "x$GFTP_GTK" = "x" ; then
-    AM_PATH_GTK(1.2.3, GFTP_GTK=gftp-gtk, AC_MSG_WARN(gFTP needs GTK+ 1.2.3 or higher for the graphical version. Only building the text version))
+    AC_DEFINE(dummy, 1, [dummy])
     GTHREAD_LIBS="-lgthread"
   else
     GTHREAD_LIBS="-lgthread-2.0"
