--- src/interactive_base.cc.orig	2007-09-01 13:06:13.000000000 +0300
+++ src/interactive_base.cc	2007-09-15 11:29:24.000000000 +0300
@@ -58,7 +58,7 @@
 		// Switch to the new graphics system now, if necessary
 		WLApplication::get()->init_graphics
 			(get_xres(), get_yres(),
-			 s.get_int("depth", 16), s.get_bool("fullscreen", false));
+			 s.get_int("depth", 32), s.get_bool("fullscreen", true));
 
 		m_show_workarea_preview = s.get_bool("workareapreview", false);
 	}
@@ -163,12 +163,12 @@
 */
 int Interactive_Base::get_xres()
 {
-	return g_options.pull_section("global")->get_int("xres", 640);
+	return g_options.pull_section("global")->get_int("xres", 1024);
 }
 
 int Interactive_Base::get_yres()
 {
-	return g_options.pull_section("global")->get_int("yres", 480);
+	return g_options.pull_section("global")->get_int("yres", 768);
 }
 
 
