--- Root_Actions_2.4.5/rootactions-servicemenu.pl	2009-08-27 13:11:01.000000000 +0300
+++ Root_Actions_2.4.5/rootactions-servicemenu.pl.new       2009-08-28 11:14:06.413524221 +0300
@@ -751,7 +751,7 @@
 
 if ( `which kdesudo` ) {
 	$SUCOMMAND = "$KDEBINPATH"."kdesudo -d --noignorebutton --"}
-else { $SUCOMMAND = "$KDEBINPATH"."kdesu -d --"}
+else { $SUCOMMAND = "$KDEBINPATH"."/usr/kde/4/lib/kde4/libexec/kdesu -d --"}
 
 $DIALOGCOMMAND = "$KDEBINPATH"."kdialog" ;
 $CONFIGCOMMAND = "$KDEBINPATH"."kreadconfig" ;
@@ -908,18 +908,22 @@
 	&delete_dialog_msgs;
 	system "$DIALOGCOMMAND --title \'$DELETETITLE\' --warningcontinuecancel \'$DELETEMSG\\n$TARGETLIST\'" ;
 
+
 	#Is deletion confirmed?
 	if ( $? eq 0 ) {
 	   # kdesu will run the command as regular user if 'Ignore' is chosen from kdesu dialog.
 	   # To prevent unwanted deletion of files, we'll run 'do_delete' instead of 'rm -r', 'do delete'
 	   # will exit if it's run as normal user, therefore the files writable for user are safe when clicking 'Ignore'
 	   exec "$SUCOMMAND \'$EXECNAME\' do_delete \'$TARGET\'";
+       
 	}
 }
 
 sub do_delete {
 	#If running as root, remove the files targeted
+
 	if ( $> eq 0 ) {
+
 	   `rm -r --preserve-root \'$TARGET\'` ;
 	}
 	exit 0;
