--- kio/kio/kprotocolmanager.cpp	2007/12/16 19:19:43	1.1
+++ kio/kio/kprotocolmanager.cpp	2007/12/16 19:20:14
@@ -73,7 +73,7 @@
 
 // DEFAULT USERAGENT STRING
 #define CFG_DEFAULT_UAGENT(X) \
-QString("Mozilla/5.0 (compatible; Konqueror/%1.%2%3) KHTML/%4.%5.%6 (like Gecko)") \
+QString("Mozilla/5.0 (compatible; Konqueror/%1.%2%3) KHTML/%4.%5.%6 (like Gecko) Pardus") \
         .arg(KDE_VERSION_MAJOR).arg(KDE_VERSION_MINOR).arg(X).arg(KDE_VERSION_MAJOR).arg(KDE_VERSION_MINOR).arg(KDE_VERSION_RELEASE)
 
 #define PRIVATE_DATA \
