--- configure~	2008-08-13 17:41:33.000000000 +0200
+++ configure	2008-08-13 17:45:45.000000000 +0200
@@ -44,7 +44,7 @@
 		echo LIBSUF=.so
         echo "CXXFLAGS+=-pthread -Wall -fPIC `mlt-config --cflags`"
 		echo "LIBFLAGS=-shared"
-        echo "LDFLAGS+=`mlt-config --libs` `mlt-config miracle --libs`"
+        echo "LDFLAGS+=`mlt-config --libs` `mlt-config miracle --libs` `mlt-config valerie --libs`"
 		;;
 esac >> config.mak
 
