--- setup.py.orig	2009-01-19 13:39:32.000000000 +0200
+++ setup.py	2009-01-26 20:49:17.000000000 +0200
@@ -633,7 +633,7 @@
     DATA_FILES = []
     if not 'bdist_egg' in sys.argv:
         # generate and install bzr.1 only with plain install, not easy_install one
-        DATA_FILES = [('man/man1', ['bzr.1'])]
+        DATA_FILES = [('share/man/man1', ['bzr.1'])]
 
     # std setup
     ARGS = {'scripts': ['bzr'],
