Index: data-boot/gfxboot.cfg
===================================================================
--- data-boot/gfxboot.cfg	(revision 21223)
+++ data-boot/gfxboot.cfg	(revision 21224)
@@ -35,7 +35,7 @@
 ; max. visible main menu items
 mainmenu.entries=8
 ; minimal main menu selection bar width
-mainmenu.bar.minwidth=300
+mainmenu.bar.minwidth=410
 ; main menu selection bar transparency (0..255)
 mainmenu.bar.transparency=100
 ; main menu colors
Index: data-install/gfxboot.cfg
===================================================================
--- data-install/gfxboot.cfg	(revision 21223)
+++ data-install/gfxboot.cfg	(revision 21224)
@@ -35,7 +35,7 @@
 ; max. visible main menu items
 mainmenu.entries=8
 ; minimal main menu selection bar width
-mainmenu.bar.minwidth=300
+mainmenu.bar.minwidth=410
 ; main menu selection bar transparency (0..255)
 mainmenu.bar.transparency=100
 ; main menu colors
