--- src/files.c
+++ src/files.c
@@ -1518,7 +1518,7 @@
 	}
 
 	if (f_open == NULL) {
-	    fd_source = open(realname, O_RDONLY | O_CREAT);
+	    fd_source = open(realname, O_RDONLY);
 
 	    if (fd_source != -1) {
 		f_source = fdopen(fd_source, "rb");
@@ -1637,7 +1637,7 @@
 	int fd_source;
 	FILE *f_source = NULL;
 
-	fd_source = open(tempname, O_RDONLY | O_CREAT);
+	fd_source = open(tempname, O_RDONLY);
 
 	if (fd_source != -1) {
 	    f_source = fdopen(fd_source, "rb");

