--- quassel-0.4.2_20090529/data/scripts/mpris.orig	2009-05-29 02:23:35.000000000 +0300
+++ quassel-0.4.2_20090529/data/scripts/mpris	2009-05-29 02:24:04.000000000 +0300
@@ -40,7 +40,7 @@
 if(keys(%d)) {
 
 ## change the following line to fit your needs ##
-print "I'm listening to ".$d{"title"}." by ".$d{"artist"};
+print "/me is listening to ".$d{"title"}." by ".$d{"artist"};
 #################################################
 
 print "\n";
