--- cdrtools-2.01.01/cdrecord/cdrecord.c	2007-07-04 02:43:51.000000000 +0300
+++ cdrtools-2.01.01/cdrecord/cdrecord.c	2007-07-04 02:44:18.000000000 +0300
@@ -361,7 +361,7 @@
 #	define	CLONE_TITLE	""
 #endif
 	if ((flags & F_MSINFO) == 0 || lverbose || flags & F_VERSION) {
-		printf("Cdrecord%s%s%s %s (%s-%s-%s) Copyright (C) 1995-2008 Jörg Schilling\n",
+		printf("Cdrecord%s%s%s %s (%s-%s-%s) Copyright (C) 1995-2008 Joerg Schilling\n",
 								PRODVD_TITLE,
 								PROBD_TITLE,
 								CLONE_TITLE,
--- cdrtools-2.01.01/readcd/readcd.c	2007-09-01 15:09:19.000000000 +0300
+++ cdrtools-2.01.01/readcd/readcd.c	2008-01-03 16:55:52.000000000 +0200
@@ -372,7 +372,7 @@
 	if (help)
 		usage(0);
 	if (pversion) {
-		printf("readcd %s (%s-%s-%s) Copyright (C) 1987, 1995-2007 Jörg Schilling\n",
+		printf("readcd %s (%s-%s-%s) Copyright (C) 1987, 1995-2007 Joerg Schilling\n",
 								cdr_version,
 								HOST_CPU, HOST_VENDOR, HOST_OS);
 		exit(0);
