--- tests/lossy_comp_test.c	2006-08-31 12:22:07.000000000 +0300
+++ tests/lossy_comp_test.c	2008-01-18 04:17:07.000000000 +0200
@@ -691,7 +691,8 @@
 lcomp_test_int (const char *filename, int filetype, int channels, double margin)
 {	SNDFILE		*file ;
 	SF_INFO		sfinfo ;
-	int			k, m, *orig, *data, sum_abs ;
+	int			k, m, *orig, *data;
+	unsigned sum_abs ;
 	long		datalen, seekpos ;
 	double		scale ;
 
