--- a/src/progs/pickit2/base/pickit2.h
+++ b/src/progs/pickit2/base/pickit2.h
@@ -27,7 +27,7 @@
 public:
   USBPort(Log::Base &log) : Pickit::USBPort(0x0033, log) {}
   virtual Pickit::Array array() const { return Array(); }
-  void fillCommand(Pickit::Array &cmd, uchar cmd, uint nbBytes, uint address, uint i, bool longAddress = true) const;
+  void fillCommand(Pickit::Array &cmd1, uchar cmd2, uint nbBytes, uint address, uint i, bool longAddress = true) const;
   Pickit::Array createCommand(uchar cmd, uint nbBytes, uint address, bool longAddress = true) const;
 
   bool readFirmwareCodeMemory(Device::Array &data, const Device::Array *vdata, ProgressMonitor &monitor);
