diff --git a/data/displaysettings.desktop b/data/displaysettings.desktop
index 91d6e9d..a8a4e10 100644
--- a/data/displaysettings.desktop
+++ b/data/displaysettings.desktop
@@ -4,11 +4,13 @@ Name[sv]=Bildskärm
 Name[tr]=Ekran Ayarları
 Name[nl]=Beeldscherminstellingen
 Name[fr]=Affichage
+Name[de]=Anzeige
 Comment=Configure display devices
 Comment[sv]=Anpassar bildskärmen
 Comment[tr]=Görüntü aygıtlarını yapılandır
 Comment[nl]=Beeldapparaten instellen
 Comment[fr]=Configuration de l'affichage
+Comment[de]=Bildschirmeinstellungen
 Exec=display-settings
 Icon=preferences-desktop-display
 Type=Application
diff --git a/data/kcm_displaydevices.desktop b/data/kcm_displaydevices.desktop
index 649b94d..21571d5 100644
--- a/data/kcm_displaydevices.desktop
+++ b/data/kcm_displaydevices.desktop
@@ -23,8 +23,10 @@ Comment[tr]=Görüntü aygıtlarını yapılandır
 Comment[sv]=Anpassar bildskärmsenheter
 Comment[nl]=Videokaart instellen
 Comment[fr]=Configuration de la carte graphique
+Comment[de]=Grafikeinstellungen
 X-KDE-Keywords=python display screen video monitor manager
 X-KDE-Keywords[tr]=python görüntü ekran video monitör yönetici
 X-KDE-Keywords[sv]=python bildskärm skärmar monitor inställningar
 X-KDE-Keywords[nl]=python beeldschermbeheer
 X-KDE-Keywords[fr]=python affichage graphique moniteur écran video
+X-KDE-Keywords[de]=python grafik monitor video manager
diff --git a/data/kcm_displaysettings.desktop b/data/kcm_displaysettings.desktop
index ca74010..ea48f1b 100644
--- a/data/kcm_displaysettings.desktop
+++ b/data/kcm_displaysettings.desktop
@@ -23,8 +23,10 @@ Comment[tr]=Ekranları boyutlandır ve döndür
 Comment[sv]=Roterar och ändrar storlek
 Comment[nl]=Grootte en rotatie van beeldscherm
 Comment[fr]=Redimensionnement et rotation de l'écran
+Comment[de]=Auflösung und Orientierung des Schirms
 X-KDE-Keywords=python display screen video monitor manager
 X-KDE-Keywords[tr]=python görüntü ekran video monitör yönetici
 X-KDE-Keywords[sv]=python bildskärm skärm monitor inställningar
 X-KDE-Keywords[nl]=python beeldschermbeheer
 X-KDE-Keywords[fr]=python video moniteur affichage écran
+X-KDE-Keywords[de]=python grafik monitor video manager
