--- security/nss/lib/Makefile	2007-07-20 00:36:49.000000000 +0300
+++ security/nss/lib/Makefile	2008-01-03 01:25:25.000000000 +0200
@@ -62,12 +62,6 @@
 DIRS := $(filter-out fortcrypt,$(DIRS))
 endif
 
-ifndef MOZILLA_CLIENT
-ifndef NSS_USE_SYSTEM_SQLITE
-DIRS := sqlite $(DIRS)
-endif
-endif
-
 #######################################################################
 # (5) Execute "global" rules. (OPTIONAL)                              #
 #######################################################################
