Index: gluon-0.71.0/creator/gluon-creator.desktop
===================================================================
--- gluon-0.71.0.orig/creator/gluon-creator.desktop
+++ gluon-0.71.0/creator/gluon-creator.desktop
@@ -23,6 +23,7 @@ GenericName[nl]=Hulpmiddel voor spelcrea
 GenericName[pt]=Ferramenta de Criação de Jogos
 GenericName[pt_BR]=Ferramenta de criação de jogos
 GenericName[sv]=Verktyg för att skapa spel
+GenericName[tr]=Oyun Oluşturma Aracı
 GenericName[uk]=Інструмент для створення ігор
 GenericName[x-test]=xxGame Creation Toolxx
 Exec=gluoncreator
