diff -ur lipstik-2.2.orig/style/lipstik.cpp lipstik-2.2/style/lipstik.cpp
--- lipstik-2.2.orig/style/lipstik.cpp	2006-02-08 19:35:16.000000000 +0200
+++ lipstik-2.2/style/lipstik.cpp	2006-12-01 10:41:43.000000000 +0200
@@ -206,7 +206,7 @@
     _drawTabHighlight = settings.readBoolEntry("/drawTabHighlight", true);
     _drawToolBarHandle = settings.readBoolEntry("/drawToolBarHandle", true);
     _coloredMenuBar = settings.readBoolEntry("/coloredMenuBar", false);
-    _menuLeftSideBar = settings.readBoolEntry("/menuLeftSideBar", true);
+    _menuLeftSideBar = settings.readBoolEntry("/menuLeftSideBar", false);
     _flatStripe = settings.readBoolEntry("/flatStripe", false);
     _customMenuStripeColor = settings.readBoolEntry("/customMenuStripeColor", false);
     _menuStripeColor.setNamedColor( settings.readEntry("/menuStripeColor", "black") );
