diff -Nuar akbib/kbibdoc.cpp kbib/kbibdoc.cpp
--- akbib/kbibdoc.cpp	2008-01-12 12:25:06.000000000 +0200
+++ kbib/kbibdoc.cpp	2008-05-14 05:21:15.000000000 +0300
@@ -14,7 +14,7 @@
  *   (at your option) any later version.                                   *
  *                                                                         *
  ***************************************************************************/
-
+#include <algorithm>
 #include <fstream>
 using namespace::std;
 
diff -Nuar akbib/scholarsearcher.cpp kbib/scholarsearcher.cpp
--- akbib/scholarsearcher.cpp	2008-01-12 12:25:06.000000000 +0200
+++ kbib/scholarsearcher.cpp	2008-05-14 05:21:50.000000000 +0300
@@ -24,7 +24,7 @@
 #include <qfile.h>
 #include <qregexp.h>
 
-#include <fstream.h>
+#include <fstream>
 
 ScholarSearcher::ScholarSearcher ( QObject *parent, const char *name )
 		: searcher ( parent, name ),m_step ( Begin ), m_started ( false )
