diff -Nur sdljump-1.0.0-old/src/setup.c sdljump-1.0.0/src/setup.c
--- sdljump-1.0.0-old/src/setup.c	2008-04-22 14:29:30.000000000 +0300
+++ sdljump-1.0.0/src/setup.c	2008-04-22 14:30:20.000000000 +0300
@@ -150,7 +150,7 @@
     
     /* Default player names */
     for (i=0; i<MAX_PLAYERS; i++) {
-        gblOps.pname[i] = malloc(sizeof(char)*(strlen(PNAME)+1));
+        gblOps.pname[i] = malloc(sizeof(char)*(strlen(PNAME)+4));
         sprintf(gblOps.pname[i],"%s%d",PNAME,i+1);
     }
 }
