diff -Nur nexuiz-2.1-old/darkplaces/fs.c nexuiz-2.1/darkplaces/fs.c
--- nexuiz-2.1-old/darkplaces/fs.c	2006-10-01 23:57:14.276644424 +0300
+++ nexuiz-2.1/darkplaces/fs.c	2006-10-02 00:16:59.475466816 +0300
@@ -1067,7 +1067,7 @@
 #ifdef DP_FS_BASEDIR
 	strlcpy(fs_basedir, DP_FS_BASEDIR, sizeof(fs_basedir));
 #else
-	strlcpy(fs_basedir, "", sizeof(fs_basedir));
+	strlcpy(fs_basedir, "/usr/share/quake1", sizeof(fs_basedir));
 
 #ifdef MACOSX
 	// FIXME: is there a better way to find the directory outside the .app?
@@ -1127,7 +1127,7 @@
 	{
 		// add the game-specific paths
 		// gamedirname1 (typically id1)
-		FS_AddGameHierarchy (gamedirname1);
+		FS_AddGameHierarchy ("nexuiz");
 
 		// add the game-specific path, if any
 		if (gamedirname2)
@@ -1137,7 +1137,7 @@
 		}
 
 		// set the com_modname (reported in server info)
-		strlcpy(com_modname, gamedirname1, sizeof(com_modname));
+		strlcpy(com_modname, "nexuiz", sizeof(com_modname));
 
 		// -game <gamedir>
 		// Adds basedir/gamedir as an override game
