--- alsa-plugins-1.0.15/pulse/pulse.c.orig	2008-03-04 18:56:22.000000000 +0100
+++ alsa-plugins-1.0.15/pulse/pulse.c	2008-03-04 18:58:33.000000000 +0100
@@ -204,7 +204,10 @@ int pulse_connect(snd_pulse_t *p, const 
     return 0;
 
 error:
-    fprintf(stderr, "*** PULSEAUDIO: Unable to connect: %s\n",
+    fprintf(stderr,
+        "*** PULSEAUDIO: Unable to connect: %s\n"
+        "*** Is your sound server running?\n"
+        "*** See: http://www.pulseaudio.org/wiki/Troubleshooting\n",
         pa_strerror(pa_context_errno(p->context)));
 
     pa_threaded_mainloop_unlock(p->mainloop);
