diff -Nur AssaultCube-old/config/menus.cfg AssaultCube/config/menus.cfg
--- AssaultCube-old/config/menus.cfg	2007-06-10 23:05:25.618722062 +0300
+++ AssaultCube/config/menus.cfg	2007-06-10 23:07:19.969338127 +0300
@@ -36,11 +36,11 @@
 
 newmenu maps
 
-genmapitems "ac_complex ac_mines ac_depot ac_desert ac_desert2 ac_snow"
+genmapitems "ac_complex ac_mines ac_depot ac_desert ac_desert2 ac_snow ac_factory fantasy illusion ac_rattrap ac_ruins snipe temple 2kingdms ac_prison ac_police_station azz octagon ac_CombatArena ac_Flood ac_RuinedFortifications Runners TheDarkCastle toll Trench TrenchWarfare2 Memico ac_castle ac_estate Bridge coy dust2 fy_grenade Labyrinth RuinedFortifications TunnelsDry 2fortac Fy_Ice_World ac_building ac_city ac_desertbase ac_inka ac_normandie aim_ak_colt fy_pool_day"
 
 newmenu maps_ctf 
 
-genmapitems "ac_mines ac_depot"
+genmapitems "ac_mines ac_depot ac_factory fantasy ac_rattrap ac_ruins snipe 2kingdms azz ac_Flood ac_RuinedFortifications Runners TheDarkCastle toll Trench coy ac_estate RuinedFortifications Bridge TunnelsDry Memico ac_desertbase fy_pool_day ac_normandie ac_city Fy_Ice_World 2fortac ac_prison temple"
 
 newmenu setup
 
@@ -206,7 +206,7 @@
 	]
 ]
 
-genbotmaps "ac_complex ac_mines ac_depot ac_desert ac_desert2 ac_snow"
+genbotmaps "ac_complex ac_mines ac_depot ac_desert ac_desert2 ac_snow ac_factory"
 
 newmenu "server operator"
 
