Index: libwnck-2.26.1/doc/Makefile.am
===================================================================
--- libwnck-2.26.1.orig/doc/Makefile.am
+++ libwnck-2.26.1/doc/Makefile.am
@@ -44,8 +44,6 @@ extra_files =
 GTKDOC_CFLAGS = -I$(top_srcdir) $(LIBWNCK_CFLAGS) -DWNCK_I_KNOW_THIS_IS_UNSTABLE
 GTKDOC_LIBS = $(top_builddir)/libwnck/libwnck-1.la $(LIBWNCK_LIBS)
 
-include $(top_srcdir)/gtk-doc.make
-
-CLEANFILES += gsf-scan.*
+CLEANFILES = gsf-scan.*
 
 .PHONY : dist-hook-local
