diff -Nur enigma-0.92-old/Makefile.in enigma-0.92/Makefile.in
--- enigma-0.92-old/Makefile.in	2006-08-20 22:44:51.733397896 +0300
+++ enigma-0.92/Makefile.in	2006-08-20 22:52:25.335439904 +0300
@@ -173,7 +173,7 @@
 #   tools -> lib-src/lua
 #   src   -> tools, lib-src
 #
-SUBDIRS = m4 lib-src tools intl src data doc po etc
+SUBDIRS = m4 lib-src tools intl src data  po etc
 EXTRA_DIST = enigma CHANGES ChangeLog.tla
 
 docdir = @datadir@/doc/enigma
@@ -538,7 +538,7 @@
 
 info-am:
 
-install-data-am: install-docDATA
+install-data-am:
 
 install-exec-am:
 
