diff -Nur enigma-0.92-old/po/Makefile.in.in enigma-0.92/po/Makefile.in.in
--- enigma-0.92-old/po/Makefile.in.in	2006-08-20 22:44:47.718008328 +0300
+++ enigma-0.92/po/Makefile.in.in	2006-08-20 22:51:04.271763456 +0300
@@ -22,7 +22,7 @@
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
-datadir = @datadir@
+datadir = /usr/share
 localedir = $(datadir)/locale
 gettextsrcdir = $(datadir)/gettext/po
 
diff -Nur enigma-0.92-old/src/Makefile.am enigma-0.92/src/Makefile.am
--- enigma-0.92-old/src/Makefile.am	2006-08-20 22:44:47.754002856 +0300
+++ enigma-0.92/src/Makefile.am	2006-08-20 22:51:04.272763304 +0300
@@ -60,7 +60,7 @@
 	video.cc video.hh		\
 	world.cc world.hh world_internal.hh
 
-datadir = @datadir@
+datadir = /usr/share
 
 if MINGW32
 
