diff -up cairo-1.2.4/src/cairo-type1-subset.c.ctype cairo-1.2.4/src/cairo-type1-subset.c
--- cairo-1.2.4/src/cairo-type1-subset.c.ctype	2007-11-28 09:27:04.000000000 -0500
+++ cairo-1.2.4/src/cairo-type1-subset.c	2007-11-28 09:27:06.000000000 -0500
@@ -45,6 +45,8 @@
 #include FT_OUTLINE_H
 #include FT_TYPE1_TABLES_H
 
+#include <ctype.h>
+
 typedef struct _cairo_type1_font_subset {
 
     cairo_scaled_font_subset_t *scaled_font_subset;
