Index: comar/network.py
===================================================================
--- comar/network.py	(revision 16471)
+++ comar/network.py	(working copy)
@@ -191,7 +191,9 @@
             return True
         return None
     
-    def startAuto(self):
+    # timeout parameter is unused and is here for compatibility reasons
+    # (Bug #6626)
+    def startAuto(self, timeout=120):
         if self.isAuto():
             self.stopAuto()
             import time
