--- kerry/src/kerryapp.cpp.old	2006-11-27 23:17:07.000000000 +0200
+++ kerry/src/kerryapp.cpp	2006-11-27 23:19:26.000000000 +0200
@@ -122,10 +122,9 @@
   globalKeys = new KGlobalAccel(this);
   globalKeys->insert( "Program:kerry", i18n("Kerry Beagle Search") );
 
-  KShortcut showDialogShortcut = KShortcut(ALT+Key_Space);
-  showDialogShortcut.append( KKey( Key_F12 ) );
+  KShortcut showDialogShortcut = KShortcut(Key_F11);
   globalKeys->insert( "Show Kerry Dialog", i18n("Show Search Dialog"), QString::null, showDialogShortcut, showDialogShortcut, hitListWindow, SLOT(showSearchDialog()) );
-  globalKeys->insert( "Search Primary Selection with Kerry", i18n("Search Primary Selection"), QString::null, CTRL+ALT+Key_Space, CTRL+ALT+Key_Space, this, SLOT(searchPrimarySelection()) );
+  globalKeys->insert( "Search Primary Selection with Kerry", i18n("Search Primary Selection"), QString::null, ALT+Key_F11, ALT+Key_F11, this, SLOT(searchPrimarySelection()) );
   configChanged();
 
   sysTrayIcon->setPixmap(sysTrayIcon->loadIcon("kerry_systemtray"));
