Index: coolplug/Makefile
===================================================================
--- coolplug.orig/Makefile
+++ coolplug/Makefile
@@ -5,10 +5,11 @@ OBJS= main.c utility.c module.c devnode.
 INCLUDES= common.h
 CFLAGS= -Os -Wall
 LDFLAGS= -Wl,-z,noexecstack
+STRIP= strip
 
 $(EXE): $(OBJS) $(INCLUDES)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $(EXE) $(OBJS)
-	strip -s --remove-section=.note --remove-section=.comment $(EXE)
+	$(STRIP) -s --remove-section=.note --remove-section=.comment $(EXE)
 
 clean:
 	rm -f $(EXE)
