diff -Nur ode-0.9-old/drawstuff/dstest/dstest.cpp ode-0.9/drawstuff/dstest/dstest.cpp
--- ode-0.9-old/drawstuff/dstest/dstest.cpp	2008-04-20 23:37:05.000000000 +0300
+++ ode-0.9/drawstuff/dstest/dstest.cpp	2008-04-20 23:37:25.000000000 +0300
@@ -116,7 +116,7 @@
   fn.step = &simLoop;
   fn.command = command;
   fn.stop = 0;
-  fn.path_to_textures = 0;	// uses default
+  fn.path_to_textures = "/usr/share/ode/examples";	// uses default
 
   // run simulation
   dsSimulationLoop (argc,argv,400,400,&fn);
