diff -Nur vdrift-2006-10-06-src-old/src/SConscript vdrift-2006-10-06-src/src/SConscript
--- vdrift-2006-10-06-src-old/src/SConscript	2006-11-14 19:47:05.145825011 +0200
+++ vdrift-2006-10-06-src/src/SConscript	2006-11-14 19:47:35.644188555 +0200
@@ -87,6 +87,6 @@
 # Install #
 #---------#
 bin_install_dir = env['destdir'] + env['prefix'] + '/' + env['bindir'] + '/'
-#os.system('mkdir -p ' + bin_install_dir)
+##os.system('mkdir -p ' + bin_install_dir)
 env.Install(bin_install_dir, vdrift)
-os.system("ln -sf %s/vdrift /usr/bin" % bin_install_dir )
+#os.system("ln -sf %s/vdrift /usr/bin" % bin_install_dir )
