--- src/mgui/bombono-dvd.desktop.orig	2009-07-26 20:31:16.000000000 +0300
+++ src/mgui/bombono-dvd.desktop	2009-12-26 22:44:55.521673442 +0200
@@ -2,7 +2,11 @@
 Encoding=UTF-8
 Type=Application
 Name=Bombono DVD
+Name[tr]=Bombono DVD
+GenericName=DVD Author
+GenericName[tr]=DVD Oluşturma
 Comment=DVD authoring program
+Comment[tr]=DVD Oluşturma uygulaması
 Icon=bombono-dvd
 Exec=bombono-dvd
 Categories=Video;AudioVideo;DiscBurning;GTK;
