Index: kdebase-workspace-4.3.4/kwin/compositingprefs.cpp
===================================================================
--- kdebase-workspace-4.3.4.orig/kwin/compositingprefs.cpp
+++ kdebase-workspace-4.3.4/kwin/compositingprefs.cpp
@@ -272,6 +272,9 @@ void CompositingPrefs::detectDriverAndVe
         {
         mDriver = "radeon";
         mVersion = Version( mGLRenderer.split(" ")[ 3 ] );
+        // Check that the version string is changed, and try the fifth element if it does
+        if (!mVersion.startsWith("20"))
+            mVersion = Version( mGLRenderer.split(' ')[ 5 ] );
         }
     else if( mGLRenderer == "Software Rasterizer" )
         {
@@ -332,6 +335,11 @@ void CompositingPrefs::applyDriverSpecif
             kDebug( 1212 ) << "radeon r200 >= 20060602, enabling compositing";
             mEnableCompositing = true;
             }
+        if( mGLRenderer.startsWith( "Mesa DRI R300" ) && mVersion >= Version( "20090101" )) // krazy:exclude=strings
+            {
+            kDebug( 1212 ) << "radeon r300 >= 20090101, enabling compositing";
+            mEnableCompositing = true;
+            }
         }
     else if( mDriver == "fglrx" )
         { // radeon r200 only ?
