commit fc6184bef3a4491eaf2f303d4f6a94e416335758
Author: Martin Gräßlin <ubuntu@martin-graesslin.com>
Date:   Fri Jan 16 17:21:22 2009 +0100

    Icons only mode for tasks plasmoid

Index: plasma/applets/tasks/abstracttaskitem.cpp
===================================================================
--- plasma/applets/tasks/abstracttaskitem.cpp.orig
+++ plasma/applets/tasks/abstracttaskitem.cpp
@@ -114,8 +114,15 @@ QSize AbstractTaskItem::basicPreferredSi
     }
 
     //kDebug() << (QObject*)this;
-    return QSize(mSize.width()*12 + m_applet->itemLeftMargin() + m_applet->itemRightMargin() + KIconLoader::SizeSmall,
-            qMax(mSize.height(), iconsize) + topMargin + bottomMargin);
+    if (m_applet->iconsOnly()) {
+        int size = qMin((int)m_applet->itemLeftMargin() + (int)m_applet->itemRightMargin() + (int)KIconLoader::SizeMedium, qMax(mSize.height(), (int)KIconLoader::SizeMedium) + topMargin + bottomMargin);
+        return QSize(size, size);
+    }
+    else
+    {
+        return QSize(mSize.width()*12 + m_applet->itemLeftMargin() + m_applet->itemRightMargin() + KIconLoader::SizeSmall, qMax(mSize.height(), iconsize) + topMargin + bottomMargin);
+    }
+
 }
 
 void AbstractTaskItem::setPreferredOffscreenSize()
@@ -602,17 +610,18 @@ void AbstractTaskItem::drawTask(QPainter
     }
 
     painter->setPen(QPen(textColor(), 1.0));
+    if (!m_applet->iconsOnly()) {
+        QRect rect = textRect(bounds).toRect();
+        if (rect.height() > 20) {
+            rect.adjust(2, 2, -2, -2); // Create a text margin
+        }
+        QTextLayout layout;
+        layout.setFont(KGlobalSettings::taskbarFont());
+        layout.setTextOption(textOption());
 
-    QRect rect = textRect(bounds).toRect();
-    if (rect.height() > 20) {
-        rect.adjust(2, 2, -2, -2); // Create a text margin
-    }
-    QTextLayout layout;
-    layout.setFont(KGlobalSettings::taskbarFont());
-    layout.setTextOption(textOption());
-
-    layoutText(layout, m_text, rect.size());
-    drawTextLayout(painter, layout, rect);
+        layoutText(layout, m_text, rect.size());
+        drawTextLayout(painter, layout, rect);
+    }
 
     TaskGroupItem *groupItem = qobject_cast<TaskGroupItem *>(this);
     if (groupItem) {
@@ -878,7 +887,9 @@ QRectF AbstractTaskItem::iconRect(const 
     QRectF bounds(b);
     const int right = bounds.right();
     //leave enough space for the text. useful in vertical panel
-    bounds.setWidth(qMax(bounds.width() / 3, qMin(minimumSize().height(), bounds.width())));
+    if (!m_applet->iconsOnly()) {
+        bounds.setWidth(qMax(bounds.width() / 3, qMin(minimumSize().height(), bounds.width())));
+    }
 
     //restore right position if the layout is RTL
     if (QApplication::layoutDirection() == Qt::RightToLeft) {
@@ -897,7 +908,12 @@ QRectF AbstractTaskItem::iconRect(const 
         }
     }
 
-    return QStyle::alignedRect(QApplication::layoutDirection(), Qt::AlignLeft | Qt::AlignVCenter,
+    Qt::Alignment align = Qt::AlignLeft;
+    if (m_applet->iconsOnly()) {
+        align = Qt::AlignCenter;
+    }
+
+    return QStyle::alignedRect(QApplication::layoutDirection(), align | Qt::AlignVCenter,
                                iconSize, bounds.toRect());
 }
 
Index: plasma/applets/tasks/tasks.cpp
===================================================================
--- plasma/applets/tasks/tasks.cpp.orig
+++ plasma/applets/tasks/tasks.cpp
@@ -82,6 +82,7 @@ void Tasks::init()
 
     const KConfigGroup cg = config();
     m_showTooltip = cg.readEntry("showTooltip", true);
+    m_iconsOnly = cg.readEntry("iconsOnly", false);
 
     m_groupManager = new TaskManager::GroupManager(this);
     Plasma::Containment* appletContainment = containment();
@@ -260,6 +261,7 @@ void Tasks::createConfigurationInterface
      parent->addPage(widget, i18n("General"), icon());
 
     m_ui.showTooltip->setChecked(m_showTooltip);
+    m_ui.iconsOnly->setChecked(m_iconsOnly);
     m_ui.showOnlyCurrentDesktop->setChecked(m_groupManager->showOnlyCurrentDesktop());
     m_ui.showOnlyCurrentScreen->setChecked(m_groupManager->showOnlyCurrentScreen());
     m_ui.showOnlyMinimized->setChecked(m_groupManager->showOnlyMinimized());
@@ -387,6 +389,13 @@ void Tasks::configAccepted()
         changed = true;
     }
 
+    if (m_iconsOnly != (m_ui.iconsOnly->checkState() == Qt::Checked)) {
+        m_iconsOnly = !m_iconsOnly;
+        KConfigGroup cg = config();
+        cg.writeEntry("iconsOnly", m_iconsOnly);
+        changed = true;
+    }
+
     if (changed) {
         emit settingsChanged();
         emit configNeedsSaving();
@@ -409,6 +418,10 @@ TaskGroupItem* Tasks::rootGroupItem()
     return m_rootGroupItem;
 }
 
+bool Tasks::iconsOnly() const
+{
+    return m_iconsOnly;
+}
 
 K_EXPORT_PLASMA_APPLET(tasks, Tasks)
 
Index: plasma/applets/tasks/tasks.h
===================================================================
--- plasma/applets/tasks/tasks.h.orig
+++ plasma/applets/tasks/tasks.h
@@ -104,6 +104,7 @@ public:
         Qt::KeyboardModifiers groupModifierKey() const;
 
         bool showToolTip() const;
+        bool iconsOnly() const;
 
         void needsVisualFocus();
 
@@ -134,6 +135,7 @@ private slots:
 
 private:
         bool m_showTooltip;
+        bool m_iconsOnly;
         Plasma::LayoutAnimator *m_animator;
         QGraphicsLinearLayout *layout;
 
Index: plasma/applets/tasks/tasksConfig.ui
===================================================================
--- plasma/applets/tasks/tasksConfig.ui.orig
+++ plasma/applets/tasks/tasksConfig.ui
@@ -1,121 +1,143 @@
-<ui version="4.0" >
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
  <class>tasksConfig</class>
- <widget class="QWidget" name="tasksConfig" >
-  <property name="geometry" >
+ <widget class="QWidget" name="tasksConfig">
+  <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>507</width>
-    <height>391</height>
+    <height>411</height>
    </rect>
   </property>
-  <layout class="QGridLayout" name="gridLayout" >
-   <item row="0" column="0" colspan="2" >
-    <widget class="QLabel" name="label_5" >
-     <property name="font" >
+  <layout class="QGridLayout" name="gridLayout">
+   <item row="0" column="0" colspan="2">
+    <widget class="QLabel" name="label_5">
+     <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
-     <property name="text" >
+     <property name="text">
       <string>Appearance</string>
      </property>
     </widget>
    </item>
-   <item row="1" column="1" >
-    <widget class="QLabel" name="label_7" >
-     <property name="text" >
-      <string>Show tooltips:</string>
-     </property>
-     <property name="alignment" >
-      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
-     </property>
-    </widget>
-   </item>
-   <item row="1" column="2" >
-    <widget class="QCheckBox" name="showTooltip" >
-     <property name="text" >
-      <string/>
-     </property>
-    </widget>
-   </item>
-   <item row="2" column="1" >
-    <widget class="QLabel" name="label_8" >
-     <property name="text" >
-      <string>Force row settings:</string>
-     </property>
-    </widget>
-   </item>
-   <item row="2" column="2" >
-    <widget class="QCheckBox" name="fillRows" >
-     <property name="text" >
-      <string/>
-     </property>
-    </widget>
-   </item>
-   <item row="3" column="1" >
-    <spacer name="verticalSpacer_6" >
-     <property name="orientation" >
-      <enum>Qt::Vertical</enum>
-     </property>
-     <property name="sizeType" >
-      <enum>QSizePolicy::Fixed</enum>
-     </property>
-     <property name="sizeHint" stdset="0" >
-      <size>
-       <width>64</width>
-       <height>5</height>
-      </size>
-     </property>
-    </spacer>
-   </item>
-   <item row="4" column="1" >
-    <widget class="QLabel" name="label_3" >
-     <property name="text" >
-      <string>Maximum rows:</string>
-     </property>
-     <property name="alignment" >
-      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
-     </property>
-    </widget>
-   </item>
-   <item row="4" column="2" >
-    <layout class="QHBoxLayout" name="horizontalLayout" >
-     <item>
-      <widget class="QSpinBox" name="maxRows" >
-       <property name="minimum" >
-        <number>1</number>
+   <item row="1" column="1" colspan="2">
+    <layout class="QFormLayout" name="formLayout">
+     <item row="0" column="0">
+      <widget class="QLabel" name="label_7">
+       <property name="text">
+        <string>Show tooltips:</string>
        </property>
-       <property name="value" >
-        <number>2</number>
+       <property name="alignment">
+        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
-     <item>
-      <spacer name="horizontalSpacer_2" >
-       <property name="orientation" >
-        <enum>Qt::Horizontal</enum>
+     <item row="0" column="1">
+      <widget class="QCheckBox" name="showTooltip">
+       <property name="text">
+        <string/>
+       </property>
+      </widget>
+     </item>
+     <item row="1" column="0">
+      <widget class="QLabel" name="label_8">
+       <property name="text">
+        <string>Force row settings:</string>
+       </property>
+      </widget>
+     </item>
+     <item row="1" column="1">
+      <widget class="QCheckBox" name="fillRows">
+       <property name="text">
+        <string/>
+       </property>
+      </widget>
+     </item>
+     <item row="2" column="0">
+      <widget class="QLabel" name="label_9">
+       <property name="text">
+        <string>Show only icons:</string>
        </property>
-       <property name="sizeHint" stdset="0" >
+       <property name="alignment">
+        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
+       </property>
+      </widget>
+     </item>
+     <item row="2" column="1">
+      <widget class="QCheckBox" name="iconsOnly">
+       <property name="text">
+        <string/>
+       </property>
+      </widget>
+     </item>
+     <item row="3" column="0">
+      <spacer name="verticalSpacer_6">
+       <property name="orientation">
+        <enum>Qt::Vertical</enum>
+       </property>
+       <property name="sizeType">
+        <enum>QSizePolicy::Fixed</enum>
+       </property>
+       <property name="sizeHint" stdset="0">
         <size>
-         <width>47</width>
-         <height>23</height>
+         <width>64</width>
+         <height>5</height>
         </size>
        </property>
       </spacer>
      </item>
+     <item row="4" column="0">
+      <widget class="QLabel" name="label_3">
+       <property name="text">
+        <string>Maximum rows:</string>
+       </property>
+       <property name="alignment">
+        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
+       </property>
+      </widget>
+     </item>
+     <item row="4" column="1">
+      <layout class="QHBoxLayout" name="horizontalLayout">
+       <item>
+        <widget class="QSpinBox" name="maxRows">
+         <property name="minimum">
+          <number>1</number>
+         </property>
+         <property name="value">
+          <number>2</number>
+         </property>
+        </widget>
+       </item>
+       <item>
+        <spacer name="horizontalSpacer_2">
+         <property name="orientation">
+          <enum>Qt::Horizontal</enum>
+         </property>
+         <property name="sizeHint" stdset="0">
+          <size>
+           <width>47</width>
+           <height>23</height>
+          </size>
+         </property>
+        </spacer>
+       </item>
+      </layout>
+     </item>
     </layout>
    </item>
-   <item row="5" column="1" >
-    <spacer name="verticalSpacer_2" >
-     <property name="orientation" >
+   <item row="2" column="1">
+    <spacer name="verticalSpacer_2">
+     <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
-     <property name="sizeType" >
+     <property name="sizeType">
       <enum>QSizePolicy::Fixed</enum>
      </property>
-     <property name="sizeHint" stdset="0" >
+     <property name="sizeHint" stdset="0">
       <size>
        <width>64</width>
        <height>10</height>
@@ -123,34 +145,34 @@
      </property>
     </spacer>
    </item>
-   <item row="6" column="0" colspan="3" >
-    <widget class="QLabel" name="label_6" >
-     <property name="font" >
+   <item row="3" column="0" colspan="2">
+    <widget class="QLabel" name="label_6">
+     <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
-     <property name="text" >
+     <property name="text">
       <string>Grouping &amp; Sorting</string>
      </property>
     </widget>
    </item>
-   <item row="7" column="1" >
-    <widget class="QLabel" name="label" >
-     <property name="text" >
+   <item row="4" column="1">
+    <widget class="QLabel" name="label">
+     <property name="text">
       <string>Grouping:</string>
      </property>
-     <property name="alignment" >
+     <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
-   <item row="7" column="2" >
-    <layout class="QHBoxLayout" name="horizontalLayout_2" >
+   <item row="4" column="2">
+    <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
-      <widget class="QComboBox" name="groupingStrategy" >
-       <property name="minimumSize" >
+      <widget class="QComboBox" name="groupingStrategy">
+       <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
@@ -159,11 +181,11 @@
       </widget>
      </item>
      <item>
-      <spacer name="horizontalSpacer_3" >
-       <property name="orientation" >
+      <spacer name="horizontalSpacer_3">
+       <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
-       <property name="sizeHint" stdset="0" >
+       <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
@@ -173,17 +195,17 @@
      </item>
     </layout>
    </item>
-   <item row="8" column="2" >
-    <layout class="QHBoxLayout" name="horizontalLayout_4" >
+   <item row="5" column="2">
+    <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
-      <spacer name="horizontalSpacer" >
-       <property name="orientation" >
+      <spacer name="horizontalSpacer">
+       <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
-       <property name="sizeType" >
+       <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
-       <property name="sizeHint" stdset="0" >
+       <property name="sizeHint" stdset="0">
         <size>
          <width>13</width>
          <height>20</height>
@@ -192,35 +214,35 @@
       </spacer>
      </item>
      <item>
-      <widget class="QCheckBox" name="groupWhenFull" >
-       <property name="enabled" >
+      <widget class="QCheckBox" name="groupWhenFull">
+       <property name="enabled">
         <bool>false</bool>
        </property>
-       <property name="text" >
+       <property name="text">
         <string>Only when the taskbar is full</string>
        </property>
-       <property name="checked" >
+       <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
-   <item row="9" column="1" >
-    <widget class="QLabel" name="label_2" >
-     <property name="text" >
+   <item row="6" column="1">
+    <widget class="QLabel" name="label_2">
+     <property name="text">
       <string>Sorting:</string>
      </property>
-     <property name="alignment" >
+     <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
-   <item row="9" column="2" >
-    <layout class="QHBoxLayout" name="horizontalLayout_3" >
+   <item row="6" column="2">
+    <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
-      <widget class="QComboBox" name="sortingStrategy" >
-       <property name="minimumSize" >
+      <widget class="QComboBox" name="sortingStrategy">
+       <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
@@ -229,11 +251,11 @@
       </widget>
      </item>
      <item>
-      <spacer name="horizontalSpacer_4" >
-       <property name="orientation" >
+      <spacer name="horizontalSpacer_4">
+       <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
-       <property name="sizeHint" stdset="0" >
+       <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
@@ -243,28 +265,28 @@
      </item>
     </layout>
    </item>
-   <item row="10" column="0" >
-    <widget class="QLabel" name="label_4" >
-     <property name="font" >
+   <item row="7" column="0">
+    <widget class="QLabel" name="label_4">
+     <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
-     <property name="text" >
+     <property name="text">
       <string>Filters</string>
      </property>
     </widget>
    </item>
-   <item row="10" column="1" >
-    <spacer name="verticalSpacer_4" >
-     <property name="orientation" >
+   <item row="7" column="1">
+    <spacer name="verticalSpacer_4">
+     <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
-     <property name="sizeType" >
+     <property name="sizeType">
       <enum>QSizePolicy::Fixed</enum>
      </property>
-     <property name="sizeHint" stdset="0" >
+     <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>10</height>
@@ -272,33 +294,33 @@
      </property>
     </spacer>
    </item>
-   <item row="11" column="1" colspan="2" >
-    <widget class="QCheckBox" name="showOnlyCurrentDesktop" >
-     <property name="text" >
+   <item row="8" column="1" colspan="2">
+    <widget class="QCheckBox" name="showOnlyCurrentDesktop">
+     <property name="text">
       <string>Only show tasks from the current desktop</string>
      </property>
     </widget>
    </item>
-   <item row="12" column="1" colspan="2" >
-    <widget class="QCheckBox" name="showOnlyCurrentScreen" >
-     <property name="text" >
+   <item row="9" column="1" colspan="2">
+    <widget class="QCheckBox" name="showOnlyCurrentScreen">
+     <property name="text">
       <string>Only show tasks from the current screen</string>
      </property>
     </widget>
    </item>
-   <item row="13" column="1" colspan="2" >
-    <widget class="QCheckBox" name="showOnlyMinimized" >
-     <property name="text" >
+   <item row="10" column="1" colspan="2">
+    <widget class="QCheckBox" name="showOnlyMinimized">
+     <property name="text">
       <string>Only show tasks that are minimized</string>
      </property>
     </widget>
    </item>
-   <item row="14" column="1" >
-    <spacer name="verticalSpacer" >
-     <property name="orientation" >
+   <item row="11" column="1">
+    <spacer name="verticalSpacer">
+     <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
-     <property name="sizeHint" stdset="0" >
+     <property name="sizeHint" stdset="0">
       <size>
        <width>0</width>
        <height>5</height>
