--- cupshelpers.py.orig	2008-01-22 10:36:48.000000000 +0200
+++ cupshelpers.py	2008-01-22 10:35:17.000000000 +0200
@@ -19,7 +19,12 @@
 ## Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
 import cups, pprint, os, tempfile, re
-from rhpl.translate import _, N_
+
+# remove RHPL and add gettext
+import gettext
+__trans = gettext.translation('system-config-printer', fallback=True)
+_ = __trans.ugettext
+
 import locale
 
 def debugprint(x):
