diff -Naur pencil_svn_20090809.orig/src/interface/mainwindow.cpp pencil_svn_20090809/src/interface/mainwindow.cpp
--- pencil_svn_20090809.orig/src/interface/mainwindow.cpp	2009-08-09 01:26:18.130057079 +0300
+++ pencil_svn_20090809/src/interface/mainwindow.cpp	2009-08-09 01:49:31.217929864 +0300
@@ -202,7 +202,7 @@
 void MainWindow::loadPlugins() {
 	qDebug() << "MainWindow loadplugins" << this << this->thread();
 	// foreach (QObject *plugin, QPluginLoader::staticInstances()) populateMenus(plugin); // static plugins
-	QDir pluginsDir = QDir(qApp->applicationDirPath());
+	QDir pluginsDir = QDir(QString("/usr/lib/pencil"));
  #if defined(Q_OS_WIN)
      //if (pluginsDir.dirName().toLower() == "debug" || pluginsDir.dirName().toLower() == "release")
      //    pluginsDir.cdUp();
