--- install.sh.orig	2007-09-17 03:20:06.000000000 +0300
+++ install.sh	2007-09-17 03:26:16.000000000 +0300
@@ -53,15 +53,6 @@
 usethisperl=$1
 echo I will be using $1 because I know it will work.
 
-echo Now I need an installation prefix.  I will use /usr/local unless
-printf "you give me a better idea here: "
-if [ -n "$backdoor" ]; then
-	prefix=$backdoor
-	printf "%s (specified on command line)\n" $prefix
-else
-	read prefix
-fi
-
 PREFIX=${prefix:-/usr/local}
 
 echo Okay, time to install this puppy.
