diff -ur libXrandr-1.2.0.orig/src/XrrMode.c libXrandr-1.2.0/src/XrrMode.c
--- libXrandr-1.2.0.orig/src/XrrMode.c	2006-12-13 11:24:25.000000000 +0200
+++ libXrandr-1.2.0/src/XrrMode.c	2007-02-26 02:56:15.130227139 +0200
@@ -56,7 +56,7 @@
     xRRCreateModeReply	    rep;
     long		    channelSize;
 
-    RRSimpleCheckExtension (dpy, info);
+    RRCheckExtension (dpy, info, None);
 
     LockDisplay(dpy);
     GetReq (RRCreateMode, req);
