diff -up alienarena-7.21/source/qcommon/files.c.BAD alienarena-7.21/source/qcommon/files.c
--- alienarena-7.21/source/qcommon/files.c.BAD	2009-01-30 19:52:35.000000000 -0500
+++ alienarena-7.21/source/qcommon/files.c	2009-01-30 19:53:25.000000000 -0500
@@ -582,6 +582,7 @@ void FS_AddGameDirectory (char *dir)
 #else
 	strcpy (search->filename, dir);
 #endif
+	Com_Printf("added %s to search paths\n", search->filename);
 	search->next = fs_searchpaths;
 	fs_searchpaths = search;
 
