diff -up alienarena-7.32/source/unix/gl_glx.c~ alienarena-7.32/source/unix/gl_glx.c
--- alienarena-7.32/source/unix/gl_glx.c~	2009-11-03 03:00:31.000000000 +0100
+++ alienarena-7.32/source/unix/gl_glx.c	2009-11-07 13:41:22.000000000 +0100
@@ -53,7 +53,7 @@ Foundation, Inc., 59 Temple Place - Suit
 #include <X11/keysym.h>
 #include <X11/cursorfont.h>
 
-#include <X11/extensions/xf86dga.h>
+#include <X11/extensions/xf86dga1.h>
 #include <X11/extensions/xf86vmode.h>
 
 #include <GL/glx.h>
