--- alienarena-7.33/source/qcommon/files.c.orig	2010-03-17 15:25:49.416878798 +0200
+++ alienarena-7.33/source/qcommon/files.c	2010-03-17 15:26:36.909897031 +0200
@@ -19,7 +19,7 @@
 */
 
 #include "qcommon.h"
-#include "../unix/glob.h"
+#include "../unix/glob.h"
 
 // define this to dissalow any data but the demo pak file
 //#define	NO_ADDONS
@@ -568,7 +568,7 @@
 ================
 FS_AddHomeAsGameDirectory
 
-Use ~/.codered/dir as fs_gamedir
+Use ~/.alienarena/dir as fs_gamedir
 ================
 */
 void FS_AddHomeAsGameDirectory (char *dir)
@@ -586,7 +586,7 @@
 		homedir = getenv("HOME");
 		if (!homedir)
 			return;
-		len = snprintf(gdir,sizeof(gdir),"%s/.codered/%s/", homedir, dir);
+		len = snprintf(gdir,sizeof(gdir),"%s/.alienarena/%s/", homedir, dir);
 	}
 
 	Com_Printf("using %s for writing\n",gdir);
