diff -ruN devil-1.7.5.orig/src-IL/include/il_internal.h devil-1.7.5/src-IL/include/il_internal.h
--- devil-1.7.5.orig/src-IL/include/il_internal.h	2008-12-29 22:02:15.000000000 -0500
+++ devil-1.7.5/src-IL/include/il_internal.h	2009-01-18 15:56:15.000000000 -0500
@@ -403,12 +403,14 @@
 ILboolean ilSaveTiff(ILconst_string FileName);
 ILboolean ilSaveTiffF(ILHANDLE File);
 ILboolean ilSaveTiffL(const void *Lump, ILuint Size);
+#ifndef IL_NO_VTF
 ILboolean ilIsValidVtf(ILconst_string FileName);
 ILboolean ilIsValidVtfF(ILHANDLE File);
 ILboolean ilIsValidVtfL(const void *Lump, ILuint Size);
 ILboolean ilLoadVtf(ILconst_string FileName);
 ILboolean ilLoadVtfF(ILHANDLE File);
 ILboolean ilLoadVtfL(const void *Lump, ILuint Size);
+#endif
 ILboolean ilLoadWal(ILconst_string FileName);
 ILboolean ilLoadWalF(ILHANDLE File);
 ILboolean ilLoadWalL(const void *Lump, ILuint Size);
