diff -Nur avidemux_2.4.1-old/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/DIA_x264.cpp avidemux_2.4.1/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/DIA_x264.cpp
--- avidemux_2.4.1-old/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/DIA_x264.cpp	2008-02-24 18:51:12.770448160 +0200
+++ avidemux_2.4.1/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/DIA_x264.cpp	2008-02-24 18:51:51.973488392 +0200
@@ -66,7 +66,7 @@
                 ,{9,    QT_TR_NOOP("7B - Ultra High (RDO on Bframes)")}};
 
                             
-        diaElemMenu parition(PX(PartitionDecision),QT_TR_NOOP("Partition Decision"),10,partitionM);
+        //diaElemMenu parition(PX(PartitionDecision),QT_TR_NOOP("Partition Decision"),10,partitionM);
         
          diaMenuEntry meM[] = {
                              {0,    QT_TR_NOOP("Diamond Search")},
@@ -95,7 +95,7 @@
          diaElemToggle    interlaced(PX(interlaced),QT_TR_NOOP("Interlaced"));
          
          diaElemFrame  frameMe(QT_TR_NOOP("Motion Estimation"));
-         frameMe.swallow(&parition);
+         //frameMe.swallow(&parition);
          frameMe.swallow(&me);
          frameMe.swallow(&rframe);
          frameMe.swallow(&range);
