diff -Naur supertux-0.3.1.orig/src/addon_manager.cpp supertux-0.3.1/src/addon_manager.cpp
--- supertux-0.3.1.orig/src/addon_manager.cpp	2008-01-03 22:59:28.000000000 +0200
+++ supertux-0.3.1/src/addon_manager.cpp	2008-05-10 12:27:02.000000000 +0300
@@ -22,6 +22,7 @@
 #include <sstream>
 #include <stdexcept>
 #include <list>
+#include <cstdlib>
 #include <physfs.h>
 #include <sys/stat.h>
 #include <stdio.h>
diff -Naur supertux-0.3.1.orig/src/console.hpp supertux-0.3.1/src/console.hpp
--- supertux-0.3.1.orig/src/console.hpp	2008-01-03 22:59:27.000000000 +0200
+++ supertux-0.3.1/src/console.hpp	2008-05-10 12:27:28.000000000 +0300
@@ -26,6 +26,7 @@
 #include <string>
 #include <sstream>
 #include <iostream>
+#include <memory>
 #include <squirrel.h>
 
 class Console;
diff -Naur supertux-0.3.1.orig/src/lisp/lexer.cpp supertux-0.3.1/src/lisp/lexer.cpp
--- supertux-0.3.1.orig/src/lisp/lexer.cpp	2008-01-03 22:59:26.000000000 +0200
+++ supertux-0.3.1/src/lisp/lexer.cpp	2008-05-10 12:27:54.000000000 +0300
@@ -21,6 +21,7 @@
 #include <sstream>
 #include <stdexcept>
 #include <iostream>
+#include <cstring>
 
 #include "lexer.hpp"
 
diff -Naur supertux-0.3.1.orig/src/sprite/sprite_manager.hpp supertux-0.3.1/src/sprite/sprite_manager.hpp
--- supertux-0.3.1.orig/src/sprite/sprite_manager.hpp	2008-01-03 22:59:26.000000000 +0200
+++ supertux-0.3.1/src/sprite/sprite_manager.hpp	2008-05-10 12:29:06.000000000 +0300
@@ -21,6 +21,7 @@
 #define SUPERTUX_SPRITE_MANAGER_H
 
 #include <map>
+#include <iterator>
 
 class SpriteData;
 class Sprite;
diff -Naur supertux-0.3.1.orig/src/textscroller.hpp supertux-0.3.1/src/textscroller.hpp
--- supertux-0.3.1.orig/src/textscroller.hpp	2008-01-03 22:59:27.000000000 +0200
+++ supertux-0.3.1/src/textscroller.hpp	2008-05-10 12:33:49.000000000 +0300
@@ -24,6 +24,7 @@
 #include <vector>
 #include <string>
 #include <map>
+#include <memory>
 
 #include "screen.hpp"
 #include "math/vector.hpp"
diff -Naur supertux-0.3.1.orig/src/tile.cpp supertux-0.3.1/src/tile.cpp
--- supertux-0.3.1.orig/src/tile.cpp	2008-01-03 22:59:27.000000000 +0200
+++ supertux-0.3.1/src/tile.cpp	2008-05-10 12:29:28.000000000 +0300
@@ -24,6 +24,7 @@
 #include <assert.h>
 #include <iostream>
 #include <stdexcept>
+#include <algorithm>
 
 #include "lisp/lisp.hpp"
 #include "tile.hpp"
diff -Naur supertux-0.3.1.orig/src/title.cpp supertux-0.3.1/src/title.cpp
--- supertux-0.3.1.orig/src/title.cpp	2008-01-03 22:59:28.000000000 +0200
+++ supertux-0.3.1/src/title.cpp	2008-05-10 12:35:35.000000000 +0300
@@ -20,6 +20,7 @@
 //  02111-1307, USA.
 #include <config.h>
 
+#include <algorithm>
 #include <iostream>
 #include <sstream>
 #include <stdexcept>
diff -Naur supertux-0.3.1.orig/src/video/sdl_texture.hpp supertux-0.3.1/src/video/sdl_texture.hpp
--- supertux-0.3.1.orig/src/video/sdl_texture.hpp	2008-01-03 22:59:22.000000000 +0200
+++ supertux-0.3.1/src/video/sdl_texture.hpp	2008-05-10 12:28:24.000000000 +0300
@@ -20,6 +20,7 @@
 #ifndef __SDL_TEXTURE_HPP__
 #define __SDL_TEXTURE_HPP__
 
+#include <algorithm>
 #include <config.h>
 
 #include <SDL.h>
