--- src/interface.c.orig	2008-05-19 17:58:29.000000000 +0300
+++ src/interface.c	2008-05-19 17:59:12.000000000 +0300
@@ -1255,11 +1255,6 @@
   gtk_widget_show (dialog_action_area4);
   gtk_button_box_set_layout (GTK_BUTTON_BOX (dialog_action_area4), GTK_BUTTONBOX_END);
 
-  button_pause = gtk_button_new_from_stock ("gtk-media-pause");
-  gtk_widget_show (button_pause);
-  gtk_dialog_add_action_widget (GTK_DIALOG (dialog_rsync), button_pause, 0);
-  gtk_tooltips_set_tip (tooltips, button_pause, _("Pause/resume rsync run"), NULL);
-
   close = gtk_button_new_from_stock ("gtk-stop");
   gtk_widget_show (close);
   gtk_dialog_add_action_widget (GTK_DIALOG (dialog_rsync), close, 0);
