Index: k3b-1.66.0/k3bsetup/k3bsetup.desktop
===================================================================
--- k3b-1.66.0.orig/k3bsetup/k3bsetup.desktop
+++ k3b-1.66.0/k3bsetup/k3bsetup.desktop
@@ -36,7 +36,7 @@ Name[x-test]=xxK3bSetupxx
 Terminal=false
 Type=Service
 Icon=k3b
-X-KDE-ServiceTypes=KCModule
+#X-KDE-ServiceTypes=KCModule
 X-KDE-Library=kcm_k3bsetup
 X-KDE-ParentApp=kcontrol
 X-KDE-System-Settings-Parent-Category=system
