Index: k3b-1.66.0/src/konqi/k3b_create_audio_cd_from_blank_medium.desktop
===================================================================
--- k3b-1.66.0.orig/src/konqi/k3b_create_audio_cd_from_blank_medium.desktop
+++ k3b-1.66.0/src/konqi/k3b_create_audio_cd_from_blank_medium.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-X-KDE-Solid-Predicate=[[StorageVolume.ignored == false AND OpticalDisc.isBlank == true ] AND OpticalDisc.discType == 'CdRecordable|CdRewritable']
+X-KDE-Solid-Predicate=[[StorageVolume.ignored == false AND OpticalDisc.blank == true ] AND [OpticalDisc.discType == 'CdRecordable' OR OpticalDisc.discType == 'CdRewritable']]
 Actions=K3bAudioCDProject;
 Type=Service
 X-KDE-Priority=TopLevel
Index: k3b-1.66.0/src/konqi/k3b_create_data_project_from_blank_medium.desktop
===================================================================
--- k3b-1.66.0.orig/src/konqi/k3b_create_data_project_from_blank_medium.desktop
+++ k3b-1.66.0/src/konqi/k3b_create_data_project_from_blank_medium.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-X-KDE-Solid-Predicate=[StorageVolume.ignored == false AND OpticalDisc.isBlank == true]
+X-KDE-Solid-Predicate=[StorageVolume.ignored == false AND OpticalDisc.blank == true]
 Actions=K3bDataProject;
 Type=Service
 X-KDE-Priority=TopLevel
