Index: k3b-1.66.0/src/k3b.notifyrc
===================================================================
--- k3b-1.66.0.orig/src/k3b.notifyrc
+++ k3b-1.66.0/src/k3b.notifyrc
@@ -88,7 +88,7 @@ Comment[tr]=İşlem başarıyla sonuçla
 Comment[uk]=Процес закінчився успішно
 Comment[x-test]=xxProcess successfully finishedxx
 Comment[zh_CN]=处理成功完成
-Sound=k3b_success1.wav
+Sound=k3b_success1.ogg
 Action=Sound
 
 [Event/FinishedWithError]
@@ -174,7 +174,7 @@ Comment[tr]=İşlem hatayla sonuçlandı
 Comment[uk]=Процес закінчився з помилками
 Comment[x-test]=xxProcess finished with errorsxx
 Comment[zh_CN]=处理完成，有错误
-Sound=k3b_error1.wav
+Sound=k3b_error1.ogg
 Action=Sound
 
 [Event/WaitingForMedium]
@@ -259,5 +259,5 @@ Comment[tr]=Kullanıcının bir ortam ta
 Comment[uk]=Користувач повинен вставити носій
 Comment[x-test]=xxThe user needs to insert a mediumxx
 Comment[zh_CN]=用户需要插入存储介质
-Sound=k3b_wait_media1.wav
+Sound=KDE-K3B-Insert-Medium.ogg
 Action=Sound
Index: k3b-1.66.0/src/sounds/CMakeLists.txt
===================================================================
--- k3b-1.66.0.orig/src/sounds/CMakeLists.txt
+++ k3b-1.66.0/src/sounds/CMakeLists.txt
@@ -1 +1 @@
-install( FILES  k3b_success1.wav k3b_error1.wav k3b_wait_media1.wav DESTINATION ${SOUND_INSTALL_DIR} )
+install( FILES  k3b_success1.ogg k3b_error1.ogg DESTINATION ${SOUND_INSTALL_DIR} )
