--- src/klamd.org.cpp	2007-12-11 11:59:10.000000000 +0200
+++ src/klamd.cpp	2007-12-11 12:04:05.000000000 +0200
@@ -815,7 +815,7 @@
 // 	
 // 	//kdDebug() << release << endl;
 // 	QString insmodCommand="insmod /lib/modules/" + release + "/dazuko.o";
-	QString insmodCommand="modprobe dazuko";
+	QString serviceCommand="service dazuko start";
 	//QString insmodCommand="insmod /opt/kde/bin/dazuko.o";
 	KProcIO *insmodproc = new KProcIO();
 	
@@ -823,7 +823,7 @@
 	insmodproc->setEnvironment("PATH",curpath + ":/usr/local/sbin:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin");
 
 	insmodproc->setUseShell(TRUE);
-	*insmodproc<<"kdesu -i klamav"<<"-c"<<insmodCommand;
+	*insmodproc<<"kdesu -i klamav"<<"-c"<<serviceCommand;
 	
 	connect( insmodproc, SIGNAL(processExited(KProcess *)),
 		SLOT(moduleLoaded()) );
