diff -Nur coreutils-7.4-old/bootstrap.conf coreutils-7.4/bootstrap.conf
--- coreutils-7.4-old/bootstrap.conf	2009-05-13 12:01:00.399402744 +0300
+++ coreutils-7.4/bootstrap.conf	2009-05-13 12:01:57.364402073 +0300
@@ -301,7 +301,7 @@
 # Build prerequisites
 buildreq="\
 autoconf   2.61
-automake   1.10b
+automake   1.10
 autopoint  -
 bison      -
 gettext    -
diff -Nur coreutils-7.4-old/configure.ac coreutils-7.4/configure.ac
--- coreutils-7.4-old/configure.ac	2009-05-13 12:01:00.568651316 +0300
+++ coreutils-7.4/configure.ac	2009-05-13 12:02:07.824463179 +0300
@@ -32,7 +32,7 @@
 AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_HEADERS([lib/config.h:lib/config.hin])
 
-AM_INIT_AUTOMAKE([1.10b dist-xz color-tests parallel-tests])
+AM_INIT_AUTOMAKE([1.10])
 # AM_SILENT_RULES([yes]) # make --enable-silent-rules the default.
 
 AC_PROG_CC_STDC
