diff -Nur coreutils-7.4-old/configure.ac coreutils-7.4/configure.ac
--- coreutils-7.4-old/configure.ac	2009-05-13 11:56:51.096526519 +0300
+++ coreutils-7.4/configure.ac	2009-05-13 11:57:11.289401605 +0300
@@ -33,7 +33,7 @@
 AC_CONFIG_HEADERS([lib/config.h:lib/config.hin])
 
 AM_INIT_AUTOMAKE([1.10b dist-xz color-tests parallel-tests])
-AM_SILENT_RULES([yes]) # make --enable-silent-rules the default.
+# AM_SILENT_RULES([yes]) # make --enable-silent-rules the default.
 
 AC_PROG_CC_STDC
 AM_PROG_CC_C_O
