--- fnfx-0.3/src/fnfxd_cmds.c.syspowerstate	2006-03-22 20:23:33.000000000 +0100
+++ fnfx-0.3/src/fnfxd_cmds.c	2006-03-22 20:25:38.000000000 +0100
@@ -362,9 +362,9 @@
     }
     else {
         if (state == 3)
-            fprintf(f, "%d", state);
+            fprintf(f, "mem");
         else if (state == 4)
-            fprintf(f, "%d", state);
+            fprintf(f, "disk");
         else {
             debug("suspend() - Not supported sleep state.");
             return 1;
--- fnfx-0.3/src/fnfx.h.syspowerstate	2006-03-22 20:23:10.000000000 +0100
+++ fnfx-0.3/src/fnfx.h	2006-03-22 20:24:21.000000000 +0100
@@ -50,7 +50,7 @@
 #define ACPI_LCD	"/proc/acpi/toshiba/lcd"
 #define ACPI_VERSION	"/proc/acpi/toshiba/version"
 #define ACPI_VIDEO	"/proc/acpi/toshiba/video"
-#define ACPI_SLEEP	"/proc/acpi/sleep"
+#define ACPI_SLEEP	"/sys/power/state"
 #define ACPI_CPU	"/proc/acpi/processor/CPU0/limit"
 #define ACPI_BLUETOOTH	"/proc/acpi/toshiba/bluetooth"
