Index: lightdm-0.3.6/liblightdm-qt/QLightDM/Makefile.am
===================================================================
--- lightdm-0.3.6.orig/liblightdm-qt/QLightDM/Makefile.am
+++ lightdm-0.3.6/liblightdm-qt/QLightDM/Makefile.am
@@ -33,7 +33,7 @@ liblightdm_qt_0include_HEADERS = \
 
 liblightdm_qt_0includedir=$(includedir)/lightdm-qt-0/lightdm
 
-liblightdm_qt_0_la_LIBADD = $(LIBLIGHTDM_QT_LIBS)
+liblightdm_qt_0_la_LIBADD = $(LIBLIGHTDM_QT_LIBS) -lQtGui
 liblightdm_qt_0_la_CXXFLAGS = $(LIBLIGHTDM_QT_CFLAGS) \
 	-DXSESSIONS_DIR=\"$(XSESSIONS_DIR)\"
 
