diff -ur ksystemlog-0.3.2.orig/ksystemlog/src/config/ksystemlog.kcfg ksystemlog-0.3.2/ksystemlog/src/config/ksystemlog.kcfg
--- ksystemlog-0.3.2.orig/ksystemlog/src/config/ksystemlog.kcfg	2005-07-06 19:40:47.000000000 +0300
+++ ksystemlog-0.3.2/ksystemlog/src/config/ksystemlog.kcfg	2006-06-21 19:08:02.000000000 +0300
@@ -70,11 +70,11 @@
 		<!-- Kernel config entries -->
 		<entry name="KernelPaths" type="StringList">
 			<label>The Kernel log files paths.</label>
-			<default>/var/log/kernel/info,/var/log/kernel/warnings,/var/log/kernel/errors</default>
+			<default>/var/log/kern.log</default>
 		</entry>
 		<entry name="KernelLevels" type="IntList">
 			<label>The Kernel log files levels.</label>
-			<default>2,4,5</default>
+			<default>0</default>
 		</entry>
 		
 		<!-- Cron config entries -->
@@ -90,11 +90,11 @@
 		<!-- Daemon config entries -->
 		<entry name="DaemonPaths" type="StringList">
 			<label>The Daemon log files paths.</label>
-			<default>/var/log/daemons/info,/var/log/daemons/warnings,/var/log/daemons/errors</default>
+			<default>/var/log/daemon.log</default>
 		</entry>
 		<entry name="DaemonLevels" type="IntList">
 			<label>The Daemon log files levels.</label>
-			<default>2,4,5</default>
+			<default>0</default>
 		</entry>
 
 		<!-- X.org config entries -->
@@ -118,19 +118,19 @@
 		<!-- Apache config entries -->
 		<entry name="ApachePaths" type="StringList">
 			<label>The Apache log files paths.</label>
-			<default>/var/log/httpd/error_log,/var/log/httpd/ssl_error_log</default>
+			<default>/var/log/apache2/error_log,/var/log/apache2/ssl_error_log</default>
 		</entry>
 		
 		<!-- Apache Access config entries -->
 		<entry name="ApacheAccessPaths" type="StringList">
 			<label>The Apache Access log files paths.</label>
-			<default>/var/log/httpd/access_log,/var/log/httpd/ssl_access_log</default>
+			<default>/var/log/apache2/access_log,/var/log/apache2/ssl_access_log</default>
 		</entry>
 		
 		<!-- Postfix config entries -->
 		<entry name="PostfixPaths" type="StringList">
 			<label>The Postfix log files paths.</label>
-			<default>/var/log/mail/info,/var/log/mail/warnings,/var/log/mail/errors</default>
+			<default>/var/log/mail.info,/var/log/mail.warn,/var/log/mail.err</default>
 		</entry>
 		<entry name="PostfixLevels" type="IntList">
 			<label>The Postfix log files levels.</label>
@@ -153,7 +153,7 @@
 		<!-- Boot config entries -->
 		<entry name="BootPath" type="String">
 			<label>The Boot log file path.</label>
-			<default>/var/log/boot.log</default>
+			<default>/var/log/syslog</default>
 		</entry>
 
 		<!-- Authentication config entries -->
@@ -178,4 +178,5 @@
 
 	</group>
 	
-</kcfg>
\ Dosya sonunda yenisatır yok.
+</kcfg>
+
