diff -ur yakuake-2.7.5.orig/yakuake/src/yakuake.desktop yakuake-2.7.5/yakuake/src/yakuake.desktop
--- yakuake-2.7.5.orig/yakuake/src/yakuake.desktop	2006-12-10 17:38:29.000000000 +0200
+++ yakuake-2.7.5/yakuake/src/yakuake.desktop	2006-12-10 17:39:30.000000000 +0200
@@ -1,7 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=Yakuake
-Name[tr]=Yakuake 
 Exec=yakuake
 Icon=yakuake
 Type=Application
@@ -14,4 +13,5 @@
 Comment[pt_BR]=Um emulador de terminal no estilo do Quake, baseado na tecnologia do Konsole do KDE. 
 Comment[sv]=En Quake-liknande terminalemulator baserad på teknologi från KDE:s Terminal.
 Comment[tr]=KDE Konsole teknolojisi üzerine kurulu Quake stili bir terminal emülatörü
+GenericName[tr]=Terminal Programı
 X-KDE-StartupNotify=false
