diff -Naur qtrans-0.2.2.1.orig/src/QTrans.desktop qtrans-0.2.2.1/src/QTrans.desktop
--- qtrans-0.2.2.1.orig/src/QTrans.desktop	2009-12-19 22:03:16.537534063 +0200
+++ qtrans-0.2.2.1/src/QTrans.desktop	2009-12-19 22:09:38.810407769 +0200
@@ -1,5 +1,7 @@
 [Desktop Entry]
-Name=qtrans
+Name=QTrans
+GenericName=A KDE Dictionary Application
+GenericName[tr]=KDE Sözlük Uygulaması
 Exec=qtrans
 Icon=qtrans
 Type=Application
