Index: libnut/muxer.c
===================================================================
--- libnut/muxer.c	(revision 658)
+++ libnut/muxer.c	(working copy)
@@ -181,6 +181,8 @@
 		if (fields > 4) put_v(tmp, 0); // reserved
 		if (fields > 5) put_v(tmp, count);
 	}
+	put_v(tmp, 0); // header_count_minus1
+	put_v(tmp, 0); // main_flags
 
 	put_header(nut->o, tmp, nut->tmp_buffer2, MAIN_STARTCODE, 0);
 }
