diff -ur glibc-2.3.6.orig/iconvdata/gen-8bit-gap-1.sh glibc-2.3.6/iconvdata/gen-8bit-gap-1.sh
--- glibc-2.3.6.orig/iconvdata/gen-8bit-gap-1.sh	2006-04-02 01:22:30.000000000 +0300
+++ glibc-2.3.6/iconvdata/gen-8bit-gap-1.sh	2006-04-02 01:26:26.000000000 +0300
@@ -1,4 +1,5 @@
 #! /bin/sh
+export LC_ALL=C
 echo "static const uint32_t iso88597_to_ucs4[96] = {"
 sed -ne '/^[^[:space:]]*[[:space:]]*.x00/d' \
     -e 's/^<U\(....\)>[[:space:]]*.x\([A-Fa-f].\).*/  [0x\2 - 0xA0] = 0x\1,/p' \
diff -ur glibc-2.3.6.orig/iconvdata/gen-8bit-gap.sh glibc-2.3.6/iconvdata/gen-8bit-gap.sh
--- glibc-2.3.6.orig/iconvdata/gen-8bit-gap.sh	2006-04-02 01:22:31.000000000 +0300
+++ glibc-2.3.6/iconvdata/gen-8bit-gap.sh	2006-04-02 01:26:31.000000000 +0300
@@ -1,4 +1,5 @@
 #! /bin/sh
+export LC_ALL=C
 echo "static const uint32_t to_ucs4[256] = {"
 sed -ne '/^[^[:space:]]*[[:space:]]*.x00/d' \
     -e 's/^<U\(....\)>[[:space:]]*.x\(..\).*/  [0x\2] = 0x\1,/p' \
diff -ur glibc-2.3.6.orig/iconvdata/gen-8bit.sh glibc-2.3.6/iconvdata/gen-8bit.sh
--- glibc-2.3.6.orig/iconvdata/gen-8bit.sh	2006-04-02 01:22:34.000000000 +0300
+++ glibc-2.3.6/iconvdata/gen-8bit.sh	2006-04-02 01:26:38.000000000 +0300
@@ -1,4 +1,5 @@
 #! /bin/sh
+export LC_ALL=C
 echo "static const uint32_t to_ucs4[256] = {"
 sed -ne '/^[^[:space:]]*[[:space:]]*.x00/d' \
     -e 's/^<U\(....\)>[[:space:]]*.x\(..\).*/  [0x\2] = 0x\1,/p' \
