diff -Nur beagle-0.2.16.2-old/acinclude.m4 beagle-0.2.16.2/acinclude.m4
--- beagle-0.2.16.2-old/acinclude.m4	2007-03-10 19:00:31.183066020 +0200
+++ beagle-0.2.16.2/acinclude.m4	2007-03-10 19:01:01.131513163 +0200
@@ -121,7 +121,7 @@
 ## this one is commonly used with AM_PATH_PYTHONDIR ...
 dnl AM_CHECK_PYMOD(MODNAME [,SYMBOL [,ACTION-IF-FOUND [,ACTION-IF-NOT-FOUND]]])
 dnl Check if a module containing a given symbol is visible to python.
-AC_DEFUN(AM_CHECK_PYMOD,
+AC_DEFUN([AM_CHECK_PYMOD],
 [AC_REQUIRE([AM_PATH_PYTHON])
 py_mod_var=`echo $1['_']$2 | sed 'y%./+-%__p_%'`
 AC_MSG_CHECKING(for ifelse([$2],[],,[$2 in ])python module $1)
