--- ext/ffmpeg/gstffmpegcodecmap.c	2006-12-07 20:07:32.000000000 +0200
+++ ext/ffmpeg/gstffmpegcodecmap.c	2007-06-27 00:32:26.000000000 +0300
@@ -523,7 +523,7 @@
       caps = gst_ff_vid_caps_new (context, "video/x-theora", NULL);
       break;
 
-    case CODEC_ID_AAC:
+    //case CODEC_ID_AAC:
     case CODEC_ID_MPEG4AAC:
       caps = gst_ff_aud_caps_new (context, "audio/mpeg",
           "mpegversion", G_TYPE_INT, 4, NULL);
@@ -2492,7 +2492,7 @@
     case CODEC_ID_THEORA:
       name = "Theora video";
       break;
-    case CODEC_ID_AAC:
+    //case CODEC_ID_AAC:
     case CODEC_ID_MPEG4AAC:
       name = "MPEG-2/4 AAC audio";
       break;
--- ext/ffmpeg/gstffmpegdemux.c	2007-06-27 00:33:10.000000000 +0300
+++ ext/ffmpeg/gstffmpegdemux.c	2007-06-27 00:34:34.000000000 +0300
@@ -131,9 +131,9 @@
   const gchar *message = NULL;
 
   switch (av_errno) {
-    case AVERROR_UNKNOWN:
-      message = "Unknown error";
-      break;
+    //case AVERROR_UNKNOWN:
+    //  message = "Unknown error";
+    //  break;
     case AVERROR_IO:
       message = "Input/output error";
       break;
