Index: xapian-core-1.2.6/xapian-config.in
===================================================================
--- xapian-core-1.2.6.orig/xapian-config.in
+++ xapian-core-1.2.6/xapian-config.in
@@ -228,7 +228,7 @@ for arg in $actions ; do
 	echo "$I"
 	;;
 
-    --libs)
+    --libs|--ltlibs)
 	if is_uninstalled ; then
 	    if [ no = "$xo_lib_xapian" ]; then
 		echo "$0: Use libtool and '$PROG_NAME --ltlibs'" 1>&2
@@ -256,7 +256,7 @@ for arg in $actions ; do
 	echo "$F$L-lxapian@LIBRARY_VERSION_SUFFIX@$D"
 	;;
 
-    --ltlibs)
+    --ltlibs_la_files_missing)
 	set_F_to_extra_ldflags
 	if is_uninstalled ; then
 	    # Don't check if libxapian@LIBRARY_VERSION_SUFFIX@.la exists yet since we want to allow
