--- ode-0.7/configure~	2006-09-22 12:15:42.000000000 +0200
+++ ode-0.7/configure	2006-09-22 12:15:42.000000000 +0200
@@ -7117,7 +7117,7 @@
 echo "${ECHO_T}$dEpsilon" >&6
 
 
-if test "$build_os" == "$target_os"
+if test "$target_cpu" != "i386"
 then
 echo "$as_me:$LINENO: checking for a Pentium CPU" >&5
 echo $ECHO_N "checking for a Pentium CPU... $ECHO_C" >&6
--- ode-0.7/configure.in~	2006-09-22 12:15:58.000000000 +0200
+++ ode-0.7/configure.in	2006-09-22 12:15:58.000000000 +0200
@@ -220,7 +220,7 @@
 
 
 dnl Check for PENTIUM
-if test "$build_os" == "$target_os"
+if test "$target_cpu" != "i386"
 then
 AC_MSG_CHECKING(for a Pentium CPU)
 AC_TRY_RUN([
