--- src/tivion.desktop.orig	2010-04-05 13:21:39.779989769 +0300
+++ src/tivion.desktop	2010-04-05 13:30:40.574989564 +0300
@@ -3,9 +3,11 @@
 Name=Tivion
 Comment=A single player to online TV streaming
 Comment[es]=Un simple reproductor de streaming para TV online
+Comment[tr]=İnternet TV
 GenericName=A single player to online TV streaming
 GenericName[es]=Un simple reproductor de streaming para TV online
-Icon=/usr/share/tivion/data/img/tivion-launcher.png
+GenericName[tr]=İnternet TV
+Icon=tivion
 Version=0.0.1
 Type=Application
 Terminal=false
