diff -Naur decibel-0.7.1_891686-orig/cmake/modules/FindTelepathyQt.cmake decibel-0.7.1_891686/cmake/modules/FindTelepathyQt.cmake
--- decibel-0.7.1_891686-orig/cmake/modules/FindTelepathyQt.cmake	2008-12-05 00:37:28.000000000 +0200
+++ decibel-0.7.1_891686/cmake/modules/FindTelepathyQt.cmake	2008-12-05 00:38:10.000000000 +0200
@@ -23,11 +23,11 @@
    # otherwise try to find it just like any other project:
    find_path(TELEPATHY_QT_INCLUDE_DIR
           NAMES QtTelepathy/Client/Channel QtTelepathy/Client/Connection
-          PATHS ${CMAKE_INSTALL_PREFIX}/include
+          PATHS /usr/qt/4/include
    )
    find_library(TELEPATHY_QT_LIBRARIES
              NAMES QtTelepathyClient
-             PATHS ${CMAKE_INSTALL_PREFIX}/lib
+             PATHS /usr/qt/4/lib
    )
 endif(KDESupport_SOURCE_DIR)
 
diff -Naur decibel-0.7.1_891686-orig/cmake/modules/FindTapiocaQt.cmake decibel-0.7.1_891686/cmake/modules/FindTapiocaQt.cmake
--- decibel-0.7.1_891686-orig/cmake/modules/FindTapiocaQt.cmake	2008-12-05 00:41:26.000000000 +0200
+++ decibel-0.7.1_891686/cmake/modules/FindTapiocaQt.cmake	2008-12-05 00:42:05.000000000 +0200
@@ -23,11 +23,11 @@
    # otherwise try to find it just like any other project:
    find_path(TAPIOCA_QT_INCLUDE_DIR
           NAMES QtTapioca/Channel QtTapioca/Connection
-          PATHS ${CMAKE_INSTALL_PREFIX}/include
+          PATHS /usr/qt/4/include
    )
    find_library(TAPIOCA_QT_LIBRARIES
              NAMES QtTapioca
-             PATHS ${CMAKE_INSTALL_PREFIX}/lib
+             PATHS /usr/qt/4/lib
    )
 endif(KDESupport_SOURCE_DIR)
 
