--- ltmain.sh	2008-02-04 09:59:40.000000000 +0200
+++ ltmain.sh	2008-02-04 10:00:59.000000000 +0200
@@ -2949,7 +2949,6 @@
 		test "X$dir" = "X$deplib" && dir="."
 		# We need an absolute path.
 		case $dir in
-		[\\/]* | [A-Za-z]:[\\/]*) absdir="$dir" ;;
 		*)
 		  absdir=`cd "$dir" && pwd`
 		  if test -z "$absdir"; then
--- libltdl/ltmain.sh	2008-02-01 18:39:51.000000000 +0200
+++ libltdl/ltmain.sh	2008-02-04 10:01:25.000000000 +0200
@@ -2935,7 +2935,6 @@
 		test "X$dir" = "X$deplib" && dir="."
 		# We need an absolute path.
 		case $dir in
-		[\\/]* | [A-Za-z]:[\\/]*) absdir="$dir" ;;
 		*)
 		  absdir=`cd "$dir" && pwd`
 		  if test -z "$absdir"; then
--- ltmain.in	2008-02-04 09:59:40.000000000 +0200
+++ ltmain.in	2008-02-04 10:01:52.000000000 +0200
@@ -2945,7 +2945,6 @@
 		test "X$dir" = "X$deplib" && dir="."
 		# We need an absolute path.
 		case $dir in
-		[\\/]* | [A-Za-z]:[\\/]*) absdir="$dir" ;;
 		*)
 		  absdir=`cd "$dir" && pwd`
 		  if test -z "$absdir"; then
