diff -Nur vdrift-2007-03-23-src-old/SConstruct vdrift-2007-03-23-src/SConstruct
--- vdrift-2007-03-23-src-old/SConstruct	2007-04-14 16:06:15.670680125 +0300
+++ vdrift-2007-03-23-src/SConstruct	2007-04-14 16:07:26.933846477 +0300
@@ -245,7 +245,7 @@
 #---------------#
 if env['release']:
     # release build, debugging off, optimizations on
-    env.Append(CCFLAGS = ['-O2', '-pipe'])
+    env.Append(CCFLAGS = ['-Wall'])
     # version is current build date
     #version = strftime("%Y-%m-%d")
     version = '2007-03-23'
