diff -urNad ghostscript-8.61.dfsg.1~/cups/pstopxl ghostscript-8.61.dfsg.1/cups/pstopxl
--- ghostscript-8.61.dfsg.1~/cups/pstopxl	2007-11-21 20:16:30.000000000 +0000
+++ ghostscript-8.61.dfsg.1/cups/pstopxl	2007-12-05 12:44:48.000000000 +0000
@@ -51,7 +51,7 @@
 
 # See if we have a filename on the command-line...
 if test -z "$6"; then
-	ifile="-"
+	ifile="-_"
 else
 	ifile="$6"
 fi
diff -urNad ghostscript-8.61.dfsg.1~/cups/pstopxl.in ghostscript-8.61.dfsg.1/cups/pstopxl.in
--- ghostscript-8.61.dfsg.1~/cups/pstopxl.in	2007-05-14 00:13:47.000000000 +0100
+++ ghostscript-8.61.dfsg.1/cups/pstopxl.in	2007-12-05 12:44:18.000000000 +0000
@@ -51,7 +51,7 @@
 
 # See if we have a filename on the command-line...
 if test -z "$6"; then
-	ifile="-"
+	ifile="-_"
 else
 	ifile="$6"
 fi
diff -urNad ghostscript-8.61.dfsg.1~/cups/pstoraster ghostscript-8.61.dfsg.1/cups/pstoraster
--- ghostscript-8.61.dfsg.1~/cups/pstoraster	2007-11-21 20:16:30.000000000 +0000
+++ ghostscript-8.61.dfsg.1/cups/pstoraster	2007-12-05 12:44:32.000000000 +0000
@@ -46,7 +46,7 @@
 
 # See if we have a filename on the command-line...
 if test -z "$6"; then
-	ifile="-"
+	ifile="-_"
 else
 	ifile="$6"
 fi
diff -urNad ghostscript-8.61.dfsg.1~/cups/pstoraster.in ghostscript-8.61.dfsg.1/cups/pstoraster.in
--- ghostscript-8.61.dfsg.1~/cups/pstoraster.in	2007-05-14 00:13:47.000000000 +0100
+++ ghostscript-8.61.dfsg.1/cups/pstoraster.in	2007-12-05 12:44:18.000000000 +0000
@@ -46,7 +46,7 @@
 
 # See if we have a filename on the command-line...
 if test -z "$6"; then
-	ifile="-"
+	ifile="-_"
 else
 	ifile="$6"
 fi
