--- trunk/gs/src/scfd.c	2007/09/25 13:31:24	8250
+++ trunk/gs/src/scfd.c	2008/07/29 04:44:05	8896
@@ -161,7 +161,7 @@
 /* makeup codes efficiently, since these are always a multiple of 64. */
 #define invert_data(rlen, black_byte, makeup_action, d)\
 	if ( rlen > qbit )\
-	{	*q++ ^= (1 << qbit) - 1;\
+	{	if (q >= ss->lbuf) *q++ ^= (1 << qbit) - 1; else q++;\
 		rlen -= qbit;\
 		switch ( rlen >> 3 )\
 		{\
