--- ./src/gdevpdtb.c.orig3	2009-04-16 13:44:13.000000000 -0400
+++ ./src/gdevpdtb.c	2009-04-16 13:46:14.000000000 -0400
@@ -190,7 +190,7 @@
 			&st_pdf_base_font, "pdf_base_font_alloc");
     const gs_font_name *pfname = pdf_choose_font_name((gs_font *)font, orig_name);
     gs_const_string font_name;
-    char fnbuf[3 + sizeof(long) / 3 + 1]; /* .F#######\0 */
+    char fnbuf[2*sizeof(long) + 3]; /* .F########\0 */
     int code;
 
     if (pbfont == 0)
