diff -Nur openal-0.0.8-old/admin/pkgconfig/openal.pc.in openal-0.0.8/admin/pkgconfig/openal.pc.in
--- openal-0.0.8-old/admin/pkgconfig/openal.pc.in	2006-11-17 13:57:19.000000000 +0200
+++ openal-0.0.8/admin/pkgconfig/openal.pc.in	2006-11-17 13:57:23.000000000 +0200
@@ -5,7 +5,6 @@
 
 Name: OpenAL
 Description:  OpenAL is a cross-platform 3D audio API.
-Requires: @requirements@
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -lopenal
 Cflags: -I${includedir}
