diff -ur soprano-2.4.1/client/localsocketclient.cpp soprano-2.4.1-missing-qdebug-header/client/localsocketclient.cpp
--- soprano-2.4.1/client/localsocketclient.cpp	2010-01-19 16:19:53.000000000 +0100
+++ soprano-2.4.1-missing-qdebug-header/client/localsocketclient.cpp	2010-03-23 21:30:53.000000000 +0100
@@ -23,6 +23,7 @@
 #include "clientconnection.h"
 #include "clientmodel.h"
 
+#include <QtCore/QDebug>
 #include <QtCore/QDir>
 #include <QtCore/QThread>
 #include <QtNetwork/QLocalSocket>
