--- setup.py.orig	2007-07-18 02:03:48.000000000 +0300
+++ setup.py	2008-01-22 04:25:49.000000000 +0200
@@ -1,7 +1,7 @@
 #-*- coding: ISO-8859-1 -*-
 # setup.py: the distutils script
 #
-# Copyright (C) 2004-2007 Gerhard Häring <gh@ghaering.de>
+# Copyright (C) 2004-2007 Gerhard Hï¿½ring <gh@ghaering.de>
 #
 # This file is part of pysqlite.
 #
@@ -105,11 +105,11 @@
         print "Fatal error: PYSQLITE_VERSION could not be detected!"
         sys.exit(1)
 
-    data_files = [("pysqlite2-doc",
+    data_files = [("share/doc/pysqlite2-doc",
                         glob.glob("doc/*.html") \
                       + glob.glob("doc/*.txt") \
                       + glob.glob("doc/*.css")),
-                   ("pysqlite2-doc/code",
+                   ("share/doc/pysqlite2-doc/code",
                         glob.glob("doc/code/*.py"))]
 
     py_modules = ["sqlite"]
