Index: bluez-utils-3.30/hcid/dbus-security.c
===================================================================
--- bluez-utils-3.30.orig/hcid/dbus-security.c
+++ bluez-utils-3.30/hcid/dbus-security.c
@@ -30,6 +30,7 @@
 #include <stdio.h>
 #include <errno.h>
 #include <stdlib.h>
+#include <limits.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
 
Index: bluez-utils-3.30/hcid/dbus-test.c
===================================================================
--- bluez-utils-3.30.orig/hcid/dbus-test.c
+++ bluez-utils-3.30/hcid/dbus-test.c
@@ -29,6 +29,7 @@
 #include <stdio.h>
 #include <errno.h>
 #include <stdlib.h>
+#include <limits.h>
 
 #include <bluetooth/bluetooth.h>
 #include <bluetooth/l2cap.h>
Index: bluez-utils-3.30/network/server.c
===================================================================
--- bluez-utils-3.30.orig/network/server.c
+++ bluez-utils-3.30/network/server.c
@@ -29,6 +29,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <errno.h>
+#include <limits.h>
 #include <sys/stat.h>
 #include <net/if.h>
 
Index: bluez-utils-3.30/network/common.c
===================================================================
--- bluez-utils-3.30.orig/network/common.c
+++ bluez-utils-3.30/network/common.c
@@ -34,6 +34,7 @@
 #include <sys/stat.h>
 #include <sys/wait.h>
 #include <net/if.h>
+#include <limits.h>
 
 #include <bluetooth/bluetooth.h>
 #include <bluetooth/l2cap.h>
Index: bluez-utils-3.30/network/connection.c
===================================================================
--- bluez-utils-3.30.orig/network/connection.c
+++ bluez-utils-3.30/network/connection.c
@@ -30,6 +30,7 @@
 #include <unistd.h>
 #include <sys/stat.h>
 #include <netinet/in.h>
+#include <limits.h>
 
 #include <bluetooth/bluetooth.h>
 #include <bluetooth/hci.h>
Index: bluez-utils-3.30/serial/manager.c
===================================================================
--- bluez-utils-3.30.orig/serial/manager.c
+++ bluez-utils-3.30/serial/manager.c
@@ -39,6 +39,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/un.h>
+#include <limits.h>
 
 #include <glib.h>
 
Index: bluez-utils-3.30/audio/device.c
===================================================================
--- bluez-utils-3.30.orig/audio/device.c
+++ bluez-utils-3.30/audio/device.c
@@ -31,6 +31,7 @@
 #include <unistd.h>
 #include <sys/stat.h>
 #include <netinet/in.h>
+#include <limits.h>
 
 #include <glib.h>
 #include <dbus/dbus.h>
