Index: linux-2.6.25/drivers/ide/ide-iops.c
===================================================================
--- linux-2.6.25.orig/drivers/ide/ide-iops.c
+++ linux-2.6.25/drivers/ide/ide-iops.c
@@ -567,6 +567,9 @@ static const struct drive_list_entry ivb
 	{ "TSSTcorp CDDVDW SH-S202J"	, "SB01"	},
 	{ "TSSTcorp CDDVDW SH-S202N"	, "SB00"	},
 	{ "TSSTcorp CDDVDW SH-S202N"	, "SB01"	},
+	{ "TSSTcorp CDDVDW SH-S202H"	, "SB01"	},
+	{ "TSSTcorp CDDVDW SH-S202H"	, "SB01"	},
+	{ "SAMSUNG SP0822N"	, "WA100-10"	},
 	{ NULL				, NULL		}
 };
 
