--- split.c	2001-08-16 14:22:20.000000000 +0300
+++ split.c	2007-11-01 01:13:38.000000000 +0200
@@ -1,3 +1,5 @@
+#include <stdlib.h>
+#include <string.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
--- func.c	2001-08-16 14:31:15.000000000 +0300
+++ func.c	2007-11-01 01:15:06.000000000 +0200
@@ -1,3 +1,4 @@
+#include <string.h>
 #include <stdlib.h>
 #include <unistd.h>
 #include <stdio.h>
