diff -Naur komparator4-0.1.orig/src/komparator4.desktop komparator4-0.1/src/komparator4.desktop
--- komparator4-0.1.orig/src/komparator4.desktop    2009-12-14 01:09:17.575611176 +0200
+++ komparator4-0.1/src/komparator4.desktop 2009-12-14 01:10:59.791610952 +0200
@@ -1,6 +1,8 @@
 [Desktop Entry]
 Name=Komparator4
 Name[x-test]=xxKomparator4xx
+GenericName=Directory Synchronization Application
+GenericName[tr]=Dizin Eşleme Uygulaması
 Exec=komparator4 %U
 Icon=komparator4
 Type=Application
