Index: kosd-0.4.2/kosd.cpp
===================================================================
--- kosd-0.4.2.orig/kosd.cpp
+++ kosd-0.4.2/kosd.cpp
@@ -50,43 +50,43 @@ KOSD::KOSD()
 	// Add volume up
 	KAction *newAct = m_actions->addAction("raiseVolume", this);
 	newAct->setText("Raise Volume");
-	newAct->setGlobalShortcut(KShortcut("Volume Up"), KAction::DefaultShortcut | KAction::ActiveShortcut);
+	//newAct->setGlobalShortcut(KShortcut("Volume Up"), KAction::DefaultShortcut | KAction::ActiveShortcut);
 	connect(newAct, SIGNAL(triggered()), this, SLOT(raiseVolume()));
 
 	// Add volume down
 	newAct = m_actions->addAction("lowerVolume", this);
 	newAct->setText("Lower Volume");
-	newAct->setGlobalShortcut(KShortcut("Volume Down"), KAction::DefaultShortcut | KAction::ActiveShortcut);
+	//newAct->setGlobalShortcut(KShortcut("Volume Down"), KAction::DefaultShortcut | KAction::ActiveShortcut);
 	connect(newAct, SIGNAL(triggered()), this, SLOT(lowerVolume()));
 
 	// Add volume mute
 	newAct = m_actions->addAction("muteVolume", this);
 	newAct->setText("Mute Volume");
-	newAct->setGlobalShortcut(KShortcut("Volume Mute"), KAction::DefaultShortcut | KAction::ActiveShortcut);
+	//newAct->setGlobalShortcut(KShortcut("Volume Mute"), KAction::DefaultShortcut | KAction::ActiveShortcut);
 	connect(newAct, SIGNAL(triggered()), this, SLOT(muteVolume()));
 
 	// Add show brightness up
 	newAct = m_actions->addAction("showBrightnessUp", this);
 	newAct->setText("Show Brightness Up");
-	newAct->setGlobalShortcut(KShortcut(), KAction::DefaultShortcut | KAction::ActiveShortcut);
+	//newAct->setGlobalShortcut(KShortcut(), KAction::DefaultShortcut | KAction::ActiveShortcut);
 	connect(newAct, SIGNAL(triggered()), this, SLOT(showBrightnessUp()));
 
 	// Add show brightness down
 	newAct = m_actions->addAction("showBrightnessDown", this);
 	newAct->setText("Show Brightness Down");
-	newAct->setGlobalShortcut(KShortcut(), KAction::DefaultShortcut | KAction::ActiveShortcut);
+	//newAct->setGlobalShortcut(KShortcut(), KAction::DefaultShortcut | KAction::ActiveShortcut);
 	connect(newAct, SIGNAL(triggered()), this, SLOT(showBrightnessDown()));
 
 	// Add show brightness
 	newAct = m_actions->addAction("showBrightness", this);
 	newAct->setText("Show Brightness");
-	newAct->setGlobalShortcut(KShortcut("Brightness"), KAction::DefaultShortcut | KAction::ActiveShortcut);
+	//newAct->setGlobalShortcut(KShortcut("Brightness"), KAction::DefaultShortcut | KAction::ActiveShortcut);
 	connect(newAct, SIGNAL(triggered()), this, SLOT(showBrightness()));
 
 	// Add show battery charge
 	newAct = m_actions->addAction("showBatteryCharge", this);
 	newAct->setText("Show Battery Charge");
-	newAct->setGlobalShortcut(KShortcut("Battery"), KAction::DefaultShortcut | KAction::ActiveShortcut);
+	//newAct->setGlobalShortcut(KShortcut("Battery"), KAction::DefaultShortcut | KAction::ActiveShortcut);
 	connect(newAct, SIGNAL(triggered()), this, SLOT(showBattery()));
 
 	m_configDialog = 0;
