Index: kosd-0.4.2/osd.cpp
===================================================================
--- kosd-0.4.2.orig/osd.cpp
+++ kosd-0.4.2/osd.cpp
@@ -39,7 +39,7 @@ using namespace std;
 #define OSD_BASE_HEIGHT 400
 
 #define ICON_BASE_SIZE 200
-#define ICON_BASE_X 100
+#define ICON_BASE_X ((OSD_BASE_WIDTH/2.0)-(ICON_BASE_SIZE/2.0))
 #define ICON_BASE_Y 50
 
 #define LABEL_BASE_WIDTH 470
@@ -186,7 +186,7 @@ void OSD::display(QString icon, QString 
 	this->updatePosition();
 	
 	m_icon = KIconLoader::global()->loadIcon(icon, KIconLoader::NoGroup, m_iconRect.width());
-	m_labelText = text;
+	m_labelText = "";/*text;*/
 
 	if (percent > 100)
 		percent = 100;
