diff -Nur icecream.orig/client/local.cpp icecream/client/local.cpp
--- icecream.orig/client/local.cpp	2008-11-10 16:28:58.000000000 +0200
+++ icecream/client/local.cpp	2008-11-10 16:29:44.000000000 +0200
@@ -30,6 +30,7 @@
 #ifdef HAVE_SIGNAL_H
 #include <signal.h>
 #endif
+#include <limits.h>
 
 #include <comm.h>
 #include "client.h"
diff -Nur icecream.orig/client/main.cpp icecream/client/main.cpp
--- icecream.orig/client/main.cpp	2008-11-10 16:28:58.000000000 +0200
+++ icecream/client/main.cpp	2008-11-10 16:29:35.000000000 +0200
@@ -56,6 +56,7 @@
 #  include <sys/stat.h>
 #endif
 #include <sys/wait.h>
+#include <limits.h>
 
 #include "client.h"
 
diff -Nur icecream.orig/client/remote.cpp icecream/client/remote.cpp
--- icecream.orig/client/remote.cpp	2008-11-10 16:28:58.000000000 +0200
+++ icecream/client/remote.cpp	2008-11-10 16:29:53.000000000 +0200
@@ -45,6 +45,7 @@
 #ifdef HAVE_RSYNC
 #include <librsync.h>
 #endif
+#include <limits.h>
 
 #include <comm.h>
 #include "client.h"
