--- util/Makefile.am	2008-09-17 09:07:05.000000000 +0200
+++ util/Makefile.am.oden	2008-09-25 11:02:28.000000000 +0200
@@ -115,19 +115,19 @@
 makefdb_CFLAGS = -DMAKE_FDB
 
 gif2mask_SOURCES = gif2mask.c
-gif2mask_LDADD = $(top_builddir)/src/libming.la
+gif2mask_LDADD = $(top_builddir)/src/libming.la $(MATHLIB) $(GIFLIB) $(ZLIB)
 
 png2dbl_SOURCES = png2dbl.c
-png2dbl_LDADD = $(top_builddir)/src/libming.la
+png2dbl_LDADD = $(top_builddir)/src/libming.la $(MATHLIB) $(PNGLIB)
 
 img2swf_SOURCES = img2swf.c
 img2swf_LDADD = $(top_builddir)/src/libming.la
 
 dbl2png_SOURCES = dbl2png.c
-dbl2png_LDADD = $(top_builddir)/src/libming.la
+dbl2png_LDADD = $(top_builddir)/src/libming.la $(MATHLIB) $(PNGLIB)
 
 gif2dbl_SOURCES = gif2dbl.c
-gif2dbl_LDADD = $(top_builddir)/src/libming.la
+gif2dbl_LDADD = $(top_builddir)/src/libming.la $(MATHLIB) $(GIFLIB) $(ZLIB)
 
 makeswf_SOURCES = makeswf.c makeswf_utils.c
 makeswf_LDADD = $(top_builddir)/src/libming.la
