
Index: ming/php_ext/Makefile.am
===================================================================
--- ming.orig/php_ext/Makefile.am	2008-12-22 14:01:29.000000000 -0500
+++ ming/php_ext/Makefile.am	2008-12-22 14:02:42.000000000 -0500
@@ -8,5 +8,5 @@
 	rm -rf tmp
 
 install: all
-	$(INSTALL) -d "$(DESTDIR)"/`php-config --extension-dir`
-	$(INSTALL) -m 755 tmp/modules/ming.so" "$(DESTDIR)"/`php-config --extension-dir`
+	$(INSTALL) -d `dirname $(prefix)`/`php-config --extension-dir`
+	$(INSTALL) -m 755 tmp/modules/ming.so `dirname $(prefix)`/`php-config --extension-dir`

