--- amarok/src/amarok_proxy.rb	2006-10-26 23:40:20.000000000 +0300
+++ amarok/src/amarok_proxy.rb	2006-11-30 15:36:59.335788092 +0200
@@ -142,11 +142,6 @@
 
   def cp_all_inward( income, output )
     myputs( "cp_all( income => #{income.inspect}, output => #{output.inspect}" )
-    if self.is_a?( LastFM ) and @engine == 'xine-engine'
-      myputs( "Using buffer fill workaround." )
-      filler = Array.new( 4096, 0 )
-      safe_write( output, filler ) # HACK: Fill xine's buffer so that xine_open() won't block
-    end
     data = income.read( 4 )
     desync( data )
     holdover = ""
