diff -Naur imagination-2.1.orig/src/support.c imagination-2.1/src/support.c
--- imagination-2.1.orig/src/support.c	2010-04-17 00:53:35.768184277 +0300
+++ imagination-2.1/src/support.c	2010-04-17 00:56:58.881309684 +0300
@@ -20,7 +20,7 @@
 #include "support.h"
 #include <glib/gstdio.h>
 
-#define PLUGINS_INSTALLED 0
+#define PLUGINS_INSTALLED 1
 
 static gboolean img_plugin_is_loaded(img_window_struct *, GModule *);
 
