--- dlls/shell32/shellpath.c.orig	2006-02-02 16:56:02.000000000 +0200
+++ dlls/shell32/shellpath.c	2006-02-05 19:43:46.000000000 +0200
@@ -2091,7 +2091,7 @@
     }
 
     /* Last but not least, the Desktop folder */
-    strcpy(szDesktopTarget, pszHome);
+    /*strcpy(szDesktopTarget, pszHome);
     if (_SHAppendToUnixPath(szDesktopTarget, DesktopW) &&
         !stat(szDesktopTarget, &statFolder) && S_ISDIR(statFolder.st_mode))
     {
@@ -2103,7 +2103,7 @@
             symlink(szDesktopTarget, pszDesktop);
             HeapFree(GetProcessHeap(), 0, pszDesktop);
         }
-    }
+    }*/
 }
 
 /* Register the default values in the registry, as some apps seem to depend
