Index: xterm-243/xterm.man
===================================================================
--- xterm-243.orig/xterm.man
+++ xterm-243/xterm.man
@@ -1474,7 +1474,7 @@ The default is ``false.''
 .TP
 .B "allowFontOps (\fPclass\fB AllowFontOps)"
 Specifies whether control sequences that set/query the font should be allowed.
-The default is ``true.''
+The default is ``false.''
 .TP 8
 .B "allowSendEvents (\fPclass\fB AllowSendEvents)"
 Specifies whether or not synthetic key and button events (generated using
@@ -1516,7 +1516,7 @@ The default is ``true.''
 .B "allowWindowOps (\fPclass\fB AllowWindowOps)"
 Specifies whether extended window control sequences (as used in dtterm)
 should be allowed.
-The default is ``true.''
+The default is ``false.''
 .TP 8
 .B "altIsNotMeta (\fPclass\fB AltIsNotMeta\fP)"
 If ``true'', treat the Alt-key as if it were the Meta-key.
Index: xterm-243/main.h
===================================================================
--- xterm-243.orig/main.h
+++ xterm-243/main.h
@@ -92,7 +92,7 @@
 #endif
 
 #ifndef DEF_ALLOW_FONT
-#define DEF_ALLOW_FONT		True
+#define DEF_ALLOW_FONT		False
 #endif
 
 #ifndef DEF_ALLOW_TCAP
