--- src/qt_qt_wrapper.cpp	2007-08-28 23:30:38.000000000 +0300
+++ src/qt_qt_wrapper.cpp	2008-01-10 18:01:08.000000000 +0200
@@ -334,7 +334,7 @@
 		printf("createQApp()\n");
 	
 	char* sessionEnv = getenv("SESSION_MANAGER");
-	if (QString(sessionEnv).endsWith(QString::number(getpid())) || cmdLine.contains("gnome-wm") || cmdLine.contains("metacity") || cmdLine.contains("xfwm4") || (getenv("GTK_QT_ENGINE_DISABLE") != NULL) ||
+	if (QString(sessionEnv).endsWith(QString::number(getpid())) || cmdLine.contains("nspluginviewer") || cmdLine.contains("metacity") || cmdLine.contains("xfwm4") || (getenv("GTK_QT_ENGINE_DISABLE") != NULL) ||
 	((qApp) && (qApp->type() == QApplication::Tty)))
 	{
 		printf("Not initializing the Gtk-Qt theme engine\n");
