diff -Nur hplip-2.7.10-old/Makefile.in hplip-2.7.10/Makefile.in
--- hplip-2.7.10-old/Makefile.in	2007-12-14 11:20:12.495559493 +0200
+++ hplip-2.7.10/Makefile.in	2007-12-14 11:20:53.473329926 +0200
@@ -1646,7 +1646,7 @@
 	$(dist_www251_DATA) $(dist_www252_DATA) $(dist_www261_DATA) \
 	$(dist_www3_DATA) $(dist_www4_DATA) $(dist_www5_DATA) \
 	$(dist_www6_DATA) $(dist_www61_DATA) $(dist_www7_DATA) \
-	$(doc_DATA) $(hplip_conf_DATA) $(hplip_desktop_DATA)
+	 $(hplip_conf_DATA) $(hplip_desktop_DATA)
 ETAGS = etags
 CTAGS = ctags
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
@@ -6808,10 +6808,10 @@
 	  echo " rm -f '$(DESTDIR)$(www7dir)/$$f'"; \
 	  rm -f "$(DESTDIR)$(www7dir)/$$f"; \
 	done
-install-docDATA: $(doc_DATA)
+install-docDATA: 
 	@$(NORMAL_INSTALL)
 	test -z "$(docdir)" || $(mkdir_p) "$(DESTDIR)$(docdir)"
-	@list='$(doc_DATA)'; for p in $$list; do \
+	@list=''; for p in $$list; do \
 	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
 	  f=$(am__strip_dir) \
 	  echo " $(docDATA_INSTALL) '$$d$$p' '$(DESTDIR)$(docdir)/$$f'"; \
@@ -6820,7 +6820,7 @@
 
 uninstall-docDATA:
 	@$(NORMAL_UNINSTALL)
-	@list='$(doc_DATA)'; for p in $$list; do \
+	@list=''; for p in $$list; do \
 	  f=$(am__strip_dir) \
 	  echo " rm -f '$(DESTDIR)$(docdir)/$$f'"; \
 	  rm -f "$(DESTDIR)$(docdir)/$$f"; \
