diff -Nur hplip-2.7.10-old/installer/core_install.py hplip-2.7.10/installer/core_install.py
--- hplip-2.7.10-old/installer/core_install.py	2007-12-14 11:27:53.555467734 +0200
+++ hplip-2.7.10/installer/core_install.py	2007-12-14 11:29:15.183058455 +0200
@@ -402,7 +402,7 @@
 
         if not found:
             try:
-                name = file('/etc/issue', 'r').read().lower().strip()
+                name = 'Pardus'.lower().strip()
             except IOError:
                 # Some O/Ss don't have /etc/issue (Mac)
                 self.distro, self.distro_version = DISTRO_UNKNOWN, '0.0'
