--- autoopts/mk-autoopts-pc.in	2007-10-07 19:58:00.000000000 +0300
+++ autoopts/mk-autoopts-pc.in	2007-12-02 12:14:27.000000000 +0200
@@ -26,8 +26,6 @@
      autogen="${bindir}/autogen${exeext}"
      ldflags="-L${libdir} -lopts"
         libs="${ldflags}"
-      libsrc="${pkgdatadir}/libopts-${dotver}.tar.gz"
- static_libs="${libdir}/libopts.a"
       cflags="-I${includedir}"
 test -n "${ldopts}" && ldflags="${ldopts}${libdir} ${ldflags}"
 test "${includedir}" = "/usr/include" && cflags=""
@@ -52,8 +50,6 @@
 	pkgdatadir="${pkgdatadir}"
 	autogen="${autogen}"
 	libs="${libs}"
-	libsrc="${libsrc}"
-	static_libs="${static_libs}"
 
 	Name:           AutoOpts
 	Description:    A semi-automated generated/library option parser
