Index: sqliteman-1.2.2/CMakeLists.txt
===================================================================
--- sqliteman-1.2.2.orig/CMakeLists.txt
+++ sqliteman-1.2.2/CMakeLists.txt
@@ -154,6 +154,6 @@ ELSE (WIN32)
     )
     INSTALL(FILES
         sqliteman/icons/sqliteman.png
-        DESTINATION "${CMAKE_INSTALL_PREFIX}/share/icons/hicolor"
+        DESTINATION "${CMAKE_INSTALL_PREFIX}/share/pixmaps"
     )
 ENDIF (WIN32)
