--- src/unix/unix.mak.orig	2008-09-22 11:25:36.000000000 +0300
+++ src/unix/unix.mak	2008-09-22 11:28:48.000000000 +0300
@@ -457,13 +457,13 @@
 endif
 
 ifdef SOUND_ARTS_TEIRA
-CONFIG  += -DSYSDEP_DSP_ARTS_TEIRA `artsc-config --cflags`
-LIBS += `artsc-config --libs`
+CONFIG  += -DSYSDEP_DSP_ARTS_TEIRA `$(ARTSCCONFIG) --cflags`
+LIBS += `$(ARTSCCONFIG) --libs`
 endif
 
 ifdef SOUND_ARTS_SMOTEK
-CONFIG  += -DSYSDEP_DSP_ARTS_SMOTEK `artsc-config --cflags`
-LIBS += `artsc-config --libs`
+CONFIG  += -DSYSDEP_DSP_ARTS_SMOTEK `$(ARTSCCONFIG) --cflags`
+LIBS += `$(ARTSCCONFIG) --libs`
 endif
 
 ifdef SOUND_SDL
--- makefile.unix.orig	2008-09-22 11:27:16.000000000 +0300
+++ makefile.unix	2008-09-22 11:27:52.000000000 +0300
@@ -804,6 +804,8 @@
 # Enable Xinerama support
 # X11_XINERAMA = 1
 
+# ARTSCCONFIG
+ARTSCCONFIG = /usr/kde/3.5/bin/artsc-config
 
 ##############################################################################
 # SDL options (only needed when SDL is the display method)
