--- share/LaTeX/Makefile.in	2007-08-31 09:35:05.000000000 +0300
+++ share/LaTeX/Makefile.in	2007-12-18 18:15:13.000000000 +0200
@@ -331,11 +331,11 @@
 
 install-cfg:
 	@if test $(PLAINTEX) != no ; then \
-	  installdir=`kpsexpand '$$TEXMFLOCAL'/tex/latex/gnuplot |sed "s%^.*share/%$(prefix)/share/%"` ; \
+	  installdir=`echo /usr/share/texmf-dist/tex/latex/gnuplot` ; \
 	  $(top_srcdir)/mkinstalldirs $(DESTDIR)$$installdir ; \
 	  if [ ! -f $(DESTDIR)$$installdir/gnuplot.cfg ]; then \
 	    $(INSTALL_DATA) $(srcdir)/gnuplot.cfg $(DESTDIR)$$installdir/gnuplot.cfg && \
-	    texhash; \
+	    /bin/true; \
 	  fi ; \
 	  fi
 
