Index: libkdepim/recentaddresses.cpp
===================================================================
--- libkdepim/recentaddresses.cpp	(revision 671908)
+++ libkdepim/recentaddresses.cpp	(revision 671909)
@@ -28,6 +28,7 @@
  *  your version.
  */
 #include "recentaddresses.h"
+#include "libemailfunctions/email.h"
 
 #include <kstaticdeleter.h>
 #include <kconfig.h>
@@ -98,24 +99,33 @@
 
 void RecentAddresses::add( const QString& entry )
 {
-    if ( !entry.isEmpty() && m_maxCount > 0 ) {
-        QString email;
-        QString fullName;
-        KABC::Addressee addr;
+  if ( !entry.isEmpty() && m_maxCount > 0 ) {
+    QStringList list = KPIM::splitEmailAddrList( entry );
+    for( QStringList::const_iterator e_it = list.begin(); e_it != list.end(); ++e_it ) {
+      KPIM::EmailParseResult errorCode = KPIM::isValidEmailAddress( *e_it );
+      if ( errorCode != KPIM::AddressOk ) 
+        continue;
+      QString email;
+      QString fullName;
+      KABC::Addressee addr;
 
-        KABC::Addressee::parseEmailAddress( entry, fullName, email );
+      KABC::Addressee::parseEmailAddress( *e_it, fullName, email );
 
-        for ( KABC::Addressee::List::Iterator it = m_addresseeList.begin();
-              it != m_addresseeList.end(); ++it )
-        {
-            if ( email == (*it).preferredEmail() )
-                return;//already inside
+      for ( KABC::Addressee::List::Iterator it = m_addresseeList.begin();
+          it != m_addresseeList.end(); ++it )
+      {
+        if ( email == (*it).preferredEmail() ) {
+          //already inside, remove it here and add it later at pos==1
+          m_addresseeList.remove( it );
+          break;
         }
-        addr.setNameFromString( fullName );
-        addr.insertEmail( email, true );
-        m_addresseeList.prepend( addr );
-        adjustSize();
+      }
+      addr.setNameFromString( fullName );
+      addr.insertEmail( email, true );
+      m_addresseeList.prepend( addr );
+      adjustSize();
     }
+  }
 }
 
 void RecentAddresses::setMaxCount( int count )
