diff -p -up ./hdlcutil/fl/Makefile.in.orig ./hdlcutil/fl/Makefile.in
--- ./hdlcutil/fl/Makefile.in.orig	2010-10-22 13:00:22.000000000 +0200
+++ ./hdlcutil/fl/Makefile.in	2010-10-22 13:00:56.000000000 +0200
@@ -64,12 +64,12 @@ am__depfiles_maybe = depfiles
 CXXCOMPILE = $(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
 	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS)
 CXXLD = $(CXX)
-CXXLINK = $(CXXLD) $(AM_CXXFLAGS) $(CXXFLAGS) $(AM_LDFLAGS) $(LDFLAGS) \
+CXXLINK = $(CXXLD) $(AM_CXXFLAGS) $(CXXFLAGS) $(LDFLAGS) \
 	-o $@
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
 	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 CCLD = $(CC)
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(LDFLAGS) -o $@
 SOURCES = $(xfhdlcchpar_SOURCES) $(xfhdlcst_SOURCES) \
 	$(xfsmdiag_SOURCES) $(xfsmmixer_SOURCES)
 DIST_SOURCES = $(xfhdlcchpar_SOURCES) $(xfhdlcst_SOURCES) \
@@ -242,16 +242,16 @@ clean-sbinPROGRAMS:
 	-test -z "$(sbin_PROGRAMS)" || rm -f $(sbin_PROGRAMS)
 xfhdlcchpar$(EXEEXT): $(xfhdlcchpar_OBJECTS) $(xfhdlcchpar_DEPENDENCIES) 
 	@rm -f xfhdlcchpar$(EXEEXT)
-	$(CXXLINK) $(xfhdlcchpar_OBJECTS) $(xfhdlcchpar_LDADD) $(LIBS)
+	$(CXXLINK) $(xfhdlcchpar_OBJECTS) $(xfhdlcchpar_LDADD) $(LIBS) $(AM_LDFLAGS)
 xfhdlcst$(EXEEXT): $(xfhdlcst_OBJECTS) $(xfhdlcst_DEPENDENCIES) 
 	@rm -f xfhdlcst$(EXEEXT)
-	$(CXXLINK) $(xfhdlcst_OBJECTS) $(xfhdlcst_LDADD) $(LIBS)
+	$(CXXLINK) $(xfhdlcst_OBJECTS) $(xfhdlcst_LDADD) $(LIBS) $(AM_LDFLAGS)
 xfsmdiag$(EXEEXT): $(xfsmdiag_OBJECTS) $(xfsmdiag_DEPENDENCIES) 
 	@rm -f xfsmdiag$(EXEEXT)
-	$(CXXLINK) $(xfsmdiag_OBJECTS) $(xfsmdiag_LDADD) $(LIBS)
+	$(CXXLINK) $(xfsmdiag_OBJECTS) $(xfsmdiag_LDADD) $(LIBS) $(AM_LDFLAGS)
 xfsmmixer$(EXEEXT): $(xfsmmixer_OBJECTS) $(xfsmmixer_DEPENDENCIES) 
 	@rm -f xfsmmixer$(EXEEXT)
-	$(CXXLINK) $(xfsmmixer_OBJECTS) $(xfsmmixer_LDADD) $(LIBS)
+	$(CXXLINK) $(xfsmmixer_OBJECTS) $(xfsmmixer_LDADD) $(LIBS) $(AM_LDFLAGS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
