diff -Nur hplip-2.8.4-old/Makefile.in hplip-2.8.4/Makefile.in
--- hplip-2.8.4-old/Makefile.in	2008-04-08 12:24:41.000000000 +0300
+++ hplip-2.8.4/Makefile.in	2008-04-08 12:24:59.000000000 +0300
@@ -1284,7 +1284,7 @@
 	$(dist_www251_DATA) $(dist_www252_DATA) $(dist_www261_DATA) \
 	$(dist_www3_DATA) $(dist_www4_DATA) $(dist_www5_DATA) \
 	$(dist_www6_DATA) $(dist_www61_DATA) $(dist_www7_DATA) \
-	$(doc_DATA) $(hplip_conf_DATA) $(hplip_desktop_DATA) \
+	 $(hplip_conf_DATA) $(hplip_desktop_DATA) \
 	$(hplip_systray_DATA)
 ETAGS = etags
 CTAGS = ctags
@@ -6072,10 +6072,10 @@
 	  echo " rm -f '$(DESTDIR)$(www7dir)/$$f'"; \
 	  rm -f "$(DESTDIR)$(www7dir)/$$f"; \
 	done
-install-docDATA: $(doc_DATA)
+install-docDATA: 
 	@$(NORMAL_INSTALL)
 	test -z "$(docdir)" || $(MKDIR_P) "$(DESTDIR)$(docdir)"
-	@list='$(doc_DATA)'; for p in $$list; do \
+	@list=''; for p in $$list; do \
 	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
 	  f=$(am__strip_dir) \
 	  echo " $(docDATA_INSTALL) '$$d$$p' '$(DESTDIR)$(docdir)/$$f'"; \
@@ -6084,7 +6084,7 @@
 
 uninstall-docDATA:
 	@$(NORMAL_UNINSTALL)
-	@list='$(doc_DATA)'; for p in $$list; do \
+	@list=''; for p in $$list; do \
 	  f=$(am__strip_dir) \
 	  echo " rm -f '$(DESTDIR)$(docdir)/$$f'"; \
 	  rm -f "$(DESTDIR)$(docdir)/$$f"; \
