--- trunk/plug-ins/common/xbm.c	2007/07/04 17:48:25	22864
+++ trunk/plug-ins/common/xbm.c	2007/07/04 23:32:15	22865
@@ -805,20 +805,34 @@
       return -1;
     }
 
-  if (width == 0)
+  if (width <= 0)
     {
       g_message (_("'%s':\nNo image width specified"),
                  gimp_filename_to_utf8 (filename));
       return -1;
     }
 
-  if (height == 0)
+  if (width > GIMP_MAX_IMAGE_SIZE)
+    {
+      g_message (_("'%s':\nImage width is larger than GIMP can handle"),
+                 gimp_filename_to_utf8 (filename));
+      return -1;
+    }
+
+  if (height <= 0)
     {
       g_message (_("'%s':\nNo image height specified"),
                  gimp_filename_to_utf8 (filename));
       return -1;
     }
 
+  if (height > GIMP_MAX_IMAGE_SIZE)
+    {
+      g_message (_("'%s':\nImage height is larger than GIMP can handle"),
+                 gimp_filename_to_utf8 (filename));
+      return -1;
+    }
+
   if (intbits == 0)
     {
       g_message (_("'%s':\nNo image data type specified"),
@@ -1063,7 +1077,7 @@
 
 #ifdef VERBOSE
       if (verbose > 1)
-	printf ("TGA: writing %dx(%d+%d) pixel region\n",
+	printf ("XBM: writing %dx(%d+%d) pixel region\n",
 		width, i, tileheight);
 #endif
 
