diff -Nur yali4-2.0.6.orig/yali4/constants.py yali4-2.0.6/yali4/constants.py
--- yali4-2.0.6.orig/yali4/constants.py	2008-12-24 20:08:26.000000000 +0200
+++ yali4-2.0.6/yali4/constants.py	2008-12-30 13:38:21.000000000 +0200
@@ -87,8 +87,8 @@
 consts.cd_repo_uri = join(consts.source_dir, "repo/pisi-index.xml.bz2")
 
 # pardus repository
-consts.pardus_repo_name = "pardus-2008.1"
-consts.pardus_repo_uri = "http://paketler.pardus.org.tr/pardus-2008.1/pisi-index.xml.bz2"
+consts.pardus_repo_name = "pardus-2008.2"
+consts.pardus_repo_uri = "http://paketler.pardus.org.tr/pardus-2008.2/pisi-index.xml.bz2"
 
 # min root partition size
 consts.min_root_size = 3500
