diff -Naur src.orig/modules/avformat/consumer_avformat.c src/modules/avformat/consumer_avformat.c
--- src.orig/modules/avformat/consumer_avformat.c	2008-04-18 14:24:48.000000000 +0300
+++ src/modules/avformat/consumer_avformat.c	2008-04-18 14:23:27.000000000 +0300
@@ -33,11 +33,11 @@
 #include <math.h>
 
 // avformat header files
-#include <avformat.h>
+#include <libavformat/avformat.h>
 #ifdef SWSCALE
-#include <swscale.h>
+#include <libswscale/swscale.h>
 #endif
-#include <opt.h>
+#include <libavcodec/opt.h>
 
 //
 // This structure should be extended and made globally available in mlt
diff -Naur src.orig/modules/avformat/factory.c src/modules/avformat/factory.c
--- src.orig/modules/avformat/factory.c	2008-04-18 14:24:48.000000000 +0300
+++ src/modules/avformat/factory.c	2008-04-18 14:23:27.000000000 +0300
@@ -31,7 +31,7 @@
 extern mlt_producer producer_avformat_init( mlt_profile profile, char *file );
 
 // ffmpeg Header files
-#include <avformat.h>
+#include <libavformat/avformat.h>
 
 // A static flag used to determine if avformat has been initialised
 static int avformat_initialised = 0;
diff -Naur src.orig/modules/avformat/filter_avcolour_space.c src/modules/avformat/filter_avcolour_space.c
--- src.orig/modules/avformat/filter_avcolour_space.c	2008-04-18 14:24:48.000000000 +0300
+++ src/modules/avformat/filter_avcolour_space.c	2008-04-18 14:23:27.000000000 +0300
@@ -22,9 +22,9 @@
 #include <framework/mlt_frame.h>
 
 // ffmpeg Header files
-#include <avformat.h>
+#include <libavformat/avformat.h>
 #ifdef SWSCALE
-#include <swscale.h>
+#include <libswscale/swscale.h>
 #endif
 
 #include <stdio.h>
diff -Naur src.orig/modules/avformat/filter_avresample.c src/modules/avformat/filter_avresample.c
--- src.orig/modules/avformat/filter_avresample.c	2008-04-18 14:24:48.000000000 +0300
+++ src/modules/avformat/filter_avresample.c	2008-04-18 14:23:27.000000000 +0300
@@ -26,7 +26,7 @@
 #include <string.h>
 
 // ffmpeg Header files
-#include <avformat.h>
+#include <libavformat/avformat.h>
 
 /** Get the audio.
 */
diff -Naur src.orig/modules/avformat/producer_avformat.c src/modules/avformat/producer_avformat.c
--- src.orig/modules/avformat/producer_avformat.c	2008-04-18 14:24:48.000000000 +0300
+++ src/modules/avformat/producer_avformat.c	2008-04-18 14:23:27.000000000 +0300
@@ -24,9 +24,9 @@
 #include <framework/mlt_frame.h>
 
 // ffmpeg Header files
-#include <avformat.h>
+#include <libavformat/avformat.h>
 #ifdef SWSCALE
-#include <swscale.h>
+#include <libswscale/swscale.h>
 #endif
 
 // System header files
