diff -Nur warsow_0.2_sdk-old/source/linux/in_x11.c warsow_0.2_sdk/source/linux/in_x11.c
--- warsow_0.2_sdk-old/source/linux/in_x11.c	2006-09-22 10:37:43.957996200 +0300
+++ warsow_0.2_sdk/source/linux/in_x11.c	2006-09-22 10:43:18.764097936 +0300
@@ -589,10 +589,8 @@
 					}
 					else
 					{
-						mx = -((int)event.xmotion.x - mwx);
-						my = -((int)event.xmotion.y - mwy);
-						mwx = event.xmotion.x;
-						mwy = event.xmotion.y;
+						mx += ((int)event.xmotion.x - mwx);
+						my += ((int)event.xmotion.y - mwy);
 
 						if (mx || my)
 							dowarp = qtrue;
