diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur l10n-orig/tr/toolkit/chrome/global/editMenuOverlay.dtd l10n/tr/toolkit/chrome/global/editMenuOverlay.dtd
--- ../l10n-orig/tr/toolkit/chrome/global/editMenuOverlay.dtd	2008-11-11 16:28:30.000000000 +0200
+++ ../l10n/tr/toolkit/chrome/global/editMenuOverlay.dtd	2008-11-11 16:28:56.000000000 +0200
@@ -14,7 +14,7 @@
 <!ENTITY findAgainCmd.label "Sonrakini bul">
 <!ENTITY findCmd.accesskey "B">
 <!ENTITY findCmd.key "F">
-<!ENTITY findCmd.label "Bul ve değiştir...">
+<!ENTITY findCmd.label "Bul">
 <!ENTITY findPreviousCmd.accesskey "n">
 <!ENTITY findPreviousCmd.label "Öncekini bul">
 <!ENTITY pasteCmd.accesskey "Y">
diff -Naur l10n-orig/tr/browser/chrome/browser/preferences/advanced.dtd l10n/tr/browser/chrome/browser/preferences/advanced.dtd
--- ../l10n-orig/tr/browser/chrome/browser/preferences/advanced.dtd	2008-11-12 10:10:41.000000000 +0200
+++ ../l10n/tr/browser/chrome/browser/preferences/advanced.dtd	2008-11-12 10:16:30.000000000 +0200
@@ -49,8 +49,8 @@
 <!ENTITY useCursorNavigation.label       "Sayfaların içinde hareket ederken her zaman işaretçi tuşlarını kullan">
 <!ENTITY useCursorNavigation.accesskey   "S">
 
-<!ENTITY searchStartTyping.label         "Ben yazmaya başlar başlamaz arama yap">
-<!ENTITY searchStartTyping.accesskey     "B">
+<!ENTITY searchStartTyping.label         "Yazmaya başlar başlamaz arama yap">
+<!ENTITY searchStartTyping.accesskey     "z">
 
 <!ENTITY useAutoScroll.label             "Otomatik kaydırmayı kullan">
 <!ENTITY useAutoScroll.accesskey         "a">
@@ -99,7 +99,7 @@
 <!ENTITY offlineAppsListRemove.label     "Kaldır...">
 <!ENTITY offlineAppsListRemove.accesskey "R">
 <!ENTITY certSelection.description       "Sunucu kişisel onay belgemi istediğinde:">
-<!ENTITY offlineNotify.label             "Bir site çevrim dışı kullanım için veri depolamak istediğinde bana söyle">
+<!ENTITY offlineNotify.label             "Bir site çevrim dışı kullanım için veri depolamak istediğinde beni uyar">
 <!ENTITY offlineNotify.accesskey         "T">
 <!ENTITY offlineNotifyExceptions.label   "Ayrıcalıklar...">
 <!ENTITY offlineNotifyExceptions.accesskey "a">
diff -Naur l10n-orig/tr/browser/chrome/browser/tabbrowser.dtd l10n/tr/browser/chrome/browser/tabbrowser.dtd
--- ../l10n-orig/tr/browser/chrome/browser/tabbrowser.dtd	2008-11-12 10:10:41.000000000 +0200
+++ ../l10n/tr/browser/chrome/browser/tabbrowser.dtd	2008-11-12 10:17:10.000000000 +0200
@@ -14,5 +14,5 @@
 <!ENTITY  bookmarkAllTabs.accesskey   "T">
 <!ENTITY  bookmarkThisTab.label        "Bu sekmeyi yer imlerine ekle...">
 <!ENTITY  bookmarkThisTab.accesskey    "B">
-<!ENTITY  undoCloseTab.label          "Kapaltılan sekmeyi aç">
+<!ENTITY  undoCloseTab.label          "Kapatılan sekmeyi aç">
 <!ENTITY  undoCloseTab.accesskey      "K">
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur l10n-orig/tr/toolkit/chrome/global/findbar.dtd l10n/tr/toolkit/chrome/global/findbar.dtd
--- ../l10n-orig/tr/toolkit/chrome/global/findbar.dtd	2008-11-11 16:28:30.000000000 +0200
+++ ../l10n/tr/toolkit/chrome/global/findbar.dtd	2008-11-11 16:28:56.000000000 +0200
@@ -13,5 +13,5 @@
 <!ENTITY highlight.label "Vurgula">
 <!ENTITY highlight.accesskey "a">
 <!ENTITY highlight.tooltiptext "Eşleşen kelimeleri vurgular (Ctrl+Enter)">
-<!ENTITY caseSensitiveCheckbox.label "Eşleşen kelimeler">
-<!ENTITY caseSensitiveCheckbox.accesskey "E">
+<!ENTITY caseSensitiveCheckbox.label "Büyük/küçük harf duyarlı">
+<!ENTITY caseSensitiveCheckbox.accesskey "h">

