--- acpica-unix-20071019/interpreter/executer/exstore.c
+++ acpica-unix-20071019/interpreter/executer/exstore.c
@@ -300,7 +300,8 @@ AcpiExDoDebugObject (
                     ACPI_DESC_TYPE_NAMED)
             {
                 ACPI_DEBUG_PRINT_RAW ((ACPI_DB_DEBUG_OBJECT,
-                    " %p - Not a valid namespace node\n"));
+                    " %p - Not a valid namespace node\n",
+				       SourceDesc->Reference.Node));
             }
             else
             {
