--- code/qcommon/files.c.orig	2007-04-17 23:16:55.000000000 +0300
+++ code/qcommon/files.c	2007-06-15 21:48:41.000000000 +0300
@@ -2808,7 +2808,7 @@
 
 	fs_debug = Cvar_Get( "fs_debug", "0", 0 );
 	fs_copyfiles = Cvar_Get( "fs_copyfiles", "0", CVAR_INIT );
-	fs_cdpath = Cvar_Get ("fs_cdpath", Sys_DefaultCDPath(), CVAR_INIT );
+	fs_cdpath = Cvar_Get ("fs_cdpath", "/usr/share/WoP", CVAR_INIT );
 	fs_basepath = Cvar_Get ("fs_basepath", Sys_DefaultInstallPath(), CVAR_INIT );
 	fs_basegame = Cvar_Get ("fs_basegame", "", CVAR_INIT );
   homePath = Sys_DefaultHomePath();
