--- plasma/dataengines/weather/ions/ion_bbcukmet.cpp	2009/11/24 19:56:50	1053778
+++ plasma/dataengines/weather/ions/ion_bbcukmet.cpp	2009/12/09 08:10:14	1060497
@@ -248,7 +248,7 @@
 void UKMETIon::findPlace(const QString& place, const QString& source)
 {
     KUrl url;
-    url = "http://news.bbc.co.uk/weather/forecast/?&search=" + place + "&itemsPerPage=200&startIndex=1&count=200";
+    url = "http://news.bbc.co.uk/weather/util/search/SearchResultsNode.xhtml?&search=" + place +"&region=world&count=500";
 
     m_job = KIO::get(url.url(), KIO::Reload, KIO::HideProgressInfo);
     m_job->addMetaData("cookies", "none"); // Disable displaying cookies
