diff -Naur kcontrol-orig/fonts/fonts.cpp kcontrol/fonts/fonts.cpp
--- kcontrol-orig/fonts/fonts.cpp	2009-06-18 17:36:38.817090778 +0300
+++ kcontrol/fonts/fonts.cpp	2009-06-18 17:39:29.413341072 +0300
@@ -513,11 +513,11 @@
 
   // NOTE: keep in sync with kdelibs/kdeui/kernel/kglobalsettings.cpp
 
-  QFont f0("Sans Serif", 10);
-  QFont f1("Monospace", 10);
+  QFont f0("Sans Serif", 9);
+  QFont f1("Monospace", 9);
   QFont f2("Sans Serif", 8);
   QFont f3("Sans Serif", 9, QFont::Bold);
-  QFont f4("Sans Serif", 10);
+  QFont f4("Sans Serif", 9);
   QFont f5("Sans Serif", 8); // smallestReadableFont
 
   defaultFontList << f0 << f1 << f5 << f2 << f0 << f3 << f4 << f0;
