diff -Naur kdebase-workspace-4.3.2-orig/kwin/effects/logout/logout.cpp kdebase-workspace-4.3.2/kwin/effects/logout/logout.cpp
--- kdebase-workspace-4.3.2-orig/kwin/effects/logout/logout.cpp	2009-10-13 16:11:40.513844054 +0300
+++ kdebase-workspace-4.3.2/kwin/effects/logout/logout.cpp	2009-10-13 16:12:00.972845069 +0300
@@ -65,7 +65,7 @@
 #ifdef KWIN_HAVE_OPENGL_COMPOSITING
     frameDelay = 0;
     KConfigGroup conf = effects->effectConfig( "Logout" );
-    useBlur = conf.readEntry( "UseBlur", true );
+    useBlur = conf.readEntry( "UseBlur", false );
     delete blurTexture;
     blurTexture = NULL;
     delete blurTarget;
