--- plasma/wallpapers/image/wallpaper.knsrc.orig	2009-07-16 19:52:35.851420129 +0300
+++ plasma/wallpapers/image/wallpaper.knsrc	2009-07-16 19:53:32.263667549 +0300
@@ -1,4 +1,4 @@
 [KNewStuff2]
-ProvidersUrl=http://download.kde.org/khotnewstuff/wallpaper-providers.xml
+ProvidersUrl=http://www.ozgurlukicin.com/tema/khotnewstuff/wallpaper-providers.xml
 StandardResource=wallpaper
 CachePolicy=resident
