diff -Naur synaptiks-0.4.0.orig/applet/plasma-applet-synaptiks.desktop synaptiks-0.4.0/applet/plasma-applet-synaptiks.desktop
--- synaptiks-0.4.0.orig/applet/plasma-applet-synaptiks.desktop	2010-05-06 01:26:40.978918546 +0300
+++ synaptiks-0.4.0/applet/plasma-applet-synaptiks.desktop	2010-05-06 01:29:24.365901574 +0300
@@ -1,5 +1,6 @@
 [Desktop Entry]
-Name=synaptiks
+Name=Synaptiks
+Name[tr]=Synaptiks
 Name[da]=synaptiks
 Name[en_GB]=synaptiks
 Name[es]=synaptiks
@@ -9,6 +10,7 @@
 Name[uk]=synaptiks
 Name[x-test]=xxsynaptiksxx
 Comment=Touchpad management applet
+Comment[tr]=Dokunmatik fare yönetimi programcığı
 Comment[da]=Applet til håndtering af touchpad
 Comment[en_GB]=Touchpad management applet
 Comment[es]=Miniaplicación de gestión del touchpad
diff -Naur synaptiks-0.4.0.orig/daemon/synaptiks.desktop synaptiks-0.4.0/daemon/synaptiks.desktop
--- synaptiks-0.4.0.orig/daemon/synaptiks.desktop	2010-05-06 01:26:40.975921866 +0300
+++ synaptiks-0.4.0/daemon/synaptiks.desktop	2010-05-06 01:30:29.849026878 +0300
@@ -8,7 +8,8 @@
 X-KDE-Kded-load-on-demand=false
 X-KDE-Kded-phase=1
 
-Name=synaptiks
+Name=Synaptiks
+Name[tr]=Synaptiks
 Name[da]=synaptiks
 Name[en_GB]=synaptiks
 Name[es]=synaptiks
@@ -19,6 +20,7 @@
 Name[x-test]=xxsynaptiksxx
 
 Comment=Touchpad control daemon
+Comment[tr]=Dokunmatik fare kontrol servisi
 Comment[da]=En kontroldæmon til touchpad
 Comment[en_GB]=Touchpad control dæmon
 Comment[es]=Demonio de control del touchpad
diff -Naur synaptiks-0.4.0.orig/kcmodule/synaptiks.desktop synaptiks-0.4.0/kcmodule/synaptiks.desktop
--- synaptiks-0.4.0.orig/kcmodule/synaptiks.desktop	2010-05-06 01:26:40.992026291 +0300
+++ synaptiks-0.4.0/kcmodule/synaptiks.desktop	2010-05-06 01:31:46.490026985 +0300
@@ -12,6 +12,7 @@
 X-KDE-Weight=65
 
 Name=Touchpad
+Name[tr]=Dokunmatik fare
 Name[da]=Touchpad
 Name[en_GB]=Touchpad
 Name[es]=Touchpad
@@ -22,6 +23,7 @@
 Name[x-test]=xxTouchpadxx
 
 Comment=Touchpad settings and management
+Comment[tr]=Dokunmatik fare yönetimi ve ayarları
 Comment[da]=Håndtering og indstilling af touchpad
 Comment[en_GB]=Touchpad settings and management
 Comment[es]=Gestión y configuración del touchpad
