diff -up ./kdesdk-4.7.0/umbrello/umbrello/codegenerators/ownedcodeblock.h.moc-fail ./kdesdk-4.7.0/umbrello/umbrello/codegenerators/ownedcodeblock.h
--- ./kdesdk-4.7.0/umbrello/umbrello/codegenerators/ownedcodeblock.h.moc-fail	2009-01-28 00:03:55.000000000 +0100
+++ ./kdesdk-4.7.0/umbrello/umbrello/codegenerators/ownedcodeblock.h	2011-08-02 18:00:16.015592150 +0200
@@ -23,7 +23,7 @@ class UMLObject;
  * Describes any codeblock which is 'owned' by a UMLobject of some sort and should
  * be in sync with that parent.
  */
-class OwnedCodeBlock : virtual public QObject
+class OwnedCodeBlock : public QObject
 {
     Q_OBJECT
 public:
