diff -up kdesdk-4.7.0/okteta/program/mainwindow.cpp.NO_STRUCTURES_TOOL kdesdk-4.7.0/okteta/program/mainwindow.cpp
--- kdesdk-4.7.0/okteta/program/mainwindow.cpp.NO_STRUCTURES_TOOL	2011-06-22 10:09:38.000000000 -0500
+++ kdesdk-4.7.0/okteta/program/mainwindow.cpp	2011-08-08 15:48:48.329320476 -0500
@@ -43,8 +43,10 @@
 #include <bytetable/bytetabletool.h>
 #include <bookmarks/bookmarkstoolview.h>
 #include <bookmarks/bookmarkstool.h>
+#ifndef NO_STRUCTURES_TOOL
 #include <structures/structtoolview.h>
 #include <structures/structtool.h>
+#endif
 // Kasten tools
 #include <document/versionview/versionviewtoolview.h>
 #include <document/versionview/versionviewtool.h>
