Index: libavcodec/alac.c
===================================================================
--- libavcodec/alac.c	(revision 15046)
+++ libavcodec/alac.c	(working copy)
@@ -597,7 +597,7 @@
     alac->context_initialized = 0;
 
     alac->numchannels = alac->avctx->channels;
-    alac->bytespersample = (avctx->bits_per_sample / 8) * alac->numchannels;
+    alac->bytespersample = 2 * alac->numchannels;
     avctx->sample_fmt = SAMPLE_FMT_S16;
 
     return 0;

