Index: libavcodec/imgconvert.c
===================================================================
--- libavcodec/imgconvert.c	(revision 13644)
+++ libavcodec/imgconvert.c	(working copy)
@@ -2106,9 +2106,7 @@
 {
     return av_picture_pad(dst, src, height, width, pix_fmt, padtop, padbottom, padleft, padright, color);
 }
-#endif
 
-#ifndef CONFIG_SWSCALE
 static uint8_t y_ccir_to_jpeg[256];
 static uint8_t y_jpeg_to_ccir[256];
 static uint8_t c_ccir_to_jpeg[256];
@@ -2628,7 +2626,7 @@
     avpicture_free(tmp);
     return ret;
 }
-#endif
+#endif /* LIBAVCODEC_VERSION_INT < ((52<<16)+(0<<8)+0) */
 
 /* NOTE: we scan all the pixels to have an exact information */
 static int get_alpha_info_pal8(const AVPicture *src, int width, int height)

