Index: libavformat/utils.c
===================================================================
--- libavformat/utils.c	(revision 15217)
+++ libavformat/utils.c	(working copy)
@@ -2444,6 +2444,10 @@
                 av_log(s, AV_LOG_ERROR, "sample rate not set\n");
                 return -1;
             }
+            if(!st->codec->block_align){
+                st->codec->block_align = st->codec->channels *
+                    av_get_bits_per_sample(st->codec->codec_id) >> 3;
+            }
             break;
         case CODEC_TYPE_VIDEO:
             if(st->codec->time_base.num<=0 || st->codec->time_base.den<=0){ //FIXME audio too?

