diff -Naur po/Makefile.in.in gyachi-mod/po/Makefile.in.in
--- gyachi/po/Makefile.in.in	2006-06-05 09:16:40.000000000 -0500
+++ gyachi-mod/po/Makefile.in.in	2007-09-22 19:36:52.000000000 -0500
@@ -28,8 +28,12 @@
 
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
-MKINSTALLDIRS = @MKINSTALLDIRS@
-mkinstalldirs = $(SHELL) $(MKINSTALLDIRS)
+#MKINSTALLDIRS = @MKINSTALLDIRS@
+#mkinstalldirs = $(SHELL) $(MKINSTALLDIRS)
+MKINSTALLDIRS = mkdir -p
+mkinstalldirs = $(MKINSTALLDIRS)
+ 
+
 
 GMSGFMT = @GMSGFMT@
 MSGFMT = @MSGFMT@
 
