--- plugins/Makefile.am	2007-11-11 00:19:20.000000000 +0200
+++ plugins/Makefile.am.new	2007-11-11 00:20:40.000000000 +0200
@@ -3,7 +3,7 @@
     gyachimcrypt.so \
     gyachiblowfish.so \
     gyachigpgme.so \
-    gyachixmms.so
+#    gyachixmms.so
 
 
 plugindir = $(pkglibdir)/plugins
@@ -24,12 +24,12 @@
 gyachigpgme_so_LDADD = @GTK_LIBS@ -lgpgme
 gyachigpgme_so_LDFLAGS = -shared -fpic -fPIC
 
-gyachixmms_so_SOURCES = gyachi-xmms.c
-gyachixmms_so_LDADD = @GTK_LIBS@ -lxmms
-gyachixmms_so_LDFLAGS = -shared -fpic -fPIC -Xlinker -Bdynamic -Wl,--export-dynamic
+#gyachixmms_so_SOURCES = gyachi-xmms.c
+#gyachixmms_so_LDADD = @GTK_LIBS@ -lxmms
+#gyachixmms_so_LDFLAGS = -shared -fpic -fPIC -Xlinker -Bdynamic -Wl,--export-dynamic
 
 # Required for localization
 localedir = $(datadir)/locale
 
-INCLUDES = $(DEPS_CFLAGS) -DLOCALEDIR=\"$(localedir)\" -I../client @GTK_CFLAGS@ @GLIB_CFLAGS@ -fpic -FPIC
-
+#INCLUDES = $(DEPS_CFLAGS) -DLOCALEDIR=\"$(localedir)\" -I../client @GTK_CFLAGS@ @GLIB_CFLAGS@
+INCLUDES = $(DEPS_CFLAGS) -DLOCALEDIR=\"$(localedir)\" -I../client @GTK_CFLAGS@ @GLIB_CFLAGS@ -shared -fpic -fPIC
