diff -Nur mplayerplug-in-old/Source/plugin-setup.cpp mplayerplug-in/Source/plugin-setup.cpp
--- mplayerplug-in-old/Source/plugin-setup.cpp	2007-11-22 19:30:05.746477489 +0200
+++ mplayerplug-in/Source/plugin-setup.cpp	2007-11-22 19:35:01.433526224 +0200
@@ -773,6 +773,8 @@
 		    instance->loop = 0;	// 0 = infinite (like mplayer)
 		} else if (isdigit((int) *(parameters->argv[i]))) {
 		    sscanf(parameters->argv[i], "%i", &instance->loop);
+            if (instance->loop == 0)
+                instance->loop = -1;
 		} else {
 		    instance->loop = -1;	// -1 = loop disabled
 		}
