diff -Nur mplayerplug-in-old/Source/plugin-threads.cpp mplayerplug-in/Source/plugin-threads.cpp
--- mplayerplug-in-old/Source/plugin-threads.cpp	2007-11-22 19:30:05.746477489 +0200
+++ mplayerplug-in/Source/plugin-threads.cpp	2007-11-22 19:38:28.664022400 +0200
@@ -872,7 +872,8 @@
 
 	pthread_testcancel();
 	if (strstr(buffer, "Cache size") != NULL
-	    || strstr(buffer, "Starting playback") != NULL) {
+	    || strstr(buffer, "Starting playback") != NULL
+        || strstr(buffer, "mmst packet_length") != NULL) {
 	    if (local_td->instance->player != NULL) {
 		if (DEBUG) {
 		    printf("Setting file mode back to blocking\n");
