diff -Naur qtm-1.0.orig/qtm-desktop.sh qtm-1.0/qtm-desktop.sh
--- qtm-1.0.orig/qtm-desktop.sh	2009-06-15 23:36:08.000000000 +0300
+++ qtm-1.0/qtm-desktop.sh	2009-06-15 23:55:37.000000000 +0300
@@ -5,7 +5,9 @@
 Type=Application
 Name=QTM
 GenericName=Blog editor
+GenericName[tr]=Günlük İstemcisi
 Comment=Weblog management application
+Comment[tr]=İnternet günlüğü yönetim uygulaması
 TryExec=$1/bin/qtm
 Exec=$1/bin/qtm
 Categories=Application;Network;X-MandrivaLinux-Internet-Other;
