--- smoon/hardware/config/app.cfg 2009-02-04 17:20:38.000000000 +0200
+++ smoon/hardware/config/app.cfg.orig 2009-03-12 20:27:53.000000000 +0200
@@ -62,10 +62,10 @@
 
 [/static]
 static_filter.on = True
-static_filter.dir = "%(top_level_dir)s/static"
+static_filter.dir = "/usr/share/smoon/hardware/static"
 
 [/favicon.ico]
 static_filter.on = True
-static_filter.file = "%(top_level_dir)s/static/images/favicon.ico"
+static_filter.file = "/usr/share/smoon/hardware/static/images/favicon.ico"
 
 
