--- smoon/setup.py	2009-03-14 19:36:22.000000000 +0200
+++ smoon/setup.py.new	2009-03-14 22:14:03.000000000 +0200
@@ -30,19 +30,19 @@
         # Python Cheeseshop
         
         # if this has widgets, uncomment the next line
-        # 'turbogears.widgets',
+         'turbogears.widgets',
         
         # if this has a tg-admin command, uncomment the next line
-        # 'turbogears.command',
+         'turbogears.command',
         
         # if this has identity providers, uncomment the next line
         # 'turbogears.identity.provider',
     
         # If this is a template plugin, uncomment the next line
-        # 'python.templating.engines',
+         'python.templating.engines',
         
         # If this is a full application, uncomment the next line
-        # 'turbogears.app',
+         'turbogears.app',
     ],
     classifiers = [
         'Development Status :: 3 - Alpha',
