diff -urN kdenetwork-4.2.3/krdc/mainwindow.cpp kdenetwork-4.2.3-new/krdc/mainwindow.cpp
--- kdenetwork-4.2.3/krdc/mainwindow.cpp	2009-04-30 15:13:15.000000000 +0600
+++ kdenetwork-4.2.3-new/krdc/mainwindow.cpp	2009-06-11 12:11:45.000000000 +0600
@@ -206,7 +206,7 @@
 
     KAction *disconnectAction = actionCollection()->addAction("disconnect");
     disconnectAction->setText(i18n("Disconnect"));
-    disconnectAction->setIcon(KIcon("system-log-out"));
+    disconnectAction->setIcon(KIcon("network-disconnect"));
     connect(disconnectAction, SIGNAL(triggered()), SLOT(disconnect()));
 
     QAction *showLocalCursorAction = actionCollection()->addAction("show_local_cursor");
@@ -712,7 +712,7 @@
         m_toolBar->addAction(actionCollection()->action("switch_fullscreen"));
 
         QAction *minimizeAction = new QAction(m_toolBar);
-        minimizeAction->setIcon(KIcon("window-suppressed"));
+        minimizeAction->setIcon(KIcon("go-down"));
         minimizeAction->setText(i18n("Minimize Fullscreen Window"));
         connect(minimizeAction, SIGNAL(triggered()), m_fullscreenWindow, SLOT(showMinimized()));
         m_toolBar->addAction(minimizeAction);
