--- Objects/stringobject.c.orig	2008-04-12 16:47:07.000000000 +0300
+++ Objects/stringobject.c	2008-04-12 16:47:39.000000000 +0300
@@ -53,7 +53,11 @@
 PyString_FromStringAndSize(const char *str, Py_ssize_t size)
 {
 	register PyStringObject *op;
-	assert(size >= 0);
+    if (size < 0) {
+        PyErr_SetString(PyExc_SystemError,
+                "Negative size passed to PyString_FromStringAndSize");
+        return NULL;
+    }
 	if (size == 0 && (op = nullstring) != NULL) {
 #ifdef COUNT_ALLOCS
 		null_strings++;
