Index: trunk/libclamav/vba_extract.c
===================================================================
--- trunk/libclamav/vba_extract.c	(revision 4310)
+++ trunk/libclamav/vba_extract.c	(revision 4311)
@@ -107,7 +107,7 @@
 	if((name == NULL) || (*name == '\0') || (size <= 0))
 		return NULL;
 
-	newname = (char *)cli_malloc(size * 7);
+	newname = (char *)cli_malloc(size * 7 + 1);
 	if(newname == NULL)
 		return NULL;
 
