diff -Nur ffmpeg-old/cmdutils.c ffmpeg/cmdutils.c
--- ffmpeg-old/cmdutils.c	2009-04-24 14:31:21.000000000 +0300
+++ ffmpeg/cmdutils.c	2009-04-24 14:36:40.000000000 +0300
@@ -290,6 +290,7 @@
 {
     fprintf(stderr, "%s version " FFMPEG_VERSION ", Copyright (c) %d-%d Fabrice Bellard, et al.\n",
             program_name, program_birth_year, this_year);
+    /*
     fprintf(stderr, "  configuration: " FFMPEG_CONFIGURATION "\n");
     print_all_lib_versions(stderr, 1);
     fprintf(stderr, "  built on " __DATE__ " " __TIME__);
@@ -298,6 +299,7 @@
 #else
     fprintf(stderr, ", using a non-gcc compiler\n");
 #endif
+    */
 }
 
 void show_version(void) {
@@ -307,7 +309,7 @@
 void show_license(void)
 {
     printf(
-#if CONFIG_NONFREE
+#if 0
     "This version of %s has nonfree parts compiled in.\n"
     "Therefore it is not legally redistributable.\n",
     program_name
