--- src/osgCal/HardwareModel.cpp~	2008-02-21 20:02:49.000000000 -0800
+++ src/osgCal/HardwareModel.cpp	2008-02-21 20:03:06.000000000 -0800
@@ -12,7 +12,7 @@
 #include "config.h"
 #endif
 
-
+#include <string.h>
 #include "osgCal/HardwareModel"
 #include "osgCal/CustomAssert"
 
--- src/osgCal/SubMeshSoftware.cpp~	2008-02-21 20:13:14.000000000 -0800
+++ src/osgCal/SubMeshSoftware.cpp	2008-02-21 20:13:51.000000000 -0800
@@ -48,6 +48,7 @@
 #include <osgCal/SubMeshSoftware>
 
 #include <iostream>
+#include <string.h>
 
 using namespace osgCal;
 
