Index: kio/kio/defaultprogress.cpp
===================================================================
--- kio/kio/defaultprogress.cpp	(revision 669953)
+++ kio/kio/defaultprogress.cpp	(working copy)
@@ -236,7 +236,7 @@
                                             unsigned long totalFiles )
 {
   if ( totalSize )
-      return i18n( "%1 % of %2 " ).arg( percent ).arg( KIO::convertSize( totalSize ) );
+      return i18n( "%1 % of %2 " ).arg( QString::number(percent) , KIO::convertSize( totalSize ) );
   else if ( totalFiles )
       return i18n( "%1 % of 1 file", "%1 % of %n files", totalFiles ).arg( percent );
   else
