diff -ur xorg-app-7.1.orig/xsm-1.0.1/configure.ac xorg-app-7.1/xsm-1.0.1/configure.ac
--- xorg-app-7.1.orig/xsm-1.0.1/configure.ac	2005-12-21 04:29:54.000000000 +0200
+++ xorg-app-7.1/xsm-1.0.1/configure.ac	2006-10-30 00:52:37.000000000 +0200
@@ -31,20 +31,6 @@
 AC_PROG_CC
 AC_PROG_INSTALL
 
-RSH=
-if [[ -z $RSH ]] ; then
-   AC_PATH_PROG(RSH,rsh)
-fi
-if [[ -z $RSH ]] ; then
-   AC_PATH_PROG(RSH,rcmd)
-fi
-if [[ -z $RSH ]] ; then
-   AC_PATH_PROG(RSH,remsh)
-fi
-if [[ -z $RSH ]] ; then
-   AC_MSG_ERROR(No rsh like program found)
-fi
-
 #if !defined(RshCmd)
 #if SystemV
 #ifdef HPArchitecture
