diff -Naur hellanzb-0.11.orig/Hellanzb/Core.py hellanzb-0.11/Hellanzb/Core.py
--- hellanzb-0.11.orig/Hellanzb/Core.py	2007-02-21 11:23:44.000000000 +0200
+++ hellanzb-0.11/Hellanzb/Core.py	2007-02-21 11:24:04.000000000 +0200
@@ -35,7 +35,7 @@
             sys.exit(1)
 
     # look for conf in this order: sys.prefix, ./, or ./etc/
-    confDirs = [os.path.join(sys.prefix, 'etc')]
+    confDirs = ['/etc', os.path.join(sys.prefix, 'etc')]
     try:
         confDirs.append(os.path.join(os.getcwd(), 'etc'))
         confDirs.append(os.getcwd())
