Index: configure
===================================================================
RCS file: /cvsroot/gpac/gpac/configure,v
retrieving revision 1.23
diff -u -r1.23 configure
--- configure	27 Feb 2007 16:18:02 -0000	1.23
+++ configure	2 Mar 2007 12:49:36 -0000
@@ -462,7 +462,7 @@
 elif test "$darwin" = "yes" ; then
 LINK3D="-lgl -lglu"
 else
-LINK3D="-lgl -lglut"
+LINK3D="-lGL -lglut"
 fi
 
 if $cc -o $TMPO $TMPC $LINK3D 2> /dev/null  ; then
