diff -ur apparmor-docs-2.0.orig/Makefile apparmor-docs-2.0/Makefile
--- apparmor-docs-2.0.orig/Makefile	2006-03-12 03:27:09.000000000 +0200
+++ apparmor-docs-2.0/Makefile	2006-03-12 03:28:05.000000000 +0200
@@ -111,7 +111,6 @@
 .PHONY: install_documents
 install_documents: ${HTMLMANPAGES}
 	mkdir -m 755 -p ${DESTDIR}/${DOCDIR} ${DESTDIR}/${DOCDIR}/papers
-	install -m 644 -D ${PAPERS} ${DESTDIR}/${DOCDIR}/papers
 
 .PHONY: install
 install: install_manpages install_documents
