Index: gtk-qt-engine/kcm_gtk/kcmgtk.cpp
===================================================================
--- gtk-qt-engine.orig/kcm_gtk/kcmgtk.cpp
+++ gtk-qt-engine/kcm_gtk/kcmgtk.cpp
@@ -393,8 +393,10 @@ void KcmGtk::save()
 	if (!found)
 	{
 		stream << "#!/bin/bash\n\n";
+		stream << "if [ -e /usr/kde/3.5/lib/kde3/kcm_kcmgtk.so ]; then\n";
 		stream << "# Make sure our customised gtkrc file is loaded.\n";
 		stream << "export GTK2_RC_FILES=$HOME/" + GTK_RC_FILE + "\n";
+		stream << "fi\n";
 	}
 	file.close();
 	
