diff -Naur xastir-1.9.8/scripts/Makefile.am xastir-1.9.8-mod/scripts/Makefile.am
--- xastir-1.9.8/scripts/Makefile.am	2010-01-31 04:12:23.000000000 +0200
+++ xastir-1.9.8-mod/scripts/Makefile.am	2010-04-14 07:54:51.000000000 +0300
@@ -3,7 +3,7 @@
 # Copyright (C) 2000-2010  The Xastir Group
 #
 
-pkglib_SCRIPTS = \
+pkgdata_SCRIPTS = \
     Coordinate.pm \
     coord-convert.pl \
     get-fcc-rac.pl \
@@ -43,5 +43,5 @@
     xastir-migrate.sh \
     Xastir_tigerpoly.py
 
-EXTRA_DIST = $(pkglib_SCRIPTS)
+EXTRA_DIST = $(pkgdata_SCRIPTS)
 
