Index: 0ad-r09786-alpha/build/premake/premake.lua
===================================================================
--- 0ad-r09786-alpha.orig/build/premake/premake.lua
+++ 0ad-r09786-alpha/build/premake/premake.lua
@@ -42,9 +42,10 @@ end
 
 -- Set up the Project
 project.name = "pyrogenesis"
-project.bindir   = "../../binaries/system"
-project.libdir   = "../../binaries/system"
+project.bindir   = "../../binaries/usr/bin"
+project.libdir   = "../../binaries/usr/lib/0ad"
 project.debugdir = "../../binaries/system"
+project.datadir = "../../binaries/usr/share/0ad"
 if not options["outpath"] then
 	error("You must specify the 'outpath' parameter")
 end
Index: 0ad-r09786-alpha/libraries/nvtt/build.sh
===================================================================
--- 0ad-r09786-alpha.orig/libraries/nvtt/build.sh
+++ 0ad-r09786-alpha/libraries/nvtt/build.sh
@@ -28,4 +28,4 @@ else
 fi
 
 cp src/build/src/nv*/${LIB_PREFIX}nv*.${LIB_EXTN} lib/
-cp src/build/src/nv*/${LIB_PREFIX}nv*.${DLL_EXTN} ../../binaries/system/
+cp src/build/src/nv*/${LIB_PREFIX}nv*.${DLL_EXTN} ../../binaries/usr/lib/0ad
Index: 0ad-r09786-alpha/libraries/spidermonkey-tip/build.sh
===================================================================
--- 0ad-r09786-alpha.orig/libraries/spidermonkey-tip/build.sh
+++ 0ad-r09786-alpha/libraries/spidermonkey-tip/build.sh
@@ -88,7 +88,7 @@ cp -L src/build-release/dist/include/* $
 cp -L src/build-debug/dist/lib/${LIB_PREFIX}mozjs-ps-debug.${LIB_EXTN} lib/
 cp -L src/build-release/dist/lib/${LIB_PREFIX}mozjs-ps-release.${LIB_EXTN} lib/
 cp -L src/build-debug/dist/bin/${LIB_PREFIX}mozjs-ps-debug.${DLL_EXTN} ../../binaries/system/
-cp -L src/build-release/dist/bin/${LIB_PREFIX}mozjs-ps-release.${DLL_EXTN} ../../binaries/system/
+cp -L src/build-release/dist/bin/${LIB_PREFIX}mozjs-ps-release.${DLL_EXTN} ../../binaries/usr/lib/0ad
 
 # Flag that it's already been built successfully so we can skip it next time
 touch .already-built
