Index: xboard-4.5.2/xboard.conf.in
===================================================================
--- xboard-4.5.2.orig/xboard.conf.in
+++ xboard-4.5.2/xboard.conf.in
@@ -57,8 +57,8 @@
 ;
 ; Engines & adjudicatons in engine-engine games
 ;
--firstChessProgram fairymax
--secondChessProgram fairymax
+-firstChessProgram gnuchess
+-secondChessProgram gnuchess
 -niceEngines 0
 -polyglotDir ""
 -adapterCommand 'polyglot -noini -ec "%fcp" -ed "%fd" -uci NalimovCache=%defaultCacheSizeEGTB'
Index: xboard-4.5.2/xboard.h
===================================================================
--- xboard-4.5.2.orig/xboard.h
+++ xboard-4.5.2/xboard.h
@@ -104,8 +104,8 @@ typedef struct {
 
 #define BORDER_X_OFFSET 3
 #define BORDER_Y_OFFSET 27
-#define FIRST_CHESS_PROGRAM	"fairymax"
-#define SECOND_CHESS_PROGRAM	"fairymax"
+#define FIRST_CHESS_PROGRAM	"gnuchess"
+#define SECOND_CHESS_PROGRAM	"gnuchess"
 #define FIRST_DIRECTORY         "."
 #define SECOND_DIRECTORY        "."
 #define SOUND_BELL              ""
