Index: libsdl1.2-1.2.13/build-scripts/makedep.sh
===================================================================
--- libsdl1.2-1.2.13.orig/build-scripts/makedep.sh	2007-12-31 05:47:51.000000000 +0100
+++ libsdl1.2-1.2.13/build-scripts/makedep.sh	2008-01-02 03:19:59.000000000 +0100
@@ -69,7 +69,7 @@
         ;;
         asm) cat >>${output}.new <<__EOF__
 
-	\$(LIBTOOL) --tag=CC --mode=compile \$(auxdir)/strip_fPIC.sh \$(NASM) $src -o \$@
+	\$(LIBTOOL) --tag=CC --mode=compile \$(auxdir)/strip_fPIC.sh \$(NASM) -I\$(srcdir)/src/hermes/ $src -o \$@
 
 __EOF__
         ;;
