diff -urN last.fm-1.3.0.62-07/src/container.cpp last.fm-1.3.0.62-20/src/container.cpp
--- last.fm-1.3.0.62-07/src/container.cpp	2007-07-02 09:45:38.000000000 -0700
+++ last.fm-1.3.0.62-20/src/container.cpp	2007-07-02 09:46:49.000000000 -0700
@@ -982,7 +982,7 @@
     // Hack to get it working with VS2005
     dirPath = qApp->applicationDirPath();
   #else
-    dirPath = qApp->applicationDirPath() + "/extensions";
+    dirPath = "/usr/lib/lastfm/extensions";
   #endif
 
   #ifndef QT_NO_DEBUG
diff -urN last.fm-1.3.0.62-07/src/lastfmapplication.cpp last.fm-1.3.0.62-20/src/lastfmapplication.cpp
--- last.fm-1.3.0.62-07/src/lastfmapplication.cpp	2007-06-27 10:12:26.000000000 -0700
+++ last.fm-1.3.0.62-20/src/lastfmapplication.cpp	2007-07-02 09:46:49.000000000 -0700
@@ -60,7 +60,7 @@
 
     // This is needed so that relative paths will work on Windows regardless
     // of where the app is launched from.
-    QDir::setCurrent( applicationDirPath() );
+    //QDir::setCurrent( applicationDirPath() );
 
     // run init() as soon as event loop starts
     QTimer::singleShot( 0, this, SLOT(init()) );
@@ -184,8 +184,8 @@
     m_lang = langCode;
 
     // Discards previously loaded translations
-    m_translatorApp.load( dataPath( "i18n/lastfm_%1" ).arg( langCode ) );
-    m_translatorQt.load( dataPath( "i18n/qt_%1" ).arg( langCode ) );
+    m_translatorApp.load( QString( "/usr/share/lastfm/i18n/lastfm_%1" ).arg( langCode ) );
+    m_translatorQt.load( QString( "/usr/share/lastfm/i18n/qt_%1" ).arg( langCode ) );
 }
 
 
diff -urN last.fm-1.3.0.62-07/src/libLastFmTools/containerutils.cpp last.fm-1.3.0.62-20/src/libLastFmTools/containerutils.cpp
--- last.fm-1.3.0.62-07/src/libLastFmTools/containerutils.cpp	2007-06-27 10:12:26.000000000 -0700
+++ last.fm-1.3.0.62-20/src/libLastFmTools/containerutils.cpp	2007-07-02 09:46:49.000000000 -0700
@@ -67,7 +67,7 @@
 
 QString dataPath( QString file )
 {
-    return QApplication::applicationDirPath() + "/data/" + file;
+    return QString( "/usr/share/lastfm/%1" ).arg( file );
 }
 
 
@@ -119,7 +119,7 @@
         // Hack to get it working with VS2005
         dirPath = qApp->applicationDirPath();
     #else
-        dirPath = qApp->applicationDirPath() + "/services";
+        dirPath = "/usr/lib/lastfm/services";
     #endif
 
     #ifndef QT_NO_DEBUG
