diff -Naur audex.orig/widgets/generalsettingswidget.cpp audex/widgets/generalsettingswidget.cpp
--- audex.orig/widgets/generalsettingswidget.cpp	2009-04-12 20:06:35.000000000 +0300
+++ audex/widgets/generalsettingswidget.cpp	2009-04-12 20:43:07.000000000 +0300
@@ -39,6 +39,7 @@
   kcfg_wikipediaLocale->addItem(QIcon(KStandardDirs::locate("data", "audex/images/spain.png")), i18n("Español"), "es");
   kcfg_wikipediaLocale->addItem(QIcon(KStandardDirs::locate("data", "audex/images/portugal.png")), i18n("Português"), "pt");
   kcfg_wikipediaLocale->addItem(QIcon(KStandardDirs::locate("data", "audex/images/sweden.png")), i18n("Svenska"), "sv");
+  kcfg_wikipediaLocale->addItem(QIcon(KStandardDirs::locate("data", "audex/images/turkey.png")), i18n("Türkçe"), "tr");
 
   urlreq_basePath->setMode(KFile::Directory|KFile::LocalOnly);
   urlreq_basePath->lineEdit()->setObjectName("kcfg_basePath");
