--- OpenCity.desktop.orig	2006-10-08 13:57:40.000000000 +0300
+++ OpenCity.desktop	2007-10-24 01:54:01.000000000 +0300
@@ -1,11 +1,13 @@
 [Desktop Entry]
-Categories=Game
+Categories=Game;Simulation;
 Comment=A 3D city simulator
 Comment[fr]=Jeu de simulation de Ville en OpenGL
+Comment[tr]=3B şehir simülasyonu
 Encoding=UTF-8
 Exec=opencity
 GenericName=City simulator
 GenericName[fr]=Simulateur de ville
+GenericName[tr]=Şehir simülatörü
 Icon=OpenCity
 Name=OpenCity
 Name[fr]=OpenCity
