--- Makefile.in	2007-05-08 00:50:34.000000000 +0300
+++ Makefile.in	2007-05-23 19:19:48.000000000 +0300
@@ -167,9 +167,9 @@
 
 	chmod -R +rx $(DESTDIR)$(dstdir)/$(PACKAGE)
 
-	ln -sf $(dstdir)/$(PACKAGE)/amsn $(DESTDIR)$(slnkdir)/amsn
-	ln -sf $(dstdir)/$(PACKAGE)/amsn-remote $(DESTDIR)$(slnkdir)/amsn-remote
-	ln -sf $(dstdir)/$(PACKAGE)/amsn-remote-CLI $(DESTDIR)$(slnkdir)/amsn-remote-CLI
+	ln -sf /usr/share/amsn/amsn $(DESTDIR)$(slnkdir)/amsn
+	ln -sf /usr/share/amsn/amsn-remote $(DESTDIR)$(slnkdir)/amsn-remote
+	ln -sf /usr/share/amsn/amsn-remote-CLI $(DESTDIR)$(slnkdir)/amsn-remote-CLI
 
 	cp $(srcdir)/amsn.desktop $(DESTDIR)$(dstdir)/applications/
 
