diff -Nur fontconfig-2.6.0-old/conf.d/Makefile.am fontconfig-2.6.0/conf.d/Makefile.am
--- fontconfig-2.6.0-old/conf.d/Makefile.am	2008-09-22 11:17:40.000000000 +0300
+++ fontconfig-2.6.0/conf.d/Makefile.am	2008-09-22 11:18:41.000000000 +0300
@@ -25,7 +25,12 @@
 	README
 
 CONF_FILES = \
+	10-antialias.conf \
 	10-autohint.conf \
+	10-hinting.conf \
+	10-hinting-full.conf \
+	10-hinting-medium.conf \
+	10-hinting-slight.conf \
 	10-no-sub-pixel.conf \
 	10-sub-pixel-bgr.conf \
 	10-sub-pixel-rgb.conf \
@@ -42,6 +47,7 @@
 	49-sansserif.conf \
 	50-user.conf \
 	51-local.conf \
+	53-monospace-lcd-filter.conf \
 	60-latin.conf \
 	65-fonts-persian.conf \
 	65-khmer.conf \
@@ -53,6 +59,10 @@
 	90-synthetic.conf
 
 CONF_LINKS = \
+	10-antialias.conf \
+	10-hinting.conf \
+	10-hinting-medium.conf \
+	10-no-sub-pixel.conf \
 	20-fix-globaladvance.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
@@ -62,6 +72,7 @@
 	49-sansserif.conf \
 	50-user.conf \
 	51-local.conf \
+	53-monospace-lcd-filter.conf \
 	60-latin.conf \
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
diff -Nur fontconfig-2.6.0-old/conf.d/Makefile.in fontconfig-2.6.0/conf.d/Makefile.in
--- fontconfig-2.6.0-old/conf.d/Makefile.in	2008-09-22 11:17:40.000000000 +0300
+++ fontconfig-2.6.0/conf.d/Makefile.in	2008-09-22 11:18:49.000000000 +0300
@@ -242,7 +242,12 @@
 	README
 
 CONF_FILES = \
+	10-antialias.conf \
 	10-autohint.conf \
+	10-hinting.conf \
+	10-hinting-full.conf \
+	10-hinting-medium.conf \
+	10-hinting-slight.conf \
 	10-no-sub-pixel.conf \
 	10-sub-pixel-bgr.conf \
 	10-sub-pixel-rgb.conf \
@@ -259,6 +264,7 @@
 	49-sansserif.conf \
 	50-user.conf \
 	51-local.conf \
+	53-monospace-lcd-filter.conf \
 	60-latin.conf \
 	65-fonts-persian.conf \
 	65-khmer.conf \
@@ -270,6 +276,10 @@
 	90-synthetic.conf
 
 CONF_LINKS = \
+	10-antialias.conf \
+	10-hinting.conf \
+	10-hinting-medium.conf \
+	10-no-sub-pixel.conf \
 	20-fix-globaladvance.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
@@ -279,6 +289,7 @@
 	49-sansserif.conf \
 	50-user.conf \
 	51-local.conf \
+	53-monospace-lcd-filter.conf \
 	60-latin.conf \
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
