diff -ur kdebluetooth-1.0_beta2.orig/configure kdebluetooth-1.0_beta2/configure
--- kdebluetooth-1.0_beta2.orig/configure	2006-10-16 13:45:32.000000000 +0300
+++ kdebluetooth-1.0_beta2/configure	2006-11-09 21:18:02.000000000 +0200
@@ -32819,7 +32819,7 @@
     BLUETOOTH_LIBS="-L$prefix/lib"
   fi
 
-  BLUETOOTH_LIBS_2_7="$BLUETOOTH_LIBS -lbluetooth -lsdp"
+  BLUETOOTH_LIBS_2_7="$BLUETOOTH_LIBS -lbluetooth"
   BLUETOOTH_LIBS_2_8="$BLUETOOTH_LIBS -lbluetooth"
 
   if test "x$bluetooth_includes" != "x" ; then
@@ -34366,7 +34366,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lsdp  $LIBS"
+LIBS="$LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -34505,7 +34505,7 @@
 
   LIB_BLUETOOTH=-lbluetooth
   if test "x${SDP_IN_SDP}" = "x1" ; then
-    LIB_SDP=-lsdp
+    LIB_SDP=
   else
     LIB_SDP=
   fi
diff -ur kdebluetooth-1.0_beta2.orig/configure.in kdebluetooth-1.0_beta2/configure.in
--- kdebluetooth-1.0_beta2.orig/configure.in	2006-10-16 13:45:16.000000000 +0300
+++ kdebluetooth-1.0_beta2/configure.in	2006-11-09 21:17:37.000000000 +0200
@@ -116,7 +116,7 @@
     BLUETOOTH_LIBS="-L$prefix/lib"
   fi
 
-  BLUETOOTH_LIBS_2_7="$BLUETOOTH_LIBS -lbluetooth -lsdp"
+  BLUETOOTH_LIBS_2_7="$BLUETOOTH_LIBS -lbluetooth"
   BLUETOOTH_LIBS_2_8="$BLUETOOTH_LIBS -lbluetooth"
 
   if test "x$bluetooth_includes" != "x" ; then
@@ -360,7 +360,7 @@
   AC_DEFINE_UNQUOTED(HAVE_QOBEX_BLUETOOTH, ${BLUETOOTH}, Defined if Bluetooth is enabled.)
   LIB_BLUETOOTH=-lbluetooth
   if test "x${SDP_IN_SDP}" = "x1" ; then
-    LIB_SDP=-lsdp
+    LIB_SDP=
   else
     LIB_SDP=
   fi
diff -ur kdebluetooth-1.0_beta2.orig/kdebluetooth/configure.in.in kdebluetooth-1.0_beta2/kdebluetooth/configure.in.in
--- kdebluetooth-1.0_beta2.orig/kdebluetooth/configure.in.in	2006-10-16 13:24:06.000000000 +0300
+++ kdebluetooth-1.0_beta2/kdebluetooth/configure.in.in	2006-11-09 21:17:37.000000000 +0200
@@ -36,7 +36,7 @@
     BLUETOOTH_LIBS="-L$prefix/lib"
   fi
 
-  BLUETOOTH_LIBS_2_7="$BLUETOOTH_LIBS -lbluetooth -lsdp"
+  BLUETOOTH_LIBS_2_7="$BLUETOOTH_LIBS -lbluetooth"
   BLUETOOTH_LIBS_2_8="$BLUETOOTH_LIBS -lbluetooth"
 
   if test "x$bluetooth_includes" != "x" ; then
diff -ur kdebluetooth-1.0_beta2.orig/kdebluetooth/libqobex/configure.in.in kdebluetooth-1.0_beta2/kdebluetooth/libqobex/configure.in.in
--- kdebluetooth-1.0_beta2.orig/kdebluetooth/libqobex/configure.in.in	2006-10-16 13:24:02.000000000 +0300
+++ kdebluetooth-1.0_beta2/kdebluetooth/libqobex/configure.in.in	2006-11-09 21:17:37.000000000 +0200
@@ -45,7 +45,7 @@
   AC_DEFINE_UNQUOTED(HAVE_QOBEX_BLUETOOTH, ${BLUETOOTH}, Defined if Bluetooth is enabled.)
   LIB_BLUETOOTH=-lbluetooth
   if test "x${SDP_IN_SDP}" = "x1" ; then
-    LIB_SDP=-lsdp
+    LIB_SDP=
   else
     LIB_SDP=
   fi
