diff -uNBwp Lib/platform.py.orig Lib/platform.py
--- Lib/platform.py.orig	2011-02-17 11:44:11.274915821 +0100
+++ Lib/platform.py	2011-02-17 11:44:50.106918239 +0100
@@ -255,7 +255,7 @@ _release_version = re.compile(r'([^0-9]+
 _supported_dists = (
     'SuSE', 'debian', 'fedora', 'redhat', 'centos',
     'mandrake', 'mandriva', 'rocks', 'slackware', 'yellowdog', 'gentoo',
-    'UnitedLinux', 'turbolinux')
+    'UnitedLinux', 'turbolinux', 'pardus')
 
 def _parse_release_file(firstline):
 
