diff -uNBwp Lib/gettext.py.orig Lib/gettext.py
--- Lib/gettext.py.orig	2010-12-24 00:45:39.000000000 +0100
+++ Lib/gettext.py	2011-02-17 20:05:46.587034898 +0100
@@ -265,6 +265,8 @@ class GNUTranslations(NullTranslations):
                     item = b_item.decode().strip()
                     if not item:
                         continue
+                    if item.startswith("#"):
+                        continue
                     if ':' in item:
                         k, v = item.split(':', 1)
                         k = k.strip().lower()
