Index: boson-0.13/bogl/bogl.cpp
===================================================================
--- boson-0.13.orig/code/bogl/bogl.cpp	2008-03-17 21:59:24.000000000 +0100
+++ boson-0.13/code/bogl/bogl.cpp	2008-03-17 22:00:29.000000000 +0100
@@ -45,8 +45,10 @@
 
 QStringList boglGetOpenGLExtensions()
 {
- QString extensions = (const char*)glGetString(GL_EXTENSIONS);
- return QStringList::split(" ", extensions);
+// QString extensions = (const char*)glGetString(GL_EXTENSIONS);
+// return QStringList::split(" ", extensions);
+// patch by Paul Wise to avoid a segfault on AMD64
+  return BoGL::bogl()->OpenGLExtensions();
 }
 
 QStringList boglGetGLUExtensions()
