Index: boson-0.13/boson/gameengine/script/pythonscript.cpp
===================================================================
--- boson-0.13.orig/code/boson/gameengine/script/pythonscript.cpp
+++ boson-0.13/code/boson/gameengine/script/pythonscript.cpp
@@ -688,7 +688,7 @@
   //  variables or submodules dict if possible.
   PyObject* key;
   PyObject* value;
-  int pos = 0;
+  Py_ssize_t pos = 0;
   while(PyDict_Next(moduledict, &pos, &key, &value))
   {
     // Check if value is any of the known types
@@ -774,7 +774,7 @@
   boDebug(700) << k_funcinfo << "Loading and merging " << PyDict_Size(submodules) << " submodules" << endl;
   PyObject* key;
   PyObject* value;
-  int pos = 0;
+  Py_ssize_t pos = 0;
   while(PyDict_Next(submodules, &pos, &key, &value))
   {
     // Check if current module already has module with this name
Index: boson-0.13/boson/gameengine/script/pythonscript.h
===================================================================
--- boson-0.13.orig/code/boson/gameengine/script/pythonscript.h
+++ boson-0.13/code/boson/gameengine/script/pythonscript.h
@@ -29,6 +29,11 @@
 struct PyMethodDef;
 typedef struct _ts PyThreadState;
 
+#include <patchlevel.h>
+/* Python < 2.5 compat */
+#if PY_VERSION_HEX < 0x02050000
+typedef int Py_ssize_t;
+#endif
 
 class PythonScript : public BosonScript
 {
