--- cmake/modules/boson/FindBosonPythonLibs.cmake.orig	2006-06-17 16:36:11.000000000 +0300
+++ cmake/modules/boson/FindBosonPythonLibs.cmake	2008-04-11 22:29:37.000000000 +0300
@@ -72,6 +72,6 @@
 
 	# PYTHON_LIBRARY is used for the cache entry only
 	# PYTHON_LIBRARIES is meant to be public
-	set(PYTHON_LIBRARIES "${PYTHON_LIBRARY} ${_python_dependency_libs}")
+	set(PYTHON_LIBRARIES ${PYTHON_LIBRARY} ${_python_dependency_libs})
 ENDIF (NOT PYTHON_LIBRARY)
 
