diff -Nur eric4-4.1.6-old/eric/QScintilla/MiniEditor.py eric4-4.1.6/eric/QScintilla/MiniEditor.py
--- eric4-4.1.6-old/eric/QScintilla/MiniEditor.py	2008-07-09 15:58:36.000000000 +0300
+++ eric4-4.1.6/eric/QScintilla/MiniEditor.py	2008-07-09 15:59:35.000000000 +0300
@@ -878,8 +878,8 @@
         
         act = E4Action(QApplication.translate('ViewManager', 'Insert new line'), 
                       QApplication.translate('ViewManager', 'Insert new line'), 
-                      QKeySequence(QApplication.translate('ViewManager', 'Return')), 
                       QKeySequence(QApplication.translate('ViewManager', 'Enter')), 
+                      QKeySequence(),
                       self.editorActGrp, 'vm_edit_insert_line')
         act.mapIt(self.esm, QsciScintilla.SCI_NEWLINE)
         act.connectIt(SIGNAL('triggered()'), self.esm, SLOT('map()'))
diff -Nur eric4-4.1.6-old/eric/ViewManager/ViewManager.py eric4-4.1.6/eric/ViewManager/ViewManager.py
--- eric4-4.1.6-old/eric/ViewManager/ViewManager.py	2008-07-09 15:58:36.000000000 +0300
+++ eric4-4.1.6/eric/ViewManager/ViewManager.py	2008-07-09 15:58:46.000000000 +0300
@@ -1499,8 +1499,8 @@
         
         act = E4Action(QApplication.translate('ViewManager', 'Insert new line'), 
                       QApplication.translate('ViewManager', 'Insert new line'), 
-                      QKeySequence(QApplication.translate('ViewManager', 'Return')), 
                       QKeySequence(QApplication.translate('ViewManager', 'Enter')), 
+                      QKeySequence(), 
                       self.editorActGrp, 'vm_edit_insert_line')
         act.mapIt(self.esm, QsciScintilla.SCI_NEWLINE)
         act.connectIt(SIGNAL('triggered()'), self.esm, SLOT('map()'))
