diff -Nur libxml++-2.22.0-old/Makefile.in libxml++-2.22.0/Makefile.in
--- libxml++-2.22.0-old/Makefile.in	2008-05-23 14:57:45.000000000 +0300
+++ libxml++-2.22.0/Makefile.in	2008-05-23 14:57:54.000000000 +0300
@@ -198,7 +198,7 @@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = libxml++ docs scripts examples MSVC_Net2003
+SUBDIRS = libxml++ docs scripts  MSVC_Net2003
 EXTRA_DIST = libxml++-2.6.pc.in libxml++.spec.in
 DIST_SUBDIRS = $(SUBDIRS) win32_msvc6
 
