--- linden/indra/SConstruct.orig	2008-04-08 05:37:53.000000000 +0300
+++ linden/indra/SConstruct	2008-04-26 14:00:19.000000000 +0300
@@ -740,7 +740,7 @@
 
         # Dataserver
         external_libs = common_external_libs + [
-            'boost_regex-gcc-mt', 'mysqlclient', 'tcmalloc', 'stacktrace',
+            'boost_regex-gcc', 'mysqlclient', 'tcmalloc', 'stacktrace',
             ]
         internal_libs = [ 'llcharacter', 'lldatabase', 'llimage', 'llimagej2coj', 'llinventory',
                 'llscene', 'llmessage', 'llvfs', 'llxml', 'llcommon', 'llmath' ]
@@ -763,7 +763,7 @@
 
         # Mapserver
         external_libs = common_external_libs + ['OSMesa16', 'kdu', 
-                'boost_regex-gcc-mt', 'iconv', 'jpeg', 'openjpeg', 'GL',
+                'boost_regex-gcc', 'iconv', 'jpeg', 'openjpeg', 'GL',
                 'mysqlclient', 'pthread', 'dl']
         internal_libs = ['llrender', 'llwindow', 'llimage', 'llimagej2coj', 'lldatabase', 'llprimitive', 'llmessage', 'llkdustatic',
                 'llxml', 'llvfs', 'llmath', 'llcommon']
@@ -773,7 +773,7 @@
         # Simulator
         external_libs = common_external_libs + [
             'hkdynamics', 'hkgeometry', 'hkmath', 'hkbase', 'hkcollide',
-            'hkactions', 'boost_regex-gcc-mt', 'openjpeg', 'dl', 'kdu',
+            'hkactions', 'boost_regex-gcc', 'openjpeg', 'dl', 'kdu',
             'mysqlclient', 'iconv', 'tcmalloc', 'stacktrace',
             ]
         internal_libs = [ 'lscript', 'llprimitive',
