diff -urNad linden~/indra/llvfs/lldir_linux.cpp linden/indra/llvfs/lldir_linux.cpp
--- linden~/indra/llvfs/lldir_linux.cpp	2007-07-14 16:25:21.000000000 +1000
+++ linden/indra/llvfs/lldir_linux.cpp	2007-07-14 16:36:51.000000000 +1000
@@ -90,7 +90,7 @@
 	mExecutablePathAndName = "";
 	mExecutableDir = tmp_str;
 	mWorkingDir = tmp_str;
-	mAppRODataDir = tmp_str;
+	mAppRODataDir = "/usr/share/secondlife";
 	mOSUserDir = getCurrentUserHome(tmp_str);
 	mOSUserAppDir = "";
 	mLindenUserDir = tmp_str;
