--- linden.orig/indra/llcommon/llhash.h	2008-04-08 05:37:54.000000000 +0300
+++ linden/indra/llcommon/llhash.h	2008-04-26 19:10:07.000000000 +0300
@@ -39,9 +39,9 @@
 #include <algorithm>
 #elif LL_DARWIN || LL_LINUX
 #  if GCC_VERSION >= 30400 // gcc 3.4 and up
-#    include <ext/hashtable.h>
+#    include <tr1/unordered_map>
 #  elif __GNUC__ >= 3
-#    include <ext/stl_hashtable.h>
+#    include <tr1/unordered_map>
 #  else
 #    include <hashtable.h>
 #  endif
