--- linden/indra/SConstruct.orig	2008-04-26 13:47:19.000000000 +0300
+++ linden/indra/SConstruct	2008-04-26 13:48:56.000000000 +0300
@@ -249,7 +249,7 @@
     ##############
 
     # Generic GCC flags
-    cflags = '-g -pipe -Wall -Wno-reorder -Wno-trigraphs -Wno-sign-compare -Werror -fexceptions '
+    cflags = '-g -pipe -Wall -Wno-reorder -Wno-trigraphs -Wno-sign-compare -Werror -fexceptions -I/usr/include/freetype2 '
     cxxflags = ''
     cppflags = '-D_FORTIFY_SOURCE=2 '
     if standalone:
@@ -294,7 +294,7 @@
             else:
                 client_external_libs += [ 'gtk-x11-2.0', 'atk-1.0', 'gmodule-2.0', 'gdk-x11-2.0', 'gdk_pixbuf-2.0', 'pango-1.0', 'pangoft2-1.0', 'pangox-1.0', 'pangoxft-1.0', 'Xinerama' , 'boost_regex']
                 incdirs = [ 'ELFIO', 'atk-1.0', 'glib-2.0', 'gtk-2.0',
-                            'llfreetype2', 'pango-1.0' ]
+                            'freetype2', 'pango-1.0' ]
                 include_dirs += ['../libraries/' + system_str + '/include/' + d
                                  for d in incdirs]
 
@@ -651,7 +651,7 @@
             external_libs += [ d[2:] for d in
                               pkgconfig('--libs-only-l').split() ]
         else:
-            external_libs += [ 'freetype', 'SDL', 'vorbisenc',
+            external_libs += [ 'freetype2', 'SDL', 'vorbisenc',
                                'vorbisfile', 'vorbis', 'ogg', 'db-4.2' ]
 
         external_libs += [ 'jpeg', 'openjpeg', 'png12', 'GL', 'GLU' ]
@@ -740,7 +740,7 @@
 
         # Dataserver
         external_libs = common_external_libs + [
-            'boost_regex-gcc', 'mysqlclient', 'tcmalloc', 'stacktrace',
+            'boost_regex-gcc-mt', 'mysqlclient', 'tcmalloc', 'stacktrace',
             ]
         internal_libs = [ 'llcharacter', 'lldatabase', 'llimage', 'llimagej2coj', 'llinventory',
                 'llscene', 'llmessage', 'llvfs', 'llxml', 'llcommon', 'llmath' ]
@@ -763,7 +763,7 @@
 
         # Mapserver
         external_libs = common_external_libs + ['OSMesa16', 'kdu', 
-                'boost_regex-gcc', 'iconv', 'jpeg', 'openjpeg', 'GL',
+                'boost_regex-gcc-mt', 'iconv', 'jpeg', 'openjpeg', 'GL',
                 'mysqlclient', 'pthread', 'dl']
         internal_libs = ['llrender', 'llwindow', 'llimage', 'llimagej2coj', 'lldatabase', 'llprimitive', 'llmessage', 'llkdustatic',
                 'llxml', 'llvfs', 'llmath', 'llcommon']
@@ -773,7 +773,7 @@
         # Simulator
         external_libs = common_external_libs + [
             'hkdynamics', 'hkgeometry', 'hkmath', 'hkbase', 'hkcollide',
-            'hkactions', 'boost_regex-gcc', 'openjpeg', 'dl', 'kdu',
+            'hkactions', 'boost_regex-gcc-mt', 'openjpeg', 'dl', 'kdu',
             'mysqlclient', 'iconv', 'tcmalloc', 'stacktrace',
             ]
         internal_libs = [ 'lscript', 'llprimitive',
