--- linden/indra/SConstruct.orig	2008-04-26 13:50:07.000000000 +0300
+++ linden/indra/SConstruct	2008-04-26 13:52:55.000000000 +0300
@@ -313,10 +313,10 @@
             # GStreamer stuff
             if enable_gstreamer:
                 cppflags += '-DLL_GSTREAMER_ENABLED=1 '
-                client_external_libs += [ 'glib-2.0', 'gobject-2.0', 'gthread-2.0' ]
-                include_dirs += [ '../libraries/' + system_str + '/include/gstreamer-0.10' ]
-                include_dirs += [ '../libraries/' + system_str + '/include/glib-2.0', '../libraries/' + system_str + '/include/glib-2.0/include' ]
-                include_dirs += [ '../libraries/' + system_str + '/include/libxml2']
+                client_external_libs += [ 'glib-2.0', 'gobject-2.0', 'gthread-2.0', 'gstreamer-0.10' ]
+                include_dirs += [ '/usr/include/gstreamer-0.10' ]
+                include_dirs += [ '/usr/include/glib-2.0', '/usr/lib/glib-2.0/include' ]
+                include_dirs += [ '/usr/include/libxml2']
             else:
                 cppflags += '-DLL_GSTREAMER_ENABLED=0 '
     else:
