diff -Nuar drgeo-1.1.0.orig/glade/drgenius2.glade drgeo-1.1.0/glade/drgenius2.glade
--- drgeo-1.1.0.orig/glade/drgenius2.glade	2003-11-24 23:24:11.000000000 +0200
+++ drgeo-1.1.0/glade/drgenius2.glade	2009-04-13 21:25:32.000000000 +0300
@@ -7195,6 +7195,14 @@
 			  <property name="selection_mode">GTK_SELECTION_BROWSE</property>
 
 			  <child>
+ 			    <widget class="GtkListItem" id="listitem570">
+ 			      <property name="visible">True</property>
+ 			      <property name="can_focus">True</property>
+ 			      <property name="label" translatable="yes">htmlview </property>
+ 			    </widget>
+ 			  </child>
+ 
+ 			  <child>
 			    <widget class="GtkListItem" id="listitem57">
 			      <property name="visible">True</property>
 			      <property name="can_focus">True</property>
diff -Nuar drgeo-1.1.0.orig/scm/preferences.scm drgeo-1.1.0/scm/preferences.scm
--- drgeo-1.1.0.orig/scm/preferences.scm	2003-11-29 20:19:07.000000000 +0200
+++ drgeo-1.1.0/scm/preferences.scm	2009-04-13 21:26:57.000000000 +0300
@@ -10,7 +10,7 @@
  (:angleStyle "Normal")
  (:scalarColor "Blue")
  (:sessionFile "session")
- (:htmlViewer "dillo ")
+ (:htmlViewer "htmlview ")
  (:undoLevel "5")
  (:figureName "Figure %d")
  (:figureFile "figure_")
