diff -Nur AssaultCube_v1.0-old/config/menus.cfg AssaultCube_v1.0/config/menus.cfg
--- AssaultCube_v1.0-old/config/menus.cfg	2008-11-28 18:56:39.000000000 +0200
+++ AssaultCube_v1.0/config/menus.cfg	2008-11-28 18:59:58.000000000 +0200
@@ -27,7 +27,7 @@
 ]
 newmenu maps
 // Add maps to the below list to include them in the normal map menu (also for botmaps).
-alias defaultmaps "ac_arctic ac_complex ac_depot ac_desert ac_desert2 ac_desert3 ac_elevation ac_keller ac_mines ac_power ac_scaffold ac_shine ac_snow ac_sunset ac_toxic ac_urban"
+alias defaultmaps "ac_arctic ac_complex ac_depot ac_desert ac_desert2 ac_desert3 ac_elevation ac_keller ac_mines ac_power ac_scaffold ac_shine ac_snow ac_sunset ac_toxic ac_urban ac_factory fantasy illusion ac_rattrap ac_ruins snipe temple 2kingdms ac_prison ac_police_station azz octagon ac_CombatArena ac_Flood ac_RuinedFortifications Runners TheDarkCastle toll Trench TrenchWarfare2 Memico ac_castle ac_estate Bridge coy dust2 fy_grenade Labyrinth RuinedFortifications TunnelsDry 2fortac Fy_Ice_World ac_building ac_city ac_desertbase ac_inka ac_normandie aim_ak_colt fy_pool_day"
 genmapitems	$defaultmaps
 menuitem	"" -1
 menuitem	"Show all maps..." [ showmenu "all maps" ]
@@ -39,7 +39,7 @@
 
 newmenu maps_flags 
 // Add maps to the below list to include them in the CTF, KTF & TKTF map menu. Maps should contain flags here.
-alias ctfmaps "ac_depot ac_desert3 ac_elevation ac_keller ac_mines ac_power ac_shine ac_sunset ac_urban"
+alias ctfmaps "ac_depot ac_desert3 ac_elevation ac_keller ac_mines ac_power ac_shine ac_sunset ac_urban ac_factory fantasy ac_rattrap ac_ruins snipe 2kingdms azz ac_Flood ac_RuinedFortifications Runners TheDarkCastle toll Trench coy ac_estate RuinedFortifications Bridge TunnelsDry Memico ac_desertbase fy_pool_day ac_normandie ac_city Fy_Ice_World 2fortac ac_prison temple"
 genmapitems $ctfmaps
 menuitem "" -1
 menuitem "Show all maps.." [showmenu "all maps"]
