diff -Nur qc-usb-messenger-1.5-old/qc-driver.c qc-usb-messenger-1.5/qc-driver.c
--- qc-usb-messenger-1.5-old/qc-driver.c	2007-01-24 15:08:04.930710718 +0200
+++ qc-usb-messenger-1.5/qc-driver.c	2007-01-24 15:08:54.175224414 +0200
@@ -207,7 +207,6 @@
 	{ USB_DEVICE(0x046D, 0x08F0) },		/* QuickCam Messenger */
 	{ USB_DEVICE(0x046D, 0x08F5) },		/* QuickCam Communicate */
 	{ USB_DEVICE(0x046D, 0x08F6) },		/* QuickCam Messenger (new) */
-	{ USB_DEVICE(0x046D, 0x08DA) },		/* QuickCam Messenger (new) */
 	{ }
 };
 MODULE_DEVICE_TABLE(usb, qc_device_table);
