Index: akonadi-googledata-1.2.0/calendar/gcalresource.desktop
===================================================================
--- akonadi-googledata-1.2.0.orig/calendar/gcalresource.desktop
+++ akonadi-googledata-1.2.0/calendar/gcalresource.desktop
@@ -36,7 +36,7 @@ Name[uk]=Ресурс Календаря
 Name[x-test]=xxAkonadi Google Calendar Resourcexx
 Name[zh_CN]=Akonadi Google 日历资源
 Name[zh_TW]=Akonadi Google 行事曆資源
-Type=AkonadiResource
+Type=Application
 Exec=akonadi_gcal_resource
 Icon=text-calendar
 
Index: akonadi-googledata-1.2.0/contacts/googledataresource.desktop
===================================================================
--- akonadi-googledata-1.2.0.orig/contacts/googledataresource.desktop
+++ akonadi-googledata-1.2.0/contacts/googledataresource.desktop
@@ -70,7 +70,7 @@ Comment[uk]=Ресурс для облі
 Comment[x-test]=xxResource for a google accountxx
 Comment[zh_CN]=Google 账户的资源
 Comment[zh_TW]=Google 帳號的資源
-Type=AkonadiResource
+Type=Application
 Exec=akonadi_googledata_resource
 Icon=text-directory
 
