Index: zorg-1.86/zorg/hwdata.py
===================================================================
--- zorg-1.86.orig/zorg/hwdata.py
+++ zorg-1.86/zorg/hwdata.py
@@ -68,7 +68,7 @@ drivers = {
 }
 
 driverPackages = {
-        "fglrx":            "xorg-video-fglrx",
+        # "fglrx":            "xorg-video-fglrx",
         "nvidia71":         "xorg-video-nvidia71",
         "nvidia96":         "xorg-video-nvidia96",
         "nvidia173":        "xorg-video-nvidia173",
