--- lib-src/Makefile.in	2007-02-12 13:33:32.000000000 +0200
+++ lib-src/Makefile.in	2008-01-19 22:02:44.000000000 +0200
@@ -48,6 +48,7 @@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_DATA = @INSTALL_DATA@
 install_sh_DATA = $(install_sh) -c -m 644
+install_so_LIB = $(install_sh) -c -m 755
 install_sh_PROGRAM = $(install_sh) -c
 install_sh_SCRIPT = $(install_sh) -c
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
@@ -77,7 +78,7 @@
 am__include = @am__include@
 am__quote = @am__quote@
 install_sh = @install_sh@
-lib_LIBRARIES = libcdd.a
+lib_LIBRARIES = libcdd.so
 
 libcdd_a_SOURCES = \
 cddcore.c \
@@ -100,7 +101,7 @@
 subdir = lib-src
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_CLEAN_FILES =
-LIBRARIES = $(lib_LIBRARIES)
+LIBRARIES = shared
 
 libcdd_a_AR = $(AR) cru
 libcdd_a_LIBADD =
@@ -141,7 +142,7 @@
 	cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@ $(am__depfiles_maybe)
 
 AR = ar
-libLIBRARIES_INSTALL = $(INSTALL_DATA)
+libLIBRARIES_INSTALL = $(install_so_LIB)
 install-libLIBRARIES: $(lib_LIBRARIES)
 	@$(NORMAL_INSTALL)
 	$(mkinstalldirs) $(DESTDIR)$(libdir)
@@ -152,14 +153,6 @@
 	    $(libLIBRARIES_INSTALL) $$p $(DESTDIR)$(libdir)/$$f; \
 	  else :; fi; \
 	done
-	@$(POST_INSTALL)
-	@list='$(lib_LIBRARIES)'; for p in $$list; do \
-	  if test -f $$p; then \
-	    p="`echo $$p | sed -e 's|^.*/||'`"; \
-	    echo " $(RANLIB) $(DESTDIR)$(libdir)/$$p"; \
-	    $(RANLIB) $(DESTDIR)$(libdir)/$$p; \
-	  else :; fi; \
-	done
 
 uninstall-libLIBRARIES:
 	@$(NORMAL_UNINSTALL)
@@ -171,10 +164,9 @@
 
 clean-libLIBRARIES:
 	-test -z "$(lib_LIBRARIES)" || rm -f $(lib_LIBRARIES)
-libcdd.a: $(libcdd_a_OBJECTS) $(libcdd_a_DEPENDENCIES) 
-	-rm -f libcdd.a
-	$(libcdd_a_AR) libcdd.a $(libcdd_a_OBJECTS) $(libcdd_a_LIBADD)
-	$(RANLIB) libcdd.a
+shared: $(libcdd_a_OBJECTS) $(libcdd_a_DEPENDENCIES) 
+	$(CC) $(CFLAGS) -fPIC -shared -Wl,-soname=libcdd.so -o libcdd.so  $(libcdd_a_OBJECTS) $(libcdd_a_LIBADD)
+
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT) core *.core

--- lib-src-gmp/Makefile.in	2007-02-12 13:33:32.000000000 +0200
+++ lib-src-gmp/Makefile.in	2008-01-19 22:07:48.000000000 +0200
@@ -48,6 +48,7 @@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_DATA = @INSTALL_DATA@
 install_sh_DATA = $(install_sh) -c -m 644
+install_so_LIB = $(install_sh) -c -m 755
 install_sh_PROGRAM = $(install_sh) -c
 install_sh_SCRIPT = $(install_sh) -c
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
@@ -77,7 +78,7 @@
 am__include = @am__include@
 am__quote = @am__quote@
 install_sh = @install_sh@
-lib_LIBRARIES = libcddgmp.a
+lib_LIBRARIES = libcddgmp.so
 
 libcddgmp_a_SOURCES = \
 cddcore.c \
@@ -110,7 +111,7 @@
 subdir = lib-src-gmp
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_CLEAN_FILES =
-LIBRARIES = $(lib_LIBRARIES)
+LIBRARIES = shared
 
 libcddgmp_a_AR = $(AR) cru
 libcddgmp_a_LIBADD =
@@ -156,7 +157,7 @@
 	cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@ $(am__depfiles_maybe)
 
 AR = ar
-libLIBRARIES_INSTALL = $(INSTALL_DATA)
+libLIBRARIES_INSTALL = $(install_so_LIB)
 install-libLIBRARIES: $(lib_LIBRARIES)
 	@$(NORMAL_INSTALL)
 	$(mkinstalldirs) $(DESTDIR)$(libdir)
@@ -167,14 +168,6 @@
 	    $(libLIBRARIES_INSTALL) $$p $(DESTDIR)$(libdir)/$$f; \
 	  else :; fi; \
 	done
-	@$(POST_INSTALL)
-	@list='$(lib_LIBRARIES)'; for p in $$list; do \
-	  if test -f $$p; then \
-	    p="`echo $$p | sed -e 's|^.*/||'`"; \
-	    echo " $(RANLIB) $(DESTDIR)$(libdir)/$$p"; \
-	    $(RANLIB) $(DESTDIR)$(libdir)/$$p; \
-	  else :; fi; \
-	done
 
 uninstall-libLIBRARIES:
 	@$(NORMAL_UNINSTALL)
@@ -186,10 +179,8 @@
 
 clean-libLIBRARIES:
 	-test -z "$(lib_LIBRARIES)" || rm -f $(lib_LIBRARIES)
-libcddgmp.a: $(libcddgmp_a_OBJECTS) $(libcddgmp_a_DEPENDENCIES) 
-	-rm -f libcddgmp.a
-	$(libcddgmp_a_AR) libcddgmp.a $(libcddgmp_a_OBJECTS) $(libcddgmp_a_LIBADD)
-	$(RANLIB) libcddgmp.a
+shared: $(libcddgmp_a_OBJECTS) $(libcddgmp_a_DEPENDENCIES)
+	$(CC) $(CFLAGS) -fPIC -shared -Wl,-soname=libcddgmp.so -o libcddgmp.so  $(libcddgmp_a_OBJECTS) $(libcddgmp_a_LIBADD)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT) core *.core
