diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kdeadmin-4.2.2-orig/CMakeLists.txt kdeadmin-4.2.2/CMakeLists.txt
--- kdeadmin-4.2.2-orig/CMakeLists.txt	2009-04-20 11:46:27.000000000 +0300
+++ kdeadmin-4.2.2/CMakeLists.txt	2009-04-20 11:47:10.000000000 +0300
@@ -13,18 +13,12 @@
 #macro_optional_add_subdirectory(kdat)
 macro_optional_add_subdirectory(strigi-analyzer)
 if ( UNIX )
-  macro_optional_add_subdirectory(kuser)
   macro_optional_add_subdirectory(kcron)
   macro_optional_add_subdirectory(ksystemlog)
   macro_optional_add_subdirectory(system-config-printer-kde)
 endif( UNIX )
 
 #macro_optional_add_subdirectory(ksysv)
-if(Q_WS_X11)
-   macro_optional_add_subdirectory(kpackage)
-   macro_optional_add_subdirectory(lilo-config)
-   macro_optional_add_subdirectory(knetworkconf)
-endif(Q_WS_X11)
 MESSAGE(STATUS "readd kdat-ksysv when porting")
 
 macro_optional_add_subdirectory(doc)
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kdeadmin-4.2.2-orig/doc/CMakeLists.txt kdeadmin-4.2.2/doc/CMakeLists.txt
--- kdeadmin-4.2.2-orig/doc/CMakeLists.txt	2009-04-20 11:46:28.000000000 +0300
+++ kdeadmin-4.2.2/doc/CMakeLists.txt	2009-04-20 11:48:17.000000000 +0300
@@ -5,16 +5,13 @@
 FIND_PROGRAM(LILO_EXECUTABLE NAMES lilo LILO  PATHS /sbin /usr/sbin /usr/local/sbin /opt/lilo  )
 
 if ( UNIX )
-  add_subdirectory(kuser)
   add_subdirectory(kcron)
   add_subdirectory(ksystemlog)
 endif( UNIX )
 
 if(Q_WS_X11)
-   add_subdirectory(kpackage)
    if(LILO_EXECUTABLE)
       add_subdirectory(lilo-config)
    endif(LILO_EXECUTABLE)
-   add_subdirectory(kcontrol)
 endif(Q_WS_X11)
 
