Index: kdelibs-4.1.85/plasma/delegate.cpp
===================================================================
--- kdelibs-4.1.85.orig/plasma/delegate.cpp	2008-12-04 09:35:59.000000000 +0100
+++ kdelibs-4.1.85/plasma/delegate.cpp	2008-12-11 18:22:51.000000000 +0100
@@ -289,7 +289,7 @@
         // which one can scan the list visually and the aesthetic qualities of the listing.
         painter->setPen(QPen(KColorScheme(QPalette::Active).foreground(KColorScheme::InactiveText), 1));
         painter->setFont(subTitleFont);
-        painter->drawText(subTitleRect, Qt::AlignLeft|Qt::AlignVCenter, "  " + subTitleText);
+        painter->drawText(subTitleRect, Qt::AlignLeft|Qt::AlignVCenter, subTitleText);
     }
 
     painter->restore();
