--- hcid/hcid.conf
+++ hcid/hcid.conf
@@ -29,16 +29,29 @@
 	# Local device name
 	#   %d - device id
 	#   %h - host name
-	name "BlueZ (%d)";
+	name "BlueZ %h (%d)";
 
 	# Local device class
+	# e.g.
+	#  0xsss100 = Computer
+	#  0xsss104 = Computer Desktop
+	#  0xsss108 = Computer Server
+	#  0xsss10c = Computer Laptop
+	# The 'sss' above defines the service-class (not quite, only the
+	# first 11 bits, the next 11 define the device-class, than 2 format bits.)
+	# See https://www.bluetooth.org/foundry/assignnumb/document/baseband
+	# for more information.
+	# 0x100bbb stands for "Object Transfer  (v-Inbox, v-Folder, ...)"
+	# 0x020bbb stands for "Networking (LAN, Ad hoc, ...)"
 	class 0x3e0100;
 
 	# Default packet type
 	#pkt_type DH1,DM1,HV1;
 
 	# Inquiry and Page scan
-	iscan enable; pscan enable;
+	# valid parameters: enable | disable
+	iscan enable;
+	pscan enable;
 
 	# Default link mode
 	#   none   - no specific policy 
