diff -Nur glob2-0.9.3-old/src/FertilityCalculatorThreadMessage.cpp glob2-0.9.3/src/FertilityCalculatorThreadMessage.cpp
--- glob2-0.9.3-old/src/FertilityCalculatorThreadMessage.cpp	2008-05-17 14:28:06.000000000 +0300
+++ glob2-0.9.3/src/FertilityCalculatorThreadMessage.cpp	2008-05-17 14:28:53.000000000 +0300
@@ -19,6 +19,7 @@
 #include "FertilityCalculatorThreadMessage.h"
 
 #include <sstream>
+#include <typeinfo>
 
 FCTUpdateCompletionPercent::FCTUpdateCompletionPercent(float percent)
 	: percent(percent)
