diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kdeadmin-4.2.0-orig/system-config-printer-kde/system-config-printer-kde.py kdeadmin-4.2.0/system-config-printer-kde/system-config-printer-kde.py
--- kdeadmin-4.2.0-orig/system-config-printer-kde/system-config-printer-kde.py	2009-02-18 11:47:45.000000000 +0200
+++ kdeadmin-4.2.0/system-config-printer-kde/system-config-printer-kde.py	2009-02-18 12:24:55.000000000 +0200
@@ -1310,11 +1310,12 @@
         debugprint ("update printer properties")
         printer = self.printer
         self.lblPMakeModel.setText(printer.make_and_model)
-        state = self.printer_states.get (printer.state, i18nc("Printer state", "Unknown"))
-	reason = printer.other_attributes.get ('printer-state-message', '')
-	if len (reason) > 0:
-	    state += ' - ' + reason
+        state = self.printer_states.get (printer.state, i18nc("Printer state", "Unknown"))[:]
+        reason = printer.other_attributes.get ('printer-state-message', '')
+        if len (reason) > 0:
+            state += ' - ' + reason
         self.lblPState.setText(state)
+
         if len (self.changed) == 0:
             debugprint ("no changes yet: full printer properties update")
             # State
