--- build/build.xml	2007-10-22 17:17:31.000000000 +0300
+++ build/build.xml2	2007-12-25 18:58:46.000000000 +0200
@@ -285,7 +285,7 @@
                  optimize="${build.optimize}"
        >
          <classpath>
-            <pathelement location="lib/servlet.jar"/>
+            <pathelement location="/usr/share/java/servlet-api.jar"/>
          </classpath>
          <include name="*.java"/>
          <include name="org/hsqldb/*.java"/>
@@ -554,7 +554,7 @@
 
     <target name="hsqldb" depends="classes14,util,swing"
             description="Build the default hsqldb.jar">
-        <jar jarfile="lib/hsqldb.jar"
+        <jar jarfile="hsqldb.jar"
              basedir="classes"
              compress="true"
          >
@@ -668,7 +668,7 @@
 <!-- ommissions where the calling methods are not actually used -->
     <target name="hsqljdbc" depends="classes14"
              description="Build the hsqljdbc.jar">
-        <jar jarfile="lib/hsqljdbc.jar" compress="true">
+        <jar jarfile="hsqljdbc.jar" compress="true">
          <manifest>
              <attribute name='Specification-Title' value='${hsqldb.title}'/>
              <attribute name='Specification-Version' value='${hsqldb.version}'/>
@@ -684,7 +684,7 @@
 
     <target name="hsqltool" depends="classes14, util"
             description="Build the hsqltool.jar">
-        <jar jarfile="lib/hsqltool.jar">
+        <jar jarfile="hsqltool.jar">
             <manifest>
                 <attribute name='Specification-Title' value='${hsqldb.title}'/>
                 <attribute name='Specification-Version' value='${hsqldb.version}'/>
@@ -733,7 +733,7 @@
 
     <target name="hsqldbutil" depends="util,swing"
             description="Build the utility hsqldbutil.jar">
-        <jar jarfile="lib/hsqldbutil.jar" compress="true">
+        <jar jarfile="hsqldbutil.jar" compress="true">
          <manifest>
              <attribute name='Specification-Title' value='${hsqldb.title}'/>
              <attribute name='Specification-Version' value='${hsqldb.version}'/>
@@ -761,7 +761,7 @@
 		   				${src}/org/hsqldb/util/RCData.java,
 		   				${src}/org/hsqldb/util/SqlTool.java"
            destdir="${basedir}/doc/src"
-           classpath="${basedir}/lib/servlet.jar;${basedir}/lib/junit.jar"
+           classpath="/usr/share/java/servlet-api.jar"
            author="true"
            version="false"
            use="true"
@@ -782,7 +782,7 @@
       <javadoc packagenames="org.hsqldb.*,org.hsqldb.jdbc.*,org.hsqldb.lib.*,org.hsqldb.util.*"
            sourcepath="${src}"
            destdir="${basedir}/doc/src"
-           classpath="${basedir}/lib/servlet.jar;${basedir}/lib/junit.jar"
+           classpath="/usr/share/java/servlet-api.jar"
            author="true"
            version="false"
            use="true"
