--- db/db_base.h.orig	2008-11-11 21:06:09.000000000 +0200
+++ db/db_base.h	2008-11-11 21:06:32.000000000 +0200
@@ -20,6 +20,9 @@
 #ifndef _DB_BASE_
 #define _DB_BASE_
 
+#include <cstdlib>
+#include <string>
+
 #include <mysql.h>
 
 // if SQL columns are not 'not null', you must use these safe_atoi, safe_atof
--- sched/edf_sim.C.orig	2008-11-11 21:07:18.000000000 +0200
+++ sched/edf_sim.C	2008-11-11 21:07:43.000000000 +0200
@@ -17,6 +17,7 @@
 // or write to the Free Software Foundation, Inc.,
 // 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 
+#include <cstdio>
 #include <vector>
 #include <algorithm>
 
--- sched/validator.C.orig	2008-11-11 21:10:03.000000000 +0200
+++ sched/validator.C	2008-11-11 21:10:31.000000000 +0200
@@ -38,6 +38,7 @@
 #include "config.h"
 #include <unistd.h>
 #include <cmath>
+#include <climits>
 #include <vector>
 
 #include "boinc_db.h"
--- sched/sched_locality.C.orig	2008-11-11 21:11:12.000000000 +0200
+++ sched/sched_locality.C	2008-11-11 21:11:42.000000000 +0200
@@ -22,6 +22,8 @@
 #include "config.h"
 #include <stdio.h>
 #include <unistd.h>
+#include <algorithm>
+#include <climits>
 #include <strings.h>
 #include <glob.h>
 #include <sys/stat.h>
--- lib/crypt.C.orig	2008-11-11 21:12:28.000000000 +0200
+++ lib/crypt.C	2008-11-11 21:12:45.000000000 +0200
@@ -26,6 +26,7 @@
 #include <cctype>
 #include <cstdio>
 #include <cstdlib>
+#include <cstring>
 #endif
 
 #include "md5_file.h"
--- lib/parse.h.orig	2008-11-11 21:13:16.000000000 +0200
+++ lib/parse.h	2008-11-11 21:13:27.000000000 +0200
@@ -25,6 +25,7 @@
 #else
 #include <cstdio>
 #include <cstdlib>
+#include <cstring>
 #include <string>
 #include <math.h>
 #ifdef solaris
--- lib/procinfo_unix.C.orig	2008-11-11 21:14:17.000000000 +0200
+++ lib/procinfo_unix.C	2008-11-11 21:24:46.000000000 +0200
@@ -30,6 +30,8 @@
 #include <stdio.h>
 
 #include <ctype.h>
+#include <cstring>
+
 #include <sys/types.h>
 #include <dirent.h>
 
--- client/cs_account.C.orig	2008-11-11 21:26:53.000000000 +0200
+++ client/cs_account.C	2008-11-11 21:27:15.000000000 +0200
@@ -23,6 +23,7 @@
 
 #ifndef _WIN32
 #include "config.h"
+#include <algorithm>
 #include <cstdio>
 #include <cassert>
 #ifdef HAVE_SYS_STAT_H
--- client/client_msgs.C.orig	2008-11-11 21:27:44.000000000 +0200
+++ client/client_msgs.C	2008-11-11 21:28:02.000000000 +0200
@@ -22,6 +22,7 @@
 #else
 #include "config.h"
 #include <cstdarg>
+#include <cstring>
 #include <deque>
 #endif
 
--- sched/edf_sim.h.orig	2008-11-11 21:33:03.000000000 +0200
+++ sched/edf_sim.h	2008-11-11 21:33:21.000000000 +0200
@@ -21,6 +21,7 @@
 #define _EDF_SIM_H
 
 #include <vector>
+#include <cstring>
 
 struct IP_RESULT {
     char name[256];
