--- acx_config.h.old	2006-01-26 05:09:06.000000000 +0200
+++ acx_config.h	2006-01-26 05:10:10.000000000 +0200
@@ -7,7 +7,7 @@
 /* set to 0 if you don't want any debugging code to be compiled in */
 /* set to 1 if you want some debugging */
 /* set to 2 if you want extensive debug log */
-#define ACX_DEBUG 2
+#define ACX_DEBUG 0
 #define ACX_DEFAULT_MSG (L_ASSOC|L_INIT)
 
 /* assume 32bit I/O width
