Index: kdelibs/admin/acinclude.m4.in
===================================================================
--- kdelibs.orig/admin/acinclude.m4.in
+++ kdelibs/admin/acinclude.m4.in
@@ -1437,6 +1437,7 @@ if test "x$kde_use_qt_mt" = "xyes"; then
     kde_use_qt_mt=no
   fi
 fi
+KDE_MT_LDFLAGS="$KDE_MT_LDFLAGS -L/usr/qt/3/lib"
 AC_SUBST(KDE_MT_LDFLAGS)
 AC_SUBST(KDE_MT_LIBS)
 
@@ -3849,6 +3850,7 @@ if test -z "$KDE_RPATH" && test "$USE_RP
     KDE_RPATH="$KDE_RPATH \$(KDE_EXTRA_RPATH)"
   fi
 fi
+KDE_RPATH="$KDE_RPATH -L/usr/qt/3/lib"
 AC_SUBST(KDE_EXTRA_RPATH)
 AC_SUBST(KDE_RPATH)
 AC_SUBST(X_RPATH)
