Index: kdelibs/kdeprint/cups/cupsdconf2/cupsd.conf.template
===================================================================
--- kdelibs.orig/kdeprint/cups/cupsdconf2/cupsd.conf.template
+++ kdelibs/kdeprint/cups/cupsdconf2/cupsd.conf.template
@@ -590,7 +590,7 @@ $$
 # <p>
 # The group name for "System" (printer administration)
 # access.  The default varies depending on the operating system, but
-# will be <b>sys</b>, <b>system</b>, or <b>root</b> (checked for in that order).</p>
+# will be <b>lpadmin</b>, <b>system</b>, or <b>root</b> (checked for in that order).</p>
 # <p>
 # <i>ex</i>: lpadmin</p>
 $$
