Index: kdelibs/kdecore/svgicons/ksvgiconengine.cpp
===================================================================
--- kdelibs.orig/kdecore/svgicons/ksvgiconengine.cpp
+++ kdelibs/kdecore/svgicons/ksvgiconengine.cpp
@@ -538,6 +538,9 @@ KSVGIconEngine::~KSVGIconEngine()
 
 bool KSVGIconEngine::load(int width, int height, const TQString &path)
 {
+    if (path.isEmpty())
+        return false;
+
 	TQDomDocument svgDocument("svg");
 	TQFile file(path);
 
