Index: kdelibs/kio/misc/Makefile.am
===================================================================
--- kdelibs.orig/kio/misc/Makefile.am
+++ kdelibs/kio/misc/Makefile.am
@@ -42,7 +42,7 @@ kmailservice_LDFLAGS = $(KDE_RPATH) $(al
 
 ktelnetservice_SOURCES = ktelnetservice.cpp
 ktelnetservice_LDADD = $(LIB_KDEUI)
-ktelnetservice_LDFLAGS = $(KDE_RPATH) $(all_libraries)
+ktelnetservice_LDFLAGS = $(all_libraries)
 
 protocol_DATA = kmailservice.protocol telnet.protocol rlogin.protocol rtsp.protocol ssh.protocol \
 		mms.protocol mmst.protocol mmsu.protocol pnm.protocol rtspt.protocol rtspu.protocol
