Index: kdecore/kglobalsettings.cpp
===================================================================
--- kdecore.orig/kglobalsettings.cpp
+++ kdecore/kglobalsettings.cpp
@@ -129,7 +129,7 @@ KGlobalSettings::TearOffHandle KGlobalSe
     int tearoff;
     bool effectsenabled;
     KConfigGroup g( KGlobal::config(), "KDE" );
-    effectsenabled = g.readBoolEntry( "EffectsEnabled", false);
+    effectsenabled = g.readBoolEntry( "EffectsEnabled", true);
     tearoff = g.readNumEntry("InsertTearOffHandle", KDE_DEFAULT_INSERTTEAROFFHANDLES);
     return effectsenabled ? (TearOffHandle) tearoff : Disable;
 }
@@ -740,8 +740,7 @@ TQRect KGlobalSettings::desktopGeometry(
 bool KGlobalSettings::showIconsOnPushButtons()
 {
     KConfigGroup g( KGlobal::config(), "KDE" );
-    return g.readBoolEntry("ShowIconsOnPushButtons",
-        KDE_DEFAULT_ICON_ON_PUSHBUTTON);
+    return g.readBoolEntry("ShowIconsOnPushButtons", true);
 }
 
 bool KGlobalSettings::showFilePreview(const KURL &url)
