Index: kdelibs/kdeprint/kmfactory.cpp
===================================================================
--- kdelibs.orig/kdeprint/kmfactory.cpp
+++ kdelibs/kdeprint/kmfactory.cpp
@@ -363,7 +363,7 @@ TQString KMFactory::autoDetect()
 			currentPrecedence = plugins[i].detectPrecedence;
 		}
 	}
-	return (pluginIndex == -1 ? TQString::fromLatin1("lpdunix") : plugins[pluginIndex].name);
+	return (pluginIndex == -1 ? TQString::fromLatin1("cups") : plugins[pluginIndex].name);
 }
 
 void KMFactory::slot_pluginChanged(pid_t pid)
