Index: kdelibs/kioslave/http/kcookiejar/kcookiejar.cpp
===================================================================
--- kdelibs.orig/kioslave/http/kcookiejar/kcookiejar.cpp
+++ kdelibs/kioslave/http/kcookiejar/kcookiejar.cpp
@@ -1528,7 +1528,7 @@ void KCookieJar::loadConfig(KConfig *_co
     m_rejectCrossDomainCookies = _config->readBoolEntry( "RejectCrossDomainCookies", true );
     m_autoAcceptSessionCookies = _config->readBoolEntry( "AcceptSessionCookies", true );
     m_ignoreCookieExpirationDate = _config->readBoolEntry( "IgnoreExpirationDate", false );
-    TQString value = _config->readEntry("CookieGlobalAdvice", L1("Ask"));
+    TQString value = _config->readEntry("CookieGlobalAdvice", L1("Accept"));
     m_globalAdvice = strToAdvice(value);
 
     // Reset current domain settings first.
