Index: kdecore/kglobalsettings.cpp
===================================================================
--- kdecore.orig/kglobalsettings.cpp
+++ kdecore/kglobalsettings.cpp
@@ -357,7 +357,7 @@ TQFont KGlobalSettings::generalFont()
         return *_generalFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _generalFont = new TQFont("Sans Serif", 10);
+    _generalFont = new TQFont("DejaVu Sans", 10);
     _generalFont->setPointSize(10);
     _generalFont->setStyleHint(TQFont::SansSerif);
 
@@ -373,7 +373,7 @@ TQFont KGlobalSettings::fixedFont()
         return *_fixedFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _fixedFont = new TQFont("Monospace", 10);
+    _fixedFont = new TQFont("DejaVu Sans Mono", 10);
     _fixedFont->setPointSize(10);
     _fixedFont->setStyleHint(TQFont::TypeWriter);
 
@@ -389,7 +389,7 @@ TQFont KGlobalSettings::toolBarFont()
         return *_toolBarFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _toolBarFont = new TQFont("Sans Serif", 10);
+    _toolBarFont = new TQFont("DejaVu Sans", 10);
     _toolBarFont->setPointSize(10);
     _toolBarFont->setStyleHint(TQFont::SansSerif);
 
@@ -405,7 +405,7 @@ TQFont KGlobalSettings::menuFont()
         return *_menuFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _menuFont = new TQFont("Sans Serif", 10);
+    _menuFont = new TQFont("DejaVu Sans", 10);
     _menuFont->setPointSize(10);
     _menuFont->setStyleHint(TQFont::SansSerif);
 
@@ -421,12 +421,12 @@ TQFont KGlobalSettings::windowTitleFont(
         return *_windowTitleFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _windowTitleFont = new TQFont("Sans Serif", 9, TQFont::Bold);
+    _windowTitleFont = new TQFont("DejaVu Sans", 9, TQFont::Bold);
     _windowTitleFont->setPointSize(10);
     _windowTitleFont->setStyleHint(TQFont::SansSerif);
 
-    KConfigGroup g( KGlobal::config(), "WM" );
-    *_windowTitleFont = g.readFontEntry("activeFont", _windowTitleFont); // inconsistency
+    //KConfigGroup g( KGlobal::config(), "WM" );
+    //*_windowTitleFont = g.readFontEntry("activeFont", _windowTitleFont); // inconsistency
 
     return *_windowTitleFont;
 }
@@ -437,7 +437,7 @@ TQFont KGlobalSettings::taskbarFont()
         return *_taskbarFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _taskbarFont = new TQFont("Sans Serif", 10);
+    _taskbarFont = new TQFont("DejaVu Sans", 10);
     _taskbarFont->setPointSize(10);
     _taskbarFont->setStyleHint(TQFont::SansSerif);
 
