Index: kdelibs/kdeprint/management/networkscanner.cpp
===================================================================
--- kdelibs.orig/kdeprint/management/networkscanner.cpp
+++ kdelibs/kdeprint/management/networkscanner.cpp
@@ -344,6 +344,8 @@ NetworkScannerConfig::NetworkScannerConf
 	mm->setFixedWidth(fontMetrics().width(mm->text())+10);
 
 	mask_ = new TQLineEdit(dummy);
+    // make sure address area is not overly small
+    mask_->setMinimumWidth(fontMetrics().width("192.168.123"));
 	mask_->setAlignment(Qt::AlignRight);
 	port_ = new TQComboBox(true,dummy);
         if ( port_->lineEdit() )
