Index: kio/kio/paste.cpp
===================================================================
--- kio.orig/kio/paste.cpp
+++ kio/kio/paste.cpp
@@ -172,6 +172,7 @@ KIO::CopyJob* KIO::pasteMimeSource( TQMi
   if ( TQTextDrag::canDecode( data ) && TQTextDrag::decode( data, text ) )
   {
       TQTextStream txtStream( ba, IO_WriteOnly );
+      txtStream.setEncoding(TQTextStream::UnicodeUTF8);
       txtStream << text;
   }
   else
