Index: kdelibs/kdeui/kaboutdialog.cpp
===================================================================
--- kdelibs.orig/kdeui/kaboutdialog.cpp
+++ kdelibs/kdeui/kaboutdialog.cpp
@@ -701,8 +701,9 @@ void KAboutContainerBase::setProduct( co
   else
     kdDebug(291) << "no kapp" << endl;
 
-  const TQString msg1 = i18n("%1 %2 (Using KDE %3)").arg(appName).arg(version).
-    arg(TQString::fromLatin1(KDE_VERSION_STRING));
+  const TQString msg1 = i18n("%1 %2 (Using KDE %3 %4)").arg(appName).arg(version).
+    arg(TQString::fromLatin1(KDE_VERSION_STRING)).
+    arg(TQString::fromLatin1(KDE_DISTRIBUTION_TEXT));
   const TQString msg2 = !year.isEmpty() ? i18n("%1 %2, %3").arg('©').arg(year).
     arg(author) : TQString::fromLatin1("");
 
