Index: open-iscsi-2.0-872-rc4-bnx2i/usr/Makefile
===================================================================
--- open-iscsi-2.0-872-rc4-bnx2i.orig/usr/Makefile
+++ open-iscsi-2.0-872-rc4-bnx2i/usr/Makefile
@@ -53,10 +53,10 @@ all: $(PROGRAMS)
 
 iscsid: $(ISCSI_LIB_SRCS) $(IPC_OBJ) $(INITIATOR_SRCS) $(DISCOVERY_SRCS) \
 	iscsid.o session_mgmt.o discoveryd.o
-	$(CC) $(CFLAGS) $^ -o $@ -L../utils/open-isns -lisns -lcrypto
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -L../utils/open-isns -lisns -lcrypto -lslp -o $@
 
 iscsiadm: $(ISCSI_LIB_SRCS) $(DISCOVERY_SRCS) iscsiadm.o session_mgmt.o
-	$(CC) $(CFLAGS) $^ -o $@ -L../utils/open-isns -lisns -lcrypto
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -L../utils/open-isns -lisns -lcrypto -lslp -o $@
 
 iscsistart: $(IPC_OBJ) $(ISCSI_LIB_SRCS) $(INITIATOR_SRCS) $(FW_BOOT_SRCS) \
 		iscsistart.o statics.o
