diff -ur module-init-tools-3.2.2.orig/configure.ac module-init-tools-3.2.2/configure.ac
--- module-init-tools-3.2.2.orig/configure.ac	2005-12-02 01:20:48.000000000 +0200
+++ module-init-tools-3.2.2/configure.ac	2006-11-10 12:11:51.000000000 +0200
@@ -10,7 +10,7 @@
 [  --enable-zlib               Handle gzipped modules],
 [if test "$enableval" = "yes"; then
   AC_DEFINE(CONFIG_USE_ZLIB)
-  zlib_flags="-Wl,-Bstatic -lz -Wl,-Bdynamic"
+  zlib_flags="-lz -Wl,-Bdynamic"
 fi])
 
 AC_PROG_CC
