diff -Nur -x '*.orig' -x '*~' poppler-0.12.2/poppler/PSOutputDev.cc poppler-0.12.2.new/poppler/PSOutputDev.cc
--- poppler-0.12.2/poppler/PSOutputDev.cc	2009-11-13 00:52:35.000000000 +0100
+++ poppler-0.12.2.new/poppler/PSOutputDev.cc	2009-12-17 23:36:22.297158221 +0100
@@ -114,7 +114,6 @@
   "    pop 3 dict begin",
   "      /PageSize exch def",
   "      /ImagingBBox null def",
-  "      /Policies 1 dict dup begin /PageSize 3 def end def",
   "      { /Duplex true def } if",
   "    currentdict end setpagedevice",
   "  } {",
