Index: libavcodec/utils.c
===================================================================
--- libavcodec/utils.c	(revision 12559)
+++ libavcodec/utils.c	(working copy)
@@ -66,7 +66,11 @@
 
     *size= FFMAX(17*min_size/16 + 32, min_size);
 
-    return av_realloc(ptr, *size);
+    ptr= av_realloc(ptr, *size);
+    if(!ptr) //we could set this to the unmodified min_size but this is safer if the user lost the ptr and uses NULL now
+        *size= 0;
+
+    return ptr;
 }
 
 static unsigned int last_static = 0;
