--- mplayer-20071230-old/mencoder.c	2007-12-30 16:06:12.000000000 +0200
+++ mplayer-20071230/mencoder.c	2007-12-31 01:46:30.168794725 +0200
@@ -319,6 +319,8 @@
 static void parse_cfgfiles( m_config_t* conf )
 {
   char *conffile;
+  if (m_config_parse_config_file(conf, MPLAYER_CONFDIR "/mencoder.conf") < 0)
+    mencoder_exit(1,MSGTR_ConfigFileError);
   if ((conffile = get_path("mencoder.conf")) == NULL) {
     mp_msg(MSGT_CPLAYER,MSGL_ERR,MSGTR_GetpathProblem);
   } else {
