Index: etc/codecs.conf
===================================================================
--- etc/codecs.conf	(revision 26271)
+++ etc/codecs.conf	(revision 26272)
@@ -9,6 +9,12 @@
 ;                   VIDEO CODECS
 ;=============================================================================
 
+videocodec ffmimic
+  fourcc ML20
+  out I420
+  driver ffmpeg
+  dll "mimic"
+
 videocodec ffkmvc
   fourcc KMVC
   out BGR8
