diff -Nur mplayer-20060923-old/help/help_mp-tr.h mplayer-20060923/help/help_mp-tr.h
--- mplayer-20060923-old/help/help_mp-tr.h	2006-10-05 14:46:25.000000000 +0300
+++ mplayer-20060923/help/help_mp-tr.h	2006-10-05 14:47:25.000000000 +0300
@@ -106,7 +106,7 @@
 #define MSGTR_Playing "%s oynatılıyor\n"
 #define MSGTR_NoSound "Ses: ses yok!\n"
 #define MSGTR_FPSforced "FPS  %5.3f olarak zorlandı (ftime: %5.3f)\n"
-#define MSGTR_CompiledWithRuntimeDetection "MPlayer runtime CPU detection ile derlendi. - Dikkat, bu özellik hızlı değildir! En iyi performansı alabilmek için, MPlayer'ı -disable-runtime-cpudetection seçeneği ile tekrar derleyiniz.\n"
+#define MSGTR_CompiledWithRuntimeDetection "MPlayer runtime CPU detection ile derlendi.\n"
 #define MSGTR_CompiledWithCPUExtensions "x86 işlemci için eklerle derlendi:"
 #define MSGTR_AvailableVideoOutputDrivers "Kullanılabilir video çıktı sürücüleri:\n"
 #define MSGTR_AvailableAudioOutputDrivers "Kullanılabilir ses çıktı sürücüleri:\n"
