diff -Nur mplayer-20080329-old/libao2/ao_pulse.c mplayer-20080329/libao2/ao_pulse.c
--- mplayer-20080329-old/libao2/ao_pulse.c	2008-03-29 18:27:22.978750413 +0200
+++ mplayer-20080329/libao2/ao_pulse.c	2008-03-29 18:31:30.989047116 +0200
@@ -121,6 +121,8 @@
     {AF_FORMAT_FLOAT_BE, PA_SAMPLE_FLOAT32BE},
     {AF_FORMAT_MU_LAW, PA_SAMPLE_ULAW},
     {AF_FORMAT_A_LAW, PA_SAMPLE_ALAW},
+    {AF_FORMAT_S32_LE, PA_SAMPLE_S32LE},
+    {AF_FORMAT_S32_BE, PA_SAMPLE_S32BE},
     {AF_FORMAT_UNKNOWN, 0}
 };
 
