Index: libmenu/menu_filesel.c
===================================================================
--- libmenu/menu_filesel.c	(revision 25660)
+++ libmenu/menu_filesel.c	(revision 25659)
@@ -427,7 +427,6 @@
   
   getcwd(wd,PATH_MAX);
   if (!path || path[0] == '\0') {
-#if 0
     char *slash = NULL;
     if (filename && !strstr(filename, "://") && (path=realpath(filename, b))) {
       slash = strrchr(path, '/');
@@ -440,7 +439,6 @@
     if (slash)
       slash[1] = '\0';
     else
-#endif
       path = wd;
   }
   if (path[0] != '/') {
