--- cdecl-2.5/cdecl.c.orig	2002-06-11 17:54:00.000000000 +0200
+++ cdecl-2.5/cdecl.c	2002-06-11 17:54:32.000000000 +0200
@@ -63,6 +63,7 @@
 #include <stdio.h>
 #include <ctype.h>
 #if __STDC__ || defined(DOS)
+#include <errno.h>
 # include <stdlib.h>
 # include <stddef.h>
 # include <string.h>
@@ -124,7 +125,7 @@
 
 #if __STDC__
   char *ds(char *), *cat(char *, ...), *visible(int);
-  int getopt(int,char **,char *);
+/*  int getopt(int,char **,char *); */
   int main(int, char **);
   int yywrap(void);
   int dostdin(void);
