diff -ur kvm-68.orig/kernel/Makefile kvm-68/kernel/Makefile
--- kvm-68.orig/kernel/Makefile	2008-05-02 14:20:13.000000000 +0300
+++ kvm-68/kernel/Makefile	2008-05-04 19:36:42.000000000 +0300
@@ -76,11 +76,6 @@
 install:
 	mkdir -p $(DESTDIR)/$(INSTALLDIR)
 	cp *.ko $(DESTDIR)/$(INSTALLDIR)
-	for i in $(ORIGMODDIR)/drivers/kvm/*.ko \
-		 $(ORIGMODDIR)/arch/$(ARCH_DIR)/kvm/*.ko; do \
-		if [ -f "$$i" ]; then mv "$$i" "$$i.orig"; fi; \
-	done
-	/sbin/depmod -a
 
 tmpspec = .tmp.kvm-kmod.spec
 
