--- configure	2008-01-25 18:05:38.000000000 +0200
+++ configure	2008-01-26 03:20:32.000000000 +0200
@@ -12245,7 +12245,7 @@
   case "$LIBEXT" in
     dll) ;;
     dylib) ac_cv_lib_soname_ncurses=`otool -L conftest$ac_exeext | grep "libncurses\\.[0-9A-Za-z.]*dylib" | sed -e "s/^.*\/\(libncurses\.[0-9A-Za-z.]*dylib\).*$/\1/"';2,$d'` ;;
-    *) ac_cv_lib_soname_ncurses=`$ac_cv_path_LDD conftest$ac_exeext | grep "libncurses\\.$LIBEXT" | sed -e "s/^.*\(libncurses\.$LIBEXT[^	 ]*\).*$/\1/"';2,$d'` ;;
+    *) ac_cv_lib_soname_ncurses=`$ac_cv_path_LDD conftest$ac_exeext | grep "libncursesw\\.$LIBEXT" | sed -e "s/^.*\(libncursesw\.$LIBEXT[^	 ]*\).*$/\1/"';2,$d'` ;;
   esac
 else
   echo "$as_me: failed program was:" >&5
