diff -Nur warsow_0.3_sdk-old/source/qcommon/files.c warsow_0.3_sdk/source/qcommon/files.c
--- warsow_0.3_sdk-old/source/qcommon/files.c	2007-05-21 14:12:19.212865664 +0300
+++ warsow_0.3_sdk/source/qcommon/files.c	2007-05-21 14:19:54.381669489 +0300
@@ -2977,9 +2977,9 @@
 	// set basepaths
 	//
 	fs_cdpath = Cvar_Get( "fs_cdpath", "", CVAR_NOSET );
-	fs_basepath = Cvar_Get( "fs_basepath", ".", CVAR_NOSET );
+	fs_basepath = Cvar_Get( "fs_basepath", "/usr/share/warsow", CVAR_NOSET );
 	if( Sys_FS_GetHomeDirectory() != NULL )
-		fs_usehomedir = Cvar_Get( "fs_usehomedir", "0", CVAR_NOSET );
+		fs_usehomedir = Cvar_Get( "fs_usehomedir", "1", CVAR_NOSET );
 
 	if( fs_cdpath->string[0] )
 		FS_AddBasePath( fs_cdpath->string );
