diff -Nur warsow_0.3_sdk-old/source/unix/unix_input.c warsow_0.3_sdk/source/unix/unix_input.c
--- warsow_0.3_sdk-old/source/unix/unix_input.c	2007-05-21 22:00:30.879751439 +0300
+++ warsow_0.3_sdk/source/unix/unix_input.c	2007-05-21 22:01:40.925102926 +0300
@@ -604,10 +604,8 @@
 					}
 					else
 					{
-						mx += ((int)event.xmotion.x - mwx);
-						my += ((int)event.xmotion.y - mwy);
-						mwx = event.xmotion.x;
-						mwy = event.xmotion.y;
+						mx += ( ( int )event.xmotion.x - mwx );
+						my += ( ( int )event.xmotion.y - mwy );
 
 						if (mx || my)
 							dowarp = qtrue;
