Index: qt-qt/mkspecs/features/qttest_p4.prf
===================================================================
--- qt-qt.orig/mkspecs/features/qttest_p4.prf
+++ qt-qt/mkspecs/features/qttest_p4.prf
@@ -17,7 +17,7 @@ symbian:{
 !contains(TARGET, ^tst_.*):TARGET = $$join(TARGET,,"tst_")
 
 isEmpty(target.path) {
-    target.path += $$[QT_INSTALL_PREFIX]/tests/qt4/$${TARGET}
+    target.path += $$[QT_INSTALL_LIBS]/qt4/tests/$${TARGET}
 }
 
 INSTALLS += target
