Index: qt-qt/mkspecs/common/linux.conf
===================================================================
--- qt-qt.orig/mkspecs/common/linux.conf
+++ qt-qt/mkspecs/common/linux.conf
@@ -57,7 +57,7 @@ QMAKE_COPY_DIR        = $(COPY) -r
 QMAKE_MOVE            = mv -f
 QMAKE_DEL_FILE        = rm -f
 QMAKE_DEL_DIR         = rmdir
-QMAKE_STRIP           = strip
+QMAKE_STRIP           =
 QMAKE_STRIPFLAGS_LIB += --strip-unneeded
 QMAKE_CHK_DIR_EXISTS  = test -d
 QMAKE_MKDIR           = mkdir -p
Index: qt-qt/mkspecs/common/g++-base.conf
===================================================================
--- qt-qt.orig/mkspecs/common/g++-base.conf
+++ qt-qt/mkspecs/common/g++-base.conf
@@ -8,14 +8,14 @@
 # you can use the manual test in tests/manual/mkspecs.
 #
 
-QMAKE_CC = gcc
+QMAKE_CC = PARDUS_CC
 
 QMAKE_LINK_C       = $$QMAKE_CC
 QMAKE_LINK_C_SHLIB = $$QMAKE_CC
 
-QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO += -O2 -g
+QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO += PARDUS_CFLAGS
 
-QMAKE_CXX = g++
+QMAKE_CXX = PARDUS_CXX
 
 QMAKE_LINK       = $$QMAKE_CXX
 QMAKE_LINK_SHLIB = $$QMAKE_CXX
Index: qt-qt/mkspecs/common/g++-unix.conf
===================================================================
--- qt-qt.orig/mkspecs/common/g++-unix.conf
+++ qt-qt/mkspecs/common/g++-unix.conf
@@ -10,5 +10,5 @@
 
 include(g++-base.conf)
 
-QMAKE_LFLAGS_RELEASE += -Wl,-O1
+QMAKE_LFLAGS_RELEASE += PARDUS_LDFLAGS
 QMAKE_LFLAGS_NOUNDEF += -Wl,--no-undefined
