--- configure	2006-07-22 20:23:34.000000000 +0300
+++ configure	2006-10-21 13:56:20.000000000 +0300
@@ -478,7 +478,7 @@
 `$sdl_config --static-libs | grep \\\-laa > /dev/null` && aa="yes"
 sdl_static_libs=`$sdl_config --static-libs`
 if [ "$aa" = "yes" ] ; then
-  sdl_static_libs="$sdl_static_libs `aalib-config --static-libs`"
+  sdl_static_libs="$sdl_static_libs `aalib-config --libs`"
 fi
 
 if $cc -o $TMPE `$sdl_config --cflags 2> /dev/null` $TMPC $sdl_static_libs 2> /dev/null; then
