--- gst/gstpad.c	2008/10/10 15:27:37	1.584
+++ gst/gstpad.c	2008/10/20 13:32:07	1.585
@@ -2497,11 +2497,11 @@
 {
   gboolean res;
 
-  /* See if pad accepts the caps */
-  if (!gst_pad_accept_caps (pad, caps))
-    goto not_accepted;
-
   if (dosetcaps) {
+    /* See if pad accepts the caps */
+    if (!gst_pad_accept_caps (pad, caps))
+      goto not_accepted;
+
     res = gst_pad_set_caps (pad, caps);
   } else {
     res = TRUE;

