--- init.c.orig	2007-06-03 23:04:16.000000000 +0300
+++ init.c	2007-06-04 00:24:27.000000000 +0300
@@ -21,7 +21,7 @@
 #define	CFG_VERSION 7	// change this when critical changes to el.cfg are made that will break it
 
 #ifndef DATA_DIR
- #define DATA_DIR "./"
+ #define DATA_DIR "/usr/share/eternal-lands"
 #endif
 
 int ini_file_size=0;
@@ -57,7 +57,7 @@
 int item_window_on_drop=1;
 int compass_direction=1;
 int buddy_log_notice=1;
-char configdir[256]="./";
+char configdir[256]="/usr/share/eternal-lands";
 char datadir[256]=DATA_DIR;
 
 char lang[10]={"en"};
