--- el.ini.orig	2007-06-04 22:12:53.000000000 +0300
+++ el.ini	2007-06-07 01:51:40.000000000 +0300
@@ -32,7 +32,7 @@
 if it is windowed, it will use the desktop BPP
 
 #video_mode = 6
-#full_screen = 0
+#full_screen = 1
 
 The no_adjust_shadows directive, if set on 1, tells the engine not to disable the shadows if
 the frame rate is too low.
@@ -199,7 +199,7 @@
 Default browser is IE, but you can easely change it. 
 
 Note that, because of Micro$oft's infinite wisdom, invoking a command is done differently
-in Win9* and in Windows NT ® ©, so you have to manually select how to invoke your browser.
+in Win9* and in Windows NT ï¿½ ï¿½, so you have to manually select how to invoke your browser.
 Be sure to have only ONE selected browser, otherwise the current browser thingy is likely not 
 to work.
 
@@ -227,7 +227,7 @@
 This option is currently only available on linux, and allows you to 
 specify a directory other than the current directory to load data from.
 For Gentoo users, the ebuild installs in /usr/share/games/eternal-lands.
-#data_dir = /usr/local/games/el/
+#data_dir = /usr/share/eternal-lands/
 
 
 #render_fog = 0
@@ -294,4 +294,4 @@
 
 #windows_on_top = 0
 
-#browser = ""
+#browser = "firefox"
