--- qemulator.py.orig	2008-03-24 01:02:28.000000000 +0200
+++ usr/local/lib/qemulator/qemulator.py	2008-03-24 01:01:42.000000000 +0200
@@ -5773,6 +5773,7 @@
     settings = {}
     settings["cdromdrives"] = [("CD-rom", "/dev/cdrom"),]
     settings["qemuapps"] = [("x86", "qemu"), 
+    ("kvm", "qemu-kvm"),
     ("x86_64", "qemu-system-x86_64"),
     ("ppc", "qemu-system-ppc"),
     ("arm", "qemu-system-arm"), 
