--- configure.in~	2006-08-17 23:29:51.000000000 +0300
+++ configure.in	2006-09-18 14:23:39.000000000 +0300
@@ -765,14 +765,14 @@
 if test "$with_colamd" = yes; then
   with_colamd=no
   AC_CHECK_HEADERS([ufsparse/colamd.h colamd/colamd.h colamd.h], [
-    AC_CHECK_LIB(colamd, colamd, [COLAMD_LIBS="-lcolamd"; with_colamd=yes])
+    AC_CHECK_LIB(colamd, colamd, [COLAMD_LIBS="-lcolamd -lcamd"; with_colamd=yes])
     if test "$with_colamd" = yes; then
       AC_DEFINE(HAVE_COLAMD, 1, [Define if the COLAMD library is used.])
       TEXINFO_COLAMD="@set HAVE_COLAMD"
       warn_colamd=
     fi
     break])
-fi 
+fi
 
 AC_SUBST(TEXINFO_COLAMD)
 
