Index: kgraphviewer-2.0.2-kde4.2.4/src/kgrapheditor.desktop
===================================================================
--- kgraphviewer-2.0.2-kde4.2.4.orig/src/kgrapheditor.desktop
+++ kgraphviewer-2.0.2-kde4.2.4/src/kgrapheditor.desktop
@@ -9,10 +9,12 @@ Name[pl]=KGraphEditor
 Name[tr]=Kgrapheditor
 Name[x-test]=xxkgrapheditorxx
 Exec=kgrapheditor %U
-Icon=kgrapheditor
+Icon=kgraphviewer
 Type=Application
 Categories=Qt;KDE;Graphics;
 X-DocPath=kgrapheditor/kgrapheditor.html
+GenericName=Graph Editor
+GenericName[tr]=Graf Düzenleyici
 Comment=A Graphviz dot graph editor for KDE
 Comment[cs]=Editor bodových grafů Graphviz pro KDE
 Comment[da]=Et program til redigering af Graphviz dot til KDE
Index: kgraphviewer-2.0.2-kde4.2.4/src/kgraphviewer.desktop
===================================================================
--- kgraphviewer-2.0.2-kde4.2.4.orig/src/kgraphviewer.desktop
+++ kgraphviewer-2.0.2-kde4.2.4/src/kgraphviewer.desktop
@@ -13,6 +13,8 @@ Icon=kgraphviewer
 Type=Application
 Categories=Qt;KDE;Graphics;
 X-DocPath=kgraphviewer/index.html
+GenericName=Graph Viewer
+GenericName[tr]=Graf Gösterici
 Comment=A Graphviz dot graph viewer for KDE
 Comment[bg]=Програма за преглед на Graphviz-графики в KDE
 Comment[cs]=Prohlížeč bodových grafů Graphviz pro KDE
