Index: man-1.6f/configure
===================================================================
--- man-1.6f.orig/configure
+++ man-1.6f/configure
@@ -26,7 +26,7 @@ LANGOPT="+lang {$LANGUAGES|all|none}"
 # (Indeed, -r may cause the terminal to get into funny states.
 # Very inconvenient. For viewing pages in strange locales, set LC_*.)
 #
-DEFAULTLESSOPT="-is"
+DEFAULTLESSOPT="-isR"
 #
 # Note that not creating any cat directories (/var/cache/man or so)
 # and not making man suid or sgid is recommended.
@@ -362,7 +362,7 @@ rm -f conftest
 #   for i in /bin:/usr/bin:$PATH
 # does not work everywhere.)
 # Some people might prefer having $PATH first in PREPATH
-PREPATH=/bin:/usr/bin:/usr/ucb:/usr/local/bin:$PATH
+PREPATH="/sbin /bin /usr/sbin /usr/bin"
 IFS0="$IFS"
 IFS=":$IFS"
 for i in $PREPATH
Index: man-1.6f/src/man.conf.in
===================================================================
--- man-1.6f.orig/src/man.conf.in
+++ man-1.6f/src/man.conf.in
@@ -36,11 +36,11 @@
 #
 # Every automatically generated MANPATH includes these fields
 #
-MANPATH	/usr/man
 MANPATH	/usr/share/man
-MANPATH	/usr/local/man
 MANPATH	/usr/local/share/man
 MANPATH	/usr/X11R6/man
+MANPATH	/usr/man
+MANPATH	/usr/local/man
 #
 # Uncomment if you want to include one of these by default
 #
