diff -Naur eina_20090427.orig/m4/efl_pthread.m4 eina_20090427/m4/efl_pthread.m4
--- eina_20090427.orig/m4/efl_pthread.m4	2009-04-27 04:46:21.000000000 +0300
+++ eina_20090427/m4/efl_pthread.m4	2009-04-27 04:47:37.000000000 +0300
@@ -33,7 +33,7 @@
 if test "x${_efl_enable_pthread}" = "xyes" ; then
 
    SAVE_CFLAGS=${CFLAGS}
-   CFLAGS="-pthread"
+   CFLAGS=""
    AC_LANG_PUSH([C])
 
    AC_COMPILE_IFELSE(
@@ -44,8 +44,8 @@
       ],
       [
        _efl_have_pthread="yes"
-       EFL_PTHREAD_CFLAGS="-pthread"
-       EFL_PTHREAD_LIBS="-pthread"
+       EFL_PTHREAD_CFLAGS=""
+       EFL_PTHREAD_LIBS="-lpthread"
        ]
    )
 
