diff -ur smb4k-0.7.5.orig/smb4k/smb4k.desktop smb4k-0.7.5/smb4k/smb4k.desktop
--- smb4k-0.7.5.orig/smb4k/smb4k.desktop	2006-05-24 08:56:15.000000000 +0300
+++ smb4k-0.7.5/smb4k/smb4k.desktop	2006-12-10 17:27:51.000000000 +0200
@@ -22,7 +22,7 @@
 GenericName[ru]=The SMB/CIFS Share Browser
 GenericName[sk]=Prehliadač SMB/CIFS zdieľaní
 GenericName[sv]=SMB Katalog programmed
-GenericName[tr]=The SMB/CIFS Share Browser
+GenericName[tr]=SMB/CIFS Paylaşım Tarayıcısı
 GenericName[uk]=Навігатор спільних ресурсів SMB/CIFS
 GenericName[zh_CN]=SMB 共享浏览器
 #
@@ -70,7 +70,7 @@
 Comment[ru]=The SMB/CIFS Share Browser
 Comment[sk]=Prehliadač SMB/CIFS zdieľaní
 Comment[sv]=SMB Katalog programmed
-Comment[tr]=The SMB/CIFS Share Browser
+Comment[tr]=SMB/CIFS Paylaşım Tarayıcısı
 Comment[uk]=Навігатор спільних ресурсів SMB/CIFS
 Comment[zh_CN]=SMB 共享浏览器
 Exec=smb4k 
