diff -ur basket-0.6.0.orig/src/basket.desktop basket-0.6.0/src/basket.desktop
--- basket-0.6.0.orig/src/basket.desktop	2006-10-16 23:25:06.000000000 +0300
+++ basket-0.6.0/src/basket.desktop	2006-12-10 17:34:27.000000000 +0200
@@ -7,6 +7,7 @@
 Name[es]=Notas de BasKet
 Name[pt]=Bloco de Notas BasKet
 Name[ru]=Альбом заметок BasKet
+Name[tr]=BasKet
 Name[xx]=xxBasKetxx
 Exec=basket
 Icon=basket
@@ -19,3 +20,4 @@
 Comment[pt]=Cuidando das suas ideias.
 Comment[ru]=Заботимся о ваших идеях.
 Comment[xx]=xxTaking care of your ideas.xx
+GenericName[tr]=Not Tutucu
