--- quassel-0.4.2_20090529/src/common/aliasmanager.cpp.orig	2009-05-29 11:35:25.000000000 +0300
+++ quassel-0.4.2_20090529/src/common/aliasmanager.cpp	2009-05-29 11:41:27.000000000 +0300
@@ -92,13 +92,17 @@
           << Alias("chanserv",  "/msg chanserv $0")
           << Alias("hs", "/msg hostserv $0")
           << Alias("hostserv", "/msg hostserv $0")
-          << Alias("back", "/quote away");
-
-#ifdef Q_OS_LINUX
-  // let's add aliases for scripts that only run on linux
-  aliases << Alias("inxi", "/exec inxi $0")
-          << Alias("sysinfo", "/exec inxi -d");
-#endif
+          << Alias("back", "/quote away")
+          << Alias("inxi", "/exec inxi $0")
+          << Alias("sysinfo", "/exec inxi -d")
+          << Alias("amarok", "/exec mpris amarok")
+          << Alias("bug", "/exec bug $0");
+
+//#ifdef Q_OS_LINUX
+  //// let's add aliases for scripts that only run on linux
+  //aliases << Alias("inxi", "/exec inxi $0")
+          //<< Alias("sysinfo", "/exec inxi -d");
+//#endif
 
   return aliases;
 }
