diff -ur konversation-1.0.1.orig/konversation/src/config/konversation.kcfg konversation-1.0.1/konversation/src/config/konversation.kcfg
--- konversation-1.0.1.orig/konversation/src/config/konversation.kcfg	2007-02-14 15:51:41.828796000 +0200
+++ konversation-1.0.1/konversation/src/config/konversation.kcfg	2007-02-14 15:56:13.574339880 +0200
@@ -670,7 +670,7 @@
   </group>
   <group name="Path Settings">
     <entry key="LogfilePath" type="Path">
-      <default code="true">KUser(KUser::UseRealUserID).homeDir()+"/logs"</default>
+      <default code="true">KUser(KUser::UseRealUserID).homeDir()+"/.kde/share/apps/konversation/logs"</default>
       <label></label>
       <whatsthis></whatsthis>
     </entry>
