--- src/3D/FontManager.cxx	2008-01-24 04:43:36.000000000 +0200
+++ src/3D/FontManager.cxx	2008-01-24 04:42:53.000000000 +0200
@@ -468,7 +468,7 @@
   float scale = size / (float)pFont->getSize();
 
   // don't include ansi codes in the length, but allow outside funcs to skip this step
-  const std::string &stripped = alreadyStripped ? text : stripAnsiCodes(text);
+  const std::string stripped = alreadyStripped ? text : stripAnsiCodes(text);
 
   return pFont->getStrLength(scale, stripped.c_str(), (int)stripped.size());
 }
