diff -ur kvm-30.orig/kernel/Makefile kvm-30/kernel/Makefile
--- kvm-30.orig/kernel/Makefile	2007-07-19 15:26:20.000000000 +0300
+++ kvm-30/kernel/Makefile	2007-07-19 16:33:11.000000000 +0300
@@ -41,10 +41,6 @@
 install:
 	mkdir -p $(DESTDIR)/$(INSTALLDIR)
 	cp *.ko $(DESTDIR)/$(INSTALLDIR)
-	for i in $(ORIGMODDIR)/drivers/kvm/*.ko; do \
-		if [ -f "$$i" ]; then mv "$$i" "$$i.orig"; fi; \
-	done
-	/sbin/depmod -a
 
 tmpspec = .tmp.kvm-kmod.spec
 RPMDIR = $$(pwd)/../RPMS
