Index: osm-gps-map-0.7.3/Makefile.am
===================================================================
--- osm-gps-map-0.7.3.orig/Makefile.am
+++ osm-gps-map-0.7.3/Makefile.am
@@ -7,7 +7,7 @@ DISTCHECK_CONFIGURE_FLAGS =
 
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
-osm_gps_mapdocdir = ${prefix}/doc/osm-gps-map
+osm_gps_mapdocdir = ${datadir}/doc/osm-gps-map
 osm_gps_mapdoc_DATA =                           \
 	README                                      \
 	COPYING                                     \
