diff -Nur cdecl-2.5-old/cdecl.c cdecl-2.5/cdecl.c
--- cdecl-2.5-old/cdecl.c	2007-07-14 00:49:21.609538945 +0300
+++ cdecl-2.5/cdecl.c	2007-07-14 01:00:44.438733129 +0300
@@ -399,7 +399,7 @@
 {
   char **matches = NULL;
 
-  if (start == 0) matches = completion_matches(text, command_completion);
+  if (start == 0) matches = rl_completion_matches(text, (rl_compentry_func_t *)  command_completion);
 
   return matches;
 }
@@ -1252,7 +1252,7 @@
 #ifdef USE_READLINE
     /* install completion handlers */
     rl_attempted_completion_function = (CPPFunction *)attempt_completion;
-    rl_completion_entry_function = (Function *)keyword_completion;
+    rl_completion_entry_function = (rl_compentry_func_t *)keyword_completion;
 #endif
 
     setprogname(argv[0]);
