--- cmake/kvirc4.desktop.cmake.orig	2009-12-02 18:06:06.000000000 +0200
+++ cmake/kvirc4.desktop.cmake	2009-12-02 18:07:14.000000000 +0200
@@ -2,8 +2,10 @@
 Name=KVIrc 4
 GenericName=IRC Client
 GenericName[pl]=Klient IRC
+GenericName[tr]=IRC İstemcisi
 Comment=Connect to Internet Relay Chat
 Comment[pl]=Połącz się z Internet Relay Chat (IRC)
+Comment[tr]=IRC'e bağlanın
 Exec=${KVIRC_BINARYNAME} -m %u
 Icon=kvirc
 Type=Application
