--- src/tools/asm/cmdlib.c	2005-10-29 04:53:09.000000000 +0300
+++ src/tools/asm/cmdlib.c	2007-03-09 02:04:32.000000000 +0200
@@ -312,7 +312,7 @@
 char *ExpandPath (const char *path)
 {
 	static char full[1024];
-	if (!qdir)
+	if (!strlen(qdir))
 		Error ("ExpandPath called without qdir set");
 	if (path[0] == '/' || path[0] == '\\' || path[1] == ':') {
 		strcpy( full, path );
@@ -325,7 +325,7 @@
 char *ExpandGamePath (const char *path)
 {
 	static char full[1024];
-	if (!qdir)
+	if (!strlen(qdir))
 		Error ("ExpandGamePath called without qdir set");
 	if (path[0] == '/' || path[0] == '\\' || path[1] == ':') {
 		strcpy( full, path );
