diff -Nur RutilTv0.16-old/configure.sh RutilTv0.16/configure.sh
--- RutilTv0.16-old/configure.sh	2007-12-01 22:05:00.627993402 +0200
+++ RutilTv0.16/configure.sh	2007-12-01 22:06:29.902421626 +0200
@@ -284,10 +284,12 @@
 # rutilt.desktop
 echo -n "Generating desktop launcher... "
 echo "[Desktop Entry]
-Name=RutilT WLAN Manager
+Name=RutilT
+Name[tr]=RutilT
 Name[fr]=Utilitaire de Configuration Wifi RutilT
 Name[it]=Strumento di Configurazione WiFi RutilT
 GenericName=WLAN Manager
+GenericName[tr]=WLAN Yönetici
 GenericName[fr]=Utilitaire de Configuration Wifi
 GenericName[it]=Strumento di Configurazione WiFi
 Comment=Manage your wireless network interfaces
