--- xbmc/utils/SystemInfo.cpp.orig	2008-12-20 02:55:22.000000000 +0200
+++ xbmc/utils/SystemInfo.cpp	2008-12-20 02:57:03.000000000 +0200
@@ -535,7 +535,7 @@
 {
   CStdString result = "";
   
-  FILE* pipe = popen("unset PYTHONHOME; unset PYTHONPATH; /usr/bin/lsb_release -d | cut -f2", "r");
+  FILE* pipe = popen("unset PYTHONHOME; unset PYTHONPATH; /bin/cat /etc/pardus-release", "r");
   if (pipe)
   {
     char buffer[256];
