--- src/StdString.h.orig	2007-07-26 20:22:46.000000000 +0300
+++ src/StdString.h	2008-04-17 22:50:01.000000000 +0300
@@ -107,6 +107,8 @@
 #include <cstdlib>
 #include <cstdarg>
 
+#include <cstring>
+
 // a very shorthand way of applying the fix for KB problem Q172398
 // (basic_string assignment bug)
 
--- src/EnumHelper.h.orig	2007-02-12 07:36:41.000000000 +0200
+++ src//EnumHelper.h	2008-04-17 22:49:25.000000000 +0300
@@ -4,6 +4,8 @@
 #include "LuaReference.h"
 #include "RageUtil.h"
 
+#include <memory>
+
 extern "C"
 {
 #include "lua-5.1/src/lua.h"
--- src/ScreenOptionsMasterPrefs.cpp.orig	2007-07-26 20:22:43.000000000 +0300
+++ src/ScreenOptionsMasterPrefs.cpp	2008-04-17 22:55:10.000000000 +0300
@@ -127,7 +127,7 @@
 }
 
 template <>
-static void MovePref<bool>( int &iSel, bool bToSel, const ConfOption *pConfOption )
+void MovePref<bool>( int &iSel, bool bToSel, const ConfOption *pConfOption )
 {
 	IPreference *pPref = IPreference::GetPreferenceByName( pConfOption->m_sPrefName );
 	ASSERT_M( pPref != NULL, pConfOption->m_sPrefName );
--- src/RageFileDriverDeflate.cpp.orig	2007-07-26 20:22:39.000000000 +0300
+++ src/RageFileDriverDeflate.cpp	2008-04-17 22:59:42.000000000 +0300
@@ -5,6 +5,8 @@
 #include "RageLog.h"
 #include "RageUtil.h"
 
+#include <memory>
+
 #if defined(_WINDOWS) || defined(_XBOX)
         #include "zlib/zlib.h"
 #if defined(_MSC_VER) && !defined(_XBOX)
