--- server/drivers/laser/lms400_cola.cc	2008-06-11 04:11:17.000000000 +0300
+++ server/drivers/laser/lms400_cola.cc.new	2008-07-07 15:11:34.000000000 +0300
@@ -6,6 +6,9 @@
 */
 #include <sys/socket.h>
 #include <netdb.h>
+#include <cstdlib>
+#include <cstring>
+#include <fstream>
 #include <libplayercore/playercore.h>
 
 #include "lms400_cola.h"
--- server/drivers/mixed/mricp/src/geometry2D.cpp	2008-07-07 15:23:07.000000000 +0300
+++ server/drivers/mixed/mricp/src/geometry2D.cpp.new	2008-07-07 15:23:18.000000000 +0300
@@ -5,6 +5,7 @@
 #include "geometry2D.h"
 #include <cmath>
 #include <cassert>
+#include <cstdio>
 
 using namespace std;
 
--- server/drivers/mixed/mricp/src/map.cpp	2008-04-13 01:15:58.000000000 +0300
+++ server/drivers/mixed/mricp/src/map.cpp.new	2008-07-07 15:32:43.000000000 +0300
@@ -21,6 +21,7 @@
 #include <math.h>
 #include <cassert>
 #include <iostream>
+#include <cstdlib>
 using namespace std;
 
 
