Index: korganizer/incidencechanger.cpp
===================================================================
--- korganizer/incidencechanger.cpp	(Revision 949917)
+++ korganizer/incidencechanger.cpp	(Revision 949918)
@@ -363,7 +363,7 @@
     // widgets sooner or later
     stdcal->setDialogParentWidget( tmpparent );
   }
-  if ( !success ) {
+  if ( !success && stdcal ) {
     // We can have a failure if the user pressed [cancel] in the resource
     // selectdialog, so check the exception.
     ErrorFormat *e = stdcal->exception();
