--- configure.orig	2008-04-04 12:04:26.000000000 +0300
+++ configure	2008-04-04 12:04:34.000000000 +0300
@@ -5608,7 +5608,7 @@
     # FIXME: Let us build with strict aliasing. bug 414641.
     CXXFLAGS="$CXXFLAGS -fno-strict-aliasing"
     # Turn on GNU specific features
-    _WARNINGS_CXXFLAGS="${_WARNINGS_CXXFLAGS} -Wall -Wconversion -Wpointer-arith -Woverloaded-virtual -Wsynth -Wno-ctor-dtor-privacy -Wno-non-virtual-dtor"
+    _WARNINGS_CXXFLAGS="${_WARNINGS_CXXFLAGS} -Wall -Wpointer-arith -Woverloaded-virtual -Wsynth -Wno-ctor-dtor-privacy -Wno-non-virtual-dtor"
     if test "$CPU_ARCH" != "ia64" && test "$CPU_ARCH" != "sparc" \
        && test -z "$INTEL_CC"; then
         # don't use -Wcast-align on ia64 or sparc, it's noisy on those platforms
