diff -ur kftpgrabber-0.6.0.orig/kftpgrabber/src/kftpgrabber.desktop kftpgrabber-0.6.0/kftpgrabber/src/kftpgrabber.desktop
--- kftpgrabber-0.6.0.orig/kftpgrabber/src/kftpgrabber.desktop	2005-05-23 15:43:31.000000000 +0300
+++ kftpgrabber-0.6.0/kftpgrabber/src/kftpgrabber.desktop	2006-12-10 17:46:42.000000000 +0200
@@ -1,7 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=KFTPGrabber
-Name[sv]=KFTPgrabber
 GenericName=FTP Client
 GenericName[ca]=Client FTP
 GenericName[cs]=FTP klient
@@ -12,6 +11,7 @@
 GenericName[it]=Client FTP
 GenericName[nl]=FTP-cliënt
 GenericName[sv]=FTP-klient
+GenericName[tr]=FTP İstemcisi
 GenericName[xh]=Umxhasi we FTP
 GenericName[zu]=Umthengi we FTP
 Comment=A graphical FTP client
