Index: sqliteman-1.2.2/sqliteman.desktop
===================================================================
--- sqliteman-1.2.2.orig/sqliteman.desktop
+++ sqliteman-1.2.2/sqliteman.desktop
@@ -1,14 +1,17 @@
 [Desktop Entry]
 Type=Application
 Name=Sqliteman
-GenericName=Sqlite admin tool
+GenericName=Sqlite Admin Tool
+GenericName[tr]=Sqlite Yönetim Aracı
 Comment=Administer and develop your Sqlite3 databases
+Comment[tr]=Sqlite3 veritabanlarınızı yönetin ve geliştirin
 Icon=sqliteman
 Exec=sqliteman %f
 Terminal=false
 StartupNotify=true
 MimeType=application/x-sqlite3;
 
+Categories=Development;Database;
 # Translations
 GenericName[cs]=Sqlite administrace
 
