--- src/settings.h.orig	2008-01-12 12:37:38.000000000 +0200
+++ src/settings.h	2008-01-12 12:38:40.000000000 +0200
@@ -3,6 +3,7 @@
 
 #include <string>
 #include <wx/colour.h>
+#include <wx/utils.h>
 
 #define DEFSETT_DEFAULT_SERVER "TAS Server"
 #define DEFSETT_DEFAULT_SERVER_HOST "taspringmaster.clan-sy.com"
@@ -12,7 +13,8 @@
 #define DEFSETT_MW_HEIGHT 600
 #define DEFSETT_MW_TOP 50
 #define DEFSETT_MW_LEFT 50
-#define DEFSETT_SPRING_DIR wxGetCwd()
+#define DEFSETT_SPRING_DIR wxGetHomeDir()+wxString::FromAscii("/.spring")
+
 #define DEFSETT_SPRING_PORT 8452
 
 class wxConfigBase;
