Index: src/common.inc
===================================================================
--- src/common.inc	(revision 19791)
+++ src/common.inc	(working copy)
@@ -451,6 +451,10 @@
         "mudur=livecd,language:pt_BR " bc.cmd dup length add sprintf
       } if
 
+      config.lang "sv_SE" eq {
+        "mudur=livecd,language:sv " bc.cmd dup length add sprintf
+      } if
+
       config.lang "tr_TR" eq {
         "mudur=livecd,language:tr " bc.cmd dup length add sprintf
       } if
Index: data-boot/languages
===================================================================
--- data-boot/languages	(revision 19791)
+++ data-boot/languages	(working copy)
@@ -7,4 +7,5 @@
 nl_NL
 pl_PL
 pt_BR
+sv_SE
 tr_TR
Index: data-install/languages
===================================================================
--- data-install/languages	(revision 19791)
+++ data-install/languages	(working copy)
@@ -7,4 +7,5 @@
 nl_NL
 pl_PL
 pt_BR
+sv_SE
 tr_TR
Index: Makefile
===================================================================
--- Makefile	(revision 19791)
+++ Makefile	(working copy)
@@ -14,7 +14,7 @@
 SUBDIRS      = fonts help-boot help-install po src
 
 DEFAULT_LANG = tr_TR
-DEFAULT_LANGUAGES = ca de en es fr it nl pl pt_BR tr
+DEFAULT_LANGUAGES = ca de en es fr it nl pl pt_BR sv tr
 
 .PHONY: all clean distclean themes $(SUBDIRS)
 
