Index: kdeedu/kturtle/src/Makefile.am
===================================================================
--- kdeedu.orig/kturtle/src/Makefile.am
+++ kdeedu/kturtle/src/Makefile.am
@@ -6,7 +6,7 @@ SUBDIRS = pics
 # added from kjots
 AM_LDFLAGS   = $(all_libraries) $(KDE_RPATH)
 
-kturtle_LDFLAGS = $(KDE_RPATH) $(all_libraries)
+kturtle_LDFLAGS = /usr/kde/3.5/lib/libkdeui.so /usr/kde/3.5/lib/libkio.so /usr/kde/3.5/lib/libkdeprint.so /usr/kde/3.5/lib/libkparts.so /usr/kde/3.5/lib/libkatepartinterfaces.so $(KDE_RPATH) $(all_libraries)
 kturtle_LDADD = $(LIB_KDEUI) $(LIB_KIO) $(LIB_KDEPRINT) $(LIB_KPARTS) -lkatepartinterfaces
 
 KDE_ICON = kturtle
