diff -Nur mplayer-20061104-old/libavformat/flvdec.c mplayer-20061104/libavformat/flvdec.c
--- mplayer-20061104-old/libavformat/flvdec.c	2006-11-04 19:15:04.255811290 +0200
+++ mplayer-20061104/libavformat/flvdec.c	2006-11-04 19:16:14.690103650 +0200
@@ -199,12 +199,7 @@
             case 3: st->codec->codec_id = CODEC_ID_FLASHSV; break;
             case 4:
                 st->codec->codec_id = CODEC_ID_VP6F;
-                if (st->codec->extradata_size != 1) {
-                    st->codec->extradata_size = 1;
-                    st->codec->extradata = av_malloc(1);
-                }
-                /* width and height adjustment */
-                st->codec->extradata[0] = get_byte(&s->pb);
+                get_byte( &s->pb ); /* width and height adjustment */
                 size--;
                 break;
             default:
