diff -Nur mplayer-20061209-old/libavformat/flvdec.c mplayer-20061209/libavformat/flvdec.c
--- mplayer-20061209-old/libavformat/flvdec.c	2006-12-09 23:00:08.135257065 +0200
+++ mplayer-20061209/libavformat/flvdec.c	2006-12-09 23:03:23.290588945 +0200
@@ -183,7 +183,7 @@
             case FLV_CODECID_PCM_BE: if (flags & FLV_AUDIO_SAMPLESIZE_MASK) st->codec->codec_id = CODEC_ID_PCM_S16BE;
                     else st->codec->codec_id = CODEC_ID_PCM_S8; break;
             case FLV_CODECID_ADPCM : st->codec->codec_id = CODEC_ID_ADPCM_SWF;                       break;
-            case FLV_CODECID_MP3   : st->codec->codec_id = CODEC_ID_MP3      ; st->need_parsing = 1; break;
+            case FLV_CODECID_MP3   : st->codec->codec_id = CODEC_ID_MP3      ; break;
             // this is not listed at FLV but at SWF, strange...
             case FLV_CODECID_PCM_LE: if (flags & FLV_AUDIO_SAMPLESIZE_MASK) st->codec->codec_id = CODEC_ID_PCM_S16LE;
                     else st->codec->codec_id = CODEC_ID_PCM_S8; break;
