diff -Nur mplayer-20061209-old/configure mplayer-20061209/configure
--- mplayer-20061209-old/configure	2006-12-09 18:34:04.029170081 +0200
+++ mplayer-20061209/configure	2006-12-09 18:38:45.675353353 +0200
@@ -6013,7 +6013,7 @@
       _live=yes && break
   done
   if test "$_live" != yes ; then
-      if cxx_check -I/usr/include/liveMedia -I/usr/include/UsageEnvironment -I/usr/include/groupsock; then
+      if cxx_check -I/usr/include/live -I/usr/include/UsageEnvironment -I/usr/include/groupsock; then
 	  _live_dist=yes
       fi
   fi
@@ -6036,7 +6036,7 @@
   _live="yes"
   _def_live='#define STREAMING_LIVE555 1'
   _ld_extra="$_ld_extra -lliveMedia -lgroupsock -lUsageEnvironment -lBasicUsageEnvironment -lstdc++"
-  _inc_extra="$_inc_extra -I/usr/include/liveMedia -I/usr/include/UsageEnvironment \
+  _inc_extra="$_inc_extra -I/usr/include/live -I/usr/include/UsageEnvironment \
              -I/usr/include/BasicUsageEnvironment -I/usr/include/groupsock"
  _inputmodules="live555 $_inputmodules"
 else
