diff -Nur mplayer-20061209-old/configure mplayer-20061209/configure
--- mplayer-20061209-old/configure	2006-12-09 20:05:08.208488561 +0200
+++ mplayer-20061209/configure	2006-12-09 20:08:06.039454145 +0200
@@ -5522,14 +5522,15 @@
 if test "$_liblzo" = auto ; then
   _liblzo=no
   cat > $TMPC << EOF
-#include <lzo1x.h>
+#include <lzo/lzo1x.h>
 int main(void) { lzo_init();return 0; }
 EOF
-  cc_check -llzo && _liblzo=yes
+  cc_check -llzo2 && _liblzo=yes
 fi
 if test "$_liblzo" = yes ; then
   _def_liblzo='#define USE_LIBLZO 1'
-  _ld_extra="$_ld_extra -llzo"
+  _ld_extra="$_ld_extra -llzo2"
+  _inc_extra="$_inc_extra -I/usr/include/lzo"
   _codecmodules="liblzo $_codecmodules"
 else
   _def_liblzo='#undef USE_LIBLZO'
