diff -Nur mplayer-20060923-old/libmpcodecs/vd_lzo.c mplayer-20060923/libmpcodecs/vd_lzo.c
--- mplayer-20060923-old/libmpcodecs/vd_lzo.c	2006-09-24 04:11:09.701009640 +0300
+++ mplayer-20060923/libmpcodecs/vd_lzo.c	2006-09-24 04:13:26.922148856 +0300
@@ -8,6 +8,8 @@
 
 #ifdef USE_LIBLZO
 #include <lzo1x.h>
+#define lzo_malloc malloc
+#define lzo_free free
 #else
 #include "native/minilzo.h"
 #define lzo_malloc malloc
