Index: src/via_mode.c
===================================================================
--- src/via_mode.c	(revision 750)
+++ src/via_mode.c	(working copy)
@@ -449,9 +449,12 @@
         if (pBIOSInfo->CrtPresent)
             pBIOSInfo->CrtActive = TRUE;
 
+#if 0
+        # FIXME : DFP must be activated with the ActiveDevice option 
         /* DFP */
         if (pBIOSInfo->DfpPresent)
             pBIOSInfo->DfpActive = TRUE;
+#endif
         
     } else {
         if (pVia->ActiveDevice & VIA_DEVICE_LCD) {
