--- src/arch/unix/ffmpeglib.c~	2009-06-15 13:45:53.000000000 +0300
+++ src/arch/unix/ffmpeglib.c	2009-06-15 13:45:53.000000000 +0300
@@ -48,7 +48,7 @@
     GET_PROC_ADDRESS_AND_TEST_AVCODEC(avcodec_encode_video);
     GET_PROC_ADDRESS_AND_TEST_AVCODEC(avpicture_fill);
     GET_PROC_ADDRESS_AND_TEST_AVCODEC(avpicture_get_size);
-    GET_PROC_ADDRESS_AND_TEST_AVCODEC(img_convert);
+//     GET_PROC_ADDRESS_AND_TEST_AVCODEC(img_convert);
     GET_PROC_ADDRESS_AND_TEST_AVCODEC(av_free);
 
     GET_PROC_ADDRESS_AND_TEST_AVFORMAT(av_register_all);
