diff -Naur vice-2.2.orig/data/fonts/Makefile.am vice-2.2/data/fonts/Makefile.am
--- vice-2.2.orig/data/fonts/Makefile.am	2010-03-12 11:53:03.862970788 +0200
+++ vice-2.2/data/fonts/Makefile.am	2010-03-12 11:53:43.528220625 +0200
@@ -7,18 +7,6 @@
 EXTRA_DIST = $(fonts_DATA) cbm-directory-charset.fon 
 
 install: install-data-am
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	@echo "Preparing fontdir, please wait..." ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	if [ ! `fc-cache --list > /dev/null 2>&1` ] ; then \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	  if test "x${HOME}" != "x" ; then \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	    mkdir ${HOME}/.fonts > /dev/null 2>&1 ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	    cp $(fontsdir)/CBM.ttf ${HOME}/.fonts ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	    fc-cache ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	  else \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	    echo '$$HOME not set... giving up.' ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	  fi \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	else \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	  echo 'fc-cache not available... giving up.' ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	fi
 @HAVE_FC_CACHE_FALSE@@SDL_COMPILE_FALSE@	@echo "Preparing fontdir..." ; \
 @HAVE_FC_CACHE_FALSE@@SDL_COMPILE_FALSE@	cd $(DESTDIR)$(fontsdir) ; \
 @HAVE_FC_CACHE_FALSE@@SDL_COMPILE_FALSE@	bdftopcf -o vice-cbm.pcf vice-cbm.bdf ; \
diff -Naur vice-2.2.orig/data/fonts/Makefile.in vice-2.2/data/fonts/Makefile.in
--- vice-2.2.orig/data/fonts/Makefile.in	2010-03-12 11:53:03.862970788 +0200
+++ vice-2.2/data/fonts/Makefile.in	2010-03-12 11:54:27.420970890 +0200
@@ -538,18 +538,6 @@
 
 
 install: install-data-am
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	@echo "Preparing fontdir, please wait..." ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	if [ ! `fc-cache --list > /dev/null 2>&1` ] ; then \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	  if test "x${HOME}" != "x" ; then \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	    mkdir ${HOME}/.fonts > /dev/null 2>&1 ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	    cp $(fontsdir)/CBM.ttf ${HOME}/.fonts ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	    fc-cache ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	  else \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	    echo '$$HOME not set... giving up.' ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	  fi \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	else \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	  echo 'fc-cache not available... giving up.' ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	fi
 @HAVE_FC_CACHE_FALSE@@SDL_COMPILE_FALSE@	@echo "Preparing fontdir..." ; \
 @HAVE_FC_CACHE_FALSE@@SDL_COMPILE_FALSE@	cd $(DESTDIR)$(fontsdir) ; \
 @HAVE_FC_CACHE_FALSE@@SDL_COMPILE_FALSE@	bdftopcf -o vice-cbm.pcf vice-cbm.bdf ; \
