Index: ecryptfs-utils-87/src/utils/ecryptfs-setup-private
===================================================================
--- ecryptfs-utils-87.orig/src/utils/ecryptfs-setup-private
+++ ecryptfs-utils-87/src/utils/ecryptfs-setup-private
@@ -340,8 +340,9 @@ echo
 # Setup private directory in home
 mkdir -m 700 -p "$CRYPTDIR" || error "$(gettext 'Could not create crypt directory')" "[$CRYPTDIR]"
 mkdir -m 700 -p "$MOUNTPOINT" || error "$(gettext 'Could not create mount directory')" "[$MOUNTPOINT]"
-ln -sf /usr/share/ecryptfs-utils/ecryptfs-mount-private.txt "$MOUNTPOINT"/README.txt
 ln -sf /usr/share/ecryptfs-utils/ecryptfs-mount-private.desktop "$MOUNTPOINT"/Access-Your-Private-Data.desktop
+# Set a nice icon for the unmounted folder
+ln -sf /usr/share/ecryptfs-utils/directory-info "$MOUNTPOINT"/.directory
 chmod 500 "$MOUNTPOINT"
 
 # Setup ~/.ecryptfs directory
